cwlVersion: sbg:draft-2
class: Workflow
label: PDX WES Tumor-Only (Xenome) with Variant Calling, MSI, and TMB scores
$namespaces:
  sbg: https://sevenbridges.com

requirements:
- class: SubworkflowFeatureRequirement
- class: InlineJavascriptRequirement

inputs:
- id: '#bed'
  label: Target BED file
  type:
  - File
  description: Target BED file
  sbg:fileTypes: TXT, BED
  sbg:includeInPorts: true
  sbg:x: -194.33344523111992
  sbg:y: 630.0982411702478
- id: '#reference'
  label: Reference FASTA file
  type:
  - File
  secondaryFiles:
  - .fai
  - ^.dict
  description: Reference FASTA file
  sbg:fileTypes: FASTA, FA
  sbg:includeInPorts: true
  sbg:suggestedValue:
    name: Homo sapiens 38 reference
    class: File
    secondaryFiles:
    - name: SequenceDictionary for Homo sapiens 38 reference
      class: File
      path: 5dcd7d94e4b0eeec709a20ff
    - name: FAI Index for Homo sapiens 38 reference
      class: File
      path: 5dcd7d94e4b0eeec709a20f2
    path: 5dcd7d94e4b0eeec709a20ef
  sbg:x: -203.33347320556655
  sbg:y: 257.76479085286473
- id: '#database'
  label: SnpEff database
  type:
  - File
  description: SnpEff database
  sbg:fileTypes: ZIP
  sbg:includeInPorts: true
  sbg:suggestedValue:
    name: SnpEFF database
    class: File
    path: 5dcd7d94e4b0eeec709a20fa
  sbg:x: 1321.0152587890625
  sbg:y: 363.6502685546875
- id: '#database_1'
  label: dbNSFP file
  type:
  - File
  secondaryFiles:
  - .tbi
  description: dbNSFP file
  sbg:fileTypes: GZ, TXT.GZ, VCF.GZ
  sbg:includeInPorts: true
  sbg:suggestedValue:
    name: dbNSFP3.5a variant chr all sorted
    class: File
    secondaryFiles:
    - name: Index for dbNSFP3.5a variant chr all sorted
      class: File
      path: 5dcd7d94e4b0eeec709a20f9
    path: 5dcd7d94e4b0eeec709a20fb
  sbg:x: 1434.251220703125
  sbg:y: 457.5042419433594
- id: '#database_for_annotation'
  label: COSMIC file
  type:
  - File
  secondaryFiles:
  - .tbi
  sbg:fileTypes: VCF.GZ
  sbg:suggestedValue:
    name: COSMICv80 coding non-coding processed
    class: File
    secondaryFiles:
    - name: Index for COSMICv80 coding non-coding processed
      class: File
      path: 5dcd7d94e4b0eeec709a20f6
    path: 5dcd7d94e4b0eeec709a2100
  sbg:x: 1608.66552734375
  sbg:y: 492.5857238769531
- id: '#intervals_string'
  label: GATK BaseRecalibrator Intervals
  type:
  - 'null'
  - string
  description: Input intervals (-L) for GATK BaseRecalibrator
  sbg:includeInPorts: true
  sbg:x: -198.60850524902344
  sbg:y: 513.7342529296875
- id: '#index_file_1'
  label: Xenome index files
  type:
  - File
  description: TGZ archive with Xenome index files
  sbg:fileTypes: TGZ, TAR.GZ
  sbg:includeInPorts: true
  sbg:suggestedValue:
    name: Xenome Index
    class: File
    path: 5dcd7d94e4b0eeec709a2101
  sbg:x: -214.0000406901043
  sbg:y: -22.66665140787761
- id: '#index_file'
  label: BWA Reference and Indices
  type:
  - File
  description: BWA reference archive
  sbg:fileTypes: TGZ, TAR.GZ
  sbg:includeInPorts: true
  sbg:suggestedValue:
    name: BWA reference and indices
    class: File
    path: 5dcd7d94e4b0eeec709a20ed
  sbg:x: -211.12738037109375
  sbg:y: 121.0541763305664
- id: '#input_pair'
  label: Input FASTQ Files
  type:
  - type: array
    items: File
  description: Input reads (tumor and normal FASTQ(.GZ) files)
  sbg:fileTypes: FASTQ, FASTQ.GZ, FQ, FQ.GZ
  sbg:includeInPorts: true
  sbg:x: -207.0000457763673
  sbg:y: 384.3333435058596
- id: '#target_intervals'
  label: Target Intervals File
  type:
  - File
  description: Target intervals
  sbg:fileTypes: INTERVAL_LIST, INTERVALS
  sbg:includeInPorts: true
  sbg:x: -205.00012715657564
  sbg:y: 748.0980936686202
- id: '#known_sites'
  label: Known Site Files
  type:
  - type: array
    items: File
  secondaryFiles:
  - .tbi
  description: Known sites for recalabration
  sbg:fileTypes: TXT, BED, VCF, VCF.GZ
  sbg:includeInPorts: true
  sbg:suggestedValue:
  - name: Known sites - Homo sapiens assembly38 known indels
    class: File
    secondaryFiles:
    - name: Index for Known sites - Homo sapiens assembly38 known indels
      class: File
      path: 5dcd7d94e4b0eeec709a20f3
    path: 5dcd7d94e4b0eeec709a20f4
  - name: Known sites - Mills and 1000G gold standard indels hg38
    class: File
    secondaryFiles:
    - name: Index for Known sites - Mills and 1000G gold standard indels hg38
      class: File
      path: 5dcd7d94e4b0eeec709a20ee
    path: 5dcd7d94e4b0eeec709a20f8
  - name: Known sites - hsa dbSNP v151 20170710
    class: File
    secondaryFiles:
    - name: Index for Known sites - hsa dbSNP v151 20170710
      class: File
      path: 5dcd7d94e4b0eeec709a20fc
    path: 5dcd7d94e4b0eeec709a20f5
  sbg:x: -197.18154907226562
  sbg:y: 866.5636596679688
- id: '#germline_resource'
  label: Germline resource
  type:
  - 'null'
  - File
  secondaryFiles:
  - .tbi
  description: Germline resource
  sbg:fileTypes: VCF.GZ
  sbg:includeInPorts: true
  sbg:suggestedValue:
    name: ExAC.0.3.GRCh38 - chr added bad lift over removed reorder primary only
    class: File
    secondaryFiles:
    - name: Index for ExAC.0.3.GRCh38 - chr added bad lift over removed reorder primary
        only
      class: File
      path: 5dcd7d94e4b0eeec709a20fe
    path: 5dcd7d94e4b0eeec709a20f1
  sbg:x: 703.3334350585942
  sbg:y: 398.3213297526044
- id: '#split_mode'
  label: Split mode
  type:
  - name: split_mode
    type: enum
    symbols:
    - File per interval
    - File per chr with alt contig in a single file
    - Output original BED
    - File per interval with alt contig in a single file
  description: |-
    Depending on selected Split Mode value, output files are generated in accordance with description below:  1. File per interval - The tool creates one interval file per line of the input BED(FAI) file. Each interval file contains a single line (one of the lines of BED(FAI) input file).  2. File per chr with alt contig in a single file - For each contig(chromosome) a single file is created containing all the intervals corresponding to it . All the intervals (lines) other than (chr1, chr2 ... chrY or 1, 2 ... Y) are saved as ("others.bed").  3. Output original BED - BED file is required for execution of this mode. If mode 3 is applied input is passed to the output.  4. File per interval with alt contig in a single file - For each chromosome a single file is created for each interval. All the intervals (lines) other than (chr1, chr2 ... chrY or 1, 2 ... Y) are saved as ("others.bed"). NOTE: Do not use option 1 (File per interval) with exome BED or a BED with a lot of GL contigs, as it will create a large number of files.
  sbg:category: Input
  sbg:suggestedValue: File per chr with alt contig in a single file
- id: '#TargetCoverage_CutOff'
  type:
  - 'null'
  - float
- id: '#sample_ploidy'
  label: Sample Ploidy
  type:
  - 'null'
  - int
  description: |-
    Ploidy (number of chromosomes) per sample. For pooled data, set to (Number of samples in each pool * Sample Ploidy).
- id: '#dbsnp'
  label: Dbsnp
  type:
  - 'null'
  - File
  secondaryFiles:
  - .tbi
  description: dbSNP file.
  sbg:fileTypes: VCF.GZ
  sbg:includeInPorts: true
  sbg:suggestedValue:
    name: hsa_dbSNP_v151
    class: File
    secondaryFiles:
    - name: Index for hsa_dbSNP_v151
      class: File
      path: 5dcd7d94e4b0eeec709a2102
    path: 5dcd7d94e4b0eeec709a20f7
  sbg:x: 1241.9063720703125
  sbg:y: 304.430419921875

outputs:
- id: '#renamedFiles_1'
  label: Tumor Sample FASTQC reports
  type:
  - 'null'
  - type: array
    items: File
  source:
  - '#FastQC.report_html'
  description: Tumor FASTQC reports
  required: false
  sbg:fileTypes: HTML
  sbg:includeInPorts: true
  sbg:x: 349.6666463216148
  sbg:y: 323.3333333333335
- id: '#output_bam_file_1'
  label: Tumor Sample BAM file
  type:
  - File
  source:
  - '#SAMtools_Index_BAM_1.output_bam_file'
  description: Tumor BAM file
  required: true
  sbg:fileTypes: BAM
  sbg:includeInPorts: true
  sbg:x: 883.333841959636
  sbg:y: -91.67888641357428
- id: '#annotated_vcf'
  label: Final annotated VCF
  type:
  - 'null'
  - File
  source:
  - '#vcftools_sort_0_1_14.output_file'
  description: Final annotated VCF
  required: false
  sbg:fileTypes: VCF
  sbg:includeInPorts: true
  sbg:x: 2014.7877197265625
  sbg:y: 364.7979431152344
- id: '#extracted_fields_file'
  label: Final annotated TSV file
  type:
  - File
  source:
  - '#SnpSift_extractFields_4_3k.extracted_fields_file'
  description: Final annotated TSV file
  required: true
  sbg:fileTypes: TAB
  sbg:includeInPorts: true
  sbg:x: 2264.5517578125
  sbg:y: 88.13920593261719
- id: '#TMB_Score'
  label: TMB Score File
  type:
  - 'null'
  - File
  source:
  - '#tmb_tumoronly.TMB_Score'
  sbg:x: 2040.6453857421875
  sbg:y: 36.68812561035156
- id: '#summary_1'
  label: Tumor Sample Coverage Report
  type:
  - 'null'
  - File
  source:
  - '#tumor_alignment_and_target_coverage_workflow.summary'
  sbg:x: 459.3224792480469
  sbg:y: 10.140352249145508
- id: '#integrated_QC_report_1'
  label: Tumor Sample Integrated QC Report
  type:
  - 'null'
  - File
  source:
  - '#tumor_alignment_and_target_coverage_workflow.integrated_QC_report'
  sbg:x: 453.6800537109375
  sbg:y: 150.48165893554688
- id: '#multiqc_report'
  label: Tumor Sample MultiQC report
  type:
  - 'null'
  - File
  source:
  - '#multiqc_2.multiqc_report'
  description: File containg MultiQC report analysis.
  sbg:fileTypes: HTML
  sbg:x: 640.0762329101562
  sbg:y: -187.80381774902344
- id: '#msi_score'
  label: MSI score
  type:
  - 'null'
  - File
  source:
  - '#msisensor_msi_0_2.msi_score'
  description: MSI score.
  sbg:fileTypes: TXT
  sbg:x: 1113.207275390625
  sbg:y: -130.77987670898438

steps:
- id: '#SBG_Prepare_Intervals'
  label: SBG Prepare Intervals
  in: []
  run:
    cwlVersion: sbg:draft-2
    class: CommandLineTool
    label: SBG Prepare Intervals
    $namespaces:
      sbg: https://sevenbridges.com

    requirements:
    - id: '#cwl-js-engine'
      class: ExpressionEngineRequirement
      requirements:
      - class: DockerRequirement
        dockerPull: rabix/js-engine
    - class: InlineJavascriptRequirement
    - class: CreateFileRequirement
      fileDef:
      - fileContent: |-
          """
          Usage:
              sbg_prepare_intervals.py [options] [--fastq FILE --bed FILE --mode INT --format STR --others STR]

          Description:
              Purpose of this tool is to split BED file into files based on the selected mode.
              If bed file is not provided fai(fasta index) file is converted to bed.

          Options:

              -h, --help            Show this message.

              -v, -V, --version     Tool version.

              -b, -B, --bed FILE    Path to input bed file.

              --fai FILE            Path to input fai file.

              --format STR          Output file format.

              --mode INT            Select input mode.

          """


          import os
          import sys
          import glob
          import shutil
          from docopt import docopt

          default_extension = '.bed'  # for output files


          def create_file(contents, contig_name, extension=default_extension):
              """function for creating a file for all intervals in a contig"""

              new_file = open("Intervals/" + contig_name + extension, "w")
              new_file.write(contents)
              new_file.close()


          def add_to_file(line, name, extension=default_extension):
              """function for adding a line to a file"""

              new_file = open("Intervals/" + name + extension, "a")
              if lformat == formats[1]:
                  sep = line.split("\t")
                  line = sep[0] + ":" + sep[1] + "-" + sep[2]
              new_file.write(line)
              new_file.close()


          def fai2bed(fai):
              """function to create a bed file from fai file"""

              region_thr = 10000000  # threshold used to determine starting point accounting for telomeres in chromosomes
              basename = fai[0:fai.rfind(".")]
              with open(fai, "r") as ins:
                  new_array = []
                  for line in ins:
                      len_reg = int(line.split()[1])
                      cutoff = 0 if (
                      len_reg < region_thr) else 0  # sd\\telomeres or start with 1
                      new_line = line.split()[0] + '\t' + str(cutoff) + '\t' + str(
                          len_reg + cutoff)
                      new_array.append(new_line)
              new_file = open(basename + ".bed", "w")
              new_file.write("\n".join(new_array))
              return basename + ".bed"


          def chr_intervals(no_of_chrms=23):
              """returns all possible designations for chromosome intervals"""

              chrms = []
              for i in range(1, no_of_chrms):
                  chrms.append("chr" + str(i))
                  chrms.append(str(i))
              chrms.extend(["x", "y", "chrx", "chry"])
              return chrms


          def mode_1(orig_file):
              """mode 1: every line is a new file"""

              with open(orig_file, "r") as ins:
                  prev = ""
                  counter = 0
                  names = []
                  for line in ins:
                      if is_header(line):
                          continue
                      if line.split()[0] == prev:
                          counter += 1
                      else:
                          counter = 0
                      suffix = "" if (counter == 0) else "_" + str(counter)
                      create_file(line, line.split()[0] + suffix)
                      names.append(line.split()[0] + suffix)
                      prev = line.split()[0]

                  create_file(str(names), "names", extension=".txt")


          def mode_2(orig_file, others_name):
              """mode 2: separate file is created for each chromosome, and one file is created for other intervals"""

              chrms = chr_intervals()
              names = []

              with open(orig_file, 'r') as ins:
                  for line in ins:
                      if is_header(line):
                          continue
                      name = line.split()[0]
                      if name.lower() in chrms:
                          name = name
                      else:
                          name = others_name
                      try:
                          add_to_file(line, name)
                          if not name in names:
                              names.append(name)
                      except:
                          raise Exception(
                              "Couldn't create or write in the file in mode 2")

                  create_file(str(names), "names", extension=".txt")


          def mode_3(orig_file, extension=default_extension):
              """mode 3: input file is staged to output"""

              orig_name = orig_file.split("/")[len(orig_file.split("/")) - 1]
              output_file = r"./Intervals/" + orig_name[
                                              0:orig_name.rfind('.')] + extension

              shutil.copyfile(orig_file, output_file)

              names = [orig_name[0:orig_name.rfind('.')]]
              create_file(str(names), "names", extension=".txt")


          def mode_4(orig_file, others_name):
              """mode 4: every interval in chromosomes is in a separate file. Other intervals are in a single file"""

              chrms = chr_intervals()
              names = []

              with open(orig_file, "r") as ins:
                  counter = {}
                  for line in ins:
                      if line.startswith('@'):
                          continue
                      name = line.split()[0].lower()
                      if name in chrms:
                          if name in counter:
                              counter[name] += 1
                          else:
                              counter[name] = 0
                          suffix = "" if (counter[name] == 0) else "_" + str(counter[name])
                          create_file(line, name + suffix)
                          names.append(name + suffix)
                          prev = name
                      else:
                          name = others_name
                          if not name in names:
                              names.append(name)
                          try:
                              add_to_file(line, name)
                          except:
                              raise Exception(
                                  "Couldn't create or write in the file in mode 4")

              create_file(str(names), "names", extension=".txt")


          def prepare_intervals():
              # reading input files and split mode from command line
              args = docopt(__doc__, version='1.0')

              bed_file = args['--bed']
              fai_file = args['--fai']
              split_mode = int(args['--mode'])

              # define file name for non-chromosomal contigs
              others_name = 'others'

              global formats, lformat
              formats = ["chr start end", "chr:start-end"]
              lformat = args['--format']
              if lformat == None:
                  lformat = formats[0]
              if not lformat in formats:
                  raise Exception('Unsuported interval format')

              if not os.path.exists(r"./Intervals"):
                  os.mkdir(r"./Intervals")
              else:
                  files = glob.glob(r"./Intervals/*")
                  for f in files:
                      os.remove(f)

              # create variable input_file taking bed_file as priority
              if bed_file:
                  input_file = bed_file
              elif fai_file:
                  input_file = fai2bed(fai_file)
              else:
                  raise Exception('No input files are provided')

              # calling adequate split mode function
              if split_mode == 1:
                  mode_1(input_file)
              elif split_mode == 2:
                  mode_2(input_file, others_name)
              elif split_mode == 3:
                  if bed_file:
                      mode_3(input_file)
                  else:
                      raise Exception('Bed file is required for mode 3')
              elif split_mode == 4:
                  mode_4(input_file, others_name)
              else:
                  raise Exception('Split mode value is not set')


          def is_header(line):
              x = line.split('\t')
              try:
                  int(x[1])
                  int(x[2])
                  header = False
              except:
                  sys.stderr.write('Line is skipped: {}'.format(line))
                  header = True
              return header


          if __name__ == '__main__':
              prepare_intervals()
        filename: sbg_prepare_intervals.py

    inputs:
    - id: '#bed_file'
      label: Input BED file
      type:
      - 'null'
      - File
      inputBinding:
        prefix: --bed
        position: 1
        separate: true
        sbg:cmdInclude: true
      description: Input BED file containing intervals. Required for modes 3 and 4.
      sbg:category: File Input
      sbg:fileTypes: BED
      sbg:stageInput: link
    - id: '#fai_file'
      label: Input FAI file
      type:
      - 'null'
      - File
      inputBinding:
        prefix: --fai
        position: 2
        separate: true
        sbg:cmdInclude: true
      description: FAI file is converted to BED format if BED file is not provided.
      sbg:category: File Input
      sbg:fileTypes: FAI
      sbg:stageInput: link
    - id: '#split_mode'
      label: Split mode
      type:
      - name: split_mode
        type: enum
        symbols:
        - File per interval
        - File per chr with alt contig in a single file
        - Output original BED
        - File per interval with alt contig in a single file
      inputBinding:
        prefix: --mode
        position: 3
        valueFrom:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              mode = $job.inputs.split_mode
              switch (mode) 
              {
                case "File per interval": 
                  return 1
                case "File per chr with alt contig in a single file": 
                  return 2
                case "Output original BED": 
                  return 3
                case "File per interval with alt contig in a single file": 
                  return 4  
              }
              return 3
            }
        separate: true
        sbg:cmdInclude: true
      description: |-
        Depending on selected Split Mode value, output files are generated in accordance with description below:  1. File per interval - The tool creates one interval file per line of the input BED(FAI) file. Each interval file contains a single line (one of the lines of BED(FAI) input file).  2. File per chr with alt contig in a single file - For each contig(chromosome) a single file is created containing all the intervals corresponding to it . All the intervals (lines) other than (chr1, chr2 ... chrY or 1, 2 ... Y) are saved as ("others.bed").  3. Output original BED - BED file is required for execution of this mode. If mode 3 is applied input is passed to the output.  4. File per interval with alt contig in a single file - For each chromosome a single file is created for each interval. All the intervals (lines) other than (chr1, chr2 ... chrY or 1, 2 ... Y) are saved as ("others.bed"). NOTE: Do not use option 1 (File per interval) with exome BED or a BED with a lot of GL contigs, as it will create a large number of files.
      sbg:category: Input
    - id: '#format'
      label: Interval format
      type:
      - 'null'
      - name: format
        type: enum
        symbols:
        - chr start end
        - chr:start-end
      description: Format of the intervals in the generated files.
      sbg:category: Input

    outputs:
    - id: '#intervals'
      label: Intervals
      type:
      - 'null'
      - type: array
        items: File
      outputBinding:
        glob: Intervals/*.bed
        sbg:metadata:
          sbg_scatter: 'true'
      description: Array of BED files genereted as per selected Split Mode.
      sbg:fileTypes: BED
    - id: '#names'
      label: Output file names
      type:
      - 'null'
      - string
      outputBinding:
        glob: Intervals/names.txt
        outputEval:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {   
             content =  $self[0].contents.replace(/\0/g, '')
             content = content.replace('[','')
             content = content.replace(']','')
             content = content.replace(/\'/g, "")
             content = content.replace(/\s/g, '')
             content_arr = content.split(",")

             return content_arr
             

            } 
        loadContents: true
      description: File containing the names of created files.
    - id: '#str_arr'
      label: String output
      type:
      - 'null'
      - type: array
        items: string
      outputBinding:
        glob:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            { 
              if ($job.inputs.bed_file){
                glob = $job.inputs.bed_file.path
            	glob = glob.split('/').slice(-1)[0]
              }
              else if($job.inputs.fai_file){
                glob = $job.inputs.fai_file.path
            	glob = glob.split('/').slice(-1)[0].split('.').slice(0, -1).join('.') + '.bed'
              }

              return glob
            }
        outputEval:
          class: Expression
          engine: '#cwl-js-engine'
          script: |
            { rows = $self[0].contents
              if (rows[rows.length -1] == '\n'){
                rows = rows.split(/\r?\n/).slice(0,-1);
              }
              else {
                rows = rows.split(/\r?\n/);
              }
              out_list = []
              for (i=0; i<rows.length; i++)
              {
                row = rows[i];
                chromosome = row.split("\t")[0];
                start = row.split("\t")[1];
                end = row.split("\t")[2];
                if(typeof start !== "undefined"){
                	interval = chromosome.concat(":", start, "-", end);
                }
                else{
                	interval = chromosome
                }    
                out_list.push(interval);
              }
              return out_list;
              	
            }
        loadContents: true
      description: Outputs BED content as strings.

    baseCommand:
    - python
    - sbg_prepare_intervals.py
    arguments:
    - position: 0
      valueFrom:
        class: Expression
        engine: '#cwl-js-engine'
        script: |-
          {	
            if ($job.inputs.format)
            	return "--format " + "\"" + $job.inputs.format + "\""
          }
      separate: true

    hints:
    - class: sbg:CPURequirement
      value: 1
    - class: sbg:MemRequirement
      value: 1000
    - class: DockerRequirement
      dockerPull: images.sbgenomics.com/bogdang/sbg_prepare_intervals:1.0
    id: admin/sbg-public-data/sbg-prepare-intervals/93
    description: |-
      Depending on selected Split Mode value, output files are generated in accordance with description below:

      1. File per interval - The tool creates one interval file per line of the input BED(FAI) file.
      Each interval file contains a single line (one of the lines of BED(FAI) input file).

      2. File per chr with alt contig in a single file - For each contig(chromosome) a single file
      is created containing all the intervals corresponding to it .
      All the intervals (lines) other than (chr1, chr2 ... chrY or 1, 2 ... Y) are saved as
      ("others.bed").

      3. Output original BED - BED file is required for execution of this mode. If mode 3 is applied input is passed to the output.

      4. File per interval with alt contig in a single file - For each chromosome a single file is created for each interval.
      All the intervals (lines) other than (chr1, chr2 ... chrY or 1, 2 ... Y) are saved as
      ("others.bed").

      ##### Common issues: 
      Do not use option 1 (File per interval) with exome BED or a BED with a lot of GL contigs, as it will create a large number of files.
    sbg:appVersion:
    - sbg:draft-2
    sbg:categories:
    - Converters
    sbg:cmdPreview: python sbg_prepare_intervals.py  --format "chr start end" --mode
      2
    sbg:content_hash: a9b2d21aa705968a9f82b0d6446b9d92b13904aaf0740cea8c0ef805d7eb2350d
    sbg:contributors:
    - admin
    sbg:createdBy: admin
    sbg:createdOn: 1473165183
    sbg:id: admin/sbg-public-data/sbg-prepare-intervals/93
    sbg:image_url:
    sbg:job:
      allocatedResources:
        cpu: 1
        mem: 1000
      inputs:
        format: chr start end
        fai_file:
          class: File
          secondaryFiles: []
          path: /path/to/fai_file.fasta.fai
          size: 0
        split_mode: File per chr with alt contig in a single file
    sbg:latestRevision: 93
    sbg:license: Apache License 2.0
    sbg:modifiedBy: admin
    sbg:modifiedOn: 1545924498
    sbg:project: admin/sbg-public-data
    sbg:projectName: SBG Public data
    sbg:publisher: sbg
    sbg:revision: 93
    sbg:revisionNotes: check for header
    sbg:revisionsInfo:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165183
      sbg:revision: 0
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165183
      sbg:revision: 1
      sbg:revisionNotes: docker file
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165183
      sbg:revision: 2
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165183
      sbg:revision: 3
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165183
      sbg:revision: 4
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165183
      sbg:revision: 5
      sbg:revisionNotes: split mode required box checked
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165183
      sbg:revision: 6
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165183
      sbg:revision: 7
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165183
      sbg:revision: 8
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165183
      sbg:revision: 9
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165183
      sbg:revision: 10
      sbg:revisionNotes: fai2bed sitno izmenjen
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165183
      sbg:revision: 11
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165183
      sbg:revision: 12
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165183
      sbg:revision: 13
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165183
      sbg:revision: 14
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165184
      sbg:revision: 15
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165184
      sbg:revision: 16
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165184
      sbg:revision: 17
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165184
      sbg:revision: 18
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165184
      sbg:revision: 19
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165184
      sbg:revision: 20
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165184
      sbg:revision: 21
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165184
      sbg:revision: 22
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165184
      sbg:revision: 23
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165184
      sbg:revision: 24
      sbg:revisionNotes: change the modes labeling
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165184
      sbg:revision: 25
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165184
      sbg:revision: 26
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165184
      sbg:revision: 27
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165184
      sbg:revision: 28
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165184
      sbg:revision: 29
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165184
      sbg:revision: 30
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165184
      sbg:revision: 31
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165184
      sbg:revision: 32
      sbg:revisionNotes: name.txt added
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165184
      sbg:revision: 33
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165184
      sbg:revision: 34
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165184
      sbg:revision: 35
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165184
      sbg:revision: 36
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165184
      sbg:revision: 37
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165184
      sbg:revision: 38
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165184
      sbg:revision: 39
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165184
      sbg:revision: 40
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165184
      sbg:revision: 41
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165184
      sbg:revision: 42
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165184
      sbg:revision: 43
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165184
      sbg:revision: 44
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165184
      sbg:revision: 45
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165184
      sbg:revision: 46
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165185
      sbg:revision: 47
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165185
      sbg:revision: 48
      sbg:revisionNotes: output names = array of strings
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165185
      sbg:revision: 49
      sbg:revisionNotes: Added format parameter.
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165185
      sbg:revision: 50
      sbg:revisionNotes: global
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165185
      sbg:revision: 51
      sbg:revisionNotes: tab
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165185
      sbg:revision: 52
      sbg:revisionNotes: tab again
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165185
      sbg:revision: 53
      sbg:revisionNotes: f
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165185
      sbg:revision: 54
      sbg:revisionNotes: '""'
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165185
      sbg:revision: 55
      sbg:revisionNotes: '""'
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165185
      sbg:revision: 56
      sbg:revisionNotes: '- ""'
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165185
      sbg:revision: 57
      sbg:revisionNotes: format required
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165186
      sbg:revision: 58
      sbg:revisionNotes: Chnaged 1000 to 0 in fai2bed
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165186
      sbg:revision: 59
      sbg:revisionNotes: like last rev
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165186
      sbg:revision: 60
      sbg:revisionNotes: Format not required anymore
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165186
      sbg:revision: 61
      sbg:revisionNotes: chr_intervals(), counter = {}, format in description
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165186
      sbg:revision: 62
      sbg:revisionNotes: Rev 60
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165186
      sbg:revision: 63
      sbg:revisionNotes: Output names is now array of strings instead of a single
        string.
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165186
      sbg:revision: 64
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165186
      sbg:revision: 65
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165186
      sbg:revision: 66
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165186
      sbg:revision: 67
      sbg:revisionNotes: Names - string array
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165186
      sbg:revision: 68
      sbg:revisionNotes: counter {}, chrms()
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165186
      sbg:revision: 69
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165186
      sbg:revision: 70
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165186
      sbg:revision: 71
      sbg:revisionNotes: Added option to define file name for other contigs
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165186
      sbg:revision: 72
      sbg:revisionNotes: sbg_scatter added to metadata of intervals.
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165187
      sbg:revision: 73
      sbg:revisionNotes: Mode 4 fix others_name
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1473165187
      sbg:revision: 74
      sbg:revisionNotes: fix (
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1476440179
      sbg:revision: 75
      sbg:revisionNotes: Mode 3 is default if split mode not selected. Split mode
        not required.
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1476440179
      sbg:revision: 76
      sbg:revisionNotes: mode 3 if split mode not set
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1476440179
      sbg:revision: 77
      sbg:revisionNotes: Docopt instead of argparse
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1476440179
      sbg:revision: 78
      sbg:revisionNotes: rev76
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1476440179
      sbg:revision: 79
      sbg:revisionNotes: Docker image with docopt
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1476440179
      sbg:revision: 80
      sbg:revisionNotes: others name removed from inputs
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1476976956
      sbg:revision: 81
      sbg:revisionNotes: split_mode set to required
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1480071816
      sbg:revision: 82
      sbg:revisionNotes: Fixed Toolkit name.
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1491908175
      sbg:revision: 83
      sbg:revisionNotes: Description changed
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1491908175
      sbg:revision: 84
      sbg:revisionNotes: Common issues added
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1501857597
      sbg:revision: 85
      sbg:revisionNotes: Change the Javascript for adding the format argument into
        the command line
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1503405681
      sbg:revision: 86
      sbg:revisionNotes: Removed lower() from mode_2
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1503405681
      sbg:revision: 87
      sbg:revisionNotes: String output added
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1504517528
      sbg:revision: 88
      sbg:revisionNotes: Glob fix for fai input
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1508493476
      sbg:revision: 89
      sbg:revisionNotes: Removed sbg:format for string output from json
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1512066802
      sbg:revision: 90
      sbg:revisionNotes: Bug fixed - if bed file is not ended with \n tool will work
        as expected.
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1545924498
      sbg:revision: 91
      sbg:revisionNotes: Ignore header lines
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1545924498
      sbg:revision: 92
      sbg:revisionNotes: Back to revision 12
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1545924498
      sbg:revision: 93
      sbg:revisionNotes: check for header
    sbg:sbgMaintained: false
    sbg:toolAuthor: Seven Bridges Genomics
    sbg:toolkit: SBGTools
    sbg:toolkitVersion: '1.0'
    sbg:validationErrors: []
  inputs:
  - id: '#SBG_Prepare_Intervals.bed_file'
    source: '#bed'
  - id: '#SBG_Prepare_Intervals.split_mode'
    source: '#split_mode'
  outputs:
  - id: '#SBG_Prepare_Intervals.intervals'
  - id: '#SBG_Prepare_Intervals.names'
  - id: '#SBG_Prepare_Intervals.str_arr'
  sbg:x: 595.0001017252608
  sbg:y: 275.0000254313153
- id: '#SnpEff'
  label: SnpEff
  in: []
  run:
    cwlVersion: sbg:draft-2
    class: CommandLineTool
    label: SnpEff
    $namespaces:
      sbg: https://sevenbridges.com

    requirements:
    - id: '#cwl-js-engine'
      class: ExpressionEngineRequirement
      requirements:
      - class: DockerRequirement
        dockerPull: rabix/js-engine
    - class: InlineJavascriptRequirement

    inputs:
    - id: '#input_format'
      label: Input format
      type:
      - 'null'
      - name: input_format
        type: enum
        symbols:
        - vcf
        - ' bed'
      inputBinding:
        prefix: -i
        position: 40
        separate: true
        sbg:cmdInclude: true
      description: 'Input format. Possible values: {vcf, txt, pileup, bed}. [Default:
        vcf].'
      sbg:category: Options
      sbg:toolDefaultValue: vcf
    - id: '#output_format'
      label: Ouput format
      type:
      - 'null'
      - name: output_format
        type: enum
        symbols:
        - vcf
        - gatk
        - bed
        - bedAnn
      inputBinding:
        prefix: -o
        position: 50
        separate: true
        sbg:cmdInclude: true
      description: 'Ouput format. Possible values: {txt, vcf, gatk, bed, bedAnn}.'
      sbg:category: Options
      sbg:toolDefaultValue: vcf
    - id: '#stats'
      label: Name of stats file (summary)
      type:
      - 'null'
      - string
      inputBinding:
        prefix: -stats
        position: 60
        valueFrom:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              if ($job.inputs.stats)
              {
                return $job.inputs.stats
              }
              else if (!$job.inputs.nostats)
              { 
                filename = $job.inputs.variants_file.path.split('/').pop()
                primename = filename.split('.vcf')[0]
                return primename.concat(".html")
              }
            }
        separate: true
        sbg:cmdInclude: true
      description: Name of stats file (summary).
      sbg:altPrefix: -s
      sbg:category: Options
      sbg:toolDefaultValue: snpEff_summary.html
    - id: '#nostats'
      label: Do not create stats (summary) file
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -noStats
        position: 70
        separate: true
        sbg:cmdInclude: true
      description: Do not create stats (summary) file.
      sbg:category: Options
    - id: '#csvstats'
      label: Create CSV summary file alongside HTML
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -csvStats
        position: 20
        valueFrom:
          class: Expression
          engine: '#cwl-js-engine'
          script: |
            {
              if ($job.inputs.csvstats)
              {
                filename = $job.inputs.variants_file.path.split('/').pop()
                primename = filename.split('.vcf')[0]
                return primename.concat(".csv")
              }
            }
        separate: true
        sbg:cmdInclude: true
      description: Create CSV summary file alongside HTML.
      sbg:category: Options
      sbg:stageInput:
    - id: '#filterinterval'
      label: |-
        Only analyze changes that intersect with the intervals specified in this file (you may use this option many times)
      type:
      - 'null'
      - type: array
        items: File
      inputBinding:
        prefix: -filterInterval
        position: 80
        separate: true
        sbg:cmdInclude: true
      description: Only analyze changes that intersect with the intervals specified
        in this file.
      sbg:altPrefix: -fi
      sbg:category: Results filter options
      sbg:fileTypes: interval
    - id: '#no_downstream'
      label: Do not show downstream changes
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -no-downstream
        position: 90
        separate: true
        sbg:cmdInclude: true
      description: Do not show downstream changes.
      sbg:category: Results filter options
    - id: '#no_intergenic'
      label: Do not show intergenic changes
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -no-intergenic
        position: 100
        separate: true
        sbg:cmdInclude: true
      description: Do not show intergenic changes.
      sbg:category: Results filter options
    - id: '#no_intron'
      label: Do not show intron changes
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -no-intron
        position: 110
        separate: true
        sbg:cmdInclude: true
      description: Do not show intron changes.
      sbg:category: Results filter options
    - id: '#no_upstream'
      label: Do not show upstream changes
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -no-upstream
        position: 120
        separate: true
        sbg:cmdInclude: true
      description: Do not show upstream changes.
      sbg:category: Results filter options
    - id: '#no_utr'
      label: Do not show 5_PRIME_UTR or 3_PRIME_UTR changes
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -no-utr
        position: 130
        separate: true
        sbg:cmdInclude: true
      description: Do not show 5_PRIME_UTR or 3_PRIME_UTR changes.
      sbg:category: Results filter options
    - id: '#cancer'
      label: Perform 'cancer' comparisons (somatic vs. germline)
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -cancer
        position: 140
        separate: true
        sbg:cmdInclude: true
      description: Perform 'cancer' comparisons (somatic vs. germline).
      sbg:category: Annotations options
      sbg:toolDefaultValue: 'False'
    - id: '#cancersamples'
      label: Two column TXT file defining 'original and derived' samples
      type:
      - 'null'
      - File
      inputBinding:
        prefix: -cancerSamples
        position: 150
        separate: true
        sbg:cmdInclude: true
      description: Two column TXT file defining 'original \t derived' samples.
      sbg:category: Annotations options
      sbg:fileTypes: TXT
    - id: '#geneid'
      label: Use gene ID instead of gene name (VCF output)
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -geneId
        position: 170
        separate: true
        sbg:cmdInclude: true
      description: 'Use gene ID instead of gene name (VCF output). Default: false.'
      sbg:category: Annotations options
      sbg:toolDefaultValue: 'False'
    - id: '#add_hgvs_anno'
      label: Use HGVS annotations for amino acid sub-field
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -hgvs
        position: 170
        separate: true
        sbg:cmdInclude: true
      description: 'Use HGVS annotations for amino acid sub-field. Default: true.'
      sbg:category: Annotations options
      sbg:toolDefaultValue: 'True'
    - id: '#add_lof_tag'
      label: Add loss of function (LOF) and nonsense mediated decay (NMD) tags
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -lof
        position: 200
        separate: true
        sbg:cmdInclude: true
      description: Add loss of function (LOF) and nonsense mediated decay (NMD) tags.
      sbg:category: Annotations options
    - id: '#sequenceontology'
      label: Use Sequence Ontology terms
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -sequenceOntology
        position: 250
        separate: true
        sbg:cmdInclude: true
      description: Use Sequence Ontology terms.
      sbg:category: Annotations options
      sbg:toolDefaultValue: 'True'
    - id: '#threads'
      label: Use multiple threads (implies '-noStats')
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -t
        position: 280
        separate: true
        sbg:cmdInclude: true
      description: "Use multiple threads (implies '-noStats'). Default: False."
      sbg:category: Other input types
      sbg:toolDefaultValue: 'False'
    - id: '#canon'
      label: Only use canonical transcripts
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -canon
        position: 300
        separate: true
        sbg:cmdInclude: true
      description: Only use canonical transcripts.
      sbg:category: Database options
    - id: '#interval'
      label: |-
        Use a custom intervals in TXT/BED/BigBed/VCF/GFF file (you may use this option many times)
      type:
      - 'null'
      - type: array
        items: File
      inputBinding:
        prefix: -interval
        position: 320
        separate: true
        sbg:cmdInclude: true
      description: |-
        Use a custom intervals in TXT/BED/BigBed/VCF/GFF file (you may use this option many times).
      sbg:category: Database options
      sbg:fileTypes: TXT, BED, BigBed, VCF, GFF
      sbg:stageInput:
    - id: '#motif'
      label: Annotate using motifs (requires Motif database)
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -motif
        position: 340
        separate: true
        sbg:cmdInclude: true
      description: Annotate using motifs (requires Motif database).
      sbg:category: Database options
      sbg:toolDefaultValue: 'True'
    - id: '#nextprot'
      label: Annotate using NextProt (requires NextProt database)
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -nextProt
        position: 350
        separate: true
        sbg:cmdInclude: true
      description: Annotate using NextProt (requires NextProt database).
      sbg:category: Database options
    - id: '#reg'
      label: Regulation track to use (this option can be used add several times)
      type:
      - 'null'
      - type: array
        items: string
      inputBinding:
        prefix: -reg
        position: 420
        separate: true
        sbg:cmdInclude: true
      description: Regulation track to use (this option can be used add several times).
      sbg:category: Database options
    - id: '#onlyreg'
      label: Only use regulation tracks
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -onlyReg
        position: 400
        separate: true
        sbg:cmdInclude: true
      description: Only use regulation tracks.
      sbg:category: Database options
    - id: '#splicesitesize'
      label: Set size for splice sites (donor and acceptor) in bases
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --spliceSiteSize
        position: 430
        separate: true
        sbg:cmdInclude: true
      description: Set size for splice sites (donor and acceptor) in bases.
      sbg:altPrefix: -ss
      sbg:category: Database options
      sbg:toolDefaultValue: '2'
    - id: '#variants_file'
      label: Input variants file
      type:
      - File
      inputBinding:
        position: 2001
        separate: true
        sbg:cmdInclude: true
      description: Input variants file.
      sbg:category: File type inputs
      sbg:fileTypes: VCF, TXT, PILEUP, BED
    - id: '#total_memory'
      label: Java memory requirement [Gb]
      type:
      - 'null'
      - int
      description: RAM requirement for the java process execution [Gb].
      sbg:category: Other input types
      sbg:toolDefaultValue: '8'
    - id: '#database'
      label: SnpEff database file
      type:
      - File
      description: |-
        SnpEff database file is zip archive that can be downloaded from the SnpEff official site, or using the SnpEff download app.
      sbg:category: File type inputs
      sbg:fileTypes: ZIP
    - id: '#classic'
      label: Use old style annotations
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -classic
        position: 10
        separate: true
        sbg:cmdInclude: true
      description: Use old style annotations instead of Sequence Ontology and Hgvs.
      sbg:category: General options
    - id: '#format_eff'
      label: Use EFF field
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -formatEff
        position: 160
        separate: true
        sbg:cmdInclude: true
      description: Use 'EFF' field compatible with older versions (instead of 'ANN').
      sbg:category: Annotations options
    - id: '#hgvs_1_letter'
      label: Use one letter Amino acid codes in HGVS
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -hgvs1LetterAa
        position: 180
        separate: true
        sbg:cmdInclude: true
      description: 'Use one letter Amino acid codes in HGVS notation. Default: false.'
      sbg:category: Annotations options
      sbg:toolDefaultValue: 'False'
    - id: '#hgvs_tr_id'
      label: Use transcript ID in HGVS
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -hgvsTrId
        position: 190
        separate: true
        sbg:cmdInclude: true
      description: 'Use transcript ID in HGVS notation. Default: false.'
      sbg:category: Annotations options
      sbg:toolDefaultValue: 'False'
    - id: '#no_hgvs'
      label: Do not add HGVS annotations
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -noHgvs
        position: 210
        separate: true
        sbg:cmdInclude: true
      description: Do not add HGVS annotations.
      sbg:category: Annotations options
      sbg:stageInput:
    - id: '#no_lof'
      label: Do not add LOF and NMD annotations
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -noLof
        position: 220
        separate: true
        sbg:cmdInclude: true
      description: Do not add LOF and NMD annotations.
      sbg:category: Annotations options
      sbg:toolDefaultValue: 'False'
    - id: '#no_shift_hgvs'
      label: Do not shift variants according to HGVS
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -noShiftHgvs
        position: 230
        separate: true
        sbg:cmdInclude: true
      description: Do not shift variants according to HGVS notation (most 3-prime
        end).
      sbg:category: Annotations options
      sbg:toolDefaultValue: 'False'
    - id: '#oicr'
      label: Add OICR tag in VCF file
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -oicr
        position: 240
        separate: true
        sbg:cmdInclude: true
      description: Add OICR tag in VCF file.
      sbg:category: Annotations options
      sbg:toolDefaultValue: 'False'
    - id: '#configuration_file'
      label: Configuration file
      type:
      - 'null'
      - File
      inputBinding:
        prefix: -c
        position: 260
        separate: true
        sbg:cmdInclude: true
      description: Specify config file.
      sbg:altPrefix: -config
      sbg:category: Generic options
      sbg:fileTypes: config
      sbg:stageInput: link
    - id: '#interaction'
      label: Annotate using interactions
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -interaction
        position: 310
        separate: true
        sbg:cmdInclude: true
      description: 'Annotate using inteactions (requires interaciton database). Default:
        true.'
      sbg:category: Database options
      sbg:toolDefaultValue: 'True'
    - id: '#max_tsl'
      label: Max TSL
      type:
      - 'null'
      - int
      inputBinding:
        prefix: -maxTSL
        position: 330
        separate: true
        sbg:cmdInclude: true
      description: Only use transcripts having Transcript Support Level lower than
        this cutoff.
      sbg:category: Database options
    - id: '#no_genome'
      label: Do not load any genomic database
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -noGenome
        position: 360
        separate: true
        sbg:cmdInclude: true
      description: Do not load any genomic database (e.g. annotate using custom files).
      sbg:category: Database options
      sbg:stageInput:
    - id: '#no_interaction'
      label: Disable interaction annotations
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -noInteraction
        position: 370
        separate: true
        sbg:cmdInclude: true
      description: Disable interaction annotations.
      sbg:category: Database options
      sbg:toolDefaultValue: 'False'
    - id: '#no_motif'
      label: Disable motif annotations
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -noMotif
        position: 380
        separate: true
        sbg:cmdInclude: true
      description: Disable motif annotations.
      sbg:category: Database options
      sbg:stageInput:
    - id: '#no_next_prot'
      label: Disable NextProt annotations
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -noNextProt
        position: 390
        separate: true
        sbg:cmdInclude: true
      description: Disable NextProt annotations.
      sbg:category: Database options
      sbg:stageInput:
    - id: '#only_protein'
      label: Only use protein coding transcripts
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -onlyProtein
        position: 410
        separate: true
        sbg:cmdInclude: true
      description: 'Only use protein coding transcripts. Default: false.'
      sbg:category: Database options
      sbg:toolDefaultValue: 'False'
    - id: '#splice_region_exons_size'
      label: Set size for splice site region within exons
      type:
      - 'null'
      - int
      inputBinding:
        prefix: -spliceRegionExonSize
        position: 440
        separate: true
        sbg:cmdInclude: true
      description: 'Set size for splice site region within exons. Default: 3 bases.'
      sbg:category: Database options
      sbg:toolDefaultValue: '3'
    - id: '#strict'
      label: Only use validated transcripts
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -strict
        position: 480
        separate: true
        sbg:cmdInclude: true
      description: |-
        Only use 'validated' transcripts (i.e. sequence has been checked). Default: false.
      sbg:category: Database options
      sbg:toolDefaultValue: 'False'
    - id: '#up_down_stream_len'
      label: Upstream downstream interval length
      type:
      - 'null'
      - int
      inputBinding:
        prefix: -upDownStreamLen
        position: 490
        separate: true
        sbg:cmdInclude: true
      description: Set upstream downstream interval length (in bases).
      sbg:altPrefix: -ud
      sbg:category: Database options
    - id: '#chr'
      label: String to prepend to chromosome names
      type:
      - 'null'
      - string
      inputBinding:
        prefix: -chr
        position: 55
        separate: true
        sbg:cmdInclude: true
      description: |-
        Prepend 'string' to chromosome name (e.g. 'chr1' instead of '1'). Only on TXT output.
      sbg:category: Options
    - id: '#no_EffectType'
      label: Do not show EffectType
      type:
      - 'null'
      - type: array
        items: string
      inputBinding:
        prefix: -no
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Do not show 'EffectType'. This option can be used several times.
      sbg:category: Results filter options
    - id: '#hgvsOld'
      label: Use old HGVS notation
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -hgvsOld
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Use old HGVS notation.
      sbg:category: Annotations options
      sbg:toolDefaultValue: 'False'
    - id: '#configOption'
      label: Override a config file option (name=value format)
      type:
      - 'null'
      - type: array
        items: string
      inputBinding:
        prefix: -configOption
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Override a config file option. Please note that the options should be entered in name=value format.
      sbg:category: Generic options
    - id: '#noExpandIUB'
      label: Disable IUB code expansion
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -noExpandIUB
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Disable IUB code expansion in input variants.
      sbg:category: Database options
    - id: '#canonList'
      label: Canon list file
      type:
      - 'null'
      - File
      inputBinding:
        prefix: -canonList
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Only use canonical transcripts, replace some transcripts using the 'gene_id 	 transcript_id' entries in <file>.
      sbg:category: Database options
    - id: '#assembly'
      label: Assembly (genome version)
      type:
      - string
      inputBinding:
        position: 2000
        separate: false
        sbg:cmdInclude: true
      description: |-
        Genome version matching the SnpEff database used (for example GRCh37.75 or GRCh38.86).
      sbg:category: Inputs
    - id: '#onlyTr'
      label: Only use transcripts from this file
      type:
      - 'null'
      - File
      inputBinding:
        prefix: -onlyTr
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: 'Only use the transcripts in this file. Format: One transcript
        ID per line.'
      sbg:category: Database options
      sbg:fileTypes: TXT
    - id: '#splice_region_intron_min'
      label: Set minimum number of bases for splice site region within intron
      type:
      - 'null'
      - int
      inputBinding:
        prefix: -spliceRegionIntronMin
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Set minimum number of bases for splice site region within intron. Default: 3 bases.
      sbg:category: Database options
      sbg:stageInput:
      sbg:toolDefaultValue: '3'
    - id: '#splice_site_region_intron_max'
      label: Set maximum number of bases for splice site region within intron
      type:
      - 'null'
      - int
      inputBinding:
        prefix: -spliceRegionIntronMax
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Set maximum number of bases for splice site region within intron. Default: 8 bases.
      sbg:category: Database options
      sbg:stageInput:
      sbg:toolDefaultValue: '8'

    outputs:
    - id: '#annotated'
      label: SnpEff annotated file
      type:
      - 'null'
      - File
      outputBinding:
        glob:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {	
              filename = $job.inputs.variants_file.path
              basename = filename.split('.').slice(0, filename.split('.').length-1).join('.').replace(/^.*[\\\/]/, '')
             
              
              if($job.inputs.output_format === "txt")
              {
                return basename.concat(".snpEff_annotated.txt")
              }
              else if ($job.inputs.output_format === "bed" || $job.inputs.output_format === "bedAnn")
              {
                return basename.concat(".snpEff_annotated.bed")
              }
              else
              {
                return basename.concat(".snpEff_annotated.vcf")
              }
            }
        sbg:inheritMetadataFrom: '#variants_file'
      description: SnpEff annotated file.
      sbg:fileTypes: VCF, TXT, GATK, BED, BEDANN
    - id: '#summary'
      label: Summary file
      type:
      - 'null'
      - type: array
        items: File
      outputBinding:
        glob: '*.{html,csv}'
        sbg:inheritMetadataFrom: '#variants_file'
      description: SnpEff summary file in HTML or CSV file format.
      sbg:fileTypes: HTML, CSV
    - id: '#summary_text'
      label: Summary
      type:
      - 'null'
      - File
      outputBinding:
        glob: '*.txt'
        sbg:inheritMetadataFrom: '#variants_file'
      description: SnpEff Summary in text format.
      sbg:fileTypes: TXT

    baseCommand:
    - class: Expression
      engine: '#cwl-js-engine'
      script: "{\n  return 'unzip -o ' + $job.inputs.database.path + ' -d /opt/snpEff\
        \ ;'\n}"
    - java
    - class: Expression
      engine: '#cwl-js-engine'
      script: |+
        {
          //java -Xmx4g path/to/snpEff/snpEff.jar -c path/to/snpEff/snpEff.config GRCh37.75 path/to/snps.vcf
          if($job.inputs.total_memory){
            mem_mb = parseInt($job.inputs.total_memory) * 1024
          	return '-Xmx'.concat(mem_mb, 'M')
          }
          	return '-Xmx8192M'
        }

    - -jar
    - /opt/snpEff/snpEff.jar
    arguments:
    - position: 5000
      valueFrom:
        class: Expression
        engine: '#cwl-js-engine'
        script: |
          {	
            filename = $job.inputs.variants_file.path
            basename = filename.split('.').slice(0, filename.split('.').length-1).join('.').replace(/^.*[\\\/]/, '')
           
            
            if($job.inputs.output_format === "txt")
            {
              name = basename.concat(".snpEff_annotated.txt")
            }
            else if ($job.inputs.output_format === "bed" || $job.inputs.output_format === "bedAnn")
            {
              name = basename.concat(".snpEff_annotated.bed")
            }
            else
            {
              name = basename.concat(".snpEff_annotated.vcf")
            }
            return '> ' + name
          }
      separate: true
    - position: 0
      valueFrom: -nodownload
      separate: true
    - position: 0
      valueFrom: -noLog
      separate: true

    hints:
    - class: sbg:CPURequirement
      value: 1
    - class: sbg:MemRequirement
      value:
        class: Expression
        engine: '#cwl-js-engine'
        script: |-
          {
            if ($job.inputs.total_memory)
            {
              return $job.inputs.total_memory * 1024
            }
            
            else
            {
              return 8192
            }
          }
    - class: DockerRequirement
      dockerPull: images.sbgenomics.com/jrandjelovic/snpeff:v4.3k
      dockerImageId: aae3dcb89b53
    id: admin/sbg-public-data/snpeff-4-3/5
    abg:revisionNotes: cosmetic, label typo
    description: |-
      **SnpEff** is a variant annotation and effect prediction​ tool, which annotates and predicts the effects of variants on genes, such as amino acid changes [1].

      *A list of **all inputs and parameters** with corresponding descriptions can be found at the end of the page.*

      ### Common Use Cases

      Typical usage assumes predicted variants (SNPs, insertions, deletions, and MNPs) as input, usually in variant call format (VCF). **SnpEff** analyzes and annotates input variants and calculates the effects they produce on known genes [1]. The output file can be in several file formats, most common being VCF.

      **SnpEff** requires an annotation database to run. Official SnpEff annotation databases can be downloaded from [here](https://sourceforge.net/projects/snpeff/files/databases/v4_3/); however, human databases are also hosted on Seven Bridges, in Public Reference Files section (files snpEff_v4_3_GRCh38.86.zip and	snpEff_v4_3_GRCh37.75.zip) and can be [imported](https://docs.sevenbridges.com/docs/copy-files-using-the-visual-interface).

      ### Changes Introduced by Seven Bridges

      * Input VCF file (**Input variants file**) is required (as opposed to default input being STDIN).
      * Parameter **Java memory requirement [Gb]** which controls the amount of RAM available to **SnpEff** was included in the wrapper.
      * The following parameters have been excluded from the wrapper:  
          * `-download`  - The tool is expected to use data from the provided database archive.
          * `-fileList` - Processing multiple files can be achieved by using batch tasks or scatter mode in workflows.
          * `-dataDir <path>`  - In the wrapper, data directory is always the same and corresponds to the location of the prepared **SnpEff** database.
          * `-download`  - Supplying a database archive as an input is required. Downloading missing data for a genome from command line is not supported.
          * `-help`, `-quiet`, `-verbose`, `-debug` and `-version` - These options are not usually included in Seven Bridges wrappers.

      ### Common Issues and Important Notes

      * Required inputs are **Input variants file** (a VCF or VCF.GZ file to be annotated), **SnpEff database file** (SnpEff database ZIP archive matching the major version of SnpEff used [2], which is 4.3 for this wrapper; e.g. snpEff_v4_3_GRCh38.86.zip or snpEff_v4_3_GRCh37.75.zip from Public Reference Files section), and **Assembly (genome version)**, which is a string representing genome version/assembly (e.g., GRCh38.86, GRCh37.75, hg19), matching the SnpEff database used (GRCh38.86 and GRCh37.75 should be used for the files in the Public Reference Files section).
      * As **SnpEff** is a java tool, it may be occasionally necessary to increase the amount of allocated RAM (default value: 8192 MB), using the **Java memory requirement [Gb]** parameter.
      * A number of **SnpEff** command line options are designed in mutually exclusive pairs (for example `-noStats` and `-stats` or `-lof` and `-noLof`) with some redundancy. These options should not be used together, to avoid task failure.
      * Multithreading parameter **Use multiple threads (implies '-noStats')** (`-t`) will disable statistics. 
      * If using VCFs with mitochondrial DNA marked as chrM with GRCh38 database, chromosome not found error can be addressed by renaming chrM to chrMT in the input VCF files, for example using sed: `sed "s/^chrM/chrMT/g" input.vcf > input_renamed.vcf`
      * Disabling statistics using **Do not create stats (summary) file** (`-noStats`) will in general speed-up execution.

      ### Performance Benchmarking

      Annotating NA12878 genome (GRCh38, ~220 Mb as VCF.GZ) with default annotation parameters, 1 CPU, and 8192 MB RAM took 25 minutes with a cost of $0.17 using on-demand default instance.
      By default, **SnpEff** is allocated 8192 MB of memory. Allocating less memory is not recommended when working with whole genome VCF files.

      *Cost can be significantly reduced by **spot instance** usage. Visit [knowledge center](https://docs.sevenbridges.com/docs/about-spot-instances) for more details.*            

      ### References

      [1] [SnpEff documentation](http://snpeff.sourceforge.net/SnpEff_manual.html)

      [2] [Official SnpEff 4.3 databases download location](https://sourceforge.net/projects/snpeff/files/databases/v4_3/)
    sbg:appVersion:
    - sbg:draft-2
    sbg:categories:
    - Annotation
    - VCF-Processing
    sbg:cmdPreview: |-
      unzip -o /path/to/database/GRCh37.75.zip -d /opt/snpEff ; java -Xmx3072M -jar /opt/snpEff/snpEff.jar  -nodownload  -noLog GRCh37.75  path/to/variants/variants_file.vcf  > variants_file.snpEff_annotated.vcf
    sbg:content_hash: a963e94586701a4e7b7c8696eaae5359035e5c02c31feb2a2f241fee4535d756b
    sbg:contributors:
    - admin
    sbg:createdBy: admin
    sbg:createdOn: 1512132000
    sbg:id: admin/sbg-public-data/snpeff-4-3/5
    sbg:image_url:
    sbg:job:
      allocatedResources:
        cpu: 1
        mem: 3072
      inputs:
        add_hgvs_anno: false
        assembly: GRCh37.75
        chr: ''
        classic: false
        configOption:
        configuration_file:
          class: File
          secondaryFiles: []
          path:
          size: 0
        csvstats: false
        database:
          secondaryFiles: []
          metadata: {}
          path: /path/to/database/GRCh37.75.zip
        format_eff: false
        hgvsOld: false
        hgvs_1_letter: false
        hgvs_tr_id: false
        input_format:
        interaction: false
        interval:
        - class: File
          secondaryFiles: []
          path: /path/to/interval-1.ext
          size: 0
        - class: File
          secondaryFiles: []
          path: /path/to/interval-2.ext
          size: 0
        max_tsl:
        noExpandIUB: false
        no_EffectType:
        - no_EffectType-string-value-1
        - no_EffectType-string-value-2
        no_genome: false
        no_hgvs: false
        no_interaction: false
        no_lof: false
        no_motif: false
        no_next_prot: false
        no_shift_hgvs: false
        nostats: false
        oicr: false
        onlyTr:
          class: File
          secondaryFiles: []
          path: /path/to/onlyTr.ext
          size: 0
        only_protein: false
        output_format:
        splice_region_exons_size:
        splice_region_intron_min: 7
        splice_site_region_intron_max: 8
        stats: ''
        strict: false
        threads: false
        total_memory: 3
        up_down_stream_len:
        variants_file:
          class: File
          secondaryFiles: []
          path: path/to/variants/variants_file.vcf
          size: 0
    sbg:latestRevision: 5
    sbg:license: GNU Lesser General Public License v3.0 only
    sbg:links:
    - id: http://snpeff.sourceforge.net/index.html
      label: Homepage
    - id: https://github.com/pcingola/SnpEff
      label: Source Code
    - id: http://snpeff.sourceforge.net/SnpEff_manual.html
      label: Wiki
    - id: http://sourceforge.net/projects/snpeff/files/snpEff_latest_core.zip
      label: Download
    - id: http://snpeff.sourceforge.net/SnpEff_paper.pdf
      label: Publication
    sbg:modifiedBy: admin
    sbg:modifiedOn: 1542279700
    sbg:project: admin/sbg-public-data
    sbg:projectName: SBG Public data
    sbg:publisher: sbg
    sbg:revision: 5
    sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpeff-4-2/31
    sbg:revisionsInfo:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1512132000
      sbg:revision: 0
      sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpeff-4-2/19
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1512132000
      sbg:revision: 1
      sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpeff-4-2/20
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1512132000
      sbg:revision: 2
      sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpeff-4-2/26
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1534244704
      sbg:revision: 3
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1542279699
      sbg:revision: 4
      sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpeff-4-2/30
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1542279700
      sbg:revision: 5
      sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpeff-4-2/31
    sbg:sbgMaintained: false
    sbg:toolAuthor: Pablo Cingolani/Broad Institue
    sbg:toolkit: SnpEff
    sbg:toolkitVersion: 4.3k
    sbg:validationErrors: []
  inputs:
  - id: '#SnpEff.add_hgvs_anno'
    default: true
  - id: '#SnpEff.add_lof_tag'
    default: true
  - id: '#SnpEff.canon'
    default: true
  - id: '#SnpEff.variants_file'
    source: '#gatk_variantannotator.annotated_variants'
  - id: '#SnpEff.database'
    source: '#database'
  - id: '#SnpEff.format_eff'
    default: true
  - id: '#SnpEff.assembly'
    default: hg38
  outputs:
  - id: '#SnpEff.annotated'
  - id: '#SnpEff.summary'
  - id: '#SnpEff.summary_text'
  sbg:x: 1499.1290283203125
  sbg:y: 311.77252197265625
- id: '#SnpSift_Annotate'
  label: SnpSift Annotate
  in: []
  run:
    cwlVersion: sbg:draft-2
    class: CommandLineTool
    label: SnpSift Annotate
    $namespaces:
      sbg: https://sevenbridges.com

    requirements:
    - id: '#cwl-js-engine'
      class: ExpressionEngineRequirement
      requirements:
      - class: DockerRequirement
        dockerPull: rabix/js-engine
    - class: InlineJavascriptRequirement

    inputs:
    - id: '#vcf_database_is_sorted'
      label: VCF database is sorted and uncompressed
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -sorted
        position: 16
        separate: true
        sbg:cmdInclude: true
      description: VCF database is sorted and uncompressed.
      sbg:category: Command options
      sbg:toolDefaultValue: 'False'
    - id: '#variant_exist'
      label: Annotate whether the variant exists in the database
      type:
      - 'null'
      - string
      inputBinding:
        prefix: -exists
        position: 2
        separate: true
        sbg:cmdInclude: true
      description: |-
        Annotate whether the variant exists or not in the database (using the supplied value as an INFO field FLAG).
      sbg:category: Command options
    - id: '#tabix'
      label: VCF database is tabix-indexed
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -tabix
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: VCF database is tabix-indexed.
      sbg:category: Command options
      sbg:toolDefaultValue: 'False'
    - id: '#no_ref_alt'
      label: Do not use REF and ALT fields
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -noAlt
        position: 10
        separate: true
        sbg:cmdInclude: true
      description: |-
        Do not use REF and ALT fields when comparing database.vcf entries to file.vcf entries.
      sbg:category: Command options
      sbg:toolDefaultValue: 'False'
    - id: '#no_id'
      label: Do not annotate ID field
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -noId
        position: 12
        separate: true
        sbg:cmdInclude: true
      description: Do not annotate ID field.
      sbg:category: Command options
      sbg:toolDefaultValue: 'False'
    - id: '#mem_mb'
      label: Java Xmx%m requirement
      type:
      - 'null'
      - int
      description: RAM requirement for the java process execution.
      sbg:category: Other input types
      sbg:toolDefaultValue: '8192'
    - id: '#max_block_size'
      label: Max block size
      type:
      - 'null'
      - int
      inputBinding:
        prefix: -maxBlockSize
        position: 8
        separate: true
        sbg:cmdInclude: true
      description: |-
        Use 'max block size' when creating index ('-sorted' command line option). Default: 16384.
      sbg:category: Command options
      sbg:toolDefaultValue: '16384'
    - id: '#input_vcf'
      label: VCF file that will be annotated
      type:
      - File
      inputBinding:
        position: 110
        separate: true
        sbg:cmdInclude: true
      description: |-
        VCF file that will be annotated with the INFO and FORMAT fields from the annotation VCF.
      required: true
      sbg:category: File type inputs
      sbg:fileTypes: VCF
    - id: '#info'
      label: Annotate using a list of INFO fields
      type:
      - 'null'
      - type: array
        items: string
      inputBinding:
        prefix: -info
        position: 5
        separate: true
        itemSeparator: ','
        sbg:cmdInclude: true
      description: Annotate using a list of INFO fields (list is a comma separated
        list of fields).
      sbg:category: Command options
      sbg:toolDefaultValue: All fields
    - id: '#id_field'
      label: Only annotate ID field
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -id
        position: 4
        separate: true
        sbg:cmdInclude: true
      description: Only annotate ID field (do not add INFO field).
      sbg:category: Command options
      sbg:toolDefaultValue: 'False'
    - id: '#force_annotate'
      label: Annotate fields even if value is absent from the annotation database
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -a
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Annotate fields, even if the database has an empty value (annotates using '.' for empty).
      sbg:category: Command options
      sbg:toolDefaultValue: 'False'
    - id: '#do_no_annotate_info_fields'
      label: Do not annotate INFO fields
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -noInfo
        position: 14
        separate: true
        sbg:cmdInclude: true
      description: Do not annotate INFO fields.
      sbg:category: Command options
      sbg:toolDefaultValue: 'False'
    - id: '#database_type'
      label: Database type
      type:
      - 'null'
      - name: database_type
        type: enum
        symbols:
        - vcf
        - dbsnp
        - clinvar
      inputBinding:
        position: 90
        valueFrom:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              if ($job.inputs.database_type == 'vcf') {return ''}
              else if ($job.inputs.database_type == 'dbsnp') {return '-dbsnp'}
              else if ($job.inputs.database_type == 'clinvar') {return '-clinvar'}
            }
        separate: true
        sbg:cmdInclude: true
      description: Type of database used.
      sbg:category: Database options
      sbg:toolDefaultValue: vcf
    - id: '#database_for_annotation'
      label: Database for annotation
      type:
      - File
      inputBinding:
        secondaryFiles:
        - .tbi
        position: 92
        valueFrom:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              type = $job.inputs.database_type
              if (type == 'dbsnp' || type == 'clinvar'){
                return '-db ' + $job.inputs.database_for_annotation.path
              } else {
                return $job.inputs.database_for_annotation.path
              }
            }
        sbg:cmdInclude: true
      description: Database for annotating the input VCF file.
      required: true
      sbg:category: Database options
      sbg:fileTypes: VCF.GZ
      sbg:stageInput: link
    - id: '#append_name_to_info_fields'
      label: String prefix for all annotated INFO fields
      type:
      - 'null'
      - string
      inputBinding:
        prefix: -name
        position: 7
        separate: true
        sbg:cmdInclude: true
      description: String to prepend to all annotated INFO fields.
      sbg:category: Command options
      sbg:toolDefaultValue: "''"

    outputs:
    - id: '#annotated_vcf'
      label: Annotated VCF
      type:
      - 'null'
      - File
      outputBinding:
        glob: '*.annotated.vcf'
        sbg:inheritMetadataFrom: '#input_vcf'
      description: Annotated VCF.
      sbg:fileTypes: VCF
    stdout:
      class: Expression
      engine: '#cwl-js-engine'
      script: |-
        {
          filename = $job.inputs.input_vcf.path
          return filename.split('.').slice(0, filename.split('.').length-1).concat("annotated.vcf").join('.').replace(/^.*[\\\/]/, '')
        }

    baseCommand:
    - java
    - class: Expression
      engine: '#cwl-js-engine'
      script: |-
        {
          if ($job.inputs.mem_mb)
          {
            return "-Xmx" + ($job.inputs.mem_mb).toString() + "m"
          }
          else
          {
            return "-Xmx8192m"
          }
        }
    - -jar
    - /opt/snpEff/SnpSift.jar
    - annotate
    - -noDownload
    - -noLog

    hints:
    - class: sbg:CPURequirement
      value: 1
    - class: sbg:MemRequirement
      value:
        class: Expression
        engine: '#cwl-js-engine'
        script: |-
          {
            if ($job.inputs.mem_mb)
            {
              return $job.inputs.mem_mb
            }
            
            else
            {
              return 8192
            }
          }
    - class: DockerRequirement
      dockerPull: images.sbgenomics.com/jrandjelovic/snpeff:v4.3k
      dockerImageId: 26f735e2c570
    id: admin/sbg-public-data/snpsift-annotate-4-3/2
    description: |-
      **SnpSift Annotate** uses fields from another VCF file (such as [dbSNP](https://www.ncbi.nlm.nih.gov/projects/SNP/snp_summary.cgi), [1000 genomes](http://www.internationalgenome.org/data), [ClinVar](https://www.ncbi.nlm.nih.gov/clinvar/), or [ExAC](http://exac.broadinstitute.org/downloads)) to annotate input VCF files [1].

      *A list of **all inputs and parameters** with corresponding descriptions can be found at the end of the page.*

      ### Common Use Cases

      **SnpSift Annotate** is typically used to annotate VCF ID and INFO fields from a "database" VCF file, such as dbSNP.

      ### Changes Introduced by Seven Bridges

      * VCF file used as the annotation database (**Database for annotation**) should be compressed and tabix-indexed (to prepare the file, please use **Tabix Bgzip** and **Tabix Index** tools which can be found under Public Apps gallery).

      ### Common Issues and Important Notes

      * Inputs **VCF file that will be annotated** and **Database for annotation** are required.
      * Note that **Database for annotation** has to be a compressed and tabix-indexed VCF file.
      * By default **SnpSift Annotate** adds **all** database INFO fields.
      You can use the **Annotate using a list of INFO fields** (`-info`) parameter if you want to select only a subset of fields from the database VCF file and **Only annotate ID field** (`-id`) parameter if you only want to annotate using ID field (no INFO fields will be added) [1].
      * Parameters **VCF database is sorted and uncompressed** (`-sorted`) and **Max block size** (`-maxBlockSize <int>`) should be used together and applied only if the annotation database VCF file is uncompressed.

      ### Performance Benchmarking

      Adding dbSNP (v150) annotations (all fields) to SnpEff-annotated NA12878 genome (GRCh38 ~3 GB as VCF) with 15 GB RAM took 2 h 23 minutes (price: $0.95 using on-demand default instance)

      *Cost can be significantly reduced by **spot instance** usage. Visit [knowledge center](https://docs.sevenbridges.com/docs/about-spot-instances) for more details.*            

      ### References

      [1] [SnpSift Annotate documentation](http://snpeff.sourceforge.net/SnpSift.html#annotate)
    sbg:appVersion:
    - sbg:draft-2
    sbg:categories:
    - Annotation
    sbg:cmdPreview: |-
      java -Xmx8192m -jar /opt/snpEff/SnpSift.jar annotate -noDownload -noLog /path/to/database_for_annotation.ext  input.vcf > input.annotated.vcf
    sbg:contributors:
    - admin
    sbg:createdBy: admin
    sbg:createdOn: 1512132000
    sbg:id: admin/sbg-public-data/snpsift-annotate-4-3/2
    sbg:image_url:
    sbg:job:
      allocatedResources:
        cpu: 1
        mem: 8192
      inputs:
        database_for_annotation:
          class: File
          secondaryFiles:
          - path: .tbi
          path: /path/to/database_for_annotation.ext
          size: 0
        database_type: vcf
        do_no_annotate_info_fields: false
        force_annotate: false
        genome_version: genome_version
        id_field: false
        info:
        - test_field1
        - test_field2
        input_vcf:
          class: File
          secondaryFiles: []
          path: input.vcf
          size: 0
        max_block_size:
        mem_mb:
        no_ref_alt: false
        tabix: true
        variant_exist: ''
        vcf_database_is_sorted: false
    sbg:latestRevision: 2
    sbg:license: GNU Lesser General Public License v3.0 only
    sbg:links:
    - id: http://snpeff.sourceforge.net/index.html
      label: Homepage
    - id: https://github.com/pcingola/SnpEff
      label: Source Code
    - id: http://snpeff.sourceforge.net/SnpEff_manual.html
      label: Wiki
    - id: https://sourceforge.net/projects/snpeff/files/snpEff_v4_2_core.zip/download
      label: Download
    - id: http://snpeff.sourceforge.net/SnpEff_paper.pdf
      label: Publication
    sbg:modifiedBy: admin
    sbg:modifiedOn: 1512140672
    sbg:project: admin/sbg-public-data
    sbg:projectName: SBG Public data
    sbg:publisher: sbg
    sbg:revision: 2
    sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-annotate-4-2/13
    sbg:revisionsInfo:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1512132000
      sbg:revision: 0
      sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-annotate-4-2/6
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1512132000
      sbg:revision: 1
      sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-annotate-4-2/12
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1512140672
      sbg:revision: 2
      sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-annotate-4-2/13
    sbg:sbgMaintained: false
    sbg:toolAuthor: Pablo Cingolani/Broad Institute
    sbg:toolkit: SnpEff
    sbg:toolkitVersion: 4.3k
    sbg:validationErrors: []
    x: 1606.6670735677094
    y: 456.66676839192735
  inputs:
  - id: '#SnpSift_Annotate.input_vcf'
    source: '#SnpSift_dbNSFP.annotated_vcf'
  - id: '#SnpSift_Annotate.id_field'
    default: true
  - id: '#SnpSift_Annotate.database_for_annotation'
    source: '#database_for_annotation'
  outputs:
  - id: '#SnpSift_Annotate.annotated_vcf'
  sbg:x: 1756.8115234375
  sbg:y: 322.2614440917969
- id: '#SnpSift_dbNSFP'
  label: SnpSift dbNSFP
  in: []
  run:
    cwlVersion: sbg:draft-2
    class: CommandLineTool
    label: SnpSift dbNSFP
    $namespaces:
      sbg: https://sevenbridges.com

    requirements:
    - id: '#cwl-js-engine'
      class: ExpressionEngineRequirement
      requirements:
      - class: DockerRequirement
        dockerPull: rabix/js-engine
    - class: InlineJavascriptRequirement

    inputs:
    - id: '#mem_mb'
      label: RAM requirement in MB
      type:
      - 'null'
      - int
      description: RAM requirement for the java process execution.
      sbg:category: Configuration
      sbg:toolDefaultValue: '2048'
    - id: '#invert_fields_to_add'
      label: Invert 'fields to add' selection
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -n
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Invert 'fields to add' selection (i.e. use all fields except the ones specified in option '-f').
      sbg:category: Options
      sbg:stageInput:
    - id: '#input_vcf'
      label: VCF file that will be annotated
      type:
      - File
      inputBinding:
        position: 12
        separate: true
        sbg:cmdInclude: true
      description: VCF file that will be annotated with dbNSFP database.
      required: true
      sbg:category: Input files
      sbg:fileTypes: VCF
    - id: '#genome_version'
      label: Genome version
      type:
      - 'null'
      - string
      inputBinding:
        prefix: -g
        position: 6
        separate: true
        sbg:cmdInclude: true
      description: Genome version.
      sbg:category: Options
    - id: '#force_anno'
      label: Annotate fields, even if the database has an empty value
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -a
        position: 1
        separate: true
        sbg:cmdInclude: true
      description: |-
        Annotate fields, even if the database has an empty value (annotates using '.' for empty).
      sbg:category: Options
      sbg:toolDefaultValue: 'False'
    - id: '#dbnsfp_series'
      label: dbNSFP version
      type:
      - name: dbnsfp_series
        type: enum
        symbols:
        - dbNSFP 2.x
        - dbNSFP 3.x
      description: |-
        Version of dbNSFP used. 2.x should be used for GRCh37 data, 3.x for GRCh38 data. This parameter only affects the default database field sets.
    - id: '#database'
      label: Database for annotation
      type:
      - File
      inputBinding:
        secondaryFiles:
        - .tbi
        prefix: -db
        position: 10
        separate: true
        sbg:cmdInclude: true
      description: Database in VCF format for the annotation of the input VCF file.
      required: true
      sbg:category: Options
      sbg:fileTypes: GZ, TXT.GZ, VCF.GZ
      sbg:stageInput: link
    - id: '#colapse'
      label: Colapse repeated values
      type:
      - 'null'
      - name: colapse
        type: enum
        symbols:
        - Don't collapse
        - Collapse
      inputBinding:
        position: 2
        valueFrom:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              if ($job.inputs.colapse === "Collapse")
              {
                return "-collapse"
              }
              else
              {
                return "-noCollapse"
              }
            }
        separate: true
        sbg:cmdInclude: true
      description: Collapse repeated values from dbNSFP.
      sbg:category: Options
      sbg:toolDefaultValue: Don't colapse
    - id: '#annotate_missing_entries'
      label: Annotate missing entries (with '.')
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -m
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Annotate fields even if there are no database matching entries (annotates using '.').
      sbg:category: Options
      sbg:stageInput:
      sbg:toolDefaultValue: 'False'
    - id: '#anno_fields'
      label: Database fields to add
      type:
      - 'null'
      - type: array
        items: string
      inputBinding:
        prefix: -f
        position: 3
        valueFrom:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              if ($job.inputs.anno_fields)
              {
                return $job.inputs.anno_fields
              }
              else if ($job.inputs.dbnsfp_series == "dbNSFP 2.x")
              {
               return "Interpro_domain,SIFT_pred,LRT_pred,MutationTaster_pred,GERP++_NR,GERP++_RS,phastCons100way_vertebrate,MutationAssessor_pred,FATHMM_pred,PROVEAN_pred,MetaSVM_pred,1000Gp1_AC,1000Gp1_AF,1000Gp1_AFR_AC,1000Gp1_AFR_AF,1000Gp1_EUR_AC,1000Gp1_EUR_AF,1000Gp1_AMR_AC,1000Gp1_AMR_AF,1000Gp1_ASN_AC,1000Gp1_ASN_AF,ESP6500_AA_AF,ESP6500_EA_AF,ExAC_AC,ExAC_AF,ExAC_Adj_AC,ExAC_Adj_AF,ExAC_AFR_AC,ExAC_AFR_AF,ExAC_AMR_AC,ExAC_AMR_AF,ExAC_EAS_AC,ExAC_EAS_AF,ExAC_FIN_AC,ExAC_FIN_AF,ExAC_NFE_AC,ExAC_NFE_AF,ExAC_SAS_AC,ExAC_SAS_AF"
              }
              else
              {
                return "Interpro_domain,SIFT_pred,LRT_pred,MutationTaster_pred,GERP++_NR,GERP++_RS,phastCons100way_vertebrate,MutationAssessor_pred,FATHMM_pred,PROVEAN_pred,MetaSVM_pred,1000Gp3_AC,1000Gp3_AF,1000Gp3_AFR_AC,1000Gp3_AFR_AF,1000Gp3_EUR_AC,1000Gp3_EUR_AF,1000Gp3_AMR_AC,1000Gp3_AMR_AF,1000Gp3_EAS_AC,1000Gp3_EAS_AF,1000Gp3_SAS_AC,1000Gp3_SAS_AF,ESP6500_AA_AC,ESP6500_AA_AF,ESP6500_EA_AC,ESP6500_EA_AF,ExAC_AC,ExAC_AF,ExAC_Adj_AC,ExAC_Adj_AF,ExAC_AFR_AC,ExAC_AFR_AF,ExAC_AMR_AC,ExAC_AMR_AF,ExAC_EAS_AC,ExAC_EAS_AF,ExAC_FIN_AC,ExAC_FIN_AF,ExAC_NFE_AC,ExAC_NFE_AF,ExAC_SAS_AC,ExAC_SAS_AF"
              }
            }
        separate: true
        itemSeparator: ','
        sbg:cmdInclude: true
      description: A comma separated list of fields to add.
      sbg:category: Options
      sbg:toolDefaultValue: |-
        Interpro_domain,SIFT_pred,LRT_pred,MutationTaster_pred,GERP++_NR,GERP++_RS,phastCons100way_vertebrate,MutationAssessor_pred,FATHMM_pred,PROVEAN_pred,MetaSVM_pred,1000Gp3_AC,1000Gp3_AF,1000Gp3_AFR_AC,1000Gp3_AFR_AF,1000Gp3_EUR_AC,1000Gp3_EUR_AF,1000Gp3_AMR_AC,1000Gp3_AMR_AF,1000Gp3_EAS_AC,1000Gp3_EAS_AF,1000Gp3_SAS_AC,1000Gp3_SAS_AF,ESP6500_AA_AC,ESP6500_AA_AF,ESP6500_EA_AC,ESP6500_EA_AF,ExAC_AC,ExAC_AF,ExAC_Adj_AC,ExAC_Adj_AF,ExAC_AFR_AC,ExAC_AFR_AF,ExAC_AMR_AC,ExAC_AMR_AF,ExAC_EAS_AC,ExAC_EAS_AF,ExAC_FIN_AC,ExAC_FIN_AF,ExAC_NFE_AC,ExAC_NFE_AF,ExAC_SAS_AC,ExAC_SAS_AF

    outputs:
    - id: '#annotated_vcf'
      label: DBNSFP annotated VCF
      type:
      - 'null'
      - File
      outputBinding:
        glob: '*.dbnsfp_anno.vcf'
        sbg:inheritMetadataFrom: '#input_vcf'
      description: DBNSFP annotated VCF.
      sbg:fileTypes: VCF
    stdout:
      class: Expression
      engine: '#cwl-js-engine'
      script: |-
        {
          filename = $job.inputs.input_vcf.path
          return filename.split('.').slice(0, filename.split('.').length-1).concat("dbnsfp_anno.vcf").join('.').replace(/^.*[\\\/]/, '')
        }

    baseCommand:
    - java
    - class: Expression
      engine: '#cwl-js-engine'
      script: |-
        {
          if ($job.inputs.mem_mb)
          {
            return "-Xmx" + ($job.inputs.mem_mb).toString() + "m"
          }
          else
          {
            return "-Xmx2048m"
          }
        }
    - -jar
    - /opt/snpEff/SnpSift.jar
    - dbnsfp
    arguments:
    - position: 5
      valueFrom:
        class: Expression
        engine: '#cwl-js-engine'
        script: "{\n  return \"-noDownload -noLog\"\n}"
      separate: true

    hints:
    - class: sbg:CPURequirement
      value: 1
    - class: sbg:MemRequirement
      value:
        class: Expression
        engine: '#cwl-js-engine'
        script: |-
          {
            if ($job.inputs.mem_mb)
            {
              return $job.inputs.mem_mb
            }
            
            else
            {
              return 2048
            }
          }
    - class: DockerRequirement
      dockerPull: images.sbgenomics.com/jrandjelovic/snpeff:v4.3k
      dockerImageId: 26f735e2c570
    id: admin/sbg-public-data/snpsift-dbnsfp-4-3/2
    description: |-
      **SnpSift dbNSFP** allows annotation with [dbNSFP](https://sites.google.com/site/jpopgen/dbNSFP) [1], an integrated database of functional predictions from multiple algorithms (SIFT, Polyphen2, LRT and MutationTaster, PhyloP and GERP++, etc.) [2].

      *A list of **all inputs and parameters** with corresponding descriptions can be found at the end of the page.*

      ### Common Use Cases

      Adding **dbNSFP** annotations to VCF files.

      ### Changes Introduced by Seven Bridges

      * To adapt the tool to newer versions of **dbNSFP** (currently 3.5), which have changed some of the fields (column headers) available for annotation, default list of annotation fields used by the tool has been somewhat updated to exclude fields no longer found in the database. Currently used default annotation fields list consists of the following fields:   
        
        `Interpro_domain,SIFT_pred, LRT_pred,MutationTaster_pred, GERP++_NR,GERP++_RS, phastCons100way_vertebrate, MutationAssessor_pred, FATHMM_pred, PROVEAN_pred, MetaSVM_pred, 1000Gp3_AC, 1000Gp3_AF, 1000Gp3_AFR_AC, 1000Gp3_AFR_AF, 1000Gp3_EUR_AC, 1000Gp3_EUR_AF, 1000Gp3_AMR_AC, 1000Gp3_AMR_AF, 1000Gp3_EAS_AC, 1000Gp3_EAS_AF, 1000Gp3_SAS_AC, 1000Gp3_SAS_AF, ESP6500_AA_AC, ESP6500_AA_AF, ESP6500_EA_AC, ESP6500_EA_AF, ExAC_AC, ExAC_AF, ExAC_Adj_AC, ExAC_Adj_AF, ExAC_AFR_AC, ExAC_AFR_AF, ExAC_AMR_AC, ExAC_AMR_AF, ExAC_EAS_AC, ExAC_EAS_AF, ExAC_FIN_AC, ExAC_FIN_AF, ExAC_NFE_AC, ExAC_NFE_AF, ExAC_SAS_AC, ExAC_SAS_AF`.

      When using dbNSFP 2.x files, the above list is modified to exclude `ESP6500_AA_AC,ESP6500_EA_AC` and replace 1000 genomes phase 3 fields with 1000 genomes phase 1 allele frequency fields found in dbNSFP 2.9.x files.

      ### Common Issues and Important Notes

      * Please note that **dbNSFP** database file used for annotation (input **Database for annotation**) should be preprocessed with bgzip and tabix before use. **dbNSFP** database files hosted on the Seven Bridges platform in the Public Reference Files section (dbNSFP_3.3c.gz and dbNSFP2.9.3.txt.gz) and older versions of the database available from SnpSift website links [2] have already been preprocessed. If you wish to preprocess your database files locally, this is one possible way (assumes a unix-based workstation) [3]:

          `unzip dbNSFPv3.x.zip`

          `head -n1 dbNSFP3.x_variant.chr1 ` > ` h`

          `cat dbNSFP3.x_variant.chr* | grep -v ^#chr | sort -k1,1 -k2,2n - | cat h - | bgzip -c ` > ` dbNSFP_3.x.gz`

          `tabix -s 1 -b 2 -e 2 dbNSFP_3.x.gz`

      * Please make sure that the **dbNSFP** version used matches the genome assembly version of your data (3.x for GRCh38 and 2.x for GRCh37).
      * Seven Bridges hosts **dbNSFP** files available for commercial use (dbNSFP_3.3c.gz and dbNSFP2.9.3.txt.gz), however please note that additional annotations are available to researchers in academic versions of **dbNSFP** database available on the **dbNSFP** website [1].

      ### Performance Benchmarking

      Annotating NA12878 genome (GRCh38, ~3 GB as VCF.GZ) with default **dbNSFP** (v3.5c) annotation fields and 8 GB RAM  took 42 minutes on the default instance (price: $0.28 using on-demand instance).

      *Cost can be significantly reduced by **spot instance** usage. Visit [knowledge center](https://docs.sevenbridges.com/docs/about-spot-instances) for more details.*            


      ### References

      [1] [dbNSFP website](https://sites.google.com/site/jpopgen/dbNSFP)

      [2] [SnpSift dbNSFP documentation](http://snpeff.sourceforge.net/SnpSift.html#dbNSFP)

      [3] [Ensembl VEP dbNSFP plugin documentation](https://github.com/Ensembl/VEP_plugins/blob/release/90/dbNSFP.pm)
    sbg:appVersion:
    - sbg:draft-2
    sbg:categories:
    - Annotation
    sbg:cmdPreview: |-
      java -Xmx5000m -jar /opt/snpEff/SnpSift.jar dbnsfp  -noDownload -noLog -db dbNSFP2.4_bgzip.vcf.gz  input.vcf > input.dbnsfp_anno.vcf
    sbg:contributors:
    - admin
    sbg:createdBy: admin
    sbg:createdOn: 1512132000
    sbg:id: admin/sbg-public-data/snpsift-dbnsfp-4-3/2
    sbg:image_url:
    sbg:job:
      allocatedResources:
        cpu: 1
        mem: 5000
      inputs:
        anno_fields:
        annotate_missing_entries: true
        colapse:
        database:
          secondaryFiles:
          - path: .tbi
          path: dbNSFP2.4_bgzip.vcf.gz
        dbnsfp_series: dbNSFP 2.x
        force_anno: false
        genome_version: genome_version-string-value
        input_vcf:
          class: File
          secondaryFiles: []
          path: input.vcf
          size: 0
        invert_fields_to_add: true
        mem_mb: 5000
    sbg:latestRevision: 2
    sbg:license: GNU Lesser General Public License v3.0 only
    sbg:links:
    - id: http://snpeff.sourceforge.net/index.html
      label: Homepage
    - id: https://github.com/pcingola/SnpEff
      label: Source Code
    - id: http://snpeff.sourceforge.net/SnpEff_manual.html
      label: Wiki
    - id: https://sourceforge.net/projects/snpeff/files/snpEff_v4_2_core.zip
      label: Download
    - id: http://snpeff.sourceforge.net/SnpEff_paper.pdf
      label: Publication
    sbg:modifiedBy: admin
    sbg:modifiedOn: 1530096338
    sbg:project: admin/sbg-public-data
    sbg:projectName: SBG Public data
    sbg:publisher: sbg
    sbg:revision: 2
    sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-dbnsfp-4-2/16
    sbg:revisionsInfo:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1512132000
      sbg:revision: 0
      sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-dbnsfp-4-2/7
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1512132000
      sbg:revision: 1
      sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-dbnsfp-4-2/14
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1530096338
      sbg:revision: 2
      sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-dbnsfp-4-2/16
    sbg:sbgMaintained: false
    sbg:toolAuthor: Pablo Cingolani/Broad Institue
    sbg:toolkit: SnpEff
    sbg:toolkitVersion: 4.3k
    sbg:validationErrors: []
    x: 1500.000203450522
    y: 401.6667175292971
  inputs:
  - id: '#SnpSift_dbNSFP.input_vcf'
    source: '#SnpEff.annotated'
  - id: '#SnpSift_dbNSFP.dbnsfp_series'
    default: dbNSFP 3.x
  - id: '#SnpSift_dbNSFP.database'
    source: '#database_1'
  - id: '#SnpSift_dbNSFP.anno_fields'
    default:
    - SIFT_score
    - SIFT_pred
    - Polyphen2_HDIV_score
    - MutationAssessor_score
    - phyloP100way_vertebrate
    - 1000Gp3_AF
    - 1000Gp3_AFR_AF
    - 1000Gp3_EUR_AF
    - 1000Gp3_AMR_AF
    - 1000Gp3_EAS_AF
    - ESP6500_AA_AF
    - ESP6500_EA_AF
    - ExAC_AC
    - ExAC_AF
  outputs:
  - id: '#SnpSift_dbNSFP.annotated_vcf'
  sbg:x: 1622.3565673828125
  sbg:y: 359.740234375
- id: '#SAMtools_Index_BAM_1'
  label: SAMtools Index BAM
  in: []
  run:
    cwlVersion: sbg:draft-2
    class: CommandLineTool
    label: SAMtools Index BAM
    $namespaces:
      sbg: https://sevenbridges.com

    requirements:
    - id: '#cwl-js-engine'
      class: ExpressionEngineRequirement
      requirements:
      - class: DockerRequirement
        dockerPull: rabix/js-engine
    - class: InlineJavascriptRequirement

    inputs:
    - id: '#input_index_file'
      label: Input indexed file
      type:
      - 'null'
      - File
      description: Input index file (BAI).
      required: false
      sbg:category: File input
      sbg:fileTypes: BAI
      sbg:stageInput: link
    - id: '#input_bam_file'
      label: BAM input file
      type:
      - File
      description: BAM input file.
      required: true
      sbg:fileTypes: BAM
      sbg:stageInput: link

    outputs:
    - id: '#output_bam_file'
      label: Output BAM file
      type:
      - File
      outputBinding:
        secondaryFiles:
        - .bai
        glob: '*.bam'
        sbg:inheritMetadataFrom: '#input_bam_file'
      description: Output BAM file with index (BAI) file.
      sbg:fileTypes: BAM
    - id: '#generated_index'
      label: Generated index file
      type:
      - 'null'
      - File
      outputBinding:
        glob: '*.bai'
        sbg:inheritMetadataFrom: '#input_bam_file'
      description: Generated index file (without the indexed data).
      sbg:fileTypes: BAI

    baseCommand:
    - class: Expression
      engine: '#cwl-js-engine'
      script: |
        {
          if ($job.inputs.input_index_file)
          {
         	return"echo Skipping index step because BAI file is provided on the input."
          }
          else
          {
            arr = [].concat($job.inputs.input_bam_file)
            return "/opt/samtools-0.1.19/samtools index " + arr[0].path
          }
        }

    hints:
    - class: sbg:AWSInstanceType
      value: c5.2xlarge;ebs-gp2;300
    - class: sbg:CPURequirement
      value: 1
    - class: sbg:MemRequirement
      value: 1000
    - class: DockerRequirement
      dockerPull: images.sbgenomics.com/markop/samtools:0.1.19
      dockerImageId: 2fb927277493
    id: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow/samtools-index-bam-1/1
    description: |-
      SAMtools Index BAM indexes sorted alignments for fast random access. Index file <aln.bam>.bai is created.
    sbg:appVersion:
    - sbg:draft-2
    sbg:categories:
    - SAM/BAM-Processing
    - Indexing
    sbg:cmdPreview: /opt/samtools-0.1.19/samtools index input.bam
    sbg:contributors:
    - jelena_randjelovic
    sbg:createdBy: jelena_randjelovic
    sbg:createdOn: 1536672693
    sbg:id: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow/samtools-index-bam-1/1
    sbg:image_url:
    sbg:job:
      allocatedResources:
        cpu: 1
        mem: 1000
      inputs:
        input_bam_file:
          class: File
          secondaryFiles: []
          path: input.bam
          size: 0
    sbg:latestRevision: 1
    sbg:license: The MIT License
    sbg:links:
    - id: http://samtools.sourceforge.net/
      label: Homepage
    - id: https://github.com/samtools/samtools
      label: Source Code
    - id: http://sourceforge.net/p/samtools/wiki/Home/
      label: Wiki
    - id: http://sourceforge.net/projects/samtools/files/
      label: Download
    - id: http://www.ncbi.nlm.nih.gov/pubmed/19505943
      label: Publication
    - id: http://www.htslib.org/doc/samtools.html
      label: Documentation
    sbg:modifiedBy: jelena_randjelovic
    sbg:modifiedOn: 1537692975
    sbg:project: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow
    sbg:projectName: PDXnet - Polishing WES JAX workflow
    sbg:publisher: sbg
    sbg:revision: 1
    sbg:revisionNotes: 300GB storage
    sbg:revisionsInfo:
    - sbg:modifiedBy: jelena_randjelovic
      sbg:modifiedOn: 1536672693
      sbg:revision: 0
      sbg:revisionNotes: Added concat guard in base command.
    - sbg:modifiedBy: jelena_randjelovic
      sbg:modifiedOn: 1537692975
      sbg:revision: 1
      sbg:revisionNotes: 300GB storage
    sbg:sbgMaintained: false
    sbg:toolAuthor: Heng Li, Sanger Institute
    sbg:toolkit: SAMtools
    sbg:toolkitVersion: v0.1.19
    sbg:validationErrors: []
    x: 730.0004069010421
    y: 11.317418416341152
  inputs:
  - id: '#SAMtools_Index_BAM_1.input_bam_file'
    source: '#tumor_alignment_and_target_coverage_workflow.output'
  outputs:
  - id: '#SAMtools_Index_BAM_1.output_bam_file'
  - id: '#SAMtools_Index_BAM_1.generated_index'
  sbg:x: 730.0004069010421
  sbg:y: 11.317418416341152
- id: '#vcfEffOnePerLine_pl'
  label: vcfEffOnePerLine.pl
  in: []
  run:
    cwlVersion: sbg:draft-2
    class: CommandLineTool
    label: vcfEffOnePerLine.pl
    $namespaces:
      sbg: https://sevenbridges.com

    requirements:
    - id: '#cwl-js-engine'
      class: ExpressionEngineRequirement
      requirements:
      - class: DockerRequirement
        dockerPull: rabix/js-engine
    - class: InlineJavascriptRequirement

    inputs:
    - id: '#input_vcf_file'
      label: Input VCF file
      type:
      - File
      inputBinding:
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Input VCF file.
      required: true
      sbg:fileTypes: VCF

    outputs:
    - id: '#one_effect_vcf'
      label: VCF with one effect per line
      type:
      - 'null'
      - File
      outputBinding:
        glob: '*oneEff.vcf'
        sbg:inheritMetadataFrom: '#input_vcf_file'
      description: VCF with one effect per line.
      sbg:fileTypes: VCF

    baseCommand:
    - cat
    arguments:
    - position: 100
      valueFrom:
        class: Expression
        engine: '#cwl-js-engine'
        script: |-
          {
            return ' \| /opt/snpEff/scripts/vcfEffOnePerLine.pl > '.concat($job.inputs.input_vcf_file.path.split('/').pop().split('.vcf')[0], 'oneEff.vcf')
          }
      separate: true

    hints:
    - class: sbg:AWSInstanceType
      value: c5.2xlarge;ebs-gp2;100
    - class: sbg:CPURequirement
      value: 1
    - class: sbg:MemRequirement
      value: 1000
    - class: DockerRequirement
      dockerPull: images.sbgenomics.com/jrandjelovic/snpeff:v4.3k
    id: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow/vcfeffoneperline-pl/2
    description: |-
      A simple wrapper for the snpEff utility script vcfEffOnePerLine.pl, which transforms a SnpEff-annotated input VCF, yielding a file with one effect per line.
    sbg:appVersion:
    - sbg:draft-2
    sbg:categories:
    - VCF-Processing
    sbg:cmdPreview: |-
      cat  /path/to/input_vcf_file.ext   | /opt/snpEff/scripts/vcfEffOnePerLine.pl > input_vcf_file.extoneEff.vcf
    sbg:contributors:
    - jelena_randjelovic
    sbg:createdBy: jelena_randjelovic
    sbg:createdOn: 1537542933
    sbg:id: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow/vcfeffoneperline-pl/2
    sbg:image_url:
    sbg:job:
      allocatedResources:
        cpu: 1
        mem: 1000
      inputs:
        input_vcf_file:
          class: File
          secondaryFiles: []
          path: /path/to/input_vcf_file.ext
          size: 0
    sbg:latestRevision: 2
    sbg:license: GNU Lesser General Public License v3.0 only
    sbg:links:
    - id: http://snpeff.sourceforge.net/SnpSift.html#Extract
      label: Documentation
    - id: https://github.com/pcingola/SnpEff
      label: Source Code
    sbg:modifiedBy: jelena_randjelovic
    sbg:modifiedOn: 1537691476
    sbg:project: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow
    sbg:projectName: PDXnet - Polishing WES JAX workflow
    sbg:publisher: sbg
    sbg:revision: 2
    sbg:revisionNotes: 100 GB storage
    sbg:revisionsInfo:
    - sbg:modifiedBy: jelena_randjelovic
      sbg:modifiedOn: 1537542933
      sbg:revision: 0
      sbg:revisionNotes:
    - sbg:modifiedBy: jelena_randjelovic
      sbg:modifiedOn: 1537543479
      sbg:revision: 1
      sbg:revisionNotes: initial setup
    - sbg:modifiedBy: jelena_randjelovic
      sbg:modifiedOn: 1537691476
      sbg:revision: 2
      sbg:revisionNotes: 100 GB storage
    sbg:sbgMaintained: false
    sbg:toolAuthor: Pablo Cingolani/Broad Institute
    sbg:toolkit: SnpEff
    sbg:toolkitVersion: 4.3k
    sbg:validationErrors: []
    x: 1708.3335367838552
    y: 365.0000508626304
  inputs:
  - id: '#vcfEffOnePerLine_pl.input_vcf_file'
    source: '#vcftools_sort_0_1_14.output_file'
  outputs:
  - id: '#vcfEffOnePerLine_pl.one_effect_vcf'
  sbg:x: 2002.38037109375
  sbg:y: 230.01698303222656
- id: '#SnpSift_extractFields_4_3k'
  label: SnpSift extractFields 4.3k
  in: []
  run:
    cwlVersion: sbg:draft-2
    class: CommandLineTool
    label: SnpSift extractFields 4.3k
    $namespaces:
      sbg: https://sevenbridges.com

    requirements:
    - id: '#cwl-js-engine'
      class: ExpressionEngineRequirement
      requirements:
      - class: DockerRequirement
        dockerPull: rabix/js-engine
    - class: InlineJavascriptRequirement

    inputs:
    - id: '#output_file_name'
      label: Output file name
      type:
      - 'null'
      - string
      description: Output file name.
      sbg:category: Options
    - id: '#multiple_fields_separator_output'
      label: Multiple fields separator for the output file
      type:
      - 'null'
      - string
      inputBinding:
        prefix: -s
        position: 1
        valueFrom:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              if ($job.inputs.multiple_fields_separator_output)
              {
                return '"'.concat($job.inputs.multiple_fields_separator_output, '"')
              }
            }
        separate: true
        sbg:cmdInclude: true
      description: Multiple fields separator for the output file.
      sbg:category: Options
      sbg:toolDefaultValue: "'\\t'"
    - id: '#mem_mb'
      label: Memory to use for the task [MB]
      type:
      - 'null'
      - int
      description: Memory to use for the task [MB].
      sbg:category: Options
      sbg:stageInput:
      sbg:toolDefaultValue: '2048'
    - id: '#input_vcf'
      label: Input VCF
      type:
      - File
      inputBinding:
        position: 10
        separate: true
        sbg:cmdInclude: true
      description: Input VCF file.
      required: true
      sbg:category: Inputs
      sbg:fileTypes: VCF, VCF.GZ
    - id: '#fields_to_extract'
      label: VCF fields to extract
      type:
      - type: array
        items: string
      inputBinding:
        position: 20
        valueFrom:
          class: Expression
          engine: '#cwl-js-engine'
          script: |+
            {
              tempout = ''
              for (k = 0, len = $job.inputs.fields_to_extract.length; k < len; k++)
              {
                tempout = tempout.concat('"', $job.inputs.fields_to_extract[k], '" ')
              }
              return tempout
            }

        separate: true
        itemSeparator: ' '
        sbg:cmdInclude: true
      description: VCF fields to extract.
      sbg:category: Inputs
    - id: '#empty_fields_formatting'
      label: Empty fields formatting
      type:
      - 'null'
      - string
      inputBinding:
        prefix: -e
        position: 1
        valueFrom:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              if ($job.inputs.empty_fields_formatting)
              {
                return '"'.concat($job.inputs.empty_fields_formatting, '"')
              }
            }
        separate: true
        sbg:cmdInclude: true
      description: Empty fields formatting (-e).
      sbg:category: Options
      sbg:toolDefaultValue: "''"

    outputs:
    - id: '#extracted_fields_file'
      label: Extracted fields output file
      type:
      - File
      outputBinding:
        glob: '*_extracted_fields.tab'
        sbg:inheritMetadataFrom: '#input_vcf'
      description: Extracted fields output file.
      sbg:fileTypes: TAB

    baseCommand:
    - java
    - class: Expression
      engine: '#cwl-js-engine'
      script: |-
        {
          if ($job.inputs.mem_mb)
          {
            return "-Xmx" + ($job.inputs.mem_mb).toString() + "m"
          }
          else
          {
            return "-Xmx2048m"
          }
        }
    - -jar
    - /opt/snpEff/SnpSift.jar
    - extractFields
    - -noDownload
    - -noLog
    arguments:
    - position: 100
      valueFrom:
        class: Expression
        engine: '#cwl-js-engine'
        script: |-
          {
            if ($job.inputs.output_file_name)
            {
              return " > ".concat($job.inputs.output_file_name, "_snpsift_extracted_fields.tab")
            }
            else
            {
              return " > ".concat($job.inputs.input_vcf.path.split("/").pop().split(".vcf")[0], "_snpsift_extracted_fields.tab")
            }
          }
      separate: true

    hints:
    - class: sbg:AWSInstanceType
      value: c5.2xlarge;ebs-gp2;100
    - class: sbg:CPURequirement
      value: 1
    - class: sbg:MemRequirement
      value:
        class: Expression
        engine: '#cwl-js-engine'
        script: |-
          {
            if ($job.inputs.mem_mb)
            {
              return $job.inputs.mem_mb
            }
            
            else
            {
              return 2048
            }
          }
    - class: DockerRequirement
      dockerPull: images.sbgenomics.com/jrandjelovic/snpeff:v4.3k
    id: |-
      jelena_randjelovic/pdxnet-polishing-wes-jax-workflow/snpsift-extract-fields-4-3-k/3
    description: |-
      **SnpSift extractFields** tool extracts fields from a VCF file to a TXT, tab separated format [1].

      ### Common Use Cases

      Extracting a subset of VCF fields of interest for further analysis from a SnpEff/SnpSift annotated VCF file.

      ### Changes Introduced by Seven Bridges

      No significant changes were introduced.

      ### Common Issues and Important Notes

      * Inputs **Input VCF** and **VCF fields to extract** are required.

      * Please note that fields to be extracted should be provided in a specific format, for example ANN[*].HGVS_P (see [here](http://snpeff.sourceforge.net/SnpSift.html#Extract) for details).

      ### Performance Benchmarking

      Typical runs take <5 minutes and cost <$0.05.

      ### References

      [1] [SnpSift extractFields documentation](http://snpeff.sourceforge.net/SnpSift.html#Extract)
    sbg:appVersion:
    - sbg:draft-2
    sbg:categories:
    - VCF-Processing
    - Annotation
    sbg:cmdPreview: |-
      java -Xmx5m -jar /opt/snpEff/SnpSift.jar extractFields -noDownload -noLog  /path/to/input_vcf.ext  "fields_to_extract-string-value-1" "fields_to_extract-string-value-2"    > output_file_name_string_snpsift_extracted_fields.tab
    sbg:contributors:
    - jelena_randjelovic
    sbg:createdBy: jelena_randjelovic
    sbg:createdOn: 1537540800
    sbg:id: |-
      jelena_randjelovic/pdxnet-polishing-wes-jax-workflow/snpsift-extract-fields-4-3-k/3
    sbg:image_url:
    sbg:job:
      allocatedResources:
        cpu: 1
        mem: 5
      inputs:
        empty_fields_formatting: ''
        fields_to_extract:
        - fields_to_extract-string-value-1
        - fields_to_extract-string-value-2
        input_vcf:
          class: File
          secondaryFiles: []
          path: /path/to/input_vcf.ext
          size: 0
        mem_mb: 5
        multiple_fields_separator_output: ''
        output_file_name: output_file_name_string
    sbg:latestRevision: 3
    sbg:license: GNU Lesser General Public License v3.0 only
    sbg:links:
    - id: http://snpeff.sourceforge.net/SnpSift.html
      label: Homepage
    - id: https://github.com/pcingola/SnpEff
      label: Source Code
    - id: http://snpeff.sourceforge.net/SnpSift.html
      label: Documentation
    - id: https://sourceforge.net/projects/snpeff/files/
      label: Download
    sbg:modifiedBy: jelena_randjelovic
    sbg:modifiedOn: 1537695029
    sbg:project: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow
    sbg:projectName: PDXnet - Polishing WES JAX workflow
    sbg:publisher: sbg
    sbg:revision: 3
    sbg:revisionNotes: improved output naming
    sbg:revisionsInfo:
    - sbg:modifiedBy: jelena_randjelovic
      sbg:modifiedOn: 1537540800
      sbg:revision: 0
      sbg:revisionNotes:
    - sbg:modifiedBy: jelena_randjelovic
      sbg:modifiedOn: 1537540818
      sbg:revision: 1
      sbg:revisionNotes: import from SBPLA
    - sbg:modifiedBy: jelena_randjelovic
      sbg:modifiedOn: 1537691639
      sbg:revision: 2
      sbg:revisionNotes: 100 GB storage
    - sbg:modifiedBy: jelena_randjelovic
      sbg:modifiedOn: 1537695029
      sbg:revision: 3
      sbg:revisionNotes: improved output naming
    sbg:sbgMaintained: false
    sbg:toolAuthor: Pablo Cingolani/Broad Institute
    sbg:toolkit: SnpEff
    sbg:toolkitVersion: 4.3k
    sbg:validationErrors: []
    x: 1858.3336385091156
    y: 328.33340962727885
  inputs:
  - id: '#SnpSift_extractFields_4_3k.input_vcf'
    source: '#vcfEffOnePerLine_pl.one_effect_vcf'
  - id: '#SnpSift_extractFields_4_3k.fields_to_extract'
    default:
    - CHROM
    - POS
    - REF
    - ALT
    - ID
    - FILTER
    - DP
    - GEN[*].DP
    - GEN[*].AF
    - LOF[*].NUMTR
    - LOF[*].PERC
    - EFF[*].GENE
    - EFF[*].EFFECT
    - EFF[*].IMPACT
    - EFF[*].FUNCLASS
    - EFF[*].RANK
    - EFF[*].CODON
    - EFF[*].AA
    - EFF[*].AA_LEN
    - EFF[*].CODING
    - EFF[*].TRID
    - dbNSFP_SIFT_score
    - dbNSFP_SIFT_pred
    - dbNSFP_Polyphen2_HDIV_score
    - dbNSFP_MutationAssessor_score
    - dbNSFP_phyloP100way_vertebrate
    - dbNSFP_1000Gp3_AF
    - dbNSFP_1000Gp3_AFR_AF
    - dbNSFP_1000Gp3_EUR_AF
    - dbNSFP_1000Gp3_AMR_AF
    - dbNSFP_1000Gp3_EAS_AF
    - dbNSFP_ESP6500_AA_AF
    - dbNSFP_ESP6500_EA_AF
    - dbNSFP_ExAC_AC
    - dbNSFP_ExAC_AF
  outputs:
  - id: '#SnpSift_extractFields_4_3k.extracted_fields_file'
  sbg:x: 2125.92529296875
  sbg:y: 182.97622680664062
- id: '#FastQC'
  label: FastQC
  in: []
  run:
    cwlVersion: sbg:draft-2
    class: CommandLineTool
    label: FastQC
    $namespaces:
      sbg: https://sevenbridges.com

    requirements:
    - id: '#cwl-js-engine'
      class: ExpressionEngineRequirement
      requirements:
      - class: DockerRequirement
        dockerPull: rabix/js-engine
    - class: InlineJavascriptRequirement

    inputs:
    - id: '#threads'
      label: Threads
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --threads
        position: 0
        valueFrom:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
            //if "threads" is not specified
            //number of threads is determined based on number of inputs
              if (! $job.inputs.threads){
                $job.inputs.threads = [].concat($job.inputs.input_fastq).length
              }
              return Math.min($job.inputs.threads,7)
            }
        separate: true
        sbg:cmdInclude: true
      description: |-
        Specifies the number of files which can be processed simultaneously.  Each thread will be allocated 250MB of memory so you shouldn't run more threads than your available memory will cope with, and not more than 6 threads on a 32 bit machine.
      sbg:altPrefix: -t
      sbg:category: Options
      sbg:toolDefaultValue: '1'
    - id: '#quiet'
      label: Quiet
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --quiet
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Supress all progress messages on stdout and only report errors.
      sbg:altPrefix: -q
      sbg:category: Options
    - id: '#nogroup'
      label: Nogroup
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --nogroup
        position: 0
        separate: false
        sbg:cmdInclude: true
      description: |-
        Disable grouping of bases for reads >50bp. All reports will show data for every base in the read.  WARNING: Using this option will cause fastqc to crash and burn if you use it on really long reads, and your plots may end up a ridiculous size. You have been warned.
      sbg:category: Options
    - id: '#nano'
      label: Nano
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --nano
        position: 0
        separate: false
        sbg:cmdInclude: true
      description: |-
        Files come from naopore sequences and are in fast5 format. In this mode you can pass in directories to process and the program will take in all fast5 files within those directories and produce a single output file from the sequences found in all files.
      sbg:category: Options
    - id: '#memory_per_job'
      label: Amount of memory allocated per job execution.
      type:
      - 'null'
      - int
      description: Amount of memory allocated per execution of FastQC job.
      sbg:category: Execution parameters
      sbg:toolDefaultValue: Determined by the number of input files
    - id: '#limits_file'
      label: Limits
      type:
      - 'null'
      - File
      inputBinding:
        prefix: --limits
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Specifies a non-default file which contains a set of criteria which will be used to determine the warn/error limits for the various modules.  This file can also be used to selectively remove some modules from the output all together.  The format needs to mirror the default limits.txt file found in the Configuration folder.
      required: false
      sbg:altPrefix: -l
      sbg:category: File inputs
      sbg:fileTypes: TXT
    - id: '#kmers'
      label: Kmers
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --kmers
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Specifies the length of Kmer to look for in the Kmer content module. Specified Kmer length must be between 2 and 10. Default length is 7 if not specified.
      sbg:altPrefix: -f
      sbg:category: Options
      sbg:toolDefaultValue: '7'
    - id: '#input_fastq'
      label: Input file
      type:
      - type: array
        items: File
      inputBinding:
        position: 100
        separate: true
        sbg:cmdInclude: true
      description: Input file.
      required: true
      sbg:category: File inputs
      sbg:fileTypes: FASTQ, FQ, FASTQ.GZ, FQ.GZ, BAM, SAM
    - id: '#format'
      label: Format
      type:
      - 'null'
      - name: format
        type: enum
        symbols:
        - bam
        - sam
        - bam_mapped
        - sam_mapped
        - fastq
      inputBinding:
        prefix: --format
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Bypasses the normal sequence file format detection and forces the program to use the specified format.  Valid formats are BAM, SAM, BAM_mapped, SAM_mapped and FASTQ.
      sbg:altPrefix: -f
      sbg:category: Options
      sbg:toolDefaultValue: FASTQ
    - id: '#cpus_per_job'
      label: Number of CPUs.
      type:
      - 'null'
      - int
      description: Number of CPUs to be allocated per execution of FastQC.
      sbg:category: Execution parameters
      sbg:toolDefaultValue: Determined by the number of input files
    - id: '#contaminants_file'
      label: Contaminants
      type:
      - 'null'
      - File
      inputBinding:
        prefix: --contaminants
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Specifies a non-default file which contains the list of contaminants to screen overrepresented sequences against. The file must contain sets of named contaminants in the form name[tab]sequence.  Lines prefixed with a hash will be ignored.
      required: false
      sbg:altPrefix: -c
      sbg:category: File inputs
      sbg:fileTypes: TXT
    - id: '#casava'
      label: Casava
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --casava
        position: 0
        separate: false
        sbg:cmdInclude: true
      description: |-
        Files come from raw casava output. Files in the same sample group (differing only by the group number) will be analysed as a set rather than individually. Sequences with the filter flag set in the header will be excluded from the analysis. Files must have the same names given to them by casava (including being gzipped and ending with .gz) otherwise they won't be grouped together correctly.
      sbg:category: Options
    - id: '#adapters_file'
      label: Adapters
      type:
      - 'null'
      - File
      inputBinding:
        prefix: --adapters
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Specifies a non-default file which contains the list of adapter sequences which will be explicity searched against the library. The file must contain sets of named adapters in the form name[tab]sequence.  Lines prefixed with a hash will be ignored.
      required: false
      sbg:altPrefix: -a
      sbg:category: File inputs
      sbg:fileTypes: TXT

    outputs:
    - id: '#report_zip'
      label: Report zip
      type:
      - 'null'
      - type: array
        items: File
      outputBinding:
        glob: '*_fastqc.zip'
        sbg:inheritMetadataFrom: '#input_fastq'
        sbg:metadata:
          __inherit__: input_fastq
      description: Zip archive of the report.
      sbg:fileTypes: ZIP
    - id: '#report_html'
      label: Report HTMLs
      type:
      - 'null'
      - type: array
        items: File
      outputBinding:
        glob: '*.html'
        sbg:inheritMetadataFrom: '#input_fastq'
      description: FastQC reports in HTML format.
      sbg:fileTypes: HTML

    baseCommand:
    - fastqc
    arguments:
    - prefix: ''
      position: 0
      valueFrom: --noextract
      separate: true
    - prefix: --outdir
      position: 0
      valueFrom: .
      separate: true

    hints:
    - class: sbg:CPURequirement
      value:
        class: Expression
        engine: '#cwl-js-engine'
        script: |-
          {
            // if cpus_per_job is set, it takes precedence
            if ($job.inputs.cpus_per_job) {
              return $job.inputs.cpus_per_job 
            }
            // if threads parameter is set, the number of CPUs is set based on that parametere
            else if ($job.inputs.threads) {
              return $job.inputs.threads
            }
            // else the number of CPUs is determined by the number of input files, up to 7 -- default
            else return Math.min([].concat($job.inputs.input_fastq).length,7)
          }
    - class: sbg:MemRequirement
      value:
        class: Expression
        engine: '#cwl-js-engine'
        script: |+
          {
            // if memory_per_job is set, it takes precedence
            if ($job.inputs.memory_per_job){
              return $job.inputs.memory_per_job
            }
            // if threads parameter is set, memory req is set based on the number of threads
            else if ($job.inputs.threads){
              return 1024 + 300*$job.inputs.threads
            }
            // else the memory req is determined by the number of input files, up to 7 -- default
            else return (1024 + 300*Math.min([].concat($job.inputs.input_fastq).length,7))
          }

    - class: DockerRequirement
      dockerPull: images.sbgenomics.com/mladenlsbg/fastqc:0.11.4
      dockerImageId: 759c4c8fbafd
    id: admin/sbg-public-data/fastqc-0-11-4/9
    description: |-
      FastQC reads a set of sequence files and produces a quality control (QC) report from each one. These reports consist of a number of different modules, each of which will help identify a different type of potential problem in your data. 

      Since it's necessary to convert the tool report in order to show them on Seven Bridges platform, it's recommended to use [FastQC Analysis workflow instead](https://igor.sbgenomics.com/public/apps#admin/sbg-public-data/fastqc-analysis/). 

      FastQC is a tool which takes a FASTQ file and runs a series of tests on it to generate a comprehensive QC report.  This report will tell you if there is anything unusual about your sequence.  Each test is flagged as a pass, warning, or fail depending on how far it departs from what you would expect from a normal large dataset with no significant biases.  It is important to stress that warnings or even failures do not necessarily mean that there is a problem with your data, only that it is unusual.  It is possible that the biological nature of your sample means that you would expect this particular bias in your results.
    sbg:appVersion:
    - sbg:draft-2
    sbg:categories:
    - FASTQ-Processing
    - Quality-Control
    - Quantification
    sbg:cmdPreview: |-
      fastqc  --noextract --outdir .  /path/to/input_fastq-1.fastq  /path/to/input_fastq-2.fastq
    sbg:contributors:
    - admin
    - mladenlSBG
    - djordje_klisic
    sbg:createdBy: mladenlSBG
    sbg:createdOn: 1447773725
    sbg:id: admin/sbg-public-data/fastqc-0-11-4/9
    sbg:image_url:
    sbg:job:
      allocatedResources:
        cpu: 2
        mem: 1624
      inputs:
        format:
        cpus_per_job:
        input_fastq:
        - class: File
          secondaryFiles: []
          path: /path/to/input_fastq-1.fastq
          size: 0
        - class: File
          secondaryFiles: []
          path: /path/to/input_fastq-2.fastq
          size: 0
        memory_per_job:
        quiet: true
        threads:
    sbg:latestRevision: 9
    sbg:license: GNU General Public License v3.0 only
    sbg:links:
    - id: http://www.bioinformatics.babraham.ac.uk/projects/fastqc/
      label: Homepage
    - id: |-
        http://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.4_source.zip
      label: Source Code
    - id: https://wiki.hpcc.msu.edu/display/Bioinfo/FastQC+Tutorial
      label: Wiki
    - id: http://www.bioinformatics.babraham.ac.uk/projects/fastqc/fastqc_v0.11.4.zip
      label: Download
    - id: http://www.bioinformatics.babraham.ac.uk/projects/fastqc
      label: Publication
    sbg:modifiedBy: admin
    sbg:modifiedOn: 1529402637
    sbg:project: admin/sbg-public-data
    sbg:projectName: SBG Public data
    sbg:publisher: sbg
    sbg:revision: 9
    sbg:revisionNotes: HTML output added.
    sbg:revisionsInfo:
    - sbg:modifiedBy: mladenlSBG
      sbg:modifiedOn: 1447773725
      sbg:revision: 0
      sbg:revisionNotes:
    - sbg:modifiedBy: djordje_klisic
      sbg:modifiedOn: 1459957440
      sbg:revision: 1
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1471861482
      sbg:revision: 2
      sbg:revisionNotes:
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1476440178
      sbg:revision: 3
      sbg:revisionNotes: Input categories added.
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1476440178
      sbg:revision: 4
      sbg:revisionNotes: |-
        FASTQ input changed from single file to array. Added better thread handling. 

        IMPORTANT NOTICE: If updating this tool in existing workflow, it's necessary to REMOVE SCATTER (uncheck it) from input_fastq or it might break the pipeline.
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1476440178
      sbg:revision: 5
      sbg:revisionNotes: |-
        FASTQ input changed from single file to array. Added better thread handling.

        IMPORTANT NOTICE: If updating this tool in existing workflow, it's necessary to REMOVE SCATTER (uncheck it) from input_fastq or it might break the pipeline.
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1489067520
      sbg:revision: 6
      sbg:revisionNotes: |-
        IMPORTANT NOTICE: If updating this tool in existing workflow, it's necessary to REMOVE SCATTER (uncheck it) from input_fastq or it might break the pipeline."

        Added automatised handling of BAM and SAM files. Also, added security measures for better automated threading handling.
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1489067520
      sbg:revision: 7
      sbg:revisionNotes: |-
        Changed the file types of limits, adapters and contaminants files to be TXT, they have to be in format name[tab]sequence. Format should be similar to the one in the Configuration folder provided with FastQC, txt files.

        "IMPORTANT NOTICE: If updating this tool in existing workflow, it's necessary to REMOVE SCATTER (uncheck it) from input_fastq or it might break the pipeline."
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1493298709
      sbg:revision: 8
      sbg:revisionNotes: |-
        * Fixed the JS expression for the CPU and Memory allocation
        * Added cpus_per_job and memory_per_job parameters
        * Removed default version for format, so the tool can handle combinations of file formats
    - sbg:modifiedBy: admin
      sbg:modifiedOn: 1529402637
      sbg:revision: 9
      sbg:revisionNotes: HTML output added.
    sbg:sbgMaintained: false
    sbg:toolAuthor: Babraham Institute
    sbg:toolkit: FastQC
    sbg:toolkitVersion: 0.11.4
    sbg:validationErrors: []
    x: 193.33333333333346
    y: 308.3333333333335
  inputs:
  - id: '#FastQC.input_fastq'
    source:
    - '#input_pair'
  outputs:
  - id: '#FastQC.report_zip'
  - id: '#FastQC.report_html'
  sbg:x: 193.33333333333346
  sbg:y: 308.3333333333335
- id: '#snpsift_split_4_3k'
  label: SnpSift Split 4.3k
  in: []
  run:
    cwlVersion: sbg:draft-2
    class: CommandLineTool
    label: SnpSift Split 4.3k
    $namespaces:
      sbg: https://sevenbridges.com

    requirements:
    - id: '#cwl-js-engine'
      class: ExpressionEngineRequirement
      requirements:
      - class: DockerRequirement
        dockerPull: rabix/js-engine
    - class: InlineJavascriptRequirement

    inputs:
    - id: '#num_lines_split'
      label: Split by <num> lines
      type:
      - 'null'
      - int
      inputBinding:
        prefix: -l
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Split by 'num' lines.
      sbg:stageInput:
    - id: '#mem_mb'
      label: Memory to use for the task [MB]
      type:
      - 'null'
      - int
      description: Memory to use for the task [MB].
      sbg:stageInput:
      sbg:toolDefaultValue: '8192'
    - id: '#join_flag'
      label: Join input VCF files
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -j
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Join input VCF files.
    - id: '#input_vcfs'
      label: VCF files to split or join
      type:
      - type: array
        items: File
      inputBinding:
        position: 5
        separate: true
        sbg:cmdInclude: true
      description: VCF files to split or join together
      required: true
      sbg:category: Input
      sbg:fileTypes: VCF, VCF.GZ
      sbg:stageInput:

    outputs:
    - id: '#output_vcfs'
      label: Split VCFs
      type:
      - 'null'
      - type: array
        items: File
      outputBinding:
        glob:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
                tempout = ''
                primename = $job.inputs.input_vcfs[0].path.split('/').pop()
                primenamebase = primename.split('.')[0]
                if ($job.inputs.input_vcfs.length > 1)
                {
                  tempout = '{'
                  for (i=0; i < $job.inputs.input_vcfs.length; i++)
                  {
                    filename = $job.inputs.input_vcfs[i].path.split('/').pop()
                    filenamebase = filename.split('_somatic.filtered.vcf')[0]
                    tempout = tempout.concat(filenamebase, '.*.vcf')
                    if (i < $job.inputs.input_vcfs.length - 1)
                    {
                      tempout = tempout.concat(',')
                    }
                  }
                  tempout = tempout.concat('}')
                }
                else
                {
                  tempout = tempout.concat(primenamebase, '.*.vcf')
                }
                return tempout
            }
        sbg:inheritMetadataFrom: '#input_vcfs'
      description: Split VCFs.
      sbg:fileTypes: VCF
    - id: '#joined_vcf'
      label: Joined VCF file
      type:
      - 'null'
      - File
      outputBinding:
        glob: '*joined.vcf'
        sbg:inheritMetadataFrom: '#input_vcfs'
      description: Joined VCF file.
      sbg:fileTypes: VCF
    stdout:
      class: Expression
      engine: '#cwl-js-engine'
      script: |-
        {
          if ($job.inputs.join_flag)
          {
            tempout = ''
            filename = $job.inputs.input_vcfs[0].path.split('/').pop()
            tempout = tempout.concat(filename.split('.vcf')[0], '.joined')
            ///for (i=0; i < $job.inputs.input_vcfs.length; i++) {
            ///  filename = $job.inputs.input_vcfs[i].path.split('/').pop()
            //  tempout = tempout.concat(filename.split('.')[0], '_')
            ///}
            tempout = tempout.concat(".vcf").replace(/^.*[\\\/]/, '')
            return tempout
          }
        }

    baseCommand:
    - java
    - class: Expression
      engine: '#cwl-js-engine'
      script: |-
        {
          if ($job.inputs.mem_mb)
          {
            return "-Xmx" + ($job.inputs.mem_mb).toString() + "m"
          }
          else
          {
            return "-Xmx8192m"
          }
        }
    - -jar
    - /opt/snpEff/SnpSift.jar
    - split

    hints:
    - class: sbg:CPURequirement
      value: 1
    - class: sbg:MemRequirement
      value: 1000
    - class: DockerRequirement
      dockerPull: images.sbgenomics.com/jrandjelovic/snpeff:v4.3k
    id: michael_lloyd/mantis/snpsift-split-4-3k/3
    description: |-
      **SnpSift split** tool splits VCF files by chromosome (default) or specified number of lines [1].

      ### Common Use Cases
      Splitting a VCF file for post-processing in parallel.

      ### Changes Introduced by Seven Bridges

      No significant changes were introduced.

      ### Common Issues and Important Notes

      * Input **Input VCF files to split or join** is required.

      ### Performance Benchmarking

      Typical runs take <5 minutes and cost <$0.05.

      ### References

      [1] [SnpSift split documentation](http://snpeff.sourceforge.net/SnpSift.html#Split)
    sbg:appVersion:
    - sbg:draft-2
    sbg:categories:
    - VCF-Processing
    sbg:cmdPreview: java -Xmx1m -jar /opt/snpEff/SnpSift.jar split  /path/to/input_vcf-1.ext
    sbg:content_hash: a776f6e7e849670dd88293d18c3dadcdfe115b36ab05be43bfcd8fe85bd6bf267
    sbg:contributors:
    - michael_lloyd
    sbg:createdBy: michael_lloyd
    sbg:createdOn: 1566395378
    sbg:id: michael_lloyd/mantis/snpsift-split-4-3k/3
    sbg:image_url:
    sbg:job:
      allocatedResources:
        cpu: 1
        mem: 1000
      inputs:
        input_vcfs:
        - class: File
          secondaryFiles: []
          path: /path/to/input_vcf-1.ext
          size: 0
        join_flag: false
        mem_mb: 1
        num_lines_split:
    sbg:latestRevision: 3
    sbg:license: GNU Lesser General Public License v3.0 only
    sbg:links:
    - id: http://snpeff.sourceforge.net/SnpSift.html
      label: Homepage
    - id: https://github.com/pcingola/SnpEff
      label: Source Code
    - id: http://snpeff.sourceforge.net/SnpSift.html
      label: Documentation
    - id: https://sourceforge.net/projects/snpeff/files/
      label: Download
    sbg:modifiedBy: michael_lloyd
    sbg:modifiedOn: 1566396199
    sbg:project: michael_lloyd/mantis
    sbg:projectName: MSI, HRD, TMB
    sbg:publisher: sbg
    sbg:revision: 3
    sbg:revisionNotes: ''
    sbg:revisionsInfo:
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1566395378
      sbg:revision: 0
      sbg:revisionNotes: Copy of michael_lloyd/variant-calling-decomposition/snpsift-split-4-3k/0
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1566396089
      sbg:revision: 1
      sbg:revisionNotes: updated output
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1566396106
      sbg:revision: 2
      sbg:revisionNotes: ''
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1566396199
      sbg:revision: 3
      sbg:revisionNotes: ''
    sbg:sbgMaintained: false
    sbg:toolAuthor: Pablo Cingolani/Broad Institute
    sbg:toolkit: SnpEff
    sbg:toolkitVersion: 4.3k
    sbg:validationErrors: []
  inputs:
  - id: '#snpsift_split_4_3k.join_flag'
    default: true
  - id: '#snpsift_split_4_3k.input_vcfs'
    source:
    - '#gatk_filtermutectcalls_4051.output'
  outputs:
  - id: '#snpsift_split_4_3k.output_vcfs'
  - id: '#snpsift_split_4_3k.joined_vcf'
  sbg:x: 1274.4981689453125
  sbg:y: 157.86251831054688
- id: '#tumor_alignment_and_target_coverage_workflow'
  label: tumor_alignment_and_target_coverage_workflow
  in: []
  run:
    cwlVersion: sbg:draft-2
    class: Workflow
    label: tumor_alignment_and_target_coverage_workflow
    $namespaces:
      sbg: https://sevenbridges.com

    requirements:
    - class: SubworkflowFeatureRequirement
    - class: InlineJavascriptRequirement

    inputs:
    - id: '#target_intervals'
      label: target_intervals_file_for_hs_metrics
      type:
      - File
      required: true
      sbg:fileTypes: TXT
      sbg:x: 833.9055175781252
      sbg:y: 494.4130045572918
    - id: '#reference'
      label: reference
      type:
      - File
      secondaryFiles:
      - .fai
      - ^.dict
      required: true
      sbg:fileTypes: FASTA, FA
      sbg:x: 626.1282552083335
      sbg:y: 292.97436523437506
    - id: '#known_sites'
      label: known_sites_for_GATK_BQSR
      type:
      - type: array
        items: File
      secondaryFiles:
      - .tbi
      required: true
      sbg:fileTypes: TXT, BED, VCF
      sbg:x: 651.6383463541669
      sbg:y: 152.57226562500006
    - id: '#index_file_1'
      label: XenomeWholeGenomeIndices
      type:
      - File
      required: true
      sbg:x: 183.07692464192715
      sbg:y: -45.84611002604167
    - id: '#index_file'
      label: bwa_ref_files
      type:
      - File
      required: true
      sbg:x: 370.7293395996094
      sbg:y: -197.05963134765625
    - id: '#bed'
      label: target_bed
      type:
      - File
      required: true
      sbg:fileTypes: TXT, BED
      sbg:includeInPorts: true
      sbg:x: 983.2644856770836
      sbg:y: 311.63033040364587
    - id: '#reads'
      label: reads
      type:
      - type: array
        items: File
      sbg:fileTypes: |-
        FASTA, FASTQ, FA, FQ, FA.GZ, FA.BZ2, FQ.GZ, FQ.BZ2, FASTA.GZ, FASTA.BZ2, FASTQ.GZ, FASTQ.BZ2
      sbg:x: -131.999989827474
      sbg:y: 92.00000000000003
    - id: '#memory_per_job'
      label: Memory per job
      type:
      - 'null'
      - int
      description: |-
        Amount of RAM memory to be used per job. Defaults to 2048 MB for single threaded jobs.
      sbg:toolDefaultValue: '2048'
    - id: '#memory_per_job_1'
      label: Memory per job
      type:
      - 'null'
      - int
      description: |-
        Amount of RAM memory to be used per job. Defaults to 2048MB for single threaded jobs.
      sbg:category: Execution options
      sbg:toolDefaultValue: '2048'
    - id: '#memory_per_job_2'
      label: Memory per job
      type:
      - 'null'
      - int
      description: |-
        Amount of RAM memory to be used per job. Defaults to 2048MB for single threaded jobs.
      required: false
      sbg:category: Execution options
      sbg:toolDefaultValue: '2048'
    - id: '#intervals_string'
      label: Genomic interval strings to include
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --intervals
        valueFrom:
          class: Expression
          engine: '#cwl-js-engine'
          script: |
            {
                if ($job.inputs.reference && $job.inputs.intervals_string){
                    fasta = $job.inputs.reference.contents
                    interval = $job.inputs.intervals_string
                    // fasta - UCSC format
                    if (fasta.indexOf(">chr") != -1){
                        // INTERVAL - UCSC format, all w/ chr
                        if (interval.indexOf("chr") != -1){
                            return interval
                        }// if (interval.indexOf("chr") == -1)
                        // interval - 1000G format, convert to UCSC
                        else{
                            if (Number(interval) || interval == "X" || interval == "Y"){
                                return "chr".concat(interval)
                            }// if (Number(interval)|| interval == "X" || interval == "Y")
                            else{
                                if (interval == "MT"){
                                    return "chrM"
                                }//if (interval == "MT")
                                else{
                                    if(Number(interval.substr(2,6))<211){
                                        n = Number(interval.substr(2,6)) - 190;
                                        s = "chr";
                                        s = s.concat(n.toString());
                                        s = s.concat("_gl");
                                        s = s.concat(interval.substr(2,6));
                                        s = s.concat("_random");
                                        return s
                                    }// if(Number(interval.substr(2,interval.length-2))<211)
                                    else{
                                        if(Number(interval.substr(2,6))<250) {
                                            s = "chrUn_gl";
                                            s = s.concat(interval.substr(2,6));
                                            return s
                                        }//if(Number(interval.substr(2,interval.length-2))<250)
                                        else {
                                            return interval
                                        }//if(Number(interval.substr(2,interval.length-2))<250) - else
                                    }// if(Number(interval.substr(2,interval.length-2))<211) - else
                                }//if (interval == "MT") - else
                            }//if (Number(interval)|| interval == "X" || interval == "Y") - else
                        }//if (interval.indexOf("chr") == -1) - else
                    }//if (fasta.indexOf(">chr") == -1)
                    // fasta - 1000G format
                    else{
                        //interval - USCS format, all w/ chr, convert to 1000G
                        if(interval.indexOf("chr") != -1){
                            if(Number(interval.substr(3,2)) && interval.length<6 || interval == "chrX" || interval == "chrY"){
                                return interval.substr(3,2)
                            }//if(Number(interval.substr(3,interval.length)) != NaN || interval == "chrX" || bsqr == "chrY")
                            else{
                                if (interval == "chrM") {
                                    return "MT"
                                }//if (interval == "chrM")
                                else{
                                    s = "GL";
                                    s = s.concat(interval.substr(8,6));
                                    s = s.concat(".1");
                                    return s
                                }//if (interval == "chrM") - else
                            }//if(Number(interval.substr(3,interval.length)) != NaN || interval == "chrX" || bsqr == "chrY") - else
                        }//if($job.inputs.interval.indexOf("chr") == -1)
                        // interval - 1000G format
                        else{
                            return interval
                        }//(interval.indexOf("chr") == -1) - else
                    }//if (fasta.indexOf(">chr") == -1) - else
                }//if ($job.inputs.fasta && $job.inputs.interval)
            }
        separate: true
        sbg:cmdInclude: true
      description: |-
        One or more genomic intervals over which to operate  This argument may be specified 0 or                              more times. Default value: null.
      sbg:altPrefix: -L
      sbg:category: Optional Arguments
      sbg:suggestedValue: chr20
    - id: '#trailing'
      label: Trailing quality threshold
      type:
      - 'null'
      - int
      inputBinding:
        position: 7
        valueFrom:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {if(typeof $job.inputs.trailing  == 'undefined' ||  $job.inputs.trailing ==null){
              	return 'TRAILING:3';
            }
             else{
               return 'TRAILING:' + $job.inputs.trailing .toString();
             }
            }
        separate: true
        sbg:cmdInclude: true
      description: |-
        As long as a base has a value below this threshold the base is removed and the next base.
      sbg:category: Processing steps
      sbg:toolDefaultValue: '3'
    - id: '#leading'
      label: Leading quality threshold
      type:
      - 'null'
      - int
      inputBinding:
        position: 6
        valueFrom:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
            	if(typeof $job.inputs.leading  == 'undefined' ||  $job.inputs.leading ==null){
              		return 'LEADING:3';
                }else{
              		return 'LEADING:' + $job.inputs.leading .toString();
                }
            }
        separate: true
        sbg:cmdInclude: true
      description: Remove low quality bases from the beginning.
      sbg:category: Processing steps
      sbg:toolDefaultValue: '3'
    - id: '#TargetCoverage_CutOff'
      type:
      - 'null'
      - float
    - id: '#threads'
      label: Number of threads
      type:
      - 'null'
      - int
      description: Number of threads.

    outputs:
    - id: '#summary'
      label: ExomeQCSummary
      type:
      - 'null'
      - File
      source:
      - '#Exome_Coverage_QC_1_0.summary'
      required: false
      sbg:includeInPorts: true
      sbg:x: 1387.7192382812505
      sbg:y: 319.00980631510424
    - id: '#output'
      label: output
      type:
      - 'null'
      - File
      source:
      - '#GATK_ApplyBQSR.output'
      required: false
      sbg:fileTypes: BAM, SAM, CRAM
      sbg:includeInPorts: true
      sbg:x: 1325.4736328125005
      sbg:y: 169.82454427083337
    - id: '#metrics_file'
      label: Picard MarkDuplicates metrics file
      type:
      - 'null'
      - File
      source:
      - '#Picard_MarkDuplicates.metrics_file'
      required: false
      sbg:fileTypes: METRICS
      sbg:includeInPorts: true
      sbg:x: 1432.0000000000005
      sbg:y: -176.0000050862631
    - id: '#hs_metrics'
      label: Picard CalculateHSMetrics Report
      type:
      - 'null'
      - File
      source:
      - '#Picard_CalculateHsMetrics.hs_metrics'
      required: false
      sbg:fileTypes: TXT
      sbg:includeInPorts: true
      sbg:x: 1401.3333333333337
      sbg:y: 502.6666666666668
    - id: '#recalibration_file'
      label: GATK BaseRecalibrator Report
      type:
      - File
      source:
      - '#GATK_BaseRecalibrator.recalibration_file'
      required: true
      sbg:fileTypes: GRP
      sbg:includeInPorts: true
      sbg:x: 1428.0000000000005
      sbg:y: 68.00000000000001
    - id: '#integrated_QC_report'
      type:
      - 'null'
      - File
      source:
      - '#qc_integrate_with_cut_off_final.integrated_QC_report'
      sbg:x: 1433.914306640625
      sbg:y: -57.130619049072266

    steps:
    - id: '#GATK_ApplyBQSR'
      label: GATK ApplyBQSR
      in: []
      run:
        cwlVersion: sbg:draft-2
        class: CommandLineTool
        label: GATK ApplyBQSR
        $namespaces:
          sbg: https://sevenbridges.com

        requirements:
        - id: '#cwl-js-engine'
          class: ExpressionEngineRequirement
          requirements:
          - class: DockerRequirement
            dockerPull: rabix/js-engine
        - class: InlineJavascriptRequirement

        inputs:
        - id: '#verbosity'
          label: Control verbosity of logging
          type:
          - 'null'
          - name: 'null'
            type: enum
            symbols:
            - ERROR
            - WARNING
            - INFO
            - DEBUG
          inputBinding:
            prefix: --verbosity
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Control verbosity of logging.  Default value: INFO. Possible values: {ERROR, WARNING,
                                          INFO, DEBUG}.
          sbg:altPrefix: -verbosity
          sbg:category: Optional Arguments
          sbg:toolDefaultValue: INFO
        - id: '#use_jdk_inflater'
          label: |-
            Whether to use the JdkInflater (as opposed to IntelInflater)  Default value: false
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --use_jdk_inflater
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Whether to use the JdkInflater (as opposed to IntelInflater)  Default value: false.
                                          Possible values: {true, false}.
          sbg:altPrefix: -jdk_inflater
          sbg:category: Optional Arguments
          sbg:stageInput:
          sbg:toolDefaultValue: 'false'
        - id: '#use_jdk_deflater'
          label: |-
            Whether to use the JdkDeflater (as opposed to IntelDeflater)  Default value: false
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --use_jdk_deflater
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Whether to use the JdkDeflater (as opposed to IntelDeflater)  Default value: false.
                                          Possible values: {true, false}.
          sbg:altPrefix: -jdk_deflater
          sbg:category: Optional Arguments
          sbg:stageInput:
          sbg:toolDefaultValue: 'false'
        - id: '#useOriginalQualities'
          label: 'Use the base quality scores from the OQ tag  Default value: false'
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --useOriginalQualities
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Use the base quality scores from the OQ tag  Default value: false. Possible values: {true,
                                          false}.
          sbg:altPrefix: -OQ
          sbg:category: Optional Arguments
          sbg:stageInput:
          sbg:toolDefaultValue: 'false'
        - id: '#static_quantized_quals'
          label: |-
            Use static quantized quality scores to a given number of levels (with -bqsr)  This
                                          argument may be specified 0 or more times
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --static_quantized_quals
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Use static quantized quality scores to a given number of levels (with -bqsr)  This
                                          argument may be specified 0 or more times. Default value: null.  Cannot be used in
                                          conjuction with argument(s) quantizationLevels (qq).
          sbg:altPrefix: -SQQ
          sbg:category: Advanced Arguments
          sbg:toolDefaultValue: 'null'
        - id: '#secondsBetweenProgressUpdates'
          label: |-
            Output traversal statistics every time this many seconds elapse  Default value: 10
          type:
          - 'null'
          - float
          inputBinding:
            prefix: --secondsBetweenProgressUpdates
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Output traversal statistics every time this many seconds elapse  Default value: 10.0.
          sbg:altPrefix: -secondsBetweenProgressUpdates
          sbg:category: Optional Arguments
          sbg:toolDefaultValue: '10'
        - id: '#sample'
          label: |-
            The name of the sample(s) to keep, filtering out all others  This argument must be
                                          specified at least once
          type:
          - 'null'
          - string
          inputBinding:
            prefix: --sample
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            The name of the sample(s) to keep, filtering out all others  This argument must be
                                          specified at least once. Required.
          sbg:altPrefix: -sample
          sbg:category: Conditional Arguments for readFilter
          sbg:toolDefaultValue:
        - id: '#round_down_quantized'
          label: 'Round quals down to nearest quantized qual  Default value: false'
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --round_down_quantized
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Round quals down to nearest quantized qual  Default value: false. Possible values: {true,
                                          false}  Cannot be used in conjuction with argument(s) quantizationLevels (qq).
          sbg:altPrefix: -RDQ
          sbg:category: Advanced Arguments
          sbg:stageInput:
          sbg:toolDefaultValue: 'false'
        - id: '#reference'
          label: 'Reference sequence  Default value: null'
          type:
          - 'null'
          - File
          inputBinding:
            prefix: --reference
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: 'Reference sequence  Default value: null.'
          required: false
          sbg:altPrefix: -R
          sbg:category: Optional Arguments
          sbg:fileTypes: FASTA, FA
          sbg:toolDefaultValue:
        - id: '#readValidationStringency'
          label: Validation stringency for all SAM/BAM/CRAM/SRA files read by this
            program
          type:
          - 'null'
          - name: 'null'
            type: enum
            symbols:
            - STRICT
            - LENIENT
            - SILENT
          inputBinding:
            prefix: --readValidationStringency
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Validation stringency for all SAM/BAM/CRAM/SRA files read by this program.  The default
                                          stringency value SILENT can improve performance when processing a BAM file in which
                                          variable-length data (read, qualities, tags) do not otherwise need to be decoded.  Default
                                          value: SILENT. Possible values: {STRICT, LENIENT, SILENT}.
          sbg:altPrefix: -VS
          sbg:category: Optional Arguments
          sbg:toolDefaultValue: SILENT
        - id: '#readName'
          label: Keep only reads with this read name  Required
          type:
          - 'null'
          - string
          inputBinding:
            prefix: --readName
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Keep only reads with this read name  Required.
          sbg:altPrefix: -readName
          sbg:category: Conditional Arguments for readFilter
          sbg:toolDefaultValue:
        - id: '#readIndex'
          label: Indices to use for the read inputs
          type:
          - 'null'
          - string
          inputBinding:
            prefix: --readIndex
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Indices to use for the read inputs. If specified, an index must be provided for every read
                                          input and in the same order as the read inputs. If this argument is not specified, the
                                          path to the index for each input will be inferred automatically.  This argument may be
                                          specified 0 or more times. Default value: null.
          sbg:altPrefix: -readIndex
          sbg:category: Optional Arguments
          sbg:toolDefaultValue: 'null'
        - id: '#readFilter'
          label: |-
            Read filters to be applied before analysis  This argument may be specified 0 or more
                                          times
          type:
          - 'null'
          - string
          inputBinding:
            prefix: --readFilter
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Read filters to be applied before analysis  This argument may be specified 0 or more
                                          times. Default value: null.
          sbg:altPrefix: -RF
          sbg:category: Optional Arguments
          sbg:toolDefaultValue: 'null'
        - id: '#quantize_quals'
          label: 'Quantize quality scores to a given number of levels  Default value:
            0'
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --quantize_quals
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Quantize quality scores to a given number of levels  Default value: 0.  Cannot be used in
                                          conjuction with argument(s) staticQuantizationQuals (SQQ) roundDown (RDQ).
          sbg:altPrefix: -qq
          sbg:category: Optional Arguments
          sbg:toolDefaultValue: '0'
        - id: '#preserve_qscores_less_than'
          label: |-
            Don't recalibrate bases with quality scores less than this threshold  Default value: 6
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --preserve_qscores_less_than
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Don't recalibrate bases with quality scores less than this threshold  Default value: 6.
          sbg:altPrefix: -preserveQ
          sbg:category: Optional Arguments
          sbg:toolDefaultValue: '6'
        - id: '#minimumMappingQuality'
          label: 'Minimum mapping quality to keep (inclusive)  Default value: 10'
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --minimumMappingQuality
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: 'Minimum mapping quality to keep (inclusive)  Default value:
            10.'
          sbg:altPrefix: -minimumMappingQuality
          sbg:category: Conditional Arguments for readFilter
          sbg:toolDefaultValue: '10'
        - id: '#minReadLength'
          label: |-
            Keep only reads with length at least equal to the specified value  Default value: 1
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --minReadLength
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Keep only reads with length at least equal to the specified value  Default value: 1.
          sbg:altPrefix: -minReadLength
          sbg:category: Conditional Arguments for readFilter
          sbg:toolDefaultValue: '1'
        - id: '#memory_per_job'
          label: Memory per job
          type:
          - 'null'
          - int
          sbg:category: Execution
          sbg:stageInput:
        - id: '#memory_overhead_per_job'
          label: Memory overhead per job
          type:
          - 'null'
          - int
          sbg:category: Execution
          sbg:stageInput:
        - id: '#maximumMappingQuality'
          label: 'Maximum mapping quality to keep (inclusive)  Default value: null'
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --maximumMappingQuality
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: 'Maximum mapping quality to keep (inclusive)  Default value:
            null.'
          sbg:altPrefix: -maximumMappingQuality
          sbg:category: Conditional Arguments for readFilter
          sbg:toolDefaultValue: 'null'
        - id: '#maxReadLength'
          label: Keep only reads with length at most equal to the specified value  Required
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --maxReadLength
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Keep only reads with length at most equal to the specified
            value  Required.
          sbg:altPrefix: -maxReadLength
          sbg:category: Conditional Arguments for readFilter
          sbg:toolDefaultValue:
        - id: '#maxFragmentLength'
          label: |-
            Keep only read pairs with fragment length at most equal to the given value  Default value:
                                          1000000
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --maxFragmentLength
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Keep only read pairs with fragment length at most equal to the given value  Default value:
                                          1000000.
          sbg:altPrefix: -maxFragmentLength
          sbg:category: Conditional Arguments for readFilter
          sbg:toolDefaultValue: '1000000'
        - id: '#library'
          label: The name of the library to keep  Required
          type:
          - 'null'
          - string
          inputBinding:
            prefix: --library
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: The name of the library to keep  Required.
          sbg:altPrefix: -library
          sbg:category: Conditional Arguments for readFilter
          sbg:toolDefaultValue:
        - id: '#lenient'
          label: 'Lenient processing of VCF files  Default value: false'
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --lenient
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Lenient processing of VCF files  Default value: false. Possible values: {true, false}.
          sbg:altPrefix: -LE
          sbg:category: Optional Arguments
          sbg:stageInput:
          sbg:toolDefaultValue: 'false'
        - id: '#keepReverse'
          label: Keep only reads on the reverse strand  Required
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --keepReverse
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: 'Keep only reads on the reverse strand  Required. Possible
            values: {true, false}.'
          sbg:altPrefix: -keepReverse
          sbg:category: Conditional Arguments for readFilter
          sbg:stageInput:
        - id: '#keepReadGroup'
          label: The name of the read group to keep  Required
          type:
          - 'null'
          - string
          inputBinding:
            prefix: --keepReadGroup
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: The name of the read group to keep  Required.
          sbg:altPrefix: -keepReadGroup
          sbg:category: Conditional Arguments for readFilter
          sbg:toolDefaultValue:
        - id: '#intervals_string'
          label: Genomic interval strings to include
          type:
          - 'null'
          - string
          inputBinding:
            prefix: --intervals
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            One or more genomic intervals over which to operate  This argument may be specified 0 or                              more times. Default value: null.
          sbg:altPrefix: -L
          sbg:category: Optional Arguments
          sbg:stageInput:
        - id: '#intervals_file'
          label: Genomic interval files to include
          type:
          - 'null'
          - File
          inputBinding:
            prefix: --intervals
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            One or more genomic intervals over which to operate  This argument may be specified 0 or
                                          more times. Default value: null.
          required: false
          sbg:altPrefix: -L
          sbg:category: Optional Arguments
          sbg:fileTypes: TXT, BED
        - id: '#interval_set_rule'
          label: 'Set merging approach to use for combining interval inputs  Default
            value: UNION'
          type:
          - 'null'
          - name: 'null'
            type: enum
            symbols:
            - UNION
            - INTERSECTION
          inputBinding:
            prefix: --interval_set_rule
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Set merging approach to use for combining interval inputs  Default value: UNION. Possible
                                          values: {UNION, INTERSECTION}.
          sbg:altPrefix: -isr
          sbg:category: Optional Arguments
          sbg:toolDefaultValue: UNION
        - id: '#interval_padding'
          label: Of padding (in bp) to add to each interval you are including
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --interval_padding
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: 'Of padding (in bp) to add to each interval you are including.  Default
            value: 0.'
          sbg:altPrefix: -ip
          sbg:category: Optional Arguments
          sbg:toolDefaultValue: '0'
        - id: '#interval_exclusion_padding'
          label: Amount of padding (in bp) to add to each interval you are excluding
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --interval_exclusion_padding
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Amount of padding (in bp) to add to each interval you are excluding.  Default value: 0.
          sbg:altPrefix: -ixp
          sbg:category: Optional Arguments
          sbg:toolDefaultValue: '0'
        - id: '#input'
          label: |-
            BAM/SAM/CRAM file containing reads  This argument must be specified at least once
          type:
          - File
          inputBinding:
            prefix: --input
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            BAM/SAM/CRAM file containing reads  This argument must be specified at least once.
                                          Required.
          required: true
          sbg:altPrefix: -I
          sbg:category: Required Arguments
          sbg:fileTypes: BAM, SAM, CRAM
        - id: '#include_interval_name_in_output_name'
          label: Include interval name in output name
          type:
          - 'null'
          - boolean
          description: Include interval name in output name.
          sbg:stageInput:
          sbg:toolDefaultValue: 'FALSE'
        - id: '#globalQScorePrior'
          label: 'Global Qscore Bayesian prior to use for BQSR  Default value: -1'
          type:
          - 'null'
          - float
          inputBinding:
            prefix: --globalQScorePrior
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: 'Global Qscore Bayesian prior to use for BQSR  Default value:
            -1.0.'
          sbg:altPrefix: -globalQScorePrior
          sbg:category: Optional Arguments
          sbg:toolDefaultValue: '-1'
        - id: '#filterTooShort'
          label: |-
            Value for which reads with less than this number of aligned bases is considered too short 
                                          Default value: 30
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --filterTooShort
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Value for which reads with less than this number of aligned bases is considered too short 
                                          Default value: 30.
          sbg:altPrefix: -filterTooShort
          sbg:category: Conditional Arguments for readFilter
          sbg:toolDefaultValue: '30'
        - id: '#exclude_intervas_string'
          label: Genomic interval strings to exclude
          type:
          - 'null'
          - string
          inputBinding:
            prefix: --excludeIntervals
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            One or more genomic intervals to exclude from processing  This argument may be specified 0                              or more times. Default value: null.
          sbg:altPrefix: -XL
          sbg:category: Optional Arguments
        - id: '#exclude_intervals_file'
          label: Genomic interval files to exclude
          type:
          - 'null'
          - File
          inputBinding:
            prefix: --excludeIntervals
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            One or more genomic intervals to exclude from processing  This argument may be specified 0
                                          or more times. Default value: null.
          required: false
          sbg:altPrefix: -XL
          sbg:category: Optional Arguments
          sbg:fileTypes: TXT, BED
        - id: '#emit_original_quals'
          label: 'Emit original base qualities under the OQ tag  Default value: false'
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --emit_original_quals
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Emit original base qualities under the OQ tag  Default value: false. Possible values:
                                          {true, false}.
          sbg:altPrefix: -EOQ
          sbg:category: Optional Arguments
          sbg:stageInput:
          sbg:toolDefaultValue: 'false'
        - id: '#dontRequireSoftClipsBothEnds'
          label: Allow a read to be filtered out based on having only 1 soft-clipped
            block
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --dontRequireSoftClipsBothEnds
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Allow a read to be filtered out based on having only 1 soft-clipped block. By default,
                                          both ends must have a soft-clipped block, setting this flag requires only 1 soft-clipped
                                          block.  Default value: false. Possible values: {true, false}.
          sbg:altPrefix: -dontRequireSoftClipsBothEnds
          sbg:category: Conditional Arguments for readFilter
          sbg:stageInput:
          sbg:toolDefaultValue: 'false'
        - id: '#disableToolDefaultReadFilters'
          label: 'Disable all tool default read filters  Default value: false'
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --disableToolDefaultReadFilters
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Disable all tool default read filters  Default value: false. Possible values: {true,
                                          false}.
          sbg:altPrefix: -disableToolDefaultReadFilters
          sbg:category: Optional Arguments
          sbg:stageInput:
          sbg:toolDefaultValue: 'false'
        - id: '#disableSequenceDictionaryValidation'
          label: |-
            If specified, do not check the sequence dictionaries from our inputs for compatibility
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --disableSequenceDictionaryValidation
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            If specified, do not check the sequence dictionaries from our inputs for compatibility.
                                          Use at your own risk!  Default value: false. Possible values: {true, false}.
          sbg:altPrefix: -disableSequenceDictionaryValidation
          sbg:category: Optional Arguments
          sbg:stageInput:
          sbg:toolDefaultValue: 'false'
        - id: '#disableReadFilter'
          label: |-
            Filters to be disabled before analysis  This argument may be specified 0 or more
                                          times
          type:
          - 'null'
          - string
          inputBinding:
            prefix: --disableReadFilter
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Filters to be disabled before analysis  This argument may be specified 0 or more
                                          times. Default value: null.
          sbg:altPrefix: -DF
          sbg:category: Optional Arguments
          sbg:toolDefaultValue: 'null'
        - id: '#disableBamIndexCaching'
          label: |-
            If true, don't cache bam indexes, this will reduce memory requirements but may harm
                                          performance if many intervals are specified
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --disableBamIndexCaching
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            If true, don't cache bam indexes, this will reduce memory requirements but may harm
                                          performance if many intervals are specified.  Caching is automatically disabled if there
                                          are no intervals specified.  Default value: false. Possible values: {true, false}.
          sbg:altPrefix: -DBIC
          sbg:category: Optional Arguments
          sbg:stageInput:
          sbg:toolDefaultValue: 'false'
        - id: '#createOutputVariantMD5'
          label: If true, create a a MD5 digest any VCF file created
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --createOutputVariantMD5
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            If true, create a a MD5 digest any VCF file created.  Default value: false. Possible
                                          values: {true, false}.
          sbg:altPrefix: -OVM
          sbg:category: Optional Arguments
          sbg:stageInput:
          sbg:toolDefaultValue: 'false'
        - id: '#createOutputVariantIndex'
          label: If true, create a VCF index when writing a coordinate-sorted VCF
            file
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --createOutputVariantIndex
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            If true, create a VCF index when writing a coordinate-sorted VCF file.  Default value:
                                          true. Possible values: {true, false}.
          sbg:altPrefix: -OVI
          sbg:category: Optional Arguments
          sbg:stageInput:
          sbg:toolDefaultValue: 'true'
        - id: '#createOutputBamMD5'
          label: |-
            If true, create a MD5 digest for any BAM/SAM/CRAM file created  Default value: false
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --createOutputBamMD5
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            If true, create a MD5 digest for any BAM/SAM/CRAM file created  Default value: false.
                                          Possible values: {true, false}.
          sbg:altPrefix: -OBM
          sbg:category: Optional Arguments
          sbg:stageInput:
          sbg:toolDefaultValue: 'false'
        - id: '#createOutputBamIndex'
          label: If true, create a BAM/CRAM index when writing a coordinate-sorted
            BAM/CRAM file
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --createOutputBamIndex
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            If true, create a BAM/CRAM index when writing a coordinate-sorted BAM/CRAM file.  Default
                                          value: true. Possible values: {true, false}.
          sbg:altPrefix: -OBI
          sbg:category: Optional Arguments
          sbg:stageInput:
          sbg:toolDefaultValue: 'true'
        - id: '#cloudPrefetchBuffer'
          label: Size of the cloud-only prefetch buffer (in MB; 0 to disable)
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --cloudPrefetchBuffer
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Size of the cloud-only prefetch buffer (in MB; 0 to disable).  Default value: 40.
          sbg:altPrefix: -CPB
          sbg:category: Optional Arguments
          sbg:toolDefaultValue: '40'
        - id: '#cloudIndexPrefetchBuffer'
          label: Size of the cloud-only prefetch buffer (in MB; 0 to disable)
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --cloudIndexPrefetchBuffer
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Size of the cloud-only prefetch buffer (in MB; 0 to disable). Defaults to
                                          cloudPrefetchBuffer if unset.  Default value: -1.
          sbg:altPrefix: -CIPB
          sbg:category: Optional Arguments
          sbg:toolDefaultValue: '-1'
        - id: '#bqsr_recal_file'
          label: Input recalibration table for BQSR  Required
          type:
          - File
          inputBinding:
            prefix: --bqsr_recal_file
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Input recalibration table for BQSR  Required.
          required: true
          sbg:altPrefix: -bqsr
          sbg:category: Required Arguments
          sbg:toolDefaultValue:
        - id: '#blackListedLanes'
          label: |-
            Keep reads with platform units not on the list  This argument must be specified at least
                                          once
          type:
          - 'null'
          - string
          inputBinding:
            prefix: --blackListedLanes
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Keep reads with platform units not on the list  This argument must be specified at least
                                          once. Required.
          sbg:altPrefix: -blackListedLanes
          sbg:category: Conditional Arguments for readFilter
          sbg:toolDefaultValue:
        - id: '#blackList'
          label: This argument must be specified at least once
          type:
          - 'null'
          - string
          inputBinding:
            prefix: --blackList
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: This argument must be specified at least once. Required.
          sbg:altPrefix: -blackList
          sbg:category: Conditional Arguments for readFilter
          sbg:toolDefaultValue:
        - id: '#ambigFilterFrac'
          label: 'Threshold fraction of ambiguous bases  Default value: 0'
          type:
          - 'null'
          - float
          inputBinding:
            prefix: --ambigFilterFrac
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: 'Threshold fraction of ambiguous bases  Default value: 0.05.'
          sbg:altPrefix:
          sbg:category: Conditional Arguments for readFilter
          sbg:toolDefaultValue: '0'
        - id: '#ambigFilterBases'
          label: Ambig Filter Bases
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --ambigFilterBases
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Threshold number of ambiguous bases. If null, uses threshold fraction; otherwise, overrides threshold fraction. Cannot be used in conjunction with argument(s) maxAmbiguousBaseFraction.
          sbg:category: Conditional Arguments for readFilter
          sbg:stageInput:
        - id: '#addOutputVCFCommandLine'
          label: If true, adds a command line header line to created VCF files
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --addOutputVCFCommandLine
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            If true, adds a command line header line to created VCF files.  Default value: true.
                                          Possible values: {true, false}.
          sbg:altPrefix: -addOutputVCFCommandLine
          sbg:category: Optional Arguments
          sbg:stageInput:
          sbg:toolDefaultValue: 'true'
        - id: '#addOutputSAMProgramRecord'
          label: If true, adds a PG tag to created SAM/BAM/CRAM files
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --addOutputSAMProgramRecord
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            If true, adds a PG tag to created SAM/BAM/CRAM files.  Default value: true. Possible
                                          values: {true, false}.
          sbg:altPrefix: -addOutputSAMProgramRecord
          sbg:category: Optional Arguments
          sbg:stageInput:
          sbg:toolDefaultValue: 'true'
        - id: '#QUIET'
          label: Whether to suppress job-summary info on System
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --QUIET
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Whether to suppress job-summary info on System.err.  Default value: false. Possible
                                          values: {true, false}.
          sbg:category: Optional Arguments
          sbg:stageInput:
          sbg:toolDefaultValue: 'false'
        - id: '#PLFilterName'
          label: |-
            Keep reads with RG:PL attribute containing this string  This argument must be specified at
                                          least once
          type:
          - 'null'
          - string
          inputBinding:
            prefix: --PLFilterName
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Keep reads with RG:PL attribute containing this string  This argument must be specified at
                                          least once. Required.
          sbg:altPrefix: -PLFilterName
          sbg:category: Conditional Arguments for readFilter
          sbg:toolDefaultValue:

        outputs:
        - id: '#output'
          label: Output
          type:
          - 'null'
          - File
          outputBinding:
            secondaryFiles:
            - .bai
            glob:
              class: Expression
              engine: '#cwl-js-engine'
              script: |-
                {
                  input = [].concat($job.inputs.input)[0].path.split('/').slice(-1)[0]
                  input = input.split('.').slice(-1)[0]
                  
                  return '*.' + input
                }
            sbg:inheritMetadataFrom: '#input'
          description: Recalibrated BAM/SAM/CRAM file
          sbg:fileTypes: BAM, SAM, CRAM

        baseCommand:
        - /opt/gatk-launch
        - --javaOptions
        - class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              if($job.inputs.memory_per_job){
              	return '\"-Xmx'.concat($job.inputs.memory_per_job, 'M') + '\"'
              }
              	return '\"-Xmx2048M\"'
            }
        - ApplyBQSR
        arguments:
        - prefix: --output
          position: 0
          valueFrom:
            class: Expression
            engine: '#cwl-js-engine'
            script: |-
              {
                interval_name = ''
                
                if ($job.inputs.include_interval_name_in_output_name){
                  if($job.inputs.intervals_file){
                    interval_path = [].concat($job.inputs.intervals_file)[0].path
                    interval_name = interval_path.split('/')[interval_path.split('/').length - 1].split('.')
                    interval_name.pop()
                    interval_name = '_' + interval_name.join('')
                  }
                }
                
                input = [].concat($job.inputs.input)[0].path.split('/').slice(-1)[0]
                input = input.split('.')
                input = input.slice(0, -1).join('.') + interval_name + '.recalibrated.' + input.slice(-1)
                
                return input
              }
          separate: true
        - position: 0
          valueFrom: --createOutputBamIndex
          separate: true
        - position: 100
          valueFrom:
            class: Expression
            engine: '#cwl-js-engine'
            script: |-
              {
                interval_name = ''
                
                if ($job.inputs.include_interval_name_in_output_name){
                  if($job.inputs.intervals_file){
                    interval_path = [].concat($job.inputs.intervals_file)[0].path
                    interval_name = interval_path.split('/')[interval_path.split('/').length - 1].split('.')
                    interval_name.pop()
                    interval_name = '_' + interval_name.join('')
                  }
                }
                
                input = [].concat($job.inputs.input)[0].path.split('/').slice(-1)[0]
                input = input.split('.')
                input_bai = input.slice(0, -1).join('.') + interval_name + '.recalibrated.bai'
                input_bam_bai = input.slice(0, -1).join('.') + interval_name + '.recalibrated.bam.bai'
                
                return ' && mv ' + input_bai + ' ' + input_bam_bai
              }
          separate: false

        hints:
        - class: sbg:CPURequirement
          value: 1
        - class: sbg:MemRequirement
          value:
            class: Expression
            engine: '#cwl-js-engine'
            script: |-
              {
                if($job.inputs.memory_per_job){
                  if($job.inputs.memory_overhead_per_job){
                  	return $job.inputs.memory_per_job + $job.inputs.memory_overhead_per_job
                  }
                  else
                		return $job.inputs.memory_per_job
                }
                else if(!$job.inputs.memory_per_job && $job.inputs.memory_overhead_per_job){
              		return 2048 + $job.inputs.memory_overhead_per_job  
                }
                else
                	return 2048
              }
        - class: DockerRequirement
          dockerPull: images.sbgenomics.com/teodora_aleksic/gatk:4.beta.2
        id: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow/gatk-applybqsr/0
        description: |-
          Apply base quality score recalibration.

          ###**Overview**  

          This tool performs the second pass in a two-stage process called Base Quality Score Recalibration (BQSR). Specifically, it recalibrates the base qualities of the input reads based on the recalibration table produced by the BaseRecalibrator tool, and outputs a recalibrated BAM or CRAM file.

          ####**Summary of the BQSR procedure**

          The goal of this procedure is to correct for systematic bias that affect the assignment of base quality scores by the sequencer. The first pass consists of calculating error empirically and finding patterns in how error varies with basecall features over all bases. The relevant observations are written to a recalibration table. The second pass consists of applying numerical corrections to each individual basecall based on the patterns identified in the first step (recorded in the recalibration table) and write out the recalibrated data to a new BAM or CRAM file.

          ###**Inputs**  

          - A BAM or CRAM file containing input read data
          - The covariates table (= recalibration file) generated by BaseRecalibrator on the input BAM or CRAM file

          ###**Output**  

          A BAM or CRAM file containing the recalibrated read data

          ###**Usage example**  

              ./gatk-launch ApplyBQSR \   
                   -R reference.fasta \ 
                   -I input.bam \  
                   -BQSR recalibration.table \  
                   -O output.bam  

          ###**Notes**

          - This tool replaces the use of PrintReads for the application of base quality score recalibration as practiced in earlier versions of GATK (2.x and 3.x).
          - You should only run ApplyBQSR with the covariates table created from the input BAM or CRAM file(s).
          - Original qualities can be retained in the output file under the "OQ" tag if desired. See the `--emit_original_quals` argument for details.  

          ###**IMPORTANT NOTICE**  

          Tools in GATK that require a fasta reference file also look for the reference file's corresponding *.fai* (fasta index) and *.dict* (fasta dictionary) files. The fasta index file allows random access to reference bases and the dictionary file is a dictionary of the contig names and sizes contained within the fasta reference. These two secondary files are essential for GATK to work properly. To append these two files to your fasta reference please use the '***SBG FASTA Indices***' tool within your GATK based workflow before using any of the GATK tools.
        sbg:appVersion:
        - sbg:draft-2
        sbg:categories:
        - GATK-4.0
        sbg:cmdPreview: |-
          /opt/gatk-launch --javaOptions "-Xmx2048M" ApplyBQSR --bqsr_recal_file /path/to/bqsr.grp --input /path/to/input.bam --output input.recalibrated.bam  --createOutputBamIndex  && mv input.recalibrated.bai input.recalibrated.bam.bai
        sbg:contributors:
        - jelena_randjelovic
        sbg:createdBy: jelena_randjelovic
        sbg:createdOn: 1536672716
        sbg:id: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow/gatk-applybqsr/0
        sbg:image_url:
        sbg:job:
          allocatedResources:
            cpu: 1
            mem: 2048
          inputs:
            QUIET: true
            addOutputSAMProgramRecord: true
            addOutputVCFCommandLine: true
            ambigFilterBases: 4
            bqsr_recal_file:
              class: File
              secondaryFiles: []
              path: /path/to/bqsr.grp
              size: 0
            createOutputBamIndex: true
            createOutputBamMD5: true
            createOutputVariantIndex: true
            createOutputVariantMD5: true
            disableBamIndexCaching: true
            disableSequenceDictionaryValidation: true
            disableToolDefaultReadFilters: true
            dontRequireSoftClipsBothEnds: true
            emit_original_quals: true
            exclude_intervas_string: ''
            include_interval_name_in_output_name: false
            input:
              class: File
              secondaryFiles: []
              path: /path/to/input.bam
              size: 0
            intervals_string: ''
            keepReverse: true
            lenient: true
            memory_overhead_per_job: 0
            memory_per_job: 2048
            reference:
              class: File
              secondaryFiles: []
              path: /path/to/reference.ext
              size: 0
            round_down_quantized: true
            useOriginalQualities: true
            use_jdk_deflater: true
            use_jdk_inflater: true
        sbg:latestRevision: 0
        sbg:license: Open source BSD (3-clause) license
        sbg:links:
        - id: https://software.broadinstitute.org/gatk/index.php
          label: Homepage
        - id: https://github.com/broadinstitute/gatk
          label: Source Code
        - id: |-
            https://gatkforums.broadinstitute.org/gatk/discussion/9881/howto-get-started-with-gatk4-beta
          label: Download
        - id: https://software.broadinstitute.org/gatk/gatkdocs/4.beta.2/
          label: Documentation
        sbg:modifiedBy: jelena_randjelovic
        sbg:modifiedOn: 1536672716
        sbg:project: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow
        sbg:projectName: PDXnet - Polishing WES JAX workflow
        sbg:publisher: sbg
        sbg:revision: 0
        sbg:revisionNotes: Copy of anuj_srivastava/pdx-wes-pipeline/gatk-4-0-applybqsr/0
        sbg:revisionsInfo:
        - sbg:modifiedBy: jelena_randjelovic
          sbg:modifiedOn: 1536672716
          sbg:revision: 0
          sbg:revisionNotes: Copy of anuj_srivastava/pdx-wes-pipeline/gatk-4-0-applybqsr/0
        sbg:sbgMaintained: false
        sbg:toolAuthor: Broad Institute
        sbg:toolkit: GATK 4.0
        sbg:toolkitVersion: Beta 2
        sbg:validationErrors: []
        sbg:wrapperAuthor: filip_tubic
        x: 1053.666666666667
        y: 171.0331624348959
      inputs:
      - id: '#GATK_ApplyBQSR.reference'
        source: '#reference'
      - id: '#GATK_ApplyBQSR.input'
        source: '#Picard_MarkDuplicates.deduped_bam'
      - id: '#GATK_ApplyBQSR.bqsr_recal_file'
        source: '#GATK_BaseRecalibrator.recalibration_file'
      outputs:
      - id: '#GATK_ApplyBQSR.output'
      sbg:x: 1053.666666666667
      sbg:y: 171.0331624348959
    - id: '#Picard_SortSam'
      label: Picard SortSam
      in: []
      run:
        cwlVersion: sbg:draft-2
        class: CommandLineTool
        label: Picard SortSam
        $namespaces:
          sbg: https://sevenbridges.com

        requirements:
        - id: '#cwl-js-engine'
          class: ExpressionEngineRequirement
          requirements:
          - class: DockerRequirement
            dockerPull: rabix/js-engine
          engineCommand: cwl-engine.js
        - class: InlineJavascriptRequirement

        inputs:
        - id: '#validation_stringency'
          label: Validation stringency
          type:
          - 'null'
          - name: validation_stringency
            type: enum
            symbols:
            - STRICT
            - LENIENT
            - SILENT
          inputBinding:
            prefix: VALIDATION_STRINGENCY=
            position: 0
            valueFrom:
              class: Expression
              engine: '#cwl-js-engine'
              script: |-
                {
                  if ($job.inputs.validation_stringency)
                  {
                    return $job.inputs.validation_stringency
                  }
                  else
                  {
                    return "SILENT"
                  }
                }
            separate: false
            sbg:cmdInclude: true
          description: |-
            Validation stringency for all SAM files read by this program. Setting stringency to SILENT can improve performance when processing a BAM file in which variable-length data (read, qualities, tags) do not otherwise need to be decoded. This option can be set to 'null' to clear the default value. Possible values: {STRICT, LENIENT, SILENT}.
          sbg:category: Other input types
          sbg:toolDefaultValue: SILENT
        - id: '#sort_order'
          label: Sort order
          type:
          - name: sort_order
            type: enum
            symbols:
            - Unsorted
            - Queryname
            - Coordinate
          inputBinding:
            prefix: SORT_ORDER=
            position: 3
            valueFrom:
              class: Expression
              engine: '#cwl-js-engine'
              script: "{\n  p = $job.inputs.sort_order.toLowerCase()\n  return p\n\
                }"
            separate: false
            sbg:cmdInclude: true
          description: |-
            Sort order of the output file. Possible values: {unsorted, queryname, coordinate}.
          sbg:altPrefix: SO
          sbg:category: Other input types
        - id: '#quiet'
          label: Quiet
          type:
          - 'null'
          - name: quiet
            type: enum
            symbols:
            - 'True'
            - 'False'
          inputBinding:
            prefix: QUIET=
            position: 0
            separate: false
            sbg:cmdInclude: true
          description: |-
            This parameter indicates whether to suppress job-summary info on System.err. This option can be set to 'null' to clear the default value. Possible values: {true, false}.
          sbg:category: Other input types
          sbg:toolDefaultValue: 'False'
        - id: '#output_type'
          label: Output format
          type:
          - 'null'
          - name: output_type
            type: enum
            symbols:
            - BAM
            - SAM
            - SAME AS INPUT
          description: |-
            Since Picard tools can output both SAM and BAM files, user can choose the format of the output file.
          sbg:category: Other input types
          sbg:toolDefaultValue: SAME AS INPUT
        - id: '#memory_per_job'
          label: Memory per job
          type:
          - 'null'
          - int
          description: |-
            Amount of RAM memory to be used per job. Defaults to 2048 MB for single threaded jobs.
          sbg:toolDefaultValue: '2048'
        - id: '#max_records_in_ram'
          label: Max records in RAM
          type:
          - 'null'
          - int
          inputBinding:
            prefix: MAX_RECORDS_IN_RAM=
            position: 0
            separate: false
            sbg:cmdInclude: true
          description: |-
            When writing SAM files that need to be sorted, this parameter will specify the number of records stored in RAM before spilling to disk. Increasing this number reduces the number of file handles needed to sort a SAM file, and increases the amount of RAM needed. This option can be set to 'null' to clear the default value.
          sbg:category: Other input types
          sbg:toolDefaultValue: '500000'
        - id: '#input_bam'
          label: Input BAM
          type:
          - File
          inputBinding:
            prefix: INPUT=
            position: 1
            separate: false
            sbg:cmdInclude: true
          description: The BAM or SAM file to sort.
          required: true
          sbg:altPrefix: I
          sbg:category: File inputs
          sbg:fileTypes: BAM, SAM
        - id: '#create_index'
          label: Create index
          type:
          - 'null'
          - name: create_index
            type: enum
            symbols:
            - 'True'
            - 'False'
          inputBinding:
            prefix: CREATE_INDEX=
            position: 5
            separate: false
            sbg:cmdInclude: true
          description: |-
            This parameter indicates whether to create a BAM index when writing a coordinate-sorted BAM file. This option can be set to 'null' to clear the default value. Possible values: {true, false}.
          sbg:category: Other input types
          sbg:toolDefaultValue: 'False'
        - id: '#compression_level'
          label: Compression level
          type:
          - 'null'
          - int
          inputBinding:
            prefix: COMPRESSION_LEVEL=
            position: 0
            separate: false
            sbg:cmdInclude: true
          description: |-
            Compression level for all compressed files created (e.g. BAM and GELI). This option can be set to 'null' to clear the default value.
          sbg:category: Other input types
          sbg:toolDefaultValue: '5'

        outputs:
        - id: '#sorted_bam'
          label: Sorted BAM/SAM
          type:
          - 'null'
          - File
          outputBinding:
            secondaryFiles:
            - ^.bai
            - .bai
            glob: '*.sorted.?am'
            sbg:inheritMetadataFrom: '#input_bam'
            sbg:metadata:
              __inherit__: input_bam
          description: Sorted BAM or SAM file.
          sbg:fileTypes: BAM, SAM

        baseCommand:
        - java
        - class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {   
              if($job.inputs.memory_per_job){
                return '-Xmx'.concat($job.inputs.memory_per_job, 'M')
              }   
              	return '-Xmx2048M'
            }
        - -jar
        - /opt/picard-tools-1.140/picard.jar
        - SortSam
        arguments:
        - prefix: OUTPUT=
          position: 0
          valueFrom:
            class: Expression
            engine: '#cwl-js-engine'
            script: |-
              {
                filename = $job.inputs.input_bam.path
                ext = $job.inputs.output_type

              if (ext === "BAM")
              {
                  return filename.split('.').slice(0, -1).concat("sorted.bam").join(".").replace(/^.*[\\\/]/, '')
                  }

              else if (ext === "SAM")
              {
                  return filename.split('.').slice(0, -1).concat("sorted.sam").join('.').replace(/^.*[\\\/]/, '')
              }

              else 
              {
              	return filename.split('.').slice(0, -1).concat("sorted."+filename.split('.').slice(-1)[0]).join(".").replace(/^.*[\\\/]/, '')
              }
              }
          separate: false
        - position: 1000
          valueFrom:
            class: Expression
            engine: '#cwl-js-engine'
            script: |-
              {
                filename = $job.inputs.input_bam.path
                
                /* figuring out output file type */
                ext = $job.inputs.output_type
                if (ext === "BAM")
                {
                  out_extension = "BAM"
                }
                else if (ext === "SAM")
                {
                  out_extension = "SAM"
                }
                else 
                {
              	out_extension = filename.split('.').slice(-1)[0].toUpperCase()
                }  
                
                /* if exist moving .bai in bam.bai */
                if ($job.inputs.create_index === 'True' && $job.inputs.sort_order === 'Coordinate' && out_extension == "BAM")
                {
                  
                  old_name = filename.split('.').slice(0, -1).concat('sorted.bai').join('.').replace(/^.*[\\\/]/, '')
                  new_name = filename.split('.').slice(0, -1).concat('sorted.bam.bai').join('.').replace(/^.*[\\\/]/, '')
                  return "; mv " + " " + old_name + " " + new_name
                }

              }
          separate: true

        hints:
        - class: sbg:CPURequirement
          value: 1
        - class: sbg:MemRequirement
          value:
            class: Expression
            engine: '#cwl-js-engine'
            script: |-
              {
                if($job.inputs.memory_per_job){
                	return $job.inputs.memory_per_job
                }
                	return 2048
              }
        - class: DockerRequirement
          dockerPull: images.sbgenomics.com/mladenlsbg/picard:1.140
          dockerImageId: eab0e70b6629
        id: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow/picard-sortsam/1
        description: |-
          Picard SortSam sorts the input SAM or BAM. Input and output formats are determined by the file extension.
        sbg:appVersion:
        - sbg:draft-2
        sbg:categories:
        - SAM/BAM-Processing
        sbg:cmdPreview: |-
          java -Xmx2048M -jar /opt/picard-tools-1.140/picard.jar SortSam OUTPUT=example.tested.sorted.bam INPUT=/root/dir/example.tested.bam SORT_ORDER=coordinate
        sbg:contributors:
        - jelena_randjelovic
        sbg:createdBy: jelena_randjelovic
        sbg:createdOn: 1536672712
        sbg:id: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow/picard-sortsam/1
        sbg:image_url:
        sbg:job:
          allocatedResources:
            cpu: 1
            mem: 2048
          inputs:
            create_index:
            input_bam:
              path: /root/dir/example.tested.bam
            memory_per_job: 2048
            output_type:
            sort_order: Coordinate
        sbg:latestRevision: 1
        sbg:license: MIT License, Apache 2.0 Licence
        sbg:links:
        - id: http://broadinstitute.github.io/picard/index.html
          label: Homepage
        - id: https://github.com/broadinstitute/picard/releases/tag/1.138
          label: Source Code
        - id: http://broadinstitute.github.io/picard/
          label: Wiki
        - id: https://github.com/broadinstitute/picard/zipball/master
          label: Download
        - id: http://broadinstitute.github.io/picard/
          label: Publication
        sbg:modifiedBy: jelena_randjelovic
        sbg:modifiedOn: 1536672763
        sbg:project: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow
        sbg:projectName: PDXnet - Polishing WES JAX workflow
        sbg:publisher: sbg
        sbg:revision: 1
        sbg:revisionNotes: Copy of anuj_srivastava/pdx-wes-pipeline/picard-sortsam-1-140/0
        sbg:revisionsInfo:
        - sbg:modifiedBy: jelena_randjelovic
          sbg:modifiedOn: 1536672712
          sbg:revision: 0
          sbg:revisionNotes: Copy of anuj_srivastava/pdx-wes-pipeline/picard-sortsam-1-140/0
        - sbg:modifiedBy: jelena_randjelovic
          sbg:modifiedOn: 1536672763
          sbg:revision: 1
          sbg:revisionNotes: Copy of anuj_srivastava/pdx-wes-pipeline/picard-sortsam-1-140/0
        sbg:sbgMaintained: false
        sbg:toolAuthor: Broad Institute
        sbg:toolkit: Picard
        sbg:toolkitVersion: '1.140'
        sbg:validationErrors: []
        x: 686.9474283854169
        y: -81.50799560546876
      inputs:
      - id: '#Picard_SortSam.sort_order'
        default: Coordinate
      - id: '#Picard_SortSam.memory_per_job'
        source: '#memory_per_job'
      - id: '#Picard_SortSam.input_bam'
        source: '#bwa_alt_aware_v3.bam_output'
      - id: '#Picard_SortSam.create_index'
        default: 'True'
      outputs:
      - id: '#Picard_SortSam.sorted_bam'
      sbg:x: 677.8410034179688
      sbg:y: -89.15899658203125
    - id: '#Picard_MarkDuplicates'
      label: Picard MarkDuplicates
      in: []
      run:
        cwlVersion: sbg:draft-2
        class: CommandLineTool
        label: Picard MarkDuplicates
        $namespaces:
          sbg: https://sevenbridges.com

        requirements:
        - id: '#cwl-js-engine'
          class: ExpressionEngineRequirement
          requirements:
          - class: DockerRequirement
            dockerPull: rabix/js-engine
          engineCommand: cwl-engine.js
        - class: InlineJavascriptRequirement

        inputs:
        - id: '#verbosity'
          label: Verbosity
          type:
          - 'null'
          - name: verbosity
            type: enum
            symbols:
            - ERROR
            - WARNING
            - INFO
            - DEBUG
          inputBinding:
            prefix: VERBOSITY=
            position: 6
            separate: false
            sbg:cmdInclude: true
          description: |-
            Control verbosity of logging. Default value: INFO. This option can be set to 'null' to clear the default value. Possible values: {ERROR, WARNING, INFO, DEBUG}.
          sbg:category: Options
          sbg:toolDefaultValue: INFO
        - id: '#validation_stringency'
          label: Validation stringency
          type:
          - 'null'
          - name: validation_stringency
            type: enum
            symbols:
            - STRICT
            - LENIENT
            - SILENT
          inputBinding:
            prefix: VALIDATION_STRINGENCY=
            position: 4
            valueFrom:
              class: Expression
              engine: '#cwl-js-engine'
              script: |-
                {
                  if ($job.inputs.validation_stringency)
                  {
                    return $job.inputs.validation_stringency
                  }
                  else
                  {
                    return "SILENT"
                  }
                }
            separate: false
            sbg:cmdInclude: true
          description: |-
            Validation stringency for all SAM files read by this program. Setting stringency to SILENT can improve performance when processing a BAM file in which variable-length data (read, qualities, tags) do not otherwise need to be decoded. Default value: STRICT. This option can be set to 'null' to clear the default value. Possible values: {STRICT, LENIENT, SILENT}.
          sbg:category: Options
          sbg:toolDefaultValue: SILENT
        - id: '#sorting_collections_size_ratio'
          label: Sorting collections size ratio
          type:
          - 'null'
          - float
          inputBinding:
            prefix: SORTING_COLLECTION_SIZE_RATIO=
            position: 9
            separate: false
            sbg:cmdInclude: true
          description: |-
            This number, plus the maximum RAM available to the JVM, determine the memory footprint used by some of the sorting collections. If you are running out of memory, try reducing this number.
          sbg:category: Options
          sbg:stageInput:
          sbg:toolDefaultValue: '0.25'
        - id: '#remove_duplicates'
          label: Remove duplicates
          type:
          - 'null'
          - name: remove_duplicates
            type: enum
            symbols:
            - 'true'
            - 'false'
          inputBinding:
            prefix: REMOVE_DUPLICATES=
            position: 3
            separate: false
            sbg:cmdInclude: true
          description: |-
            If this parameter is set to true, duplicates will not be written to the output file. If set to false, duplicates will be written with the appropriate flags set. Default value: false. This option can be set to 'null' to clear the default value. Possible values: {true, false}.
          sbg:category: Options
          sbg:toolDefaultValue: 'false'
        - id: '#read_name_regex'
          label: Read name regex
          type:
          - 'null'
          - string
          inputBinding:
            prefix: READ_NAME_REGEX=
            position: 9
            separate: false
            sbg:cmdInclude: true
          description: |-
            Regular expression that can be used to parse read names in the incoming SAM file. Read names are parsed to extract three variables: tile/region, x coordinate and y coordinate. These values are used to estimate the rate of optical duplication in order to give a more accurate estimated library size. Set this option to null to disable optical duplicate detection. The regular expression should contain three capture groups for the three variables, in order. It must match the entire read name. Note that if the default regex is specified, a regex match is not actually done, but instead the read name is split on colon character. For 5 element names, the 3rd, 4th and 5th elements are assumed to be tile, x and y values, respectively. For 7 element names (CASAVA 1.8), the 5th, 6th, and 7th elements are assumed to be tile, x and y values, respectively. Default value: [a-zA-Z0-9]+:[0-9]:([0-9]+):([0-9]+):([0-9]+).*. This option can be set to 'null' to clear the default value.
          sbg:category: Options
          sbg:toolDefaultValue: '[a-zA-Z0-9]+:[0-9]:([0-9]+):([0-9]+):([0-9]+).*'
        - id: '#quiet'
          label: Quiet
          type:
          - 'null'
          - name: quiet
            type: enum
            symbols:
            - 'True'
            - 'False'
          inputBinding:
            prefix: QUIET=
            position: 4
            separate: false
            sbg:cmdInclude: true
          description: |-
            This parameter indicates whether to suppress job-summary info on System.err. Possible values: {True, False}.
          sbg:category: Options
          sbg:toolDefaultValue: 'False'
        - id: '#program_record'
          label: Program record ID
          type:
          - 'null'
          - string
          inputBinding:
            prefix: PROGRAM_RECORD_ID=
            position: 10
            separate: false
            sbg:cmdInclude: true
          description: |-
            The program record ID for the @PG record(s) created by this program. Set to null to disable PG record creation. This string may have a suffix appended to avoid collision with other program record IDs. Default value: MarkDuplicates. This option can be set to 'null' to clear the default value.
          sbg:altPrefix: PG
          sbg:category: Options
          sbg:toolDefaultValue: MarkDuplicates
        - id: '#program_group_ver'
          label: Program group version
          type:
          - 'null'
          - string
          inputBinding:
            prefix: PROGRAM_GROUP_VERSION=
            position: 9
            separate: false
            sbg:cmdInclude: true
          description: |-
            Value of version number (VN) tag of program group (PG) record to be created. If not specified, the version will be detected automatically. Default value: null.
          sbg:altPrefix: PG_VERSION=
          sbg:category: Options
        - id: '#program_group_name'
          label: Program group name
          type:
          - 'null'
          - string
          inputBinding:
            prefix: PROGRAM_GROUP_NAME=
            position: 9
            separate: false
            sbg:cmdInclude: true
          description: |-
            Value of program name (PN) tag of program group (PG) record to be created. Default value: MarkDuplicates. This option can be set to 'null' to clear the default value.
          sbg:altPrefix: PG_NAME=
          sbg:category: Options
          sbg:toolDefaultValue: MarkDuplicates
        - id: '#program_group_command_line'
          label: Program group command line
          type:
          - 'null'
          - string
          inputBinding:
            prefix: PROGRAM_GROUP_COMMAND_LINE=
            position: 9
            separate: false
            sbg:cmdInclude: true
          description: |-
            Value of command line (CL) tag of program group (PG) record to be created. If not supplied the command line will be detected automatically. Default value: null.
          sbg:altPrefix: PG_COMMAND=
          sbg:category: Options
          sbg:toolDefaultValue:
        - id: '#output_type'
          label: Output format
          type:
          - 'null'
          - name: output_type
            type: enum
            symbols:
            - BAM
            - SAM
            - SAME AS INPUT
          description: |-
            Since Picard tools can output both SAM and BAM files, user can choose the format of the output file.
          sbg:category: Options
          sbg:toolDefaultValue: SAME AS INPUT
        - id: '#optical_duplicate_pixel_distance'
          label: Optical duplicate pixel distance
          type:
          - 'null'
          - int
          inputBinding:
            prefix: OPTICAL_DUPLICATE_PIXEL_DISTANCE=
            position: 9
            separate: false
            sbg:cmdInclude: true
          description: |-
            The maximum offset between two duplicate clusters in order to consider them optical duplicates. This should usually be set to some fairly small number (e.g. 5-10 pixels) unless using later versions of the Illumina pipeline that multiply pixel values by 10, in which case 50-100 is more normal. Default value: 100. This option can be set to 'null' to clear the default value.
          sbg:category: Options
          sbg:toolDefaultValue: '100'
        - id: '#memory_per_job'
          label: Memory per job
          type:
          - 'null'
          - int
          description: |-
            Amount of RAM memory to be used per job. Defaults to 2048MB for single threaded jobs.
          sbg:category: Execution options
          sbg:toolDefaultValue: '2048'
        - id: '#max_sequences_for_disk_read_ends_map'
          label: Max sequences for disk read ends map
          type:
          - 'null'
          - int
          inputBinding:
            prefix: MAX_SEQUENCES_FOR_DISK_READ_ENDS_MAP=
            position: 9
            separate: false
            sbg:cmdInclude: true
          description: |-
            This option is obsolete. ReadEnds will always be spilled to disk. Default value: 50000. This option can be set to 'null' to clear the default value.
          sbg:altPrefix: MAX_SEQS=
          sbg:category: Options
          sbg:toolDefaultValue: '50000'
        - id: '#max_records_in_ram'
          label: Max records in RAM
          type:
          - 'null'
          - int
          inputBinding:
            prefix: MAX_RECORDS_IN_RAM=
            position: 4
            separate: false
            sbg:cmdInclude: true
          description: |-
            When writing SAM files that need to be sorted, this parameter will specify the number of records stored in RAM before spilling to disk. Increasing this number reduces the number of file handles needed to sort a SAM file, and increases the amount of RAM needed. Default value: 500000. This option can be set to 'null' to clear the default value.
          sbg:category: Options
          sbg:toolDefaultValue: '500000'
        - id: '#max_file_handles_for_read_ends_map'
          label: Max file handles for read ends map
          type:
          - 'null'
          - int
          inputBinding:
            prefix: MAX_FILE_HANDLES_FOR_READ_ENDS_MAP=
            position: 9
            separate: false
            sbg:cmdInclude: true
          description: |-
            Maximum number of file handles to keep open when spilling read ends to disk. Set this number a little lower than the per-process maximum number of file that may be open. This number can be found by executing the 'ulimit -n' command on a Unix system. Default value: 8000. This option can be set to 'null' to clear the default value.
          sbg:altPrefix: MAX_FILE_HANDLES=
          sbg:category: Options
          sbg:toolDefaultValue: '8000'
        - id: '#input_bam'
          label: Input bam
          type:
          - type: array
            items: File
          inputBinding:
            prefix: INPUT=
            position: 0
            separate: false
            sbg:cmdInclude: true
          description: |-
            This parameter indicates one or more input SAM or BAM files to analyze. Must be coordinate sorted.  Default value: null. This option may be specified 0 or more times.
          required: true
          sbg:altPrefix: I
          sbg:category: File inputs
          sbg:fileTypes: SAM, BAM
        - id: '#duplicate_scoring_strategy'
          label: Duplicate scoring strategy
          type:
          - 'null'
          - name: duplicate_scoring_strategy
            type: enum
            symbols:
            - SUM_OF_BASE_QUALITIES
            - TOTAL_MAPPED_REFERENCE_LENGTH
          inputBinding:
            prefix: DUPLICATE_SCORING_STRATEGY=
            position: 0
            separate: false
            sbg:cmdInclude: true
          description: |-
            This parameter indicates the scoring strategy for choosing the non-duplicate among candidates.
          sbg:altPrefix: DS
          sbg:category: Options
          sbg:toolDefaultValue: SUM_OF_BASE_QUALITIES
        - id: '#create_index'
          label: Create index
          type:
          - 'null'
          - name: create_index
            type: enum
            symbols:
            - 'True'
            - 'False'
          inputBinding:
            prefix: CREATE_INDEX=
            position: 5
            separate: false
            sbg:cmdInclude: true
          description: |-
            This parameter indicates whether to create a BAM index when writing a coordinate-sorted BAM file. Default value: False. This option can be set to 'null' to clear the default value. Possible values: {True, False}.
          sbg:category: Options
          sbg:toolDefaultValue: 'False'
        - id: '#compression_level'
          label: Compression level
          type:
          - 'null'
          - int
          inputBinding:
            prefix: COMPRESSION_LEVEL=
            position: 4
            separate: false
            sbg:cmdInclude: true
          description: |-
            Compression level for all compressed files created (e.g. BAM and GELI). Default value: 5. This option can be set to 'null' to clear the default value.
          sbg:category: Options
          sbg:toolDefaultValue: '5'
        - id: '#comment'
          label: Comment
          type:
          - 'null'
          - type: array
            items: string
          inputBinding:
            prefix: COMMENT=
            position: 9
            separate: false
            sbg:cmdInclude: true
          description: |-
            Comment(s) to include in the output file's header. Default value: null. This option may be specified 0 or more times.
          sbg:altPrefix: CO
          sbg:category: Options
          sbg:stageInput:
        - id: '#assume_sorted'
          label: Assume sorted
          type:
          - 'null'
          - name: assume_sorted
            type: enum
            symbols:
            - 'true'
            - 'false'
          inputBinding:
            prefix: ASSUME_SORTED=
            position: 0
            separate: false
            sbg:cmdInclude: true
          description: |-
            If this parameter is set to true, it is assumed that the input file is coordinate sorted even if the header says otherwise.  Default value: false. This option can be set to 'null' to clear the default value. Possible values: {true, false}.
          sbg:altPrefix: AS
          sbg:category: Options
          sbg:toolDefaultValue: 'false'

        outputs:
        - id: '#metrics_file'
          label: Metrics file
          type:
          - 'null'
          - File
          outputBinding:
            glob: '*.metrics'
            sbg:inheritMetadataFrom: '#input_bam'
            sbg:metadata:
              __inherit__: input_bam
          description: File to which the duplication metrics will be written.
          sbg:fileTypes: METRICS
        - id: '#deduped_bam'
          label: Deduped BAM
          type:
          - 'null'
          - File
          outputBinding:
            secondaryFiles:
            - ^.bai
            - .bai
            glob: '*.deduped.?am'
            sbg:inheritMetadataFrom: '#input_bam'
            sbg:metadata:
              __inherit__: input_bam
          description: The output file to which marked records will be written.
          sbg:fileTypes: BAM, SAM

        baseCommand:
        - java
        - class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {   
              if($job.inputs.memory_per_job){
                return '-Xmx'.concat($job.inputs.memory_per_job, 'M')
              }   
              	return '-Xmx2048M'
            }
        - -jar
        - /opt/picard-tools-1.140/picard.jar
        - MarkDuplicates
        arguments:
        - prefix: METRICS_FILE=
          position: 0
          valueFrom:
            class: Expression
            engine: '#cwl-js-engine'
            script: |-
              {
                input_bam = [].concat($job.inputs.input_bam)
                filename =input_bam[0].path.split('/').slice(-1)[0];
                return filename.split('.').slice(0, -1).concat("metrics").join(".");
              }
          separate: false
        - prefix: OUTPUT=
          position: 0
          valueFrom:
            class: Expression
            engine: '#cwl-js-engine'
            script: |-
              {
                input_bam = [].concat($job.inputs.input_bam)
                filename = input_bam[0].path.split('/').slice(-1)[0];
                ext = $job.inputs.output_type;
                filebase = filename.split('.').slice(0, -1).join('.') + '.deduped.';
                
              if (ext === "BAM")
              {
                  return filebase + 'bam';
                  }

              else if (ext === "SAM")
              {
                  return filebase + 'sam';
              }

              else 
              {
              	return filebase + filename.split('.').slice(-1)[0];
              }
              }
          separate: false
        - position: 2001
          valueFrom:
            class: Expression
            engine: '#cwl-js-engine'
            script: |-
              {
                input_bam = [].concat($job.inputs.input_bam)
                filename = input_bam[0].path
                
                /* figuring out output file type */
                ext = $job.inputs.output_type
                if (ext === "BAM")
                {
                  out_extension = "BAM"
                }
                else if (ext === "SAM")
                {
                  out_extension = "SAM"
                }
                else 
                {
              	out_extension = filename.split('.').slice(-1)[0].toUpperCase()
                }  
                
                /* if exist moving .bai in bam.bai */
                if ($job.inputs.create_index === 'True' && out_extension == "BAM")
                {
                  
                  old_name = filename.split('.').slice(0, -1).concat('deduped.bai').join('.').replace(/^.*[\\\/]/, '')
                  new_name = filename.split('.').slice(0, -1).concat('deduped.bam.bai').join('.').replace(/^.*[\\\/]/, '')
                  return "; mv " + " " + old_name + " " + new_name
                }

              }
          separate: true

        hints:
        - class: sbg:CPURequirement
          value: 1
        - class: sbg:MemRequirement
          value:
            class: Expression
            engine: '#cwl-js-engine'
            script: |-
              {
                if($job.inputs.memory_per_job){
                	return $job.inputs.memory_per_job
                }
                	return 2048
              }
        - class: DockerRequirement
          dockerPull: images.sbgenomics.com/mladenlsbg/picard:1.140
          dockerImageId: eab0e70b6629
        id: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow/picard-markduplicates/1
        description: |-
          Picard MarkDuplicates examines aligned records in the supplied SAM or BAM file to locate duplicate molecules. All records are then written to the output file with the duplicate records flagged.
        sbg:appVersion:
        - sbg:draft-2
        sbg:categories:
        - SAM/BAM-Processing
        sbg:cmdPreview: |-
          java -Xmx2048M -jar /opt/picard-tools-1.140/picard.jar MarkDuplicates INPUT=input_bam.ext METRICS_FILE=input_bam.metrics OUTPUT=input_bam.deduped.bam
        sbg:contributors:
        - jelena_randjelovic
        sbg:createdBy: jelena_randjelovic
        sbg:createdOn: 1536672715
        sbg:id: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow/picard-markduplicates/1
        sbg:image_url:
        sbg:job:
          allocatedResources:
            cpu: 1
            mem: 2048
          inputs:
            comment:
            - comment-string-value-1
            - comment-string-value-2
            duplicate_scoring_strategy: SUM_OF_BASE_QUALITIES
            input_bam:
            - class: File
              secondaryFiles: []
              path: input_bam.ext
              size: 0
            max_file_handles_for_read_ends_map: 0
            memory_per_job: 0
            output_type: BAM
            sorting_collections_size_ratio: 2.163350957715713
        sbg:latestRevision: 1
        sbg:license: MIT License, Apache 2.0 Licence
        sbg:links:
        - id: http://broadinstitute.github.io/picard/
          label: Homepage
        - id: https://github.com/broadinstitute/picard/releases/tag/1.140
          label: Source Code
        - id: http://broadinstitute.github.io/picard/
          label: Wiki
        - id: https://github.com/broadinstitute/picard/zipball/master
          label: Download
        - id: http://broadinstitute.github.io/picard/
          label: Publication
        sbg:modifiedBy: jelena_randjelovic
        sbg:modifiedOn: 1536672764
        sbg:project: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow
        sbg:projectName: PDXnet - Polishing WES JAX workflow
        sbg:publisher: sbg
        sbg:revision: 1
        sbg:revisionNotes: Copy of anuj_srivastava/pdx-wes-pipeline/picard-markduplicates-1-140/0
        sbg:revisionsInfo:
        - sbg:modifiedBy: jelena_randjelovic
          sbg:modifiedOn: 1536672715
          sbg:revision: 0
          sbg:revisionNotes: Copy of anuj_srivastava/pdx-wes-pipeline/picard-markduplicates-1-140/0
        - sbg:modifiedBy: jelena_randjelovic
          sbg:modifiedOn: 1536672764
          sbg:revision: 1
          sbg:revisionNotes: Copy of anuj_srivastava/pdx-wes-pipeline/picard-markduplicates-1-140/0
        sbg:sbgMaintained: false
        sbg:toolAuthor: Broad Institute
        sbg:toolkit: Picard
        sbg:toolkitVersion: '1.140'
        sbg:validationErrors: []
        x: 791.4589843750002
        y: -16.669921875000004
      hints:
      - class: sbg:AWSInstanceType
        value: c5.4xlarge;ebs-gp2;1024
      inputs:
      - id: '#Picard_MarkDuplicates.memory_per_job'
        source: '#memory_per_job_1'
      - id: '#Picard_MarkDuplicates.input_bam'
        source:
        - '#Picard_SortSam.sorted_bam'
      - id: '#Picard_MarkDuplicates.create_index'
        default: 'True'
      outputs:
      - id: '#Picard_MarkDuplicates.metrics_file'
      - id: '#Picard_MarkDuplicates.deduped_bam'
      sbg:x: 794.0596313476562
      sbg:y: 2.3577497005462646
    - id: '#Exome_Coverage_QC_1_0'
      label: Exome Coverage QC 1.0
      in: []
      run:
        cwlVersion: sbg:draft-2
        class: Workflow
        label: Exome Coverage QC 1.0
        $namespaces:
          sbg: https://sevenbridges.com

        requirements:
        - class: SubworkflowFeatureRequirement
        - class: InlineJavascriptRequirement

        inputs:
        - id: '#bed'
          label: bed
          type:
          - File
          description: Target interval BED
          required: true
          sbg:includeInPorts: true
          sbg:x: 94
          sbg:y: 149
        - id: '#bam'
          label: bam
          type:
          - File
          description: BAM
          required: true
          sbg:includeInPorts: true
          sbg:x: 103
          sbg:y: 510

        outputs:
        - id: '#summary'
          label: summary
          type:
          - 'null'
          - File
          source:
          - '#SBG_Coverage.summary'
          description: Coverage summary
          sbg:includeInPorts: true
          sbg:x: 1457
          sbg:y: 210
        - id: '#coverage'
          label: coverage
          type:
          - 'null'
          - File
          source:
          - '#SBG_Coverage.coverage'
          description: Coverage BED
          sbg:includeInPorts: true
          sbg:x: 1457
          sbg:y: 516

        steps:
        - id: '#SBG_Coverage'
          label: SBG Coverage
          in: []
          run:
            cwlVersion: sbg:draft-2
            class: CommandLineTool
            label: SBG Coverage
            $namespaces:
              sbg: https://sevenbridges.com

            requirements:
            - id: '#cwl-js-engine'
              class: ExpressionEngineRequirement
              requirements:
              - class: DockerRequirement
                dockerPull: rabix/js-engine
            - class: InlineJavascriptRequirement

            inputs:
            - id: '#bam'
              label: Input BAM or BED
              type:
              - File
              inputBinding:
                prefix: --bam
                position: 0
                separate: true
                sbg:cmdInclude: true
              description: Input BAM or BED file for coverage calculation.
              sbg:category: ''
            - id: '#bed'
              label: Input BED file
              type:
              - File
              inputBinding:
                prefix: --bed
                position: 0
                separate: true
                sbg:cmdInclude: true
              description: BED file with defined feature intervals.
              sbg:category: ''
            - id: '#strandness'
              label: Strandness
              type:
              - 'null'
              - name: strandness
                type: enum
                symbols:
                - disregard
                - same
                - opposite
              inputBinding:
                prefix: --strandness
                position: 0
                separate: true
                sbg:cmdInclude: true
              description: |-
                Features in bam are only counted towards coverage in bed if they are the same strand. Default: disregard
              sbg:category: ''
            - id: '#depth_per_base'
              label: Depth per base
              type:
              - 'null'
              - boolean
              inputBinding:
                prefix: --depth_per_base
                position: 0
                separate: false
                sbg:cmdInclude: true
              description: Report the depth at each base position.
              sbg:category: ''
            - id: '#counts'
              label: Counts
              type:
              - 'null'
              - boolean
              inputBinding:
                prefix: --counts
                position: 0
                separate: false
                sbg:cmdInclude: true
              description: Only report the count of overlaps, don't compute fraction,
                etc.
              sbg:category: ''
            - id: '#split'
              label: Split
              type:
              - 'null'
              - boolean
              inputBinding:
                prefix: --split
                position: 0
                separate: false
                sbg:cmdInclude: true
              description: |-
                Treat BAM entries as distinct BED intervals when computing coverage. Uses CIGAR 'N' and 'D' operations to infer the blocks for computing coverage.
              sbg:category: ''
            - id: '#isbam'
              label: Input is a BAM file
              type:
              - 'null'
              - boolean
              inputBinding:
                prefix: --isbam
                position: 0
                separate: false
                sbg:cmdInclude: true
              description: Input is a BAM file.
              sbg:category: ''

            outputs:
            - id: '#coverage'
              type:
              - 'null'
              - File
              outputBinding:
                glob:
                  class: Expression
                  engine: '#cwl-js-engine'
                  script: |-
                    {
                    	bam = $job.inputs.bam.path
                    	bam_split = bam.split('/')
                        if (bam_split.length > 1) {
                        	bam_split = bam_split[bam_split.length-1].split('.')
                        }
                    	if (bam_split.length > 1)
                    	{
                    		base1 = bam_split.slice(0, bam_split.length-1)
                    	}
                      	bed = $job.inputs.bed.path
                    	bed_split = bed.split('/')
                        if (bed_split.length > 1) {
                        	bed_split = bed_split[bed_split.length-1].split('.')
                        }
                    	if (bed_split.length > 1)
                    	{
                    		base2 = bed_split.slice(0, bed_split.length-1)
                    	}
                        retval = base1.concat(base2).concat('bed')
                        return retval.join('.').replace(/^.*[\\\/]/, '')
                    }
                sbg:inheritMetadataFrom: '#bam'
                sbg:metadata:
                  file_format: BED
            - id: '#summary'
              type:
              - 'null'
              - File
              outputBinding:
                glob:
                  class: Expression
                  engine: '#cwl-js-engine'
                  script: |-
                    {
                    	bam = $job.inputs.bam.path
                    	bam_split = bam.split('/')
                        if (bam_split.length > 1) {
                        	bam_split = bam_split[bam_split.length-1].split('.')
                        }
                    	if (bam_split.length > 1)
                    	{
                    		base1 = bam_split.slice(0, bam_split.length-1)
                    	}
                      	bed = $job.inputs.bed.path
                    	bed_split = bed.split('/')
                        if (bed_split.length > 1) {
                        	bed_split = bed_split[bed_split.length-1].split('.')
                        }
                    	if (bed_split.length > 1)
                    	{
                    		base2 = bed_split.slice(0, bed_split.length-1)
                    	}
                        retval = base1.concat(base2).concat('summary')
                        return retval.join('.').replace(/^.*[\\\/]/, '')
                    }
                sbg:inheritMetadataFrom: '#bam'
                sbg:metadata:
                  file_format: TEXT
            - id: '#per_interval'
              type:
              - 'null'
              - File
              outputBinding:
                glob:
                  class: Expression
                  engine: '#cwl-js-engine'
                  script: |-
                    {
                    	bam = $job.inputs.bam.path
                    	bam_split = bam.split('/')
                        if (bam_split.length > 1) {
                        	bam_split = bam_split[bam_split.length-1].split('.')
                        }
                    	if (bam_split.length > 1)
                    	{
                    		base1 = bam_split.slice(0, bam_split.length-1)
                    	}
                      	bed = $job.inputs.bed.path
                    	bed_split = bed.split('/')
                        if (bed_split.length > 1) {
                        	bed_split = bed_split[bed_split.length-1].split('.')
                        }
                    	if (bed_split.length > 1)
                    	{
                    		base2 = bed_split.slice(0, bed_split.length-1)
                    	}
                        retval = base1.concat(base2).concat('per_interval').concat('bed')
                        return retval.join('.').replace(/^.*[\\\/]/, '')
                    }
                sbg:inheritMetadataFrom: '#bam'
                sbg:metadata:
                  file_format: BED

            baseCommand:
            - python
            - /opt/sbg_coverage.py

            hints:
            - class: sbg:CPURequirement
              value: 1
            - class: sbg:MemRequirement
              value: 7168
            - class: DockerRequirement
              dockerPull: images.sbgenomics.com/tziotas/sbg_coverage:1.0
            id: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow/sbg-coverage/0
            description: |-
              SBG Coverage calculates the depth and breadth of coverage of exons defined in a BED file versus reads defined in a BAM file. It uses BEDTools 2.22.1 and exports the summary file.
            sbg:appVersion:
            - sbg:draft-2
            sbg:categories:
            - Analysis
            sbg:cmdPreview: |-
              python /opt/sbg_coverage.py --bam /test/example_human_Illumina.sorted.bam --bed /test/Agilent_SureSelect_50Mb.b37.bed --strandness strandness --depth_per_base --counts --split --isbam /test/example_human_Illumina.sorted.bam /test/Agilent_SureSelect_50Mb.b37.bed
            sbg:contributors:
            - jelena_randjelovic
            sbg:createdBy: jelena_randjelovic
            sbg:createdOn: 1536672719
            sbg:id: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow/sbg-coverage/0
            sbg:image_url:
            sbg:job:
              allocatedResources:
                cpu: 1
                mem: 7168
              inputs:
                bam:
                  class: File
                  secondaryFiles: []
                  path: /test/example_human_Illumina.sorted.bam
                  size: 0
                bed:
                  class: File
                  secondaryFiles: []
                  path: /test/Agilent_SureSelect_50Mb.b37.bed
                  size: 0
                counts: true
                depth_per_base: depth_per_base
                isbam: true
                split: true
                strandness: strandness
            sbg:latestRevision: 0
            sbg:license: Apache License 2.0
            sbg:modifiedBy: jelena_randjelovic
            sbg:modifiedOn: 1536672719
            sbg:project: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow
            sbg:projectName: PDXnet - Polishing WES JAX workflow
            sbg:publisher: sbg
            sbg:revision: 0
            sbg:revisionNotes:
            sbg:revisionsInfo:
            - sbg:modifiedBy: jelena_randjelovic
              sbg:modifiedOn: 1536672719
              sbg:revision: 0
              sbg:revisionNotes:
            sbg:sbgMaintained: false
            sbg:toolAuthor: Seven Bridges Genomics
            sbg:toolkit: SBGTools
            sbg:validationErrors: []
          inputs:
          - id: '#SBG_Coverage.bam'
            source: '#bam'
          - id: '#SBG_Coverage.bed'
            source: '#bed'
          - id: '#SBG_Coverage.isbam'
            default: true
          outputs:
          - id: '#SBG_Coverage.coverage'
          - id: '#SBG_Coverage.summary'
          - id: '#SBG_Coverage.per_interval'
          sbg:x: 813
          sbg:y: 345
        id: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow/exome-coverage-qc-1-0/1
        description: |-
          Exome Coverage QC calculates histograms, per-base reports and BedGraph summaries of feature coverage (aligned sequences for example) for a given genome. This extended version additionally extracts and creates a text file containing summary coverage stats.
        sbg:appVersion:
        - sbg:draft-2
        sbg:canvas_x: 0
        sbg:canvas_y: 0
        sbg:canvas_zoom: 1
        sbg:categories:
        - Quality-Control
        sbg:contributors:
        - jelena_randjelovic
        sbg:createdBy: jelena_randjelovic
        sbg:createdOn: 1536672721
        sbg:id: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow/exome-coverage-qc-1-0/1
        sbg:image_url:
        sbg:latestRevision: 1
        sbg:license: GNU General Public License v3.0 only
        sbg:modifiedBy: jelena_randjelovic
        sbg:modifiedOn: 1536672766
        sbg:original_source: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow/exome-coverage-qc-1-0/1
        sbg:project: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow
        sbg:projectName: PDXnet - Polishing WES JAX workflow
        sbg:publisher: sbg
        sbg:revision: 1
        sbg:revisionNotes: Copy of admin/sbg-public-data/exome-coverage-qc-1-0/3
        sbg:revisionsInfo:
        - sbg:modifiedBy: jelena_randjelovic
          sbg:modifiedOn: 1536672721
          sbg:revision: 0
          sbg:revisionNotes: Copy of admin/sbg-public-data/exome-coverage-qc-1-0/3
        - sbg:modifiedBy: jelena_randjelovic
          sbg:modifiedOn: 1536672766
          sbg:revision: 1
          sbg:revisionNotes: Copy of admin/sbg-public-data/exome-coverage-qc-1-0/3
        sbg:sbgMaintained: false
        sbg:toolAuthor: Seven Bridges Genomics
        sbg:toolkit: SBGTools
        sbg:toolkitVersion: '1.0'
        sbg:validationErrors: []
        x: 1260.666666666667
        y: 395.68229166666674
      inputs:
      - id: '#Exome_Coverage_QC_1_0.bed'
        source: '#bed'
      - id: '#Exome_Coverage_QC_1_0.bam'
        source: '#GATK_ApplyBQSR.output'
      outputs:
      - id: '#Exome_Coverage_QC_1_0.summary'
      - id: '#Exome_Coverage_QC_1_0.coverage'
      sbg:x: 1260.666666666667
      sbg:y: 395.68229166666674
    - id: '#Picard_CalculateHsMetrics'
      label: Picard CalculateHsMetrics
      in: []
      run:
        cwlVersion: sbg:draft-2
        class: CommandLineTool
        label: Picard CalculateHsMetrics
        $namespaces:
          sbg: https://sevenbridges.com

        requirements:
        - id: '#cwl-js-engine'
          class: ExpressionEngineRequirement
          requirements:
          - class: DockerRequirement
            dockerPull: rabix/js-engine
          engineCommand: cwl-engine.js
        - class: InlineJavascriptRequirement

        inputs:
        - id: '#verbosity'
          label: Verbosity
          type:
          - 'null'
          - name: verbosity
            type: enum
            symbols:
            - ERROR
            - WARNING
            - INFO
            - DEBUG
          inputBinding:
            prefix: VERBOSITY=
            position: 6
            separate: false
            sbg:cmdInclude: true
          description: |-
            Control verbosity of logging. Default value: INFO. This option can be set to 'null' to clear the default value. Possible values: {ERROR, WARNING, INFO, DEBUG}.
          required: false
          sbg:category: Options
          sbg:toolDefaultValue: INFO
        - id: '#validation_stringency'
          label: Validation stringency
          type:
          - 'null'
          - name: validation_stringency
            type: enum
            symbols:
            - STRICT
            - LENIENT
            - SILENT
          inputBinding:
            prefix: VALIDATION_STRINGENCY=
            position: 4
            separate: false
            sbg:cmdInclude: true
          description: |-
            Validation stringency for all SAM files read by this program. Setting stringency to SILENT can improve performance when processing a BAM file in which variable-length data (read, qualities, tags) do not otherwise need to be decoded. Default value: STRICT. This option can be set to 'null' to clear the default value. Possible values: {STRICT, LENIENT, SILENT}.
          required: false
          sbg:category: Options
          sbg:toolDefaultValue: STRICT
        - id: '#target_intervals'
          label: Target intervals
          type:
          - File
          inputBinding:
            prefix: TARGET_INTERVALS=
            position: 0
            separate: false
            sbg:cmdInclude: true
          description: An interval list file that contains the locations of the targets.
          required: true
          sbg:altPrefix: TI
          sbg:category: File inputs
          sbg:fileTypes: TXT
        - id: '#reference'
          label: Reference
          type:
          - File
          inputBinding:
            secondaryFiles:
            - .fai
            - ^.dict
            prefix: R=
            position: 3
            separate: false
            sbg:cmdInclude: true
          description: The reference sequence fasta aligned to.  Required.
          required: true
        - id: '#quiet'
          label: Quiet
          type:
          - 'null'
          - name: quiet
            type: enum
            symbols:
            - 'True'
            - 'False'
          inputBinding:
            prefix: QUIET=
            position: 4
            separate: false
            sbg:cmdInclude: true
          description: |-
            Whether to suppress job-summary info on System.err. Default value: false. This option can be set to 'null' to clear the default value. Possible values: {true, false}.
          required: false
          sbg:category: Options
          sbg:toolDefaultValue: 'False'
        - id: '#per_target_coverage'
          label: Per target coverage
          type:
          - 'null'
          - name: per_target_coverage
            type: enum
            symbols:
            - 'True'
            - 'False'
          description: An optional file to output per target coverage information
            to.
          required: false
          sbg:category: Options
          sbg:toolDefaultValue: 'False'
        - id: '#metric_accumulation_level'
          label: Metric accumulation level
          type:
          - 'null'
          - type: array
            items:
              name: metric_accumulation_level
              type: enum
              symbols:
              - ALL_READS
              - SAMPLE
              - LIBRARY
              - READ_GROUP
          inputBinding:
            prefix: METRIC_ACCUMULATION_LEVEL=
            position: 0
            separate: false
            sbg:cmdInclude: true
          description: |-
            The level(s) at which to accumulate metrics.  Default value: [ALL_READS]. This option can be set to 'null' to clear the default value. Possible values: {ALL_READS, SAMPLE, LIBRARY, READ_GROUP}.
          required: false
          sbg:altPrefix: LEVEL
          sbg:category: Options
          sbg:toolDefaultValue: ALL_READS
        - id: '#memory_per_job'
          label: Memory per job
          type:
          - 'null'
          - int
          description: |-
            Amount of RAM memory to be used per job. Defaults to 2048MB for single threaded jobs.
          required: false
          sbg:category: Execution options
          sbg:toolDefaultValue: '2048'
        - id: '#max_records_in_ram'
          label: Max records in RAM
          type:
          - 'null'
          - int
          inputBinding:
            prefix: MAX_RECORDS_IN_RAM=
            position: 4
            separate: false
            sbg:cmdInclude: true
          description: |-
            When writing SAM files that need to be sorted, this will specify the number of records stored in RAM before spilling to disk. Increasing this number reduces the number of file handles needed to sort a SAM file, and increases the amount of RAM needed. Default value: 500000. This option can be set to 'null' to clear the default value.
          required: false
          sbg:category: Options
          sbg:toolDefaultValue: '500000'
        - id: '#input_bam'
          label: Input
          type:
          - File
          inputBinding:
            prefix: INPUT=
            position: 0
            separate: false
            sbg:cmdInclude: true
          description: An aligned SAM or BAM file.
          required: true
          sbg:altPrefix: I
          sbg:category: File inputs
          sbg:fileTypes: SAM, BAM
        - id: '#compression_level'
          label: Compression level
          type:
          - 'null'
          - int
          inputBinding:
            prefix: COMPRESSION_LEVEL=
            position: 4
            separate: false
            sbg:cmdInclude: true
          description: |-
            Compression level for all compressed files created (e.g. BAM and GELI). Default value: 5. This option can be set to 'null' to clear the default value.
          required: false
          sbg:category: Options
          sbg:toolDefaultValue: '5'
        - id: '#bait_set_name'
          label: Bait set name
          type:
          - 'null'
          - string
          inputBinding:
            prefix: BAIT_SET_NAME=
            position: 0
            separate: false
            sbg:cmdInclude: true
          description: |-
            Bait set name. If not provided it is inferred from the filename of the bait intervals.
          required: false
          sbg:altPrefix: N
          sbg:category: Options
        - id: '#bait_intervals'
          label: Bait intervals
          type:
          - File
          inputBinding:
            prefix: BAIT_INTERVALS=
            position: 0
            separate: false
            sbg:cmdInclude: true
          description: An interval list file that contains the locations of the baits
            used.
          required: true
          sbg:altPrefix: BI
          sbg:category: File inputs
          sbg:fileTypes: TXT

        outputs:
        - id: '#hs_metrics'
          label: HS metrics
          type:
          - 'null'
          - File
          outputBinding:
            glob: '*.hsmetrics.txt'
            sbg:inheritMetadataFrom: '#input_bam'
            sbg:metadata:
              __inherit__: input_file
          description: Output metrics file.
          sbg:fileTypes: TXT
        - id: '#coverage'
          label: Coverage
          type:
          - 'null'
          - File
          outputBinding:
            glob: '*.coverage.bed'
            sbg:inheritMetadataFrom: '#input_bam'
          description: |-
            Per target coverage. An optional file to output per target coverage information to.
          sbg:fileTypes: BED

        baseCommand:
        - java
        - class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {   
              if($job.inputs.memory_per_job){
                return '-Xmx'.concat($job.inputs.memory_per_job, 'M')
              }   
              	return '-Xmx2048M'
            }
        - -jar
        - /opt/picard-tools-1.140/picard.jar
        - CalculateHsMetrics
        arguments:
        - prefix: OUTPUT=
          position: 0
          valueFrom:
            class: Expression
            engine: '#cwl-js-engine'
            script: |
              {
                if ($job.inputs.input_bam)
                {
                filename = $job.inputs.input_bam.path

                  return filename.split('.').slice(0, -1).concat("hsmetrics.txt").join(".").replace(/^.*[\\\/]/, '')
                }
              }
          separate: false
        - prefix: PER_TARGET_COVERAGE=
          position: 0
          valueFrom:
            class: Expression
            engine: '#cwl-js-engine'
            script: |
              {
                if ($job.inputs.per_target_coverage)
                {
                	filename = $job.inputs.input_bam.path

                  return filename.split('.').slice(0, -1).concat("coverage.bed").join(".").replace(/^.*[\\\/]/, '')
                }
              }
          separate: false

        hints:
        - class: sbg:CPURequirement
          value: 1
        - class: sbg:MemRequirement
          value:
            class: Expression
            engine: '#cwl-js-engine'
            script: |-
              {
                if($job.inputs.memory_per_job){
                	return $job.inputs.memory_per_job
                }
                	return 2048
              }
        - class: DockerRequirement
          dockerPull: images.sbgenomics.com/mladenlsbg/picard:1.140
          dockerImageId: eab0e70b6629
        id: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow/picard-calculatehsmetrics/1
        appUrl: |-
          /u/jelena_randjelovic/pdxnet-cnv-testing-area/apps/#jelena_randjelovic/pdxnet-cnv-testing-area/picard-calculatehsmetrics/0
        description: |-
          Calculates a set of Hybrid Selection specific metrics from an aligned SAMor BAM file. If a reference sequence is provided, AT/GC dropout metrics will be calculated, and the PER_TARGET_COVERAGE option can be used to output GC and mean coverage information for every target.
        sbg:appVersion:
        - sbg:draft-2
        sbg:categories:
        - SAM/BAM-Processing
        - Quality-Control
        - Quantification
        sbg:cmdPreview: |-
          java -Xmx2048M -jar /opt/picard-tools-1.140/picard.jar CalculateHsMetrics INPUT=lolic.bam BAIT_INTERVALS=bait_intervals.ext TARGET_INTERVALS=target_intervals.ext OUTPUT=lolic.hsmetrics.txt PER_TARGET_COVERAGE=lolic.coverage.bed
        sbg:contributors:
        - jelena_randjelovic
        sbg:createdBy: jelena_randjelovic
        sbg:createdOn: 1536672727
        sbg:id: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow/picard-calculatehsmetrics/1
        sbg:image_url:
        sbg:job:
          allocatedResources:
            cpu: 1
            mem: 2048
          inputs:
            bait_intervals:
              class: File
              secondaryFiles: []
              path: bait_intervals.ext
              size: 0
            bait_set_name: bait_set_name
            input_bam:
              path: lolic.bam
            memory_per_job: 0
            metric_accumulation_level:
            - ALL_READS
            per_target_coverage: 'True'
            target_intervals:
              class: File
              secondaryFiles: []
              path: target_intervals.ext
              size: 0
        sbg:latestRevision: 1
        sbg:license: MIT License, Apache 2.0 Licence
        sbg:links:
        - id: http://broadinstitute.github.io/picard/
          label: Homepage
        - id: https://github.com/broadinstitute/picard/releases/tag/1.140
          label: Source Code
        - id: http://broadinstitute.github.io/picard/
          label: Wiki
        - id: https://github.com/broadinstitute/picard/zipball/master
          label: Download
        - id: http://broadinstitute.github.io/picard/
          label: Publication
        sbg:modifiedBy: jelena_randjelovic
        sbg:modifiedOn: 1536672770
        sbg:project: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow
        sbg:projectName: PDXnet - Polishing WES JAX workflow
        sbg:publisher: sbg
        sbg:revision: 1
        sbg:revisionNotes: Copy of admin/sbg-public-data/picard-calculatehsmetrics-1-140/2
        sbg:revisionsInfo:
        - sbg:modifiedBy: jelena_randjelovic
          sbg:modifiedOn: 1536672727
          sbg:revision: 0
          sbg:revisionNotes: Copy of admin/sbg-public-data/picard-calculatehsmetrics-1-140/2
        - sbg:modifiedBy: jelena_randjelovic
          sbg:modifiedOn: 1536672770
          sbg:revision: 1
          sbg:revisionNotes: Copy of admin/sbg-public-data/picard-calculatehsmetrics-1-140/2
        sbg:sbgMaintained: false
        sbg:toolAuthor: Broad Institute
        sbg:toolkit: Picard
        sbg:toolkitVersion: '1.140'
        sbg:validationErrors: []
        x: 1048.6748046875005
        y: 471.4627278645834
      inputs:
      - id: '#Picard_CalculateHsMetrics.target_intervals'
        source: '#target_intervals'
      - id: '#Picard_CalculateHsMetrics.reference'
        source: '#reference'
      - id: '#Picard_CalculateHsMetrics.memory_per_job'
        source: '#memory_per_job_2'
      - id: '#Picard_CalculateHsMetrics.input_bam'
        source: '#GATK_ApplyBQSR.output'
      - id: '#Picard_CalculateHsMetrics.bait_intervals'
        source: '#target_intervals'
      outputs:
      - id: '#Picard_CalculateHsMetrics.hs_metrics'
      - id: '#Picard_CalculateHsMetrics.coverage'
      sbg:x: 1048.6748046875005
      sbg:y: 471.4627278645834
    - id: '#GATK_BaseRecalibrator'
      label: GATK BaseRecalibrator
      in: []
      run:
        cwlVersion: sbg:draft-2
        class: CommandLineTool
        label: GATK BaseRecalibrator
        $namespaces:
          sbg: https://sevenbridges.com

        requirements:
        - id: '#cwl-js-engine'
          class: ExpressionEngineRequirement
          requirements:
          - class: DockerRequirement
            dockerPull: rabix/js-engine
        - class: InlineJavascriptRequirement

        inputs:
        - id: '#verbosity'
          label: Control verbosity of logging
          type:
          - 'null'
          - name: 'null'
            type: enum
            symbols:
            - ERROR
            - WARNING
            - INFO
            - DEBUG
          inputBinding:
            prefix: --verbosity
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Control verbosity of logging.  Default value: INFO. Possible values: {ERROR, WARNING,
                                          INFO, DEBUG}.
          required: false
          sbg:altPrefix: -verbosity
          sbg:category: Optional Arguments
        - id: '#use_jdk_inflater'
          label: |-
            Whether to use the JdkInflater (as opposed to IntelInflater)  Default value: false
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --use_jdk_inflater
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Whether to use the JdkInflater (as opposed to IntelInflater)  Default value: false.
                                          Possible values: {true, false}.
          required: false
          sbg:altPrefix: -jdk_inflater
          sbg:category: Optional Arguments
          sbg:stageInput:
        - id: '#use_jdk_deflater'
          label: |-
            Whether to use the JdkDeflater (as opposed to IntelDeflater)  Default value: false
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --use_jdk_deflater
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Whether to use the JdkDeflater (as opposed to IntelDeflater)  Default value: false.
                                          Possible values: {true, false}.
          required: false
          sbg:altPrefix: -jdk_deflater
          sbg:category: Optional Arguments
          sbg:stageInput:
        - id: '#useOriginalQualities'
          label: 'Use the base quality scores from the OQ tag  Default value: false'
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --useOriginalQualities
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Use the base quality scores from the OQ tag  Default value: false. Possible values: {true,
                                          false}.
          required: false
          sbg:altPrefix: -OQ
          sbg:category: Optional Arguments
          sbg:stageInput:
        - id: '#secondsBetweenProgressUpdates'
          label: |-
            Output traversal statistics every time this many seconds elapse  Default value: 10
          type:
          - 'null'
          - float
          inputBinding:
            prefix: --secondsBetweenProgressUpdates
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Output traversal statistics every time this many seconds elapse  Default value: 10.0.
          required: false
          sbg:altPrefix: -secondsBetweenProgressUpdates
          sbg:category: Optional Arguments
        - id: '#sample'
          label: |-
            The name of the sample(s) to keep, filtering out all others  This argument must be
                                          specified at least once
          type:
          - 'null'
          - string
          inputBinding:
            prefix: --sample
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            The name of the sample(s) to keep, filtering out all others  This argument must be
                                          specified at least once. Required.
          required: false
          sbg:altPrefix: -sample
          sbg:category: Conditional Arguments for readFilter
        - id: '#reference'
          label: Reference sequence file  Required
          type:
          - File
          inputBinding:
            secondaryFiles:
            - .fai
            - ^.dict
            prefix: --reference
            position: 0
            separate: true
            loadContents: true
            sbg:cmdInclude: true
          description: Reference sequence file  Required.
          required: true
          sbg:altPrefix: -R
          sbg:category: Required Arguments
          sbg:fileTypes: FASTA, FA
        - id: '#reads'
          label: |-
            BAM/SAM/CRAM file containing reads  This argument must be specified at least once
          type:
          - type: array
            items: File
          inputBinding:
            secondaryFiles:
            - .bai
            prefix: --input
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            BAM/SAM/CRAM file containing reads  This argument must be specified at least once.
                                          Required.
          required: true
          sbg:altPrefix: -I
          sbg:category: Required Arguments
          sbg:fileTypes: CRAM, BAM
          sbg:stageInput:
        - id: '#readValidationStringency'
          label: Validation stringency for all SAM/BAM/CRAM/SRA files read by this
            program
          type:
          - 'null'
          - name: 'null'
            type: enum
            symbols:
            - STRICT
            - LENIENT
            - SILENT
          inputBinding:
            prefix: --readValidationStringency
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Validation stringency for all SAM/BAM/CRAM/SRA files read by this program.  The default
                                          stringency value SILENT can improve performance when processing a BAM file in which
                                          variable-length data (read, qualities, tags) do not otherwise need to be decoded.  Default
                                          value: SILENT. Possible values: {STRICT, LENIENT, SILENT}.
          required: false
          sbg:altPrefix: -VS
          sbg:category: Optional Arguments
        - id: '#readName'
          label: Keep only reads with this read name  Required
          type:
          - 'null'
          - string
          inputBinding:
            prefix: --readName
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Keep only reads with this read name  Required.
          required: false
          sbg:altPrefix: -readName
          sbg:category: Conditional Arguments for readFilter
        - id: '#readIndex'
          label: Indices to use for the read inputs
          type:
          - 'null'
          - string
          inputBinding:
            prefix: --readIndex
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Indices to use for the read inputs. If specified, an index must be provided for every read
                                          input and in the same order as the read inputs. If this argument is not specified, the
                                          path to the index for each input will be inferred automatically.  This argument may be
                                          specified 0 or more times. Default value: null.
          required: false
          sbg:altPrefix: -readIndex
          sbg:category: Optional Arguments
        - id: '#readFilter'
          label: |-
            Read filters to be applied before analysis  This argument may be specified 0 or more
                                          times
          type:
          - 'null'
          - string
          inputBinding:
            prefix: --readFilter
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Read filters to be applied before analysis  This argument may be specified 0 or more
                                          times. Default value: null.
          required: false
          sbg:altPrefix: -RF
          sbg:category: Optional Arguments
        - id: '#quantizing_levels'
          label: 'Number of distinct quality scores in the quantized output  Default
            value: 16'
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --quantizing_levels
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: 'Number of distinct quality scores in the quantized output  Default
            value: 16.'
          required: false
          sbg:altPrefix: -ql
          sbg:category: Optional Arguments
        - id: '#preserve_qscores_less_than'
          label: |-
            Don't recalibrate bases with quality scores less than this threshold (with -bqsr)  Default
                                          value: 6
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --preserve_qscores_less_than
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Don't recalibrate bases with quality scores less than this threshold (with -bqsr)  Default
                                          value: 6.
          required: false
          sbg:altPrefix: -preserveQ
          sbg:category: Optional Arguments
        - id: '#mismatches_default_quality'
          label: 'Default quality for the base mismatches covariate  Default value:
            -1'
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --mismatches_default_quality
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: 'Default quality for the base mismatches covariate  Default
            value: -1.'
          required: false
          sbg:altPrefix: -mdq
          sbg:category: Optional Arguments
        - id: '#mismatches_context_size'
          label: 'Size of the k-mer context to be used for base mismatches  Default
            value: 2'
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --mismatches_context_size
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: 'Size of the k-mer context to be used for base mismatches  Default
            value: 2.'
          required: false
          sbg:altPrefix: -mcs
          sbg:category: Optional Arguments
        - id: '#minimumMappingQuality'
          label: 'Minimum mapping quality to keep (inclusive)  Default value: 10'
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --minimumMappingQuality
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: 'Minimum mapping quality to keep (inclusive)  Default value:
            10.'
          required: false
          sbg:altPrefix: -minimumMappingQuality
          sbg:category: Conditional Arguments for readFilter
        - id: '#minReadLength'
          label: |-
            Keep only reads with length at least equal to the specified value  Default value: 1
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --minReadLength
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Keep only reads with length at least equal to the specified value  Default value: 1.
          required: false
          sbg:altPrefix: -minReadLength
          sbg:category: Conditional Arguments for readFilter
        - id: '#memory_per_job'
          label: Memory Per Job
          type:
          - 'null'
          - int
          description: Memory per job
          required: false
          sbg:stageInput:
        - id: '#memory_overhead_per_job'
          label: Memory Overhead Per Job
          type:
          - 'null'
          - int
          description: Memory Overhead Per Job
          required: false
          sbg:stageInput:
        - id: '#maximum_cycle_value'
          label: 'The maximum cycle value permitted for the Cycle covariate  Default
            value: 500'
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --maximum_cycle_value
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: 'The maximum cycle value permitted for the Cycle covariate  Default
            value: 500.'
          required: false
          sbg:altPrefix: -maxCycle
          sbg:category: Optional Arguments
        - id: '#maximumMappingQuality'
          label: 'Maximum mapping quality to keep (inclusive)  Default value: null'
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --maximumMappingQuality
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: 'Maximum mapping quality to keep (inclusive)  Default value:
            null.'
          required: false
          sbg:altPrefix: -maximumMappingQuality
          sbg:category: Conditional Arguments for readFilter
        - id: '#maxReadLength'
          label: Keep only reads with length at most equal to the specified value  Required
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --maxReadLength
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Keep only reads with length at most equal to the specified
            value  Required.
          required: false
          sbg:altPrefix: -maxReadLength
          sbg:category: Conditional Arguments for readFilter
        - id: '#maxFragmentLength'
          label: |-
            Keep only read pairs with fragment length at most equal to the given value  Default value:
                                          1000000
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --maxFragmentLength
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Keep only read pairs with fragment length at most equal to the given value  Default value:
                                          1000000.
          required: false
          sbg:altPrefix: -maxFragmentLength
          sbg:category: Conditional Arguments for readFilter
        - id: '#low_quality_tail'
          label: |-
            Minimum quality for the bases in the tail of the reads to be considered  Default value: 2
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --low_quality_tail
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Minimum quality for the bases in the tail of the reads to be considered  Default value: 2.
          required: false
          sbg:altPrefix: -lqt
          sbg:category: Optional Arguments
        - id: '#library'
          label: The name of the library to keep  Required
          type:
          - 'null'
          - string
          inputBinding:
            prefix: --library
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: The name of the library to keep  Required.
          required: false
          sbg:altPrefix: -library
          sbg:category: Conditional Arguments for readFilter
        - id: '#lenient'
          label: 'Lenient processing of VCF files  Default value: false'
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --lenient
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Lenient processing of VCF files  Default value: false. Possible values: {true, false}.
          required: false
          sbg:altPrefix: -LE
          sbg:category: Optional Arguments
          sbg:stageInput:
        - id: '#known_sites'
          label: |-
            One or more databases of known polymorphic sites used to exclude regions around known
                                          polymorphisms from analysis
          type:
          - type: array
            items: File
          inputBinding:
            secondaryFiles:
            - .tbi
            position: 0
            valueFrom:
              class: Expression
              engine: '#cwl-js-engine'
              script: "{\n\treturn ''\n}"
            sbg:cmdInclude: true
          description: |-
            One or more databases of known polymorphic sites used to exclude regions around known
                                          polymorphisms from analysis.  This argument must be specified at least once. Required.
          required: true
          sbg:altPrefix: -knownSites
          sbg:category: Required Arguments
          sbg:fileTypes: TXT, BED, VCF
        - id: '#keepReverse'
          label: Keep only reads on the reverse strand  Required
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --keepReverse
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: 'Keep only reads on the reverse strand  Required. Possible
            values: {true, false}.'
          required: false
          sbg:altPrefix: -keepReverse
          sbg:category: Conditional Arguments for readFilter
          sbg:stageInput:
        - id: '#keepReadGroup'
          label: The name of the read group to keep  Required
          type:
          - 'null'
          - string
          inputBinding:
            prefix: --keepReadGroup
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: The name of the read group to keep  Required.
          required: false
          sbg:altPrefix: -keepReadGroup
          sbg:category: Conditional Arguments for readFilter
        - id: '#intervals_string'
          label: Genomic interval strings to include
          type:
          - 'null'
          - string
          inputBinding:
            prefix: --intervals
            position: 0
            valueFrom:
              class: Expression
              engine: '#cwl-js-engine'
              script: |
                {
                    if ($job.inputs.reference && $job.inputs.intervals_string){
                        fasta = $job.inputs.reference.contents
                        interval = $job.inputs.intervals_string
                        // fasta - UCSC format
                        if (fasta.indexOf(">chr") != -1){
                            // INTERVAL - UCSC format, all w/ chr
                            if (interval.indexOf("chr") != -1){
                                return interval
                            }// if (interval.indexOf("chr") == -1)
                            // interval - 1000G format, convert to UCSC
                            else{
                                if (Number(interval) || interval == "X" || interval == "Y"){
                                    return "chr".concat(interval)
                                }// if (Number(interval)|| interval == "X" || interval == "Y")
                                else{
                                    if (interval == "MT"){
                                        return "chrM"
                                    }//if (interval == "MT")
                                    else{
                                        if(Number(interval.substr(2,6))<211){
                                            n = Number(interval.substr(2,6)) - 190;
                                            s = "chr";
                                            s = s.concat(n.toString());
                                            s = s.concat("_gl");
                                            s = s.concat(interval.substr(2,6));
                                            s = s.concat("_random");
                                            return s
                                        }// if(Number(interval.substr(2,interval.length-2))<211)
                                        else{
                                            if(Number(interval.substr(2,6))<250) {
                                                s = "chrUn_gl";
                                                s = s.concat(interval.substr(2,6));
                                                return s
                                            }//if(Number(interval.substr(2,interval.length-2))<250)
                                            else {
                                                return interval
                                            }//if(Number(interval.substr(2,interval.length-2))<250) - else
                                        }// if(Number(interval.substr(2,interval.length-2))<211) - else
                                    }//if (interval == "MT") - else
                                }//if (Number(interval)|| interval == "X" || interval == "Y") - else
                            }//if (interval.indexOf("chr") == -1) - else
                        }//if (fasta.indexOf(">chr") == -1)
                        // fasta - 1000G format
                        else{
                            //interval - USCS format, all w/ chr, convert to 1000G
                            if(interval.indexOf("chr") != -1){
                                if(Number(interval.substr(3,2)) && interval.length<6 || interval == "chrX" || interval == "chrY"){
                                    return interval.substr(3,2)
                                }//if(Number(interval.substr(3,interval.length)) != NaN || interval == "chrX" || bsqr == "chrY")
                                else{
                                    if (interval == "chrM") {
                                        return "MT"
                                    }//if (interval == "chrM")
                                    else{
                                        s = "GL";
                                        s = s.concat(interval.substr(8,6));
                                        s = s.concat(".1");
                                        return s
                                    }//if (interval == "chrM") - else
                                }//if(Number(interval.substr(3,interval.length)) != NaN || interval == "chrX" || bsqr == "chrY") - else
                            }//if($job.inputs.interval.indexOf("chr") == -1)
                            // interval - 1000G format
                            else{
                                return interval
                            }//(interval.indexOf("chr") == -1) - else
                        }//if (fasta.indexOf(">chr") == -1) - else
                    }//if ($job.inputs.fasta && $job.inputs.interval)
                }
            separate: true
            sbg:cmdInclude: true
          description: |-
            One or more genomic intervals over which to operate  This argument may be specified 0 or                              more times. Default value: null.
          required: false
          sbg:altPrefix: -L
          sbg:category: Optional Arguments
        - id: '#intervals_file'
          label: Genomic interval files to include
          type:
          - 'null'
          - File
          inputBinding:
            prefix: --intervals
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            One or more genomic intervals over which to operate  This argument may be specified 0 or
                                          more times. Default value: null.
          required: false
          sbg:altPrefix: -L
          sbg:category: Optional Arguments
          sbg:fileTypes: TXT, BED
        - id: '#interval_set_rule'
          label: 'Set merging approach to use for combining interval inputs  Default
            value: UNION'
          type:
          - 'null'
          - name: 'null'
            type: enum
            symbols:
            - UNION
            - INTERSECTION
          inputBinding:
            prefix: --interval_set_rule
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Set merging approach to use for combining interval inputs  Default value: UNION. Possible
                                          values: {UNION, INTERSECTION}.
          required: false
          sbg:altPrefix: -isr
          sbg:category: Optional Arguments
        - id: '#interval_padding'
          label: Amount of padding (in bp) to add to each interval you are including
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --interval_padding
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Amount of padding (in bp) to add to each interval you are including.  Default value: 0.
          required: false
          sbg:altPrefix: -ip
          sbg:category: Optional Arguments
        - id: '#interval_exclusion_padding'
          label: Amount of padding (in bp) to add to each interval you are excluding
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --interval_exclusion_padding
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Amount of padding (in bp) to add to each interval you are excluding.  Default value: 0.
          required: false
          sbg:altPrefix: -ixp
          sbg:category: Optional Arguments
        - id: '#insertions_default_quality'
          label: 'Default quality for the base insertions covariate  Default value:
            45'
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --insertions_default_quality
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: 'Default quality for the base insertions covariate  Default
            value: 45.'
          required: false
          sbg:altPrefix: -idq
          sbg:category: Optional Arguments
        - id: '#indels_context_size'
          label: |-
            Size of the k-mer context to be used for base insertions and deletions  Default value: 3
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --indels_context_size
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Size of the k-mer context to be used for base insertions and deletions  Default value: 3.
          required: false
          sbg:altPrefix: -ics
          sbg:category: Optional Arguments
        - id: '#filterTooShort'
          label: |-
            Value for which reads with less than this number of aligned bases is considered too short 
                                          Default value: 30
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --filterTooShort
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Value for which reads with less than this number of aligned bases is considered too short 
                                          Default value: 30.
          required: false
          sbg:altPrefix: -filterTooShort
          sbg:category: Conditional Arguments for readFilter
        - id: '#exclude_intervals_string'
          label: Genomic interval strings to exclude
          type:
          - 'null'
          - string
          inputBinding:
            prefix: --excludeIntervals
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            One or more genomic intervals to exclude from processing  This argument may be specified 0                              or more times. Default value: null.
          required: false
          sbg:altPrefix: -XL
          sbg:category: Optional Arguments
          sbg:stageInput:
        - id: '#exclude_intervals_file'
          label: Genomic interval files to exclude
          type:
          - 'null'
          - File
          inputBinding:
            prefix: --excludeIntervals
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            One or more genomic intervals to exclude from processing  This argument may be specified 0
                                          or more times. Default value: null.
          required: false
          sbg:altPrefix: -XL
          sbg:category: Optional Arguments
          sbg:fileTypes: TXT, BED
        - id: '#dontRequireSoftClipsBothEnds'
          label: Allow a read to be filtered out based on having only 1 soft-clipped
            block
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --dontRequireSoftClipsBothEnds
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Allow a read to be filtered out based on having only 1 soft-clipped block. By default,
                                          both ends must have a soft-clipped block, setting this flag requires only 1 soft-clipped
                                          block.  Default value: false. Possible values: {true, false}.
          required: false
          sbg:altPrefix: -dontRequireSoftClipsBothEnds
          sbg:category: Conditional Arguments for readFilter
          sbg:stageInput:
        - id: '#disableToolDefaultReadFilters'
          label: 'Disable all tool default read filters  Default value: false'
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --disableToolDefaultReadFilters
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Disable all tool default read filters  Default value: false. Possible values: {true,
                                          false}.
          required: false
          sbg:altPrefix: -disableToolDefaultReadFilters
          sbg:category: Optional Arguments
          sbg:stageInput:
        - id: '#disableSequenceDictionaryValidation'
          label: |-
            If specified, do not check the sequence dictionaries from our inputs for compatibility
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --disableSequenceDictionaryValidation
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            If specified, do not check the sequence dictionaries from our inputs for compatibility.
                                          Use at your own risk!  Default value: false. Possible values: {true, false}.
          required: false
          sbg:altPrefix: -disableSequenceDictionaryValidation
          sbg:category: Optional Arguments
          sbg:stageInput:
        - id: '#disableReadFilter'
          label: |-
            Read filters to be disabled before analysis  This argument may be specified 0 or more
                                          times
          type:
          - 'null'
          - string
          inputBinding:
            prefix: --disableReadFilter
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Read filters to be disabled before analysis  This argument may be specified 0 or more
                                          times. Default value: null.
          required: false
          sbg:altPrefix: -DF
          sbg:category: Optional Arguments
        - id: '#disableBamIndexCaching'
          label: |-
            If true, don't cache bam indexes, this will reduce memory requirements but may harm
                                          performance if many intervals are specified
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --disableBamIndexCaching
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            If true, don't cache bam indexes, this will reduce memory requirements but may harm
                                          performance if many intervals are specified.  Caching is automatically disabled if there
                                          are no intervals specified.  Default value: false. Possible values: {true, false}.
          required: false
          sbg:altPrefix: -DBIC
          sbg:category: Optional Arguments
          sbg:stageInput:
        - id: '#deletions_default_quality'
          label: 'Default quality for the base deletions covariate  Default value:
            45'
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --deletions_default_quality
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: 'Default quality for the base deletions covariate  Default
            value: 45.'
          required: false
          sbg:altPrefix: -ddq
          sbg:category: Optional Arguments
        - id: '#defaultBaseQualities'
          label: 'Assign a default base quality  Default value: -1'
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --defaultBaseQualities
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: 'Assign a default base quality  Default value: -1.'
          required: false
          sbg:altPrefix: -DBQ
          sbg:category: Optional Arguments
        - id: '#createOutputVariantMD5'
          label: If true, create a a MD5 digest any VCF file created
          type:
          - 'null'
          - float
          inputBinding:
            prefix: --createOutputVariantMD5
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            If true, create a a MD5 digest any VCF file created.  Default value: false. Possible
                                          values: {true, false}.
          required: false
          sbg:altPrefix: -OVM
          sbg:category: Optional Arguments
          sbg:stageInput:
        - id: '#createOutputVariantIndex'
          label: If true, create a VCF index when writing a coordinate-sorted VCF
            file
          type:
          - 'null'
          - float
          inputBinding:
            prefix: --createOutputVariantIndex
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            If true, create a VCF index when writing a coordinate-sorted VCF file.  Default value:
                                          true. Possible values: {true, false}.
          required: false
          sbg:altPrefix: -OVI
          sbg:category: Optional Arguments
          sbg:stageInput:
        - id: '#createOutputBamMD5'
          label: |-
            If true, create a MD5 digest for any BAM/SAM/CRAM file created  Default value: false
          type:
          - 'null'
          - float
          inputBinding:
            prefix: --createOutputBamMD5
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            If true, create a MD5 digest for any BAM/SAM/CRAM file created  Default value: false.
                                          Possible values: {true, false}.
          required: false
          sbg:altPrefix: -OBM
          sbg:category: Optional Arguments
          sbg:stageInput:
        - id: '#createOutputBamIndex'
          label: If true, create a BAM/CRAM index when writing a coordinate-sorted
            BAM/CRAM file
          type:
          - 'null'
          - float
          inputBinding:
            prefix: --createOutputBamIndex
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            If true, create a BAM/CRAM index when writing a coordinate-sorted BAM/CRAM file.  Default
                                          value: true. Possible values: {true, false}.
          required: false
          sbg:altPrefix: -OBI
          sbg:category: Optional Arguments
          sbg:stageInput:
        - id: '#cloudPrefetchBuffer'
          label: Size of the cloud-only prefetch buffer (in MB; 0 to disable)
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --cloudPrefetchBuffer
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Size of the cloud-only prefetch buffer (in MB; 0 to disable).  Default value: 40.
          required: false
          sbg:altPrefix: -CPB
          sbg:category: Optional Arguments
        - id: '#cloudIndexPrefetchBuffer'
          label: Size of the cloud-only prefetch buffer (in MB; 0 to disable)
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --cloudIndexPrefetchBuffer
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Size of the cloud-only prefetch buffer (in MB; 0 to disable). Defaults to
                                          cloudPrefetchBuffer if unset.  Default value: -1.
          required: false
          sbg:altPrefix: -CIPB
          sbg:category: Optional Arguments
        - id: '#bqsrBAQGapOpenPenalty'
          label: BQSR BAQ gap open penalty (Phred Scaled)
          type:
          - 'null'
          - float
          inputBinding:
            prefix: --bqsrBAQGapOpenPenalty
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            BQSR BAQ gap open penalty (Phred Scaled).  Default value is 40.  30 is perhaps better for
                                          whole genome call sets  Default value: 40.0.
          required: false
          sbg:altPrefix: -bqsrBAQGOP
          sbg:category: Optional Arguments
        - id: '#blackListedLanes'
          label: |-
            Keep reads with platform units not on the list  This argument must be specified at least
                                          once
          type:
          - 'null'
          - string
          inputBinding:
            prefix: --blackListedLanes
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Keep reads with platform units not on the list  This argument must be specified at least
                                          once. Required.
          required: false
          sbg:altPrefix: -blackListedLanes
          sbg:category: Conditional Arguments for readFilter
        - id: '#blackList'
          label: This argument must be specified at least once
          type:
          - 'null'
          - string
          inputBinding:
            prefix: --blackList
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: This argument must be specified at least once. Required.
          required: false
          sbg:altPrefix: -blackList
          sbg:category: Conditional Arguments for readFilter
        - id: '#binary_tag_name'
          label: 'The binary tag covariate name if using it  Default value: null'
          type:
          - 'null'
          - string
          inputBinding:
            prefix: --binary_tag_name
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: 'The binary tag covariate name if using it  Default value:
            null.'
          required: false
          sbg:altPrefix: -bintag
          sbg:category: Optional Arguments
        - id: '#ambigFilterFrac'
          label: Threshold fraction of non-regular bases (e
          type:
          - 'null'
          - float
          inputBinding:
            prefix: --ambigFilterFrac
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Threshold fraction of non-regular bases (e.g. N) above which to filter  Default value:
                                          0.05.
          required: false
          sbg:altPrefix: -ambigFilterFrac
          sbg:category: Conditional Arguments for readFilter
        - id: '#ambigFilterBases'
          label: Ambig Filter Bases
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --ambigFilterBases
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Threshold number of ambiguous bases. If null, uses threshold fraction; otherwise, overrides threshold fraction. Cannot be used in conjunction with argument(s) maxAmbiguousBaseFraction.
          required: false
          sbg:category: Conditional Arguments for readFilter
          sbg:stageInput:
        - id: '#addOutputSAMProgramRecord'
          label: If true, adds a PG tag to created SAM/BAM/CRAM files
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --addOutputSAMProgramRecord
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            If true, adds a PG tag to created SAM/BAM/CRAM files.  Default value: true. Possible
                                          values: {true, false}.
          required: false
          sbg:altPrefix: -addOutputSAMProgramRecord
          sbg:category: Optional Arguments
          sbg:stageInput:
        - id: '#QUIET'
          label: Whether to suppress job-summary info on System
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --QUIET
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Whether to suppress job-summary info on System.err.  Default value: false. Possible
                                          values: {true, false}.
          required: false
          sbg:category: Optional Arguments
          sbg:stageInput:
        - id: '#PLFilterName'
          label: |-
            Keep reads with RG:PL attribute containing this string  This argument must be specified at
                                          least once
          type:
          - 'null'
          - string
          inputBinding:
            prefix: --PLFilterName
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Keep reads with RG:PL attribute containing this string  This argument must be specified at
                                          least once. Required.
          required: false
          sbg:altPrefix: -PLFilterName
          sbg:category: Conditional Arguments for readFilter

        outputs:
        - id: '#recalibration_file'
          label: BQSR Table
          type:
          - File
          outputBinding:
            glob: '*.recal_data.grp'
            sbg:inheritMetadataFrom: '#reads'
          description: The output recalibration table file to create.
          sbg:fileTypes: GRP
        - id: '#performance_summary'
          label: Performance Summary
          type:
          - 'null'
          - File
          outputBinding:
            glob: '*.txt'
            sbg:inheritMetadataFrom: '#reads'
          description: Test

        baseCommand:
        - /opt/gatk-launch
        - --javaOptions
        - class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              if($job.inputs.memory_per_job){
              	return '\"-Xmx'.concat($job.inputs.memory_per_job, 'M') + '\"'
              }
              	return '\"-Xmx2048M\"'
            }
        - BaseRecalibrator
        arguments:
        - prefix: ''
          position: 0
          valueFrom:
            class: Expression
            engine: '#cwl-js-engine'
            script: |-
              {
                read_name = [].concat($job.inputs.reads)[0].path.replace(/^.*[\\\/]/, '').split('.')
                read_namebase = read_name.slice(0, read_name.length-1).join('.')
                return '--output ' + read_namebase + '.recal_data.grp'
              }
          separate: true
        - position: 0
          valueFrom: --createOutputBamIndex
          separate: true
        - position: 0
          valueFrom:
            class: Expression
            engine: '#cwl-js-engine'
            script: |-
              {
                known_sites = $job.inputs.known_sites

                cmd = []

                for (i = 0; i < known_sites.length; i++) {
                  cmd.push('--knownSites', known_sites[i].path)
                }

                return cmd.join(' ')
              }
          separate: true

        hints:
        - class: sbg:CPURequirement
          value: 1
        - class: sbg:MemRequirement
          value:
            class: Expression
            engine: '#cwl-js-engine'
            script: |-
              {
                if($job.inputs.memory_per_job){
                  if($job.inputs.memory_overhead_per_job){
                  	return $job.inputs.memory_per_job + $job.inputs.memory_overhead_per_job
                  }
                  else
                		return $job.inputs.memory_per_job
                }
                else if(!$job.inputs.memory_per_job && $job.inputs.memory_overhead_per_job){
              		return 2048 + $job.inputs.memory_overhead_per_job  
                }
                else
                	return 2048
              }
        - class: DockerRequirement
          dockerPull: images.sbgenomics.com/teodora_aleksic/gatk:4.beta.2
          dockerImageId: 3c3b8e0ed4e5
        id: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow/gatk-baserecalibrator/3
        appUrl: |-
          /u/jelena_randjelovic/pdxnet-polishing-wes-jax-workflow/apps/#jelena_randjelovic/pdxnet-polishing-wes-jax-workflow/gatk-baserecalibrator/3
        description: |-
          First pass of the base quality score recalibration -- Generates recalibration table based on various covariates (such as read group, reported quality score, machine cycle, and nucleotide context).

          ###**Overview**  

          This walker is designed to work as the first pass in a two-pass processing step. It does a by-locus traversal operating only at sites that are not in dbSNP. We assume that all reference mismatches we see are therefore errors and indicative of poor base quality. This walker generates tables based on various user-specified covariates (such as read group, reported quality score, cycle, and context). Since there is a large amount of data one can then calculate an empirical probability of error given the particular covariates seen at this site, where p(error) = num mismatches / num observations. The output file is a table (of the several covariate values, num observations, num mismatches, empirical quality score).

          Note: ReadGroupCovariate and QualityScoreCovariate are required covariates and will be added for the user regardless of whether or not they were specified.

          ###**Inputs**  

          The input read data whose base quality scores need to be assessed.
          A database of known polymorphic sites to skip over.

          ###**Output**  

          A GATK Report file with many tables:

          - The list of arguments
          - The quantized qualities table
          - The recalibration table by read group
          - The recalibration table by quality score
          - The recalibration table for all the optional covariates

          The GATK Report is intended to be easy to read by humans or computers. Check out the documentation of the GATKReport to learn how to manipulate this table.

          ###**Usage example**  

               ./gatk-launch BaseRecalibrator \ 
               	 -R reference.fasta \ 
                   -I input.bam \   
                   --knownSites latest_dbsnp.vcf \  
                   -O recalibration.table 

          ###**IMPORTANT NOTICE**  

          Tools in GATK that require a fasta reference file also look for the reference file's corresponding *.fai* (fasta index) and *.dict* (fasta dictionary) files. The fasta index file allows random access to reference bases and the dictionary file is a dictionary of the contig names and sizes contained within the fasta reference. These two secondary files are essential for GATK to work properly. To append these two files to your fasta reference please use the '***SBG FASTA Indices***' tool within your GATK based workflow before using any of the GATK tools.
        sbg:appVersion:
        - sbg:draft-2
        sbg:categories:
        - GATK-4.0
        sbg:cmdPreview: |-
          /opt/gatk-launch --javaOptions "-Xmx4M" BaseRecalibrator --reference /path/to/reference.ext --input /path/to/reads-1.ext --input /path/to/reads-2.ext  --output reads-1.recal_data.grp  --createOutputBamIndex  --knownSites /path/to/known_sites-1.ext --knownSites /path/to/known_sites-2.ext
        sbg:contributors:
        - jelena_randjelovic
        sbg:createdBy: jelena_randjelovic
        sbg:createdOn: 1536672723
        sbg:id: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow/gatk-baserecalibrator/3
        sbg:image_url:
        sbg:job:
          allocatedResources:
            cpu: 1
            mem: 5
          inputs:
            QUIET: true
            addOutputSAMProgramRecord: true
            ambigFilterBases: 2
            cpu_per_job: 2
            createOutputBamIndex: 6.802465483383213
            createOutputBamMD5: 1.7560880637573613
            createOutputVariantIndex: 5.081587263149359
            createOutputVariantMD5: 0.6317428673703886
            disableBamIndexCaching: true
            disableSequenceDictionaryValidation: true
            disableToolDefaultReadFilters: true
            dontRequireSoftClipsBothEnds: true
            exclude_intervals_file:
              class: File
              secondaryFiles: []
              path: /path/to/exclude_intervals_file.ext
              size: 0
            exclude_intervals_string: exclude_intervals_string-string-value
            intervals_file:
              class: File
              secondaryFiles: []
              path: /path/to/intervals_file.ext
              size: 0
            intervals_string: intervals_string-string-value
            keepReverse: true
            known_sites:
            - class: File
              secondaryFiles: []
              path: /path/to/known_sites-1.ext
              size: 0
            - class: File
              secondaryFiles: []
              path: /path/to/known_sites-2.ext
              size: 0
            lenient: true
            memory_overhead_per_job: 1
            memory_per_job: 4
            no_cmdline_in_header: true
            reads:
            - class: File
              secondaryFiles: []
              path: /path/to/reads-1.ext
              size: 0
            - class: File
              secondaryFiles: []
              path: /path/to/reads-2.ext
              size: 0
            reference:
              class: File
              secondaryFiles:
              - path: .fai
              - path: ^.dict
              path: /path/to/reference.ext
              size: 0
            secondsBetweenProgressUpdates: 1
            threads_per_job: 2
            useOriginalQualities: true
            use_jdk_deflater: true
            use_jdk_inflater: true
        sbg:latestRevision: 3
        sbg:license: Open source BSD (3-clause) license
        sbg:links:
        - id: https://software.broadinstitute.org/gatk/index.php
          label: Homepage
        - id: https://github.com/broadinstitute/gatk
          label: Source Code
        - id: |-
            https://gatkforums.broadinstitute.org/gatk/discussion/9881/howto-get-started-with-gatk4-beta
          label: Download
        - id: https://software.broadinstitute.org/gatk/gatkdocs/4.beta.2/
          label: Documentation
        sbg:modifiedBy: jelena_randjelovic
        sbg:modifiedOn: 1537514967
        sbg:project: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow
        sbg:projectName: PDXnet - Polishing WES JAX workflow
        sbg:publisher: sbg
        sbg:revision: 3
        sbg:revisionNotes: revert to 1
        sbg:revisionsInfo:
        - sbg:modifiedBy: jelena_randjelovic
          sbg:modifiedOn: 1536672723
          sbg:revision: 0
          sbg:revisionNotes:
        - sbg:modifiedBy: jelena_randjelovic
          sbg:modifiedOn: 1536672768
          sbg:revision: 1
          sbg:revisionNotes:
        - sbg:modifiedBy: jelena_randjelovic
          sbg:modifiedOn: 1537426568
          sbg:revision: 2
          sbg:revisionNotes: cleared intervals expression
        - sbg:modifiedBy: jelena_randjelovic
          sbg:modifiedOn: 1537514967
          sbg:revision: 3
          sbg:revisionNotes: revert to 1
        sbg:sbgMaintained: false
        sbg:toolAuthor: Broad Institute
        sbg:toolkit: GATK 4.0
        sbg:toolkitVersion: Beta 2
        sbg:validationErrors: []
        x: 903.3333333333336
        y: 130.34895833333337
      inputs:
      - id: '#GATK_BaseRecalibrator.reference'
        source: '#reference'
      - id: '#GATK_BaseRecalibrator.reads'
        source:
        - '#Picard_MarkDuplicates.deduped_bam'
      - id: '#GATK_BaseRecalibrator.known_sites'
        source:
        - '#known_sites'
      - id: '#GATK_BaseRecalibrator.intervals_string'
        default: chr20
        source: '#intervals_string'
      outputs:
      - id: '#GATK_BaseRecalibrator.recalibration_file'
      - id: '#GATK_BaseRecalibrator.performance_summary'
      sbg:x: 903.3333333333336
      sbg:y: 130.34895833333337
    - id: '#SBG_Split_Pair_by_Metadata_1'
      label: SBG Split Pair by Metadata
      in: []
      run:
        cwlVersion: sbg:draft-2
        class: CommandLineTool
        label: SBG Split Pair by Metadata
        $namespaces:
          sbg: https://sevenbridges.com

        requirements:
        - id: '#cwl-js-engine'
          class: ExpressionEngineRequirement
          requirements:
          - class: DockerRequirement
            dockerPull: rabix/js-engine
        - class: InlineJavascriptRequirement

        inputs:
        - id: '#metadata_criteria'
          label: Metadata criteria
          type:
          - name: metadata_criteria
            type: record
            fields:
            - label: Output 1
              name: output_1
              type:
              - 'null'
              - string
              description: Output 1 metadta name:value pair.
              sbg:category: Execution
            - label: Output 2
              name: output_2
              type:
              - 'null'
              - string
              description: Output 2 metadta name:value pair.
              sbg:category: Execution
          description: Metadata name:value pair for splitting the pair.
          sbg:category: Execution
        - id: '#input_pair'
          label: Input pair
          type:
          - type: array
            items: File
          description: Input pair to be split on two outputs.
          required: true
          sbg:category: File Input
          sbg:stageInput: link

        outputs:
        - id: '#output_files_2'
          label: Output Files 2
          type:
          - 'null'
          - type: array
            items: File
          outputBinding:
            outputEval:
              class: Expression
              engine: '#cwl-js-engine'
              script: |-
                {
                 //return $job.inputs.metadata_criteria
                  outlist = []
                  
                  if ($job.inputs.metadata_criteria.output_2){
                    
                    parts = $job.inputs.metadata_criteria.output_2.split(':')
                    
                    if (parts.length < 2){
                      
                      return ''
                      
                    }
                    
                    for (i=0;i<$job.inputs.input_pair.length;i++){
                    
                    
                    	if ($job.inputs.input_pair[i].metadata[parts[0]] == parts[1]) {
                      
                      		outlist.push($job.inputs.input_pair[i])
                      
                    	}
                    }
                    
                  }
                  
                  return outlist
                }
            sbg:inheritMetadataFrom: '#input_pair'
          description: Output files 2 based on the appropriate metadata.
        - id: '#output_files_1'
          label: Output files 1
          type:
          - 'null'
          - type: array
            items: File
          outputBinding:
            outputEval:
              class: Expression
              engine: '#cwl-js-engine'
              script: |-
                {
                 //return $job.inputs.metadata_criteria
                  outlist = []
                  
                  if ($job.inputs.metadata_criteria.output_1){
                    
                    parts = $job.inputs.metadata_criteria.output_1.split(':')
                    
                    if (parts.length < 2){
                      
                      return ''
                      
                    }
                    
                    for (i=0;i<$job.inputs.input_pair.length;i++){
                    
                    
                    	if ($job.inputs.input_pair[i].metadata[parts[0]] == parts[1]) {
                      
                      		outlist.push($job.inputs.input_pair[i])
                      
                    	}
                    }
                    
                  }
                  
                  return outlist
                }
            sbg:inheritMetadataFrom: '#input_pair'
          description: Output files 1 based on the appropriate metadata.

        baseCommand: []

        hints:
        - class: sbg:CPURequirement
          value: 1
        - class: sbg:MemRequirement
          value: 1000
        - class: DockerRequirement
          dockerPull: ubuntu:14.04
        id: |-
          jelena_randjelovic/pdxnet-polishing-wes-jax-workflow/sbg-split-pair-by-metadata/0
        description: |-
          This tool is a "junction" tool sorting files from a single array input into multiple arrays. It shouldn't be used as a single tool.
          Because of the general purpose the input files do not have any set requirements and they depend on the use case and pipeline. The splitting criteria is given in the "key:value" form, and each output contains all the files that have metadat[key]==value.
          Built-in metadat fields are all written in lower case with underscores instead of spaces (Sample type is "sample_type"), while custom metadata fields can be written with any desired string.
        sbg:appVersion:
        - sbg:draft-2
        sbg:cmdPreview: ''
        sbg:contributors:
        - jelena_randjelovic
        sbg:createdBy: jelena_randjelovic
        sbg:createdOn: 1536672753
        sbg:id: |-
          jelena_randjelovic/pdxnet-polishing-wes-jax-workflow/sbg-split-pair-by-metadata/0
        sbg:image_url:
        sbg:job:
          allocatedResources:
            cpu: 1
            mem: 1000
          inputs:
            input_pair:
            - class: File
              secondaryFiles: []
              metadata:
                sample_type: Primary Tumor
              path: /path/to/input_pair-1.ext
              size: 0
            - class: File
              secondaryFiles: []
              metadata:
                sample_type: nada
              path: /path/to/input_pair-2.ext
              size: 0
            - class: File
              secondaryFiles: []
              metadata:
                sample_type: nada
              path: akajsdhkasdksj
              size: 0
            - class: File
              secondaryFiles: []
              metadata:
                sample_type: stagod
              path: alksjdlaksdlk
              size: 0
            metadata_criteria:
              output_1: paired_end:1
              output_2: paired_end:2
        sbg:latestRevision: 0
        sbg:license: Internal
        sbg:modifiedBy: jelena_randjelovic
        sbg:modifiedOn: 1536672753
        sbg:project: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow
        sbg:projectName: PDXnet - Polishing WES JAX workflow
        sbg:publisher: sbg
        sbg:revision: 0
        sbg:revisionNotes:
        sbg:revisionsInfo:
        - sbg:modifiedBy: jelena_randjelovic
          sbg:modifiedOn: 1536672753
          sbg:revision: 0
          sbg:revisionNotes:
        sbg:sbgMaintained: false
        sbg:toolAuthor: Seven Bridges
        sbg:validationErrors: []
        x: 262.66666666666674
        y: 90.66666666666669
      inputs:
      - id: '#SBG_Split_Pair_by_Metadata_1.metadata_criteria'
        default:
          output_1: paired_end:1
          output_2: paired_end:2
      - id: '#SBG_Split_Pair_by_Metadata_1.input_pair'
        source:
        - '#Trimmomatic___JAX_settings.output_trimmed'
      outputs:
      - id: '#SBG_Split_Pair_by_Metadata_1.output_files_2'
      - id: '#SBG_Split_Pair_by_Metadata_1.output_files_1'
      sbg:x: 229.0550537109375
      sbg:y: 95.88302612304688
    - id: '#XenomeV2'
      label: XenomeV2
      in: []
      run:
        cwlVersion: sbg:draft-2
        class: CommandLineTool
        label: XenomeV2
        $namespaces:
          sbg: https://sevenbridges.com

        requirements:
        - id: '#cwl-js-engine'
          class: ExpressionEngineRequirement
          requirements:
          - class: DockerRequirement
            dockerPull: rabix/js-engine
        - class: InlineJavascriptRequirement

        inputs:
        - id: '#indices_name'
          type:
          - 'null'
          - string
          inputBinding:
            prefix: -P
            position: 0
            valueFrom:
              class: Expression
              engine: '#cwl-js-engine'
              script: '"hg38_broad_NOD_based_on_mm10_k25"'
            separate: true
            sbg:cmdInclude: true
          sbg:toolDefaultValue: fastq
        - id: '#index_file'
          label: Xenome Indices
          type:
          - File
          required: true
          sbg:stageInput: copy
        - id: '#fastq_reverse'
          type:
          - File
          inputBinding:
            prefix: -i
            position: 5
            separate: true
            sbg:cmdInclude: true
          required: true
        - id: '#fastq_forward'
          type:
          - File
          inputBinding:
            prefix: -i
            position: 4
            separate: true
            sbg:cmdInclude: true
          required: true
          sbg:fileTypes: fastq

        outputs:
        - id: '#stats_file'
          type:
          - 'null'
          - File
          outputBinding:
            glob:
              class: Expression
              engine: '#cwl-js-engine'
              script: '"Sample.Xenome.Classification.Stats.txt"'
            sbg:inheritMetadataFrom: '#fastq_forward'
        - id: '#human_reverse'
          label: Human Specific Fastq files
          type:
          - 'null'
          - File
          outputBinding:
            glob:
              class: Expression
              engine: '#cwl-js-engine'
              script: '"*human_2.fastq"'
            sbg:inheritMetadataFrom: '#fastq_reverse'
        - id: '#human_forward'
          type:
          - 'null'
          - File
          outputBinding:
            glob:
              class: Expression
              engine: '#cwl-js-engine'
              script: '"*human_1.fastq"'
            sbg:inheritMetadataFrom: '#fastq_forward'
        stdout: Sample.Xenome.Classification.Stats.txt

        baseCommand:
        - tar
        - -xzvf
        - class: Expression
          engine: '#cwl-js-engine'
          script: $job.inputs.index_file.path
        - --strip-components
        - 1;
        - xenome
        - classify
        arguments:
        - prefix: -T
          position: 0
          valueFrom: '15'
          separate: true
        - prefix: --pairs
          position: 1
          valueFrom:
            class: Expression
            engine: '#cwl-js-engine'
            script: '""'
          separate: true
        - prefix: --host-name
          position: 2
          valueFrom: 'mouse  '
          separate: true
        - prefix: --graft-name
          position: 3
          valueFrom: human
          separate: true

        hints:
        - class: sbg:AWSInstanceType
          value: c5.4xlarge;ebs-gp2;300
        - class: sbg:CPURequirement
          value: 15
        - class: sbg:MemRequirement
          value: 30000
        - class: DockerRequirement
          dockerPull: pgc-images.sbgenomics.com/d3b-bixu/xenome2
        id: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow/xenomev2/2
        sbg:appVersion:
        - sbg:draft-2
        sbg:cmdPreview: |-
          tar -xzvf /path/to/index_file.ext --strip-components 1; xenome classify  -T 15 --preserve-read-order  --host-name mouse   --graft-name human -i /path/to/input.ext -i /path/to/input.ext > Sample.Xenome.Classification.Stats.txt
        sbg:contributors:
        - jelena_randjelovic
        sbg:createdBy: jelena_randjelovic
        sbg:createdOn: 1536672725
        sbg:id: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow/xenomev2/2
        sbg:image_url:
        sbg:job:
          allocatedResources:
            cpu: 15
            mem: 30000
          inputs:
            fastq_forward:
              class: File
              secondaryFiles: []
              contents: file contents
              path: /path/to/input.ext
              size: 0
            fastq_reverse:
              class: File
              secondaryFiles: []
              contents: file contents
              path: /path/to/input.ext
              size: 0
            index_file:
              class: File
              secondaryFiles: []
              path: /path/to/index_file.ext
              size: 0
            indices_name: indices_name-string-value
        sbg:latestRevision: 2
        sbg:modifiedBy: jelena_randjelovic
        sbg:modifiedOn: 1537691411
        sbg:project: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow
        sbg:projectName: PDXnet - Polishing WES JAX workflow
        sbg:publisher: sbg
        sbg:revision: 2
        sbg:revisionNotes: 300 GB storage
        sbg:revisionsInfo:
        - sbg:modifiedBy: jelena_randjelovic
          sbg:modifiedOn: 1536672725
          sbg:revision: 0
          sbg:revisionNotes: Copy of anuj_srivastava/pdx-wes-pipeline/xenome/15
        - sbg:modifiedBy: jelena_randjelovic
          sbg:modifiedOn: 1537377399
          sbg:revision: 1
          sbg:revisionNotes: increased T to 15
        - sbg:modifiedBy: jelena_randjelovic
          sbg:modifiedOn: 1537691411
          sbg:revision: 2
          sbg:revisionNotes: 300 GB storage
        sbg:sbgMaintained: false
        sbg:validationErrors: []
        x: 416.7936197916668
        y: 3.8512776692708344
      inputs:
      - id: '#XenomeV2.index_file'
        source: '#index_file_1'
      - id: '#XenomeV2.fastq_reverse'
        source: '#SBG_Split_Pair_by_Metadata_1.output_files_2'
      - id: '#XenomeV2.fastq_forward'
        source: '#SBG_Split_Pair_by_Metadata_1.output_files_1'
      outputs:
      - id: '#XenomeV2.stats_file'
      - id: '#XenomeV2.human_reverse'
      - id: '#XenomeV2.human_forward'
      sbg:x: 380.5894470214844
      sbg:y: 3.176605463027954
    - id: '#Trimmomatic___JAX_settings'
      label: Trimmomatic - JAX settings
      in: []
      run:
        cwlVersion: sbg:draft-2
        class: CommandLineTool
        label: Trimmomatic - JAX settings
        $namespaces:
          sbg: https://sevenbridges.com

        requirements:
        - id: '#cwl-js-engine'
          class: ExpressionEngineRequirement
          requirements:
          - class: DockerRequirement
            dockerPull: rabix/js-engine
        - class: InlineJavascriptRequirement
        - class: CreateFileRequirement
          fileDef:
          - fileContent:
              class: Expression
              engine: '#cwl-js-engine'
              script: |-
                { 	if($job.inputs.illuminaClip){
                  		var x = [].concat($job.inputs.illuminaClip.adapterList);
                    	return x.join("\n"); 
                	}

                }
            filename: adapterList.ext

        inputs:
        - id: '#reads'
          label: Fastq Files
          type:
          - type: array
            items: File
          inputBinding:
            position: 1
            valueFrom:
              class: Expression
              engine: '#cwl-js-engine'
              script: |
                { 
                	var list = [].concat($job.inputs.reads)
                        
                  	if (list.length == 1){ 
                    	return list[0].path;   
                  	} 
                  
                  	if (list.length == 2){  
                      	if(list[0].metadata){
                          
                          	if(list[0].metadata.paired_end==1){
                          		return list[0].path + ' ' + list[1].path;
                      		}
                          
                        	if(list[0].metadata.paired_end==2){
                            	return list[1].path + ' ' + list[0].path;
                      		}
                    	}
                	}
                }    
            sbg:cmdInclude: true
          description: Fastq files to be trimmed.
          sbg:category: Input files
          sbg:fileTypes: |-
            FASTA, FASTQ, FA, FQ, FA.GZ, FA.BZ2, FQ.GZ, FQ.BZ2, FASTA.GZ, FASTA.BZ2, FASTQ.GZ, FASTQ.BZ2
        - id: '#leading'
          label: Leading quality threshold
          type:
          - 'null'
          - int
          inputBinding:
            position: 6
            valueFrom:
              class: Expression
              engine: '#cwl-js-engine'
              script: |-
                {
                	if(typeof $job.inputs.leading  == 'undefined' ||  $job.inputs.leading ==null){
                  		return 'LEADING:3';
                    }else{
                  		return 'LEADING:' + $job.inputs.leading .toString();
                    }
                }
            separate: true
            sbg:cmdInclude: true
          description: Remove low quality bases from the beginning.
          sbg:category: Processing steps
          sbg:toolDefaultValue: '3'
        - id: '#trailing'
          label: Trailing quality threshold
          type:
          - 'null'
          - int
          inputBinding:
            position: 7
            valueFrom:
              class: Expression
              engine: '#cwl-js-engine'
              script: |-
                {if(typeof $job.inputs.trailing  == 'undefined' ||  $job.inputs.trailing ==null){
                  	return 'TRAILING:3';
                }
                 else{
                   return 'TRAILING:' + $job.inputs.trailing .toString();
                 }
                }
            separate: true
            sbg:cmdInclude: true
          description: |-
            As long as a base has a value below this threshold the base is removed and the next base.
          sbg:category: Processing steps
          sbg:toolDefaultValue: '3'
        - id: '#minLen'
          label: Minimal length
          type:
          - 'null'
          - int
          inputBinding:
            position: 9
            valueFrom:
              class: Expression
              engine: '#cwl-js-engine'
              script: |-
                {
                	if(typeof $job.inputs.minLen  == 'undefined' ||  $job.inputs.minLen ==null){
                  		return 'MINLEN:36';
                	}else{
                 		return 'MINLEN:' + $job.inputs.minLen.toString()
                	}
                }
            separate: true
            sbg:cmdInclude: true
          description: This module removes reads that fall below the specified minimal
            length.
          sbg:category: Processing steps
        - id: '#crop'
          label: Crop reads
          type:
          - 'null'
          - int
          inputBinding:
            prefix: 'CROP:'
            position: 10
            separate: false
            sbg:cmdInclude: true
          description: The number of bases to keep, from the start of the read.
          sbg:category: Processing steps
          sbg:toolDefaultValue: None
        - id: '#headCrop'
          label: Removes the specified number of bases
          type:
          - 'null'
          - int
          inputBinding:
            prefix: 'HEADCROP:'
            position: 10
            separate: false
            sbg:cmdInclude: true
          description: The number of bases to remove from the start of the read.
          sbg:category: Processing steps
          sbg:toolDefaultValue: None
        - id: '#maxInfo'
          label: Adaptive quality trim
          type:
          - 'null'
          - name: maxInfo
            type: record
            fields:
            - label: Target length
              name: targetLength
              type:
              - 'null'
              - int
              inputBinding:
                position: 11
                valueFrom:
                  class: Expression
                  engine: '#cwl-js-engine'
                  script: |-
                    {
                    	var s = ''; 
                      
                      	if($job.inputs.maxInfo.targetLength!=null){
                          
                      		s = s + $job.inputs.maxInfo.targetLength.toString(); 
                       		if($job.inputs.maxInfo.strictness!=null){
                      
                         		s = s + ':' + $job.inputs.maxInfo.strictness.toFixed(1).toString(); 
                         		return 'MAXINFO:' + s; 
                      		}
                      	}
                      
                      
                      return ''; 
                    }
                separate: true
                sbg:cmdInclude: 'true'
              description: |-
                This specifies the read length which is likely to allow the location of the read within the target sequence to be determined.
              sbg:category: Processing steps - Max Info
              sbg:stageInput:
              sbg:toolDefaultValue: None
            - label: Strictness
              name: strictness
              type:
              - 'null'
              - float
              description: |-
                This value, which should be set between 0 and 1, specifies the balance between preserving as much read length as possible vs. removal of incorrect bases. A low value of this parameter (<0.2) favours longer reads, while a high value (>0.8) favours read correctness.
              sbg:category: Processing steps - Max Info
              sbg:stageInput:
              sbg:toolDefaultValue: None
          inputBinding:
            position: 11
            separate: true
            sbg:cmdInclude: 'true'
          description: |-
            Performs an adaptive quality trim, balancing the benefits of retaining longer reads against the costs of retaining bases with errors.
          sbg:category: Processing steps - Max Info
          sbg:toolDefaultValue: None
        - id: '#qualityEncoding'
          label: Quality Encoding
          type:
          - 'null'
          - name: qualityEncoding
            type: enum
            symbols:
            - FASTQ file to base 33
            - FASTQ file to base 64
          inputBinding:
            position: 12
            valueFrom:
              class: Expression
              engine: '#cwl-js-engine'
              script: |-
                {

                  if($job.inputs.qualityEncoding == 'FASTQ file to base 33'){
                  	return '-phred33'; 
                  }
                  
                  if($job.inputs.qualityEncoding == 'FASTQ file to base 64'){
                  	return '-phred64'; 
                  }
                  
                }
            separate: true
            sbg:cmdInclude: true
          description: Quality encoding. If not selected, it is detected automatically.
          sbg:category: Processing steps
          sbg:toolDefaultValue: None
        - id: '#avgQual'
          label: Average Quality
          type:
          - 'null'
          - int
          inputBinding:
            prefix: 'AVGQUAL:'
            position: 9
            separate: false
            sbg:cmdInclude: true
          description: Drop the read if the average quality is below the specified
            level.
          sbg:category: Processing steps
          sbg:stageInput:
          sbg:toolDefaultValue: None
        - id: '#noCPUs'
          label: No. CPUs
          type:
          - 'null'
          - int
          description: Number of CPUs to use.
          sbg:category: Input parameters
          sbg:toolDefaultValue: '8'

        outputs:
        - id: '#output_trimmed'
          label: Output Files - Trimmed
          type:
          - 'null'
          - type: array
            items: File
          outputBinding:
            glob:
              class: Expression
              engine: '#cwl-js-engine'
              script: |
                {
                  reads = [].concat($job.inputs.reads);
                    
                  return (reads.length==2) ? '*_trimmed_paired.*' : '*_trimmed.*'
                  
                }
            sbg:inheritMetadataFrom: '#reads'
            sbg:metadata:
              paired_end:
                class: Expression
                engine: '#cwl-js-engine'
                script: |-
                  {
                    reads = [].concat($job.inputs.reads);
                    if (reads.length==1){ 
                      return 1;
                    }
                    
                    filename = $self.path.split("/").pop().split("_trimmed")[0];
                    inputname= reads[0].path.split("/").pop().slice(0,filename.length);
                    
                    if(filename == inputname){
                      return reads[0].metadata.paired_end;
                    }else{
                      return reads[1].metadata.paired_end;
                    }
                  }
          description: Trimmed FASTQ output reads.
          sbg:fileTypes: FASTA, FASTQ, FA, FQ, FASTQ.GZ, FQ.GZ, FASTQ.BZ2, FQ.BZ2
        - id: '#unpaired_output'
          label: Unpaired files
          type:
          - 'null'
          - type: array
            items: File
          outputBinding:
            glob:
              class: Expression
              engine: '#cwl-js-engine'
              script: "{return '*_trimmed_unpaired.*' }\n\n"
            sbg:inheritMetadataFrom: '#reads'
            sbg:metadata:
              paired_end:
                class: Expression
                engine: '#cwl-js-engine'
                script: |-
                  {
                    reads = [].concat($job.inputs.reads);

                    if (reads.length==1){ 
                      return 1;
                    }
                    
                    filename = $self.path.split("/").pop().split("_trimmed")[0];
                    inputname= reads[0].path.split("/").pop().slice(0,filename.length);
                    
                    if(filename == inputname){
                      return reads[0].metadata.paired_end;
                    }else{
                      return reads[1].metadata.paired_end;
                    }
                  }
          description: Trimmed FASTQ output reads.
          sbg:fileTypes: FASTA, FASTQ, FA, FQ, FASTQ.GZ, FQ.GZ, FASTQ.BZ2, FQ.BZ2
        - id: '#log_file'
          label: Log file
          type:
          - 'null'
          - File
          outputBinding:
            glob: '*.txt'
            sbg:inheritMetadataFrom: '#reads'
          description: Trimlog file creates a log of all read trimmings.
          sbg:fileTypes: .TXT

        baseCommand:
        - java
        - -jar
        - /opt/trimmomatic-0.36.jar
        - class: Expression
          engine: '#cwl-js-engine'
          script: |-
            { // PE or SE	
              	reads = [].concat($job.inputs.reads);
              
              	return  (reads.length==2) ? 'PE' : 'SE';
            } 
        - -threads
        - $(nproc)
        - class: Expression
          engine: '#cwl-js-engine'
          script: |+
            { // Log file name as read name without extension
              
              	var s = [].concat($job.inputs.reads)[0].path.split('/').pop()
                
            	s = s.replace(/.FASTA/i,'').replace(/.GZ/i,'')
                s = s.replace(/.FASTQ/i,'').replace(/.FQ/i,'').replace(/.FA/i,'');

              	return '-trimlog ' + s + '_logFile.txt';
            }




        arguments:
        - position: 3
          valueFrom:
            class: Expression
            engine: '#cwl-js-engine'
            script: |-
              { // Output name of paired_end 1 file 
                 
                var ext = ['FASTA', 'FASTQ', 'FA', 'FQ'];
                var list = [].concat($job.inputs.reads)
                var s = list[0].path.split('/').pop().split('.'); 
                
                if(list.length == 2){
                  if(list[0].metadata){
                    if(list[0].metadata.paired_end==1 && list[1].metadata.paired_end==2){
                        var s = list[0].path.split('/').pop().split('.'); 
                    }
                    if(list[0].metadata.paired_end==2 && list[1].metadata.paired_end==1){
                        var s = list[1].path.split('/').pop().split('.');
                    }
                  }
                }  

                if(!s){
                  return ''; 
                }
                
                if (ext.indexOf(s[s.length-1].toUpperCase())!=-1){ //check if gz extention
                		var i=1;
                }else{
                  	var i=2;
                }  
                
                if(list.length == 1){
                  return s.slice(0,-i).join('.') + '_trimmed.' + s.slice(-i).join('.');
                }  
                  
                if(list.length == 2) {
                  return s.slice(0,-i).join('.') + '_trimmed_paired.'   + s.slice(-i).join('.') + ' ' + s.slice(0,-i).join('.') + '_trimmed_unpaired.' + s.slice(-i).join('.');
                }
                
              }
          separate: true
        - position: 4
          valueFrom:
            class: Expression
            engine: '#cwl-js-engine'
            script: |-
              {// Output name of paired_end 2 file (if exists)
                 
                var ext = ['FASTA', 'FASTQ', 'FA', 'FQ'];
                var list = [].concat($job.inputs.reads) 
                
                if(list.length == 2){
                  if(list[0].metadata){
                    if(list[0].metadata.paired_end==1 && list[1].metadata.paired_end==2){
                        var s = list[1].path.split('/').pop().split('.'); 
                    }
                    if(list[0].metadata.paired_end==2 && list[1].metadata.paired_end==1){
                        var s = list[0].path.split('/').pop().split('.');
                    }
                  }
                  
                  if(!s){
                  	return ''; 
                	}
                  
                  
                  if (ext.indexOf(s[s.length-1].toUpperCase())!=-1) { //check if gz extention
                    	var i=1;
              	} else {
                    	var i=2;
              	}  

                  return s.slice(0,-i).join('.') + '_trimmed_paired.'   + s.slice(-i).join('.') + ' ' 
                  	 + s.slice(0,-i).join('.') + '_trimmed_unpaired.' + s.slice(-i).join('.');
               }  

                
                  
              }
               
          separate: true

        hints:
        - class: sbg:AWSInstanceType
          value: c5.4xlarge;ebs-gp2;400
        - class: sbg:CPURequirement
          value:
            class: Expression
            engine: '#cwl-js-engine'
            script: "{\n  if(!$job.inputs.noCPUs){\n  return 8; \n  }\n  \n  return\
              \ $job.inputs.noCPUs\n}"
        - class: sbg:MemRequirement
          value: 30000
        - class: DockerRequirement
          dockerPull: images.sbgenomics.com/svetozar_nesic/trimmomatic:v1
        id: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow/trimmomatic-public/2
        description: |-
          **Trimmomatic** performs a variety of useful trimming tasks for illumina paired-end and single ended data. It works with FASTQ files (using phred + 33 or phred + 64 quality scores, depending on the Illumina pipeline used), either uncompressed or gzipped. Use of gzip format is determined based on the .gz extension automatically.

          There are two major modes of the program: **paired-end** mode and **single-end** mode. The **paired-end** mode will maintain correspondence of read pairs and also use the additional information contained in paired reads to better find adapter or PCR primer fragments introduced by the library preparation process [1].

          *A list of **all inputs and parameters** with corresponding descriptions can be found at the bottom of the page.*

          ### Common Use Cases

          For single-ended data, one input FASTQ file is specified and one output file is returned. For paired-end data, two input FASTQ files are specified, and 4 output files are returned, 2 for the 'paired' output where both reads survived the processing, and 2 for corresponding 'unpaired' output where a read survived, but the partner read did not.

          Most important trimming options are:

          * **Fasta with adapters**: Cut adapter and other illumina-specific sequences from the read.
          * **Sliding window**: Perform a sliding window trimming, cutting once the average quality within the window falls below a threshold.
          * **Leading quality threshold**: Cut bases off the start of a read, if below a threshold quality
          * **Trailing quality threshold**: Cut bases off the end of a read, if below a threshold quality
          * **Crop reads**: Cut the read to a specified length
          * **Head crop reads**: Cut the specified number of bases from the start of the read
          * **Minimum length**: Drop the read if it is below a specified length
          * **Average Quality**: Drop the read if the average quality is below a specified level


          ### Changes Introduced by Seven Bridges

          * Adapters could be supplied as a file (**File With Adapters**), list of strings (**Array of Adapters**) or selected from a list of known adapters(**Fasta files with Adapters**, `ILLUMINACLIP`). When the file is provided, the list of strings and the list of known adapters are ignored. When the list of strings is provided, the list of known adapters is ignored.

          * Output filenames are the same as the input filenames with the addition of the '\_trimmed' suffix prior to the file extension for single-end files. In case of paired-end files, the '\_trimmed\_paired' suffix is added to both filenames, while unpaired read files have the '\_trimmed\_unpaired' suffix added prior to the file extension.


          ### Common Issues and Important Notes

          * For paired-end read files, it is important to properly set the **Paired-end** metadata field on your read files.

          ### Performance Benchmarking 

          Trimming is not a computationally heavy procedure (as alignment is). Therefore, all benchmarking was performed on the default c5.2xlarge (8 cpus, 16Gb RAM) instance (AWS), while using the maximum number of threads the instance allows, showing that the tool benefits from multithreading.  

          | Input size [Gb] | Paired-end | Duration [min] |  Cost [$] |   Instance  |
          |:---------------:|:----------:|:--------------:|:------------:|:-----------:|
          |       210      |     Yes    |       670       |     6     | c5.2xlarge  |
          |       17.4      |     Yes    |       54       |     0.5     | c5.2xlarge  |
          |       4.5       |     Yes    |       16       |     0.15     |  c5.2xlarge |
          |       0.2       |     Yes    |        3       |     0.04     |  c5.2xlarge |

          *Cost can be significantly reduced by using **spot instances**. Visit the [Knowledge Center](https://docs.sevenbridges.com/docs/about-spot-instances) for more details.*

          ### References

          [1] [Trimmomatic paper](https://www.ncbi.nlm.nih.gov/pubmed/24695404)
        sbg:appVersion:
        - sbg:draft-2
        sbg:categories:
        - FASTQ-Processing
        - Quality-Control
        sbg:cmdPreview: |-
          java -jar /opt/trimmomatic-0.36.jar PE -threads $(nproc) -trimlog sample_pe1_logFile.txt /path/to/sample_pe1.fq /path/to/sample_pe2.fq  sample_pe1_trimmed_paired.fq sample_pe1_trimmed_unpaired.fq  sample_pe2_trimmed_paired.fq sample_pe2_trimmed_unpaired.fq
        sbg:contributors:
        - jelena_randjelovic
        sbg:createdBy: jelena_randjelovic
        sbg:createdOn: 1537842340
        sbg:id: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow/trimmomatic-public/2
        sbg:image_url:
        sbg:job:
          allocatedResources:
            cpu: 6
            mem: 30000
          inputs:
            avgQual:
            crop:
            headCrop:
            leading:
            maxInfo:
              strictness:
              targetLength:
            minLen: 36
            noCPUs: 6
            qualityEncoding:
            reads:
            - class: File
              secondaryFiles: []
              metadata:
                paired_end: '1'
              path: /path/to/sample_pe1.fq
              size: 5000
            - class: File
              secondaryFiles: []
              metadata:
                paired_end: '2'
              path: /path/to/sample_pe2.fq
              size: 0
            trailing:
        sbg:latestRevision: 2
        sbg:license: GNU General Public License v3.0 only
        sbg:links:
        - id: http://www.usadellab.org/cms/index.php?page=trimmomatic
          label: Homepage
        - id: |-
            http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/Trimmomatic-Src-0.36.zip
          label: Source code
        - id: |-
            http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/TrimmomaticManual_V0.32.pdf
          label: Manual
        - id: https://www.ncbi.nlm.nih.gov/pubmed/24695404
          label: Publication
        - id: |-
            http://www.usadellab.org/cms/uploads/supplementary/Trimmomatic/Trimmomatic-Src-0.36.zip
          label: Download
        - id: https://wiki.gacrc.uga.edu/wiki/Trimmomatic
          label: Wiki
        sbg:modifiedBy: jelena_randjelovic
        sbg:modifiedOn: 1538755729
        sbg:project: jelena_randjelovic/pdxnet-polishing-wes-jax-workflow
        sbg:projectName: PDXnet - Polishing WES JAX workflow
        sbg:publisher: sbg
        sbg:revision: 2
        sbg:revisionNotes: set mem to 30000
        sbg:revisionsInfo:
        - sbg:modifiedBy: jelena_randjelovic
          sbg:modifiedOn: 1537842340
          sbg:revision: 0
          sbg:revisionNotes: Copy of admin/sbg-public-data/trimmomatic/17
        - sbg:modifiedBy: jelena_randjelovic
          sbg:modifiedOn: 1537844256
          sbg:revision: 1
          sbg:revisionNotes: c5.4xlarge; 400 GB; JAX commandline settings
        - sbg:modifiedBy: jelena_randjelovic
          sbg:modifiedOn: 1538755729
          sbg:revision: 2
          sbg:revisionNotes: set mem to 30000
        sbg:sbgMaintained: false
        sbg:toolAuthor: |-
          Bolger A. M., Lohse M.,  Usadel B /  Max Planck Institute of Molecular Plant Physiology, Am Mühlenberg & RWTH Aachen & Forschungszentrum Jülich
        sbg:toolkit: Trimmomatic
        sbg:toolkitVersion: '0.36'
        sbg:validationErrors: []
      inputs:
      - id: '#Trimmomatic___JAX_settings.reads'
        source:
        - '#reads'
      - id: '#Trimmomatic___JAX_settings.leading'
        source: '#leading'
      - id: '#Trimmomatic___JAX_settings.trailing'
        source: '#trailing'
      outputs:
      - id: '#Trimmomatic___JAX_settings.output_trimmed'
      - id: '#Trimmomatic___JAX_settings.unpaired_output'
      - id: '#Trimmomatic___JAX_settings.log_file'
      sbg:x: 26.666666666666675
      sbg:y: 92.00000000000003
    - id: '#qc_integrate_with_cut_off_final'
      label: QC_Integrate_with_cut-off_final
      in: []
      run:
        cwlVersion: sbg:draft-2
        class: CommandLineTool
        label: QC_Integrate_with_cut-off_final
        $namespaces:
          sbg: https://sevenbridges.com

        requirements:
        - id: '#cwl-js-engine'
          class: ExpressionEngineRequirement
          requirements:
          - class: DockerRequirement
            dockerPull: rabix/js-engine
        - class: InlineJavascriptRequirement
        - class: CreateFileRequirement
          fileDef:
          - fileContent: |-
              #!/usr/bin/perl
              use strict;
              use warnings;

              open(FILEIN1, $ARGV[0]) || die "cannot open the $ARGV[0]";####Xenome Report
              open(FILEIN2, $ARGV[1]) || die "cannot open the $ARGV[1]";####Picard Report
              open(FILEIN3, $ARGV[2]) || die "cannot open the $ARGV[2]";####Target Coverage

              my $flag = 0;

              while(my $readFile1 = <FILEIN1>)
              {
               if(($readFile1 =~ /^\s*count.*?percent.*?class.*$/) && ($flag == 0))
               {
                  $flag = 1;
                  print "####Xenome Report####\n";
                  print "####$readFile1";
                  next;
               }
               elsif($flag == 1)
               {

                 print "$readFile1";
                   
               }

              }


               $flag = 0;

              my @header = ();

              while(my $readFile2 = <FILEIN2>)
              {
               if(($readFile2 =~ /^LIBRARY/) && ($flag == 0))
               {
                  $flag = 1;
                  print "\n\n####Picard MarkDuplicate Report####\n";
                  @header = split("\t", $readFile2);
                  #print "$#header\n";
                  #print "$readFile2";
                  next;
               }
               elsif($flag == 1)
               {
                 $flag = 2;
                 next;    
               }
               elsif($flag == 2)
               {
              my $i= 0;
                  my @array = split("\t", $readFile2);

                  for(; $i <=$#array-1; $i++)
                  {
                     print "$header[$i]\t$array[$i]\n";

                  }
                  if($i == $#array)
                  {
                    chomp $header[$i];
                    print "$header[$i]\t$array[$i]"
                  }

              $flag = 3;
               }

              }

              print "\n\n####Picard Hs metrics  Report####\n";


              $flag = 1;

              my @header2 = ();

              while(my $readFile3 = <FILEIN3>)
              {
               if(($readFile3 =~ /^##\s+METRICS CLASS.*$/) && ($flag==1)) 
               {
                  $flag = 2;
                  next;
               }
               elsif($flag == 2)
               {
              #print "$readFile3";
                    @header2 = split("\t",$readFile3);
                    $flag =3;
                    next;
                }
               elsif($flag==3)
                {
                     my @fields = split("\t",$readFile3);

              #print ">>$#fields<<\n";
              #print "$readFile3";

              $flag = 4;
                    for(my $j=0; $j <=$#header2; $j++)
                    {
                          if($header2[$j] =~ /MEAN_TARGET_COVERAGE|ZERO_CVG_TARGETS_PCT|PCT_TARGET_BASES_2X|PCT_TARGET_BASES_10X|PCT_TARGET_BASES_20X|PCT_TARGET_BASES_30X|PCT_TARGET_BASES_40X|PCT_TARGET_BASES_50X|PCT_TARGET_BASES_100X/)
                           {
                             print "$header2[$j]\t$fields[$j]\n";
                           }
                          if($header2[$j] =~ /PCT_TARGET_BASES_20X.*$/)
                           {
                              
                              my $coverage = $fields[$j];
                               
                               if($coverage < $ARGV[3])
                                 {
                                   print "Target coverage is less than given cut-off of   $ARGV[3] % : The target coverage is $coverage .\n";
                                 }

                           }
                    }

                }
              }
            filename: QC_report.pl

        inputs:
        - id: '#XenomeReport'
          type:
          - File
          inputBinding:
            position: 1
            separate: true
            sbg:cmdInclude: true
          required: true
        - id: '#PicardMarkDupReport'
          type:
          - File
          inputBinding:
            position: 2
            separate: true
            sbg:cmdInclude: true
          required: true
        - id: '#PicardHSMetrics'
          type:
          - File
          inputBinding:
            position: 3
            separate: true
            sbg:cmdInclude: true
          required: true
        - id: '#TargetCoverage_CutOff'
          label: TargetCoverage
          type:
          - 'null'
          - float
          inputBinding:
            position: 4
            valueFrom:
              class: Expression
              engine: '#cwl-js-engine'
              script: |-
                { if ($job.inputs.Cut_Off == undefined)
                {
                  return 0.75
                }
                 
                 else
                 {
                   return $job.inputs.Cut_Off
                 }

                }
            separate: true
            sbg:cmdInclude: true
          description: |-
            values between (0 and 1; default is 0.75 which means 75% of target region is covered with 20 reads or more)
          sbg:toolDefaultValue: '0.75'

        outputs:
        - id: '#integrated_QC_report'
          type:
          - 'null'
          - File
          outputBinding:
            glob: '*integrated_QC_report.txt'
            sbg:inheritMetadataFrom: '#XenomeReport'
        - id: '#error_log'
          label: Error logs
          type:
          - 'null'
          - type: array
            items: File
          outputBinding:
            glob: '*.log'
            outputEval:
              class: Expression
              engine: '#cwl-js-engine'
              script: |-
                {n=false;e=0;j=0;for(i in $self){if($self[i].path.includes('exit_code.log'))e=i;if ($self[i].path.includes('job.err.log'))j=i;};if(!(parseInt($self[e].contents) === 0))n=true;if(n==true){s='\n\n#####     ERROR!     #####\n';throw s + $self[j].contents};return $self;}
            loadContents: true
          description: Exit code log and error log
          sbg:fileTypes: LOG

        baseCommand:
        - perl
        - QC_report.pl
        arguments:
        - prefix: '>'
          position: 5
          valueFrom:
            class: Expression
            engine: '#cwl-js-engine'
            script: |-
              {
                if (($job.inputs.PicardMarkDupReport.metadata) && ($job.inputs.PicardMarkDupReport.metadata['sample_id']))
                {
                  return $job.inputs.PicardMarkDupReport.metadata['sample_id'].concat('_integrated_QC_report.txt')
                }
                else
                {
                  return "Sample.Integrated.QC.report.txt"
                }
              }
          separate: true
        - prefix: ;
          position: 10
          valueFrom: echo $? > exit_code.log
          separate: true

        hints:
        - class: sbg:AWSInstanceType
          value: c5.2xlarge;ebs-gp2;100
        - class: sbg:CPURequirement
          value: 1
        - class: sbg:MemRequirement
          value: 1000
        - class: DockerRequirement
          dockerPull: perl
        id: michael_lloyd/mantis/qc-integrate-with-cut-off-final/3
        sbg:appVersion:
        - sbg:draft-2
        sbg:cmdPreview: |-
          perl QC_report.pl  /path/to/StatsFIle.ext  /path/to/PicardMarkDupReport.ext  /path/to/ExomeQCReport.ext > Sample.Integrated.QC.report.txt ; echo $? > exit_code.log
        sbg:content_hash: a83fd4695b7d96d7bf8db7c52fbc50e8501b62ddc6b60b812386904032481caaf
        sbg:contributors:
        - michael_lloyd
        sbg:createdBy: michael_lloyd
        sbg:createdOn: 1566838882
        sbg:id: michael_lloyd/mantis/qc-integrate-with-cut-off-final/3
        sbg:image_url:
        sbg:job:
          allocatedResources:
            cpu: 1
            mem: 1000
          inputs:
            Cut_Off: 3.8166036384162147
            PicardHSMetrics:
              class: File
              secondaryFiles: []
              path: /path/to/ExomeQCReport.ext
              size: 0
            PicardMarkDupReport:
              class: File
              secondaryFiles: []
              path: /path/to/PicardMarkDupReport.ext
              size: 0
            XenomeReport:
              class: File
              secondaryFiles: []
              path: /path/to/StatsFIle.ext
              size: 0
        sbg:latestRevision: 3
        sbg:modifiedBy: michael_lloyd
        sbg:modifiedOn: 1566919498
        sbg:project: michael_lloyd/mantis
        sbg:projectName: MSI, HRD, TMB
        sbg:publisher: sbg
        sbg:revision: 3
        sbg:revisionNotes: updated grob
        sbg:revisionsInfo:
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1566838882
          sbg:revision: 0
          sbg:revisionNotes: |-
            Copy of michael_lloyd/variant-calling-decomposition/qc-integrate-with-cut-off-final/0
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1566839024
          sbg:revision: 1
          sbg:revisionNotes: added dynamic naming
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1566839224
          sbg:revision: 2
          sbg:revisionNotes: removed 'die' from script. Instead it prints a warning
            to the ouput
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1566919498
          sbg:revision: 3
          sbg:revisionNotes: updated grob
        sbg:sbgMaintained: false
        sbg:validationErrors: []
      inputs:
      - id: '#qc_integrate_with_cut_off_final.XenomeReport'
        source: '#XenomeV2.stats_file'
      - id: '#qc_integrate_with_cut_off_final.PicardMarkDupReport'
        source: '#Picard_MarkDuplicates.metrics_file'
      - id: '#qc_integrate_with_cut_off_final.PicardHSMetrics'
        source: '#Picard_CalculateHsMetrics.hs_metrics'
      - id: '#qc_integrate_with_cut_off_final.TargetCoverage_CutOff'
        source: '#TargetCoverage_CutOff'
      outputs:
      - id: '#qc_integrate_with_cut_off_final.integrated_QC_report'
      - id: '#qc_integrate_with_cut_off_final.error_log'
      sbg:x: 1192.875732421875
      sbg:y: -40.26350784301758
    - id: '#bwa_alt_aware_v3'
      label: bwa-alt-aware_v3
      in: []
      run:
        cwlVersion: sbg:draft-2
        class: CommandLineTool
        label: bwa-alt-aware_v3
        $namespaces:
          sbg: https://sevenbridges.com

        requirements:
        - id: '#cwl-js-engine'
          class: ExpressionEngineRequirement
          requirements:
          - class: DockerRequirement
            dockerPull: rabix/js-engine
        - class: InlineJavascriptRequirement

        inputs:
        - id: '#threads'
          label: Number of threads
          type:
          - 'null'
          - int
          inputBinding:
            prefix: -t
            position: 1
            valueFrom:
              class: Expression
              engine: '#cwl-js-engine'
              script: |-
                {
                  if ($job.inputs.threads)
                  {
                    return $job.inputs.threads
                  }
                  else
                  {
                    return 15
                  }
                }
            separate: true
            sbg:cmdInclude: true
          description: Number of threads.
        - id: '#reverse_fastq'
          label: Reverse FASTQ file
          type:
          - File
          inputBinding:
            position: 6
            separate: true
            sbg:cmdInclude: true
          description: Reverse FASTQ file.
          required: true
          sbg:fileTypes: FASTQ, FASTQ.GZ, FQ, FQ.GZ
        - id: '#read_group'
          label: Read group
          type:
          - 'null'
          - string
          inputBinding:
            prefix: -R
            position: 2
            valueFrom:
              class: Expression
              engine: '#cwl-js-engine'
              script: |-
                {
                  if ($job.inputs.read_group)
                  {
                    return $job.inputs.read_group
                  }
                  else
                  {
                    return "\'\@RG\\tID:1\\tPL:Illumina\\tSM:ExomeSample'"
                  }
                }
            separate: true
            sbg:cmdInclude: true
          description: Read group.
          sbg:toolDefaultValue: "\"\\'\\@RG\\\\tID:1\\\\tPL:Illumina\\\\tSM:ExomeSample'\""
        - id: '#prefix_bwa'
          label: Prefix BWA
          type:
          - 'null'
          - string
          inputBinding:
            prefix: -H
            position: 4
            valueFrom:
              class: Expression
              engine: '#cwl-js-engine'
              script: |-
                {
                  if ($job.inputs.prefix_bwa)
                  {
                    return $job.inputs.prefix_bwa
                  }
                  else
                  {
                    return "hs38DH.fa"
                  }
                }
            separate: true
            sbg:cmdInclude: true
          description: Prefix BWA.
          sbg:toolDefaultValue: hs38DH.fa
        - id: '#outfile_name'
          label: Output file name
          type:
          - 'null'
          - string
          inputBinding:
            prefix: -o
            position: 3
            valueFrom:
              class: Expression
              engine: '#cwl-js-engine'
              script: |-
                {
                  test = [].concat($job.inputs.forward_fastq)
                  if ($job.inputs.outfile_name)
                  {
                    return $job.inputs.outfile_name
                  }
                  else if ((test[0].metadata) && (test[0].metadata['sample_id']) && (test[0].metadata['case_id']))
                  {
                    return test[0].metadata['case_id'].concat('_',test[0].metadata['sample_id'].concat("_out_bwa.bam"))
                  }
                  else if ((test[0].metadata) && (test[0].metadata['sample_id']))
                  {
                    return test[0].metadata['sample_id'].concat("_out_bwa.bam")
                  }
                    else if ((test[0].metadata) && (test[0].metadata['case_id']))
                  {
                    return test[0].metadata['case_id'].concat("_normal_out_bwa.bam")
                  }  
                  else
                  {
                    return "out_bwa.bam"
                  }
                }
            separate: true
            sbg:cmdInclude: true
          description: Output file name.
          sbg:toolDefaultValue: out_bwa.bam
        - id: '#index_file'
          label: BWA_indices
          type:
          - File
          required: true
        - id: '#forward_fastq'
          label: Forward FASTQ file
          type:
          - File
          inputBinding:
            position: 5
            separate: true
            sbg:cmdInclude: true
          description: Forward FASTQ file.
          required: true
          sbg:fileTypes: FASTQ, FASTQ.GZ, FQ, FQ.GZ

        outputs:
        - id: '#bam_output'
          label: Aligned BAM output
          type:
          - 'null'
          - File
          outputBinding:
            glob: '*.bam'
            sbg:inheritMetadataFrom: '#forward_fastq'
            sbg:metadata:
              case_id:
                class: Expression
                engine: '#cwl-js-engine'
                script: |-
                  {
                    test = [].concat($job.inputs.forward_fastq)
                    if ((test[0].metadata) && (test[0].metadata['case_id']))
                    {
                      return test[0].metadata['case_id']
                    }
                    else
                    {
                      return ''
                    }
                  }
              sample_id:
                class: Expression
                engine: '#cwl-js-engine'
                script: |-
                  {
                    test = [].concat($job.inputs.forward_fastq)
                    if ((test[0].metadata) && (test[0].metadata['sample_id']))
                    {
                      return test[0].metadata['sample_id']
                    }
                    else
                    {
                      return ''
                    }
                  }
              sample_type:
                class: Expression
                engine: '#cwl-js-engine'
                script: |-
                  {
                    test = [].concat($job.inputs.forward_fastq)
                    if ((test[0].metadata) && (test[0].metadata['sample_type']))
                    {
                      return test[0].metadata['sample_type']
                    }
                    else
                    {
                      return ''
                    }
                  }
          description: Aligned BAM output.
          sbg:fileTypes: BAM

        baseCommand:
        - tar
        - -xzvf
        - class: Expression
          engine: '#cwl-js-engine'
          script: $job.inputs.index_file.path
        - --strip-components
        - '1'
        - ;
        - run-bwamem
        arguments:
        - position: 1000
          valueFrom: '|sh'
          separate: true

        hints:
        - class: sbg:AWSInstanceType
          value: c5.4xlarge;ebs-gp2;500
        - class: sbg:CPURequirement
          value:
            class: Expression
            engine: '#cwl-js-engine'
            script: |-
              {
                if ($job.inputs.threads)
                {
                  return $job.inputs.threads
                }
                else
                {
                  return 15
                }
              }
        - class: sbg:MemRequirement
          value: 5000
        - class: DockerRequirement
          dockerPull: pgc-images.sbgenomics.com/d3b-bixu/bwa-alt-aware
        id: michael_lloyd/mantis/bwa-alt-aware-v3/4
        sbg:appVersion:
        - sbg:draft-2
        sbg:cmdPreview: |-
          tar -xzvf /path/to/index_file.ext --strip-components 1 ; run-bwamem   /path/to/ForwardFastq.ext  /path/to/ReverseFastq.ext  |sh
        sbg:content_hash: ac5e18dedba6bc13d8c7358e599c0e95503e2baa84c6a302c704b98e126f069e6
        sbg:contributors:
        - michael_lloyd
        sbg:createdBy: michael_lloyd
        sbg:createdOn: 1566052707
        sbg:id: michael_lloyd/mantis/bwa-alt-aware-v3/4
        sbg:image_url:
        sbg:job:
          allocatedResources:
            cpu: 4
            mem: 5000
          inputs:
            forward_fastq:
              class: File
              secondaryFiles: []
              metadata:
                case_id: test_case_id
                sample_id: test_sample_id
                sample_type: test_type
              path: /path/to/ForwardFastq.ext
              size: 0
            index_file:
              class: File
              secondaryFiles: []
              path: /path/to/index_file.ext
              size: 0
            outfile_name: ''
            prefix_bwa: Prefix_Bwa-string-value
            read_group: Read_Group-string-value
            reverse_fastq:
              class: File
              secondaryFiles: []
              path: /path/to/ReverseFastq.ext
              size: 0
            threads: 4
        sbg:latestRevision: 4
        sbg:modifiedBy: michael_lloyd
        sbg:modifiedOn: 1566925497
        sbg:project: michael_lloyd/mantis
        sbg:projectName: MSI, HRD, TMB
        sbg:publisher: sbg
        sbg:revision: 4
        sbg:revisionNotes: added 'inherit metadata'
        sbg:revisionsInfo:
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1566052707
          sbg:revision: 0
          sbg:revisionNotes: Copy of michael_lloyd/variant-calling-decomposition/bwa-alt-aware-v3/1
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1566243035
          sbg:revision: 1
          sbg:revisionNotes: BAM naming change.
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1566243308
          sbg:revision: 2
          sbg:revisionNotes: ''
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1566305054
          sbg:revision: 3
          sbg:revisionNotes: ''
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1566925497
          sbg:revision: 4
          sbg:revisionNotes: added 'inherit metadata'
        sbg:sbgMaintained: false
        sbg:validationErrors: []
      inputs:
      - id: '#bwa_alt_aware_v3.threads'
        source: '#threads'
      - id: '#bwa_alt_aware_v3.reverse_fastq'
        source: '#XenomeV2.human_reverse'
      - id: '#bwa_alt_aware_v3.read_group'
        default: '"@RG\tID:1\tPL:Illumina\tSM:tumor"'
      - id: '#bwa_alt_aware_v3.index_file'
        source: '#index_file'
      - id: '#bwa_alt_aware_v3.forward_fastq'
        source: '#XenomeV2.human_forward'
      outputs:
      - id: '#bwa_alt_aware_v3.bam_output'
      sbg:x: 530.9862670898438
      sbg:y: -102.87158203125
    id: michael_lloyd/mantis/tumor-alignment-and-target-coverage-workflow/9
    sbg:appVersion:
    - sbg:draft-2
    sbg:canvas_x: 347
    sbg:canvas_y: 125
    sbg:canvas_zoom: 0.7499999999999998
    sbg:content_hash: a73aa6c160094461a0309966f4e4a5fd206dd566f931575725d8d90158722aa19
    sbg:contributors:
    - michael_lloyd
    sbg:createdBy: michael_lloyd
    sbg:createdOn: 1566052687
    sbg:id: michael_lloyd/mantis/tumor-alignment-and-target-coverage-workflow/9
    sbg:image_url:
    sbg:latestRevision: 9
    sbg:modifiedBy: michael_lloyd
    sbg:modifiedOn: 1572961486
    sbg:original_source: michael_lloyd/mantis/tumor-alignment-and-target-coverage-workflow/9
    sbg:project: michael_lloyd/mantis
    sbg:projectName: MSI, HRD, TMB
    sbg:publisher: sbg
    sbg:revision: 9
    sbg:revisionNotes: increased instance request for Picard Markduplicates
    sbg:revisionsInfo:
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1566052687
      sbg:revision: 0
      sbg:revisionNotes: |-
        Copy of michael_lloyd/variant-calling-decomposition/tumor-alignment-and-target-coverage-workflow/0
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1566839381
      sbg:revision: 1
      sbg:revisionNotes: updated dynamic naming of QC integrate steps
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1566848846
      sbg:revision: 2
      sbg:revisionNotes: ''
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1566849403
      sbg:revision: 3
      sbg:revisionNotes: removed bad link to markduplicates
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1566849478
      sbg:revision: 4
      sbg:revisionNotes: ''
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1566919515
      sbg:revision: 5
      sbg:revisionNotes: ''
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1566925665
      sbg:revision: 6
      sbg:revisionNotes: added metadata inherit statement to bwa tool
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1566929866
      sbg:revision: 7
      sbg:revisionNotes: ''
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1566930464
      sbg:revision: 8
      sbg:revisionNotes: changed to correct BWA version
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1572961486
      sbg:revision: 9
      sbg:revisionNotes: increased instance request for Picard Markduplicates
    sbg:sbgMaintained: false
    sbg:validationErrors: []
  inputs:
  - id: '#tumor_alignment_and_target_coverage_workflow.target_intervals'
    source: '#target_intervals'
  - id: '#tumor_alignment_and_target_coverage_workflow.reference'
    source: '#reference'
  - id: '#tumor_alignment_and_target_coverage_workflow.known_sites'
    source:
    - '#known_sites'
  - id: '#tumor_alignment_and_target_coverage_workflow.index_file_1'
    source: '#index_file_1'
  - id: '#tumor_alignment_and_target_coverage_workflow.index_file'
    source: '#index_file'
  - id: '#tumor_alignment_and_target_coverage_workflow.bed'
    source: '#bed'
  - id: '#tumor_alignment_and_target_coverage_workflow.reads'
    source:
    - '#input_pair'
  - id: '#tumor_alignment_and_target_coverage_workflow.memory_per_job'
    default: 15000
  - id: '#tumor_alignment_and_target_coverage_workflow.memory_per_job_1'
    default: 15000
  - id: '#tumor_alignment_and_target_coverage_workflow.memory_per_job_2'
    default: 15000
  - id: '#tumor_alignment_and_target_coverage_workflow.intervals_string'
    default: chr20
    source: '#intervals_string'
  - id: '#tumor_alignment_and_target_coverage_workflow.TargetCoverage_CutOff'
    source: '#TargetCoverage_CutOff'
  outputs:
  - id: '#tumor_alignment_and_target_coverage_workflow.summary'
  - id: '#tumor_alignment_and_target_coverage_workflow.output'
  - id: '#tumor_alignment_and_target_coverage_workflow.metrics_file'
  - id: '#tumor_alignment_and_target_coverage_workflow.hs_metrics'
  - id: '#tumor_alignment_and_target_coverage_workflow.recalibration_file'
  - id: '#tumor_alignment_and_target_coverage_workflow.integrated_QC_report'
  sbg:x: 190.66123962402344
  sbg:y: 88.91834259033203
- id: '#multiqc_2'
  label: MultiQC
  in: []
  run:
    cwlVersion: sbg:draft-2
    class: CommandLineTool
    label: MultiQC
    $namespaces:
      sbg: https://sevenbridges.com

    requirements:
    - id: '#cwl-js-engine'
      class: ExpressionEngineRequirement
      requirements:
      - class: DockerRequirement
        dockerPull: rabix/js-engine
    - class: InlineJavascriptRequirement

    inputs:
    - id: '#input_files'
      label: Input files
      type:
      - type: array
        items: File
      description: |-
        Input reports. Can be reports from any of the tools supported by MultiQC. Complete list of supported tools is avilable in the documentation.
      sbg:category: Inputs
      sbg:stageInput: link
    - id: '#fullnames'
      label: Full names
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -s
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Do not clean the sample names (leave as full file name).
      sbg:category: Options
      sbg:stageInput:
      sbg:toolDefaultValue: off
    - id: '#title'
      label: Title
      type:
      - 'null'
      - string
      inputBinding:
        position: 0
        valueFrom:
          class: Expression
          engine: '#cwl-js-engine'
          script: "{\n  if ($job.inputs.title) {\n    return \"-i '\" + $job.inputs.title\
            \ + \"'\"\n  }\n}"
        separate: true
        sbg:cmdInclude: true
      description: Report title. Printed as page header.
      sbg:category: Options
      sbg:toolDefaultValue: None
    - id: '#filename'
      label: Filename
      type:
      - 'null'
      - string
      inputBinding:
        position: 0
        valueFrom:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              function flatten(files){
                var a = []
              	for(var i=0;i<files.length;i++){
                  if(files[i]){
                    if(files[i].constructor == Array) {
                      a = a.concat(flatten(files[i]))
                    } else {
                      a = a.concat(files[i])
                    }
                  }
                }
                return a
              }
              
              arr = [].concat($job.inputs.input_files)
              return_array = []
              return_array = flatten(arr)
              new_list = return_array
              
              if (new_list[0].metadata && new_list[0].metadata.sample_id) {
                var x = new_list[0].metadata.sample_id
              } else {
                var x = new_list[0].path.split('/').pop().split('.')[0].split('_')[0]
              }
              
              if ($job.inputs.filename) {
                return "-n " + $job.inputs.filename
              } else {
                return "-n " + x + '.multiqc_report'
              }
                                                                                                     
            }
        separate: true
        sbg:cmdInclude: true
      description: Report filename.
      sbg:category: Options
      sbg:toolDefaultValue: sample_id.multiqc_report
    - id: '#template'
      label: Template
      type:
      - 'null'
      - name: template
        type: enum
        symbols:
        - default
        - simple
        - sections
      inputBinding:
        prefix: -t
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Report template to use.
      sbg:category: Options
      sbg:toolDefaultValue: default
    - id: '#data_format'
      label: Data format
      type:
      - 'null'
      - name: data_format
        type: enum
        symbols:
        - tsv
        - yaml
        - json
      inputBinding:
        prefix: -k
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Output parsed data in a different format.
      sbg:category: Options
      sbg:stageInput:
      sbg:toolDefaultValue: tsv
    - id: '#exclude'
      label: Exclude module
      type:
      - 'null'
      - type: array
        items: string
      inputBinding:
        prefix: -e
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Do not use this module. Can specify multiple times.
      sbg:category: Options
      sbg:stageInput:
      sbg:toolDefaultValue: None
    - id: '#module'
      label: Use only this module
      type:
      - 'null'
      - type: array
        items: string
      inputBinding:
        prefix: -m
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Use only this module. Can specify multiple times.
      sbg:category: Options
      sbg:stageInput:
      sbg:toolDefaultValue: None
    - id: '#flat'
      label: Flat
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --flat
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Use only flat plots (static images).
      sbg:category: Options
      sbg:stageInput:
      sbg:toolDefaultValue: off
    - id: '#interactive'
      label: Interactive
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --interactive
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Use only interactive plots (HighCharts Javascript).
      sbg:category: Options
      sbg:stageInput:
      sbg:toolDefaultValue: off
    - id: '#config'
      label: Config file
      type:
      - 'null'
      - type: array
        items: File
      inputBinding:
        prefix: -c
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Specific config file to load, after those in the current working
        directory.
      sbg:category: Options
      sbg:fileTypes: CONFIG, TXT
      sbg:stageInput:
    - id: '#comment'
      label: Comment
      type:
      - 'null'
      - string
      inputBinding:
        prefix: -b
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Custom comment, will be printed at the top of the report.
      sbg:altPrefix: --comment
      sbg:category: Options
      sbg:toolDefaultValue: None
    - id: '#ignore_files'
      label: Ignore files
      type:
      - 'null'
      - type: array
        items: string
      inputBinding:
        prefix: -x
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Ignore analysis files (glob expression).
      sbg:category: Options
      sbg:stageInput:
      sbg:toolDefaultValue: None
    - id: '#ignore_samples'
      label: Ignore samples
      type:
      - 'null'
      - type: array
        items: string
      inputBinding:
        prefix: --ignore-samples
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Ignore sample names (glob expression).
      sbg:category: Options
      sbg:stageInput:
      sbg:toolDefaultValue: None
    - id: '#sample_names'
      label: Sample names file
      type:
      - 'null'
      - File
      inputBinding:
        prefix: --sample-names
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        File containing alternative sample names. Allows one-click batch renaming in reports.
      sbg:category: Options
      sbg:fileTypes: TXT
    - id: '#export'
      label: Export plots
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -p
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Export plots as static images in addition to the report.
      sbg:category: Options
      sbg:stageInput:
      sbg:toolDefaultValue: 'False'
    - id: '#cl_config'
      label: Config YAML
      type:
      - 'null'
      - type: array
        items: string
      inputBinding:
        prefix: --cl_config
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Specify MultiQC config YAML directly here.
      sbg:category: Options
      sbg:stageInput:
      sbg:toolDefaultValue: None
    - id: '#tag'
      label: Tag
      type:
      - 'null'
      - type: array
        items: string
      inputBinding:
        prefix: --tag
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Use only modules which tagged with this keyword, eg. RNA.
      sbg:category: Options
      sbg:toolDefaultValue: None
    - id: '#view_tags'
      label: View tags
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --view_tags
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: View the available tags and which modules they load.
      sbg:category: Options
      sbg:stageInput:
      sbg:toolDefaultValue: Off

    outputs:
    - id: '#multiqc_report'
      label: MultiQC report
      type:
      - 'null'
      - File
      outputBinding:
        glob:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              function flatten(files){
                var a = []
              	for(var i=0;i<files.length;i++){
                  if(files[i]){
                    if(files[i].constructor == Array) {
                      a = a.concat(flatten(files[i]))
                    } else {
                      a = a.concat(files[i])
                    }
                  }
                }
                return a
              }
              
              arr = [].concat($job.inputs.input_files)
              return_array = []
              return_array = flatten(arr)
              new_list = return_array
              
              if (new_list[0].metadata && new_list[0].metadata.sample_id) {
                var x = new_list[0].metadata.sample_id
              } else {
                var x = new_list[0].path.split('/').pop().split('.')[0].split('_')[0]
              }
              
              if ($job.inputs.filename) {
                return $job.inputs.filename + ".html"
              } else {
                return x + '.multiqc_report.html'
              }
                                                                                                     
            }
        sbg:inheritMetadataFrom: '#input_files'
      description: File containg MultiQC report analysis.
      sbg:fileTypes: HTML
    - id: '#data_archive'
      label: Data archive
      type:
      - 'null'
      - File
      outputBinding:
        glob:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              function flatten(files){
                var a = []
              	for(var i=0;i<files.length;i++){
                  if(files[i]){
                    if(files[i].constructor == Array) {
                      a = a.concat(flatten(files[i]))
                    } else {
                      a = a.concat(files[i])
                    }
                  }
                }
                return a
              }
              
              arr = [].concat($job.inputs.input_files)
              return_array = []
              return_array = flatten(arr)
              new_list = return_array
              
              if (new_list[0].metadata && new_list[0].metadata.sample_id) {
                var x = new_list[0].metadata.sample_id
              } else {
                var x = new_list[0].path.split('/').pop().split('.')[0].split('_')[0]
              }
              
              if ($job.inputs.filename) {
                return $job.inputs.filename + "_data.zip"
              } else {
                return x + '.multiqc_report_data.zip'
              }
                                                                                                     
            }
        sbg:inheritMetadataFrom: '#input_files'
      description: |-
        An archive containing parsed data that can be used for further downstream analysis.
      sbg:fileTypes: ZIP
    - id: '#static_plots'
      label: Static plots
      type:
      - 'null'
      - type: array
        items: File
      outputBinding:
        glob:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              return "{multiqc_plots/pdf/*.pdf, multiqc_plots/png/*.png, multiqc_plots/svg/*.svg}"
            }
        sbg:inheritMetadataFrom: '#input_files'
      description: |-
        Static plots from the report in pdf, png and svg format, if the the 'Export' option was specified.
      sbg:fileTypes: PDF, PNG, SVG

    baseCommand:
    - class: Expression
      engine: '#cwl-js-engine'
      script: |-
        {
          return "find -name '*.qualimap.zip' -exec sh -c \'unzip -d \"${1%.*}\" \"$1\"\' _ {} \\; ;"
        }
    - class: Expression
      engine: '#cwl-js-engine'
      script: "{\n  return \"find -iname \\\\*.tar -exec tar -xvf {} \\\\; ;\"\n}\n\
        \n"
    - multiqc
    arguments:
    - position: 100
      valueFrom: .
      separate: false
    - prefix: ''
      position: 1
      valueFrom: -z
      separate: true

    hints:
    - class: sbg:AWSInstanceType
      value: c5.2xlarge;ebs-gp2;100
    - class: sbg:CPURequirement
      value: 1
    - class: sbg:MemRequirement
      value: 1000
    - class: DockerRequirement
      dockerPull: images.sbgenomics.com/uros_sipetic/multiqc:1.3
    id: michael_lloyd/mantis/multiqc-1/2
    description: |-
      **MultiQC** aggregates results from bioinformatics analyses across many samples into a single report [1]. 

      __MultiQC__ searches a given directory for analysis logs and compiles an HTML report. It's a general purpose tool, perfect for aggregating and summarizing the output from numerous bioinformatics tools [1].

      *A list of **all inputs and parameters** with corresponding descriptions can be found at the end of the page.*

      ### Common Use Cases

      - To successfully run __MultiQC__, just supply it with outputs from one of the many currently supported modules. The list of all supported tools and how to properly forward their outputs into __MultiQC__ can be found via the following [link](http://multiqc.info/docs/#multiqc-modules). As an output, a collective report with all of the analyses you provided in the input will be generated in an interactive HTML format as shown in [1]. 

      ### Changes Introduced by Seven Bridges

      * Taking into consideration that the platform currently is unable to directly display non-native and interactive HTML pages, instead of downloading the __MultiQC__ report and viewing it in your browser locally, you can either use the __SBG Html2b64__ app from the Public Apps Gallery to turn your 'simple' template report into a platform friendly HTML document and view it on the platform, or you can simply resort to the PDF file (use *View as PDF* option under **...** to open it on the platform), which is by default always produced in addition to the HTML report (under default **MultiQC** parameters).

      ### Common Issues and Important Notes

      - All output files will be prefixed by the input **Sample ID** metadata field (if it exists, otherwise the prefix will be taken from the file name). If there are multiple samples, the prefix will be the first input filename.

      ### Performance Benchmarking

      __MultiQC__ is a tool that reports metrics, and as such does not require a lot of resources. Almost all **MultiQC** tasks will finish within 5 minutes when running on the default instance, costing around $0.05 using on-demand instances. 

      *Cost can be significantly reduced by **spot instance** usage. Visit [knowledge center](https://docs.sevenbridges.com/docs/about-spot-instances) for more details.*

      ### References

      [1] [MultiQC homepage](http://multiqc.info/)
    sbg:appVersion:
    - sbg:draft-2
    sbg:categories:
    - Plotting-and-Rendering
    sbg:cmdPreview: |-
      find -name '*.qualimap.zip' -exec sh -c 'unzip -d "${1%.*}" "$1"' _ {} \; ; find -iname \*.tar -exec tar -xvf {} \; ; multiqc  -z .
    sbg:content_hash: a0414dc647dd140b541b28ed8039ef3e9a24b4c15b29eec6c8693e531699d6a62
    sbg:contributors:
    - michael_lloyd
    sbg:createdBy: michael_lloyd
    sbg:createdOn: 1567083925
    sbg:id: michael_lloyd/mantis/multiqc-1/2
    sbg:image_url:
    sbg:job:
      allocatedResources:
        cpu: 1
        mem: 1000
      inputs:
        cl_config:
        - cl_config-string-value-1
        - cl_config-string-value-2
        comment: ''
        config:
        - class: File
          secondaryFiles: []
          path: /path/to/config-1.ext
          size: 0
        - class: File
          secondaryFiles: []
          path: /path/to/config-2.ext
          size: 0
        data_format:
        exclude:
        - exclude-string-value-1
        - exclude-string-value-2
        export: false
        filename: ''
        flat: false
        fullnames: false
        ignore_files:
        - ignore_files-string-value-1
        - ignore_files-string-value-2
        ignore_samples:
        - ignore_samples-string-value-1
        - ignore_samples-string-value-2
        input_files:
        - class: File
          secondaryFiles: []
          path: /path/to/input_files-1.ext
          size: 0
        - class: File
          secondaryFiles: []
          path: /path/to/input_files-2.ext
          size: 0
        interactive: false
        module:
        - module-string-value-1
        - module-string-value-2
        tag:
        - tah-string-value-1
        - tah-string-value-2
        template:
        title: ''
        view_tags: true
    sbg:latestRevision: 2
    sbg:license: GNU General Public License v3.0 only
    sbg:links:
    - id: http://multiqc.info/
      label: Homepage
    - id: https://github.com/ewels/MultiQC
      label: Source Code
    - id: https://github.com/ewels/MultiQC/archive/v1.3.tar.gz
      label: Download
    - id: |-
        http://bioinformatics.oxfordjournals.org/content/early/2016/06/23/bioinformatics.btw354
      label: Publication
    - id: http://multiqc.info/docs/
      label: Documentation
    sbg:modifiedBy: michael_lloyd
    sbg:modifiedOn: 1568040078
    sbg:project: michael_lloyd/mantis
    sbg:projectName: MSI, HRD, TMB
    sbg:publisher: sbg
    sbg:revision: 2
    sbg:revisionNotes: corrected grob statement
    sbg:revisionsInfo:
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1567083925
      sbg:revision: 0
      sbg:revisionNotes: Copy of michael_lloyd/variant-calling-decomposition/multiqc-1/0
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1567085036
      sbg:revision: 1
      sbg:revisionNotes: updated filenaming to deal with PDMR naming conventions
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1568040078
      sbg:revision: 2
      sbg:revisionNotes: corrected grob statement
    sbg:sbgMaintained: false
    sbg:toolAuthor: Phil Ewels
    sbg:toolkit: MultiQC
    sbg:toolkitVersion: '1.3'
    sbg:validationErrors: []
  inputs:
  - id: '#multiqc_2.input_files'
    source:
    - '#tumor_alignment_and_target_coverage_workflow.recalibration_file'
    - '#tumor_alignment_and_target_coverage_workflow.metrics_file'
    - '#tumor_alignment_and_target_coverage_workflow.hs_metrics'
    - '#FastQC.report_zip'
  outputs:
  - id: '#multiqc_2.multiqc_report'
  - id: '#multiqc_2.data_archive'
  - id: '#multiqc_2.static_plots'
  sbg:x: 452.88201904296875
  sbg:y: -129.2266082763672
- id: '#tmb_tumoronly'
  label: TMB-TumorOnly
  in: []
  run:
    cwlVersion: sbg:draft-2
    class: Workflow
    label: TMB-TumorOnly
    $namespaces:
      sbg: https://sevenbridges.com

    requirements:
    - class: SubworkflowFeatureRequirement
    - class: InlineJavascriptRequirement

    inputs:
    - id: '#snpEFF_Database'
      type:
      - File
      sbg:fileTypes: ZIP
      sbg:includeInPorts: true
      sbg:x: 428.84844970703125
      sbg:y: -132.7667236328125
    - id: '#COSMIC_snpSift_Database'
      type:
      - File
      sbg:fileTypes: VCF.GZ
      sbg:includeInPorts: true
      sbg:x: 910.9334716796875
      sbg:y: -98.51634979248047
    - id: '#raw_vcf'
      type:
      - 'null'
      - File
      sbg:includeInPorts: true
      sbg:x: -683.9304809570312
      sbg:y: -818.4986572265625
    - id: '#dbNSFP_Database'
      type:
      - File
      sbg:fileTypes: GZ, TXT.GZ, VCF.GZ
      sbg:includeInPorts: true
      sbg:x: 662.803466796875
      sbg:y: -113.94287109375
    - id: '#Coverage_BedFile'
      type:
      - File
      sbg:fileTypes: BED
      sbg:includeInPorts: true
      sbg:x: 1196.267578125
      sbg:y: -867.7504272460938

    outputs:
    - id: '#TMB_Score'
      type:
      - 'null'
      - File
      source:
      - '#tmb_calc.TMB_Score'
      sbg:x: 2188.110595703125
      sbg:y: -686.37451171875

    steps:
    - id: '#snpsift_filter_4_3'
      label: SnpSift filter
      in: []
      run:
        cwlVersion: sbg:draft-2
        class: CommandLineTool
        label: SnpSift filter
        $namespaces:
          sbg: https://sevenbridges.com

        requirements:
        - id: '#cwl-js-engine'
          class: ExpressionEngineRequirement
          requirements:
          - class: DockerRequirement
            dockerPull: rabix/js-engine
        - class: InlineJavascriptRequirement

        inputs:
        - id: '#mem_mb'
          label: Memory to use for the task [Mb]
          type:
          - 'null'
          - int
          description: Memory to use for the task [Mb]
          sbg:category: Options
          sbg:stageInput:
          sbg:toolDefaultValue: '4096'
        - id: '#input_vcf'
          label: Input VCF
          type:
          - 'null'
          - File
          inputBinding:
            prefix: -f
            position: 100
            separate: true
            sbg:cmdInclude: true
          description: 'VCF input file. Default: STDIN'
          sbg:fileTypes: VCF, VCF.GZ
        - id: '#expression'
          label: Filter expression (without quotes)
          type:
          - 'null'
          - string
          inputBinding:
            position: 55
            valueFrom:
              class: Expression
              engine: '#cwl-js-engine'
              script: |-
                {
                  if ($job.inputs.expression)
                  {
                    return '"'.concat($job.inputs.expression,'"')
                  }
                  else
                  {
                    return ''
                  }
                }
            separate: true
            sbg:cmdInclude: true
          description: |-
            Expression to use for filtering. Please note that wrapping quotes are added automatically by the wrapper and should not be used as part of the filtering expression.
          sbg:category: Inputs
        - id: '#addFilter'
          label: Add a string to FILTER field
          type:
          - 'null'
          - string
          inputBinding:
            prefix: -a
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: "Add a string to FILTER VCF field if 'expression' is true.\
            \ Default: '' (none)."
          sbg:altPrefix: --addFilter
          sbg:category: Options
          sbg:toolDefaultValue: "''"
        - id: '#exprFile'
          label: Expression file
          type:
          - 'null'
          - File
          inputBinding:
            prefix: -e
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Read expression from a file.
          sbg:altPrefix: --exprFile
          sbg:category: Options
        - id: '#filterId'
          label: Filter ID
          type:
          - 'null'
          - string
          inputBinding:
            prefix: -i
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            ID for this filter (##FILTER tag in header and FILTER VCF field). Default: 'SnpSift'.
          sbg:altPrefix: --filterId
          sbg:category: Options
          sbg:toolDefaultValue: SnpSift
        - id: '#inverse'
          label: Inverse
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -n
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Inverse. Show lines that do not match filter expression.
          sbg:altPrefix: --inverse
          sbg:category: Options
          sbg:stageInput:
        - id: '#pass'
          label: Pass
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -p
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Use 'PASS' field instead of filtering out VCF entries.
          sbg:altPrefix: --pass
          sbg:category: Options
        - id: '#rmFilter'
          label: Remove string from FILTER field
          type:
          - 'null'
          - string
          inputBinding:
            prefix: -r
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Remove a string from FILTER VCF field if 'expression' is true (and 'str' is in the field). Default: '' (none).
          sbg:altPrefix: --rmFilter
          sbg:category: Options
          sbg:toolDefaultValue: "''"
        - id: '#set'
          label: Set File
          type:
          - 'null'
          - File
          inputBinding:
            prefix: -s
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Create a SET using 'file'.
          sbg:altPrefix: --set
          sbg:category: Options
        - id: '#errMissing'
          label: Error on missing fields
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --errMissing
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: 'Error if a field is missing. Default: False.'
          sbg:category: Options
          sbg:stageInput:
          sbg:toolDefaultValue: 'False'
        - id: '#format'
          label: Format
          type:
          - 'null'
          - name: format
            type: enum
            symbols:
            - '2'
            - '3'
          inputBinding:
            prefix: --format
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: 'SnpEff format version: {2, 3}. Default: Auto.'
          sbg:stageInput:
          sbg:toolDefaultValue: Auto
        - id: '#galaxy'
          label: Galaxy
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --galaxy
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Used from Galaxy (expressions have been sanitized).
          sbg:category: Options
          sbg:stageInput:

        outputs:
        - id: '#filtered_vcf'
          label: Filtered VCF
          type:
          - 'null'
          - File
          outputBinding:
            glob: '*.filtered.vcf'
            sbg:inheritMetadataFrom: '#input_vcf'
          description: Filtered VCF.
          sbg:fileTypes: VCF
        stdout:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              filename = $job.inputs.input_vcf.path
              return filename.split('.').slice(0, filename.split('.').length-1).concat("filtered.vcf").join('.').replace(/^.*[\\\/]/, '')
            }

        baseCommand:
        - java
        - class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              if ($job.inputs.mem_mb)
              {
                return "-Xmx" + ($job.inputs.mem_mb).toString() + "m"
              }
              else
              {
                return "-Xmx4096m"
              }
            }
        - -jar
        - /opt/snpEff/SnpSift.jar
        - filter

        hints:
        - class: sbg:CPURequirement
          value: 1
        - class: sbg:MemRequirement
          value:
            class: Expression
            engine: '#cwl-js-engine'
            script: |-
              {
                if ($job.inputs.mem_mb)
                {
                  return $job.inputs.mem_mb
                }
                
                else
                {
                  return 4096
                }
              }
        - class: DockerRequirement
          dockerPull: images.sbgenomics.com/jrandjelovic/snpeff:v4.3r
        id: admin/sbg-public-data/snpsift-filter-4-3/2
        description: |-
          **SnpSift filter** filters SnpEff-annotated VCF files using arbitrary expressions, for instance "(QUAL > 30) | (exists INDEL) | ( countHet() > 2 )". The actual expressions can be quite complex, allowing for a lot of flexibility [1].

          *A list of **all inputs and parameters** with corresponding descriptions can be found at the end of the page.*

          ### Common Use Cases

          The **SnpSift filter** tool is used for filtering SnpEff-annotated VCF files to extract variants of interest.

          ### Changes Introduced by Seven Bridges

          * No modifications to the original tool representation have been made. 

          ### Common Issues and Important Notes

          * Parameter **Filter expression**: Wrapping quotes ("") are added automatically by the wrapper and should not be included in the filter expression. For example, filtering expression: "( CHROM = 'chr1' )" should be written as `( CHROM = 'chr1' )`.
          * **SnpSift filter** filtering syntax allows for creation of complex, flexible filters with different operators and functions. Please check the [official documentation](http://snpeff.sourceforge.net/SnpSift.html#filter) for examples of filter syntax and valid filtering fields before using the tool.

          ### Performance Benchmarking

          Typical runs take <10 minutes and cost <$0.1 on an on-demand default instance.

          ### References

          [1] [SnpSift filter documentation](http://snpeff.sourceforge.net/SnpSift.html#filter)
        sbg:appVersion:
        - sbg:draft-2
        sbg:categories:
        - Annotation
        - VCF-Processing
        sbg:cmdPreview: java -Xmx5m -jar /opt/snpEff/SnpSift.jar filter  > input_vcf.filtered.vcf
        sbg:content_hash:
        sbg:contributors:
        - admin
        sbg:createdBy: admin
        sbg:createdOn: 1512132000
        sbg:id: admin/sbg-public-data/snpsift-filter-4-3/2
        sbg:image_url:
        sbg:job:
          allocatedResources:
            cpu: 1
            mem: 5
          inputs:
            format: '2'
            addFilter: addFilter-string-value
            errMissing: true
            exprFile:
              class: File
              secondaryFiles: []
              path: /path/to/exprFile.ext
              size: 0
            expression: expression
            filterId: ''
            galaxy: true
            input_vcf:
              class: File
              secondaryFiles: []
              path: /path/to/input_vcf.ext
              size: 0
            inverse: true
            mem_mb: 5
            pass: true
            rmFilter: rmFilter-string-value
            set:
              class: File
              secondaryFiles: []
              path: /path/to/set.ext
              size: 0
        sbg:latestRevision: 2
        sbg:license: GNU Lesser General Public License v3.0 only
        sbg:links:
        - id: http://snpeff.sourceforge.net/SnpSift.html#filter
          label: Homepage
        - id: https://github.com/pcingola/SnpEff
          label: Source Code
        sbg:modifiedBy: admin
        sbg:modifiedOn: 1533292196
        sbg:project: admin/sbg-public-data
        sbg:projectName: SBG Public data
        sbg:publisher: sbg
        sbg:revision: 2
        sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-filter/9
        sbg:revisionsInfo:
        - sbg:modifiedBy: admin
          sbg:modifiedOn: 1512132000
          sbg:revision: 0
          sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-filter/5
        - sbg:modifiedBy: admin
          sbg:modifiedOn: 1512132000
          sbg:revision: 1
          sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-filter/8
        - sbg:modifiedBy: admin
          sbg:modifiedOn: 1533292196
          sbg:revision: 2
          sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-filter/9
        sbg:sbgMaintained: false
        sbg:toolAuthor: Pablo Cingolani/Broad Institue
        sbg:toolkit: SnpEff
        sbg:toolkitVersion: 4.3r
        sbg:validationErrors: []
      inputs:
      - id: '#snpsift_filter_4_3.input_vcf'
        source: '#filter_trim_py_tumoronly.filtered_vcf'
      - id: '#snpsift_filter_4_3.expression'
        default: ALT_AF[ANY] < 5
      - id: '#snpsift_filter_4_3.addFilter'
        default: lowAF
      - id: '#snpsift_filter_4_3.rmFilter'
        default: PASS
      outputs:
      - id: '#snpsift_filter_4_3.filtered_vcf'
      sbg:x: -338.35003662109375
      sbg:y: -648.0833129882812
    - id: '#snpsift_filter_4_4'
      label: SnpSift filter
      in: []
      run:
        cwlVersion: sbg:draft-2
        class: CommandLineTool
        label: SnpSift filter
        $namespaces:
          sbg: https://sevenbridges.com

        requirements:
        - id: '#cwl-js-engine'
          class: ExpressionEngineRequirement
          requirements:
          - class: DockerRequirement
            dockerPull: rabix/js-engine
        - class: InlineJavascriptRequirement

        inputs:
        - id: '#mem_mb'
          label: Memory to use for the task [Mb]
          type:
          - 'null'
          - int
          description: Memory to use for the task [Mb]
          sbg:category: Options
          sbg:stageInput:
          sbg:toolDefaultValue: '4096'
        - id: '#input_vcf'
          label: Input VCF
          type:
          - 'null'
          - File
          inputBinding:
            prefix: -f
            position: 100
            separate: true
            sbg:cmdInclude: true
          description: 'VCF input file. Default: STDIN'
          sbg:fileTypes: VCF, VCF.GZ
        - id: '#expression'
          label: Filter expression (without quotes)
          type:
          - 'null'
          - string
          inputBinding:
            position: 55
            valueFrom:
              class: Expression
              engine: '#cwl-js-engine'
              script: |-
                {
                  if ($job.inputs.expression)
                  {
                    return '"'.concat($job.inputs.expression,'"')
                  }
                  else
                  {
                    return ''
                  }
                }
            separate: true
            sbg:cmdInclude: true
          description: |-
            Expression to use for filtering. Please note that wrapping quotes are added automatically by the wrapper and should not be used as part of the filtering expression.
          sbg:category: Inputs
        - id: '#addFilter'
          label: Add a string to FILTER field
          type:
          - 'null'
          - string
          inputBinding:
            prefix: -a
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: "Add a string to FILTER VCF field if 'expression' is true.\
            \ Default: '' (none)."
          sbg:altPrefix: --addFilter
          sbg:category: Options
          sbg:toolDefaultValue: "''"
        - id: '#exprFile'
          label: Expression file
          type:
          - 'null'
          - File
          inputBinding:
            prefix: -e
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Read expression from a file.
          sbg:altPrefix: --exprFile
          sbg:category: Options
        - id: '#filterId'
          label: Filter ID
          type:
          - 'null'
          - string
          inputBinding:
            prefix: -i
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            ID for this filter (##FILTER tag in header and FILTER VCF field). Default: 'SnpSift'.
          sbg:altPrefix: --filterId
          sbg:category: Options
          sbg:toolDefaultValue: SnpSift
        - id: '#inverse'
          label: Inverse
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -n
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Inverse. Show lines that do not match filter expression.
          sbg:altPrefix: --inverse
          sbg:category: Options
          sbg:stageInput:
        - id: '#pass'
          label: Pass
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -p
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Use 'PASS' field instead of filtering out VCF entries.
          sbg:altPrefix: --pass
          sbg:category: Options
        - id: '#rmFilter'
          label: Remove string from FILTER field
          type:
          - 'null'
          - string
          inputBinding:
            prefix: -r
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Remove a string from FILTER VCF field if 'expression' is true (and 'str' is in the field). Default: '' (none).
          sbg:altPrefix: --rmFilter
          sbg:category: Options
          sbg:toolDefaultValue: "''"
        - id: '#set'
          label: Set File
          type:
          - 'null'
          - File
          inputBinding:
            prefix: -s
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Create a SET using 'file'.
          sbg:altPrefix: --set
          sbg:category: Options
        - id: '#errMissing'
          label: Error on missing fields
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --errMissing
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: 'Error if a field is missing. Default: False.'
          sbg:category: Options
          sbg:stageInput:
          sbg:toolDefaultValue: 'False'
        - id: '#format'
          label: Format
          type:
          - 'null'
          - name: format
            type: enum
            symbols:
            - '2'
            - '3'
          inputBinding:
            prefix: --format
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: 'SnpEff format version: {2, 3}. Default: Auto.'
          sbg:stageInput:
          sbg:toolDefaultValue: Auto
        - id: '#galaxy'
          label: Galaxy
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --galaxy
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Used from Galaxy (expressions have been sanitized).
          sbg:category: Options
          sbg:stageInput:

        outputs:
        - id: '#filtered_vcf'
          label: Filtered VCF
          type:
          - 'null'
          - File
          outputBinding:
            glob: '*.filtered.vcf'
            sbg:inheritMetadataFrom: '#input_vcf'
          description: Filtered VCF.
          sbg:fileTypes: VCF
        stdout:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              filename = $job.inputs.input_vcf.path
              return filename.split('.').slice(0, filename.split('.').length-1).concat("filtered.vcf").join('.').replace(/^.*[\\\/]/, '')
            }

        baseCommand:
        - java
        - class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              if ($job.inputs.mem_mb)
              {
                return "-Xmx" + ($job.inputs.mem_mb).toString() + "m"
              }
              else
              {
                return "-Xmx4096m"
              }
            }
        - -jar
        - /opt/snpEff/SnpSift.jar
        - filter

        hints:
        - class: sbg:CPURequirement
          value: 1
        - class: sbg:MemRequirement
          value:
            class: Expression
            engine: '#cwl-js-engine'
            script: |-
              {
                if ($job.inputs.mem_mb)
                {
                  return $job.inputs.mem_mb
                }
                
                else
                {
                  return 4096
                }
              }
        - class: DockerRequirement
          dockerPull: images.sbgenomics.com/jrandjelovic/snpeff:v4.3r
        id: admin/sbg-public-data/snpsift-filter-4-3/2
        description: |-
          **SnpSift filter** filters SnpEff-annotated VCF files using arbitrary expressions, for instance "(QUAL > 30) | (exists INDEL) | ( countHet() > 2 )". The actual expressions can be quite complex, allowing for a lot of flexibility [1].

          *A list of **all inputs and parameters** with corresponding descriptions can be found at the end of the page.*

          ### Common Use Cases

          The **SnpSift filter** tool is used for filtering SnpEff-annotated VCF files to extract variants of interest.

          ### Changes Introduced by Seven Bridges

          * No modifications to the original tool representation have been made. 

          ### Common Issues and Important Notes

          * Parameter **Filter expression**: Wrapping quotes ("") are added automatically by the wrapper and should not be included in the filter expression. For example, filtering expression: "( CHROM = 'chr1' )" should be written as `( CHROM = 'chr1' )`.
          * **SnpSift filter** filtering syntax allows for creation of complex, flexible filters with different operators and functions. Please check the [official documentation](http://snpeff.sourceforge.net/SnpSift.html#filter) for examples of filter syntax and valid filtering fields before using the tool.

          ### Performance Benchmarking

          Typical runs take <10 minutes and cost <$0.1 on an on-demand default instance.

          ### References

          [1] [SnpSift filter documentation](http://snpeff.sourceforge.net/SnpSift.html#filter)
        sbg:appVersion:
        - sbg:draft-2
        sbg:categories:
        - Annotation
        - VCF-Processing
        sbg:cmdPreview: java -Xmx5m -jar /opt/snpEff/SnpSift.jar filter  > input_vcf.filtered.vcf
        sbg:content_hash:
        sbg:contributors:
        - admin
        sbg:createdBy: admin
        sbg:createdOn: 1512132000
        sbg:id: admin/sbg-public-data/snpsift-filter-4-3/2
        sbg:image_url:
        sbg:job:
          allocatedResources:
            cpu: 1
            mem: 5
          inputs:
            format: '2'
            addFilter: addFilter-string-value
            errMissing: true
            exprFile:
              class: File
              secondaryFiles: []
              path: /path/to/exprFile.ext
              size: 0
            expression: expression
            filterId: ''
            galaxy: true
            input_vcf:
              class: File
              secondaryFiles: []
              path: /path/to/input_vcf.ext
              size: 0
            inverse: true
            mem_mb: 5
            pass: true
            rmFilter: rmFilter-string-value
            set:
              class: File
              secondaryFiles: []
              path: /path/to/set.ext
              size: 0
        sbg:latestRevision: 2
        sbg:license: GNU Lesser General Public License v3.0 only
        sbg:links:
        - id: http://snpeff.sourceforge.net/SnpSift.html#filter
          label: Homepage
        - id: https://github.com/pcingola/SnpEff
          label: Source Code
        sbg:modifiedBy: admin
        sbg:modifiedOn: 1533292196
        sbg:project: admin/sbg-public-data
        sbg:projectName: SBG Public data
        sbg:publisher: sbg
        sbg:revision: 2
        sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-filter/9
        sbg:revisionsInfo:
        - sbg:modifiedBy: admin
          sbg:modifiedOn: 1512132000
          sbg:revision: 0
          sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-filter/5
        - sbg:modifiedBy: admin
          sbg:modifiedOn: 1512132000
          sbg:revision: 1
          sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-filter/8
        - sbg:modifiedBy: admin
          sbg:modifiedOn: 1533292196
          sbg:revision: 2
          sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-filter/9
        sbg:sbgMaintained: false
        sbg:toolAuthor: Pablo Cingolani/Broad Institue
        sbg:toolkit: SnpEff
        sbg:toolkitVersion: 4.3r
        sbg:validationErrors: []
      inputs:
      - id: '#snpsift_filter_4_4.input_vcf'
        source: '#snpsift_filter_4_3.filtered_vcf'
      - id: '#snpsift_filter_4_4.expression'
        default: FS > 60
      - id: '#snpsift_filter_4_4.addFilter'
        default: strandBias
      - id: '#snpsift_filter_4_4.rmFilter'
        default: PASS
      outputs:
      - id: '#snpsift_filter_4_4.filtered_vcf'
      sbg:x: -180.06483459472656
      sbg:y: -576.2888793945312
    - id: '#snpsift_filter_4_5'
      label: SnpSift filter
      in: []
      run:
        cwlVersion: sbg:draft-2
        class: CommandLineTool
        label: SnpSift filter
        $namespaces:
          sbg: https://sevenbridges.com

        requirements:
        - id: '#cwl-js-engine'
          class: ExpressionEngineRequirement
          requirements:
          - class: DockerRequirement
            dockerPull: rabix/js-engine
        - class: InlineJavascriptRequirement

        inputs:
        - id: '#mem_mb'
          label: Memory to use for the task [Mb]
          type:
          - 'null'
          - int
          description: Memory to use for the task [Mb]
          sbg:category: Options
          sbg:stageInput:
          sbg:toolDefaultValue: '4096'
        - id: '#input_vcf'
          label: Input VCF
          type:
          - 'null'
          - File
          inputBinding:
            prefix: -f
            position: 100
            separate: true
            sbg:cmdInclude: true
          description: 'VCF input file. Default: STDIN'
          sbg:fileTypes: VCF, VCF.GZ
        - id: '#expression'
          label: Filter expression (without quotes)
          type:
          - 'null'
          - string
          inputBinding:
            position: 55
            valueFrom:
              class: Expression
              engine: '#cwl-js-engine'
              script: |-
                {
                  if ($job.inputs.expression)
                  {
                    return '"'.concat($job.inputs.expression,'"')
                  }
                  else
                  {
                    return ''
                  }
                }
            separate: true
            sbg:cmdInclude: true
          description: |-
            Expression to use for filtering. Please note that wrapping quotes are added automatically by the wrapper and should not be used as part of the filtering expression.
          sbg:category: Inputs
        - id: '#addFilter'
          label: Add a string to FILTER field
          type:
          - 'null'
          - string
          inputBinding:
            prefix: -a
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: "Add a string to FILTER VCF field if 'expression' is true.\
            \ Default: '' (none)."
          sbg:altPrefix: --addFilter
          sbg:category: Options
          sbg:toolDefaultValue: "''"
        - id: '#exprFile'
          label: Expression file
          type:
          - 'null'
          - File
          inputBinding:
            prefix: -e
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Read expression from a file.
          sbg:altPrefix: --exprFile
          sbg:category: Options
        - id: '#filterId'
          label: Filter ID
          type:
          - 'null'
          - string
          inputBinding:
            prefix: -i
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            ID for this filter (##FILTER tag in header and FILTER VCF field). Default: 'SnpSift'.
          sbg:altPrefix: --filterId
          sbg:category: Options
          sbg:toolDefaultValue: SnpSift
        - id: '#inverse'
          label: Inverse
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -n
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Inverse. Show lines that do not match filter expression.
          sbg:altPrefix: --inverse
          sbg:category: Options
          sbg:stageInput:
        - id: '#pass'
          label: Pass
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -p
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Use 'PASS' field instead of filtering out VCF entries.
          sbg:altPrefix: --pass
          sbg:category: Options
        - id: '#rmFilter'
          label: Remove string from FILTER field
          type:
          - 'null'
          - string
          inputBinding:
            prefix: -r
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Remove a string from FILTER VCF field if 'expression' is true (and 'str' is in the field). Default: '' (none).
          sbg:altPrefix: --rmFilter
          sbg:category: Options
          sbg:toolDefaultValue: "''"
        - id: '#set'
          label: Set File
          type:
          - 'null'
          - File
          inputBinding:
            prefix: -s
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Create a SET using 'file'.
          sbg:altPrefix: --set
          sbg:category: Options
        - id: '#errMissing'
          label: Error on missing fields
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --errMissing
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: 'Error if a field is missing. Default: False.'
          sbg:category: Options
          sbg:stageInput:
          sbg:toolDefaultValue: 'False'
        - id: '#format'
          label: Format
          type:
          - 'null'
          - name: format
            type: enum
            symbols:
            - '2'
            - '3'
          inputBinding:
            prefix: --format
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: 'SnpEff format version: {2, 3}. Default: Auto.'
          sbg:stageInput:
          sbg:toolDefaultValue: Auto
        - id: '#galaxy'
          label: Galaxy
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --galaxy
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Used from Galaxy (expressions have been sanitized).
          sbg:category: Options
          sbg:stageInput:

        outputs:
        - id: '#filtered_vcf'
          label: Filtered VCF
          type:
          - 'null'
          - File
          outputBinding:
            glob: '*.filtered.vcf'
            sbg:inheritMetadataFrom: '#input_vcf'
          description: Filtered VCF.
          sbg:fileTypes: VCF
        stdout:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              filename = $job.inputs.input_vcf.path
              return filename.split('.').slice(0, filename.split('.').length-1).concat("filtered.vcf").join('.').replace(/^.*[\\\/]/, '')
            }

        baseCommand:
        - java
        - class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              if ($job.inputs.mem_mb)
              {
                return "-Xmx" + ($job.inputs.mem_mb).toString() + "m"
              }
              else
              {
                return "-Xmx4096m"
              }
            }
        - -jar
        - /opt/snpEff/SnpSift.jar
        - filter

        hints:
        - class: sbg:CPURequirement
          value: 1
        - class: sbg:MemRequirement
          value:
            class: Expression
            engine: '#cwl-js-engine'
            script: |-
              {
                if ($job.inputs.mem_mb)
                {
                  return $job.inputs.mem_mb
                }
                
                else
                {
                  return 4096
                }
              }
        - class: DockerRequirement
          dockerPull: images.sbgenomics.com/jrandjelovic/snpeff:v4.3r
        id: admin/sbg-public-data/snpsift-filter-4-3/2
        description: |-
          **SnpSift filter** filters SnpEff-annotated VCF files using arbitrary expressions, for instance "(QUAL > 30) | (exists INDEL) | ( countHet() > 2 )". The actual expressions can be quite complex, allowing for a lot of flexibility [1].

          *A list of **all inputs and parameters** with corresponding descriptions can be found at the end of the page.*

          ### Common Use Cases

          The **SnpSift filter** tool is used for filtering SnpEff-annotated VCF files to extract variants of interest.

          ### Changes Introduced by Seven Bridges

          * No modifications to the original tool representation have been made. 

          ### Common Issues and Important Notes

          * Parameter **Filter expression**: Wrapping quotes ("") are added automatically by the wrapper and should not be included in the filter expression. For example, filtering expression: "( CHROM = 'chr1' )" should be written as `( CHROM = 'chr1' )`.
          * **SnpSift filter** filtering syntax allows for creation of complex, flexible filters with different operators and functions. Please check the [official documentation](http://snpeff.sourceforge.net/SnpSift.html#filter) for examples of filter syntax and valid filtering fields before using the tool.

          ### Performance Benchmarking

          Typical runs take <10 minutes and cost <$0.1 on an on-demand default instance.

          ### References

          [1] [SnpSift filter documentation](http://snpeff.sourceforge.net/SnpSift.html#filter)
        sbg:appVersion:
        - sbg:draft-2
        sbg:categories:
        - Annotation
        - VCF-Processing
        sbg:cmdPreview: java -Xmx5m -jar /opt/snpEff/SnpSift.jar filter  > input_vcf.filtered.vcf
        sbg:content_hash:
        sbg:contributors:
        - admin
        sbg:createdBy: admin
        sbg:createdOn: 1512132000
        sbg:id: admin/sbg-public-data/snpsift-filter-4-3/2
        sbg:image_url:
        sbg:job:
          allocatedResources:
            cpu: 1
            mem: 5
          inputs:
            format: '2'
            addFilter: addFilter-string-value
            errMissing: true
            exprFile:
              class: File
              secondaryFiles: []
              path: /path/to/exprFile.ext
              size: 0
            expression: expression
            filterId: ''
            galaxy: true
            input_vcf:
              class: File
              secondaryFiles: []
              path: /path/to/input_vcf.ext
              size: 0
            inverse: true
            mem_mb: 5
            pass: true
            rmFilter: rmFilter-string-value
            set:
              class: File
              secondaryFiles: []
              path: /path/to/set.ext
              size: 0
        sbg:latestRevision: 2
        sbg:license: GNU Lesser General Public License v3.0 only
        sbg:links:
        - id: http://snpeff.sourceforge.net/SnpSift.html#filter
          label: Homepage
        - id: https://github.com/pcingola/SnpEff
          label: Source Code
        sbg:modifiedBy: admin
        sbg:modifiedOn: 1533292196
        sbg:project: admin/sbg-public-data
        sbg:projectName: SBG Public data
        sbg:publisher: sbg
        sbg:revision: 2
        sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-filter/9
        sbg:revisionsInfo:
        - sbg:modifiedBy: admin
          sbg:modifiedOn: 1512132000
          sbg:revision: 0
          sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-filter/5
        - sbg:modifiedBy: admin
          sbg:modifiedOn: 1512132000
          sbg:revision: 1
          sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-filter/8
        - sbg:modifiedBy: admin
          sbg:modifiedOn: 1533292196
          sbg:revision: 2
          sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-filter/9
        sbg:sbgMaintained: false
        sbg:toolAuthor: Pablo Cingolani/Broad Institue
        sbg:toolkit: SnpEff
        sbg:toolkitVersion: 4.3r
        sbg:validationErrors: []
      inputs:
      - id: '#snpsift_filter_4_5.input_vcf'
        source: '#snpsift_filter_4_4.filtered_vcf'
      - id: '#snpsift_filter_4_5.expression'
        default: MQ < 40
      - id: '#snpsift_filter_4_5.addFilter'
        default: lowMQ
      - id: '#snpsift_filter_4_5.rmFilter'
        default: PASS
      outputs:
      - id: '#snpsift_filter_4_5.filtered_vcf'
      sbg:x: -18.167282104492188
      sbg:y: -520.2623291015625
    - id: '#snpsift_filter_4_6'
      label: SnpSift filter
      in: []
      run:
        cwlVersion: sbg:draft-2
        class: CommandLineTool
        label: SnpSift filter
        $namespaces:
          sbg: https://sevenbridges.com

        requirements:
        - id: '#cwl-js-engine'
          class: ExpressionEngineRequirement
          requirements:
          - class: DockerRequirement
            dockerPull: rabix/js-engine
        - class: InlineJavascriptRequirement

        inputs:
        - id: '#mem_mb'
          label: Memory to use for the task [Mb]
          type:
          - 'null'
          - int
          description: Memory to use for the task [Mb]
          sbg:category: Options
          sbg:stageInput:
          sbg:toolDefaultValue: '4096'
        - id: '#input_vcf'
          label: Input VCF
          type:
          - 'null'
          - File
          inputBinding:
            prefix: -f
            position: 100
            separate: true
            sbg:cmdInclude: true
          description: 'VCF input file. Default: STDIN'
          sbg:fileTypes: VCF, VCF.GZ
        - id: '#expression'
          label: Filter expression (without quotes)
          type:
          - 'null'
          - string
          inputBinding:
            position: 55
            valueFrom:
              class: Expression
              engine: '#cwl-js-engine'
              script: |-
                {
                  if ($job.inputs.expression)
                  {
                    return '"'.concat($job.inputs.expression,'"')
                  }
                  else
                  {
                    return ''
                  }
                }
            separate: true
            sbg:cmdInclude: true
          description: |-
            Expression to use for filtering. Please note that wrapping quotes are added automatically by the wrapper and should not be used as part of the filtering expression.
          sbg:category: Inputs
        - id: '#addFilter'
          label: Add a string to FILTER field
          type:
          - 'null'
          - string
          inputBinding:
            prefix: -a
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: "Add a string to FILTER VCF field if 'expression' is true.\
            \ Default: '' (none)."
          sbg:altPrefix: --addFilter
          sbg:category: Options
          sbg:toolDefaultValue: "''"
        - id: '#exprFile'
          label: Expression file
          type:
          - 'null'
          - File
          inputBinding:
            prefix: -e
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Read expression from a file.
          sbg:altPrefix: --exprFile
          sbg:category: Options
        - id: '#filterId'
          label: Filter ID
          type:
          - 'null'
          - string
          inputBinding:
            prefix: -i
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            ID for this filter (##FILTER tag in header and FILTER VCF field). Default: 'SnpSift'.
          sbg:altPrefix: --filterId
          sbg:category: Options
          sbg:toolDefaultValue: SnpSift
        - id: '#inverse'
          label: Inverse
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -n
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Inverse. Show lines that do not match filter expression.
          sbg:altPrefix: --inverse
          sbg:category: Options
          sbg:stageInput:
        - id: '#pass'
          label: Pass
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -p
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Use 'PASS' field instead of filtering out VCF entries.
          sbg:altPrefix: --pass
          sbg:category: Options
        - id: '#rmFilter'
          label: Remove string from FILTER field
          type:
          - 'null'
          - string
          inputBinding:
            prefix: -r
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Remove a string from FILTER VCF field if 'expression' is true (and 'str' is in the field). Default: '' (none).
          sbg:altPrefix: --rmFilter
          sbg:category: Options
          sbg:toolDefaultValue: "''"
        - id: '#set'
          label: Set File
          type:
          - 'null'
          - File
          inputBinding:
            prefix: -s
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Create a SET using 'file'.
          sbg:altPrefix: --set
          sbg:category: Options
        - id: '#errMissing'
          label: Error on missing fields
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --errMissing
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: 'Error if a field is missing. Default: False.'
          sbg:category: Options
          sbg:stageInput:
          sbg:toolDefaultValue: 'False'
        - id: '#format'
          label: Format
          type:
          - 'null'
          - name: format
            type: enum
            symbols:
            - '2'
            - '3'
          inputBinding:
            prefix: --format
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: 'SnpEff format version: {2, 3}. Default: Auto.'
          sbg:stageInput:
          sbg:toolDefaultValue: Auto
        - id: '#galaxy'
          label: Galaxy
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --galaxy
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Used from Galaxy (expressions have been sanitized).
          sbg:category: Options
          sbg:stageInput:

        outputs:
        - id: '#filtered_vcf'
          label: Filtered VCF
          type:
          - 'null'
          - File
          outputBinding:
            glob: '*.filtered.vcf'
            sbg:inheritMetadataFrom: '#input_vcf'
          description: Filtered VCF.
          sbg:fileTypes: VCF
        stdout:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              filename = $job.inputs.input_vcf.path
              return filename.split('.').slice(0, filename.split('.').length-1).concat("filtered.vcf").join('.').replace(/^.*[\\\/]/, '')
            }

        baseCommand:
        - java
        - class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              if ($job.inputs.mem_mb)
              {
                return "-Xmx" + ($job.inputs.mem_mb).toString() + "m"
              }
              else
              {
                return "-Xmx4096m"
              }
            }
        - -jar
        - /opt/snpEff/SnpSift.jar
        - filter

        hints:
        - class: sbg:CPURequirement
          value: 1
        - class: sbg:MemRequirement
          value:
            class: Expression
            engine: '#cwl-js-engine'
            script: |-
              {
                if ($job.inputs.mem_mb)
                {
                  return $job.inputs.mem_mb
                }
                
                else
                {
                  return 4096
                }
              }
        - class: DockerRequirement
          dockerPull: images.sbgenomics.com/jrandjelovic/snpeff:v4.3r
        id: admin/sbg-public-data/snpsift-filter-4-3/2
        description: |-
          **SnpSift filter** filters SnpEff-annotated VCF files using arbitrary expressions, for instance "(QUAL > 30) | (exists INDEL) | ( countHet() > 2 )". The actual expressions can be quite complex, allowing for a lot of flexibility [1].

          *A list of **all inputs and parameters** with corresponding descriptions can be found at the end of the page.*

          ### Common Use Cases

          The **SnpSift filter** tool is used for filtering SnpEff-annotated VCF files to extract variants of interest.

          ### Changes Introduced by Seven Bridges

          * No modifications to the original tool representation have been made. 

          ### Common Issues and Important Notes

          * Parameter **Filter expression**: Wrapping quotes ("") are added automatically by the wrapper and should not be included in the filter expression. For example, filtering expression: "( CHROM = 'chr1' )" should be written as `( CHROM = 'chr1' )`.
          * **SnpSift filter** filtering syntax allows for creation of complex, flexible filters with different operators and functions. Please check the [official documentation](http://snpeff.sourceforge.net/SnpSift.html#filter) for examples of filter syntax and valid filtering fields before using the tool.

          ### Performance Benchmarking

          Typical runs take <10 minutes and cost <$0.1 on an on-demand default instance.

          ### References

          [1] [SnpSift filter documentation](http://snpeff.sourceforge.net/SnpSift.html#filter)
        sbg:appVersion:
        - sbg:draft-2
        sbg:categories:
        - Annotation
        - VCF-Processing
        sbg:cmdPreview: java -Xmx5m -jar /opt/snpEff/SnpSift.jar filter  > input_vcf.filtered.vcf
        sbg:content_hash:
        sbg:contributors:
        - admin
        sbg:createdBy: admin
        sbg:createdOn: 1512132000
        sbg:id: admin/sbg-public-data/snpsift-filter-4-3/2
        sbg:image_url:
        sbg:job:
          allocatedResources:
            cpu: 1
            mem: 5
          inputs:
            format: '2'
            addFilter: addFilter-string-value
            errMissing: true
            exprFile:
              class: File
              secondaryFiles: []
              path: /path/to/exprFile.ext
              size: 0
            expression: expression
            filterId: ''
            galaxy: true
            input_vcf:
              class: File
              secondaryFiles: []
              path: /path/to/input_vcf.ext
              size: 0
            inverse: true
            mem_mb: 5
            pass: true
            rmFilter: rmFilter-string-value
            set:
              class: File
              secondaryFiles: []
              path: /path/to/set.ext
              size: 0
        sbg:latestRevision: 2
        sbg:license: GNU Lesser General Public License v3.0 only
        sbg:links:
        - id: http://snpeff.sourceforge.net/SnpSift.html#filter
          label: Homepage
        - id: https://github.com/pcingola/SnpEff
          label: Source Code
        sbg:modifiedBy: admin
        sbg:modifiedOn: 1533292196
        sbg:project: admin/sbg-public-data
        sbg:projectName: SBG Public data
        sbg:publisher: sbg
        sbg:revision: 2
        sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-filter/9
        sbg:revisionsInfo:
        - sbg:modifiedBy: admin
          sbg:modifiedOn: 1512132000
          sbg:revision: 0
          sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-filter/5
        - sbg:modifiedBy: admin
          sbg:modifiedOn: 1512132000
          sbg:revision: 1
          sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-filter/8
        - sbg:modifiedBy: admin
          sbg:modifiedOn: 1533292196
          sbg:revision: 2
          sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-filter/9
        sbg:sbgMaintained: false
        sbg:toolAuthor: Pablo Cingolani/Broad Institue
        sbg:toolkit: SnpEff
        sbg:toolkitVersion: 4.3r
        sbg:validationErrors: []
      inputs:
      - id: '#snpsift_filter_4_6.input_vcf'
        source: '#snpsift_filter_4_5.filtered_vcf'
      - id: '#snpsift_filter_4_6.expression'
        default: MQRankSum < -12.5
      - id: '#snpsift_filter_4_6.addFilter'
        default: lowMQRankSum
      - id: '#snpsift_filter_4_6.rmFilter'
        default: PASS
      outputs:
      - id: '#snpsift_filter_4_6.filtered_vcf'
      sbg:x: 145.5419921875
      sbg:y: -472.5
    - id: '#snpsift_filter_4_7'
      label: SnpSift filter
      in: []
      run:
        cwlVersion: sbg:draft-2
        class: CommandLineTool
        label: SnpSift filter
        $namespaces:
          sbg: https://sevenbridges.com

        requirements:
        - id: '#cwl-js-engine'
          class: ExpressionEngineRequirement
          requirements:
          - class: DockerRequirement
            dockerPull: rabix/js-engine
        - class: InlineJavascriptRequirement

        inputs:
        - id: '#mem_mb'
          label: Memory to use for the task [Mb]
          type:
          - 'null'
          - int
          description: Memory to use for the task [Mb]
          sbg:category: Options
          sbg:stageInput:
          sbg:toolDefaultValue: '4096'
        - id: '#input_vcf'
          label: Input VCF
          type:
          - 'null'
          - File
          inputBinding:
            prefix: -f
            position: 100
            separate: true
            sbg:cmdInclude: true
          description: 'VCF input file. Default: STDIN'
          sbg:fileTypes: VCF, VCF.GZ
        - id: '#expression'
          label: Filter expression (without quotes)
          type:
          - 'null'
          - string
          inputBinding:
            position: 55
            valueFrom:
              class: Expression
              engine: '#cwl-js-engine'
              script: |-
                {
                  if ($job.inputs.expression)
                  {
                    return '"'.concat($job.inputs.expression,'"')
                  }
                  else
                  {
                    return ''
                  }
                }
            separate: true
            sbg:cmdInclude: true
          description: |-
            Expression to use for filtering. Please note that wrapping quotes are added automatically by the wrapper and should not be used as part of the filtering expression.
          sbg:category: Inputs
        - id: '#addFilter'
          label: Add a string to FILTER field
          type:
          - 'null'
          - string
          inputBinding:
            prefix: -a
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: "Add a string to FILTER VCF field if 'expression' is true.\
            \ Default: '' (none)."
          sbg:altPrefix: --addFilter
          sbg:category: Options
          sbg:toolDefaultValue: "''"
        - id: '#exprFile'
          label: Expression file
          type:
          - 'null'
          - File
          inputBinding:
            prefix: -e
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Read expression from a file.
          sbg:altPrefix: --exprFile
          sbg:category: Options
        - id: '#filterId'
          label: Filter ID
          type:
          - 'null'
          - string
          inputBinding:
            prefix: -i
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            ID for this filter (##FILTER tag in header and FILTER VCF field). Default: 'SnpSift'.
          sbg:altPrefix: --filterId
          sbg:category: Options
          sbg:toolDefaultValue: SnpSift
        - id: '#inverse'
          label: Inverse
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -n
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Inverse. Show lines that do not match filter expression.
          sbg:altPrefix: --inverse
          sbg:category: Options
          sbg:stageInput:
        - id: '#pass'
          label: Pass
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -p
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Use 'PASS' field instead of filtering out VCF entries.
          sbg:altPrefix: --pass
          sbg:category: Options
        - id: '#rmFilter'
          label: Remove string from FILTER field
          type:
          - 'null'
          - string
          inputBinding:
            prefix: -r
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Remove a string from FILTER VCF field if 'expression' is true (and 'str' is in the field). Default: '' (none).
          sbg:altPrefix: --rmFilter
          sbg:category: Options
          sbg:toolDefaultValue: "''"
        - id: '#set'
          label: Set File
          type:
          - 'null'
          - File
          inputBinding:
            prefix: -s
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Create a SET using 'file'.
          sbg:altPrefix: --set
          sbg:category: Options
        - id: '#errMissing'
          label: Error on missing fields
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --errMissing
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: 'Error if a field is missing. Default: False.'
          sbg:category: Options
          sbg:stageInput:
          sbg:toolDefaultValue: 'False'
        - id: '#format'
          label: Format
          type:
          - 'null'
          - name: format
            type: enum
            symbols:
            - '2'
            - '3'
          inputBinding:
            prefix: --format
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: 'SnpEff format version: {2, 3}. Default: Auto.'
          sbg:stageInput:
          sbg:toolDefaultValue: Auto
        - id: '#galaxy'
          label: Galaxy
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --galaxy
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Used from Galaxy (expressions have been sanitized).
          sbg:category: Options
          sbg:stageInput:

        outputs:
        - id: '#filtered_vcf'
          label: Filtered VCF
          type:
          - 'null'
          - File
          outputBinding:
            glob: '*.filtered.vcf'
            sbg:inheritMetadataFrom: '#input_vcf'
          description: Filtered VCF.
          sbg:fileTypes: VCF
        stdout:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              filename = $job.inputs.input_vcf.path
              return filename.split('.').slice(0, filename.split('.').length-1).concat("filtered.vcf").join('.').replace(/^.*[\\\/]/, '')
            }

        baseCommand:
        - java
        - class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              if ($job.inputs.mem_mb)
              {
                return "-Xmx" + ($job.inputs.mem_mb).toString() + "m"
              }
              else
              {
                return "-Xmx4096m"
              }
            }
        - -jar
        - /opt/snpEff/SnpSift.jar
        - filter

        hints:
        - class: sbg:CPURequirement
          value: 1
        - class: sbg:MemRequirement
          value:
            class: Expression
            engine: '#cwl-js-engine'
            script: |-
              {
                if ($job.inputs.mem_mb)
                {
                  return $job.inputs.mem_mb
                }
                
                else
                {
                  return 4096
                }
              }
        - class: DockerRequirement
          dockerPull: images.sbgenomics.com/jrandjelovic/snpeff:v4.3r
        id: admin/sbg-public-data/snpsift-filter-4-3/2
        description: |-
          **SnpSift filter** filters SnpEff-annotated VCF files using arbitrary expressions, for instance "(QUAL > 30) | (exists INDEL) | ( countHet() > 2 )". The actual expressions can be quite complex, allowing for a lot of flexibility [1].

          *A list of **all inputs and parameters** with corresponding descriptions can be found at the end of the page.*

          ### Common Use Cases

          The **SnpSift filter** tool is used for filtering SnpEff-annotated VCF files to extract variants of interest.

          ### Changes Introduced by Seven Bridges

          * No modifications to the original tool representation have been made. 

          ### Common Issues and Important Notes

          * Parameter **Filter expression**: Wrapping quotes ("") are added automatically by the wrapper and should not be included in the filter expression. For example, filtering expression: "( CHROM = 'chr1' )" should be written as `( CHROM = 'chr1' )`.
          * **SnpSift filter** filtering syntax allows for creation of complex, flexible filters with different operators and functions. Please check the [official documentation](http://snpeff.sourceforge.net/SnpSift.html#filter) for examples of filter syntax and valid filtering fields before using the tool.

          ### Performance Benchmarking

          Typical runs take <10 minutes and cost <$0.1 on an on-demand default instance.

          ### References

          [1] [SnpSift filter documentation](http://snpeff.sourceforge.net/SnpSift.html#filter)
        sbg:appVersion:
        - sbg:draft-2
        sbg:categories:
        - Annotation
        - VCF-Processing
        sbg:cmdPreview: java -Xmx5m -jar /opt/snpEff/SnpSift.jar filter  > input_vcf.filtered.vcf
        sbg:content_hash:
        sbg:contributors:
        - admin
        sbg:createdBy: admin
        sbg:createdOn: 1512132000
        sbg:id: admin/sbg-public-data/snpsift-filter-4-3/2
        sbg:image_url:
        sbg:job:
          allocatedResources:
            cpu: 1
            mem: 5
          inputs:
            format: '2'
            addFilter: addFilter-string-value
            errMissing: true
            exprFile:
              class: File
              secondaryFiles: []
              path: /path/to/exprFile.ext
              size: 0
            expression: expression
            filterId: ''
            galaxy: true
            input_vcf:
              class: File
              secondaryFiles: []
              path: /path/to/input_vcf.ext
              size: 0
            inverse: true
            mem_mb: 5
            pass: true
            rmFilter: rmFilter-string-value
            set:
              class: File
              secondaryFiles: []
              path: /path/to/set.ext
              size: 0
        sbg:latestRevision: 2
        sbg:license: GNU Lesser General Public License v3.0 only
        sbg:links:
        - id: http://snpeff.sourceforge.net/SnpSift.html#filter
          label: Homepage
        - id: https://github.com/pcingola/SnpEff
          label: Source Code
        sbg:modifiedBy: admin
        sbg:modifiedOn: 1533292196
        sbg:project: admin/sbg-public-data
        sbg:projectName: SBG Public data
        sbg:publisher: sbg
        sbg:revision: 2
        sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-filter/9
        sbg:revisionsInfo:
        - sbg:modifiedBy: admin
          sbg:modifiedOn: 1512132000
          sbg:revision: 0
          sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-filter/5
        - sbg:modifiedBy: admin
          sbg:modifiedOn: 1512132000
          sbg:revision: 1
          sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-filter/8
        - sbg:modifiedBy: admin
          sbg:modifiedOn: 1533292196
          sbg:revision: 2
          sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-filter/9
        sbg:sbgMaintained: false
        sbg:toolAuthor: Pablo Cingolani/Broad Institue
        sbg:toolkit: SnpEff
        sbg:toolkitVersion: 4.3r
        sbg:validationErrors: []
      inputs:
      - id: '#snpsift_filter_4_7.input_vcf'
        source: '#snpsift_filter_4_6.filtered_vcf'
      - id: '#snpsift_filter_4_7.expression'
        default: ReadPosRankSum < -8
      - id: '#snpsift_filter_4_7.addFilter'
        default: lowReadPosRankSum
      - id: '#snpsift_filter_4_7.rmFilter'
        default: PASS
      outputs:
      - id: '#snpsift_filter_4_7.filtered_vcf'
      sbg:x: 309.0456848144531
      sbg:y: -407.882080078125
    - id: '#snpeff'
      label: SnpEff
      in: []
      run:
        cwlVersion: sbg:draft-2
        class: CommandLineTool
        label: SnpEff
        $namespaces:
          sbg: https://sevenbridges.com

        requirements:
        - id: '#cwl-js-engine'
          class: ExpressionEngineRequirement
          requirements:
          - class: DockerRequirement
            dockerPull: rabix/js-engine
        - class: InlineJavascriptRequirement

        inputs:
        - id: '#variants_file'
          label: Input variants file
          type:
          - File
          inputBinding:
            position: 2001
            separate: true
            sbg:cmdInclude: true
          description: Input variants file.
          required: true
          sbg:category: File type inputs
          sbg:fileTypes: VCF, TXT, PILEUP, BED
        - id: '#up_down_stream_len'
          label: Upstream downstream interval length
          type:
          - 'null'
          - int
          inputBinding:
            prefix: -upDownStreamLen
            position: 490
            separate: true
            sbg:cmdInclude: true
          description: Set upstream downstream interval length (in bases).
          sbg:altPrefix: -ud
          sbg:category: Database options
        - id: '#total_memory'
          label: Java memory requirement [Gb]
          type:
          - 'null'
          - int
          description: RAM requirement for the java process execution [Gb].
          sbg:category: Other input types
          sbg:toolDefaultValue: '8'
        - id: '#threads'
          label: Use multiple threads (implies '-noStats')
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -t
            position: 280
            separate: true
            sbg:cmdInclude: true
          description: "Use multiple threads (implies '-noStats'). Default: False."
          sbg:category: Other input types
          sbg:toolDefaultValue: 'False'
        - id: '#strict'
          label: Only use validated transcripts
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -strict
            position: 480
            separate: true
            sbg:cmdInclude: true
          description: |-
            Only use 'validated' transcripts (i.e. sequence has been checked). Default: false.
          sbg:category: Database options
          sbg:toolDefaultValue: 'False'
        - id: '#stats'
          label: Name of stats file (summary)
          type:
          - 'null'
          - string
          inputBinding:
            prefix: -stats
            position: 60
            valueFrom:
              class: Expression
              engine: '#cwl-js-engine'
              script: |-
                {
                  if ($job.inputs.stats)
                  {
                    filename = $job.inputs.variants_file.path.split('/').pop()
                    primename = filename.split('.vcf')[0]
                    return primename.concat(".html")
                  }
                }
            separate: true
            sbg:cmdInclude: true
          description: Name of stats file (summary).
          sbg:altPrefix: -s
          sbg:category: Options
          sbg:toolDefaultValue: snpEff_summary.html
        - id: '#splicesitesize'
          label: Set size for splice sites (donor and acceptor) in bases
          type:
          - 'null'
          - int
          inputBinding:
            prefix: --spliceSiteSize
            position: 430
            separate: true
            sbg:cmdInclude: true
          description: Set size for splice sites (donor and acceptor) in bases.
          sbg:altPrefix: -ss
          sbg:category: Database options
          sbg:toolDefaultValue: '2'
        - id: '#splice_site_region_intron_max'
          label: Set maximum number of bases for splice site region within intron
          type:
          - 'null'
          - int
          inputBinding:
            prefix: -spliceRegionIntronMax
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Set maximum number of bases for splice site region within intron. Default: 8 bases.
          sbg:category: Database options
          sbg:stageInput:
          sbg:toolDefaultValue: '8'
        - id: '#splice_region_intron_min'
          label: Set minimum number of bases for splice site region within intron
          type:
          - 'null'
          - int
          inputBinding:
            prefix: -spliceRegionIntronMin
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Set minimum number of bases for splice site region within intron. Default: 3 bases.
          sbg:category: Database options
          sbg:stageInput:
          sbg:toolDefaultValue: '3'
        - id: '#splice_region_exons_size'
          label: Set size for splice site region within exons
          type:
          - 'null'
          - int
          inputBinding:
            prefix: -spliceRegionExonSize
            position: 440
            separate: true
            sbg:cmdInclude: true
          description: 'Set size for splice site region within exons. Default: 3 bases.'
          sbg:category: Database options
          sbg:toolDefaultValue: '3'
        - id: '#sequenceontology'
          label: Use Sequence Ontology terms
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -sequenceOntology
            position: 250
            separate: true
            sbg:cmdInclude: true
          description: Use Sequence Ontology terms.
          sbg:category: Annotations options
          sbg:toolDefaultValue: 'True'
        - id: '#reg'
          label: Regulation track to use (this option can be used add several times)
          type:
          - 'null'
          - type: array
            items: string
          inputBinding:
            prefix: -reg
            position: 420
            separate: true
            sbg:cmdInclude: true
          description: Regulation track to use (this option can be used add several
            times).
          sbg:category: Database options
        - id: '#output_format'
          label: Ouput format
          type:
          - 'null'
          - name: output_format
            type: enum
            symbols:
            - vcf
            - gatk
            - bed
            - bedAnn
          inputBinding:
            prefix: -o
            position: 50
            separate: true
            sbg:cmdInclude: true
          description: 'Ouput format. Possible values: {txt, vcf, gatk, bed, bedAnn}.'
          sbg:category: Options
          sbg:toolDefaultValue: vcf
        - id: '#onlyreg'
          label: Only use regulation tracks
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -onlyReg
            position: 400
            separate: true
            sbg:cmdInclude: true
          description: Only use regulation tracks.
          sbg:category: Database options
        - id: '#only_protein'
          label: Only use protein coding transcripts
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -onlyProtein
            position: 410
            separate: true
            sbg:cmdInclude: true
          description: 'Only use protein coding transcripts. Default: false.'
          sbg:category: Database options
          sbg:toolDefaultValue: 'False'
        - id: '#onlyTr'
          label: Only use transcripts from this file
          type:
          - 'null'
          - File
          inputBinding:
            prefix: -onlyTr
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: 'Only use the transcripts in this file. Format: One transcript
            ID per line.'
          required: false
          sbg:category: Database options
          sbg:fileTypes: TXT
        - id: '#oicr'
          label: Add OICR tag in VCF file
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -oicr
            position: 240
            separate: true
            sbg:cmdInclude: true
          description: Add OICR tag in VCF file.
          sbg:category: Annotations options
          sbg:toolDefaultValue: 'False'
        - id: '#nostats'
          label: Do not create stats (summary) file
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -noStats
            position: 70
            separate: true
            sbg:cmdInclude: true
          description: Do not create stats (summary) file.
          sbg:category: Options
        - id: '#no_utr'
          label: Do not show 5_PRIME_UTR or 3_PRIME_UTR changes
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -no-utr
            position: 130
            separate: true
            sbg:cmdInclude: true
          description: Do not show 5_PRIME_UTR or 3_PRIME_UTR changes.
          sbg:category: Results filter options
        - id: '#no_upstream'
          label: Do not show upstream changes
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -no-upstream
            position: 120
            separate: true
            sbg:cmdInclude: true
          description: Do not show upstream changes.
          sbg:category: Results filter options
        - id: '#no_shift_hgvs'
          label: Do not shift variants according to HGVS
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -noShiftHgvs
            position: 230
            separate: true
            sbg:cmdInclude: true
          description: Do not shift variants according to HGVS notation (most 3-prime
            end).
          sbg:category: Annotations options
          sbg:toolDefaultValue: 'False'
        - id: '#no_next_prot'
          label: Disable NextProt annotations
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -noNextProt
            position: 390
            separate: true
            sbg:cmdInclude: true
          description: Disable NextProt annotations.
          sbg:category: Database options
          sbg:stageInput:
        - id: '#no_motif'
          label: Disable motif annotations
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -noMotif
            position: 380
            separate: true
            sbg:cmdInclude: true
          description: Disable motif annotations.
          sbg:category: Database options
          sbg:stageInput:
        - id: '#no_lof'
          label: Do not add LOF and NMD annotations
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -noLof
            position: 220
            separate: true
            sbg:cmdInclude: true
          description: Do not add LOF and NMD annotations.
          sbg:category: Annotations options
          sbg:toolDefaultValue: 'False'
        - id: '#no_intron'
          label: Do not show intron changes
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -no-intron
            position: 110
            separate: true
            sbg:cmdInclude: true
          description: Do not show intron changes.
          sbg:category: Results filter options
        - id: '#no_intergenic'
          label: Do not show intergenic changes
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -no-intergenic
            position: 100
            separate: true
            sbg:cmdInclude: true
          description: Do not show intergenic changes.
          sbg:category: Results filter options
        - id: '#no_interaction'
          label: Disable interaction annotations
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -noInteraction
            position: 370
            separate: true
            sbg:cmdInclude: true
          description: Disable interaction annotations.
          sbg:category: Database options
          sbg:toolDefaultValue: 'False'
        - id: '#no_hgvs'
          label: Do not add HGVS annotations
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -noHgvs
            position: 210
            separate: true
            sbg:cmdInclude: true
          description: Do not add HGVS annotations.
          sbg:category: Annotations options
          sbg:stageInput:
        - id: '#no_genome'
          label: Do not load any genomic database
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -noGenome
            position: 360
            separate: true
            sbg:cmdInclude: true
          description: Do not load any genomic database (e.g. annotate using custom
            files).
          sbg:category: Database options
          sbg:stageInput:
        - id: '#no_downstream'
          label: Do not show downstream changes
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -no-downstream
            position: 90
            separate: true
            sbg:cmdInclude: true
          description: Do not show downstream changes.
          sbg:category: Results filter options
        - id: '#no_EffectType'
          label: Do not show EffectType
          type:
          - 'null'
          - type: array
            items: string
          inputBinding:
            prefix: -no
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Do not show 'EffectType'. This option can be used several times.
          sbg:category: Results filter options
        - id: '#noExpandIUB'
          label: Disable IUB code expansion
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -noExpandIUB
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Disable IUB code expansion in input variants.
          sbg:category: Database options
        - id: '#nextprot'
          label: Annotate using NextProt (requires NextProt database)
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -nextProt
            position: 350
            separate: true
            sbg:cmdInclude: true
          description: Annotate using NextProt (requires NextProt database).
          sbg:category: Database options
        - id: '#motif'
          label: Annotate using motifs (requires Motif database)
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -motif
            position: 340
            separate: true
            sbg:cmdInclude: true
          description: Annotate using motifs (requires Motif database).
          sbg:category: Database options
          sbg:toolDefaultValue: 'True'
        - id: '#max_tsl'
          label: Max TSL
          type:
          - 'null'
          - int
          inputBinding:
            prefix: -maxTSL
            position: 330
            separate: true
            sbg:cmdInclude: true
          description: Only use transcripts having Transcript Support Level lower
            than this cutoff.
          sbg:category: Database options
        - id: '#interval'
          label: |-
            Use a custom intervals in TXT/BED/BigBed/VCF/GFF file (you may use this option many times)
          type:
          - 'null'
          - type: array
            items: File
          inputBinding:
            prefix: -interval
            position: 320
            separate: true
            sbg:cmdInclude: true
          description: |-
            Use a custom intervals in TXT/BED/BigBed/VCF/GFF file (you may use this option many times).
          required: false
          sbg:category: Database options
          sbg:fileTypes: TXT, BED, BigBed, VCF, GFF
          sbg:stageInput:
        - id: '#interaction'
          label: Annotate using interactions
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -interaction
            position: 310
            separate: true
            sbg:cmdInclude: true
          description: 'Annotate using inteactions (requires interaciton database).
            Default: true.'
          sbg:category: Database options
          sbg:toolDefaultValue: 'True'
        - id: '#input_format'
          label: Input format
          type:
          - 'null'
          - name: input_format
            type: enum
            symbols:
            - vcf
            - ' bed'
          inputBinding:
            prefix: -i
            position: 40
            separate: true
            sbg:cmdInclude: true
          description: 'Input format. Possible values: {vcf, txt, pileup, bed}. [Default:
            vcf].'
          sbg:category: Options
          sbg:toolDefaultValue: vcf
        - id: '#hgvs_tr_id'
          label: Use transcript ID in HGVS
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -hgvsTrId
            position: 190
            separate: true
            sbg:cmdInclude: true
          description: 'Use transcript ID in HGVS notation. Default: false.'
          sbg:category: Annotations options
          sbg:toolDefaultValue: 'False'
        - id: '#hgvs_1_letter'
          label: Use one letter Amino acid codes in HGVS
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -hgvs1LetterAa
            position: 180
            separate: true
            sbg:cmdInclude: true
          description: 'Use one letter Amino acid codes in HGVS notation. Default:
            false.'
          sbg:category: Annotations options
          sbg:toolDefaultValue: 'False'
        - id: '#hgvsOld'
          label: Use old HGVS notation
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -hgvsOld
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Use old HGVS notation.
          sbg:category: Annotations options
          sbg:toolDefaultValue: 'False'
        - id: '#geneid'
          label: Use gene ID instead of gene name (VCF output)
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -geneId
            position: 170
            separate: true
            sbg:cmdInclude: true
          description: 'Use gene ID instead of gene name (VCF output). Default: false.'
          sbg:category: Annotations options
          sbg:toolDefaultValue: 'False'
        - id: '#format_eff'
          label: Use EFF field
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -formatEff
            position: 160
            separate: true
            sbg:cmdInclude: true
          description: Use 'EFF' field compatible with older versions (instead of
            'ANN').
          sbg:category: Annotations options
        - id: '#filterinterval'
          label: |-
            Only analyze changes that intersect with the intervals specified in this file (you may use this option many times)
          type:
          - 'null'
          - type: array
            items: File
          inputBinding:
            prefix: -filterInterval
            position: 80
            separate: true
            sbg:cmdInclude: true
          description: Only analyze changes that intersect with the intervals specified
            in this file.
          required: false
          sbg:altPrefix: -fi
          sbg:category: Results filter options
          sbg:fileTypes: interval
        - id: '#database'
          label: SnpEff database file
          type:
          - File
          description: |-
            SnpEff database file is zip archive that can be downloaded from the SnpEff official site, or using the SnpEff download app.
          required: true
          sbg:category: File type inputs
          sbg:fileTypes: ZIP
        - id: '#csvstats'
          label: Create CSV summary file alongside HTML
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -csvStats
            position: 20
            valueFrom:
              class: Expression
              engine: '#cwl-js-engine'
              script: |
                {
                  if ($job.inputs.csvstats)
                  {
                    filename = $job.inputs.variants_file.path.split('/').pop()
                    primename = filename.split('.vcf')[0]
                    return primename.concat(".csv")
                  }
                }
            separate: true
            sbg:cmdInclude: true
          description: Create CSV summary file alongside HTML.
          sbg:category: Options
          sbg:stageInput:
        - id: '#configuration_file'
          label: Configuration file
          type:
          - 'null'
          - File
          inputBinding:
            prefix: -c
            position: 260
            separate: true
            sbg:cmdInclude: true
          description: Specify config file.
          required: false
          sbg:altPrefix: -config
          sbg:category: Generic options
          sbg:fileTypes: config
          sbg:stageInput: link
        - id: '#configOption'
          label: Override a config file option (name=value format)
          type:
          - 'null'
          - type: array
            items: string
          inputBinding:
            prefix: -configOption
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Override a config file option. Please note that the options should be entered in name=value format.
          sbg:category: Generic options
        - id: '#classic'
          label: Use old style annotations
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -classic
            position: 10
            separate: true
            sbg:cmdInclude: true
          description: Use old style annotations instead of Sequence Ontology and
            Hgvs.
          sbg:category: General options
        - id: '#chr'
          label: String to prepend to chromosome names
          type:
          - 'null'
          - string
          inputBinding:
            prefix: -chr
            position: 55
            separate: true
            sbg:cmdInclude: true
          description: |-
            Prepend 'string' to chromosome name (e.g. 'chr1' instead of '1'). Only on TXT output.
          sbg:category: Options
        - id: '#canonList'
          label: Canon list file
          type:
          - 'null'
          - File
          inputBinding:
            prefix: -canonList
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Only use canonical transcripts, replace some transcripts using the 'gene_id 	 transcript_id' entries in <file>.
          required: false
          sbg:category: Database options
        - id: '#canon'
          label: Only use canonical transcripts
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -canon
            position: 300
            separate: true
            sbg:cmdInclude: true
          description: Only use canonical transcripts.
          sbg:category: Database options
        - id: '#cancersamples'
          label: Two column TXT file defining 'original and derived' samples
          type:
          - 'null'
          - File
          inputBinding:
            prefix: -cancerSamples
            position: 150
            separate: true
            sbg:cmdInclude: true
          description: Two column TXT file defining 'original \t derived' samples.
          required: false
          sbg:category: Annotations options
          sbg:fileTypes: TXT
        - id: '#cancer'
          label: Perform 'cancer' comparisons (somatic vs. germline)
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -cancer
            position: 140
            separate: true
            sbg:cmdInclude: true
          description: Perform 'cancer' comparisons (somatic vs. germline).
          sbg:category: Annotations options
          sbg:toolDefaultValue: 'False'
        - id: '#assembly'
          label: Assembly (genome version)
          type:
          - string
          inputBinding:
            position: 2000
            separate: false
            sbg:cmdInclude: true
          description: |-
            Genome version matching the SnpEff database used (for example GRCh37.75 or GRCh38.86).
          sbg:category: Inputs
        - id: '#add_lof_tag'
          label: Add loss of function (LOF) and nonsense mediated decay (NMD) tags
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -lof
            position: 200
            separate: true
            sbg:cmdInclude: true
          description: Add loss of function (LOF) and nonsense mediated decay (NMD)
            tags.
          sbg:category: Annotations options
        - id: '#add_hgvs_anno'
          label: Use HGVS annotations for amino acid sub-field
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -hgvs
            position: 170
            separate: true
            sbg:cmdInclude: true
          description: 'Use HGVS annotations for amino acid sub-field. Default: true.'
          sbg:category: Annotations options
          sbg:toolDefaultValue: 'True'

        outputs:
        - id: '#summary_text'
          label: Summary
          type:
          - 'null'
          - File
          outputBinding:
            glob: '*.txt'
            sbg:inheritMetadataFrom: '#variants_file'
          description: SnpEff Summary in text format.
          sbg:fileTypes: TXT
        - id: '#summary'
          label: Summary file
          type:
          - 'null'
          - type: array
            items: File
          outputBinding:
            glob: '*.{html,csv}'
            sbg:inheritMetadataFrom: '#variants_file'
          description: SnpEff summary file in HTML or CSV file format.
          sbg:fileTypes: HTML, CSV
        - id: '#annotated'
          label: SnpEff annotated file
          type:
          - 'null'
          - File
          outputBinding:
            glob:
              class: Expression
              engine: '#cwl-js-engine'
              script: |-
                {	
                  filename = $job.inputs.variants_file.path
                  basename = filename.split('.').slice(0, filename.split('.').length-1).join('.').replace(/^.*[\\\/]/, '')
                 
                  
                  if($job.inputs.output_format === "txt")
                  {
                    return basename.concat(".snpEff_annotated.txt")
                  }
                  else if ($job.inputs.output_format === "bed" || $job.inputs.output_format === "bedAnn")
                  {
                    return basename.concat(".snpEff_annotated.bed")
                  }
                  else
                  {
                    return basename.concat(".snpEff_annotated.vcf")
                  }
                }
            sbg:inheritMetadataFrom: '#variants_file'
          description: SnpEff annotated file.
          sbg:fileTypes: VCF, TXT, GATK, BED, BEDANN

        baseCommand:
        - class: Expression
          engine: '#cwl-js-engine'
          script: "{\n  return 'unzip -o ' + $job.inputs.database.path + ' -d /opt/snpEff\
            \ ;'\n}"
        - java
        - class: Expression
          engine: '#cwl-js-engine'
          script: |+
            {
              //java -Xmx4g path/to/snpEff/snpEff.jar -c path/to/snpEff/snpEff.config GRCh37.75 path/to/snps.vcf
              if($job.inputs.total_memory){
                mem_mb = parseInt($job.inputs.total_memory) * 1024
              	return '-Xmx'.concat(mem_mb, 'M')
              }
              	return '-Xmx8192M'
            }

        - -jar
        - /opt/snpEff/snpEff.jar
        arguments:
        - position: 5000
          valueFrom:
            class: Expression
            engine: '#cwl-js-engine'
            script: |
              {	
                filename = $job.inputs.variants_file.path
                basename = filename.split('.').slice(0, filename.split('.').length-1).join('.').replace(/^.*[\\\/]/, '')
               
                
                if($job.inputs.output_format === "txt")
                {
                  name = basename.concat(".snpEff_annotated.txt")
                }
                else if ($job.inputs.output_format === "bed" || $job.inputs.output_format === "bedAnn")
                {
                  name = basename.concat(".snpEff_annotated.bed")
                }
                else
                {
                  name = basename.concat(".snpEff_annotated.vcf")
                }
                return '> ' + name
              }
          separate: true
        - position: 0
          valueFrom: -nodownload
          separate: true
        - position: 0
          valueFrom: -noLog
          separate: true

        hints:
        - class: sbg:CPURequirement
          value: 1
        - class: sbg:MemRequirement
          value:
            class: Expression
            engine: '#cwl-js-engine'
            script: |-
              {
                if ($job.inputs.total_memory)
                {
                  return $job.inputs.total_memory * 1024
                }
                
                else
                {
                  return 8192
                }
              }
        - class: DockerRequirement
          dockerPull: images.sbgenomics.com/jrandjelovic/snpeff:v4.3k
          dockerImageId: aae3dcb89b53
        id: michael_lloyd/variant-calling-decomposition/snpeff/0
        abg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpeff-4-2/29
        description: |-
          **SnpEff** is a variant annotation and effect prediction​ tool, which annotates and predicts the effects of variants on genes, such as amino acid changes [1].

          *A list of **all inputs and parameters** with corresponding descriptions can be found at the end of the page.*

          ### Common Use Cases

          Typical usage assumes predicted variants (SNPs, insertions, deletions, and MNPs) as input, usually in variant call format (VCF). **SnpEff** analyzes and annotates input variants and calculates the effects they produce on known genes [1]. The output file can be in several file formats, most common being VCF.

          **SnpEff** requires an annotation database to run. Official SnpEff annotation databases can be downloaded from [here](https://sourceforge.net/projects/snpeff/files/databases/v4_3/); however, human databases are also hosted on Seven Bridges, in Public Reference Files section (files snpEff_v4_3_GRCh38.86.zip and	snpEff_v4_3_GRCh37.75.zip) and can be [imported](https://docs.sevenbridges.com/docs/copy-files-using-the-visual-interface).

          ### Changes Introduced by Seven Bridges

          * Input VCF file (**Input variants file**) is required (as opposed to default input being STDIN).
          * Parameter **Java memory requirement [Gb]** which controls the amount of RAM available to **SnpEff** was included in the wrapper.
          * The following parameters have been excluded from the wrapper:  
              * `-download`  - The tool is expected to use data from the provided database archive.
              * `-fileList` - Processing multiple files can be achieved by using batch tasks or scatter mode in workflows.
              * `-dataDir <path>`  - In the wrapper, data directory is always the same and corresponds to the location of the prepared **SnpEff** database.
              * `-download`  - Supplying a database archive as an input is required. Downloading missing data for a genome from command line is not supported.
              * `-help`, `-quiet`, `-verbose`, `-debug` and `-version` - These options are not usually included in Seven Bridges wrappers.

          ### Common Issues and Important Notes

          * Required inputs are **Input variants file** (a VCF or VCF.GZ file to be annotated), **SnpEff database file** (SnpEff database ZIP archive matching the major version of SnpEff used [2], which is 4.3 for this wrapper; e.g. snpEff_v4_3_GRCh38.86.zip or snpEff_v4_3_GRCh37.75.zip from Public Reference Files section), and **Assembly (genome version)**, which is a string representing genome version/assembly (e.g., GRCh38.86, GRCh37.75, hg19), matching the SnpEff database used (GRCh38.86 and GRCh37.75 should be used for the files in the Public Reference Files section).
          * As **SnpEff** is a java tool, it may be occasionally necessary to increase the amount of allocated RAM (default value: 8192 MB), using the **Java memory requirement [Gb]** parameter.
          * A number of **SnpEff** command line options are designed in mutually exclusive pairs (for example `-noStats` and `-stats` or `-lof` and `-noLof`) with some redundancy. These options should not be used together, to avoid task failure.
          * Multithreading parameter **Use multiple threads (implies '-noStats')** (`-t`) will disable statistics. 
          * Disabling statistics using **Do not create stats (summary) file** (`-noStats`) will in general speed-up execution.

          ### Performance Benchmarking

          Annotating NA12878 genome (GRCh38, ~220 Mb as VCF.GZ) with default annotation parameters, 1 CPU, and 8192 MB RAM took 25 minutes with a cost of $0.17 using on-demand default instance.
          By default, **SnpEff** is allocated 8192 MB of memory. Allocating less memory is not recommended when working with whole genome VCF files.

          *Cost can be significantly reduced by **spot instance** usage. Visit [knowledge center](https://docs.sevenbridges.com/docs/about-spot-instances) for more details.*            

          ### References

          [1] [SnpEff documentation](http://snpeff.sourceforge.net/SnpEff_manual.html)

          [2] [Official SnpEff 4.3 databases download location](https://sourceforge.net/projects/snpeff/files/databases/v4_3/)
        sbg:appVersion:
        - sbg:draft-2
        sbg:categories:
        - Annotation
        - VCF-Processing
        sbg:cmdPreview: |-
          unzip -o /path/to/database/GRCh37.75.zip -d /opt/snpEff ; java -Xmx3072M -jar /opt/snpEff/snpEff.jar  -nodownload  -noLog GRCh37.75  path/to/variants/variants_file.vcf  > variants_file.snpEff_annotated.vcf
        sbg:content_hash: a1f0830cf4a189d917e82af4f6f1abbd3dfa7759940ab3e6d94f3e2fc5adcc5b1
        sbg:contributors:
        - jelena_randjelovic
        sbg:createdBy: jelena_randjelovic
        sbg:createdOn: 1552058593
        sbg:id: michael_lloyd/variant-calling-decomposition/snpeff/0
        sbg:image_url:
        sbg:job:
          allocatedResources:
            cpu: 1
            mem: 3072
          inputs:
            add_hgvs_anno: false
            assembly: GRCh37.75
            chr: ''
            classic: false
            configOption:
            configuration_file:
              class: File
              secondaryFiles: []
              path:
              size: 0
            csvstats: true
            database:
              secondaryFiles: []
              metadata: {}
              path: /path/to/database/GRCh37.75.zip
            format_eff: false
            hgvsOld: true
            hgvs_1_letter: false
            hgvs_tr_id: false
            input_format: vcf
            interaction: false
            interval:
            - class: File
              secondaryFiles: []
              path: /path/to/interval-1.ext
              size: 0
            - class: File
              secondaryFiles: []
              path: /path/to/interval-2.ext
              size: 0
            max_tsl:
            noExpandIUB: true
            no_EffectType:
            - no_EffectType-string-value-1
            - no_EffectType-string-value-2
            no_genome: false
            no_hgvs: false
            no_interaction: false
            no_lof: false
            no_motif: false
            no_next_prot: false
            no_shift_hgvs: false
            nostats: false
            oicr: false
            onlyTr:
              class: File
              secondaryFiles: []
              path: /path/to/onlyTr.ext
              size: 0
            only_protein: false
            output_format: vcf
            splice_region_exons_size:
            splice_region_intron_min: 7
            splice_site_region_intron_max: 8
            stats: st
            strict: false
            threads: false
            total_memory: 3
            up_down_stream_len:
            variants_file:
              class: File
              secondaryFiles: []
              path: path/to/variants/variants_file.vcf
              size: 0
        sbg:latestRevision: 0
        sbg:license: GNU Lesser General Public License v3.0 only
        sbg:links:
        - id: http://snpeff.sourceforge.net/index.html
          label: Homepage
        - id: https://github.com/pcingola/SnpEff
          label: Source Code
        - id: http://snpeff.sourceforge.net/SnpEff_manual.html
          label: Wiki
        - id: http://sourceforge.net/projects/snpeff/files/snpEff_latest_core.zip
          label: Download
        - id: http://snpeff.sourceforge.net/SnpEff_paper.pdf
          label: Publication
        sbg:modifiedBy: jelena_randjelovic
        sbg:modifiedOn: 1552058593
        sbg:project: michael_lloyd/variant-calling-decomposition
        sbg:projectName: variant_calling_decomposition
        sbg:publisher: sbg
        sbg:revision: 0
        sbg:revisionNotes:
        sbg:revisionsInfo:
        - sbg:modifiedBy: jelena_randjelovic
          sbg:modifiedOn: 1552058593
          sbg:revision: 0
          sbg:revisionNotes:
        sbg:sbgMaintained: false
        sbg:toolAuthor: Pablo Cingolani/Broad Institue
        sbg:toolkit: SnpEff
        sbg:toolkitVersion: 4.3k
        sbg:validationErrors: []
      inputs:
      - id: '#snpeff.variants_file'
        source: '#snpsift_filter_4_7.filtered_vcf'
      - id: '#snpeff.nostats'
        default: true
      - id: '#snpeff.no_hgvs'
        default: false
      - id: '#snpeff.format_eff'
        default: true
      - id: '#snpeff.database'
        source: '#snpEFF_Database'
      - id: '#snpeff.canon'
        default: true
      - id: '#snpeff.assembly'
        default: hg38
      - id: '#snpeff.add_lof_tag'
        default: true
      - id: '#snpeff.add_hgvs_anno'
        default: true
      outputs:
      - id: '#snpeff.summary_text'
      - id: '#snpeff.summary'
      - id: '#snpeff.annotated'
      sbg:x: 532.6160888671875
      sbg:y: -307.71112060546875
    - id: '#snpsift_dbnsfp'
      label: SnpSift dbNSFP
      in: []
      run:
        cwlVersion: sbg:draft-2
        class: CommandLineTool
        label: SnpSift dbNSFP
        $namespaces:
          sbg: https://sevenbridges.com

        requirements:
        - id: '#cwl-js-engine'
          class: ExpressionEngineRequirement
          requirements:
          - class: DockerRequirement
            dockerPull: rabix/js-engine
        - class: InlineJavascriptRequirement

        inputs:
        - id: '#mem_mb'
          label: RAM requirement in MB
          type:
          - 'null'
          - int
          description: RAM requirement for the java process execution.
          sbg:category: Configuration
          sbg:toolDefaultValue: '2048'
        - id: '#invert_fields_to_add'
          label: Invert 'fields to add' selection
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -n
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Invert 'fields to add' selection (i.e. use all fields except the ones specified in option '-f').
          sbg:category: Options
          sbg:stageInput:
        - id: '#input_vcf'
          label: VCF file that will be annotated
          type:
          - File
          inputBinding:
            position: 12
            separate: true
            sbg:cmdInclude: true
          description: VCF file that will be annotated with dbNSFP database.
          required: true
          sbg:category: Input files
          sbg:fileTypes: VCF
        - id: '#genome_version'
          label: Genome version
          type:
          - 'null'
          - string
          inputBinding:
            prefix: -g
            position: 6
            separate: true
            sbg:cmdInclude: true
          description: Genome version.
          sbg:category: Options
        - id: '#force_anno'
          label: Annotate fields, even if the database has an empty value
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -a
            position: 1
            separate: true
            sbg:cmdInclude: true
          description: |-
            Annotate fields, even if the database has an empty value (annotates using '.' for empty).
          sbg:category: Options
          sbg:toolDefaultValue: 'False'
        - id: '#dbnsfp_series'
          label: dbNSFP version
          type:
          - name: dbnsfp_series
            type: enum
            symbols:
            - dbNSFP 2.x
            - dbNSFP 3.x
          description: |-
            Version of dbNSFP used. 2.x should be used for GRCh37 data, 3.x for GRCh38 data. This parameter only affects the default database field sets.
        - id: '#database'
          label: Database for annotation
          type:
          - File
          inputBinding:
            secondaryFiles:
            - .tbi
            prefix: -db
            position: 10
            separate: true
            sbg:cmdInclude: true
          description: Database in VCF format for the annotation of the input VCF
            file.
          required: true
          sbg:category: Options
          sbg:fileTypes: GZ, TXT.GZ, VCF.GZ
          sbg:stageInput: link
        - id: '#colapse'
          label: Colapse repeated values
          type:
          - 'null'
          - name: colapse
            type: enum
            symbols:
            - Don't collapse
            - Collapse
          inputBinding:
            position: 2
            valueFrom:
              class: Expression
              engine: '#cwl-js-engine'
              script: |-
                {
                  if ($job.inputs.colapse === "Collapse")
                  {
                    return "-collapse"
                  }
                  else
                  {
                    return "-noCollapse"
                  }
                }
            separate: true
            sbg:cmdInclude: true
          description: Collapse repeated values from dbNSFP.
          sbg:category: Options
          sbg:toolDefaultValue: Don't colapse
        - id: '#annotate_missing_entries'
          label: Annotate missing entries (with '.')
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -m
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Annotate fields even if there are no database matching entries (annotates using '.').
          sbg:category: Options
          sbg:stageInput:
          sbg:toolDefaultValue: 'False'
        - id: '#anno_fields'
          label: Database fields to add
          type:
          - 'null'
          - type: array
            items: string
          inputBinding:
            prefix: -f
            position: 3
            valueFrom:
              class: Expression
              engine: '#cwl-js-engine'
              script: |-
                {
                  if ($job.inputs.anno_fields)
                  {
                    return $job.inputs.anno_fields
                  }
                  else if ($job.inputs.dbnsfp_series == "dbNSFP 2.x")
                  {
                   return "Interpro_domain,SIFT_pred,LRT_pred,MutationTaster_pred,GERP++_NR,GERP++_RS,phastCons100way_vertebrate,MutationAssessor_pred,FATHMM_pred,PROVEAN_pred,MetaSVM_pred,1000Gp1_AC,1000Gp1_AF,1000Gp1_AFR_AC,1000Gp1_AFR_AF,1000Gp1_EUR_AC,1000Gp1_EUR_AF,1000Gp1_AMR_AC,1000Gp1_AMR_AF,1000Gp1_ASN_AC,1000Gp1_ASN_AF,ESP6500_AA_AF,ESP6500_EA_AF,ExAC_AC,ExAC_AF,ExAC_Adj_AC,ExAC_Adj_AF,ExAC_AFR_AC,ExAC_AFR_AF,ExAC_AMR_AC,ExAC_AMR_AF,ExAC_EAS_AC,ExAC_EAS_AF,ExAC_FIN_AC,ExAC_FIN_AF,ExAC_NFE_AC,ExAC_NFE_AF,ExAC_SAS_AC,ExAC_SAS_AF"
                  }
                  else
                  {
                    return "Interpro_domain,SIFT_pred,LRT_pred,MutationTaster_pred,GERP++_NR,GERP++_RS,phastCons100way_vertebrate,MutationAssessor_pred,FATHMM_pred,PROVEAN_pred,MetaSVM_pred,1000Gp3_AC,1000Gp3_AF,1000Gp3_AFR_AC,1000Gp3_AFR_AF,1000Gp3_EUR_AC,1000Gp3_EUR_AF,1000Gp3_AMR_AC,1000Gp3_AMR_AF,1000Gp3_EAS_AC,1000Gp3_EAS_AF,1000Gp3_SAS_AC,1000Gp3_SAS_AF,ESP6500_AA_AC,ESP6500_AA_AF,ESP6500_EA_AC,ESP6500_EA_AF,ExAC_AC,ExAC_AF,ExAC_Adj_AC,ExAC_Adj_AF,ExAC_AFR_AC,ExAC_AFR_AF,ExAC_AMR_AC,ExAC_AMR_AF,ExAC_EAS_AC,ExAC_EAS_AF,ExAC_FIN_AC,ExAC_FIN_AF,ExAC_NFE_AC,ExAC_NFE_AF,ExAC_SAS_AC,ExAC_SAS_AF"
                  }
                }
            separate: true
            itemSeparator: ','
            sbg:cmdInclude: true
          description: A comma separated list of fields to add.
          sbg:category: Options
          sbg:toolDefaultValue: |-
            Interpro_domain,SIFT_pred,LRT_pred,MutationTaster_pred,GERP++_NR,GERP++_RS,phastCons100way_vertebrate,MutationAssessor_pred,FATHMM_pred,PROVEAN_pred,MetaSVM_pred,1000Gp3_AC,1000Gp3_AF,1000Gp3_AFR_AC,1000Gp3_AFR_AF,1000Gp3_EUR_AC,1000Gp3_EUR_AF,1000Gp3_AMR_AC,1000Gp3_AMR_AF,1000Gp3_EAS_AC,1000Gp3_EAS_AF,1000Gp3_SAS_AC,1000Gp3_SAS_AF,ESP6500_AA_AC,ESP6500_AA_AF,ESP6500_EA_AC,ESP6500_EA_AF,ExAC_AC,ExAC_AF,ExAC_Adj_AC,ExAC_Adj_AF,ExAC_AFR_AC,ExAC_AFR_AF,ExAC_AMR_AC,ExAC_AMR_AF,ExAC_EAS_AC,ExAC_EAS_AF,ExAC_FIN_AC,ExAC_FIN_AF,ExAC_NFE_AC,ExAC_NFE_AF,ExAC_SAS_AC,ExAC_SAS_AF

        outputs:
        - id: '#annotated_vcf'
          label: DBNSFP annotated VCF
          type:
          - 'null'
          - File
          outputBinding:
            glob: '*.dbnsfp_anno.vcf'
            sbg:inheritMetadataFrom: '#input_vcf'
          description: DBNSFP annotated VCF.
          sbg:fileTypes: VCF
        stdout:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              filename = $job.inputs.input_vcf.path
              return filename.split('.').slice(0, filename.split('.').length-1).concat("dbnsfp_anno.vcf").join('.').replace(/^.*[\\\/]/, '')
            }

        baseCommand:
        - java
        - class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              if ($job.inputs.mem_mb)
              {
                return "-Xmx" + ($job.inputs.mem_mb).toString() + "m"
              }
              else
              {
                return "-Xmx2048m"
              }
            }
        - -jar
        - /opt/snpEff/SnpSift.jar
        - dbnsfp
        arguments:
        - position: 5
          valueFrom:
            class: Expression
            engine: '#cwl-js-engine'
            script: "{\n  return \"-noDownload -noLog\"\n}"
          separate: true

        hints:
        - class: sbg:CPURequirement
          value: 1
        - class: sbg:MemRequirement
          value:
            class: Expression
            engine: '#cwl-js-engine'
            script: |-
              {
                if ($job.inputs.mem_mb)
                {
                  return $job.inputs.mem_mb
                }
                
                else
                {
                  return 2048
                }
              }
        - class: DockerRequirement
          dockerPull: images.sbgenomics.com/jrandjelovic/snpeff:v4.3k
          dockerImageId: 26f735e2c570
        id: michael_lloyd/variant-calling-decomposition/snpsift-dbnsfp/0
        description: |-
          **SnpSift dbNSFP** allows annotation with [dbNSFP](https://sites.google.com/site/jpopgen/dbNSFP) [1], an integrated database of functional predictions from multiple algorithms (SIFT, Polyphen2, LRT and MutationTaster, PhyloP and GERP++, etc.) [2].

          *A list of **all inputs and parameters** with corresponding descriptions can be found at the end of the page.*

          ### Common Use Cases

          Adding **dbNSFP** annotations to VCF files.

          ### Changes Introduced by Seven Bridges

          * To adapt the tool to newer versions of **dbNSFP** (currently 3.5), which have changed some of the fields (column headers) available for annotation, default list of annotation fields used by the tool has been somewhat updated to exclude fields no longer found in the database. Currently used default annotation fields list consists of the following fields:   
            
            `Interpro_domain,SIFT_pred, LRT_pred,MutationTaster_pred, GERP++_NR,GERP++_RS, phastCons100way_vertebrate, MutationAssessor_pred, FATHMM_pred, PROVEAN_pred, MetaSVM_pred, 1000Gp3_AC, 1000Gp3_AF, 1000Gp3_AFR_AC, 1000Gp3_AFR_AF, 1000Gp3_EUR_AC, 1000Gp3_EUR_AF, 1000Gp3_AMR_AC, 1000Gp3_AMR_AF, 1000Gp3_EAS_AC, 1000Gp3_EAS_AF, 1000Gp3_SAS_AC, 1000Gp3_SAS_AF, ESP6500_AA_AC, ESP6500_AA_AF, ESP6500_EA_AC, ESP6500_EA_AF, ExAC_AC, ExAC_AF, ExAC_Adj_AC, ExAC_Adj_AF, ExAC_AFR_AC, ExAC_AFR_AF, ExAC_AMR_AC, ExAC_AMR_AF, ExAC_EAS_AC, ExAC_EAS_AF, ExAC_FIN_AC, ExAC_FIN_AF, ExAC_NFE_AC, ExAC_NFE_AF, ExAC_SAS_AC, ExAC_SAS_AF`.

          When using dbNSFP 2.x files, the above list is modified to exclude `ESP6500_AA_AC,ESP6500_EA_AC` and replace 1000 genomes phase 3 fields with 1000 genomes phase 1 allele frequency fields found in dbNSFP 2.9.x files.

          ### Common Issues and Important Notes

          * Please note that **dbNSFP** database file used for annotation (input **Database for annotation**) should be preprocessed with bgzip and tabix before use. **dbNSFP** database files hosted on the Seven Bridges platform in the Public Reference Files section (dbNSFP_3.3c.gz and dbNSFP2.9.3.txt.gz) and older versions of the database available from SnpSift website links [2] have already been preprocessed. If you wish to preprocess your database files locally, this is one possible way (assumes a unix-based workstation) [3]:

              `unzip dbNSFPv3.x.zip`

              `head -n1 dbNSFP3.x_variant.chr1 ` > ` h`

              `cat dbNSFP3.x_variant.chr* | grep -v ^#chr | sort -k1,1 -k2,2n - | cat h - | bgzip -c ` > ` dbNSFP_3.x.gz`

              `tabix -s 1 -b 2 -e 2 dbNSFP_3.x.gz`

          * Please make sure that the **dbNSFP** version used matches the genome assembly version of your data (3.x for GRCh38 and 2.x for GRCh37).
          * Seven Bridges hosts **dbNSFP** files available for commercial use (dbNSFP_3.3c.gz and dbNSFP2.9.3.txt.gz), however please note that additional annotations are available to researchers in academic versions of **dbNSFP** database available on the **dbNSFP** website [1].

          ### Performance Benchmarking

          Annotating NA12878 genome (GRCh38, ~3 GB as VCF.GZ) with default **dbNSFP** (v3.5c) annotation fields and 8 GB RAM  took 42 minutes on the default instance (price: $0.28 using on-demand instance).

          *Cost can be significantly reduced by **spot instance** usage. Visit [knowledge center](https://docs.sevenbridges.com/docs/about-spot-instances) for more details.*            


          ### References

          [1] [dbNSFP website](https://sites.google.com/site/jpopgen/dbNSFP)

          [2] [SnpSift dbNSFP documentation](http://snpeff.sourceforge.net/SnpSift.html#dbNSFP)

          [3] [Ensembl VEP dbNSFP plugin documentation](https://github.com/Ensembl/VEP_plugins/blob/release/90/dbNSFP.pm)
        sbg:appVersion:
        - sbg:draft-2
        sbg:categories:
        - Annotation
        sbg:cmdPreview: |-
          java -Xmx5000m -jar /opt/snpEff/SnpSift.jar dbnsfp  -noDownload -noLog -db dbNSFP2.4_bgzip.vcf.gz  input.vcf > input.dbnsfp_anno.vcf
        sbg:content_hash: ae2cba441af820b7edeecd9e13ebb0637e8f0a8c57a9752968cbba8aa116a9f98
        sbg:contributors:
        - jelena_randjelovic
        sbg:createdBy: jelena_randjelovic
        sbg:createdOn: 1552058598
        sbg:id: michael_lloyd/variant-calling-decomposition/snpsift-dbnsfp/0
        sbg:image_url:
        sbg:job:
          allocatedResources:
            cpu: 1
            mem: 5000
          inputs:
            anno_fields:
            annotate_missing_entries: true
            colapse:
            database:
              secondaryFiles:
              - path: .tbi
              path: dbNSFP2.4_bgzip.vcf.gz
            dbnsfp_series: dbNSFP 2.x
            force_anno: false
            genome_version: genome_version-string-value
            input_vcf:
              class: File
              secondaryFiles: []
              path: input.vcf
              size: 0
            invert_fields_to_add: true
            mem_mb: 5000
        sbg:latestRevision: 0
        sbg:license: GNU Lesser General Public License v3.0 only
        sbg:links:
        - id: http://snpeff.sourceforge.net/index.html
          label: Homepage
        - id: https://github.com/pcingola/SnpEff
          label: Source Code
        - id: http://snpeff.sourceforge.net/SnpEff_manual.html
          label: Wiki
        - id: https://sourceforge.net/projects/snpeff/files/snpEff_v4_2_core.zip
          label: Download
        - id: http://snpeff.sourceforge.net/SnpEff_paper.pdf
          label: Publication
        sbg:modifiedBy: jelena_randjelovic
        sbg:modifiedOn: 1552058598
        sbg:project: michael_lloyd/variant-calling-decomposition
        sbg:projectName: variant_calling_decomposition
        sbg:publisher: sbg
        sbg:revision: 0
        sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-dbnsfp-4-2/16
        sbg:revisionsInfo:
        - sbg:modifiedBy: jelena_randjelovic
          sbg:modifiedOn: 1552058598
          sbg:revision: 0
          sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-dbnsfp-4-2/16
        sbg:sbgMaintained: false
        sbg:toolAuthor: Pablo Cingolani/Broad Institue
        sbg:toolkit: SnpEff
        sbg:toolkitVersion: 4.3k
        sbg:validationErrors: []
      inputs:
      - id: '#snpsift_dbnsfp.input_vcf'
        source: '#snpeff.annotated'
      - id: '#snpsift_dbnsfp.dbnsfp_series'
        default: dbNSFP 3.x
      - id: '#snpsift_dbnsfp.database'
        source: '#dbNSFP_Database'
      - id: '#snpsift_dbnsfp.anno_fields'
        default:
        - SIFT_score
        - SIFT_pred
        - Polyphen2_HDIV_score
        - MutationAssessor_score
        - phyloP100way_vertebrate
        - 1000Gp3_AF
        - 1000Gp3_AFR_AF
        - 1000Gp3_EUR_AF
        - 1000Gp3_AMR_AF
        - 1000Gp3_EAS_AF
        - ESP6500_AA_AF
        - ESP6500_EA_AF
        - ExAC_AC
        - ExAC_AF
      outputs:
      - id: '#snpsift_dbnsfp.annotated_vcf'
      sbg:x: 756.4259643554688
      sbg:y: -332.5968933105469
    - id: '#snpsift_annotate'
      label: SnpSift Annotate
      in: []
      run:
        cwlVersion: sbg:draft-2
        class: CommandLineTool
        label: SnpSift Annotate
        $namespaces:
          sbg: https://sevenbridges.com

        requirements:
        - id: '#cwl-js-engine'
          class: ExpressionEngineRequirement
          requirements:
          - class: DockerRequirement
            dockerPull: rabix/js-engine
        - class: InlineJavascriptRequirement

        inputs:
        - id: '#vcf_database_is_sorted'
          label: VCF database is sorted and uncompressed
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -sorted
            position: 16
            separate: true
            sbg:cmdInclude: true
          description: VCF database is sorted and uncompressed.
          sbg:category: Command options
          sbg:toolDefaultValue: 'False'
        - id: '#variant_exist'
          label: Annotate whether the variant exists in the database
          type:
          - 'null'
          - string
          inputBinding:
            prefix: -exists
            position: 2
            separate: true
            sbg:cmdInclude: true
          description: |-
            Annotate whether the variant exists or not in the database (using the supplied value as an INFO field FLAG).
          sbg:category: Command options
        - id: '#tabix'
          label: VCF database is tabix-indexed
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -tabix
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: VCF database is tabix-indexed.
          sbg:category: Command options
          sbg:toolDefaultValue: 'False'
        - id: '#no_ref_alt'
          label: Do not use REF and ALT fields
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -noAlt
            position: 10
            separate: true
            sbg:cmdInclude: true
          description: |-
            Do not use REF and ALT fields when comparing database.vcf entries to file.vcf entries.
          sbg:category: Command options
          sbg:toolDefaultValue: 'False'
        - id: '#no_id'
          label: Do not annotate ID field
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -noId
            position: 12
            separate: true
            sbg:cmdInclude: true
          description: Do not annotate ID field.
          sbg:category: Command options
          sbg:toolDefaultValue: 'False'
        - id: '#mem_mb'
          label: Java Xmx%m requirement
          type:
          - 'null'
          - int
          description: RAM requirement for the java process execution.
          sbg:category: Other input types
          sbg:toolDefaultValue: '8192'
        - id: '#max_block_size'
          label: Max block size
          type:
          - 'null'
          - int
          inputBinding:
            prefix: -maxBlockSize
            position: 8
            separate: true
            sbg:cmdInclude: true
          description: |-
            Use 'max block size' when creating index ('-sorted' command line option). Default: 16384.
          sbg:category: Command options
          sbg:toolDefaultValue: '16384'
        - id: '#input_vcf'
          label: VCF file that will be annotated
          type:
          - File
          inputBinding:
            position: 110
            separate: true
            sbg:cmdInclude: true
          description: |-
            VCF file that will be annotated with the INFO and FORMAT fields from the annotation VCF.
          required: true
          sbg:category: File type inputs
          sbg:fileTypes: VCF
        - id: '#info'
          label: Annotate using a list of INFO fields
          type:
          - 'null'
          - type: array
            items: string
          inputBinding:
            prefix: -info
            position: 5
            separate: true
            itemSeparator: ','
            sbg:cmdInclude: true
          description: Annotate using a list of INFO fields (list is a comma separated
            list of fields).
          sbg:category: Command options
          sbg:toolDefaultValue: All fields
        - id: '#id_field'
          label: Only annotate ID field
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -id
            position: 4
            separate: true
            sbg:cmdInclude: true
          description: Only annotate ID field (do not add INFO field).
          sbg:category: Command options
          sbg:toolDefaultValue: 'False'
        - id: '#force_annotate'
          label: Annotate fields even if value is absent from the annotation database
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -a
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Annotate fields, even if the database has an empty value (annotates using '.' for empty).
          sbg:category: Command options
          sbg:toolDefaultValue: 'False'
        - id: '#do_no_annotate_info_fields'
          label: Do not annotate INFO fields
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -noInfo
            position: 14
            separate: true
            sbg:cmdInclude: true
          description: Do not annotate INFO fields.
          sbg:category: Command options
          sbg:toolDefaultValue: 'False'
        - id: '#database_type'
          label: Database type
          type:
          - 'null'
          - name: database_type
            type: enum
            symbols:
            - vcf
            - dbsnp
            - clinvar
          inputBinding:
            position: 90
            valueFrom:
              class: Expression
              engine: '#cwl-js-engine'
              script: |-
                {
                  if ($job.inputs.database_type == 'vcf') {return ''}
                  else if ($job.inputs.database_type == 'dbsnp') {return '-dbsnp'}
                  else if ($job.inputs.database_type == 'clinvar') {return '-clinvar'}
                }
            separate: true
            sbg:cmdInclude: true
          description: Type of database used.
          sbg:category: Database options
          sbg:toolDefaultValue: vcf
        - id: '#database_for_annotation'
          label: Database for annotation
          type:
          - File
          inputBinding:
            secondaryFiles:
            - .tbi
            position: 92
            valueFrom:
              class: Expression
              engine: '#cwl-js-engine'
              script: |-
                {
                  type = $job.inputs.database_type
                  if (type == 'dbsnp' || type == 'clinvar'){
                    return '-db ' + $job.inputs.database_for_annotation.path
                  } else {
                    return $job.inputs.database_for_annotation.path
                  }
                }
            sbg:cmdInclude: true
          description: Database for annotating the input VCF file.
          required: true
          sbg:category: Database options
          sbg:fileTypes: VCF.GZ
          sbg:stageInput: link
        - id: '#append_name_to_info_fields'
          label: String prefix for all annotated INFO fields
          type:
          - 'null'
          - string
          inputBinding:
            prefix: -name
            position: 7
            separate: true
            sbg:cmdInclude: true
          description: String to prepend to all annotated INFO fields.
          sbg:category: Command options
          sbg:toolDefaultValue: "''"

        outputs:
        - id: '#annotated_vcf'
          label: Annotated VCF
          type:
          - 'null'
          - File
          outputBinding:
            glob: '*.annotated.vcf'
            sbg:inheritMetadataFrom: '#input_vcf'
          description: Annotated VCF.
          sbg:fileTypes: VCF
        stdout:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              filename = $job.inputs.input_vcf.path
              return filename.split('.').slice(0, filename.split('.').length-1).concat("annotated.vcf").join('.').replace(/^.*[\\\/]/, '')
            }

        baseCommand:
        - java
        - class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              if ($job.inputs.mem_mb)
              {
                return "-Xmx" + ($job.inputs.mem_mb).toString() + "m"
              }
              else
              {
                return "-Xmx8192m"
              }
            }
        - -jar
        - /opt/snpEff/SnpSift.jar
        - annotate
        - -noDownload
        - -noLog

        hints:
        - class: sbg:CPURequirement
          value: 1
        - class: sbg:MemRequirement
          value:
            class: Expression
            engine: '#cwl-js-engine'
            script: |-
              {
                if ($job.inputs.mem_mb)
                {
                  return $job.inputs.mem_mb
                }
                
                else
                {
                  return 8192
                }
              }
        - class: DockerRequirement
          dockerPull: images.sbgenomics.com/jrandjelovic/snpeff:v4.3k
          dockerImageId: 26f735e2c570
        id: michael_lloyd/variant-calling-decomposition/snpsift-annotate/0
        description: |-
          **SnpSift Annotate** uses fields from another VCF file (such as [dbSNP](https://www.ncbi.nlm.nih.gov/projects/SNP/snp_summary.cgi), [1000 genomes](http://www.internationalgenome.org/data), [ClinVar](https://www.ncbi.nlm.nih.gov/clinvar/), or [ExAC](http://exac.broadinstitute.org/downloads)) to annotate input VCF files [1].

          *A list of **all inputs and parameters** with corresponding descriptions can be found at the end of the page.*

          ### Common Use Cases

          **SnpSift Annotate** is typically used to annotate VCF ID and INFO fields from a "database" VCF file, such as dbSNP.

          ### Changes Introduced by Seven Bridges

          * VCF file used as the annotation database (**Database for annotation**) should be compressed and tabix-indexed (to prepare the file, please use **Tabix Bgzip** and **Tabix Index** tools which can be found under Public Apps gallery).

          ### Common Issues and Important Notes

          * Inputs **VCF file that will be annotated** and **Database for annotation** are required.
          * Note that **Database for annotation** has to be a compressed and tabix-indexed VCF file.
          * By default **SnpSift Annotate** adds **all** database INFO fields.
          You can use the **Annotate using a list of INFO fields** (`-info`) parameter if you want to select only a subset of fields from the database VCF file and **Only annotate ID field** (`-id`) parameter if you only want to annotate using ID field (no INFO fields will be added) [1].
          * Parameters **VCF database is sorted and uncompressed** (`-sorted`) and **Max block size** (`-maxBlockSize <int>`) should be used together and applied only if the annotation database VCF file is uncompressed.

          ### Performance Benchmarking

          Adding dbSNP (v150) annotations (all fields) to SnpEff-annotated NA12878 genome (GRCh38 ~3 GB as VCF) with 15 GB RAM took 2 h 23 minutes (price: $0.95 using on-demand default instance)

          *Cost can be significantly reduced by **spot instance** usage. Visit [knowledge center](https://docs.sevenbridges.com/docs/about-spot-instances) for more details.*            

          ### References

          [1] [SnpSift Annotate documentation](http://snpeff.sourceforge.net/SnpSift.html#annotate)
        sbg:appVersion:
        - sbg:draft-2
        sbg:categories:
        - Annotation
        sbg:cmdPreview: |-
          java -Xmx8192m -jar /opt/snpEff/SnpSift.jar annotate -noDownload -noLog /path/to/database_for_annotation.ext  input.vcf > input.annotated.vcf
        sbg:content_hash: a2e461d81f2040c7e15d6946268e57c698b100ceb0cb56037d0c43348789945cf
        sbg:contributors:
        - jelena_randjelovic
        sbg:createdBy: jelena_randjelovic
        sbg:createdOn: 1552058596
        sbg:id: michael_lloyd/variant-calling-decomposition/snpsift-annotate/0
        sbg:image_url:
        sbg:job:
          allocatedResources:
            cpu: 1
            mem: 8192
          inputs:
            database_for_annotation:
              class: File
              secondaryFiles:
              - path: .tbi
              path: /path/to/database_for_annotation.ext
              size: 0
            database_type: vcf
            do_no_annotate_info_fields: false
            force_annotate: false
            genome_version: genome_version
            id_field: false
            info:
            - test_field1
            - test_field2
            input_vcf:
              class: File
              secondaryFiles: []
              path: input.vcf
              size: 0
            max_block_size:
            mem_mb:
            no_ref_alt: false
            tabix: true
            variant_exist: ''
            vcf_database_is_sorted: false
        sbg:latestRevision: 0
        sbg:license: GNU Lesser General Public License v3.0 only
        sbg:links:
        - id: http://snpeff.sourceforge.net/index.html
          label: Homepage
        - id: https://github.com/pcingola/SnpEff
          label: Source Code
        - id: http://snpeff.sourceforge.net/SnpEff_manual.html
          label: Wiki
        - id: https://sourceforge.net/projects/snpeff/files/snpEff_v4_2_core.zip/download
          label: Download
        - id: http://snpeff.sourceforge.net/SnpEff_paper.pdf
          label: Publication
        sbg:modifiedBy: jelena_randjelovic
        sbg:modifiedOn: 1552058596
        sbg:project: michael_lloyd/variant-calling-decomposition
        sbg:projectName: variant_calling_decomposition
        sbg:publisher: sbg
        sbg:revision: 0
        sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-annotate-4-2/13
        sbg:revisionsInfo:
        - sbg:modifiedBy: jelena_randjelovic
          sbg:modifiedOn: 1552058596
          sbg:revision: 0
          sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-annotate-4-2/13
        sbg:sbgMaintained: false
        sbg:toolAuthor: Pablo Cingolani/Broad Institute
        sbg:toolkit: SnpEff
        sbg:toolkitVersion: 4.3k
        sbg:validationErrors: []
      inputs:
      - id: '#snpsift_annotate.input_vcf'
        source: '#snpsift_dbnsfp.annotated_vcf'
      - id: '#snpsift_annotate.id_field'
        default: true
      - id: '#snpsift_annotate.database_for_annotation'
        source: '#COSMIC_snpSift_Database'
      outputs:
      - id: '#snpsift_annotate.annotated_vcf'
      sbg:x: 949.8062133789062
      sbg:y: -380.2660217285156
    - id: '#snpsift_filter_4_8'
      label: SnpSift filter
      in: []
      run:
        cwlVersion: sbg:draft-2
        class: CommandLineTool
        label: SnpSift filter
        $namespaces:
          sbg: https://sevenbridges.com

        requirements:
        - id: '#cwl-js-engine'
          class: ExpressionEngineRequirement
          requirements:
          - class: DockerRequirement
            dockerPull: rabix/js-engine
        - class: InlineJavascriptRequirement

        inputs:
        - id: '#mem_mb'
          label: Memory to use for the task [Mb]
          type:
          - 'null'
          - int
          description: Memory to use for the task [Mb]
          sbg:category: Options
          sbg:stageInput:
          sbg:toolDefaultValue: '4096'
        - id: '#input_vcf'
          label: Input VCF
          type:
          - 'null'
          - File
          inputBinding:
            prefix: -f
            position: 100
            separate: true
            sbg:cmdInclude: true
          description: 'VCF input file. Default: STDIN'
          sbg:fileTypes: VCF, VCF.GZ
        - id: '#expression'
          label: Filter expression (without quotes)
          type:
          - 'null'
          - string
          inputBinding:
            position: 55
            valueFrom:
              class: Expression
              engine: '#cwl-js-engine'
              script: |-
                {
                  if ($job.inputs.expression)
                  {
                    return '"'.concat($job.inputs.expression,'"')
                  }
                  else
                  {
                    return ''
                  }
                }
            separate: true
            sbg:cmdInclude: true
          description: |-
            Expression to use for filtering. Please note that wrapping quotes are added automatically by the wrapper and should not be used as part of the filtering expression.
          sbg:category: Inputs
        - id: '#addFilter'
          label: Add a string to FILTER field
          type:
          - 'null'
          - string
          inputBinding:
            prefix: -a
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: "Add a string to FILTER VCF field if 'expression' is true.\
            \ Default: '' (none)."
          sbg:altPrefix: --addFilter
          sbg:category: Options
          sbg:toolDefaultValue: "''"
        - id: '#exprFile'
          label: Expression file
          type:
          - 'null'
          - File
          inputBinding:
            prefix: -e
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Read expression from a file.
          sbg:altPrefix: --exprFile
          sbg:category: Options
        - id: '#filterId'
          label: Filter ID
          type:
          - 'null'
          - string
          inputBinding:
            prefix: -i
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            ID for this filter (##FILTER tag in header and FILTER VCF field). Default: 'SnpSift'.
          sbg:altPrefix: --filterId
          sbg:category: Options
          sbg:toolDefaultValue: SnpSift
        - id: '#inverse'
          label: Inverse
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -n
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Inverse. Show lines that do not match filter expression.
          sbg:altPrefix: --inverse
          sbg:category: Options
          sbg:stageInput:
        - id: '#pass'
          label: Pass
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: -p
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Use 'PASS' field instead of filtering out VCF entries.
          sbg:altPrefix: --pass
          sbg:category: Options
        - id: '#rmFilter'
          label: Remove string from FILTER field
          type:
          - 'null'
          - string
          inputBinding:
            prefix: -r
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: |-
            Remove a string from FILTER VCF field if 'expression' is true (and 'str' is in the field). Default: '' (none).
          sbg:altPrefix: --rmFilter
          sbg:category: Options
          sbg:toolDefaultValue: "''"
        - id: '#set'
          label: Set File
          type:
          - 'null'
          - File
          inputBinding:
            prefix: -s
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Create a SET using 'file'.
          sbg:altPrefix: --set
          sbg:category: Options
        - id: '#errMissing'
          label: Error on missing fields
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --errMissing
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: 'Error if a field is missing. Default: False.'
          sbg:category: Options
          sbg:stageInput:
          sbg:toolDefaultValue: 'False'
        - id: '#format'
          label: Format
          type:
          - 'null'
          - name: format
            type: enum
            symbols:
            - '2'
            - '3'
          inputBinding:
            prefix: --format
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: 'SnpEff format version: {2, 3}. Default: Auto.'
          sbg:stageInput:
          sbg:toolDefaultValue: Auto
        - id: '#galaxy'
          label: Galaxy
          type:
          - 'null'
          - boolean
          inputBinding:
            prefix: --galaxy
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Used from Galaxy (expressions have been sanitized).
          sbg:category: Options
          sbg:stageInput:

        outputs:
        - id: '#filtered_vcf'
          label: Filtered VCF
          type:
          - 'null'
          - File
          outputBinding:
            glob: '*.filtered.vcf'
            sbg:inheritMetadataFrom: '#input_vcf'
          description: Filtered VCF.
          sbg:fileTypes: VCF
        stdout:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              filename = $job.inputs.input_vcf.path
              return filename.split('.').slice(0, filename.split('.').length-1).concat("filtered.vcf").join('.').replace(/^.*[\\\/]/, '')
            }

        baseCommand:
        - java
        - class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              if ($job.inputs.mem_mb)
              {
                return "-Xmx" + ($job.inputs.mem_mb).toString() + "m"
              }
              else
              {
                return "-Xmx4096m"
              }
            }
        - -jar
        - /opt/snpEff/SnpSift.jar
        - filter

        hints:
        - class: sbg:CPURequirement
          value: 1
        - class: sbg:MemRequirement
          value:
            class: Expression
            engine: '#cwl-js-engine'
            script: |-
              {
                if ($job.inputs.mem_mb)
                {
                  return $job.inputs.mem_mb
                }
                
                else
                {
                  return 4096
                }
              }
        - class: DockerRequirement
          dockerPull: images.sbgenomics.com/jrandjelovic/snpeff:v4.3r
        id: admin/sbg-public-data/snpsift-filter-4-3/2
        description: |-
          **SnpSift filter** filters SnpEff-annotated VCF files using arbitrary expressions, for instance "(QUAL > 30) | (exists INDEL) | ( countHet() > 2 )". The actual expressions can be quite complex, allowing for a lot of flexibility [1].

          *A list of **all inputs and parameters** with corresponding descriptions can be found at the end of the page.*

          ### Common Use Cases

          The **SnpSift filter** tool is used for filtering SnpEff-annotated VCF files to extract variants of interest.

          ### Changes Introduced by Seven Bridges

          * No modifications to the original tool representation have been made. 

          ### Common Issues and Important Notes

          * Parameter **Filter expression**: Wrapping quotes ("") are added automatically by the wrapper and should not be included in the filter expression. For example, filtering expression: "( CHROM = 'chr1' )" should be written as `( CHROM = 'chr1' )`.
          * **SnpSift filter** filtering syntax allows for creation of complex, flexible filters with different operators and functions. Please check the [official documentation](http://snpeff.sourceforge.net/SnpSift.html#filter) for examples of filter syntax and valid filtering fields before using the tool.

          ### Performance Benchmarking

          Typical runs take <10 minutes and cost <$0.1 on an on-demand default instance.

          ### References

          [1] [SnpSift filter documentation](http://snpeff.sourceforge.net/SnpSift.html#filter)
        sbg:appVersion:
        - sbg:draft-2
        sbg:categories:
        - Annotation
        - VCF-Processing
        sbg:cmdPreview: java -Xmx5m -jar /opt/snpEff/SnpSift.jar filter  > input_vcf.filtered.vcf
        sbg:content_hash:
        sbg:contributors:
        - admin
        sbg:createdBy: admin
        sbg:createdOn: 1512132000
        sbg:id: admin/sbg-public-data/snpsift-filter-4-3/2
        sbg:image_url:
        sbg:job:
          allocatedResources:
            cpu: 1
            mem: 5
          inputs:
            format: '2'
            addFilter: addFilter-string-value
            errMissing: true
            exprFile:
              class: File
              secondaryFiles: []
              path: /path/to/exprFile.ext
              size: 0
            expression: expression
            filterId: ''
            galaxy: true
            input_vcf:
              class: File
              secondaryFiles: []
              path: /path/to/input_vcf.ext
              size: 0
            inverse: true
            mem_mb: 5
            pass: true
            rmFilter: rmFilter-string-value
            set:
              class: File
              secondaryFiles: []
              path: /path/to/set.ext
              size: 0
        sbg:latestRevision: 2
        sbg:license: GNU Lesser General Public License v3.0 only
        sbg:links:
        - id: http://snpeff.sourceforge.net/SnpSift.html#filter
          label: Homepage
        - id: https://github.com/pcingola/SnpEff
          label: Source Code
        sbg:modifiedBy: admin
        sbg:modifiedOn: 1533292196
        sbg:project: admin/sbg-public-data
        sbg:projectName: SBG Public data
        sbg:publisher: sbg
        sbg:revision: 2
        sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-filter/9
        sbg:revisionsInfo:
        - sbg:modifiedBy: admin
          sbg:modifiedOn: 1512132000
          sbg:revision: 0
          sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-filter/5
        - sbg:modifiedBy: admin
          sbg:modifiedOn: 1512132000
          sbg:revision: 1
          sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-filter/8
        - sbg:modifiedBy: admin
          sbg:modifiedOn: 1533292196
          sbg:revision: 2
          sbg:revisionNotes: Copy of jrandjelovic/snpeff-4-3/snpsift-filter/9
        sbg:sbgMaintained: false
        sbg:toolAuthor: Pablo Cingolani/Broad Institue
        sbg:toolkit: SnpEff
        sbg:toolkitVersion: 4.3r
        sbg:validationErrors: []
      inputs:
      - id: '#snpsift_filter_4_8.input_vcf'
        source: '#snpsift_annotate.annotated_vcf'
      - id: '#snpsift_filter_4_8.expression'
        default: |-
          (ALT_AF[ANY] >= 90 & dbNSFP_1000Gp3_AF[ANY] >= 0.0095) | ((ALT_AF[ANY] >= 40 & ALT_AF[ANY] <= 60) & dbNSFP_1000Gp3_AF[ANY] >= 0.0095)| (ALT_AF[ANY] >= 90 & dbNSFP_ESP6500_AA_AF[ANY] >= 0.0095)| ((ALT_AF[ANY] >= 40 & ALT_AF[ANY] <= 60) & dbNSFP_ESP6500_AA_AF[ANY] >= 0.0095)|(ALT_AF[ANY] >= 90 & dbNSFP_ESP6500_EA_AF[ANY] >= 0.0095)|((ALT_AF[ANY] >= 40 & ALT_AF[ANY] <= 60) & dbNSFP_ESP6500_EA_AF[ANY] >= 0.0095)|(ALT_AF[ANY] >= 90 & dbNSFP_ExAC_AF[ANY] >= 0.0095)|((ALT_AF[ANY] >= 40 & ALT_AF[ANY] <= 60) & dbNSFP_ExAC_AF[ANY] >= 0.0095)
      - id: '#snpsift_filter_4_8.addFilter'
        default: PutativeGermline
      - id: '#snpsift_filter_4_8.rmFilter'
        default: PASS
      outputs:
      - id: '#snpsift_filter_4_8.filtered_vcf'
      sbg:x: 1125.281494140625
      sbg:y: -430.7641906738281
    - id: '#vcfeffoneperline_pl'
      label: vcfEffOnePerLine.pl
      in: []
      run:
        cwlVersion: sbg:draft-2
        class: CommandLineTool
        label: vcfEffOnePerLine.pl
        $namespaces:
          sbg: https://sevenbridges.com

        requirements:
        - id: '#cwl-js-engine'
          class: ExpressionEngineRequirement
          requirements:
          - class: DockerRequirement
            dockerPull: rabix/js-engine
        - class: InlineJavascriptRequirement

        inputs:
        - id: '#input_vcf_file'
          label: Input VCF file
          type:
          - File
          inputBinding:
            position: 0
            separate: true
            sbg:cmdInclude: true
          description: Input VCF file.
          required: true
          sbg:fileTypes: VCF

        outputs:
        - id: '#one_effect_vcf'
          label: VCF with one effect per line
          type:
          - 'null'
          - File
          outputBinding:
            glob: '*oneEff.vcf'
            sbg:inheritMetadataFrom: '#input_vcf_file'
          description: VCF with one effect per line.
          sbg:fileTypes: VCF

        baseCommand:
        - cat
        arguments:
        - position: 100
          valueFrom:
            class: Expression
            engine: '#cwl-js-engine'
            script: |-
              {
                return ' \| /opt/snpEff/scripts/vcfEffOnePerLine.pl > '.concat($job.inputs.input_vcf_file.path.split('/').pop().split('.vcf')[0], 'oneEff.vcf')
              }
          separate: true

        hints:
        - class: sbg:AWSInstanceType
          value: c5.2xlarge;ebs-gp2;100
        - class: sbg:CPURequirement
          value: 1
        - class: sbg:MemRequirement
          value: 1000
        - class: DockerRequirement
          dockerPull: images.sbgenomics.com/jrandjelovic/snpeff:v4.3k
        id: michael_lloyd/variant-calling-decomposition/vcfeffoneperline-pl/0
        description: |-
          A simple wrapper for the snpEff utility script vcfEffOnePerLine.pl, which transforms a SnpEff-annotated input VCF, yielding a file with one effect per line.
        sbg:appVersion:
        - sbg:draft-2
        sbg:categories:
        - VCF-Processing
        sbg:cmdPreview: |-
          cat  /path/to/input_vcf_file.ext   | /opt/snpEff/scripts/vcfEffOnePerLine.pl > input_vcf_file.extoneEff.vcf
        sbg:content_hash: a8b5fb3fadbe47b6b3c6607f16b6ab8a4225d72036ad68ccf474936565e36875f
        sbg:contributors:
        - jelena_randjelovic
        sbg:createdBy: jelena_randjelovic
        sbg:createdOn: 1552058607
        sbg:id: michael_lloyd/variant-calling-decomposition/vcfeffoneperline-pl/0
        sbg:image_url:
        sbg:job:
          allocatedResources:
            cpu: 1
            mem: 1000
          inputs:
            input_vcf_file:
              class: File
              secondaryFiles: []
              path: /path/to/input_vcf_file.ext
              size: 0
        sbg:latestRevision: 0
        sbg:license: GNU Lesser General Public License v3.0 only
        sbg:links:
        - id: http://snpeff.sourceforge.net/SnpSift.html#Extract
          label: Documentation
        - id: https://github.com/pcingola/SnpEff
          label: Source Code
        sbg:modifiedBy: jelena_randjelovic
        sbg:modifiedOn: 1552058607
        sbg:project: michael_lloyd/variant-calling-decomposition
        sbg:projectName: variant_calling_decomposition
        sbg:publisher: sbg
        sbg:revision: 0
        sbg:revisionNotes: 100 GB storage
        sbg:revisionsInfo:
        - sbg:modifiedBy: jelena_randjelovic
          sbg:modifiedOn: 1552058607
          sbg:revision: 0
          sbg:revisionNotes: 100 GB storage
        sbg:sbgMaintained: false
        sbg:toolAuthor: Pablo Cingolani/Broad Institute
        sbg:toolkit: SnpEff
        sbg:toolkitVersion: 4.3k
        sbg:validationErrors: []
      inputs:
      - id: '#vcfeffoneperline_pl.input_vcf_file'
        source: '#snpsift_filter_4_8.filtered_vcf'
      outputs:
      - id: '#vcfeffoneperline_pl.one_effect_vcf'
      sbg:x: 1296.6124267578125
      sbg:y: -477.9543151855469
    - id: '#snpsift_extractfields_4_3k'
      label: SnpSift extractFields 4.3k
      in: []
      run:
        cwlVersion: sbg:draft-2
        class: CommandLineTool
        label: SnpSift extractFields 4.3k
        $namespaces:
          sbg: https://sevenbridges.com

        requirements:
        - id: '#cwl-js-engine'
          class: ExpressionEngineRequirement
          requirements:
          - class: DockerRequirement
            dockerPull: rabix/js-engine
        - class: InlineJavascriptRequirement

        inputs:
        - id: '#output_file_name'
          label: Output file name
          type:
          - 'null'
          - string
          description: Output file name.
          sbg:category: Options
        - id: '#multiple_fields_separator_output'
          label: Multiple fields separator for the output file
          type:
          - 'null'
          - string
          inputBinding:
            prefix: -s
            position: 1
            valueFrom:
              class: Expression
              engine: '#cwl-js-engine'
              script: |-
                {
                  if ($job.inputs.multiple_fields_separator_output)
                  {
                    return '"'.concat($job.inputs.multiple_fields_separator_output, '"')
                  }
                }
            separate: true
            sbg:cmdInclude: true
          description: Multiple fields separator for the output file.
          sbg:category: Options
          sbg:toolDefaultValue: "'\\t'"
        - id: '#mem_mb'
          label: Memory to use for the task [MB]
          type:
          - 'null'
          - int
          description: Memory to use for the task [MB].
          sbg:category: Options
          sbg:stageInput:
          sbg:toolDefaultValue: '2048'
        - id: '#input_vcf'
          label: Input VCF
          type:
          - File
          inputBinding:
            position: 10
            separate: true
            sbg:cmdInclude: true
          description: Input VCF file.
          required: true
          sbg:category: Inputs
          sbg:fileTypes: VCF, VCF.GZ
        - id: '#fields_to_extract'
          label: VCF fields to extract
          type:
          - type: array
            items: string
          inputBinding:
            position: 20
            valueFrom:
              class: Expression
              engine: '#cwl-js-engine'
              script: |+
                {
                  tempout = ''
                  for (k = 0, len = $job.inputs.fields_to_extract.length; k < len; k++)
                  {
                    tempout = tempout.concat('"', $job.inputs.fields_to_extract[k], '" ')
                  }
                  return tempout
                }

            separate: true
            itemSeparator: ' '
            sbg:cmdInclude: true
          description: VCF fields to extract.
          sbg:category: Inputs
        - id: '#empty_fields_formatting'
          label: Empty fields formatting
          type:
          - 'null'
          - string
          inputBinding:
            prefix: -e
            position: 1
            valueFrom:
              class: Expression
              engine: '#cwl-js-engine'
              script: |-
                {
                  if ($job.inputs.empty_fields_formatting)
                  {
                    return '"'.concat($job.inputs.empty_fields_formatting, '"')
                  }
                }
            separate: true
            sbg:cmdInclude: true
          description: Empty fields formatting (-e).
          sbg:category: Options
          sbg:toolDefaultValue: "''"

        outputs:
        - id: '#extracted_fields_file'
          label: Extracted fields output file
          type:
          - File
          outputBinding:
            glob: '*_extracted_fields.tab'
            sbg:inheritMetadataFrom: '#input_vcf'
          description: Extracted fields output file.
          sbg:fileTypes: TAB

        baseCommand:
        - java
        - class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              if ($job.inputs.mem_mb)
              {
                return "-Xmx" + ($job.inputs.mem_mb).toString() + "m"
              }
              else
              {
                return "-Xmx2048m"
              }
            }
        - -jar
        - /opt/snpEff/SnpSift.jar
        - extractFields
        - -noDownload
        - -noLog
        arguments:
        - position: 100
          valueFrom:
            class: Expression
            engine: '#cwl-js-engine'
            script: |-
              {
                if ($job.inputs.output_file_name)
                {
                  return " > ".concat($job.inputs.output_file_name, "_snpsift_extracted_fields.tab")
                }
                else
                {
                  return " > ".concat($job.inputs.input_vcf.path.split("/").pop().split(".vcf")[0], "_snpsift_extracted_fields.tab")
                }
              }
          separate: true

        hints:
        - class: sbg:AWSInstanceType
          value: c5.2xlarge;ebs-gp2;100
        - class: sbg:CPURequirement
          value: 1
        - class: sbg:MemRequirement
          value:
            class: Expression
            engine: '#cwl-js-engine'
            script: |-
              {
                if ($job.inputs.mem_mb)
                {
                  return $job.inputs.mem_mb
                }
                
                else
                {
                  return 2048
                }
              }
        - class: DockerRequirement
          dockerPull: images.sbgenomics.com/jrandjelovic/snpeff:v4.3k
        id: michael_lloyd/variant-calling-decomposition/snpsift-extractfields-4-3k/0
        description: |-
          **SnpSift extractFields** tool extracts fields from a VCF file to a TXT, tab separated format [1].

          ### Common Use Cases

          Extracting a subset of VCF fields of interest for further analysis from a SnpEff/SnpSift annotated VCF file.

          ### Changes Introduced by Seven Bridges

          No significant changes were introduced.

          ### Common Issues and Important Notes

          * Inputs **Input VCF** and **VCF fields to extract** are required.

          * Please note that fields to be extracted should be provided in a specific format, for example ANN[*].HGVS_P (see [here](http://snpeff.sourceforge.net/SnpSift.html#Extract) for details).

          ### Performance Benchmarking

          Typical runs take <5 minutes and cost <$0.05.

          ### References

          [1] [SnpSift extractFields documentation](http://snpeff.sourceforge.net/SnpSift.html#Extract)
        sbg:appVersion:
        - sbg:draft-2
        sbg:categories:
        - VCF-Processing
        - Annotation
        sbg:cmdPreview: |-
          java -Xmx5m -jar /opt/snpEff/SnpSift.jar extractFields -noDownload -noLog  /path/to/input_vcf.ext  "fields_to_extract-string-value-1" "fields_to_extract-string-value-2"    > output_file_name_string_snpsift_extracted_fields.tab
        sbg:content_hash: a0b441251cf22c7782f64ef2c420b73c23c3b3aad9b379e9b17f3c4aa46a1b4a9
        sbg:contributors:
        - jelena_randjelovic
        sbg:createdBy: jelena_randjelovic
        sbg:createdOn: 1552058609
        sbg:id: michael_lloyd/variant-calling-decomposition/snpsift-extractfields-4-3k/0
        sbg:image_url:
        sbg:job:
          allocatedResources:
            cpu: 1
            mem: 5
          inputs:
            empty_fields_formatting: ''
            fields_to_extract:
            - fields_to_extract-string-value-1
            - fields_to_extract-string-value-2
            input_vcf:
              class: File
              secondaryFiles: []
              path: /path/to/input_vcf.ext
              size: 0
            mem_mb: 5
            multiple_fields_separator_output: ''
            output_file_name: output_file_name_string
        sbg:latestRevision: 0
        sbg:license: GNU Lesser General Public License v3.0 only
        sbg:links:
        - id: http://snpeff.sourceforge.net/SnpSift.html
          label: Homepage
        - id: https://github.com/pcingola/SnpEff
          label: Source Code
        - id: http://snpeff.sourceforge.net/SnpSift.html
          label: Documentation
        - id: https://sourceforge.net/projects/snpeff/files/
          label: Download
        sbg:modifiedBy: jelena_randjelovic
        sbg:modifiedOn: 1552058609
        sbg:project: michael_lloyd/variant-calling-decomposition
        sbg:projectName: variant_calling_decomposition
        sbg:publisher: sbg
        sbg:revision: 0
        sbg:revisionNotes: improved output naming
        sbg:revisionsInfo:
        - sbg:modifiedBy: jelena_randjelovic
          sbg:modifiedOn: 1552058609
          sbg:revision: 0
          sbg:revisionNotes: improved output naming
        sbg:sbgMaintained: false
        sbg:toolAuthor: Pablo Cingolani/Broad Institute
        sbg:toolkit: SnpEff
        sbg:toolkitVersion: 4.3k
        sbg:validationErrors: []
      inputs:
      - id: '#snpsift_extractfields_4_3k.input_vcf'
        source: '#vcfeffoneperline_pl.one_effect_vcf'
      - id: '#snpsift_extractfields_4_3k.fields_to_extract'
        default:
        - CHROM
        - POS
        - REF
        - ALT
        - ID
        - FILTER
        - DP
        - LOF[*].NUMTR
        - LOF[*].PERC
        - EFF[*].GENE
        - EFF[*].EFFECT
        - EFF[*].IMPACT
        - EFF[*].FUNCLASS
        - EFF[*].RANK
        - EFF[*].CODON
        - EFF[*].AA
        - EFF[*].AA_LEN
        - EFF[*].CODING
        - EFF[*].TRID
        - dbNSFP_SIFT_score
        - dbNSFP_SIFT_pred
        - dbNSFP_Polyphen2_HDIV_score
        - dbNSFP_MutationAssessor_score
        - dbNSFP_phyloP100way_vertebrate
        - dbNSFP_1000Gp3_AF
        - dbNSFP_1000Gp3_AFR_AF
        - dbNSFP_1000Gp3_EUR_AF
        - dbNSFP_1000Gp3_AMR_AF
        - dbNSFP_1000Gp3_EAS_AF
        - dbNSFP_ESP6500_AA_AF
        - dbNSFP_ESP6500_EA_AF
        - dbNSFP_ExAC_AC
        - dbNSFP_ExAC_AF
      outputs:
      - id: '#snpsift_extractfields_4_3k.extracted_fields_file'
      sbg:x: 1484.680908203125
      sbg:y: -557.6919555664062
    - id: '#tmb_vcf_manipulations'
      label: TMB VCF Manipulations
      in: []
      run:
        cwlVersion: sbg:draft-2
        class: CommandLineTool
        label: TMB VCF Manipulations
        $namespaces:
          sbg: https://sevenbridges.com

        requirements:
        - id: '#cwl-js-engine'
          class: ExpressionEngineRequirement
          requirements:
          - class: DockerRequirement
            dockerPull: rabix/js-engine
        - class: InlineJavascriptRequirement

        inputs:
        - id: '#input_vcf'
          type:
          - File
        - id: '#Window_File'
          type:
          - File
          inputBinding:
            prefix: -a
            position: 0
            separate: true

        outputs:
        - id: '#HM_tab_File'
          type:
          - 'null'
          - File
          outputBinding:
            glob: '*.HM.tab'
            sbg:inheritMetadataFrom: '#input_vcf'

        baseCommand:
        - class: Expression
          engine: '#cwl-js-engine'
          script: "'cat ' + $job.inputs.input_vcf.path"
        - '|'
        - grep
        - '"HIGH\|MODERATE"'
        - '|'
        - awk
        - -F
        - "'\\t'"
        - "'{"
        - if($5
        - '!~'
        - '"rs"'
        - '&&'
        - ($6==""||$6=="PASS"||$6=="."))
        - print
        - $1,$2-1,$2
        - "}'|sort|uniq|tr"
        - "'"
        - "'"
        - "'\\t'"
        - '>'
        - count2;
        - echo
        - '"chr\tstart\tend\tLength\tHM"'
        - '>'
        - class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
            	input_1 = [].concat($job.inputs.input_vcf)
              	filename = $job.inputs.input_vcf.path.split('/').slice(-1)[0];
              	pass_value = filename.split('pyfiltered.filtered.filtered.filtered.filtered.filtered.snpEff_annotated.dbnsfp_anno.annotated.filteredoneEff_snpsift_extracted_fields.tab')[0]
              	return pass_value +'HM.tab;'
            }
        - bedtools
        - coverage
        arguments:
        - prefix: -b
          position: 1
          valueFrom: count2
          separate: true
        - prefix: ''
          position: 3
          valueFrom:
            class: Expression
            engine: '#cwl-js-engine'
            script: |+
              {
              	input_1 = [].concat($job.inputs.input_vcf)
                	filename = $job.inputs.input_vcf.path.split('/').slice(-1)[0];
                	pass_value = filename.split('pyfiltered.filtered.filtered.filtered.filtered.filtered.snpEff_annotated.dbnsfp_anno.annotated.filteredoneEff_snpsift_extracted_fields.tab')[0]
                	return '| cut -f 1-5 >> ' + pass_value +'HM.tab'
              }

          separate: true

        hints:
        - class: DockerRequirement
          dockerPull: images.sbgenomics.com/thedzo/bedtools:2.25.0
        id: michael_lloyd/mantis/tmb-vcf-manipulations/10
        sbg:appVersion:
        - sbg:draft-2
        sbg:content_hash: af67c7ccb4857e36a03ba6f60278234aad0ec41734a638e0d8bf11c2e7a226c7a
        sbg:contributors:
        - michael_lloyd
        sbg:createdBy: michael_lloyd
        sbg:createdOn: 1565984228
        sbg:id: michael_lloyd/mantis/tmb-vcf-manipulations/10
        sbg:image_url:
        sbg:latestRevision: 10
        sbg:modifiedBy: michael_lloyd
        sbg:modifiedOn: 1566240783
        sbg:project: michael_lloyd/mantis
        sbg:projectName: MSI, HRD, TMB
        sbg:publisher: sbg
        sbg:revision: 10
        sbg:revisionNotes: reversed a and b in bedtools coverage
        sbg:revisionsInfo:
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1565984228
          sbg:revision: 0
          sbg:revisionNotes:
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1565985182
          sbg:revision: 1
          sbg:revisionNotes: ''
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1565985226
          sbg:revision: 2
          sbg:revisionNotes: ''
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1565985236
          sbg:revision: 3
          sbg:revisionNotes: ''
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1565985663
          sbg:revision: 4
          sbg:revisionNotes: ''
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1565985938
          sbg:revision: 5
          sbg:revisionNotes: ''
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1566064268
          sbg:revision: 6
          sbg:revisionNotes: updated call
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1566064690
          sbg:revision: 7
          sbg:revisionNotes: updated call
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1566064997
          sbg:revision: 8
          sbg:revisionNotes: ''
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1566065179
          sbg:revision: 9
          sbg:revisionNotes: ''
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1566240783
          sbg:revision: 10
          sbg:revisionNotes: reversed a and b in bedtools coverage
        sbg:sbgMaintained: false
        sbg:validationErrors: []
      inputs:
      - id: '#tmb_vcf_manipulations.input_vcf'
        source: '#snpsift_extractfields_4_3k.extracted_fields_file'
      - id: '#tmb_vcf_manipulations.Window_File'
        source: '#bed_sort.OverlapBed'
      outputs:
      - id: '#tmb_vcf_manipulations.HM_tab_File'
      sbg:x: 1717.0118408203125
      sbg:y: -637.882080078125
    - id: '#tmb_calc'
      label: TMB_Calc
      in: []
      run:
        cwlVersion: sbg:draft-2
        class: CommandLineTool
        label: TMB_Calc
        $namespaces:
          sbg: https://sevenbridges.com

        requirements:
        - id: '#cwl-js-engine'
          class: ExpressionEngineRequirement
          requirements:
          - class: DockerRequirement
            dockerPull: rabix/js-engine
        - class: InlineJavascriptRequirement
        - class: CreateFileRequirement
          fileDef:
          - fileContent: |-
              #!/usr/bin/env Rscript
              args = commandArgs(trailingOnly=TRUE)

              data <- read.table(args[1], header=T)

              d<-data[!((data$chr== "chr3" & data$start=="195000000") | (data$chr == "chr11" & data$start=="1000000")  |  (data$chr == "chr19" & data$start == "8000000") |        (data$chr == "chr7" & data$start == "101000000") | (data$chr == "chr6" & data$start== "29000000")),]

              print(sum(d$HM))
              print(sum(d$Length))

              TMB<- (sum(d$HM)/ sum(d$Length))*1000000
               
              write.table(TMB, file=args[2], quote=F, row.names=F, col.names=F)
            filename: TMB_Final.R

        inputs:
        - id: '#HM_tab_File'
          type:
          - File
          inputBinding:
            position: 0

        outputs:
        - id: '#TMB_Score'
          type:
          - 'null'
          - File
          outputBinding:
            glob: '*TMB_Score.txt'
            sbg:inheritMetadataFrom: '#HM_tab_File'

        baseCommand:
        - Rscript
        - TMB_Final.R
        arguments:
        - prefix: ''
          position: 1
          valueFrom:
            class: Expression
            engine: '#cwl-js-engine'
            script: |
              {
              	input_1 = [].concat($job.inputs.HM_tab_File)
                	filename = $job.inputs.HM_tab_File.path.split('/').slice(-1)[0];
                	pass_value = filename.split('.HM.tab')[0]
                	return pass_value + '_TMB_Score.txt'
              }
          separate: true

        hints:
        - class: DockerRequirement
          dockerPull: pgc-images.sbgenomics.com/d3b-bixu/sequenza_hg38:v6
        id: michael_lloyd/mantis/tmb-calc/4
        sbg:appVersion:
        - sbg:draft-2
        sbg:content_hash: a1e19659fcf295c13ef78e7a874a10f3d92134eb117c15242a26442697a0a7439
        sbg:contributors:
        - michael_lloyd
        sbg:createdBy: michael_lloyd
        sbg:createdOn: 1565985315
        sbg:id: michael_lloyd/mantis/tmb-calc/4
        sbg:image_url:
        sbg:latestRevision: 4
        sbg:modifiedBy: michael_lloyd
        sbg:modifiedOn: 1566065246
        sbg:project: michael_lloyd/mantis
        sbg:projectName: MSI, HRD, TMB
        sbg:publisher: sbg
        sbg:revision: 4
        sbg:revisionNotes: updated R code
        sbg:revisionsInfo:
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1565985315
          sbg:revision: 0
          sbg:revisionNotes:
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1565985621
          sbg:revision: 1
          sbg:revisionNotes: ''
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1565985635
          sbg:revision: 2
          sbg:revisionNotes: ''
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1565985862
          sbg:revision: 3
          sbg:revisionNotes: ''
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1566065246
          sbg:revision: 4
          sbg:revisionNotes: updated R code
        sbg:sbgMaintained: false
        sbg:validationErrors: []
      inputs:
      - id: '#tmb_calc.HM_tab_File'
        source: '#tmb_vcf_manipulations.HM_tab_File'
      outputs:
      - id: '#tmb_calc.TMB_Score'
      sbg:x: 1970.9852294921875
      sbg:y: -634.0265502929688
    - id: '#bed_sort'
      label: bedmap
      in: []
      run:
        cwlVersion: sbg:draft-2
        class: CommandLineTool
        label: bedmap
        $namespaces:
          sbg: https://sevenbridges.com

        requirements:
        - id: '#cwl-js-engine'
          class: ExpressionEngineRequirement
          requirements:
          - class: DockerRequirement
            dockerPull: rabix/js-engine
        - class: InlineJavascriptRequirement
        - class: CreateFileRequirement
          fileDef:
          - fileContent: |-
              chr1	0	1000000
              chr1	1000000	2000000
              chr1	2000000	3000000
              chr1	3000000	4000000
              chr1	4000000	5000000
              chr1	5000000	6000000
              chr1	6000000	7000000
              chr1	7000000	8000000
              chr1	8000000	9000000
              chr1	9000000	10000000
              chr1	10000000	11000000
              chr1	11000000	12000000
              chr1	12000000	13000000
              chr1	13000000	14000000
              chr1	14000000	15000000
              chr1	15000000	16000000
              chr1	16000000	17000000
              chr1	17000000	18000000
              chr1	18000000	19000000
              chr1	19000000	20000000
              chr1	20000000	21000000
              chr1	21000000	22000000
              chr1	22000000	23000000
              chr1	23000000	24000000
              chr1	24000000	25000000
              chr1	25000000	26000000
              chr1	26000000	27000000
              chr1	27000000	28000000
              chr1	28000000	29000000
              chr1	29000000	30000000
              chr1	30000000	31000000
              chr1	31000000	32000000
              chr1	32000000	33000000
              chr1	33000000	34000000
              chr1	34000000	35000000
              chr1	35000000	36000000
              chr1	36000000	37000000
              chr1	37000000	38000000
              chr1	38000000	39000000
              chr1	39000000	40000000
              chr1	40000000	41000000
              chr1	41000000	42000000
              chr1	42000000	43000000
              chr1	43000000	44000000
              chr1	44000000	45000000
              chr1	45000000	46000000
              chr1	46000000	47000000
              chr1	47000000	48000000
              chr1	48000000	49000000
              chr1	49000000	50000000
              chr1	50000000	51000000
              chr1	51000000	52000000
              chr1	52000000	53000000
              chr1	53000000	54000000
              chr1	54000000	55000000
              chr1	55000000	56000000
              chr1	56000000	57000000
              chr1	57000000	58000000
              chr1	58000000	59000000
              chr1	59000000	60000000
              chr1	60000000	61000000
              chr1	61000000	62000000
              chr1	62000000	63000000
              chr1	63000000	64000000
              chr1	64000000	65000000
              chr1	65000000	66000000
              chr1	66000000	67000000
              chr1	67000000	68000000
              chr1	68000000	69000000
              chr1	69000000	70000000
              chr1	70000000	71000000
              chr1	71000000	72000000
              chr1	72000000	73000000
              chr1	73000000	74000000
              chr1	74000000	75000000
              chr1	75000000	76000000
              chr1	76000000	77000000
              chr1	77000000	78000000
              chr1	78000000	79000000
              chr1	79000000	80000000
              chr1	80000000	81000000
              chr1	81000000	82000000
              chr1	82000000	83000000
              chr1	83000000	84000000
              chr1	84000000	85000000
              chr1	85000000	86000000
              chr1	86000000	87000000
              chr1	87000000	88000000
              chr1	88000000	89000000
              chr1	89000000	90000000
              chr1	90000000	91000000
              chr1	91000000	92000000
              chr1	92000000	93000000
              chr1	93000000	94000000
              chr1	94000000	95000000
              chr1	95000000	96000000
              chr1	96000000	97000000
              chr1	97000000	98000000
              chr1	98000000	99000000
              chr1	99000000	100000000
              chr1	100000000	101000000
              chr1	101000000	102000000
              chr1	102000000	103000000
              chr1	103000000	104000000
              chr1	104000000	105000000
              chr1	105000000	106000000
              chr1	106000000	107000000
              chr1	107000000	108000000
              chr1	108000000	109000000
              chr1	109000000	110000000
              chr1	110000000	111000000
              chr1	111000000	112000000
              chr1	112000000	113000000
              chr1	113000000	114000000
              chr1	114000000	115000000
              chr1	115000000	116000000
              chr1	116000000	117000000
              chr1	117000000	118000000
              chr1	118000000	119000000
              chr1	119000000	120000000
              chr1	120000000	121000000
              chr1	121000000	122000000
              chr1	122000000	123000000
              chr1	123000000	124000000
              chr1	124000000	125000000
              chr1	125000000	126000000
              chr1	126000000	127000000
              chr1	127000000	128000000
              chr1	128000000	129000000
              chr1	129000000	130000000
              chr1	130000000	131000000
              chr1	131000000	132000000
              chr1	132000000	133000000
              chr1	133000000	134000000
              chr1	134000000	135000000
              chr1	135000000	136000000
              chr1	136000000	137000000
              chr1	137000000	138000000
              chr1	138000000	139000000
              chr1	139000000	140000000
              chr1	140000000	141000000
              chr1	141000000	142000000
              chr1	142000000	143000000
              chr1	143000000	144000000
              chr1	144000000	145000000
              chr1	145000000	146000000
              chr1	146000000	147000000
              chr1	147000000	148000000
              chr1	148000000	149000000
              chr1	149000000	150000000
              chr1	150000000	151000000
              chr1	151000000	152000000
              chr1	152000000	153000000
              chr1	153000000	154000000
              chr1	154000000	155000000
              chr1	155000000	156000000
              chr1	156000000	157000000
              chr1	157000000	158000000
              chr1	158000000	159000000
              chr1	159000000	160000000
              chr1	160000000	161000000
              chr1	161000000	162000000
              chr1	162000000	163000000
              chr1	163000000	164000000
              chr1	164000000	165000000
              chr1	165000000	166000000
              chr1	166000000	167000000
              chr1	167000000	168000000
              chr1	168000000	169000000
              chr1	169000000	170000000
              chr1	170000000	171000000
              chr1	171000000	172000000
              chr1	172000000	173000000
              chr1	173000000	174000000
              chr1	174000000	175000000
              chr1	175000000	176000000
              chr1	176000000	177000000
              chr1	177000000	178000000
              chr1	178000000	179000000
              chr1	179000000	180000000
              chr1	180000000	181000000
              chr1	181000000	182000000
              chr1	182000000	183000000
              chr1	183000000	184000000
              chr1	184000000	185000000
              chr1	185000000	186000000
              chr1	186000000	187000000
              chr1	187000000	188000000
              chr1	188000000	189000000
              chr1	189000000	190000000
              chr1	190000000	191000000
              chr1	191000000	192000000
              chr1	192000000	193000000
              chr1	193000000	194000000
              chr1	194000000	195000000
              chr1	195000000	196000000
              chr1	196000000	197000000
              chr1	197000000	198000000
              chr1	198000000	199000000
              chr1	199000000	200000000
              chr1	200000000	201000000
              chr1	201000000	202000000
              chr1	202000000	203000000
              chr1	203000000	204000000
              chr1	204000000	205000000
              chr1	205000000	206000000
              chr1	206000000	207000000
              chr1	207000000	208000000
              chr1	208000000	209000000
              chr1	209000000	210000000
              chr1	210000000	211000000
              chr1	211000000	212000000
              chr1	212000000	213000000
              chr1	213000000	214000000
              chr1	214000000	215000000
              chr1	215000000	216000000
              chr1	216000000	217000000
              chr1	217000000	218000000
              chr1	218000000	219000000
              chr1	219000000	220000000
              chr1	220000000	221000000
              chr1	221000000	222000000
              chr1	222000000	223000000
              chr1	223000000	224000000
              chr1	224000000	225000000
              chr1	225000000	226000000
              chr1	226000000	227000000
              chr1	227000000	228000000
              chr1	228000000	229000000
              chr1	229000000	230000000
              chr1	230000000	231000000
              chr1	231000000	232000000
              chr1	232000000	233000000
              chr1	233000000	234000000
              chr1	234000000	235000000
              chr1	235000000	236000000
              chr1	236000000	237000000
              chr1	237000000	238000000
              chr1	238000000	239000000
              chr1	239000000	240000000
              chr1	240000000	241000000
              chr1	241000000	242000000
              chr1	242000000	243000000
              chr1	243000000	244000000
              chr1	244000000	245000000
              chr1	245000000	246000000
              chr1	246000000	247000000
              chr1	247000000	248000000
              chr1	248000000	248956422
              chr10	0	1000000
              chr10	1000000	2000000
              chr10	2000000	3000000
              chr10	3000000	4000000
              chr10	4000000	5000000
              chr10	5000000	6000000
              chr10	6000000	7000000
              chr10	7000000	8000000
              chr10	8000000	9000000
              chr10	9000000	10000000
              chr10	10000000	11000000
              chr10	11000000	12000000
              chr10	12000000	13000000
              chr10	13000000	14000000
              chr10	14000000	15000000
              chr10	15000000	16000000
              chr10	16000000	17000000
              chr10	17000000	18000000
              chr10	18000000	19000000
              chr10	19000000	20000000
              chr10	20000000	21000000
              chr10	21000000	22000000
              chr10	22000000	23000000
              chr10	23000000	24000000
              chr10	24000000	25000000
              chr10	25000000	26000000
              chr10	26000000	27000000
              chr10	27000000	28000000
              chr10	28000000	29000000
              chr10	29000000	30000000
              chr10	30000000	31000000
              chr10	31000000	32000000
              chr10	32000000	33000000
              chr10	33000000	34000000
              chr10	34000000	35000000
              chr10	35000000	36000000
              chr10	36000000	37000000
              chr10	37000000	38000000
              chr10	38000000	39000000
              chr10	39000000	40000000
              chr10	40000000	41000000
              chr10	41000000	42000000
              chr10	42000000	43000000
              chr10	43000000	44000000
              chr10	44000000	45000000
              chr10	45000000	46000000
              chr10	46000000	47000000
              chr10	47000000	48000000
              chr10	48000000	49000000
              chr10	49000000	50000000
              chr10	50000000	51000000
              chr10	51000000	52000000
              chr10	52000000	53000000
              chr10	53000000	54000000
              chr10	54000000	55000000
              chr10	55000000	56000000
              chr10	56000000	57000000
              chr10	57000000	58000000
              chr10	58000000	59000000
              chr10	59000000	60000000
              chr10	60000000	61000000
              chr10	61000000	62000000
              chr10	62000000	63000000
              chr10	63000000	64000000
              chr10	64000000	65000000
              chr10	65000000	66000000
              chr10	66000000	67000000
              chr10	67000000	68000000
              chr10	68000000	69000000
              chr10	69000000	70000000
              chr10	70000000	71000000
              chr10	71000000	72000000
              chr10	72000000	73000000
              chr10	73000000	74000000
              chr10	74000000	75000000
              chr10	75000000	76000000
              chr10	76000000	77000000
              chr10	77000000	78000000
              chr10	78000000	79000000
              chr10	79000000	80000000
              chr10	80000000	81000000
              chr10	81000000	82000000
              chr10	82000000	83000000
              chr10	83000000	84000000
              chr10	84000000	85000000
              chr10	85000000	86000000
              chr10	86000000	87000000
              chr10	87000000	88000000
              chr10	88000000	89000000
              chr10	89000000	90000000
              chr10	90000000	91000000
              chr10	91000000	92000000
              chr10	92000000	93000000
              chr10	93000000	94000000
              chr10	94000000	95000000
              chr10	95000000	96000000
              chr10	96000000	97000000
              chr10	97000000	98000000
              chr10	98000000	99000000
              chr10	99000000	100000000
              chr10	100000000	101000000
              chr10	101000000	102000000
              chr10	102000000	103000000
              chr10	103000000	104000000
              chr10	104000000	105000000
              chr10	105000000	106000000
              chr10	106000000	107000000
              chr10	107000000	108000000
              chr10	108000000	109000000
              chr10	109000000	110000000
              chr10	110000000	111000000
              chr10	111000000	112000000
              chr10	112000000	113000000
              chr10	113000000	114000000
              chr10	114000000	115000000
              chr10	115000000	116000000
              chr10	116000000	117000000
              chr10	117000000	118000000
              chr10	118000000	119000000
              chr10	119000000	120000000
              chr10	120000000	121000000
              chr10	121000000	122000000
              chr10	122000000	123000000
              chr10	123000000	124000000
              chr10	124000000	125000000
              chr10	125000000	126000000
              chr10	126000000	127000000
              chr10	127000000	128000000
              chr10	128000000	129000000
              chr10	129000000	130000000
              chr10	130000000	131000000
              chr10	131000000	132000000
              chr10	132000000	133000000
              chr10	133000000	133797422
              chr10_GL383545v1_alt	0	179254
              chr10_GL383546v1_alt	0	309802
              chr10_KI270824v1_alt	0	181496
              chr10_KI270825v1_alt	0	188315
              chr11	0	1000000
              chr11	1000000	2000000
              chr11	2000000	3000000
              chr11	3000000	4000000
              chr11	4000000	5000000
              chr11	5000000	6000000
              chr11	6000000	7000000
              chr11	7000000	8000000
              chr11	8000000	9000000
              chr11	9000000	10000000
              chr11	10000000	11000000
              chr11	11000000	12000000
              chr11	12000000	13000000
              chr11	13000000	14000000
              chr11	14000000	15000000
              chr11	15000000	16000000
              chr11	16000000	17000000
              chr11	17000000	18000000
              chr11	18000000	19000000
              chr11	19000000	20000000
              chr11	20000000	21000000
              chr11	21000000	22000000
              chr11	22000000	23000000
              chr11	23000000	24000000
              chr11	24000000	25000000
              chr11	25000000	26000000
              chr11	26000000	27000000
              chr11	27000000	28000000
              chr11	28000000	29000000
              chr11	29000000	30000000
              chr11	30000000	31000000
              chr11	31000000	32000000
              chr11	32000000	33000000
              chr11	33000000	34000000
              chr11	34000000	35000000
              chr11	35000000	36000000
              chr11	36000000	37000000
              chr11	37000000	38000000
              chr11	38000000	39000000
              chr11	39000000	40000000
              chr11	40000000	41000000
              chr11	41000000	42000000
              chr11	42000000	43000000
              chr11	43000000	44000000
              chr11	44000000	45000000
              chr11	45000000	46000000
              chr11	46000000	47000000
              chr11	47000000	48000000
              chr11	48000000	49000000
              chr11	49000000	50000000
              chr11	50000000	51000000
              chr11	51000000	52000000
              chr11	52000000	53000000
              chr11	53000000	54000000
              chr11	54000000	55000000
              chr11	55000000	56000000
              chr11	56000000	57000000
              chr11	57000000	58000000
              chr11	58000000	59000000
              chr11	59000000	60000000
              chr11	60000000	61000000
              chr11	61000000	62000000
              chr11	62000000	63000000
              chr11	63000000	64000000
              chr11	64000000	65000000
              chr11	65000000	66000000
              chr11	66000000	67000000
              chr11	67000000	68000000
              chr11	68000000	69000000
              chr11	69000000	70000000
              chr11	70000000	71000000
              chr11	71000000	72000000
              chr11	72000000	73000000
              chr11	73000000	74000000
              chr11	74000000	75000000
              chr11	75000000	76000000
              chr11	76000000	77000000
              chr11	77000000	78000000
              chr11	78000000	79000000
              chr11	79000000	80000000
              chr11	80000000	81000000
              chr11	81000000	82000000
              chr11	82000000	83000000
              chr11	83000000	84000000
              chr11	84000000	85000000
              chr11	85000000	86000000
              chr11	86000000	87000000
              chr11	87000000	88000000
              chr11	88000000	89000000
              chr11	89000000	90000000
              chr11	90000000	91000000
              chr11	91000000	92000000
              chr11	92000000	93000000
              chr11	93000000	94000000
              chr11	94000000	95000000
              chr11	95000000	96000000
              chr11	96000000	97000000
              chr11	97000000	98000000
              chr11	98000000	99000000
              chr11	99000000	100000000
              chr11	100000000	101000000
              chr11	101000000	102000000
              chr11	102000000	103000000
              chr11	103000000	104000000
              chr11	104000000	105000000
              chr11	105000000	106000000
              chr11	106000000	107000000
              chr11	107000000	108000000
              chr11	108000000	109000000
              chr11	109000000	110000000
              chr11	110000000	111000000
              chr11	111000000	112000000
              chr11	112000000	113000000
              chr11	113000000	114000000
              chr11	114000000	115000000
              chr11	115000000	116000000
              chr11	116000000	117000000
              chr11	117000000	118000000
              chr11	118000000	119000000
              chr11	119000000	120000000
              chr11	120000000	121000000
              chr11	121000000	122000000
              chr11	122000000	123000000
              chr11	123000000	124000000
              chr11	124000000	125000000
              chr11	125000000	126000000
              chr11	126000000	127000000
              chr11	127000000	128000000
              chr11	128000000	129000000
              chr11	129000000	130000000
              chr11	130000000	131000000
              chr11	131000000	132000000
              chr11	132000000	133000000
              chr11	133000000	134000000
              chr11	134000000	135000000
              chr11	135000000	135086622
              chr11_GL383547v1_alt	0	154407
              chr11_JH159136v1_alt	0	200998
              chr11_JH159137v1_alt	0	191409
              chr11_KI270721v1_random	0	100316
              chr11_KI270826v1_alt	0	186169
              chr11_KI270827v1_alt	0	67707
              chr11_KI270829v1_alt	0	204059
              chr11_KI270830v1_alt	0	177092
              chr11_KI270831v1_alt	0	296895
              chr11_KI270832v1_alt	0	210133
              chr11_KI270902v1_alt	0	106711
              chr11_KI270903v1_alt	0	214625
              chr11_KI270927v1_alt	0	218612
              chr12	0	1000000
              chr12	1000000	2000000
              chr12	2000000	3000000
              chr12	3000000	4000000
              chr12	4000000	5000000
              chr12	5000000	6000000
              chr12	6000000	7000000
              chr12	7000000	8000000
              chr12	8000000	9000000
              chr12	9000000	10000000
              chr12	10000000	11000000
              chr12	11000000	12000000
              chr12	12000000	13000000
              chr12	13000000	14000000
              chr12	14000000	15000000
              chr12	15000000	16000000
              chr12	16000000	17000000
              chr12	17000000	18000000
              chr12	18000000	19000000
              chr12	19000000	20000000
              chr12	20000000	21000000
              chr12	21000000	22000000
              chr12	22000000	23000000
              chr12	23000000	24000000
              chr12	24000000	25000000
              chr12	25000000	26000000
              chr12	26000000	27000000
              chr12	27000000	28000000
              chr12	28000000	29000000
              chr12	29000000	30000000
              chr12	30000000	31000000
              chr12	31000000	32000000
              chr12	32000000	33000000
              chr12	33000000	34000000
              chr12	34000000	35000000
              chr12	35000000	36000000
              chr12	36000000	37000000
              chr12	37000000	38000000
              chr12	38000000	39000000
              chr12	39000000	40000000
              chr12	40000000	41000000
              chr12	41000000	42000000
              chr12	42000000	43000000
              chr12	43000000	44000000
              chr12	44000000	45000000
              chr12	45000000	46000000
              chr12	46000000	47000000
              chr12	47000000	48000000
              chr12	48000000	49000000
              chr12	49000000	50000000
              chr12	50000000	51000000
              chr12	51000000	52000000
              chr12	52000000	53000000
              chr12	53000000	54000000
              chr12	54000000	55000000
              chr12	55000000	56000000
              chr12	56000000	57000000
              chr12	57000000	58000000
              chr12	58000000	59000000
              chr12	59000000	60000000
              chr12	60000000	61000000
              chr12	61000000	62000000
              chr12	62000000	63000000
              chr12	63000000	64000000
              chr12	64000000	65000000
              chr12	65000000	66000000
              chr12	66000000	67000000
              chr12	67000000	68000000
              chr12	68000000	69000000
              chr12	69000000	70000000
              chr12	70000000	71000000
              chr12	71000000	72000000
              chr12	72000000	73000000
              chr12	73000000	74000000
              chr12	74000000	75000000
              chr12	75000000	76000000
              chr12	76000000	77000000
              chr12	77000000	78000000
              chr12	78000000	79000000
              chr12	79000000	80000000
              chr12	80000000	81000000
              chr12	81000000	82000000
              chr12	82000000	83000000
              chr12	83000000	84000000
              chr12	84000000	85000000
              chr12	85000000	86000000
              chr12	86000000	87000000
              chr12	87000000	88000000
              chr12	88000000	89000000
              chr12	89000000	90000000
              chr12	90000000	91000000
              chr12	91000000	92000000
              chr12	92000000	93000000
              chr12	93000000	94000000
              chr12	94000000	95000000
              chr12	95000000	96000000
              chr12	96000000	97000000
              chr12	97000000	98000000
              chr12	98000000	99000000
              chr12	99000000	100000000
              chr12	100000000	101000000
              chr12	101000000	102000000
              chr12	102000000	103000000
              chr12	103000000	104000000
              chr12	104000000	105000000
              chr12	105000000	106000000
              chr12	106000000	107000000
              chr12	107000000	108000000
              chr12	108000000	109000000
              chr12	109000000	110000000
              chr12	110000000	111000000
              chr12	111000000	112000000
              chr12	112000000	113000000
              chr12	113000000	114000000
              chr12	114000000	115000000
              chr12	115000000	116000000
              chr12	116000000	117000000
              chr12	117000000	118000000
              chr12	118000000	119000000
              chr12	119000000	120000000
              chr12	120000000	121000000
              chr12	121000000	122000000
              chr12	122000000	123000000
              chr12	123000000	124000000
              chr12	124000000	125000000
              chr12	125000000	126000000
              chr12	126000000	127000000
              chr12	127000000	128000000
              chr12	128000000	129000000
              chr12	129000000	130000000
              chr12	130000000	131000000
              chr12	131000000	132000000
              chr12	132000000	133000000
              chr12	133000000	133275309
              chr12_GL383549v1_alt	0	120804
              chr12_GL383550v2_alt	0	169178
              chr12_GL383551v1_alt	0	184319
              chr12_GL383552v1_alt	0	138655
              chr12_GL383553v2_alt	0	152874
              chr12_GL877875v1_alt	0	167313
              chr12_GL877876v1_alt	0	408271
              chr12_KI270833v1_alt	0	76061
              chr12_KI270834v1_alt	0	119498
              chr12_KI270835v1_alt	0	238139
              chr12_KI270836v1_alt	0	56134
              chr12_KI270837v1_alt	0	40090
              chr12_KI270904v1_alt	0	572349
              chr13	0	1000000
              chr13	1000000	2000000
              chr13	2000000	3000000
              chr13	3000000	4000000
              chr13	4000000	5000000
              chr13	5000000	6000000
              chr13	6000000	7000000
              chr13	7000000	8000000
              chr13	8000000	9000000
              chr13	9000000	10000000
              chr13	10000000	11000000
              chr13	11000000	12000000
              chr13	12000000	13000000
              chr13	13000000	14000000
              chr13	14000000	15000000
              chr13	15000000	16000000
              chr13	16000000	17000000
              chr13	17000000	18000000
              chr13	18000000	19000000
              chr13	19000000	20000000
              chr13	20000000	21000000
              chr13	21000000	22000000
              chr13	22000000	23000000
              chr13	23000000	24000000
              chr13	24000000	25000000
              chr13	25000000	26000000
              chr13	26000000	27000000
              chr13	27000000	28000000
              chr13	28000000	29000000
              chr13	29000000	30000000
              chr13	30000000	31000000
              chr13	31000000	32000000
              chr13	32000000	33000000
              chr13	33000000	34000000
              chr13	34000000	35000000
              chr13	35000000	36000000
              chr13	36000000	37000000
              chr13	37000000	38000000
              chr13	38000000	39000000
              chr13	39000000	40000000
              chr13	40000000	41000000
              chr13	41000000	42000000
              chr13	42000000	43000000
              chr13	43000000	44000000
              chr13	44000000	45000000
              chr13	45000000	46000000
              chr13	46000000	47000000
              chr13	47000000	48000000
              chr13	48000000	49000000
              chr13	49000000	50000000
              chr13	50000000	51000000
              chr13	51000000	52000000
              chr13	52000000	53000000
              chr13	53000000	54000000
              chr13	54000000	55000000
              chr13	55000000	56000000
              chr13	56000000	57000000
              chr13	57000000	58000000
              chr13	58000000	59000000
              chr13	59000000	60000000
              chr13	60000000	61000000
              chr13	61000000	62000000
              chr13	62000000	63000000
              chr13	63000000	64000000
              chr13	64000000	65000000
              chr13	65000000	66000000
              chr13	66000000	67000000
              chr13	67000000	68000000
              chr13	68000000	69000000
              chr13	69000000	70000000
              chr13	70000000	71000000
              chr13	71000000	72000000
              chr13	72000000	73000000
              chr13	73000000	74000000
              chr13	74000000	75000000
              chr13	75000000	76000000
              chr13	76000000	77000000
              chr13	77000000	78000000
              chr13	78000000	79000000
              chr13	79000000	80000000
              chr13	80000000	81000000
              chr13	81000000	82000000
              chr13	82000000	83000000
              chr13	83000000	84000000
              chr13	84000000	85000000
              chr13	85000000	86000000
              chr13	86000000	87000000
              chr13	87000000	88000000
              chr13	88000000	89000000
              chr13	89000000	90000000
              chr13	90000000	91000000
              chr13	91000000	92000000
              chr13	92000000	93000000
              chr13	93000000	94000000
              chr13	94000000	95000000
              chr13	95000000	96000000
              chr13	96000000	97000000
              chr13	97000000	98000000
              chr13	98000000	99000000
              chr13	99000000	100000000
              chr13	100000000	101000000
              chr13	101000000	102000000
              chr13	102000000	103000000
              chr13	103000000	104000000
              chr13	104000000	105000000
              chr13	105000000	106000000
              chr13	106000000	107000000
              chr13	107000000	108000000
              chr13	108000000	109000000
              chr13	109000000	110000000
              chr13	110000000	111000000
              chr13	111000000	112000000
              chr13	112000000	113000000
              chr13	113000000	114000000
              chr13	114000000	114364328
              chr13_KI270838v1_alt	0	306913
              chr13_KI270839v1_alt	0	180306
              chr13_KI270840v1_alt	0	191684
              chr13_KI270841v1_alt	0	169134
              chr13_KI270842v1_alt	0	37287
              chr13_KI270843v1_alt	0	103832
              chr14	0	1000000
              chr14	1000000	2000000
              chr14	2000000	3000000
              chr14	3000000	4000000
              chr14	4000000	5000000
              chr14	5000000	6000000
              chr14	6000000	7000000
              chr14	7000000	8000000
              chr14	8000000	9000000
              chr14	9000000	10000000
              chr14	10000000	11000000
              chr14	11000000	12000000
              chr14	12000000	13000000
              chr14	13000000	14000000
              chr14	14000000	15000000
              chr14	15000000	16000000
              chr14	16000000	17000000
              chr14	17000000	18000000
              chr14	18000000	19000000
              chr14	19000000	20000000
              chr14	20000000	21000000
              chr14	21000000	22000000
              chr14	22000000	23000000
              chr14	23000000	24000000
              chr14	24000000	25000000
              chr14	25000000	26000000
              chr14	26000000	27000000
              chr14	27000000	28000000
              chr14	28000000	29000000
              chr14	29000000	30000000
              chr14	30000000	31000000
              chr14	31000000	32000000
              chr14	32000000	33000000
              chr14	33000000	34000000
              chr14	34000000	35000000
              chr14	35000000	36000000
              chr14	36000000	37000000
              chr14	37000000	38000000
              chr14	38000000	39000000
              chr14	39000000	40000000
              chr14	40000000	41000000
              chr14	41000000	42000000
              chr14	42000000	43000000
              chr14	43000000	44000000
              chr14	44000000	45000000
              chr14	45000000	46000000
              chr14	46000000	47000000
              chr14	47000000	48000000
              chr14	48000000	49000000
              chr14	49000000	50000000
              chr14	50000000	51000000
              chr14	51000000	52000000
              chr14	52000000	53000000
              chr14	53000000	54000000
              chr14	54000000	55000000
              chr14	55000000	56000000
              chr14	56000000	57000000
              chr14	57000000	58000000
              chr14	58000000	59000000
              chr14	59000000	60000000
              chr14	60000000	61000000
              chr14	61000000	62000000
              chr14	62000000	63000000
              chr14	63000000	64000000
              chr14	64000000	65000000
              chr14	65000000	66000000
              chr14	66000000	67000000
              chr14	67000000	68000000
              chr14	68000000	69000000
              chr14	69000000	70000000
              chr14	70000000	71000000
              chr14	71000000	72000000
              chr14	72000000	73000000
              chr14	73000000	74000000
              chr14	74000000	75000000
              chr14	75000000	76000000
              chr14	76000000	77000000
              chr14	77000000	78000000
              chr14	78000000	79000000
              chr14	79000000	80000000
              chr14	80000000	81000000
              chr14	81000000	82000000
              chr14	82000000	83000000
              chr14	83000000	84000000
              chr14	84000000	85000000
              chr14	85000000	86000000
              chr14	86000000	87000000
              chr14	87000000	88000000
              chr14	88000000	89000000
              chr14	89000000	90000000
              chr14	90000000	91000000
              chr14	91000000	92000000
              chr14	92000000	93000000
              chr14	93000000	94000000
              chr14	94000000	95000000
              chr14	95000000	96000000
              chr14	96000000	97000000
              chr14	97000000	98000000
              chr14	98000000	99000000
              chr14	99000000	100000000
              chr14	100000000	101000000
              chr14	101000000	102000000
              chr14	102000000	103000000
              chr14	103000000	104000000
              chr14	104000000	105000000
              chr14	105000000	106000000
              chr14	106000000	107000000
              chr14	107000000	107043718
              chr14_GL000009v2_random	0	201709
              chr14_GL000194v1_random	0	191469
              chr14_GL000225v1_random	0	211173
              chr14_KI270722v1_random	0	194050
              chr14_KI270723v1_random	0	38115
              chr14_KI270724v1_random	0	39555
              chr14_KI270725v1_random	0	172810
              chr14_KI270726v1_random	0	43739
              chr14_KI270844v1_alt	0	322166
              chr14_KI270845v1_alt	0	180703
              chr14_KI270846v1_alt	0	1000000
              chr14_KI270846v1_alt	1000000	1351393
              chr14_KI270847v1_alt	0	1000000
              chr14_KI270847v1_alt	1000000	1511111
              chr15	0	1000000
              chr15	1000000	2000000
              chr15	2000000	3000000
              chr15	3000000	4000000
              chr15	4000000	5000000
              chr15	5000000	6000000
              chr15	6000000	7000000
              chr15	7000000	8000000
              chr15	8000000	9000000
              chr15	9000000	10000000
              chr15	10000000	11000000
              chr15	11000000	12000000
              chr15	12000000	13000000
              chr15	13000000	14000000
              chr15	14000000	15000000
              chr15	15000000	16000000
              chr15	16000000	17000000
              chr15	17000000	18000000
              chr15	18000000	19000000
              chr15	19000000	20000000
              chr15	20000000	21000000
              chr15	21000000	22000000
              chr15	22000000	23000000
              chr15	23000000	24000000
              chr15	24000000	25000000
              chr15	25000000	26000000
              chr15	26000000	27000000
              chr15	27000000	28000000
              chr15	28000000	29000000
              chr15	29000000	30000000
              chr15	30000000	31000000
              chr15	31000000	32000000
              chr15	32000000	33000000
              chr15	33000000	34000000
              chr15	34000000	35000000
              chr15	35000000	36000000
              chr15	36000000	37000000
              chr15	37000000	38000000
              chr15	38000000	39000000
              chr15	39000000	40000000
              chr15	40000000	41000000
              chr15	41000000	42000000
              chr15	42000000	43000000
              chr15	43000000	44000000
              chr15	44000000	45000000
              chr15	45000000	46000000
              chr15	46000000	47000000
              chr15	47000000	48000000
              chr15	48000000	49000000
              chr15	49000000	50000000
              chr15	50000000	51000000
              chr15	51000000	52000000
              chr15	52000000	53000000
              chr15	53000000	54000000
              chr15	54000000	55000000
              chr15	55000000	56000000
              chr15	56000000	57000000
              chr15	57000000	58000000
              chr15	58000000	59000000
              chr15	59000000	60000000
              chr15	60000000	61000000
              chr15	61000000	62000000
              chr15	62000000	63000000
              chr15	63000000	64000000
              chr15	64000000	65000000
              chr15	65000000	66000000
              chr15	66000000	67000000
              chr15	67000000	68000000
              chr15	68000000	69000000
              chr15	69000000	70000000
              chr15	70000000	71000000
              chr15	71000000	72000000
              chr15	72000000	73000000
              chr15	73000000	74000000
              chr15	74000000	75000000
              chr15	75000000	76000000
              chr15	76000000	77000000
              chr15	77000000	78000000
              chr15	78000000	79000000
              chr15	79000000	80000000
              chr15	80000000	81000000
              chr15	81000000	82000000
              chr15	82000000	83000000
              chr15	83000000	84000000
              chr15	84000000	85000000
              chr15	85000000	86000000
              chr15	86000000	87000000
              chr15	87000000	88000000
              chr15	88000000	89000000
              chr15	89000000	90000000
              chr15	90000000	91000000
              chr15	91000000	92000000
              chr15	92000000	93000000
              chr15	93000000	94000000
              chr15	94000000	95000000
              chr15	95000000	96000000
              chr15	96000000	97000000
              chr15	97000000	98000000
              chr15	98000000	99000000
              chr15	99000000	100000000
              chr15	100000000	101000000
              chr15	101000000	101991189
              chr15_GL383554v1_alt	0	296527
              chr15_GL383555v2_alt	0	388773
              chr15_KI270727v1_random	0	448248
              chr15_KI270848v1_alt	0	327382
              chr15_KI270849v1_alt	0	244917
              chr15_KI270850v1_alt	0	430880
              chr15_KI270851v1_alt	0	263054
              chr15_KI270852v1_alt	0	478999
              chr15_KI270905v1_alt	0	1000000
              chr15_KI270905v1_alt	1000000	2000000
              chr15_KI270905v1_alt	2000000	3000000
              chr15_KI270905v1_alt	3000000	4000000
              chr15_KI270905v1_alt	4000000	5000000
              chr15_KI270905v1_alt	5000000	5161414
              chr15_KI270906v1_alt	0	196384
              chr16	0	1000000
              chr16	1000000	2000000
              chr16	2000000	3000000
              chr16	3000000	4000000
              chr16	4000000	5000000
              chr16	5000000	6000000
              chr16	6000000	7000000
              chr16	7000000	8000000
              chr16	8000000	9000000
              chr16	9000000	10000000
              chr16	10000000	11000000
              chr16	11000000	12000000
              chr16	12000000	13000000
              chr16	13000000	14000000
              chr16	14000000	15000000
              chr16	15000000	16000000
              chr16	16000000	17000000
              chr16	17000000	18000000
              chr16	18000000	19000000
              chr16	19000000	20000000
              chr16	20000000	21000000
              chr16	21000000	22000000
              chr16	22000000	23000000
              chr16	23000000	24000000
              chr16	24000000	25000000
              chr16	25000000	26000000
              chr16	26000000	27000000
              chr16	27000000	28000000
              chr16	28000000	29000000
              chr16	29000000	30000000
              chr16	30000000	31000000
              chr16	31000000	32000000
              chr16	32000000	33000000
              chr16	33000000	34000000
              chr16	34000000	35000000
              chr16	35000000	36000000
              chr16	36000000	37000000
              chr16	37000000	38000000
              chr16	38000000	39000000
              chr16	39000000	40000000
              chr16	40000000	41000000
              chr16	41000000	42000000
              chr16	42000000	43000000
              chr16	43000000	44000000
              chr16	44000000	45000000
              chr16	45000000	46000000
              chr16	46000000	47000000
              chr16	47000000	48000000
              chr16	48000000	49000000
              chr16	49000000	50000000
              chr16	50000000	51000000
              chr16	51000000	52000000
              chr16	52000000	53000000
              chr16	53000000	54000000
              chr16	54000000	55000000
              chr16	55000000	56000000
              chr16	56000000	57000000
              chr16	57000000	58000000
              chr16	58000000	59000000
              chr16	59000000	60000000
              chr16	60000000	61000000
              chr16	61000000	62000000
              chr16	62000000	63000000
              chr16	63000000	64000000
              chr16	64000000	65000000
              chr16	65000000	66000000
              chr16	66000000	67000000
              chr16	67000000	68000000
              chr16	68000000	69000000
              chr16	69000000	70000000
              chr16	70000000	71000000
              chr16	71000000	72000000
              chr16	72000000	73000000
              chr16	73000000	74000000
              chr16	74000000	75000000
              chr16	75000000	76000000
              chr16	76000000	77000000
              chr16	77000000	78000000
              chr16	78000000	79000000
              chr16	79000000	80000000
              chr16	80000000	81000000
              chr16	81000000	82000000
              chr16	82000000	83000000
              chr16	83000000	84000000
              chr16	84000000	85000000
              chr16	85000000	86000000
              chr16	86000000	87000000
              chr16	87000000	88000000
              chr16	88000000	89000000
              chr16	89000000	90000000
              chr16	90000000	90338345
              chr16_GL383556v1_alt	0	192462
              chr16_GL383557v1_alt	0	89672
              chr16_KI270728v1_random	0	1000000
              chr16_KI270728v1_random	1000000	1872759
              chr16_KI270853v1_alt	0	1000000
              chr16_KI270853v1_alt	1000000	2000000
              chr16_KI270853v1_alt	2000000	2659700
              chr16_KI270854v1_alt	0	134193
              chr16_KI270855v1_alt	0	232857
              chr16_KI270856v1_alt	0	63982
              chr17	0	1000000
              chr17	1000000	2000000
              chr17	2000000	3000000
              chr17	3000000	4000000
              chr17	4000000	5000000
              chr17	5000000	6000000
              chr17	6000000	7000000
              chr17	7000000	8000000
              chr17	8000000	9000000
              chr17	9000000	10000000
              chr17	10000000	11000000
              chr17	11000000	12000000
              chr17	12000000	13000000
              chr17	13000000	14000000
              chr17	14000000	15000000
              chr17	15000000	16000000
              chr17	16000000	17000000
              chr17	17000000	18000000
              chr17	18000000	19000000
              chr17	19000000	20000000
              chr17	20000000	21000000
              chr17	21000000	22000000
              chr17	22000000	23000000
              chr17	23000000	24000000
              chr17	24000000	25000000
              chr17	25000000	26000000
              chr17	26000000	27000000
              chr17	27000000	28000000
              chr17	28000000	29000000
              chr17	29000000	30000000
              chr17	30000000	31000000
              chr17	31000000	32000000
              chr17	32000000	33000000
              chr17	33000000	34000000
              chr17	34000000	35000000
              chr17	35000000	36000000
              chr17	36000000	37000000
              chr17	37000000	38000000
              chr17	38000000	39000000
              chr17	39000000	40000000
              chr17	40000000	41000000
              chr17	41000000	42000000
              chr17	42000000	43000000
              chr17	43000000	44000000
              chr17	44000000	45000000
              chr17	45000000	46000000
              chr17	46000000	47000000
              chr17	47000000	48000000
              chr17	48000000	49000000
              chr17	49000000	50000000
              chr17	50000000	51000000
              chr17	51000000	52000000
              chr17	52000000	53000000
              chr17	53000000	54000000
              chr17	54000000	55000000
              chr17	55000000	56000000
              chr17	56000000	57000000
              chr17	57000000	58000000
              chr17	58000000	59000000
              chr17	59000000	60000000
              chr17	60000000	61000000
              chr17	61000000	62000000
              chr17	62000000	63000000
              chr17	63000000	64000000
              chr17	64000000	65000000
              chr17	65000000	66000000
              chr17	66000000	67000000
              chr17	67000000	68000000
              chr17	68000000	69000000
              chr17	69000000	70000000
              chr17	70000000	71000000
              chr17	71000000	72000000
              chr17	72000000	73000000
              chr17	73000000	74000000
              chr17	74000000	75000000
              chr17	75000000	76000000
              chr17	76000000	77000000
              chr17	77000000	78000000
              chr17	78000000	79000000
              chr17	79000000	80000000
              chr17	80000000	81000000
              chr17	81000000	82000000
              chr17	82000000	83000000
              chr17	83000000	83257441
              chr17_GL000205v2_random	0	185591
              chr17_GL000258v2_alt	0	1000000
              chr17_GL000258v2_alt	1000000	1821992
              chr17_GL383563v3_alt	0	375691
              chr17_GL383564v2_alt	0	133151
              chr17_GL383565v1_alt	0	223995
              chr17_GL383566v1_alt	0	90219
              chr17_JH159146v1_alt	0	278131
              chr17_JH159147v1_alt	0	70345
              chr17_JH159148v1_alt	0	88070
              chr17_KI270729v1_random	0	280839
              chr17_KI270730v1_random	0	112551
              chr17_KI270857v1_alt	0	1000000
              chr17_KI270857v1_alt	1000000	2000000
              chr17_KI270857v1_alt	2000000	2877074
              chr17_KI270858v1_alt	0	235827
              chr17_KI270859v1_alt	0	108763
              chr17_KI270860v1_alt	0	178921
              chr17_KI270861v1_alt	0	196688
              chr17_KI270862v1_alt	0	391357
              chr17_KI270907v1_alt	0	137721
              chr17_KI270908v1_alt	0	1000000
              chr17_KI270908v1_alt	1000000	1423190
              chr17_KI270909v1_alt	0	325800
              chr17_KI270910v1_alt	0	157099
              chr18	0	1000000
              chr18	1000000	2000000
              chr18	2000000	3000000
              chr18	3000000	4000000
              chr18	4000000	5000000
              chr18	5000000	6000000
              chr18	6000000	7000000
              chr18	7000000	8000000
              chr18	8000000	9000000
              chr18	9000000	10000000
              chr18	10000000	11000000
              chr18	11000000	12000000
              chr18	12000000	13000000
              chr18	13000000	14000000
              chr18	14000000	15000000
              chr18	15000000	16000000
              chr18	16000000	17000000
              chr18	17000000	18000000
              chr18	18000000	19000000
              chr18	19000000	20000000
              chr18	20000000	21000000
              chr18	21000000	22000000
              chr18	22000000	23000000
              chr18	23000000	24000000
              chr18	24000000	25000000
              chr18	25000000	26000000
              chr18	26000000	27000000
              chr18	27000000	28000000
              chr18	28000000	29000000
              chr18	29000000	30000000
              chr18	30000000	31000000
              chr18	31000000	32000000
              chr18	32000000	33000000
              chr18	33000000	34000000
              chr18	34000000	35000000
              chr18	35000000	36000000
              chr18	36000000	37000000
              chr18	37000000	38000000
              chr18	38000000	39000000
              chr18	39000000	40000000
              chr18	40000000	41000000
              chr18	41000000	42000000
              chr18	42000000	43000000
              chr18	43000000	44000000
              chr18	44000000	45000000
              chr18	45000000	46000000
              chr18	46000000	47000000
              chr18	47000000	48000000
              chr18	48000000	49000000
              chr18	49000000	50000000
              chr18	50000000	51000000
              chr18	51000000	52000000
              chr18	52000000	53000000
              chr18	53000000	54000000
              chr18	54000000	55000000
              chr18	55000000	56000000
              chr18	56000000	57000000
              chr18	57000000	58000000
              chr18	58000000	59000000
              chr18	59000000	60000000
              chr18	60000000	61000000
              chr18	61000000	62000000
              chr18	62000000	63000000
              chr18	63000000	64000000
              chr18	64000000	65000000
              chr18	65000000	66000000
              chr18	66000000	67000000
              chr18	67000000	68000000
              chr18	68000000	69000000
              chr18	69000000	70000000
              chr18	70000000	71000000
              chr18	71000000	72000000
              chr18	72000000	73000000
              chr18	73000000	74000000
              chr18	74000000	75000000
              chr18	75000000	76000000
              chr18	76000000	77000000
              chr18	77000000	78000000
              chr18	78000000	79000000
              chr18	79000000	80000000
              chr18	80000000	80373285
              chr18_GL383567v1_alt	0	289831
              chr18_GL383568v1_alt	0	104552
              chr18_GL383569v1_alt	0	167950
              chr18_GL383570v1_alt	0	164789
              chr18_GL383571v1_alt	0	198278
              chr18_GL383572v1_alt	0	159547
              chr18_KI270863v1_alt	0	167999
              chr18_KI270864v1_alt	0	111737
              chr18_KI270911v1_alt	0	157710
              chr18_KI270912v1_alt	0	174061
              chr19	0	1000000
              chr19	1000000	2000000
              chr19	2000000	3000000
              chr19	3000000	4000000
              chr19	4000000	5000000
              chr19	5000000	6000000
              chr19	6000000	7000000
              chr19	7000000	8000000
              chr19	8000000	9000000
              chr19	9000000	10000000
              chr19	10000000	11000000
              chr19	11000000	12000000
              chr19	12000000	13000000
              chr19	13000000	14000000
              chr19	14000000	15000000
              chr19	15000000	16000000
              chr19	16000000	17000000
              chr19	17000000	18000000
              chr19	18000000	19000000
              chr19	19000000	20000000
              chr19	20000000	21000000
              chr19	21000000	22000000
              chr19	22000000	23000000
              chr19	23000000	24000000
              chr19	24000000	25000000
              chr19	25000000	26000000
              chr19	26000000	27000000
              chr19	27000000	28000000
              chr19	28000000	29000000
              chr19	29000000	30000000
              chr19	30000000	31000000
              chr19	31000000	32000000
              chr19	32000000	33000000
              chr19	33000000	34000000
              chr19	34000000	35000000
              chr19	35000000	36000000
              chr19	36000000	37000000
              chr19	37000000	38000000
              chr19	38000000	39000000
              chr19	39000000	40000000
              chr19	40000000	41000000
              chr19	41000000	42000000
              chr19	42000000	43000000
              chr19	43000000	44000000
              chr19	44000000	45000000
              chr19	45000000	46000000
              chr19	46000000	47000000
              chr19	47000000	48000000
              chr19	48000000	49000000
              chr19	49000000	50000000
              chr19	50000000	51000000
              chr19	51000000	52000000
              chr19	52000000	53000000
              chr19	53000000	54000000
              chr19	54000000	55000000
              chr19	55000000	56000000
              chr19	56000000	57000000
              chr19	57000000	58000000
              chr19	58000000	58617616
              chr19_GL000209v2_alt	0	177381
              chr19_GL383573v1_alt	0	385657
              chr19_GL383574v1_alt	0	155864
              chr19_GL383575v2_alt	0	170222
              chr19_GL383576v1_alt	0	188024
              chr19_GL949746v1_alt	0	987716
              chr19_GL949747v2_alt	0	729520
              chr19_GL949748v2_alt	0	1000000
              chr19_GL949748v2_alt	1000000	1064304
              chr19_GL949749v2_alt	0	1000000
              chr19_GL949749v2_alt	1000000	1091841
              chr19_GL949750v2_alt	0	1000000
              chr19_GL949750v2_alt	1000000	1066390
              chr19_GL949751v2_alt	0	1000000
              chr19_GL949751v2_alt	1000000	1002683
              chr19_GL949752v1_alt	0	987100
              chr19_GL949753v2_alt	0	796479
              chr19_KI270865v1_alt	0	52969
              chr19_KI270866v1_alt	0	43156
              chr19_KI270867v1_alt	0	233762
              chr19_KI270868v1_alt	0	61734
              chr19_KI270882v1_alt	0	248807
              chr19_KI270883v1_alt	0	170399
              chr19_KI270884v1_alt	0	157053
              chr19_KI270885v1_alt	0	171027
              chr19_KI270886v1_alt	0	204239
              chr19_KI270887v1_alt	0	209512
              chr19_KI270888v1_alt	0	155532
              chr19_KI270889v1_alt	0	170698
              chr19_KI270890v1_alt	0	184499
              chr19_KI270891v1_alt	0	170680
              chr19_KI270914v1_alt	0	205194
              chr19_KI270915v1_alt	0	170665
              chr19_KI270916v1_alt	0	184516
              chr19_KI270917v1_alt	0	190932
              chr19_KI270918v1_alt	0	123111
              chr19_KI270919v1_alt	0	170701
              chr19_KI270920v1_alt	0	198005
              chr19_KI270921v1_alt	0	282224
              chr19_KI270922v1_alt	0	187935
              chr19_KI270923v1_alt	0	189352
              chr19_KI270929v1_alt	0	186203
              chr19_KI270930v1_alt	0	200773
              chr19_KI270931v1_alt	0	170148
              chr19_KI270932v1_alt	0	215732
              chr19_KI270933v1_alt	0	170537
              chr19_KI270938v1_alt	0	1000000
              chr19_KI270938v1_alt	1000000	1066800
              chr1_GL383518v1_alt	0	182439
              chr1_GL383519v1_alt	0	110268
              chr1_GL383520v2_alt	0	366580
              chr1_KI270706v1_random	0	175055
              chr1_KI270707v1_random	0	32032
              chr1_KI270708v1_random	0	127682
              chr1_KI270709v1_random	0	66860
              chr1_KI270710v1_random	0	40176
              chr1_KI270711v1_random	0	42210
              chr1_KI270712v1_random	0	176043
              chr1_KI270713v1_random	0	40745
              chr1_KI270714v1_random	0	41717
              chr1_KI270759v1_alt	0	425601
              chr1_KI270760v1_alt	0	109528
              chr1_KI270761v1_alt	0	165834
              chr1_KI270762v1_alt	0	354444
              chr1_KI270763v1_alt	0	911658
              chr1_KI270764v1_alt	0	50258
              chr1_KI270765v1_alt	0	185285
              chr1_KI270766v1_alt	0	256271
              chr1_KI270892v1_alt	0	162212
              chr2	0	1000000
              chr2	1000000	2000000
              chr2	2000000	3000000
              chr2	3000000	4000000
              chr2	4000000	5000000
              chr2	5000000	6000000
              chr2	6000000	7000000
              chr2	7000000	8000000
              chr2	8000000	9000000
              chr2	9000000	10000000
              chr2	10000000	11000000
              chr2	11000000	12000000
              chr2	12000000	13000000
              chr2	13000000	14000000
              chr2	14000000	15000000
              chr2	15000000	16000000
              chr2	16000000	17000000
              chr2	17000000	18000000
              chr2	18000000	19000000
              chr2	19000000	20000000
              chr2	20000000	21000000
              chr2	21000000	22000000
              chr2	22000000	23000000
              chr2	23000000	24000000
              chr2	24000000	25000000
              chr2	25000000	26000000
              chr2	26000000	27000000
              chr2	27000000	28000000
              chr2	28000000	29000000
              chr2	29000000	30000000
              chr2	30000000	31000000
              chr2	31000000	32000000
              chr2	32000000	33000000
              chr2	33000000	34000000
              chr2	34000000	35000000
              chr2	35000000	36000000
              chr2	36000000	37000000
              chr2	37000000	38000000
              chr2	38000000	39000000
              chr2	39000000	40000000
              chr2	40000000	41000000
              chr2	41000000	42000000
              chr2	42000000	43000000
              chr2	43000000	44000000
              chr2	44000000	45000000
              chr2	45000000	46000000
              chr2	46000000	47000000
              chr2	47000000	48000000
              chr2	48000000	49000000
              chr2	49000000	50000000
              chr2	50000000	51000000
              chr2	51000000	52000000
              chr2	52000000	53000000
              chr2	53000000	54000000
              chr2	54000000	55000000
              chr2	55000000	56000000
              chr2	56000000	57000000
              chr2	57000000	58000000
              chr2	58000000	59000000
              chr2	59000000	60000000
              chr2	60000000	61000000
              chr2	61000000	62000000
              chr2	62000000	63000000
              chr2	63000000	64000000
              chr2	64000000	65000000
              chr2	65000000	66000000
              chr2	66000000	67000000
              chr2	67000000	68000000
              chr2	68000000	69000000
              chr2	69000000	70000000
              chr2	70000000	71000000
              chr2	71000000	72000000
              chr2	72000000	73000000
              chr2	73000000	74000000
              chr2	74000000	75000000
              chr2	75000000	76000000
              chr2	76000000	77000000
              chr2	77000000	78000000
              chr2	78000000	79000000
              chr2	79000000	80000000
              chr2	80000000	81000000
              chr2	81000000	82000000
              chr2	82000000	83000000
              chr2	83000000	84000000
              chr2	84000000	85000000
              chr2	85000000	86000000
              chr2	86000000	87000000
              chr2	87000000	88000000
              chr2	88000000	89000000
              chr2	89000000	90000000
              chr2	90000000	91000000
              chr2	91000000	92000000
              chr2	92000000	93000000
              chr2	93000000	94000000
              chr2	94000000	95000000
              chr2	95000000	96000000
              chr2	96000000	97000000
              chr2	97000000	98000000
              chr2	98000000	99000000
              chr2	99000000	100000000
              chr2	100000000	101000000
              chr2	101000000	102000000
              chr2	102000000	103000000
              chr2	103000000	104000000
              chr2	104000000	105000000
              chr2	105000000	106000000
              chr2	106000000	107000000
              chr2	107000000	108000000
              chr2	108000000	109000000
              chr2	109000000	110000000
              chr2	110000000	111000000
              chr2	111000000	112000000
              chr2	112000000	113000000
              chr2	113000000	114000000
              chr2	114000000	115000000
              chr2	115000000	116000000
              chr2	116000000	117000000
              chr2	117000000	118000000
              chr2	118000000	119000000
              chr2	119000000	120000000
              chr2	120000000	121000000
              chr2	121000000	122000000
              chr2	122000000	123000000
              chr2	123000000	124000000
              chr2	124000000	125000000
              chr2	125000000	126000000
              chr2	126000000	127000000
              chr2	127000000	128000000
              chr2	128000000	129000000
              chr2	129000000	130000000
              chr2	130000000	131000000
              chr2	131000000	132000000
              chr2	132000000	133000000
              chr2	133000000	134000000
              chr2	134000000	135000000
              chr2	135000000	136000000
              chr2	136000000	137000000
              chr2	137000000	138000000
              chr2	138000000	139000000
              chr2	139000000	140000000
              chr2	140000000	141000000
              chr2	141000000	142000000
              chr2	142000000	143000000
              chr2	143000000	144000000
              chr2	144000000	145000000
              chr2	145000000	146000000
              chr2	146000000	147000000
              chr2	147000000	148000000
              chr2	148000000	149000000
              chr2	149000000	150000000
              chr2	150000000	151000000
              chr2	151000000	152000000
              chr2	152000000	153000000
              chr2	153000000	154000000
              chr2	154000000	155000000
              chr2	155000000	156000000
              chr2	156000000	157000000
              chr2	157000000	158000000
              chr2	158000000	159000000
              chr2	159000000	160000000
              chr2	160000000	161000000
              chr2	161000000	162000000
              chr2	162000000	163000000
              chr2	163000000	164000000
              chr2	164000000	165000000
              chr2	165000000	166000000
              chr2	166000000	167000000
              chr2	167000000	168000000
              chr2	168000000	169000000
              chr2	169000000	170000000
              chr2	170000000	171000000
              chr2	171000000	172000000
              chr2	172000000	173000000
              chr2	173000000	174000000
              chr2	174000000	175000000
              chr2	175000000	176000000
              chr2	176000000	177000000
              chr2	177000000	178000000
              chr2	178000000	179000000
              chr2	179000000	180000000
              chr2	180000000	181000000
              chr2	181000000	182000000
              chr2	182000000	183000000
              chr2	183000000	184000000
              chr2	184000000	185000000
              chr2	185000000	186000000
              chr2	186000000	187000000
              chr2	187000000	188000000
              chr2	188000000	189000000
              chr2	189000000	190000000
              chr2	190000000	191000000
              chr2	191000000	192000000
              chr2	192000000	193000000
              chr2	193000000	194000000
              chr2	194000000	195000000
              chr2	195000000	196000000
              chr2	196000000	197000000
              chr2	197000000	198000000
              chr2	198000000	199000000
              chr2	199000000	200000000
              chr2	200000000	201000000
              chr2	201000000	202000000
              chr2	202000000	203000000
              chr2	203000000	204000000
              chr2	204000000	205000000
              chr2	205000000	206000000
              chr2	206000000	207000000
              chr2	207000000	208000000
              chr2	208000000	209000000
              chr2	209000000	210000000
              chr2	210000000	211000000
              chr2	211000000	212000000
              chr2	212000000	213000000
              chr2	213000000	214000000
              chr2	214000000	215000000
              chr2	215000000	216000000
              chr2	216000000	217000000
              chr2	217000000	218000000
              chr2	218000000	219000000
              chr2	219000000	220000000
              chr2	220000000	221000000
              chr2	221000000	222000000
              chr2	222000000	223000000
              chr2	223000000	224000000
              chr2	224000000	225000000
              chr2	225000000	226000000
              chr2	226000000	227000000
              chr2	227000000	228000000
              chr2	228000000	229000000
              chr2	229000000	230000000
              chr2	230000000	231000000
              chr2	231000000	232000000
              chr2	232000000	233000000
              chr2	233000000	234000000
              chr2	234000000	235000000
              chr2	235000000	236000000
              chr2	236000000	237000000
              chr2	237000000	238000000
              chr2	238000000	239000000
              chr2	239000000	240000000
              chr2	240000000	241000000
              chr2	241000000	242000000
              chr2	242000000	242193529
              chr20	0	1000000
              chr20	1000000	2000000
              chr20	2000000	3000000
              chr20	3000000	4000000
              chr20	4000000	5000000
              chr20	5000000	6000000
              chr20	6000000	7000000
              chr20	7000000	8000000
              chr20	8000000	9000000
              chr20	9000000	10000000
              chr20	10000000	11000000
              chr20	11000000	12000000
              chr20	12000000	13000000
              chr20	13000000	14000000
              chr20	14000000	15000000
              chr20	15000000	16000000
              chr20	16000000	17000000
              chr20	17000000	18000000
              chr20	18000000	19000000
              chr20	19000000	20000000
              chr20	20000000	21000000
              chr20	21000000	22000000
              chr20	22000000	23000000
              chr20	23000000	24000000
              chr20	24000000	25000000
              chr20	25000000	26000000
              chr20	26000000	27000000
              chr20	27000000	28000000
              chr20	28000000	29000000
              chr20	29000000	30000000
              chr20	30000000	31000000
              chr20	31000000	32000000
              chr20	32000000	33000000
              chr20	33000000	34000000
              chr20	34000000	35000000
              chr20	35000000	36000000
              chr20	36000000	37000000
              chr20	37000000	38000000
              chr20	38000000	39000000
              chr20	39000000	40000000
              chr20	40000000	41000000
              chr20	41000000	42000000
              chr20	42000000	43000000
              chr20	43000000	44000000
              chr20	44000000	45000000
              chr20	45000000	46000000
              chr20	46000000	47000000
              chr20	47000000	48000000
              chr20	48000000	49000000
              chr20	49000000	50000000
              chr20	50000000	51000000
              chr20	51000000	52000000
              chr20	52000000	53000000
              chr20	53000000	54000000
              chr20	54000000	55000000
              chr20	55000000	56000000
              chr20	56000000	57000000
              chr20	57000000	58000000
              chr20	58000000	59000000
              chr20	59000000	60000000
              chr20	60000000	61000000
              chr20	61000000	62000000
              chr20	62000000	63000000
              chr20	63000000	64000000
              chr20	64000000	64444167
              chr20_GL383577v2_alt	0	128386
              chr20_KI270869v1_alt	0	118774
              chr20_KI270870v1_alt	0	183433
              chr20_KI270871v1_alt	0	58661
              chr21	0	1000000
              chr21	1000000	2000000
              chr21	2000000	3000000
              chr21	3000000	4000000
              chr21	4000000	5000000
              chr21	5000000	6000000
              chr21	6000000	7000000
              chr21	7000000	8000000
              chr21	8000000	9000000
              chr21	9000000	10000000
              chr21	10000000	11000000
              chr21	11000000	12000000
              chr21	12000000	13000000
              chr21	13000000	14000000
              chr21	14000000	15000000
              chr21	15000000	16000000
              chr21	16000000	17000000
              chr21	17000000	18000000
              chr21	18000000	19000000
              chr21	19000000	20000000
              chr21	20000000	21000000
              chr21	21000000	22000000
              chr21	22000000	23000000
              chr21	23000000	24000000
              chr21	24000000	25000000
              chr21	25000000	26000000
              chr21	26000000	27000000
              chr21	27000000	28000000
              chr21	28000000	29000000
              chr21	29000000	30000000
              chr21	30000000	31000000
              chr21	31000000	32000000
              chr21	32000000	33000000
              chr21	33000000	34000000
              chr21	34000000	35000000
              chr21	35000000	36000000
              chr21	36000000	37000000
              chr21	37000000	38000000
              chr21	38000000	39000000
              chr21	39000000	40000000
              chr21	40000000	41000000
              chr21	41000000	42000000
              chr21	42000000	43000000
              chr21	43000000	44000000
              chr21	44000000	45000000
              chr21	45000000	46000000
              chr21	46000000	46709983
              chr21_GL383578v2_alt	0	63917
              chr21_GL383579v2_alt	0	201197
              chr21_GL383580v2_alt	0	74653
              chr21_GL383581v2_alt	0	116689
              chr21_KI270872v1_alt	0	82692
              chr21_KI270873v1_alt	0	143900
              chr21_KI270874v1_alt	0	166743
              chr22	0	1000000
              chr22	1000000	2000000
              chr22	2000000	3000000
              chr22	3000000	4000000
              chr22	4000000	5000000
              chr22	5000000	6000000
              chr22	6000000	7000000
              chr22	7000000	8000000
              chr22	8000000	9000000
              chr22	9000000	10000000
              chr22	10000000	11000000
              chr22	11000000	12000000
              chr22	12000000	13000000
              chr22	13000000	14000000
              chr22	14000000	15000000
              chr22	15000000	16000000
              chr22	16000000	17000000
              chr22	17000000	18000000
              chr22	18000000	19000000
              chr22	19000000	20000000
              chr22	20000000	21000000
              chr22	21000000	22000000
              chr22	22000000	23000000
              chr22	23000000	24000000
              chr22	24000000	25000000
              chr22	25000000	26000000
              chr22	26000000	27000000
              chr22	27000000	28000000
              chr22	28000000	29000000
              chr22	29000000	30000000
              chr22	30000000	31000000
              chr22	31000000	32000000
              chr22	32000000	33000000
              chr22	33000000	34000000
              chr22	34000000	35000000
              chr22	35000000	36000000
              chr22	36000000	37000000
              chr22	37000000	38000000
              chr22	38000000	39000000
              chr22	39000000	40000000
              chr22	40000000	41000000
              chr22	41000000	42000000
              chr22	42000000	43000000
              chr22	43000000	44000000
              chr22	44000000	45000000
              chr22	45000000	46000000
              chr22	46000000	47000000
              chr22	47000000	48000000
              chr22	48000000	49000000
              chr22	49000000	50000000
              chr22	50000000	50818468
              chr22_GL383582v2_alt	0	162811
              chr22_GL383583v2_alt	0	96924
              chr22_KB663609v1_alt	0	74013
              chr22_KI270731v1_random	0	150754
              chr22_KI270732v1_random	0	41543
              chr22_KI270733v1_random	0	179772
              chr22_KI270734v1_random	0	165050
              chr22_KI270735v1_random	0	42811
              chr22_KI270736v1_random	0	181920
              chr22_KI270737v1_random	0	103838
              chr22_KI270738v1_random	0	99375
              chr22_KI270739v1_random	0	73985
              chr22_KI270875v1_alt	0	259914
              chr22_KI270876v1_alt	0	263666
              chr22_KI270877v1_alt	0	101331
              chr22_KI270878v1_alt	0	186262
              chr22_KI270879v1_alt	0	304135
              chr22_KI270928v1_alt	0	176103
              chr2_GL383521v1_alt	0	143390
              chr2_GL383522v1_alt	0	123821
              chr2_GL582966v2_alt	0	96131
              chr2_KI270715v1_random	0	161471
              chr2_KI270716v1_random	0	153799
              chr2_KI270767v1_alt	0	161578
              chr2_KI270768v1_alt	0	110099
              chr2_KI270769v1_alt	0	120616
              chr2_KI270770v1_alt	0	136240
              chr2_KI270771v1_alt	0	110395
              chr2_KI270772v1_alt	0	133041
              chr2_KI270773v1_alt	0	70887
              chr2_KI270774v1_alt	0	223625
              chr2_KI270775v1_alt	0	138019
              chr2_KI270776v1_alt	0	174166
              chr2_KI270893v1_alt	0	161218
              chr2_KI270894v1_alt	0	214158
              chr3	0	1000000
              chr3	1000000	2000000
              chr3	2000000	3000000
              chr3	3000000	4000000
              chr3	4000000	5000000
              chr3	5000000	6000000
              chr3	6000000	7000000
              chr3	7000000	8000000
              chr3	8000000	9000000
              chr3	9000000	10000000
              chr3	10000000	11000000
              chr3	11000000	12000000
              chr3	12000000	13000000
              chr3	13000000	14000000
              chr3	14000000	15000000
              chr3	15000000	16000000
              chr3	16000000	17000000
              chr3	17000000	18000000
              chr3	18000000	19000000
              chr3	19000000	20000000
              chr3	20000000	21000000
              chr3	21000000	22000000
              chr3	22000000	23000000
              chr3	23000000	24000000
              chr3	24000000	25000000
              chr3	25000000	26000000
              chr3	26000000	27000000
              chr3	27000000	28000000
              chr3	28000000	29000000
              chr3	29000000	30000000
              chr3	30000000	31000000
              chr3	31000000	32000000
              chr3	32000000	33000000
              chr3	33000000	34000000
              chr3	34000000	35000000
              chr3	35000000	36000000
              chr3	36000000	37000000
              chr3	37000000	38000000
              chr3	38000000	39000000
              chr3	39000000	40000000
              chr3	40000000	41000000
              chr3	41000000	42000000
              chr3	42000000	43000000
              chr3	43000000	44000000
              chr3	44000000	45000000
              chr3	45000000	46000000
              chr3	46000000	47000000
              chr3	47000000	48000000
              chr3	48000000	49000000
              chr3	49000000	50000000
              chr3	50000000	51000000
              chr3	51000000	52000000
              chr3	52000000	53000000
              chr3	53000000	54000000
              chr3	54000000	55000000
              chr3	55000000	56000000
              chr3	56000000	57000000
              chr3	57000000	58000000
              chr3	58000000	59000000
              chr3	59000000	60000000
              chr3	60000000	61000000
              chr3	61000000	62000000
              chr3	62000000	63000000
              chr3	63000000	64000000
              chr3	64000000	65000000
              chr3	65000000	66000000
              chr3	66000000	67000000
              chr3	67000000	68000000
              chr3	68000000	69000000
              chr3	69000000	70000000
              chr3	70000000	71000000
              chr3	71000000	72000000
              chr3	72000000	73000000
              chr3	73000000	74000000
              chr3	74000000	75000000
              chr3	75000000	76000000
              chr3	76000000	77000000
              chr3	77000000	78000000
              chr3	78000000	79000000
              chr3	79000000	80000000
              chr3	80000000	81000000
              chr3	81000000	82000000
              chr3	82000000	83000000
              chr3	83000000	84000000
              chr3	84000000	85000000
              chr3	85000000	86000000
              chr3	86000000	87000000
              chr3	87000000	88000000
              chr3	88000000	89000000
              chr3	89000000	90000000
              chr3	90000000	91000000
              chr3	91000000	92000000
              chr3	92000000	93000000
              chr3	93000000	94000000
              chr3	94000000	95000000
              chr3	95000000	96000000
              chr3	96000000	97000000
              chr3	97000000	98000000
              chr3	98000000	99000000
              chr3	99000000	100000000
              chr3	100000000	101000000
              chr3	101000000	102000000
              chr3	102000000	103000000
              chr3	103000000	104000000
              chr3	104000000	105000000
              chr3	105000000	106000000
              chr3	106000000	107000000
              chr3	107000000	108000000
              chr3	108000000	109000000
              chr3	109000000	110000000
              chr3	110000000	111000000
              chr3	111000000	112000000
              chr3	112000000	113000000
              chr3	113000000	114000000
              chr3	114000000	115000000
              chr3	115000000	116000000
              chr3	116000000	117000000
              chr3	117000000	118000000
              chr3	118000000	119000000
              chr3	119000000	120000000
              chr3	120000000	121000000
              chr3	121000000	122000000
              chr3	122000000	123000000
              chr3	123000000	124000000
              chr3	124000000	125000000
              chr3	125000000	126000000
              chr3	126000000	127000000
              chr3	127000000	128000000
              chr3	128000000	129000000
              chr3	129000000	130000000
              chr3	130000000	131000000
              chr3	131000000	132000000
              chr3	132000000	133000000
              chr3	133000000	134000000
              chr3	134000000	135000000
              chr3	135000000	136000000
              chr3	136000000	137000000
              chr3	137000000	138000000
              chr3	138000000	139000000
              chr3	139000000	140000000
              chr3	140000000	141000000
              chr3	141000000	142000000
              chr3	142000000	143000000
              chr3	143000000	144000000
              chr3	144000000	145000000
              chr3	145000000	146000000
              chr3	146000000	147000000
              chr3	147000000	148000000
              chr3	148000000	149000000
              chr3	149000000	150000000
              chr3	150000000	151000000
              chr3	151000000	152000000
              chr3	152000000	153000000
              chr3	153000000	154000000
              chr3	154000000	155000000
              chr3	155000000	156000000
              chr3	156000000	157000000
              chr3	157000000	158000000
              chr3	158000000	159000000
              chr3	159000000	160000000
              chr3	160000000	161000000
              chr3	161000000	162000000
              chr3	162000000	163000000
              chr3	163000000	164000000
              chr3	164000000	165000000
              chr3	165000000	166000000
              chr3	166000000	167000000
              chr3	167000000	168000000
              chr3	168000000	169000000
              chr3	169000000	170000000
              chr3	170000000	171000000
              chr3	171000000	172000000
              chr3	172000000	173000000
              chr3	173000000	174000000
              chr3	174000000	175000000
              chr3	175000000	176000000
              chr3	176000000	177000000
              chr3	177000000	178000000
              chr3	178000000	179000000
              chr3	179000000	180000000
              chr3	180000000	181000000
              chr3	181000000	182000000
              chr3	182000000	183000000
              chr3	183000000	184000000
              chr3	184000000	185000000
              chr3	185000000	186000000
              chr3	186000000	187000000
              chr3	187000000	188000000
              chr3	188000000	189000000
              chr3	189000000	190000000
              chr3	190000000	191000000
              chr3	191000000	192000000
              chr3	192000000	193000000
              chr3	193000000	194000000
              chr3	194000000	195000000
              chr3	195000000	196000000
              chr3	196000000	197000000
              chr3	197000000	198000000
              chr3	198000000	198295559
              chr3_GL000221v1_random	0	155397
              chr3_GL383526v1_alt	0	180671
              chr3_JH636055v2_alt	0	173151
              chr3_KI270777v1_alt	0	173649
              chr3_KI270778v1_alt	0	248252
              chr3_KI270779v1_alt	0	205312
              chr3_KI270780v1_alt	0	224108
              chr3_KI270781v1_alt	0	113034
              chr3_KI270782v1_alt	0	162429
              chr3_KI270783v1_alt	0	109187
              chr3_KI270784v1_alt	0	184404
              chr3_KI270895v1_alt	0	162896
              chr3_KI270924v1_alt	0	166540
              chr3_KI270934v1_alt	0	163458
              chr3_KI270935v1_alt	0	197351
              chr3_KI270936v1_alt	0	164170
              chr3_KI270937v1_alt	0	165607
              chr4	0	1000000
              chr4	1000000	2000000
              chr4	2000000	3000000
              chr4	3000000	4000000
              chr4	4000000	5000000
              chr4	5000000	6000000
              chr4	6000000	7000000
              chr4	7000000	8000000
              chr4	8000000	9000000
              chr4	9000000	10000000
              chr4	10000000	11000000
              chr4	11000000	12000000
              chr4	12000000	13000000
              chr4	13000000	14000000
              chr4	14000000	15000000
              chr4	15000000	16000000
              chr4	16000000	17000000
              chr4	17000000	18000000
              chr4	18000000	19000000
              chr4	19000000	20000000
              chr4	20000000	21000000
              chr4	21000000	22000000
              chr4	22000000	23000000
              chr4	23000000	24000000
              chr4	24000000	25000000
              chr4	25000000	26000000
              chr4	26000000	27000000
              chr4	27000000	28000000
              chr4	28000000	29000000
              chr4	29000000	30000000
              chr4	30000000	31000000
              chr4	31000000	32000000
              chr4	32000000	33000000
              chr4	33000000	34000000
              chr4	34000000	35000000
              chr4	35000000	36000000
              chr4	36000000	37000000
              chr4	37000000	38000000
              chr4	38000000	39000000
              chr4	39000000	40000000
              chr4	40000000	41000000
              chr4	41000000	42000000
              chr4	42000000	43000000
              chr4	43000000	44000000
              chr4	44000000	45000000
              chr4	45000000	46000000
              chr4	46000000	47000000
              chr4	47000000	48000000
              chr4	48000000	49000000
              chr4	49000000	50000000
              chr4	50000000	51000000
              chr4	51000000	52000000
              chr4	52000000	53000000
              chr4	53000000	54000000
              chr4	54000000	55000000
              chr4	55000000	56000000
              chr4	56000000	57000000
              chr4	57000000	58000000
              chr4	58000000	59000000
              chr4	59000000	60000000
              chr4	60000000	61000000
              chr4	61000000	62000000
              chr4	62000000	63000000
              chr4	63000000	64000000
              chr4	64000000	65000000
              chr4	65000000	66000000
              chr4	66000000	67000000
              chr4	67000000	68000000
              chr4	68000000	69000000
              chr4	69000000	70000000
              chr4	70000000	71000000
              chr4	71000000	72000000
              chr4	72000000	73000000
              chr4	73000000	74000000
              chr4	74000000	75000000
              chr4	75000000	76000000
              chr4	76000000	77000000
              chr4	77000000	78000000
              chr4	78000000	79000000
              chr4	79000000	80000000
              chr4	80000000	81000000
              chr4	81000000	82000000
              chr4	82000000	83000000
              chr4	83000000	84000000
              chr4	84000000	85000000
              chr4	85000000	86000000
              chr4	86000000	87000000
              chr4	87000000	88000000
              chr4	88000000	89000000
              chr4	89000000	90000000
              chr4	90000000	91000000
              chr4	91000000	92000000
              chr4	92000000	93000000
              chr4	93000000	94000000
              chr4	94000000	95000000
              chr4	95000000	96000000
              chr4	96000000	97000000
              chr4	97000000	98000000
              chr4	98000000	99000000
              chr4	99000000	100000000
              chr4	100000000	101000000
              chr4	101000000	102000000
              chr4	102000000	103000000
              chr4	103000000	104000000
              chr4	104000000	105000000
              chr4	105000000	106000000
              chr4	106000000	107000000
              chr4	107000000	108000000
              chr4	108000000	109000000
              chr4	109000000	110000000
              chr4	110000000	111000000
              chr4	111000000	112000000
              chr4	112000000	113000000
              chr4	113000000	114000000
              chr4	114000000	115000000
              chr4	115000000	116000000
              chr4	116000000	117000000
              chr4	117000000	118000000
              chr4	118000000	119000000
              chr4	119000000	120000000
              chr4	120000000	121000000
              chr4	121000000	122000000
              chr4	122000000	123000000
              chr4	123000000	124000000
              chr4	124000000	125000000
              chr4	125000000	126000000
              chr4	126000000	127000000
              chr4	127000000	128000000
              chr4	128000000	129000000
              chr4	129000000	130000000
              chr4	130000000	131000000
              chr4	131000000	132000000
              chr4	132000000	133000000
              chr4	133000000	134000000
              chr4	134000000	135000000
              chr4	135000000	136000000
              chr4	136000000	137000000
              chr4	137000000	138000000
              chr4	138000000	139000000
              chr4	139000000	140000000
              chr4	140000000	141000000
              chr4	141000000	142000000
              chr4	142000000	143000000
              chr4	143000000	144000000
              chr4	144000000	145000000
              chr4	145000000	146000000
              chr4	146000000	147000000
              chr4	147000000	148000000
              chr4	148000000	149000000
              chr4	149000000	150000000
              chr4	150000000	151000000
              chr4	151000000	152000000
              chr4	152000000	153000000
              chr4	153000000	154000000
              chr4	154000000	155000000
              chr4	155000000	156000000
              chr4	156000000	157000000
              chr4	157000000	158000000
              chr4	158000000	159000000
              chr4	159000000	160000000
              chr4	160000000	161000000
              chr4	161000000	162000000
              chr4	162000000	163000000
              chr4	163000000	164000000
              chr4	164000000	165000000
              chr4	165000000	166000000
              chr4	166000000	167000000
              chr4	167000000	168000000
              chr4	168000000	169000000
              chr4	169000000	170000000
              chr4	170000000	171000000
              chr4	171000000	172000000
              chr4	172000000	173000000
              chr4	173000000	174000000
              chr4	174000000	175000000
              chr4	175000000	176000000
              chr4	176000000	177000000
              chr4	177000000	178000000
              chr4	178000000	179000000
              chr4	179000000	180000000
              chr4	180000000	181000000
              chr4	181000000	182000000
              chr4	182000000	183000000
              chr4	183000000	184000000
              chr4	184000000	185000000
              chr4	185000000	186000000
              chr4	186000000	187000000
              chr4	187000000	188000000
              chr4	188000000	189000000
              chr4	189000000	190000000
              chr4	190000000	190214555
              chr4_GL000008v2_random	0	209709
              chr4_GL000257v2_alt	0	586476
              chr4_GL383527v1_alt	0	164536
              chr4_GL383528v1_alt	0	376187
              chr4_KI270785v1_alt	0	119912
              chr4_KI270786v1_alt	0	244096
              chr4_KI270787v1_alt	0	111943
              chr4_KI270788v1_alt	0	158965
              chr4_KI270789v1_alt	0	205944
              chr4_KI270790v1_alt	0	220246
              chr4_KI270896v1_alt	0	378547
              chr4_KI270925v1_alt	0	555799
              chr5	0	1000000
              chr5	1000000	2000000
              chr5	2000000	3000000
              chr5	3000000	4000000
              chr5	4000000	5000000
              chr5	5000000	6000000
              chr5	6000000	7000000
              chr5	7000000	8000000
              chr5	8000000	9000000
              chr5	9000000	10000000
              chr5	10000000	11000000
              chr5	11000000	12000000
              chr5	12000000	13000000
              chr5	13000000	14000000
              chr5	14000000	15000000
              chr5	15000000	16000000
              chr5	16000000	17000000
              chr5	17000000	18000000
              chr5	18000000	19000000
              chr5	19000000	20000000
              chr5	20000000	21000000
              chr5	21000000	22000000
              chr5	22000000	23000000
              chr5	23000000	24000000
              chr5	24000000	25000000
              chr5	25000000	26000000
              chr5	26000000	27000000
              chr5	27000000	28000000
              chr5	28000000	29000000
              chr5	29000000	30000000
              chr5	30000000	31000000
              chr5	31000000	32000000
              chr5	32000000	33000000
              chr5	33000000	34000000
              chr5	34000000	35000000
              chr5	35000000	36000000
              chr5	36000000	37000000
              chr5	37000000	38000000
              chr5	38000000	39000000
              chr5	39000000	40000000
              chr5	40000000	41000000
              chr5	41000000	42000000
              chr5	42000000	43000000
              chr5	43000000	44000000
              chr5	44000000	45000000
              chr5	45000000	46000000
              chr5	46000000	47000000
              chr5	47000000	48000000
              chr5	48000000	49000000
              chr5	49000000	50000000
              chr5	50000000	51000000
              chr5	51000000	52000000
              chr5	52000000	53000000
              chr5	53000000	54000000
              chr5	54000000	55000000
              chr5	55000000	56000000
              chr5	56000000	57000000
              chr5	57000000	58000000
              chr5	58000000	59000000
              chr5	59000000	60000000
              chr5	60000000	61000000
              chr5	61000000	62000000
              chr5	62000000	63000000
              chr5	63000000	64000000
              chr5	64000000	65000000
              chr5	65000000	66000000
              chr5	66000000	67000000
              chr5	67000000	68000000
              chr5	68000000	69000000
              chr5	69000000	70000000
              chr5	70000000	71000000
              chr5	71000000	72000000
              chr5	72000000	73000000
              chr5	73000000	74000000
              chr5	74000000	75000000
              chr5	75000000	76000000
              chr5	76000000	77000000
              chr5	77000000	78000000
              chr5	78000000	79000000
              chr5	79000000	80000000
              chr5	80000000	81000000
              chr5	81000000	82000000
              chr5	82000000	83000000
              chr5	83000000	84000000
              chr5	84000000	85000000
              chr5	85000000	86000000
              chr5	86000000	87000000
              chr5	87000000	88000000
              chr5	88000000	89000000
              chr5	89000000	90000000
              chr5	90000000	91000000
              chr5	91000000	92000000
              chr5	92000000	93000000
              chr5	93000000	94000000
              chr5	94000000	95000000
              chr5	95000000	96000000
              chr5	96000000	97000000
              chr5	97000000	98000000
              chr5	98000000	99000000
              chr5	99000000	100000000
              chr5	100000000	101000000
              chr5	101000000	102000000
              chr5	102000000	103000000
              chr5	103000000	104000000
              chr5	104000000	105000000
              chr5	105000000	106000000
              chr5	106000000	107000000
              chr5	107000000	108000000
              chr5	108000000	109000000
              chr5	109000000	110000000
              chr5	110000000	111000000
              chr5	111000000	112000000
              chr5	112000000	113000000
              chr5	113000000	114000000
              chr5	114000000	115000000
              chr5	115000000	116000000
              chr5	116000000	117000000
              chr5	117000000	118000000
              chr5	118000000	119000000
              chr5	119000000	120000000
              chr5	120000000	121000000
              chr5	121000000	122000000
              chr5	122000000	123000000
              chr5	123000000	124000000
              chr5	124000000	125000000
              chr5	125000000	126000000
              chr5	126000000	127000000
              chr5	127000000	128000000
              chr5	128000000	129000000
              chr5	129000000	130000000
              chr5	130000000	131000000
              chr5	131000000	132000000
              chr5	132000000	133000000
              chr5	133000000	134000000
              chr5	134000000	135000000
              chr5	135000000	136000000
              chr5	136000000	137000000
              chr5	137000000	138000000
              chr5	138000000	139000000
              chr5	139000000	140000000
              chr5	140000000	141000000
              chr5	141000000	142000000
              chr5	142000000	143000000
              chr5	143000000	144000000
              chr5	144000000	145000000
              chr5	145000000	146000000
              chr5	146000000	147000000
              chr5	147000000	148000000
              chr5	148000000	149000000
              chr5	149000000	150000000
              chr5	150000000	151000000
              chr5	151000000	152000000
              chr5	152000000	153000000
              chr5	153000000	154000000
              chr5	154000000	155000000
              chr5	155000000	156000000
              chr5	156000000	157000000
              chr5	157000000	158000000
              chr5	158000000	159000000
              chr5	159000000	160000000
              chr5	160000000	161000000
              chr5	161000000	162000000
              chr5	162000000	163000000
              chr5	163000000	164000000
              chr5	164000000	165000000
              chr5	165000000	166000000
              chr5	166000000	167000000
              chr5	167000000	168000000
              chr5	168000000	169000000
              chr5	169000000	170000000
              chr5	170000000	171000000
              chr5	171000000	172000000
              chr5	172000000	173000000
              chr5	173000000	174000000
              chr5	174000000	175000000
              chr5	175000000	176000000
              chr5	176000000	177000000
              chr5	177000000	178000000
              chr5	178000000	179000000
              chr5	179000000	180000000
              chr5	180000000	181000000
              chr5	181000000	181538259
              chr5_GL000208v1_random	0	92689
              chr5_GL339449v2_alt	0	1000000
              chr5_GL339449v2_alt	1000000	1612928
              chr5_GL383530v1_alt	0	101241
              chr5_GL383531v1_alt	0	173459
              chr5_GL383532v1_alt	0	82728
              chr5_GL949742v1_alt	0	226852
              chr5_KI270791v1_alt	0	195710
              chr5_KI270792v1_alt	0	179043
              chr5_KI270793v1_alt	0	126136
              chr5_KI270794v1_alt	0	164558
              chr5_KI270795v1_alt	0	131892
              chr5_KI270796v1_alt	0	172708
              chr5_KI270897v1_alt	0	1000000
              chr5_KI270897v1_alt	1000000	1144418
              chr5_KI270898v1_alt	0	130957
              chr6	0	1000000
              chr6	1000000	2000000
              chr6	2000000	3000000
              chr6	3000000	4000000
              chr6	4000000	5000000
              chr6	5000000	6000000
              chr6	6000000	7000000
              chr6	7000000	8000000
              chr6	8000000	9000000
              chr6	9000000	10000000
              chr6	10000000	11000000
              chr6	11000000	12000000
              chr6	12000000	13000000
              chr6	13000000	14000000
              chr6	14000000	15000000
              chr6	15000000	16000000
              chr6	16000000	17000000
              chr6	17000000	18000000
              chr6	18000000	19000000
              chr6	19000000	20000000
              chr6	20000000	21000000
              chr6	21000000	22000000
              chr6	22000000	23000000
              chr6	23000000	24000000
              chr6	24000000	25000000
              chr6	25000000	26000000
              chr6	26000000	27000000
              chr6	27000000	28000000
              chr6	28000000	29000000
              chr6	29000000	30000000
              chr6	30000000	31000000
              chr6	31000000	32000000
              chr6	32000000	33000000
              chr6	33000000	34000000
              chr6	34000000	35000000
              chr6	35000000	36000000
              chr6	36000000	37000000
              chr6	37000000	38000000
              chr6	38000000	39000000
              chr6	39000000	40000000
              chr6	40000000	41000000
              chr6	41000000	42000000
              chr6	42000000	43000000
              chr6	43000000	44000000
              chr6	44000000	45000000
              chr6	45000000	46000000
              chr6	46000000	47000000
              chr6	47000000	48000000
              chr6	48000000	49000000
              chr6	49000000	50000000
              chr6	50000000	51000000
              chr6	51000000	52000000
              chr6	52000000	53000000
              chr6	53000000	54000000
              chr6	54000000	55000000
              chr6	55000000	56000000
              chr6	56000000	57000000
              chr6	57000000	58000000
              chr6	58000000	59000000
              chr6	59000000	60000000
              chr6	60000000	61000000
              chr6	61000000	62000000
              chr6	62000000	63000000
              chr6	63000000	64000000
              chr6	64000000	65000000
              chr6	65000000	66000000
              chr6	66000000	67000000
              chr6	67000000	68000000
              chr6	68000000	69000000
              chr6	69000000	70000000
              chr6	70000000	71000000
              chr6	71000000	72000000
              chr6	72000000	73000000
              chr6	73000000	74000000
              chr6	74000000	75000000
              chr6	75000000	76000000
              chr6	76000000	77000000
              chr6	77000000	78000000
              chr6	78000000	79000000
              chr6	79000000	80000000
              chr6	80000000	81000000
              chr6	81000000	82000000
              chr6	82000000	83000000
              chr6	83000000	84000000
              chr6	84000000	85000000
              chr6	85000000	86000000
              chr6	86000000	87000000
              chr6	87000000	88000000
              chr6	88000000	89000000
              chr6	89000000	90000000
              chr6	90000000	91000000
              chr6	91000000	92000000
              chr6	92000000	93000000
              chr6	93000000	94000000
              chr6	94000000	95000000
              chr6	95000000	96000000
              chr6	96000000	97000000
              chr6	97000000	98000000
              chr6	98000000	99000000
              chr6	99000000	100000000
              chr6	100000000	101000000
              chr6	101000000	102000000
              chr6	102000000	103000000
              chr6	103000000	104000000
              chr6	104000000	105000000
              chr6	105000000	106000000
              chr6	106000000	107000000
              chr6	107000000	108000000
              chr6	108000000	109000000
              chr6	109000000	110000000
              chr6	110000000	111000000
              chr6	111000000	112000000
              chr6	112000000	113000000
              chr6	113000000	114000000
              chr6	114000000	115000000
              chr6	115000000	116000000
              chr6	116000000	117000000
              chr6	117000000	118000000
              chr6	118000000	119000000
              chr6	119000000	120000000
              chr6	120000000	121000000
              chr6	121000000	122000000
              chr6	122000000	123000000
              chr6	123000000	124000000
              chr6	124000000	125000000
              chr6	125000000	126000000
              chr6	126000000	127000000
              chr6	127000000	128000000
              chr6	128000000	129000000
              chr6	129000000	130000000
              chr6	130000000	131000000
              chr6	131000000	132000000
              chr6	132000000	133000000
              chr6	133000000	134000000
              chr6	134000000	135000000
              chr6	135000000	136000000
              chr6	136000000	137000000
              chr6	137000000	138000000
              chr6	138000000	139000000
              chr6	139000000	140000000
              chr6	140000000	141000000
              chr6	141000000	142000000
              chr6	142000000	143000000
              chr6	143000000	144000000
              chr6	144000000	145000000
              chr6	145000000	146000000
              chr6	146000000	147000000
              chr6	147000000	148000000
              chr6	148000000	149000000
              chr6	149000000	150000000
              chr6	150000000	151000000
              chr6	151000000	152000000
              chr6	152000000	153000000
              chr6	153000000	154000000
              chr6	154000000	155000000
              chr6	155000000	156000000
              chr6	156000000	157000000
              chr6	157000000	158000000
              chr6	158000000	159000000
              chr6	159000000	160000000
              chr6	160000000	161000000
              chr6	161000000	162000000
              chr6	162000000	163000000
              chr6	163000000	164000000
              chr6	164000000	165000000
              chr6	165000000	166000000
              chr6	166000000	167000000
              chr6	167000000	168000000
              chr6	168000000	169000000
              chr6	169000000	170000000
              chr6	170000000	170805979
              chr6_GL000250v2_alt	0	1000000
              chr6_GL000250v2_alt	1000000	2000000
              chr6_GL000250v2_alt	2000000	3000000
              chr6_GL000250v2_alt	3000000	4000000
              chr6_GL000250v2_alt	4000000	4672374
              chr6_GL000251v2_alt	0	1000000
              chr6_GL000251v2_alt	1000000	2000000
              chr6_GL000251v2_alt	2000000	3000000
              chr6_GL000251v2_alt	3000000	4000000
              chr6_GL000251v2_alt	4000000	4795265
              chr6_GL000252v2_alt	0	1000000
              chr6_GL000252v2_alt	1000000	2000000
              chr6_GL000252v2_alt	2000000	3000000
              chr6_GL000252v2_alt	3000000	4000000
              chr6_GL000252v2_alt	4000000	4604811
              chr6_GL000253v2_alt	0	1000000
              chr6_GL000253v2_alt	1000000	2000000
              chr6_GL000253v2_alt	2000000	3000000
              chr6_GL000253v2_alt	3000000	4000000
              chr6_GL000253v2_alt	4000000	4677643
              chr6_GL000254v2_alt	0	1000000
              chr6_GL000254v2_alt	1000000	2000000
              chr6_GL000254v2_alt	2000000	3000000
              chr6_GL000254v2_alt	3000000	4000000
              chr6_GL000254v2_alt	4000000	4827813
              chr6_GL000255v2_alt	0	1000000
              chr6_GL000255v2_alt	1000000	2000000
              chr6_GL000255v2_alt	2000000	3000000
              chr6_GL000255v2_alt	3000000	4000000
              chr6_GL000255v2_alt	4000000	4606388
              chr6_GL000256v2_alt	0	1000000
              chr6_GL000256v2_alt	1000000	2000000
              chr6_GL000256v2_alt	2000000	3000000
              chr6_GL000256v2_alt	3000000	4000000
              chr6_GL000256v2_alt	4000000	4929269
              chr6_GL383533v1_alt	0	124736
              chr6_KB021644v2_alt	0	185823
              chr6_KI270758v1_alt	0	76752
              chr6_KI270797v1_alt	0	197536
              chr6_KI270798v1_alt	0	271782
              chr6_KI270799v1_alt	0	152148
              chr6_KI270800v1_alt	0	175808
              chr6_KI270801v1_alt	0	870480
              chr6_KI270802v1_alt	0	75005
              chr7	0	1000000
              chr7	1000000	2000000
              chr7	2000000	3000000
              chr7	3000000	4000000
              chr7	4000000	5000000
              chr7	5000000	6000000
              chr7	6000000	7000000
              chr7	7000000	8000000
              chr7	8000000	9000000
              chr7	9000000	10000000
              chr7	10000000	11000000
              chr7	11000000	12000000
              chr7	12000000	13000000
              chr7	13000000	14000000
              chr7	14000000	15000000
              chr7	15000000	16000000
              chr7	16000000	17000000
              chr7	17000000	18000000
              chr7	18000000	19000000
              chr7	19000000	20000000
              chr7	20000000	21000000
              chr7	21000000	22000000
              chr7	22000000	23000000
              chr7	23000000	24000000
              chr7	24000000	25000000
              chr7	25000000	26000000
              chr7	26000000	27000000
              chr7	27000000	28000000
              chr7	28000000	29000000
              chr7	29000000	30000000
              chr7	30000000	31000000
              chr7	31000000	32000000
              chr7	32000000	33000000
              chr7	33000000	34000000
              chr7	34000000	35000000
              chr7	35000000	36000000
              chr7	36000000	37000000
              chr7	37000000	38000000
              chr7	38000000	39000000
              chr7	39000000	40000000
              chr7	40000000	41000000
              chr7	41000000	42000000
              chr7	42000000	43000000
              chr7	43000000	44000000
              chr7	44000000	45000000
              chr7	45000000	46000000
              chr7	46000000	47000000
              chr7	47000000	48000000
              chr7	48000000	49000000
              chr7	49000000	50000000
              chr7	50000000	51000000
              chr7	51000000	52000000
              chr7	52000000	53000000
              chr7	53000000	54000000
              chr7	54000000	55000000
              chr7	55000000	56000000
              chr7	56000000	57000000
              chr7	57000000	58000000
              chr7	58000000	59000000
              chr7	59000000	60000000
              chr7	60000000	61000000
              chr7	61000000	62000000
              chr7	62000000	63000000
              chr7	63000000	64000000
              chr7	64000000	65000000
              chr7	65000000	66000000
              chr7	66000000	67000000
              chr7	67000000	68000000
              chr7	68000000	69000000
              chr7	69000000	70000000
              chr7	70000000	71000000
              chr7	71000000	72000000
              chr7	72000000	73000000
              chr7	73000000	74000000
              chr7	74000000	75000000
              chr7	75000000	76000000
              chr7	76000000	77000000
              chr7	77000000	78000000
              chr7	78000000	79000000
              chr7	79000000	80000000
              chr7	80000000	81000000
              chr7	81000000	82000000
              chr7	82000000	83000000
              chr7	83000000	84000000
              chr7	84000000	85000000
              chr7	85000000	86000000
              chr7	86000000	87000000
              chr7	87000000	88000000
              chr7	88000000	89000000
              chr7	89000000	90000000
              chr7	90000000	91000000
              chr7	91000000	92000000
              chr7	92000000	93000000
              chr7	93000000	94000000
              chr7	94000000	95000000
              chr7	95000000	96000000
              chr7	96000000	97000000
              chr7	97000000	98000000
              chr7	98000000	99000000
              chr7	99000000	100000000
              chr7	100000000	101000000
              chr7	101000000	102000000
              chr7	102000000	103000000
              chr7	103000000	104000000
              chr7	104000000	105000000
              chr7	105000000	106000000
              chr7	106000000	107000000
              chr7	107000000	108000000
              chr7	108000000	109000000
              chr7	109000000	110000000
              chr7	110000000	111000000
              chr7	111000000	112000000
              chr7	112000000	113000000
              chr7	113000000	114000000
              chr7	114000000	115000000
              chr7	115000000	116000000
              chr7	116000000	117000000
              chr7	117000000	118000000
              chr7	118000000	119000000
              chr7	119000000	120000000
              chr7	120000000	121000000
              chr7	121000000	122000000
              chr7	122000000	123000000
              chr7	123000000	124000000
              chr7	124000000	125000000
              chr7	125000000	126000000
              chr7	126000000	127000000
              chr7	127000000	128000000
              chr7	128000000	129000000
              chr7	129000000	130000000
              chr7	130000000	131000000
              chr7	131000000	132000000
              chr7	132000000	133000000
              chr7	133000000	134000000
              chr7	134000000	135000000
              chr7	135000000	136000000
              chr7	136000000	137000000
              chr7	137000000	138000000
              chr7	138000000	139000000
              chr7	139000000	140000000
              chr7	140000000	141000000
              chr7	141000000	142000000
              chr7	142000000	143000000
              chr7	143000000	144000000
              chr7	144000000	145000000
              chr7	145000000	146000000
              chr7	146000000	147000000
              chr7	147000000	148000000
              chr7	148000000	149000000
              chr7	149000000	150000000
              chr7	150000000	151000000
              chr7	151000000	152000000
              chr7	152000000	153000000
              chr7	153000000	154000000
              chr7	154000000	155000000
              chr7	155000000	156000000
              chr7	156000000	157000000
              chr7	157000000	158000000
              chr7	158000000	159000000
              chr7	159000000	159345973
              chr7_GL383534v2_alt	0	119183
              chr7_KI270803v1_alt	0	1000000
              chr7_KI270803v1_alt	1000000	1111570
              chr7_KI270804v1_alt	0	157952
              chr7_KI270805v1_alt	0	209988
              chr7_KI270806v1_alt	0	158166
              chr7_KI270807v1_alt	0	126434
              chr7_KI270808v1_alt	0	271455
              chr7_KI270809v1_alt	0	209586
              chr7_KI270899v1_alt	0	190869
              chr8	0	1000000
              chr8	1000000	2000000
              chr8	2000000	3000000
              chr8	3000000	4000000
              chr8	4000000	5000000
              chr8	5000000	6000000
              chr8	6000000	7000000
              chr8	7000000	8000000
              chr8	8000000	9000000
              chr8	9000000	10000000
              chr8	10000000	11000000
              chr8	11000000	12000000
              chr8	12000000	13000000
              chr8	13000000	14000000
              chr8	14000000	15000000
              chr8	15000000	16000000
              chr8	16000000	17000000
              chr8	17000000	18000000
              chr8	18000000	19000000
              chr8	19000000	20000000
              chr8	20000000	21000000
              chr8	21000000	22000000
              chr8	22000000	23000000
              chr8	23000000	24000000
              chr8	24000000	25000000
              chr8	25000000	26000000
              chr8	26000000	27000000
              chr8	27000000	28000000
              chr8	28000000	29000000
              chr8	29000000	30000000
              chr8	30000000	31000000
              chr8	31000000	32000000
              chr8	32000000	33000000
              chr8	33000000	34000000
              chr8	34000000	35000000
              chr8	35000000	36000000
              chr8	36000000	37000000
              chr8	37000000	38000000
              chr8	38000000	39000000
              chr8	39000000	40000000
              chr8	40000000	41000000
              chr8	41000000	42000000
              chr8	42000000	43000000
              chr8	43000000	44000000
              chr8	44000000	45000000
              chr8	45000000	46000000
              chr8	46000000	47000000
              chr8	47000000	48000000
              chr8	48000000	49000000
              chr8	49000000	50000000
              chr8	50000000	51000000
              chr8	51000000	52000000
              chr8	52000000	53000000
              chr8	53000000	54000000
              chr8	54000000	55000000
              chr8	55000000	56000000
              chr8	56000000	57000000
              chr8	57000000	58000000
              chr8	58000000	59000000
              chr8	59000000	60000000
              chr8	60000000	61000000
              chr8	61000000	62000000
              chr8	62000000	63000000
              chr8	63000000	64000000
              chr8	64000000	65000000
              chr8	65000000	66000000
              chr8	66000000	67000000
              chr8	67000000	68000000
              chr8	68000000	69000000
              chr8	69000000	70000000
              chr8	70000000	71000000
              chr8	71000000	72000000
              chr8	72000000	73000000
              chr8	73000000	74000000
              chr8	74000000	75000000
              chr8	75000000	76000000
              chr8	76000000	77000000
              chr8	77000000	78000000
              chr8	78000000	79000000
              chr8	79000000	80000000
              chr8	80000000	81000000
              chr8	81000000	82000000
              chr8	82000000	83000000
              chr8	83000000	84000000
              chr8	84000000	85000000
              chr8	85000000	86000000
              chr8	86000000	87000000
              chr8	87000000	88000000
              chr8	88000000	89000000
              chr8	89000000	90000000
              chr8	90000000	91000000
              chr8	91000000	92000000
              chr8	92000000	93000000
              chr8	93000000	94000000
              chr8	94000000	95000000
              chr8	95000000	96000000
              chr8	96000000	97000000
              chr8	97000000	98000000
              chr8	98000000	99000000
              chr8	99000000	100000000
              chr8	100000000	101000000
              chr8	101000000	102000000
              chr8	102000000	103000000
              chr8	103000000	104000000
              chr8	104000000	105000000
              chr8	105000000	106000000
              chr8	106000000	107000000
              chr8	107000000	108000000
              chr8	108000000	109000000
              chr8	109000000	110000000
              chr8	110000000	111000000
              chr8	111000000	112000000
              chr8	112000000	113000000
              chr8	113000000	114000000
              chr8	114000000	115000000
              chr8	115000000	116000000
              chr8	116000000	117000000
              chr8	117000000	118000000
              chr8	118000000	119000000
              chr8	119000000	120000000
              chr8	120000000	121000000
              chr8	121000000	122000000
              chr8	122000000	123000000
              chr8	123000000	124000000
              chr8	124000000	125000000
              chr8	125000000	126000000
              chr8	126000000	127000000
              chr8	127000000	128000000
              chr8	128000000	129000000
              chr8	129000000	130000000
              chr8	130000000	131000000
              chr8	131000000	132000000
              chr8	132000000	133000000
              chr8	133000000	134000000
              chr8	134000000	135000000
              chr8	135000000	136000000
              chr8	136000000	137000000
              chr8	137000000	138000000
              chr8	138000000	139000000
              chr8	139000000	140000000
              chr8	140000000	141000000
              chr8	141000000	142000000
              chr8	142000000	143000000
              chr8	143000000	144000000
              chr8	144000000	145000000
              chr8	145000000	145138636
              chr8_KI270810v1_alt	0	374415
              chr8_KI270811v1_alt	0	292436
              chr8_KI270812v1_alt	0	282736
              chr8_KI270813v1_alt	0	300230
              chr8_KI270814v1_alt	0	141812
              chr8_KI270815v1_alt	0	132244
              chr8_KI270816v1_alt	0	305841
              chr8_KI270817v1_alt	0	158983
              chr8_KI270818v1_alt	0	145606
              chr8_KI270819v1_alt	0	133535
              chr8_KI270820v1_alt	0	36640
              chr8_KI270821v1_alt	0	985506
              chr8_KI270822v1_alt	0	624492
              chr8_KI270900v1_alt	0	318687
              chr8_KI270901v1_alt	0	136959
              chr8_KI270926v1_alt	0	229282
              chr9	0	1000000
              chr9	1000000	2000000
              chr9	2000000	3000000
              chr9	3000000	4000000
              chr9	4000000	5000000
              chr9	5000000	6000000
              chr9	6000000	7000000
              chr9	7000000	8000000
              chr9	8000000	9000000
              chr9	9000000	10000000
              chr9	10000000	11000000
              chr9	11000000	12000000
              chr9	12000000	13000000
              chr9	13000000	14000000
              chr9	14000000	15000000
              chr9	15000000	16000000
              chr9	16000000	17000000
              chr9	17000000	18000000
              chr9	18000000	19000000
              chr9	19000000	20000000
              chr9	20000000	21000000
              chr9	21000000	22000000
              chr9	22000000	23000000
              chr9	23000000	24000000
              chr9	24000000	25000000
              chr9	25000000	26000000
              chr9	26000000	27000000
              chr9	27000000	28000000
              chr9	28000000	29000000
              chr9	29000000	30000000
              chr9	30000000	31000000
              chr9	31000000	32000000
              chr9	32000000	33000000
              chr9	33000000	34000000
              chr9	34000000	35000000
              chr9	35000000	36000000
              chr9	36000000	37000000
              chr9	37000000	38000000
              chr9	38000000	39000000
              chr9	39000000	40000000
              chr9	40000000	41000000
              chr9	41000000	42000000
              chr9	42000000	43000000
              chr9	43000000	44000000
              chr9	44000000	45000000
              chr9	45000000	46000000
              chr9	46000000	47000000
              chr9	47000000	48000000
              chr9	48000000	49000000
              chr9	49000000	50000000
              chr9	50000000	51000000
              chr9	51000000	52000000
              chr9	52000000	53000000
              chr9	53000000	54000000
              chr9	54000000	55000000
              chr9	55000000	56000000
              chr9	56000000	57000000
              chr9	57000000	58000000
              chr9	58000000	59000000
              chr9	59000000	60000000
              chr9	60000000	61000000
              chr9	61000000	62000000
              chr9	62000000	63000000
              chr9	63000000	64000000
              chr9	64000000	65000000
              chr9	65000000	66000000
              chr9	66000000	67000000
              chr9	67000000	68000000
              chr9	68000000	69000000
              chr9	69000000	70000000
              chr9	70000000	71000000
              chr9	71000000	72000000
              chr9	72000000	73000000
              chr9	73000000	74000000
              chr9	74000000	75000000
              chr9	75000000	76000000
              chr9	76000000	77000000
              chr9	77000000	78000000
              chr9	78000000	79000000
              chr9	79000000	80000000
              chr9	80000000	81000000
              chr9	81000000	82000000
              chr9	82000000	83000000
              chr9	83000000	84000000
              chr9	84000000	85000000
              chr9	85000000	86000000
              chr9	86000000	87000000
              chr9	87000000	88000000
              chr9	88000000	89000000
              chr9	89000000	90000000
              chr9	90000000	91000000
              chr9	91000000	92000000
              chr9	92000000	93000000
              chr9	93000000	94000000
              chr9	94000000	95000000
              chr9	95000000	96000000
              chr9	96000000	97000000
              chr9	97000000	98000000
              chr9	98000000	99000000
              chr9	99000000	100000000
              chr9	100000000	101000000
              chr9	101000000	102000000
              chr9	102000000	103000000
              chr9	103000000	104000000
              chr9	104000000	105000000
              chr9	105000000	106000000
              chr9	106000000	107000000
              chr9	107000000	108000000
              chr9	108000000	109000000
              chr9	109000000	110000000
              chr9	110000000	111000000
              chr9	111000000	112000000
              chr9	112000000	113000000
              chr9	113000000	114000000
              chr9	114000000	115000000
              chr9	115000000	116000000
              chr9	116000000	117000000
              chr9	117000000	118000000
              chr9	118000000	119000000
              chr9	119000000	120000000
              chr9	120000000	121000000
              chr9	121000000	122000000
              chr9	122000000	123000000
              chr9	123000000	124000000
              chr9	124000000	125000000
              chr9	125000000	126000000
              chr9	126000000	127000000
              chr9	127000000	128000000
              chr9	128000000	129000000
              chr9	129000000	130000000
              chr9	130000000	131000000
              chr9	131000000	132000000
              chr9	132000000	133000000
              chr9	133000000	134000000
              chr9	134000000	135000000
              chr9	135000000	136000000
              chr9	136000000	137000000
              chr9	137000000	138000000
              chr9	138000000	138394717
              chr9_GL383539v1_alt	0	162988
              chr9_GL383540v1_alt	0	71551
              chr9_GL383541v1_alt	0	171286
              chr9_GL383542v1_alt	0	60032
              chr9_KI270717v1_random	0	40062
              chr9_KI270718v1_random	0	38054
              chr9_KI270719v1_random	0	176845
              chr9_KI270720v1_random	0	39050
              chr9_KI270823v1_alt	0	439082
              chrM	0	16569
              chrUn_GL000195v1	0	182896
              chrUn_GL000213v1	0	164239
              chrUn_GL000214v1	0	137718
              chrUn_GL000216v2	0	176608
              chrUn_GL000218v1	0	161147
              chrUn_GL000219v1	0	179198
              chrUn_GL000220v1	0	161802
              chrUn_GL000224v1	0	179693
              chrUn_GL000226v1	0	15008
              chrUn_KI270302v1	0	2274
              chrUn_KI270303v1	0	1942
              chrUn_KI270304v1	0	2165
              chrUn_KI270305v1	0	1472
              chrUn_KI270310v1	0	1201
              chrUn_KI270311v1	0	12399
              chrUn_KI270312v1	0	998
              chrUn_KI270315v1	0	2276
              chrUn_KI270316v1	0	1444
              chrUn_KI270317v1	0	37690
              chrUn_KI270320v1	0	4416
              chrUn_KI270322v1	0	21476
              chrUn_KI270329v1	0	1040
              chrUn_KI270330v1	0	1652
              chrUn_KI270333v1	0	2699
              chrUn_KI270334v1	0	1368
              chrUn_KI270335v1	0	1048
              chrUn_KI270336v1	0	1026
              chrUn_KI270337v1	0	1121
              chrUn_KI270338v1	0	1428
              chrUn_KI270340v1	0	1428
              chrUn_KI270362v1	0	3530
              chrUn_KI270363v1	0	1803
              chrUn_KI270364v1	0	2855
              chrUn_KI270366v1	0	8320
              chrUn_KI270371v1	0	2805
              chrUn_KI270372v1	0	1650
              chrUn_KI270373v1	0	1451
              chrUn_KI270374v1	0	2656
              chrUn_KI270375v1	0	2378
              chrUn_KI270376v1	0	1136
              chrUn_KI270378v1	0	1048
              chrUn_KI270379v1	0	1045
              chrUn_KI270381v1	0	1930
              chrUn_KI270382v1	0	4215
              chrUn_KI270383v1	0	1750
              chrUn_KI270384v1	0	1658
              chrUn_KI270385v1	0	990
              chrUn_KI270386v1	0	1788
              chrUn_KI270387v1	0	1537
              chrUn_KI270388v1	0	1216
              chrUn_KI270389v1	0	1298
              chrUn_KI270390v1	0	2387
              chrUn_KI270391v1	0	1484
              chrUn_KI270392v1	0	971
              chrUn_KI270393v1	0	1308
              chrUn_KI270394v1	0	970
              chrUn_KI270395v1	0	1143
              chrUn_KI270396v1	0	1880
              chrUn_KI270411v1	0	2646
              chrUn_KI270412v1	0	1179
              chrUn_KI270414v1	0	2489
              chrUn_KI270417v1	0	2043
              chrUn_KI270418v1	0	2145
              chrUn_KI270419v1	0	1029
              chrUn_KI270420v1	0	2321
              chrUn_KI270422v1	0	1445
              chrUn_KI270423v1	0	981
              chrUn_KI270424v1	0	2140
              chrUn_KI270425v1	0	1884
              chrUn_KI270429v1	0	1361
              chrUn_KI270435v1	0	92983
              chrUn_KI270438v1	0	112505
              chrUn_KI270442v1	0	392061
              chrUn_KI270448v1	0	7992
              chrUn_KI270465v1	0	1774
              chrUn_KI270466v1	0	1233
              chrUn_KI270467v1	0	3920
              chrUn_KI270468v1	0	4055
              chrUn_KI270507v1	0	5353
              chrUn_KI270508v1	0	1951
              chrUn_KI270509v1	0	2318
              chrUn_KI270510v1	0	2415
              chrUn_KI270511v1	0	8127
              chrUn_KI270512v1	0	22689
              chrUn_KI270515v1	0	6361
              chrUn_KI270516v1	0	1300
              chrUn_KI270517v1	0	3253
              chrUn_KI270518v1	0	2186
              chrUn_KI270519v1	0	138126
              chrUn_KI270521v1	0	7642
              chrUn_KI270522v1	0	5674
              chrUn_KI270528v1	0	2983
              chrUn_KI270529v1	0	1899
              chrUn_KI270530v1	0	2168
              chrUn_KI270538v1	0	91309
              chrUn_KI270539v1	0	993
              chrUn_KI270544v1	0	1202
              chrUn_KI270548v1	0	1599
              chrUn_KI270579v1	0	31033
              chrUn_KI270580v1	0	1553
              chrUn_KI270581v1	0	7046
              chrUn_KI270582v1	0	6504
              chrUn_KI270583v1	0	1400
              chrUn_KI270584v1	0	4513
              chrUn_KI270587v1	0	2969
              chrUn_KI270588v1	0	6158
              chrUn_KI270589v1	0	44474
              chrUn_KI270590v1	0	4685
              chrUn_KI270591v1	0	5796
              chrUn_KI270593v1	0	3041
              chrUn_KI270741v1	0	157432
              chrUn_KI270742v1	0	186739
              chrUn_KI270743v1	0	210658
              chrUn_KI270744v1	0	168472
              chrUn_KI270745v1	0	41891
              chrUn_KI270746v1	0	66486
              chrUn_KI270747v1	0	198735
              chrUn_KI270748v1	0	93321
              chrUn_KI270749v1	0	158759
              chrUn_KI270750v1	0	148850
              chrUn_KI270751v1	0	150742
              chrUn_KI270752v1	0	27745
              chrUn_KI270753v1	0	62944
              chrUn_KI270754v1	0	40191
              chrUn_KI270755v1	0	36723
              chrUn_KI270756v1	0	79590
              chrUn_KI270757v1	0	71251
              chrX	0	1000000
              chrX	1000000	2000000
              chrX	2000000	3000000
              chrX	3000000	4000000
              chrX	4000000	5000000
              chrX	5000000	6000000
              chrX	6000000	7000000
              chrX	7000000	8000000
              chrX	8000000	9000000
              chrX	9000000	10000000
              chrX	10000000	11000000
              chrX	11000000	12000000
              chrX	12000000	13000000
              chrX	13000000	14000000
              chrX	14000000	15000000
              chrX	15000000	16000000
              chrX	16000000	17000000
              chrX	17000000	18000000
              chrX	18000000	19000000
              chrX	19000000	20000000
              chrX	20000000	21000000
              chrX	21000000	22000000
              chrX	22000000	23000000
              chrX	23000000	24000000
              chrX	24000000	25000000
              chrX	25000000	26000000
              chrX	26000000	27000000
              chrX	27000000	28000000
              chrX	28000000	29000000
              chrX	29000000	30000000
              chrX	30000000	31000000
              chrX	31000000	32000000
              chrX	32000000	33000000
              chrX	33000000	34000000
              chrX	34000000	35000000
              chrX	35000000	36000000
              chrX	36000000	37000000
              chrX	37000000	38000000
              chrX	38000000	39000000
              chrX	39000000	40000000
              chrX	40000000	41000000
              chrX	41000000	42000000
              chrX	42000000	43000000
              chrX	43000000	44000000
              chrX	44000000	45000000
              chrX	45000000	46000000
              chrX	46000000	47000000
              chrX	47000000	48000000
              chrX	48000000	49000000
              chrX	49000000	50000000
              chrX	50000000	51000000
              chrX	51000000	52000000
              chrX	52000000	53000000
              chrX	53000000	54000000
              chrX	54000000	55000000
              chrX	55000000	56000000
              chrX	56000000	57000000
              chrX	57000000	58000000
              chrX	58000000	59000000
              chrX	59000000	60000000
              chrX	60000000	61000000
              chrX	61000000	62000000
              chrX	62000000	63000000
              chrX	63000000	64000000
              chrX	64000000	65000000
              chrX	65000000	66000000
              chrX	66000000	67000000
              chrX	67000000	68000000
              chrX	68000000	69000000
              chrX	69000000	70000000
              chrX	70000000	71000000
              chrX	71000000	72000000
              chrX	72000000	73000000
              chrX	73000000	74000000
              chrX	74000000	75000000
              chrX	75000000	76000000
              chrX	76000000	77000000
              chrX	77000000	78000000
              chrX	78000000	79000000
              chrX	79000000	80000000
              chrX	80000000	81000000
              chrX	81000000	82000000
              chrX	82000000	83000000
              chrX	83000000	84000000
              chrX	84000000	85000000
              chrX	85000000	86000000
              chrX	86000000	87000000
              chrX	87000000	88000000
              chrX	88000000	89000000
              chrX	89000000	90000000
              chrX	90000000	91000000
              chrX	91000000	92000000
              chrX	92000000	93000000
              chrX	93000000	94000000
              chrX	94000000	95000000
              chrX	95000000	96000000
              chrX	96000000	97000000
              chrX	97000000	98000000
              chrX	98000000	99000000
              chrX	99000000	100000000
              chrX	100000000	101000000
              chrX	101000000	102000000
              chrX	102000000	103000000
              chrX	103000000	104000000
              chrX	104000000	105000000
              chrX	105000000	106000000
              chrX	106000000	107000000
              chrX	107000000	108000000
              chrX	108000000	109000000
              chrX	109000000	110000000
              chrX	110000000	111000000
              chrX	111000000	112000000
              chrX	112000000	113000000
              chrX	113000000	114000000
              chrX	114000000	115000000
              chrX	115000000	116000000
              chrX	116000000	117000000
              chrX	117000000	118000000
              chrX	118000000	119000000
              chrX	119000000	120000000
              chrX	120000000	121000000
              chrX	121000000	122000000
              chrX	122000000	123000000
              chrX	123000000	124000000
              chrX	124000000	125000000
              chrX	125000000	126000000
              chrX	126000000	127000000
              chrX	127000000	128000000
              chrX	128000000	129000000
              chrX	129000000	130000000
              chrX	130000000	131000000
              chrX	131000000	132000000
              chrX	132000000	133000000
              chrX	133000000	134000000
              chrX	134000000	135000000
              chrX	135000000	136000000
              chrX	136000000	137000000
              chrX	137000000	138000000
              chrX	138000000	139000000
              chrX	139000000	140000000
              chrX	140000000	141000000
              chrX	141000000	142000000
              chrX	142000000	143000000
              chrX	143000000	144000000
              chrX	144000000	145000000
              chrX	145000000	146000000
              chrX	146000000	147000000
              chrX	147000000	148000000
              chrX	148000000	149000000
              chrX	149000000	150000000
              chrX	150000000	151000000
              chrX	151000000	152000000
              chrX	152000000	153000000
              chrX	153000000	154000000
              chrX	154000000	155000000
              chrX	155000000	156000000
              chrX	156000000	156040895
              chrX_KI270880v1_alt	0	284869
              chrX_KI270881v1_alt	0	144206
              chrX_KI270913v1_alt	0	274009
              chrY	0	1000000
              chrY	1000000	2000000
              chrY	2000000	3000000
              chrY	3000000	4000000
              chrY	4000000	5000000
              chrY	5000000	6000000
              chrY	6000000	7000000
              chrY	7000000	8000000
              chrY	8000000	9000000
              chrY	9000000	10000000
              chrY	10000000	11000000
              chrY	11000000	12000000
              chrY	12000000	13000000
              chrY	13000000	14000000
              chrY	14000000	15000000
              chrY	15000000	16000000
              chrY	16000000	17000000
              chrY	17000000	18000000
              chrY	18000000	19000000
              chrY	19000000	20000000
              chrY	20000000	21000000
              chrY	21000000	22000000
              chrY	22000000	23000000
              chrY	23000000	24000000
              chrY	24000000	25000000
              chrY	25000000	26000000
              chrY	26000000	27000000
              chrY	27000000	28000000
              chrY	28000000	29000000
              chrY	29000000	30000000
              chrY	30000000	31000000
              chrY	31000000	32000000
              chrY	32000000	33000000
              chrY	33000000	34000000
              chrY	34000000	35000000
              chrY	35000000	36000000
              chrY	36000000	37000000
              chrY	37000000	38000000
              chrY	38000000	39000000
              chrY	39000000	40000000
              chrY	40000000	41000000
              chrY	41000000	42000000
              chrY	42000000	43000000
              chrY	43000000	44000000
              chrY	44000000	45000000
              chrY	45000000	46000000
              chrY	46000000	47000000
              chrY	47000000	48000000
              chrY	48000000	49000000
              chrY	49000000	50000000
              chrY	50000000	51000000
              chrY	51000000	52000000
              chrY	52000000	53000000
              chrY	53000000	54000000
              chrY	54000000	55000000
              chrY	55000000	56000000
              chrY	56000000	57000000
              chrY	57000000	57227415
              chrY_KI270740v1_random	0	37240
            filename: hg38_chrom_sizes.window.1000000.bed

        inputs:
        - id: '#Sorted_Whole_Exome_Capture_Bed'
          type:
          - File
          description: Bed file must be sorted
          sbg:fileTypes: BED

        outputs:
        - id: '#OverlapBed'
          type:
          - 'null'
          - File
          outputBinding:
            glob:
              class: Expression
              engine: '#cwl-js-engine'
              script: |
                {
                  //sufix = "test";
                    //output_file=$job.inputs.Window_Bed.path + "_hex.coverage_" + $job.inputs.Sorted_Whole_Exome_Capture_Bed.nameroot + ".bed"

                    basename = 'hg38_chrom_sizes.window.1000000'
                  
                    
                    filepath_cap = $job.inputs.Sorted_Whole_Exome_Capture_Bed.path
                    filename_cap = filepath_cap.split("/").pop()
                    basename_cap = filename_cap.substr(0,filename_cap.lastIndexOf("."))
                  
                    output = basename + "_hex.coverage_" + basename_cap + ".bed"
                    
                    return output
                }
        stdout:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              //sufix = "test";
                //output_file=$job.inputs.Window_Bed.path + "_hex.coverage_" + $job.inputs.Sorted_Whole_Exome_Capture_Bed.nameroot + ".bed"

                basename = 'hg38_chrom_sizes.window.1000000'
              
                
                filepath_cap = $job.inputs.Sorted_Whole_Exome_Capture_Bed.path
                filename_cap = filepath_cap.split("/").pop()
                basename_cap = filename_cap.substr(0,filename_cap.lastIndexOf("."))
              
                output = basename + "_hex.coverage_" + basename_cap + ".bed"
                
                return output
            }


              

        baseCommand:
        - class: Expression
          engine: '#cwl-js-engine'
          script: |-
            'bedops -m ' + $job.inputs.Sorted_Whole_Exome_Capture_Bed.path + ' > Sorted_Whole_Exome_Capture.merged.bed;'
        - bedmap
        arguments:
        - prefix: ''
          position: 0
          valueFrom: --echo
          separate: false
        - prefix: ''
          position: 0
          valueFrom: --bases
          separate: false
        - prefix: ''
          position: 0
          valueFrom: --delim '\t'
          separate: false
        - prefix: ''
          position: 1
          valueFrom: hg38_chrom_sizes.window.1000000.bed
          separate: true
        - prefix: ''
          position: 2
          valueFrom: Sorted_Whole_Exome_Capture.merged.bed
          separate: true

        hints:
        - class: DockerRequirement
          dockerPull: pgc-images.sbgenomics.com/d3b-bixu/bedops:v2
        id: michael_lloyd/mantis/bed-sort/2
        sbg:appVersion:
        - sbg:draft-2
        sbg:content_hash: a793f644c514ea8fef4241e541faafbbf212c82f116b12674d87b745ea36f30b0
        sbg:contributors:
        - michael_lloyd
        sbg:createdBy: michael_lloyd
        sbg:createdOn: 1566228535
        sbg:id: michael_lloyd/mantis/bed-sort/2
        sbg:image_url:
        sbg:latestRevision: 2
        sbg:modifiedBy: michael_lloyd
        sbg:modifiedOn: 1566236195
        sbg:project: michael_lloyd/mantis
        sbg:projectName: MSI, HRD, TMB
        sbg:publisher: sbg
        sbg:revision: 2
        sbg:revisionNotes: ''
        sbg:revisionsInfo:
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1566228535
          sbg:revision: 0
          sbg:revisionNotes: Copy of michael_lloyd/mantis/bedops/3
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1566228770
          sbg:revision: 1
          sbg:revisionNotes: ''
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1566236195
          sbg:revision: 2
          sbg:revisionNotes: ''
        sbg:sbgMaintained: false
        sbg:validationErrors: []
      inputs:
      - id: '#bed_sort.Sorted_Whole_Exome_Capture_Bed'
        source: '#bedops.OverlapBed'
      outputs:
      - id: '#bed_sort.OverlapBed'
      sbg:x: 1531.6337890625
      sbg:y: -725.4197387695312
    - id: '#bedops'
      label: Sort-Bed
      in: []
      run:
        cwlVersion: sbg:draft-2
        class: CommandLineTool
        label: Sort-Bed
        $namespaces:
          sbg: https://sevenbridges.com

        requirements:
        - id: '#cwl-js-engine'
          class: ExpressionEngineRequirement
          requirements:
          - class: DockerRequirement
            dockerPull: rabix/js-engine
        - class: InlineJavascriptRequirement
        - class: CreateFileRequirement
          fileDef:
          - fileContent: |-
              chr1	0	1000000
              chr1	1000000	2000000
              chr1	2000000	3000000
              chr1	3000000	4000000
              chr1	4000000	5000000
              chr1	5000000	6000000
              chr1	6000000	7000000
              chr1	7000000	8000000
              chr1	8000000	9000000
              chr1	9000000	10000000
              chr1	10000000	11000000
              chr1	11000000	12000000
              chr1	12000000	13000000
              chr1	13000000	14000000
              chr1	14000000	15000000
              chr1	15000000	16000000
              chr1	16000000	17000000
              chr1	17000000	18000000
              chr1	18000000	19000000
              chr1	19000000	20000000
              chr1	20000000	21000000
              chr1	21000000	22000000
              chr1	22000000	23000000
              chr1	23000000	24000000
              chr1	24000000	25000000
              chr1	25000000	26000000
              chr1	26000000	27000000
              chr1	27000000	28000000
              chr1	28000000	29000000
              chr1	29000000	30000000
              chr1	30000000	31000000
              chr1	31000000	32000000
              chr1	32000000	33000000
              chr1	33000000	34000000
              chr1	34000000	35000000
              chr1	35000000	36000000
              chr1	36000000	37000000
              chr1	37000000	38000000
              chr1	38000000	39000000
              chr1	39000000	40000000
              chr1	40000000	41000000
              chr1	41000000	42000000
              chr1	42000000	43000000
              chr1	43000000	44000000
              chr1	44000000	45000000
              chr1	45000000	46000000
              chr1	46000000	47000000
              chr1	47000000	48000000
              chr1	48000000	49000000
              chr1	49000000	50000000
              chr1	50000000	51000000
              chr1	51000000	52000000
              chr1	52000000	53000000
              chr1	53000000	54000000
              chr1	54000000	55000000
              chr1	55000000	56000000
              chr1	56000000	57000000
              chr1	57000000	58000000
              chr1	58000000	59000000
              chr1	59000000	60000000
              chr1	60000000	61000000
              chr1	61000000	62000000
              chr1	62000000	63000000
              chr1	63000000	64000000
              chr1	64000000	65000000
              chr1	65000000	66000000
              chr1	66000000	67000000
              chr1	67000000	68000000
              chr1	68000000	69000000
              chr1	69000000	70000000
              chr1	70000000	71000000
              chr1	71000000	72000000
              chr1	72000000	73000000
              chr1	73000000	74000000
              chr1	74000000	75000000
              chr1	75000000	76000000
              chr1	76000000	77000000
              chr1	77000000	78000000
              chr1	78000000	79000000
              chr1	79000000	80000000
              chr1	80000000	81000000
              chr1	81000000	82000000
              chr1	82000000	83000000
              chr1	83000000	84000000
              chr1	84000000	85000000
              chr1	85000000	86000000
              chr1	86000000	87000000
              chr1	87000000	88000000
              chr1	88000000	89000000
              chr1	89000000	90000000
              chr1	90000000	91000000
              chr1	91000000	92000000
              chr1	92000000	93000000
              chr1	93000000	94000000
              chr1	94000000	95000000
              chr1	95000000	96000000
              chr1	96000000	97000000
              chr1	97000000	98000000
              chr1	98000000	99000000
              chr1	99000000	100000000
              chr1	100000000	101000000
              chr1	101000000	102000000
              chr1	102000000	103000000
              chr1	103000000	104000000
              chr1	104000000	105000000
              chr1	105000000	106000000
              chr1	106000000	107000000
              chr1	107000000	108000000
              chr1	108000000	109000000
              chr1	109000000	110000000
              chr1	110000000	111000000
              chr1	111000000	112000000
              chr1	112000000	113000000
              chr1	113000000	114000000
              chr1	114000000	115000000
              chr1	115000000	116000000
              chr1	116000000	117000000
              chr1	117000000	118000000
              chr1	118000000	119000000
              chr1	119000000	120000000
              chr1	120000000	121000000
              chr1	121000000	122000000
              chr1	122000000	123000000
              chr1	123000000	124000000
              chr1	124000000	125000000
              chr1	125000000	126000000
              chr1	126000000	127000000
              chr1	127000000	128000000
              chr1	128000000	129000000
              chr1	129000000	130000000
              chr1	130000000	131000000
              chr1	131000000	132000000
              chr1	132000000	133000000
              chr1	133000000	134000000
              chr1	134000000	135000000
              chr1	135000000	136000000
              chr1	136000000	137000000
              chr1	137000000	138000000
              chr1	138000000	139000000
              chr1	139000000	140000000
              chr1	140000000	141000000
              chr1	141000000	142000000
              chr1	142000000	143000000
              chr1	143000000	144000000
              chr1	144000000	145000000
              chr1	145000000	146000000
              chr1	146000000	147000000
              chr1	147000000	148000000
              chr1	148000000	149000000
              chr1	149000000	150000000
              chr1	150000000	151000000
              chr1	151000000	152000000
              chr1	152000000	153000000
              chr1	153000000	154000000
              chr1	154000000	155000000
              chr1	155000000	156000000
              chr1	156000000	157000000
              chr1	157000000	158000000
              chr1	158000000	159000000
              chr1	159000000	160000000
              chr1	160000000	161000000
              chr1	161000000	162000000
              chr1	162000000	163000000
              chr1	163000000	164000000
              chr1	164000000	165000000
              chr1	165000000	166000000
              chr1	166000000	167000000
              chr1	167000000	168000000
              chr1	168000000	169000000
              chr1	169000000	170000000
              chr1	170000000	171000000
              chr1	171000000	172000000
              chr1	172000000	173000000
              chr1	173000000	174000000
              chr1	174000000	175000000
              chr1	175000000	176000000
              chr1	176000000	177000000
              chr1	177000000	178000000
              chr1	178000000	179000000
              chr1	179000000	180000000
              chr1	180000000	181000000
              chr1	181000000	182000000
              chr1	182000000	183000000
              chr1	183000000	184000000
              chr1	184000000	185000000
              chr1	185000000	186000000
              chr1	186000000	187000000
              chr1	187000000	188000000
              chr1	188000000	189000000
              chr1	189000000	190000000
              chr1	190000000	191000000
              chr1	191000000	192000000
              chr1	192000000	193000000
              chr1	193000000	194000000
              chr1	194000000	195000000
              chr1	195000000	196000000
              chr1	196000000	197000000
              chr1	197000000	198000000
              chr1	198000000	199000000
              chr1	199000000	200000000
              chr1	200000000	201000000
              chr1	201000000	202000000
              chr1	202000000	203000000
              chr1	203000000	204000000
              chr1	204000000	205000000
              chr1	205000000	206000000
              chr1	206000000	207000000
              chr1	207000000	208000000
              chr1	208000000	209000000
              chr1	209000000	210000000
              chr1	210000000	211000000
              chr1	211000000	212000000
              chr1	212000000	213000000
              chr1	213000000	214000000
              chr1	214000000	215000000
              chr1	215000000	216000000
              chr1	216000000	217000000
              chr1	217000000	218000000
              chr1	218000000	219000000
              chr1	219000000	220000000
              chr1	220000000	221000000
              chr1	221000000	222000000
              chr1	222000000	223000000
              chr1	223000000	224000000
              chr1	224000000	225000000
              chr1	225000000	226000000
              chr1	226000000	227000000
              chr1	227000000	228000000
              chr1	228000000	229000000
              chr1	229000000	230000000
              chr1	230000000	231000000
              chr1	231000000	232000000
              chr1	232000000	233000000
              chr1	233000000	234000000
              chr1	234000000	235000000
              chr1	235000000	236000000
              chr1	236000000	237000000
              chr1	237000000	238000000
              chr1	238000000	239000000
              chr1	239000000	240000000
              chr1	240000000	241000000
              chr1	241000000	242000000
              chr1	242000000	243000000
              chr1	243000000	244000000
              chr1	244000000	245000000
              chr1	245000000	246000000
              chr1	246000000	247000000
              chr1	247000000	248000000
              chr1	248000000	248956422
              chr10	0	1000000
              chr10	1000000	2000000
              chr10	2000000	3000000
              chr10	3000000	4000000
              chr10	4000000	5000000
              chr10	5000000	6000000
              chr10	6000000	7000000
              chr10	7000000	8000000
              chr10	8000000	9000000
              chr10	9000000	10000000
              chr10	10000000	11000000
              chr10	11000000	12000000
              chr10	12000000	13000000
              chr10	13000000	14000000
              chr10	14000000	15000000
              chr10	15000000	16000000
              chr10	16000000	17000000
              chr10	17000000	18000000
              chr10	18000000	19000000
              chr10	19000000	20000000
              chr10	20000000	21000000
              chr10	21000000	22000000
              chr10	22000000	23000000
              chr10	23000000	24000000
              chr10	24000000	25000000
              chr10	25000000	26000000
              chr10	26000000	27000000
              chr10	27000000	28000000
              chr10	28000000	29000000
              chr10	29000000	30000000
              chr10	30000000	31000000
              chr10	31000000	32000000
              chr10	32000000	33000000
              chr10	33000000	34000000
              chr10	34000000	35000000
              chr10	35000000	36000000
              chr10	36000000	37000000
              chr10	37000000	38000000
              chr10	38000000	39000000
              chr10	39000000	40000000
              chr10	40000000	41000000
              chr10	41000000	42000000
              chr10	42000000	43000000
              chr10	43000000	44000000
              chr10	44000000	45000000
              chr10	45000000	46000000
              chr10	46000000	47000000
              chr10	47000000	48000000
              chr10	48000000	49000000
              chr10	49000000	50000000
              chr10	50000000	51000000
              chr10	51000000	52000000
              chr10	52000000	53000000
              chr10	53000000	54000000
              chr10	54000000	55000000
              chr10	55000000	56000000
              chr10	56000000	57000000
              chr10	57000000	58000000
              chr10	58000000	59000000
              chr10	59000000	60000000
              chr10	60000000	61000000
              chr10	61000000	62000000
              chr10	62000000	63000000
              chr10	63000000	64000000
              chr10	64000000	65000000
              chr10	65000000	66000000
              chr10	66000000	67000000
              chr10	67000000	68000000
              chr10	68000000	69000000
              chr10	69000000	70000000
              chr10	70000000	71000000
              chr10	71000000	72000000
              chr10	72000000	73000000
              chr10	73000000	74000000
              chr10	74000000	75000000
              chr10	75000000	76000000
              chr10	76000000	77000000
              chr10	77000000	78000000
              chr10	78000000	79000000
              chr10	79000000	80000000
              chr10	80000000	81000000
              chr10	81000000	82000000
              chr10	82000000	83000000
              chr10	83000000	84000000
              chr10	84000000	85000000
              chr10	85000000	86000000
              chr10	86000000	87000000
              chr10	87000000	88000000
              chr10	88000000	89000000
              chr10	89000000	90000000
              chr10	90000000	91000000
              chr10	91000000	92000000
              chr10	92000000	93000000
              chr10	93000000	94000000
              chr10	94000000	95000000
              chr10	95000000	96000000
              chr10	96000000	97000000
              chr10	97000000	98000000
              chr10	98000000	99000000
              chr10	99000000	100000000
              chr10	100000000	101000000
              chr10	101000000	102000000
              chr10	102000000	103000000
              chr10	103000000	104000000
              chr10	104000000	105000000
              chr10	105000000	106000000
              chr10	106000000	107000000
              chr10	107000000	108000000
              chr10	108000000	109000000
              chr10	109000000	110000000
              chr10	110000000	111000000
              chr10	111000000	112000000
              chr10	112000000	113000000
              chr10	113000000	114000000
              chr10	114000000	115000000
              chr10	115000000	116000000
              chr10	116000000	117000000
              chr10	117000000	118000000
              chr10	118000000	119000000
              chr10	119000000	120000000
              chr10	120000000	121000000
              chr10	121000000	122000000
              chr10	122000000	123000000
              chr10	123000000	124000000
              chr10	124000000	125000000
              chr10	125000000	126000000
              chr10	126000000	127000000
              chr10	127000000	128000000
              chr10	128000000	129000000
              chr10	129000000	130000000
              chr10	130000000	131000000
              chr10	131000000	132000000
              chr10	132000000	133000000
              chr10	133000000	133797422
              chr10_GL383545v1_alt	0	179254
              chr10_GL383546v1_alt	0	309802
              chr10_KI270824v1_alt	0	181496
              chr10_KI270825v1_alt	0	188315
              chr11	0	1000000
              chr11	1000000	2000000
              chr11	2000000	3000000
              chr11	3000000	4000000
              chr11	4000000	5000000
              chr11	5000000	6000000
              chr11	6000000	7000000
              chr11	7000000	8000000
              chr11	8000000	9000000
              chr11	9000000	10000000
              chr11	10000000	11000000
              chr11	11000000	12000000
              chr11	12000000	13000000
              chr11	13000000	14000000
              chr11	14000000	15000000
              chr11	15000000	16000000
              chr11	16000000	17000000
              chr11	17000000	18000000
              chr11	18000000	19000000
              chr11	19000000	20000000
              chr11	20000000	21000000
              chr11	21000000	22000000
              chr11	22000000	23000000
              chr11	23000000	24000000
              chr11	24000000	25000000
              chr11	25000000	26000000
              chr11	26000000	27000000
              chr11	27000000	28000000
              chr11	28000000	29000000
              chr11	29000000	30000000
              chr11	30000000	31000000
              chr11	31000000	32000000
              chr11	32000000	33000000
              chr11	33000000	34000000
              chr11	34000000	35000000
              chr11	35000000	36000000
              chr11	36000000	37000000
              chr11	37000000	38000000
              chr11	38000000	39000000
              chr11	39000000	40000000
              chr11	40000000	41000000
              chr11	41000000	42000000
              chr11	42000000	43000000
              chr11	43000000	44000000
              chr11	44000000	45000000
              chr11	45000000	46000000
              chr11	46000000	47000000
              chr11	47000000	48000000
              chr11	48000000	49000000
              chr11	49000000	50000000
              chr11	50000000	51000000
              chr11	51000000	52000000
              chr11	52000000	53000000
              chr11	53000000	54000000
              chr11	54000000	55000000
              chr11	55000000	56000000
              chr11	56000000	57000000
              chr11	57000000	58000000
              chr11	58000000	59000000
              chr11	59000000	60000000
              chr11	60000000	61000000
              chr11	61000000	62000000
              chr11	62000000	63000000
              chr11	63000000	64000000
              chr11	64000000	65000000
              chr11	65000000	66000000
              chr11	66000000	67000000
              chr11	67000000	68000000
              chr11	68000000	69000000
              chr11	69000000	70000000
              chr11	70000000	71000000
              chr11	71000000	72000000
              chr11	72000000	73000000
              chr11	73000000	74000000
              chr11	74000000	75000000
              chr11	75000000	76000000
              chr11	76000000	77000000
              chr11	77000000	78000000
              chr11	78000000	79000000
              chr11	79000000	80000000
              chr11	80000000	81000000
              chr11	81000000	82000000
              chr11	82000000	83000000
              chr11	83000000	84000000
              chr11	84000000	85000000
              chr11	85000000	86000000
              chr11	86000000	87000000
              chr11	87000000	88000000
              chr11	88000000	89000000
              chr11	89000000	90000000
              chr11	90000000	91000000
              chr11	91000000	92000000
              chr11	92000000	93000000
              chr11	93000000	94000000
              chr11	94000000	95000000
              chr11	95000000	96000000
              chr11	96000000	97000000
              chr11	97000000	98000000
              chr11	98000000	99000000
              chr11	99000000	100000000
              chr11	100000000	101000000
              chr11	101000000	102000000
              chr11	102000000	103000000
              chr11	103000000	104000000
              chr11	104000000	105000000
              chr11	105000000	106000000
              chr11	106000000	107000000
              chr11	107000000	108000000
              chr11	108000000	109000000
              chr11	109000000	110000000
              chr11	110000000	111000000
              chr11	111000000	112000000
              chr11	112000000	113000000
              chr11	113000000	114000000
              chr11	114000000	115000000
              chr11	115000000	116000000
              chr11	116000000	117000000
              chr11	117000000	118000000
              chr11	118000000	119000000
              chr11	119000000	120000000
              chr11	120000000	121000000
              chr11	121000000	122000000
              chr11	122000000	123000000
              chr11	123000000	124000000
              chr11	124000000	125000000
              chr11	125000000	126000000
              chr11	126000000	127000000
              chr11	127000000	128000000
              chr11	128000000	129000000
              chr11	129000000	130000000
              chr11	130000000	131000000
              chr11	131000000	132000000
              chr11	132000000	133000000
              chr11	133000000	134000000
              chr11	134000000	135000000
              chr11	135000000	135086622
              chr11_GL383547v1_alt	0	154407
              chr11_JH159136v1_alt	0	200998
              chr11_JH159137v1_alt	0	191409
              chr11_KI270721v1_random	0	100316
              chr11_KI270826v1_alt	0	186169
              chr11_KI270827v1_alt	0	67707
              chr11_KI270829v1_alt	0	204059
              chr11_KI270830v1_alt	0	177092
              chr11_KI270831v1_alt	0	296895
              chr11_KI270832v1_alt	0	210133
              chr11_KI270902v1_alt	0	106711
              chr11_KI270903v1_alt	0	214625
              chr11_KI270927v1_alt	0	218612
              chr12	0	1000000
              chr12	1000000	2000000
              chr12	2000000	3000000
              chr12	3000000	4000000
              chr12	4000000	5000000
              chr12	5000000	6000000
              chr12	6000000	7000000
              chr12	7000000	8000000
              chr12	8000000	9000000
              chr12	9000000	10000000
              chr12	10000000	11000000
              chr12	11000000	12000000
              chr12	12000000	13000000
              chr12	13000000	14000000
              chr12	14000000	15000000
              chr12	15000000	16000000
              chr12	16000000	17000000
              chr12	17000000	18000000
              chr12	18000000	19000000
              chr12	19000000	20000000
              chr12	20000000	21000000
              chr12	21000000	22000000
              chr12	22000000	23000000
              chr12	23000000	24000000
              chr12	24000000	25000000
              chr12	25000000	26000000
              chr12	26000000	27000000
              chr12	27000000	28000000
              chr12	28000000	29000000
              chr12	29000000	30000000
              chr12	30000000	31000000
              chr12	31000000	32000000
              chr12	32000000	33000000
              chr12	33000000	34000000
              chr12	34000000	35000000
              chr12	35000000	36000000
              chr12	36000000	37000000
              chr12	37000000	38000000
              chr12	38000000	39000000
              chr12	39000000	40000000
              chr12	40000000	41000000
              chr12	41000000	42000000
              chr12	42000000	43000000
              chr12	43000000	44000000
              chr12	44000000	45000000
              chr12	45000000	46000000
              chr12	46000000	47000000
              chr12	47000000	48000000
              chr12	48000000	49000000
              chr12	49000000	50000000
              chr12	50000000	51000000
              chr12	51000000	52000000
              chr12	52000000	53000000
              chr12	53000000	54000000
              chr12	54000000	55000000
              chr12	55000000	56000000
              chr12	56000000	57000000
              chr12	57000000	58000000
              chr12	58000000	59000000
              chr12	59000000	60000000
              chr12	60000000	61000000
              chr12	61000000	62000000
              chr12	62000000	63000000
              chr12	63000000	64000000
              chr12	64000000	65000000
              chr12	65000000	66000000
              chr12	66000000	67000000
              chr12	67000000	68000000
              chr12	68000000	69000000
              chr12	69000000	70000000
              chr12	70000000	71000000
              chr12	71000000	72000000
              chr12	72000000	73000000
              chr12	73000000	74000000
              chr12	74000000	75000000
              chr12	75000000	76000000
              chr12	76000000	77000000
              chr12	77000000	78000000
              chr12	78000000	79000000
              chr12	79000000	80000000
              chr12	80000000	81000000
              chr12	81000000	82000000
              chr12	82000000	83000000
              chr12	83000000	84000000
              chr12	84000000	85000000
              chr12	85000000	86000000
              chr12	86000000	87000000
              chr12	87000000	88000000
              chr12	88000000	89000000
              chr12	89000000	90000000
              chr12	90000000	91000000
              chr12	91000000	92000000
              chr12	92000000	93000000
              chr12	93000000	94000000
              chr12	94000000	95000000
              chr12	95000000	96000000
              chr12	96000000	97000000
              chr12	97000000	98000000
              chr12	98000000	99000000
              chr12	99000000	100000000
              chr12	100000000	101000000
              chr12	101000000	102000000
              chr12	102000000	103000000
              chr12	103000000	104000000
              chr12	104000000	105000000
              chr12	105000000	106000000
              chr12	106000000	107000000
              chr12	107000000	108000000
              chr12	108000000	109000000
              chr12	109000000	110000000
              chr12	110000000	111000000
              chr12	111000000	112000000
              chr12	112000000	113000000
              chr12	113000000	114000000
              chr12	114000000	115000000
              chr12	115000000	116000000
              chr12	116000000	117000000
              chr12	117000000	118000000
              chr12	118000000	119000000
              chr12	119000000	120000000
              chr12	120000000	121000000
              chr12	121000000	122000000
              chr12	122000000	123000000
              chr12	123000000	124000000
              chr12	124000000	125000000
              chr12	125000000	126000000
              chr12	126000000	127000000
              chr12	127000000	128000000
              chr12	128000000	129000000
              chr12	129000000	130000000
              chr12	130000000	131000000
              chr12	131000000	132000000
              chr12	132000000	133000000
              chr12	133000000	133275309
              chr12_GL383549v1_alt	0	120804
              chr12_GL383550v2_alt	0	169178
              chr12_GL383551v1_alt	0	184319
              chr12_GL383552v1_alt	0	138655
              chr12_GL383553v2_alt	0	152874
              chr12_GL877875v1_alt	0	167313
              chr12_GL877876v1_alt	0	408271
              chr12_KI270833v1_alt	0	76061
              chr12_KI270834v1_alt	0	119498
              chr12_KI270835v1_alt	0	238139
              chr12_KI270836v1_alt	0	56134
              chr12_KI270837v1_alt	0	40090
              chr12_KI270904v1_alt	0	572349
              chr13	0	1000000
              chr13	1000000	2000000
              chr13	2000000	3000000
              chr13	3000000	4000000
              chr13	4000000	5000000
              chr13	5000000	6000000
              chr13	6000000	7000000
              chr13	7000000	8000000
              chr13	8000000	9000000
              chr13	9000000	10000000
              chr13	10000000	11000000
              chr13	11000000	12000000
              chr13	12000000	13000000
              chr13	13000000	14000000
              chr13	14000000	15000000
              chr13	15000000	16000000
              chr13	16000000	17000000
              chr13	17000000	18000000
              chr13	18000000	19000000
              chr13	19000000	20000000
              chr13	20000000	21000000
              chr13	21000000	22000000
              chr13	22000000	23000000
              chr13	23000000	24000000
              chr13	24000000	25000000
              chr13	25000000	26000000
              chr13	26000000	27000000
              chr13	27000000	28000000
              chr13	28000000	29000000
              chr13	29000000	30000000
              chr13	30000000	31000000
              chr13	31000000	32000000
              chr13	32000000	33000000
              chr13	33000000	34000000
              chr13	34000000	35000000
              chr13	35000000	36000000
              chr13	36000000	37000000
              chr13	37000000	38000000
              chr13	38000000	39000000
              chr13	39000000	40000000
              chr13	40000000	41000000
              chr13	41000000	42000000
              chr13	42000000	43000000
              chr13	43000000	44000000
              chr13	44000000	45000000
              chr13	45000000	46000000
              chr13	46000000	47000000
              chr13	47000000	48000000
              chr13	48000000	49000000
              chr13	49000000	50000000
              chr13	50000000	51000000
              chr13	51000000	52000000
              chr13	52000000	53000000
              chr13	53000000	54000000
              chr13	54000000	55000000
              chr13	55000000	56000000
              chr13	56000000	57000000
              chr13	57000000	58000000
              chr13	58000000	59000000
              chr13	59000000	60000000
              chr13	60000000	61000000
              chr13	61000000	62000000
              chr13	62000000	63000000
              chr13	63000000	64000000
              chr13	64000000	65000000
              chr13	65000000	66000000
              chr13	66000000	67000000
              chr13	67000000	68000000
              chr13	68000000	69000000
              chr13	69000000	70000000
              chr13	70000000	71000000
              chr13	71000000	72000000
              chr13	72000000	73000000
              chr13	73000000	74000000
              chr13	74000000	75000000
              chr13	75000000	76000000
              chr13	76000000	77000000
              chr13	77000000	78000000
              chr13	78000000	79000000
              chr13	79000000	80000000
              chr13	80000000	81000000
              chr13	81000000	82000000
              chr13	82000000	83000000
              chr13	83000000	84000000
              chr13	84000000	85000000
              chr13	85000000	86000000
              chr13	86000000	87000000
              chr13	87000000	88000000
              chr13	88000000	89000000
              chr13	89000000	90000000
              chr13	90000000	91000000
              chr13	91000000	92000000
              chr13	92000000	93000000
              chr13	93000000	94000000
              chr13	94000000	95000000
              chr13	95000000	96000000
              chr13	96000000	97000000
              chr13	97000000	98000000
              chr13	98000000	99000000
              chr13	99000000	100000000
              chr13	100000000	101000000
              chr13	101000000	102000000
              chr13	102000000	103000000
              chr13	103000000	104000000
              chr13	104000000	105000000
              chr13	105000000	106000000
              chr13	106000000	107000000
              chr13	107000000	108000000
              chr13	108000000	109000000
              chr13	109000000	110000000
              chr13	110000000	111000000
              chr13	111000000	112000000
              chr13	112000000	113000000
              chr13	113000000	114000000
              chr13	114000000	114364328
              chr13_KI270838v1_alt	0	306913
              chr13_KI270839v1_alt	0	180306
              chr13_KI270840v1_alt	0	191684
              chr13_KI270841v1_alt	0	169134
              chr13_KI270842v1_alt	0	37287
              chr13_KI270843v1_alt	0	103832
              chr14	0	1000000
              chr14	1000000	2000000
              chr14	2000000	3000000
              chr14	3000000	4000000
              chr14	4000000	5000000
              chr14	5000000	6000000
              chr14	6000000	7000000
              chr14	7000000	8000000
              chr14	8000000	9000000
              chr14	9000000	10000000
              chr14	10000000	11000000
              chr14	11000000	12000000
              chr14	12000000	13000000
              chr14	13000000	14000000
              chr14	14000000	15000000
              chr14	15000000	16000000
              chr14	16000000	17000000
              chr14	17000000	18000000
              chr14	18000000	19000000
              chr14	19000000	20000000
              chr14	20000000	21000000
              chr14	21000000	22000000
              chr14	22000000	23000000
              chr14	23000000	24000000
              chr14	24000000	25000000
              chr14	25000000	26000000
              chr14	26000000	27000000
              chr14	27000000	28000000
              chr14	28000000	29000000
              chr14	29000000	30000000
              chr14	30000000	31000000
              chr14	31000000	32000000
              chr14	32000000	33000000
              chr14	33000000	34000000
              chr14	34000000	35000000
              chr14	35000000	36000000
              chr14	36000000	37000000
              chr14	37000000	38000000
              chr14	38000000	39000000
              chr14	39000000	40000000
              chr14	40000000	41000000
              chr14	41000000	42000000
              chr14	42000000	43000000
              chr14	43000000	44000000
              chr14	44000000	45000000
              chr14	45000000	46000000
              chr14	46000000	47000000
              chr14	47000000	48000000
              chr14	48000000	49000000
              chr14	49000000	50000000
              chr14	50000000	51000000
              chr14	51000000	52000000
              chr14	52000000	53000000
              chr14	53000000	54000000
              chr14	54000000	55000000
              chr14	55000000	56000000
              chr14	56000000	57000000
              chr14	57000000	58000000
              chr14	58000000	59000000
              chr14	59000000	60000000
              chr14	60000000	61000000
              chr14	61000000	62000000
              chr14	62000000	63000000
              chr14	63000000	64000000
              chr14	64000000	65000000
              chr14	65000000	66000000
              chr14	66000000	67000000
              chr14	67000000	68000000
              chr14	68000000	69000000
              chr14	69000000	70000000
              chr14	70000000	71000000
              chr14	71000000	72000000
              chr14	72000000	73000000
              chr14	73000000	74000000
              chr14	74000000	75000000
              chr14	75000000	76000000
              chr14	76000000	77000000
              chr14	77000000	78000000
              chr14	78000000	79000000
              chr14	79000000	80000000
              chr14	80000000	81000000
              chr14	81000000	82000000
              chr14	82000000	83000000
              chr14	83000000	84000000
              chr14	84000000	85000000
              chr14	85000000	86000000
              chr14	86000000	87000000
              chr14	87000000	88000000
              chr14	88000000	89000000
              chr14	89000000	90000000
              chr14	90000000	91000000
              chr14	91000000	92000000
              chr14	92000000	93000000
              chr14	93000000	94000000
              chr14	94000000	95000000
              chr14	95000000	96000000
              chr14	96000000	97000000
              chr14	97000000	98000000
              chr14	98000000	99000000
              chr14	99000000	100000000
              chr14	100000000	101000000
              chr14	101000000	102000000
              chr14	102000000	103000000
              chr14	103000000	104000000
              chr14	104000000	105000000
              chr14	105000000	106000000
              chr14	106000000	107000000
              chr14	107000000	107043718
              chr14_GL000009v2_random	0	201709
              chr14_GL000194v1_random	0	191469
              chr14_GL000225v1_random	0	211173
              chr14_KI270722v1_random	0	194050
              chr14_KI270723v1_random	0	38115
              chr14_KI270724v1_random	0	39555
              chr14_KI270725v1_random	0	172810
              chr14_KI270726v1_random	0	43739
              chr14_KI270844v1_alt	0	322166
              chr14_KI270845v1_alt	0	180703
              chr14_KI270846v1_alt	0	1000000
              chr14_KI270846v1_alt	1000000	1351393
              chr14_KI270847v1_alt	0	1000000
              chr14_KI270847v1_alt	1000000	1511111
              chr15	0	1000000
              chr15	1000000	2000000
              chr15	2000000	3000000
              chr15	3000000	4000000
              chr15	4000000	5000000
              chr15	5000000	6000000
              chr15	6000000	7000000
              chr15	7000000	8000000
              chr15	8000000	9000000
              chr15	9000000	10000000
              chr15	10000000	11000000
              chr15	11000000	12000000
              chr15	12000000	13000000
              chr15	13000000	14000000
              chr15	14000000	15000000
              chr15	15000000	16000000
              chr15	16000000	17000000
              chr15	17000000	18000000
              chr15	18000000	19000000
              chr15	19000000	20000000
              chr15	20000000	21000000
              chr15	21000000	22000000
              chr15	22000000	23000000
              chr15	23000000	24000000
              chr15	24000000	25000000
              chr15	25000000	26000000
              chr15	26000000	27000000
              chr15	27000000	28000000
              chr15	28000000	29000000
              chr15	29000000	30000000
              chr15	30000000	31000000
              chr15	31000000	32000000
              chr15	32000000	33000000
              chr15	33000000	34000000
              chr15	34000000	35000000
              chr15	35000000	36000000
              chr15	36000000	37000000
              chr15	37000000	38000000
              chr15	38000000	39000000
              chr15	39000000	40000000
              chr15	40000000	41000000
              chr15	41000000	42000000
              chr15	42000000	43000000
              chr15	43000000	44000000
              chr15	44000000	45000000
              chr15	45000000	46000000
              chr15	46000000	47000000
              chr15	47000000	48000000
              chr15	48000000	49000000
              chr15	49000000	50000000
              chr15	50000000	51000000
              chr15	51000000	52000000
              chr15	52000000	53000000
              chr15	53000000	54000000
              chr15	54000000	55000000
              chr15	55000000	56000000
              chr15	56000000	57000000
              chr15	57000000	58000000
              chr15	58000000	59000000
              chr15	59000000	60000000
              chr15	60000000	61000000
              chr15	61000000	62000000
              chr15	62000000	63000000
              chr15	63000000	64000000
              chr15	64000000	65000000
              chr15	65000000	66000000
              chr15	66000000	67000000
              chr15	67000000	68000000
              chr15	68000000	69000000
              chr15	69000000	70000000
              chr15	70000000	71000000
              chr15	71000000	72000000
              chr15	72000000	73000000
              chr15	73000000	74000000
              chr15	74000000	75000000
              chr15	75000000	76000000
              chr15	76000000	77000000
              chr15	77000000	78000000
              chr15	78000000	79000000
              chr15	79000000	80000000
              chr15	80000000	81000000
              chr15	81000000	82000000
              chr15	82000000	83000000
              chr15	83000000	84000000
              chr15	84000000	85000000
              chr15	85000000	86000000
              chr15	86000000	87000000
              chr15	87000000	88000000
              chr15	88000000	89000000
              chr15	89000000	90000000
              chr15	90000000	91000000
              chr15	91000000	92000000
              chr15	92000000	93000000
              chr15	93000000	94000000
              chr15	94000000	95000000
              chr15	95000000	96000000
              chr15	96000000	97000000
              chr15	97000000	98000000
              chr15	98000000	99000000
              chr15	99000000	100000000
              chr15	100000000	101000000
              chr15	101000000	101991189
              chr15_GL383554v1_alt	0	296527
              chr15_GL383555v2_alt	0	388773
              chr15_KI270727v1_random	0	448248
              chr15_KI270848v1_alt	0	327382
              chr15_KI270849v1_alt	0	244917
              chr15_KI270850v1_alt	0	430880
              chr15_KI270851v1_alt	0	263054
              chr15_KI270852v1_alt	0	478999
              chr15_KI270905v1_alt	0	1000000
              chr15_KI270905v1_alt	1000000	2000000
              chr15_KI270905v1_alt	2000000	3000000
              chr15_KI270905v1_alt	3000000	4000000
              chr15_KI270905v1_alt	4000000	5000000
              chr15_KI270905v1_alt	5000000	5161414
              chr15_KI270906v1_alt	0	196384
              chr16	0	1000000
              chr16	1000000	2000000
              chr16	2000000	3000000
              chr16	3000000	4000000
              chr16	4000000	5000000
              chr16	5000000	6000000
              chr16	6000000	7000000
              chr16	7000000	8000000
              chr16	8000000	9000000
              chr16	9000000	10000000
              chr16	10000000	11000000
              chr16	11000000	12000000
              chr16	12000000	13000000
              chr16	13000000	14000000
              chr16	14000000	15000000
              chr16	15000000	16000000
              chr16	16000000	17000000
              chr16	17000000	18000000
              chr16	18000000	19000000
              chr16	19000000	20000000
              chr16	20000000	21000000
              chr16	21000000	22000000
              chr16	22000000	23000000
              chr16	23000000	24000000
              chr16	24000000	25000000
              chr16	25000000	26000000
              chr16	26000000	27000000
              chr16	27000000	28000000
              chr16	28000000	29000000
              chr16	29000000	30000000
              chr16	30000000	31000000
              chr16	31000000	32000000
              chr16	32000000	33000000
              chr16	33000000	34000000
              chr16	34000000	35000000
              chr16	35000000	36000000
              chr16	36000000	37000000
              chr16	37000000	38000000
              chr16	38000000	39000000
              chr16	39000000	40000000
              chr16	40000000	41000000
              chr16	41000000	42000000
              chr16	42000000	43000000
              chr16	43000000	44000000
              chr16	44000000	45000000
              chr16	45000000	46000000
              chr16	46000000	47000000
              chr16	47000000	48000000
              chr16	48000000	49000000
              chr16	49000000	50000000
              chr16	50000000	51000000
              chr16	51000000	52000000
              chr16	52000000	53000000
              chr16	53000000	54000000
              chr16	54000000	55000000
              chr16	55000000	56000000
              chr16	56000000	57000000
              chr16	57000000	58000000
              chr16	58000000	59000000
              chr16	59000000	60000000
              chr16	60000000	61000000
              chr16	61000000	62000000
              chr16	62000000	63000000
              chr16	63000000	64000000
              chr16	64000000	65000000
              chr16	65000000	66000000
              chr16	66000000	67000000
              chr16	67000000	68000000
              chr16	68000000	69000000
              chr16	69000000	70000000
              chr16	70000000	71000000
              chr16	71000000	72000000
              chr16	72000000	73000000
              chr16	73000000	74000000
              chr16	74000000	75000000
              chr16	75000000	76000000
              chr16	76000000	77000000
              chr16	77000000	78000000
              chr16	78000000	79000000
              chr16	79000000	80000000
              chr16	80000000	81000000
              chr16	81000000	82000000
              chr16	82000000	83000000
              chr16	83000000	84000000
              chr16	84000000	85000000
              chr16	85000000	86000000
              chr16	86000000	87000000
              chr16	87000000	88000000
              chr16	88000000	89000000
              chr16	89000000	90000000
              chr16	90000000	90338345
              chr16_GL383556v1_alt	0	192462
              chr16_GL383557v1_alt	0	89672
              chr16_KI270728v1_random	0	1000000
              chr16_KI270728v1_random	1000000	1872759
              chr16_KI270853v1_alt	0	1000000
              chr16_KI270853v1_alt	1000000	2000000
              chr16_KI270853v1_alt	2000000	2659700
              chr16_KI270854v1_alt	0	134193
              chr16_KI270855v1_alt	0	232857
              chr16_KI270856v1_alt	0	63982
              chr17	0	1000000
              chr17	1000000	2000000
              chr17	2000000	3000000
              chr17	3000000	4000000
              chr17	4000000	5000000
              chr17	5000000	6000000
              chr17	6000000	7000000
              chr17	7000000	8000000
              chr17	8000000	9000000
              chr17	9000000	10000000
              chr17	10000000	11000000
              chr17	11000000	12000000
              chr17	12000000	13000000
              chr17	13000000	14000000
              chr17	14000000	15000000
              chr17	15000000	16000000
              chr17	16000000	17000000
              chr17	17000000	18000000
              chr17	18000000	19000000
              chr17	19000000	20000000
              chr17	20000000	21000000
              chr17	21000000	22000000
              chr17	22000000	23000000
              chr17	23000000	24000000
              chr17	24000000	25000000
              chr17	25000000	26000000
              chr17	26000000	27000000
              chr17	27000000	28000000
              chr17	28000000	29000000
              chr17	29000000	30000000
              chr17	30000000	31000000
              chr17	31000000	32000000
              chr17	32000000	33000000
              chr17	33000000	34000000
              chr17	34000000	35000000
              chr17	35000000	36000000
              chr17	36000000	37000000
              chr17	37000000	38000000
              chr17	38000000	39000000
              chr17	39000000	40000000
              chr17	40000000	41000000
              chr17	41000000	42000000
              chr17	42000000	43000000
              chr17	43000000	44000000
              chr17	44000000	45000000
              chr17	45000000	46000000
              chr17	46000000	47000000
              chr17	47000000	48000000
              chr17	48000000	49000000
              chr17	49000000	50000000
              chr17	50000000	51000000
              chr17	51000000	52000000
              chr17	52000000	53000000
              chr17	53000000	54000000
              chr17	54000000	55000000
              chr17	55000000	56000000
              chr17	56000000	57000000
              chr17	57000000	58000000
              chr17	58000000	59000000
              chr17	59000000	60000000
              chr17	60000000	61000000
              chr17	61000000	62000000
              chr17	62000000	63000000
              chr17	63000000	64000000
              chr17	64000000	65000000
              chr17	65000000	66000000
              chr17	66000000	67000000
              chr17	67000000	68000000
              chr17	68000000	69000000
              chr17	69000000	70000000
              chr17	70000000	71000000
              chr17	71000000	72000000
              chr17	72000000	73000000
              chr17	73000000	74000000
              chr17	74000000	75000000
              chr17	75000000	76000000
              chr17	76000000	77000000
              chr17	77000000	78000000
              chr17	78000000	79000000
              chr17	79000000	80000000
              chr17	80000000	81000000
              chr17	81000000	82000000
              chr17	82000000	83000000
              chr17	83000000	83257441
              chr17_GL000205v2_random	0	185591
              chr17_GL000258v2_alt	0	1000000
              chr17_GL000258v2_alt	1000000	1821992
              chr17_GL383563v3_alt	0	375691
              chr17_GL383564v2_alt	0	133151
              chr17_GL383565v1_alt	0	223995
              chr17_GL383566v1_alt	0	90219
              chr17_JH159146v1_alt	0	278131
              chr17_JH159147v1_alt	0	70345
              chr17_JH159148v1_alt	0	88070
              chr17_KI270729v1_random	0	280839
              chr17_KI270730v1_random	0	112551
              chr17_KI270857v1_alt	0	1000000
              chr17_KI270857v1_alt	1000000	2000000
              chr17_KI270857v1_alt	2000000	2877074
              chr17_KI270858v1_alt	0	235827
              chr17_KI270859v1_alt	0	108763
              chr17_KI270860v1_alt	0	178921
              chr17_KI270861v1_alt	0	196688
              chr17_KI270862v1_alt	0	391357
              chr17_KI270907v1_alt	0	137721
              chr17_KI270908v1_alt	0	1000000
              chr17_KI270908v1_alt	1000000	1423190
              chr17_KI270909v1_alt	0	325800
              chr17_KI270910v1_alt	0	157099
              chr18	0	1000000
              chr18	1000000	2000000
              chr18	2000000	3000000
              chr18	3000000	4000000
              chr18	4000000	5000000
              chr18	5000000	6000000
              chr18	6000000	7000000
              chr18	7000000	8000000
              chr18	8000000	9000000
              chr18	9000000	10000000
              chr18	10000000	11000000
              chr18	11000000	12000000
              chr18	12000000	13000000
              chr18	13000000	14000000
              chr18	14000000	15000000
              chr18	15000000	16000000
              chr18	16000000	17000000
              chr18	17000000	18000000
              chr18	18000000	19000000
              chr18	19000000	20000000
              chr18	20000000	21000000
              chr18	21000000	22000000
              chr18	22000000	23000000
              chr18	23000000	24000000
              chr18	24000000	25000000
              chr18	25000000	26000000
              chr18	26000000	27000000
              chr18	27000000	28000000
              chr18	28000000	29000000
              chr18	29000000	30000000
              chr18	30000000	31000000
              chr18	31000000	32000000
              chr18	32000000	33000000
              chr18	33000000	34000000
              chr18	34000000	35000000
              chr18	35000000	36000000
              chr18	36000000	37000000
              chr18	37000000	38000000
              chr18	38000000	39000000
              chr18	39000000	40000000
              chr18	40000000	41000000
              chr18	41000000	42000000
              chr18	42000000	43000000
              chr18	43000000	44000000
              chr18	44000000	45000000
              chr18	45000000	46000000
              chr18	46000000	47000000
              chr18	47000000	48000000
              chr18	48000000	49000000
              chr18	49000000	50000000
              chr18	50000000	51000000
              chr18	51000000	52000000
              chr18	52000000	53000000
              chr18	53000000	54000000
              chr18	54000000	55000000
              chr18	55000000	56000000
              chr18	56000000	57000000
              chr18	57000000	58000000
              chr18	58000000	59000000
              chr18	59000000	60000000
              chr18	60000000	61000000
              chr18	61000000	62000000
              chr18	62000000	63000000
              chr18	63000000	64000000
              chr18	64000000	65000000
              chr18	65000000	66000000
              chr18	66000000	67000000
              chr18	67000000	68000000
              chr18	68000000	69000000
              chr18	69000000	70000000
              chr18	70000000	71000000
              chr18	71000000	72000000
              chr18	72000000	73000000
              chr18	73000000	74000000
              chr18	74000000	75000000
              chr18	75000000	76000000
              chr18	76000000	77000000
              chr18	77000000	78000000
              chr18	78000000	79000000
              chr18	79000000	80000000
              chr18	80000000	80373285
              chr18_GL383567v1_alt	0	289831
              chr18_GL383568v1_alt	0	104552
              chr18_GL383569v1_alt	0	167950
              chr18_GL383570v1_alt	0	164789
              chr18_GL383571v1_alt	0	198278
              chr18_GL383572v1_alt	0	159547
              chr18_KI270863v1_alt	0	167999
              chr18_KI270864v1_alt	0	111737
              chr18_KI270911v1_alt	0	157710
              chr18_KI270912v1_alt	0	174061
              chr19	0	1000000
              chr19	1000000	2000000
              chr19	2000000	3000000
              chr19	3000000	4000000
              chr19	4000000	5000000
              chr19	5000000	6000000
              chr19	6000000	7000000
              chr19	7000000	8000000
              chr19	8000000	9000000
              chr19	9000000	10000000
              chr19	10000000	11000000
              chr19	11000000	12000000
              chr19	12000000	13000000
              chr19	13000000	14000000
              chr19	14000000	15000000
              chr19	15000000	16000000
              chr19	16000000	17000000
              chr19	17000000	18000000
              chr19	18000000	19000000
              chr19	19000000	20000000
              chr19	20000000	21000000
              chr19	21000000	22000000
              chr19	22000000	23000000
              chr19	23000000	24000000
              chr19	24000000	25000000
              chr19	25000000	26000000
              chr19	26000000	27000000
              chr19	27000000	28000000
              chr19	28000000	29000000
              chr19	29000000	30000000
              chr19	30000000	31000000
              chr19	31000000	32000000
              chr19	32000000	33000000
              chr19	33000000	34000000
              chr19	34000000	35000000
              chr19	35000000	36000000
              chr19	36000000	37000000
              chr19	37000000	38000000
              chr19	38000000	39000000
              chr19	39000000	40000000
              chr19	40000000	41000000
              chr19	41000000	42000000
              chr19	42000000	43000000
              chr19	43000000	44000000
              chr19	44000000	45000000
              chr19	45000000	46000000
              chr19	46000000	47000000
              chr19	47000000	48000000
              chr19	48000000	49000000
              chr19	49000000	50000000
              chr19	50000000	51000000
              chr19	51000000	52000000
              chr19	52000000	53000000
              chr19	53000000	54000000
              chr19	54000000	55000000
              chr19	55000000	56000000
              chr19	56000000	57000000
              chr19	57000000	58000000
              chr19	58000000	58617616
              chr19_GL000209v2_alt	0	177381
              chr19_GL383573v1_alt	0	385657
              chr19_GL383574v1_alt	0	155864
              chr19_GL383575v2_alt	0	170222
              chr19_GL383576v1_alt	0	188024
              chr19_GL949746v1_alt	0	987716
              chr19_GL949747v2_alt	0	729520
              chr19_GL949748v2_alt	0	1000000
              chr19_GL949748v2_alt	1000000	1064304
              chr19_GL949749v2_alt	0	1000000
              chr19_GL949749v2_alt	1000000	1091841
              chr19_GL949750v2_alt	0	1000000
              chr19_GL949750v2_alt	1000000	1066390
              chr19_GL949751v2_alt	0	1000000
              chr19_GL949751v2_alt	1000000	1002683
              chr19_GL949752v1_alt	0	987100
              chr19_GL949753v2_alt	0	796479
              chr19_KI270865v1_alt	0	52969
              chr19_KI270866v1_alt	0	43156
              chr19_KI270867v1_alt	0	233762
              chr19_KI270868v1_alt	0	61734
              chr19_KI270882v1_alt	0	248807
              chr19_KI270883v1_alt	0	170399
              chr19_KI270884v1_alt	0	157053
              chr19_KI270885v1_alt	0	171027
              chr19_KI270886v1_alt	0	204239
              chr19_KI270887v1_alt	0	209512
              chr19_KI270888v1_alt	0	155532
              chr19_KI270889v1_alt	0	170698
              chr19_KI270890v1_alt	0	184499
              chr19_KI270891v1_alt	0	170680
              chr19_KI270914v1_alt	0	205194
              chr19_KI270915v1_alt	0	170665
              chr19_KI270916v1_alt	0	184516
              chr19_KI270917v1_alt	0	190932
              chr19_KI270918v1_alt	0	123111
              chr19_KI270919v1_alt	0	170701
              chr19_KI270920v1_alt	0	198005
              chr19_KI270921v1_alt	0	282224
              chr19_KI270922v1_alt	0	187935
              chr19_KI270923v1_alt	0	189352
              chr19_KI270929v1_alt	0	186203
              chr19_KI270930v1_alt	0	200773
              chr19_KI270931v1_alt	0	170148
              chr19_KI270932v1_alt	0	215732
              chr19_KI270933v1_alt	0	170537
              chr19_KI270938v1_alt	0	1000000
              chr19_KI270938v1_alt	1000000	1066800
              chr1_GL383518v1_alt	0	182439
              chr1_GL383519v1_alt	0	110268
              chr1_GL383520v2_alt	0	366580
              chr1_KI270706v1_random	0	175055
              chr1_KI270707v1_random	0	32032
              chr1_KI270708v1_random	0	127682
              chr1_KI270709v1_random	0	66860
              chr1_KI270710v1_random	0	40176
              chr1_KI270711v1_random	0	42210
              chr1_KI270712v1_random	0	176043
              chr1_KI270713v1_random	0	40745
              chr1_KI270714v1_random	0	41717
              chr1_KI270759v1_alt	0	425601
              chr1_KI270760v1_alt	0	109528
              chr1_KI270761v1_alt	0	165834
              chr1_KI270762v1_alt	0	354444
              chr1_KI270763v1_alt	0	911658
              chr1_KI270764v1_alt	0	50258
              chr1_KI270765v1_alt	0	185285
              chr1_KI270766v1_alt	0	256271
              chr1_KI270892v1_alt	0	162212
              chr2	0	1000000
              chr2	1000000	2000000
              chr2	2000000	3000000
              chr2	3000000	4000000
              chr2	4000000	5000000
              chr2	5000000	6000000
              chr2	6000000	7000000
              chr2	7000000	8000000
              chr2	8000000	9000000
              chr2	9000000	10000000
              chr2	10000000	11000000
              chr2	11000000	12000000
              chr2	12000000	13000000
              chr2	13000000	14000000
              chr2	14000000	15000000
              chr2	15000000	16000000
              chr2	16000000	17000000
              chr2	17000000	18000000
              chr2	18000000	19000000
              chr2	19000000	20000000
              chr2	20000000	21000000
              chr2	21000000	22000000
              chr2	22000000	23000000
              chr2	23000000	24000000
              chr2	24000000	25000000
              chr2	25000000	26000000
              chr2	26000000	27000000
              chr2	27000000	28000000
              chr2	28000000	29000000
              chr2	29000000	30000000
              chr2	30000000	31000000
              chr2	31000000	32000000
              chr2	32000000	33000000
              chr2	33000000	34000000
              chr2	34000000	35000000
              chr2	35000000	36000000
              chr2	36000000	37000000
              chr2	37000000	38000000
              chr2	38000000	39000000
              chr2	39000000	40000000
              chr2	40000000	41000000
              chr2	41000000	42000000
              chr2	42000000	43000000
              chr2	43000000	44000000
              chr2	44000000	45000000
              chr2	45000000	46000000
              chr2	46000000	47000000
              chr2	47000000	48000000
              chr2	48000000	49000000
              chr2	49000000	50000000
              chr2	50000000	51000000
              chr2	51000000	52000000
              chr2	52000000	53000000
              chr2	53000000	54000000
              chr2	54000000	55000000
              chr2	55000000	56000000
              chr2	56000000	57000000
              chr2	57000000	58000000
              chr2	58000000	59000000
              chr2	59000000	60000000
              chr2	60000000	61000000
              chr2	61000000	62000000
              chr2	62000000	63000000
              chr2	63000000	64000000
              chr2	64000000	65000000
              chr2	65000000	66000000
              chr2	66000000	67000000
              chr2	67000000	68000000
              chr2	68000000	69000000
              chr2	69000000	70000000
              chr2	70000000	71000000
              chr2	71000000	72000000
              chr2	72000000	73000000
              chr2	73000000	74000000
              chr2	74000000	75000000
              chr2	75000000	76000000
              chr2	76000000	77000000
              chr2	77000000	78000000
              chr2	78000000	79000000
              chr2	79000000	80000000
              chr2	80000000	81000000
              chr2	81000000	82000000
              chr2	82000000	83000000
              chr2	83000000	84000000
              chr2	84000000	85000000
              chr2	85000000	86000000
              chr2	86000000	87000000
              chr2	87000000	88000000
              chr2	88000000	89000000
              chr2	89000000	90000000
              chr2	90000000	91000000
              chr2	91000000	92000000
              chr2	92000000	93000000
              chr2	93000000	94000000
              chr2	94000000	95000000
              chr2	95000000	96000000
              chr2	96000000	97000000
              chr2	97000000	98000000
              chr2	98000000	99000000
              chr2	99000000	100000000
              chr2	100000000	101000000
              chr2	101000000	102000000
              chr2	102000000	103000000
              chr2	103000000	104000000
              chr2	104000000	105000000
              chr2	105000000	106000000
              chr2	106000000	107000000
              chr2	107000000	108000000
              chr2	108000000	109000000
              chr2	109000000	110000000
              chr2	110000000	111000000
              chr2	111000000	112000000
              chr2	112000000	113000000
              chr2	113000000	114000000
              chr2	114000000	115000000
              chr2	115000000	116000000
              chr2	116000000	117000000
              chr2	117000000	118000000
              chr2	118000000	119000000
              chr2	119000000	120000000
              chr2	120000000	121000000
              chr2	121000000	122000000
              chr2	122000000	123000000
              chr2	123000000	124000000
              chr2	124000000	125000000
              chr2	125000000	126000000
              chr2	126000000	127000000
              chr2	127000000	128000000
              chr2	128000000	129000000
              chr2	129000000	130000000
              chr2	130000000	131000000
              chr2	131000000	132000000
              chr2	132000000	133000000
              chr2	133000000	134000000
              chr2	134000000	135000000
              chr2	135000000	136000000
              chr2	136000000	137000000
              chr2	137000000	138000000
              chr2	138000000	139000000
              chr2	139000000	140000000
              chr2	140000000	141000000
              chr2	141000000	142000000
              chr2	142000000	143000000
              chr2	143000000	144000000
              chr2	144000000	145000000
              chr2	145000000	146000000
              chr2	146000000	147000000
              chr2	147000000	148000000
              chr2	148000000	149000000
              chr2	149000000	150000000
              chr2	150000000	151000000
              chr2	151000000	152000000
              chr2	152000000	153000000
              chr2	153000000	154000000
              chr2	154000000	155000000
              chr2	155000000	156000000
              chr2	156000000	157000000
              chr2	157000000	158000000
              chr2	158000000	159000000
              chr2	159000000	160000000
              chr2	160000000	161000000
              chr2	161000000	162000000
              chr2	162000000	163000000
              chr2	163000000	164000000
              chr2	164000000	165000000
              chr2	165000000	166000000
              chr2	166000000	167000000
              chr2	167000000	168000000
              chr2	168000000	169000000
              chr2	169000000	170000000
              chr2	170000000	171000000
              chr2	171000000	172000000
              chr2	172000000	173000000
              chr2	173000000	174000000
              chr2	174000000	175000000
              chr2	175000000	176000000
              chr2	176000000	177000000
              chr2	177000000	178000000
              chr2	178000000	179000000
              chr2	179000000	180000000
              chr2	180000000	181000000
              chr2	181000000	182000000
              chr2	182000000	183000000
              chr2	183000000	184000000
              chr2	184000000	185000000
              chr2	185000000	186000000
              chr2	186000000	187000000
              chr2	187000000	188000000
              chr2	188000000	189000000
              chr2	189000000	190000000
              chr2	190000000	191000000
              chr2	191000000	192000000
              chr2	192000000	193000000
              chr2	193000000	194000000
              chr2	194000000	195000000
              chr2	195000000	196000000
              chr2	196000000	197000000
              chr2	197000000	198000000
              chr2	198000000	199000000
              chr2	199000000	200000000
              chr2	200000000	201000000
              chr2	201000000	202000000
              chr2	202000000	203000000
              chr2	203000000	204000000
              chr2	204000000	205000000
              chr2	205000000	206000000
              chr2	206000000	207000000
              chr2	207000000	208000000
              chr2	208000000	209000000
              chr2	209000000	210000000
              chr2	210000000	211000000
              chr2	211000000	212000000
              chr2	212000000	213000000
              chr2	213000000	214000000
              chr2	214000000	215000000
              chr2	215000000	216000000
              chr2	216000000	217000000
              chr2	217000000	218000000
              chr2	218000000	219000000
              chr2	219000000	220000000
              chr2	220000000	221000000
              chr2	221000000	222000000
              chr2	222000000	223000000
              chr2	223000000	224000000
              chr2	224000000	225000000
              chr2	225000000	226000000
              chr2	226000000	227000000
              chr2	227000000	228000000
              chr2	228000000	229000000
              chr2	229000000	230000000
              chr2	230000000	231000000
              chr2	231000000	232000000
              chr2	232000000	233000000
              chr2	233000000	234000000
              chr2	234000000	235000000
              chr2	235000000	236000000
              chr2	236000000	237000000
              chr2	237000000	238000000
              chr2	238000000	239000000
              chr2	239000000	240000000
              chr2	240000000	241000000
              chr2	241000000	242000000
              chr2	242000000	242193529
              chr20	0	1000000
              chr20	1000000	2000000
              chr20	2000000	3000000
              chr20	3000000	4000000
              chr20	4000000	5000000
              chr20	5000000	6000000
              chr20	6000000	7000000
              chr20	7000000	8000000
              chr20	8000000	9000000
              chr20	9000000	10000000
              chr20	10000000	11000000
              chr20	11000000	12000000
              chr20	12000000	13000000
              chr20	13000000	14000000
              chr20	14000000	15000000
              chr20	15000000	16000000
              chr20	16000000	17000000
              chr20	17000000	18000000
              chr20	18000000	19000000
              chr20	19000000	20000000
              chr20	20000000	21000000
              chr20	21000000	22000000
              chr20	22000000	23000000
              chr20	23000000	24000000
              chr20	24000000	25000000
              chr20	25000000	26000000
              chr20	26000000	27000000
              chr20	27000000	28000000
              chr20	28000000	29000000
              chr20	29000000	30000000
              chr20	30000000	31000000
              chr20	31000000	32000000
              chr20	32000000	33000000
              chr20	33000000	34000000
              chr20	34000000	35000000
              chr20	35000000	36000000
              chr20	36000000	37000000
              chr20	37000000	38000000
              chr20	38000000	39000000
              chr20	39000000	40000000
              chr20	40000000	41000000
              chr20	41000000	42000000
              chr20	42000000	43000000
              chr20	43000000	44000000
              chr20	44000000	45000000
              chr20	45000000	46000000
              chr20	46000000	47000000
              chr20	47000000	48000000
              chr20	48000000	49000000
              chr20	49000000	50000000
              chr20	50000000	51000000
              chr20	51000000	52000000
              chr20	52000000	53000000
              chr20	53000000	54000000
              chr20	54000000	55000000
              chr20	55000000	56000000
              chr20	56000000	57000000
              chr20	57000000	58000000
              chr20	58000000	59000000
              chr20	59000000	60000000
              chr20	60000000	61000000
              chr20	61000000	62000000
              chr20	62000000	63000000
              chr20	63000000	64000000
              chr20	64000000	64444167
              chr20_GL383577v2_alt	0	128386
              chr20_KI270869v1_alt	0	118774
              chr20_KI270870v1_alt	0	183433
              chr20_KI270871v1_alt	0	58661
              chr21	0	1000000
              chr21	1000000	2000000
              chr21	2000000	3000000
              chr21	3000000	4000000
              chr21	4000000	5000000
              chr21	5000000	6000000
              chr21	6000000	7000000
              chr21	7000000	8000000
              chr21	8000000	9000000
              chr21	9000000	10000000
              chr21	10000000	11000000
              chr21	11000000	12000000
              chr21	12000000	13000000
              chr21	13000000	14000000
              chr21	14000000	15000000
              chr21	15000000	16000000
              chr21	16000000	17000000
              chr21	17000000	18000000
              chr21	18000000	19000000
              chr21	19000000	20000000
              chr21	20000000	21000000
              chr21	21000000	22000000
              chr21	22000000	23000000
              chr21	23000000	24000000
              chr21	24000000	25000000
              chr21	25000000	26000000
              chr21	26000000	27000000
              chr21	27000000	28000000
              chr21	28000000	29000000
              chr21	29000000	30000000
              chr21	30000000	31000000
              chr21	31000000	32000000
              chr21	32000000	33000000
              chr21	33000000	34000000
              chr21	34000000	35000000
              chr21	35000000	36000000
              chr21	36000000	37000000
              chr21	37000000	38000000
              chr21	38000000	39000000
              chr21	39000000	40000000
              chr21	40000000	41000000
              chr21	41000000	42000000
              chr21	42000000	43000000
              chr21	43000000	44000000
              chr21	44000000	45000000
              chr21	45000000	46000000
              chr21	46000000	46709983
              chr21_GL383578v2_alt	0	63917
              chr21_GL383579v2_alt	0	201197
              chr21_GL383580v2_alt	0	74653
              chr21_GL383581v2_alt	0	116689
              chr21_KI270872v1_alt	0	82692
              chr21_KI270873v1_alt	0	143900
              chr21_KI270874v1_alt	0	166743
              chr22	0	1000000
              chr22	1000000	2000000
              chr22	2000000	3000000
              chr22	3000000	4000000
              chr22	4000000	5000000
              chr22	5000000	6000000
              chr22	6000000	7000000
              chr22	7000000	8000000
              chr22	8000000	9000000
              chr22	9000000	10000000
              chr22	10000000	11000000
              chr22	11000000	12000000
              chr22	12000000	13000000
              chr22	13000000	14000000
              chr22	14000000	15000000
              chr22	15000000	16000000
              chr22	16000000	17000000
              chr22	17000000	18000000
              chr22	18000000	19000000
              chr22	19000000	20000000
              chr22	20000000	21000000
              chr22	21000000	22000000
              chr22	22000000	23000000
              chr22	23000000	24000000
              chr22	24000000	25000000
              chr22	25000000	26000000
              chr22	26000000	27000000
              chr22	27000000	28000000
              chr22	28000000	29000000
              chr22	29000000	30000000
              chr22	30000000	31000000
              chr22	31000000	32000000
              chr22	32000000	33000000
              chr22	33000000	34000000
              chr22	34000000	35000000
              chr22	35000000	36000000
              chr22	36000000	37000000
              chr22	37000000	38000000
              chr22	38000000	39000000
              chr22	39000000	40000000
              chr22	40000000	41000000
              chr22	41000000	42000000
              chr22	42000000	43000000
              chr22	43000000	44000000
              chr22	44000000	45000000
              chr22	45000000	46000000
              chr22	46000000	47000000
              chr22	47000000	48000000
              chr22	48000000	49000000
              chr22	49000000	50000000
              chr22	50000000	50818468
              chr22_GL383582v2_alt	0	162811
              chr22_GL383583v2_alt	0	96924
              chr22_KB663609v1_alt	0	74013
              chr22_KI270731v1_random	0	150754
              chr22_KI270732v1_random	0	41543
              chr22_KI270733v1_random	0	179772
              chr22_KI270734v1_random	0	165050
              chr22_KI270735v1_random	0	42811
              chr22_KI270736v1_random	0	181920
              chr22_KI270737v1_random	0	103838
              chr22_KI270738v1_random	0	99375
              chr22_KI270739v1_random	0	73985
              chr22_KI270875v1_alt	0	259914
              chr22_KI270876v1_alt	0	263666
              chr22_KI270877v1_alt	0	101331
              chr22_KI270878v1_alt	0	186262
              chr22_KI270879v1_alt	0	304135
              chr22_KI270928v1_alt	0	176103
              chr2_GL383521v1_alt	0	143390
              chr2_GL383522v1_alt	0	123821
              chr2_GL582966v2_alt	0	96131
              chr2_KI270715v1_random	0	161471
              chr2_KI270716v1_random	0	153799
              chr2_KI270767v1_alt	0	161578
              chr2_KI270768v1_alt	0	110099
              chr2_KI270769v1_alt	0	120616
              chr2_KI270770v1_alt	0	136240
              chr2_KI270771v1_alt	0	110395
              chr2_KI270772v1_alt	0	133041
              chr2_KI270773v1_alt	0	70887
              chr2_KI270774v1_alt	0	223625
              chr2_KI270775v1_alt	0	138019
              chr2_KI270776v1_alt	0	174166
              chr2_KI270893v1_alt	0	161218
              chr2_KI270894v1_alt	0	214158
              chr3	0	1000000
              chr3	1000000	2000000
              chr3	2000000	3000000
              chr3	3000000	4000000
              chr3	4000000	5000000
              chr3	5000000	6000000
              chr3	6000000	7000000
              chr3	7000000	8000000
              chr3	8000000	9000000
              chr3	9000000	10000000
              chr3	10000000	11000000
              chr3	11000000	12000000
              chr3	12000000	13000000
              chr3	13000000	14000000
              chr3	14000000	15000000
              chr3	15000000	16000000
              chr3	16000000	17000000
              chr3	17000000	18000000
              chr3	18000000	19000000
              chr3	19000000	20000000
              chr3	20000000	21000000
              chr3	21000000	22000000
              chr3	22000000	23000000
              chr3	23000000	24000000
              chr3	24000000	25000000
              chr3	25000000	26000000
              chr3	26000000	27000000
              chr3	27000000	28000000
              chr3	28000000	29000000
              chr3	29000000	30000000
              chr3	30000000	31000000
              chr3	31000000	32000000
              chr3	32000000	33000000
              chr3	33000000	34000000
              chr3	34000000	35000000
              chr3	35000000	36000000
              chr3	36000000	37000000
              chr3	37000000	38000000
              chr3	38000000	39000000
              chr3	39000000	40000000
              chr3	40000000	41000000
              chr3	41000000	42000000
              chr3	42000000	43000000
              chr3	43000000	44000000
              chr3	44000000	45000000
              chr3	45000000	46000000
              chr3	46000000	47000000
              chr3	47000000	48000000
              chr3	48000000	49000000
              chr3	49000000	50000000
              chr3	50000000	51000000
              chr3	51000000	52000000
              chr3	52000000	53000000
              chr3	53000000	54000000
              chr3	54000000	55000000
              chr3	55000000	56000000
              chr3	56000000	57000000
              chr3	57000000	58000000
              chr3	58000000	59000000
              chr3	59000000	60000000
              chr3	60000000	61000000
              chr3	61000000	62000000
              chr3	62000000	63000000
              chr3	63000000	64000000
              chr3	64000000	65000000
              chr3	65000000	66000000
              chr3	66000000	67000000
              chr3	67000000	68000000
              chr3	68000000	69000000
              chr3	69000000	70000000
              chr3	70000000	71000000
              chr3	71000000	72000000
              chr3	72000000	73000000
              chr3	73000000	74000000
              chr3	74000000	75000000
              chr3	75000000	76000000
              chr3	76000000	77000000
              chr3	77000000	78000000
              chr3	78000000	79000000
              chr3	79000000	80000000
              chr3	80000000	81000000
              chr3	81000000	82000000
              chr3	82000000	83000000
              chr3	83000000	84000000
              chr3	84000000	85000000
              chr3	85000000	86000000
              chr3	86000000	87000000
              chr3	87000000	88000000
              chr3	88000000	89000000
              chr3	89000000	90000000
              chr3	90000000	91000000
              chr3	91000000	92000000
              chr3	92000000	93000000
              chr3	93000000	94000000
              chr3	94000000	95000000
              chr3	95000000	96000000
              chr3	96000000	97000000
              chr3	97000000	98000000
              chr3	98000000	99000000
              chr3	99000000	100000000
              chr3	100000000	101000000
              chr3	101000000	102000000
              chr3	102000000	103000000
              chr3	103000000	104000000
              chr3	104000000	105000000
              chr3	105000000	106000000
              chr3	106000000	107000000
              chr3	107000000	108000000
              chr3	108000000	109000000
              chr3	109000000	110000000
              chr3	110000000	111000000
              chr3	111000000	112000000
              chr3	112000000	113000000
              chr3	113000000	114000000
              chr3	114000000	115000000
              chr3	115000000	116000000
              chr3	116000000	117000000
              chr3	117000000	118000000
              chr3	118000000	119000000
              chr3	119000000	120000000
              chr3	120000000	121000000
              chr3	121000000	122000000
              chr3	122000000	123000000
              chr3	123000000	124000000
              chr3	124000000	125000000
              chr3	125000000	126000000
              chr3	126000000	127000000
              chr3	127000000	128000000
              chr3	128000000	129000000
              chr3	129000000	130000000
              chr3	130000000	131000000
              chr3	131000000	132000000
              chr3	132000000	133000000
              chr3	133000000	134000000
              chr3	134000000	135000000
              chr3	135000000	136000000
              chr3	136000000	137000000
              chr3	137000000	138000000
              chr3	138000000	139000000
              chr3	139000000	140000000
              chr3	140000000	141000000
              chr3	141000000	142000000
              chr3	142000000	143000000
              chr3	143000000	144000000
              chr3	144000000	145000000
              chr3	145000000	146000000
              chr3	146000000	147000000
              chr3	147000000	148000000
              chr3	148000000	149000000
              chr3	149000000	150000000
              chr3	150000000	151000000
              chr3	151000000	152000000
              chr3	152000000	153000000
              chr3	153000000	154000000
              chr3	154000000	155000000
              chr3	155000000	156000000
              chr3	156000000	157000000
              chr3	157000000	158000000
              chr3	158000000	159000000
              chr3	159000000	160000000
              chr3	160000000	161000000
              chr3	161000000	162000000
              chr3	162000000	163000000
              chr3	163000000	164000000
              chr3	164000000	165000000
              chr3	165000000	166000000
              chr3	166000000	167000000
              chr3	167000000	168000000
              chr3	168000000	169000000
              chr3	169000000	170000000
              chr3	170000000	171000000
              chr3	171000000	172000000
              chr3	172000000	173000000
              chr3	173000000	174000000
              chr3	174000000	175000000
              chr3	175000000	176000000
              chr3	176000000	177000000
              chr3	177000000	178000000
              chr3	178000000	179000000
              chr3	179000000	180000000
              chr3	180000000	181000000
              chr3	181000000	182000000
              chr3	182000000	183000000
              chr3	183000000	184000000
              chr3	184000000	185000000
              chr3	185000000	186000000
              chr3	186000000	187000000
              chr3	187000000	188000000
              chr3	188000000	189000000
              chr3	189000000	190000000
              chr3	190000000	191000000
              chr3	191000000	192000000
              chr3	192000000	193000000
              chr3	193000000	194000000
              chr3	194000000	195000000
              chr3	195000000	196000000
              chr3	196000000	197000000
              chr3	197000000	198000000
              chr3	198000000	198295559
              chr3_GL000221v1_random	0	155397
              chr3_GL383526v1_alt	0	180671
              chr3_JH636055v2_alt	0	173151
              chr3_KI270777v1_alt	0	173649
              chr3_KI270778v1_alt	0	248252
              chr3_KI270779v1_alt	0	205312
              chr3_KI270780v1_alt	0	224108
              chr3_KI270781v1_alt	0	113034
              chr3_KI270782v1_alt	0	162429
              chr3_KI270783v1_alt	0	109187
              chr3_KI270784v1_alt	0	184404
              chr3_KI270895v1_alt	0	162896
              chr3_KI270924v1_alt	0	166540
              chr3_KI270934v1_alt	0	163458
              chr3_KI270935v1_alt	0	197351
              chr3_KI270936v1_alt	0	164170
              chr3_KI270937v1_alt	0	165607
              chr4	0	1000000
              chr4	1000000	2000000
              chr4	2000000	3000000
              chr4	3000000	4000000
              chr4	4000000	5000000
              chr4	5000000	6000000
              chr4	6000000	7000000
              chr4	7000000	8000000
              chr4	8000000	9000000
              chr4	9000000	10000000
              chr4	10000000	11000000
              chr4	11000000	12000000
              chr4	12000000	13000000
              chr4	13000000	14000000
              chr4	14000000	15000000
              chr4	15000000	16000000
              chr4	16000000	17000000
              chr4	17000000	18000000
              chr4	18000000	19000000
              chr4	19000000	20000000
              chr4	20000000	21000000
              chr4	21000000	22000000
              chr4	22000000	23000000
              chr4	23000000	24000000
              chr4	24000000	25000000
              chr4	25000000	26000000
              chr4	26000000	27000000
              chr4	27000000	28000000
              chr4	28000000	29000000
              chr4	29000000	30000000
              chr4	30000000	31000000
              chr4	31000000	32000000
              chr4	32000000	33000000
              chr4	33000000	34000000
              chr4	34000000	35000000
              chr4	35000000	36000000
              chr4	36000000	37000000
              chr4	37000000	38000000
              chr4	38000000	39000000
              chr4	39000000	40000000
              chr4	40000000	41000000
              chr4	41000000	42000000
              chr4	42000000	43000000
              chr4	43000000	44000000
              chr4	44000000	45000000
              chr4	45000000	46000000
              chr4	46000000	47000000
              chr4	47000000	48000000
              chr4	48000000	49000000
              chr4	49000000	50000000
              chr4	50000000	51000000
              chr4	51000000	52000000
              chr4	52000000	53000000
              chr4	53000000	54000000
              chr4	54000000	55000000
              chr4	55000000	56000000
              chr4	56000000	57000000
              chr4	57000000	58000000
              chr4	58000000	59000000
              chr4	59000000	60000000
              chr4	60000000	61000000
              chr4	61000000	62000000
              chr4	62000000	63000000
              chr4	63000000	64000000
              chr4	64000000	65000000
              chr4	65000000	66000000
              chr4	66000000	67000000
              chr4	67000000	68000000
              chr4	68000000	69000000
              chr4	69000000	70000000
              chr4	70000000	71000000
              chr4	71000000	72000000
              chr4	72000000	73000000
              chr4	73000000	74000000
              chr4	74000000	75000000
              chr4	75000000	76000000
              chr4	76000000	77000000
              chr4	77000000	78000000
              chr4	78000000	79000000
              chr4	79000000	80000000
              chr4	80000000	81000000
              chr4	81000000	82000000
              chr4	82000000	83000000
              chr4	83000000	84000000
              chr4	84000000	85000000
              chr4	85000000	86000000
              chr4	86000000	87000000
              chr4	87000000	88000000
              chr4	88000000	89000000
              chr4	89000000	90000000
              chr4	90000000	91000000
              chr4	91000000	92000000
              chr4	92000000	93000000
              chr4	93000000	94000000
              chr4	94000000	95000000
              chr4	95000000	96000000
              chr4	96000000	97000000
              chr4	97000000	98000000
              chr4	98000000	99000000
              chr4	99000000	100000000
              chr4	100000000	101000000
              chr4	101000000	102000000
              chr4	102000000	103000000
              chr4	103000000	104000000
              chr4	104000000	105000000
              chr4	105000000	106000000
              chr4	106000000	107000000
              chr4	107000000	108000000
              chr4	108000000	109000000
              chr4	109000000	110000000
              chr4	110000000	111000000
              chr4	111000000	112000000
              chr4	112000000	113000000
              chr4	113000000	114000000
              chr4	114000000	115000000
              chr4	115000000	116000000
              chr4	116000000	117000000
              chr4	117000000	118000000
              chr4	118000000	119000000
              chr4	119000000	120000000
              chr4	120000000	121000000
              chr4	121000000	122000000
              chr4	122000000	123000000
              chr4	123000000	124000000
              chr4	124000000	125000000
              chr4	125000000	126000000
              chr4	126000000	127000000
              chr4	127000000	128000000
              chr4	128000000	129000000
              chr4	129000000	130000000
              chr4	130000000	131000000
              chr4	131000000	132000000
              chr4	132000000	133000000
              chr4	133000000	134000000
              chr4	134000000	135000000
              chr4	135000000	136000000
              chr4	136000000	137000000
              chr4	137000000	138000000
              chr4	138000000	139000000
              chr4	139000000	140000000
              chr4	140000000	141000000
              chr4	141000000	142000000
              chr4	142000000	143000000
              chr4	143000000	144000000
              chr4	144000000	145000000
              chr4	145000000	146000000
              chr4	146000000	147000000
              chr4	147000000	148000000
              chr4	148000000	149000000
              chr4	149000000	150000000
              chr4	150000000	151000000
              chr4	151000000	152000000
              chr4	152000000	153000000
              chr4	153000000	154000000
              chr4	154000000	155000000
              chr4	155000000	156000000
              chr4	156000000	157000000
              chr4	157000000	158000000
              chr4	158000000	159000000
              chr4	159000000	160000000
              chr4	160000000	161000000
              chr4	161000000	162000000
              chr4	162000000	163000000
              chr4	163000000	164000000
              chr4	164000000	165000000
              chr4	165000000	166000000
              chr4	166000000	167000000
              chr4	167000000	168000000
              chr4	168000000	169000000
              chr4	169000000	170000000
              chr4	170000000	171000000
              chr4	171000000	172000000
              chr4	172000000	173000000
              chr4	173000000	174000000
              chr4	174000000	175000000
              chr4	175000000	176000000
              chr4	176000000	177000000
              chr4	177000000	178000000
              chr4	178000000	179000000
              chr4	179000000	180000000
              chr4	180000000	181000000
              chr4	181000000	182000000
              chr4	182000000	183000000
              chr4	183000000	184000000
              chr4	184000000	185000000
              chr4	185000000	186000000
              chr4	186000000	187000000
              chr4	187000000	188000000
              chr4	188000000	189000000
              chr4	189000000	190000000
              chr4	190000000	190214555
              chr4_GL000008v2_random	0	209709
              chr4_GL000257v2_alt	0	586476
              chr4_GL383527v1_alt	0	164536
              chr4_GL383528v1_alt	0	376187
              chr4_KI270785v1_alt	0	119912
              chr4_KI270786v1_alt	0	244096
              chr4_KI270787v1_alt	0	111943
              chr4_KI270788v1_alt	0	158965
              chr4_KI270789v1_alt	0	205944
              chr4_KI270790v1_alt	0	220246
              chr4_KI270896v1_alt	0	378547
              chr4_KI270925v1_alt	0	555799
              chr5	0	1000000
              chr5	1000000	2000000
              chr5	2000000	3000000
              chr5	3000000	4000000
              chr5	4000000	5000000
              chr5	5000000	6000000
              chr5	6000000	7000000
              chr5	7000000	8000000
              chr5	8000000	9000000
              chr5	9000000	10000000
              chr5	10000000	11000000
              chr5	11000000	12000000
              chr5	12000000	13000000
              chr5	13000000	14000000
              chr5	14000000	15000000
              chr5	15000000	16000000
              chr5	16000000	17000000
              chr5	17000000	18000000
              chr5	18000000	19000000
              chr5	19000000	20000000
              chr5	20000000	21000000
              chr5	21000000	22000000
              chr5	22000000	23000000
              chr5	23000000	24000000
              chr5	24000000	25000000
              chr5	25000000	26000000
              chr5	26000000	27000000
              chr5	27000000	28000000
              chr5	28000000	29000000
              chr5	29000000	30000000
              chr5	30000000	31000000
              chr5	31000000	32000000
              chr5	32000000	33000000
              chr5	33000000	34000000
              chr5	34000000	35000000
              chr5	35000000	36000000
              chr5	36000000	37000000
              chr5	37000000	38000000
              chr5	38000000	39000000
              chr5	39000000	40000000
              chr5	40000000	41000000
              chr5	41000000	42000000
              chr5	42000000	43000000
              chr5	43000000	44000000
              chr5	44000000	45000000
              chr5	45000000	46000000
              chr5	46000000	47000000
              chr5	47000000	48000000
              chr5	48000000	49000000
              chr5	49000000	50000000
              chr5	50000000	51000000
              chr5	51000000	52000000
              chr5	52000000	53000000
              chr5	53000000	54000000
              chr5	54000000	55000000
              chr5	55000000	56000000
              chr5	56000000	57000000
              chr5	57000000	58000000
              chr5	58000000	59000000
              chr5	59000000	60000000
              chr5	60000000	61000000
              chr5	61000000	62000000
              chr5	62000000	63000000
              chr5	63000000	64000000
              chr5	64000000	65000000
              chr5	65000000	66000000
              chr5	66000000	67000000
              chr5	67000000	68000000
              chr5	68000000	69000000
              chr5	69000000	70000000
              chr5	70000000	71000000
              chr5	71000000	72000000
              chr5	72000000	73000000
              chr5	73000000	74000000
              chr5	74000000	75000000
              chr5	75000000	76000000
              chr5	76000000	77000000
              chr5	77000000	78000000
              chr5	78000000	79000000
              chr5	79000000	80000000
              chr5	80000000	81000000
              chr5	81000000	82000000
              chr5	82000000	83000000
              chr5	83000000	84000000
              chr5	84000000	85000000
              chr5	85000000	86000000
              chr5	86000000	87000000
              chr5	87000000	88000000
              chr5	88000000	89000000
              chr5	89000000	90000000
              chr5	90000000	91000000
              chr5	91000000	92000000
              chr5	92000000	93000000
              chr5	93000000	94000000
              chr5	94000000	95000000
              chr5	95000000	96000000
              chr5	96000000	97000000
              chr5	97000000	98000000
              chr5	98000000	99000000
              chr5	99000000	100000000
              chr5	100000000	101000000
              chr5	101000000	102000000
              chr5	102000000	103000000
              chr5	103000000	104000000
              chr5	104000000	105000000
              chr5	105000000	106000000
              chr5	106000000	107000000
              chr5	107000000	108000000
              chr5	108000000	109000000
              chr5	109000000	110000000
              chr5	110000000	111000000
              chr5	111000000	112000000
              chr5	112000000	113000000
              chr5	113000000	114000000
              chr5	114000000	115000000
              chr5	115000000	116000000
              chr5	116000000	117000000
              chr5	117000000	118000000
              chr5	118000000	119000000
              chr5	119000000	120000000
              chr5	120000000	121000000
              chr5	121000000	122000000
              chr5	122000000	123000000
              chr5	123000000	124000000
              chr5	124000000	125000000
              chr5	125000000	126000000
              chr5	126000000	127000000
              chr5	127000000	128000000
              chr5	128000000	129000000
              chr5	129000000	130000000
              chr5	130000000	131000000
              chr5	131000000	132000000
              chr5	132000000	133000000
              chr5	133000000	134000000
              chr5	134000000	135000000
              chr5	135000000	136000000
              chr5	136000000	137000000
              chr5	137000000	138000000
              chr5	138000000	139000000
              chr5	139000000	140000000
              chr5	140000000	141000000
              chr5	141000000	142000000
              chr5	142000000	143000000
              chr5	143000000	144000000
              chr5	144000000	145000000
              chr5	145000000	146000000
              chr5	146000000	147000000
              chr5	147000000	148000000
              chr5	148000000	149000000
              chr5	149000000	150000000
              chr5	150000000	151000000
              chr5	151000000	152000000
              chr5	152000000	153000000
              chr5	153000000	154000000
              chr5	154000000	155000000
              chr5	155000000	156000000
              chr5	156000000	157000000
              chr5	157000000	158000000
              chr5	158000000	159000000
              chr5	159000000	160000000
              chr5	160000000	161000000
              chr5	161000000	162000000
              chr5	162000000	163000000
              chr5	163000000	164000000
              chr5	164000000	165000000
              chr5	165000000	166000000
              chr5	166000000	167000000
              chr5	167000000	168000000
              chr5	168000000	169000000
              chr5	169000000	170000000
              chr5	170000000	171000000
              chr5	171000000	172000000
              chr5	172000000	173000000
              chr5	173000000	174000000
              chr5	174000000	175000000
              chr5	175000000	176000000
              chr5	176000000	177000000
              chr5	177000000	178000000
              chr5	178000000	179000000
              chr5	179000000	180000000
              chr5	180000000	181000000
              chr5	181000000	181538259
              chr5_GL000208v1_random	0	92689
              chr5_GL339449v2_alt	0	1000000
              chr5_GL339449v2_alt	1000000	1612928
              chr5_GL383530v1_alt	0	101241
              chr5_GL383531v1_alt	0	173459
              chr5_GL383532v1_alt	0	82728
              chr5_GL949742v1_alt	0	226852
              chr5_KI270791v1_alt	0	195710
              chr5_KI270792v1_alt	0	179043
              chr5_KI270793v1_alt	0	126136
              chr5_KI270794v1_alt	0	164558
              chr5_KI270795v1_alt	0	131892
              chr5_KI270796v1_alt	0	172708
              chr5_KI270897v1_alt	0	1000000
              chr5_KI270897v1_alt	1000000	1144418
              chr5_KI270898v1_alt	0	130957
              chr6	0	1000000
              chr6	1000000	2000000
              chr6	2000000	3000000
              chr6	3000000	4000000
              chr6	4000000	5000000
              chr6	5000000	6000000
              chr6	6000000	7000000
              chr6	7000000	8000000
              chr6	8000000	9000000
              chr6	9000000	10000000
              chr6	10000000	11000000
              chr6	11000000	12000000
              chr6	12000000	13000000
              chr6	13000000	14000000
              chr6	14000000	15000000
              chr6	15000000	16000000
              chr6	16000000	17000000
              chr6	17000000	18000000
              chr6	18000000	19000000
              chr6	19000000	20000000
              chr6	20000000	21000000
              chr6	21000000	22000000
              chr6	22000000	23000000
              chr6	23000000	24000000
              chr6	24000000	25000000
              chr6	25000000	26000000
              chr6	26000000	27000000
              chr6	27000000	28000000
              chr6	28000000	29000000
              chr6	29000000	30000000
              chr6	30000000	31000000
              chr6	31000000	32000000
              chr6	32000000	33000000
              chr6	33000000	34000000
              chr6	34000000	35000000
              chr6	35000000	36000000
              chr6	36000000	37000000
              chr6	37000000	38000000
              chr6	38000000	39000000
              chr6	39000000	40000000
              chr6	40000000	41000000
              chr6	41000000	42000000
              chr6	42000000	43000000
              chr6	43000000	44000000
              chr6	44000000	45000000
              chr6	45000000	46000000
              chr6	46000000	47000000
              chr6	47000000	48000000
              chr6	48000000	49000000
              chr6	49000000	50000000
              chr6	50000000	51000000
              chr6	51000000	52000000
              chr6	52000000	53000000
              chr6	53000000	54000000
              chr6	54000000	55000000
              chr6	55000000	56000000
              chr6	56000000	57000000
              chr6	57000000	58000000
              chr6	58000000	59000000
              chr6	59000000	60000000
              chr6	60000000	61000000
              chr6	61000000	62000000
              chr6	62000000	63000000
              chr6	63000000	64000000
              chr6	64000000	65000000
              chr6	65000000	66000000
              chr6	66000000	67000000
              chr6	67000000	68000000
              chr6	68000000	69000000
              chr6	69000000	70000000
              chr6	70000000	71000000
              chr6	71000000	72000000
              chr6	72000000	73000000
              chr6	73000000	74000000
              chr6	74000000	75000000
              chr6	75000000	76000000
              chr6	76000000	77000000
              chr6	77000000	78000000
              chr6	78000000	79000000
              chr6	79000000	80000000
              chr6	80000000	81000000
              chr6	81000000	82000000
              chr6	82000000	83000000
              chr6	83000000	84000000
              chr6	84000000	85000000
              chr6	85000000	86000000
              chr6	86000000	87000000
              chr6	87000000	88000000
              chr6	88000000	89000000
              chr6	89000000	90000000
              chr6	90000000	91000000
              chr6	91000000	92000000
              chr6	92000000	93000000
              chr6	93000000	94000000
              chr6	94000000	95000000
              chr6	95000000	96000000
              chr6	96000000	97000000
              chr6	97000000	98000000
              chr6	98000000	99000000
              chr6	99000000	100000000
              chr6	100000000	101000000
              chr6	101000000	102000000
              chr6	102000000	103000000
              chr6	103000000	104000000
              chr6	104000000	105000000
              chr6	105000000	106000000
              chr6	106000000	107000000
              chr6	107000000	108000000
              chr6	108000000	109000000
              chr6	109000000	110000000
              chr6	110000000	111000000
              chr6	111000000	112000000
              chr6	112000000	113000000
              chr6	113000000	114000000
              chr6	114000000	115000000
              chr6	115000000	116000000
              chr6	116000000	117000000
              chr6	117000000	118000000
              chr6	118000000	119000000
              chr6	119000000	120000000
              chr6	120000000	121000000
              chr6	121000000	122000000
              chr6	122000000	123000000
              chr6	123000000	124000000
              chr6	124000000	125000000
              chr6	125000000	126000000
              chr6	126000000	127000000
              chr6	127000000	128000000
              chr6	128000000	129000000
              chr6	129000000	130000000
              chr6	130000000	131000000
              chr6	131000000	132000000
              chr6	132000000	133000000
              chr6	133000000	134000000
              chr6	134000000	135000000
              chr6	135000000	136000000
              chr6	136000000	137000000
              chr6	137000000	138000000
              chr6	138000000	139000000
              chr6	139000000	140000000
              chr6	140000000	141000000
              chr6	141000000	142000000
              chr6	142000000	143000000
              chr6	143000000	144000000
              chr6	144000000	145000000
              chr6	145000000	146000000
              chr6	146000000	147000000
              chr6	147000000	148000000
              chr6	148000000	149000000
              chr6	149000000	150000000
              chr6	150000000	151000000
              chr6	151000000	152000000
              chr6	152000000	153000000
              chr6	153000000	154000000
              chr6	154000000	155000000
              chr6	155000000	156000000
              chr6	156000000	157000000
              chr6	157000000	158000000
              chr6	158000000	159000000
              chr6	159000000	160000000
              chr6	160000000	161000000
              chr6	161000000	162000000
              chr6	162000000	163000000
              chr6	163000000	164000000
              chr6	164000000	165000000
              chr6	165000000	166000000
              chr6	166000000	167000000
              chr6	167000000	168000000
              chr6	168000000	169000000
              chr6	169000000	170000000
              chr6	170000000	170805979
              chr6_GL000250v2_alt	0	1000000
              chr6_GL000250v2_alt	1000000	2000000
              chr6_GL000250v2_alt	2000000	3000000
              chr6_GL000250v2_alt	3000000	4000000
              chr6_GL000250v2_alt	4000000	4672374
              chr6_GL000251v2_alt	0	1000000
              chr6_GL000251v2_alt	1000000	2000000
              chr6_GL000251v2_alt	2000000	3000000
              chr6_GL000251v2_alt	3000000	4000000
              chr6_GL000251v2_alt	4000000	4795265
              chr6_GL000252v2_alt	0	1000000
              chr6_GL000252v2_alt	1000000	2000000
              chr6_GL000252v2_alt	2000000	3000000
              chr6_GL000252v2_alt	3000000	4000000
              chr6_GL000252v2_alt	4000000	4604811
              chr6_GL000253v2_alt	0	1000000
              chr6_GL000253v2_alt	1000000	2000000
              chr6_GL000253v2_alt	2000000	3000000
              chr6_GL000253v2_alt	3000000	4000000
              chr6_GL000253v2_alt	4000000	4677643
              chr6_GL000254v2_alt	0	1000000
              chr6_GL000254v2_alt	1000000	2000000
              chr6_GL000254v2_alt	2000000	3000000
              chr6_GL000254v2_alt	3000000	4000000
              chr6_GL000254v2_alt	4000000	4827813
              chr6_GL000255v2_alt	0	1000000
              chr6_GL000255v2_alt	1000000	2000000
              chr6_GL000255v2_alt	2000000	3000000
              chr6_GL000255v2_alt	3000000	4000000
              chr6_GL000255v2_alt	4000000	4606388
              chr6_GL000256v2_alt	0	1000000
              chr6_GL000256v2_alt	1000000	2000000
              chr6_GL000256v2_alt	2000000	3000000
              chr6_GL000256v2_alt	3000000	4000000
              chr6_GL000256v2_alt	4000000	4929269
              chr6_GL383533v1_alt	0	124736
              chr6_KB021644v2_alt	0	185823
              chr6_KI270758v1_alt	0	76752
              chr6_KI270797v1_alt	0	197536
              chr6_KI270798v1_alt	0	271782
              chr6_KI270799v1_alt	0	152148
              chr6_KI270800v1_alt	0	175808
              chr6_KI270801v1_alt	0	870480
              chr6_KI270802v1_alt	0	75005
              chr7	0	1000000
              chr7	1000000	2000000
              chr7	2000000	3000000
              chr7	3000000	4000000
              chr7	4000000	5000000
              chr7	5000000	6000000
              chr7	6000000	7000000
              chr7	7000000	8000000
              chr7	8000000	9000000
              chr7	9000000	10000000
              chr7	10000000	11000000
              chr7	11000000	12000000
              chr7	12000000	13000000
              chr7	13000000	14000000
              chr7	14000000	15000000
              chr7	15000000	16000000
              chr7	16000000	17000000
              chr7	17000000	18000000
              chr7	18000000	19000000
              chr7	19000000	20000000
              chr7	20000000	21000000
              chr7	21000000	22000000
              chr7	22000000	23000000
              chr7	23000000	24000000
              chr7	24000000	25000000
              chr7	25000000	26000000
              chr7	26000000	27000000
              chr7	27000000	28000000
              chr7	28000000	29000000
              chr7	29000000	30000000
              chr7	30000000	31000000
              chr7	31000000	32000000
              chr7	32000000	33000000
              chr7	33000000	34000000
              chr7	34000000	35000000
              chr7	35000000	36000000
              chr7	36000000	37000000
              chr7	37000000	38000000
              chr7	38000000	39000000
              chr7	39000000	40000000
              chr7	40000000	41000000
              chr7	41000000	42000000
              chr7	42000000	43000000
              chr7	43000000	44000000
              chr7	44000000	45000000
              chr7	45000000	46000000
              chr7	46000000	47000000
              chr7	47000000	48000000
              chr7	48000000	49000000
              chr7	49000000	50000000
              chr7	50000000	51000000
              chr7	51000000	52000000
              chr7	52000000	53000000
              chr7	53000000	54000000
              chr7	54000000	55000000
              chr7	55000000	56000000
              chr7	56000000	57000000
              chr7	57000000	58000000
              chr7	58000000	59000000
              chr7	59000000	60000000
              chr7	60000000	61000000
              chr7	61000000	62000000
              chr7	62000000	63000000
              chr7	63000000	64000000
              chr7	64000000	65000000
              chr7	65000000	66000000
              chr7	66000000	67000000
              chr7	67000000	68000000
              chr7	68000000	69000000
              chr7	69000000	70000000
              chr7	70000000	71000000
              chr7	71000000	72000000
              chr7	72000000	73000000
              chr7	73000000	74000000
              chr7	74000000	75000000
              chr7	75000000	76000000
              chr7	76000000	77000000
              chr7	77000000	78000000
              chr7	78000000	79000000
              chr7	79000000	80000000
              chr7	80000000	81000000
              chr7	81000000	82000000
              chr7	82000000	83000000
              chr7	83000000	84000000
              chr7	84000000	85000000
              chr7	85000000	86000000
              chr7	86000000	87000000
              chr7	87000000	88000000
              chr7	88000000	89000000
              chr7	89000000	90000000
              chr7	90000000	91000000
              chr7	91000000	92000000
              chr7	92000000	93000000
              chr7	93000000	94000000
              chr7	94000000	95000000
              chr7	95000000	96000000
              chr7	96000000	97000000
              chr7	97000000	98000000
              chr7	98000000	99000000
              chr7	99000000	100000000
              chr7	100000000	101000000
              chr7	101000000	102000000
              chr7	102000000	103000000
              chr7	103000000	104000000
              chr7	104000000	105000000
              chr7	105000000	106000000
              chr7	106000000	107000000
              chr7	107000000	108000000
              chr7	108000000	109000000
              chr7	109000000	110000000
              chr7	110000000	111000000
              chr7	111000000	112000000
              chr7	112000000	113000000
              chr7	113000000	114000000
              chr7	114000000	115000000
              chr7	115000000	116000000
              chr7	116000000	117000000
              chr7	117000000	118000000
              chr7	118000000	119000000
              chr7	119000000	120000000
              chr7	120000000	121000000
              chr7	121000000	122000000
              chr7	122000000	123000000
              chr7	123000000	124000000
              chr7	124000000	125000000
              chr7	125000000	126000000
              chr7	126000000	127000000
              chr7	127000000	128000000
              chr7	128000000	129000000
              chr7	129000000	130000000
              chr7	130000000	131000000
              chr7	131000000	132000000
              chr7	132000000	133000000
              chr7	133000000	134000000
              chr7	134000000	135000000
              chr7	135000000	136000000
              chr7	136000000	137000000
              chr7	137000000	138000000
              chr7	138000000	139000000
              chr7	139000000	140000000
              chr7	140000000	141000000
              chr7	141000000	142000000
              chr7	142000000	143000000
              chr7	143000000	144000000
              chr7	144000000	145000000
              chr7	145000000	146000000
              chr7	146000000	147000000
              chr7	147000000	148000000
              chr7	148000000	149000000
              chr7	149000000	150000000
              chr7	150000000	151000000
              chr7	151000000	152000000
              chr7	152000000	153000000
              chr7	153000000	154000000
              chr7	154000000	155000000
              chr7	155000000	156000000
              chr7	156000000	157000000
              chr7	157000000	158000000
              chr7	158000000	159000000
              chr7	159000000	159345973
              chr7_GL383534v2_alt	0	119183
              chr7_KI270803v1_alt	0	1000000
              chr7_KI270803v1_alt	1000000	1111570
              chr7_KI270804v1_alt	0	157952
              chr7_KI270805v1_alt	0	209988
              chr7_KI270806v1_alt	0	158166
              chr7_KI270807v1_alt	0	126434
              chr7_KI270808v1_alt	0	271455
              chr7_KI270809v1_alt	0	209586
              chr7_KI270899v1_alt	0	190869
              chr8	0	1000000
              chr8	1000000	2000000
              chr8	2000000	3000000
              chr8	3000000	4000000
              chr8	4000000	5000000
              chr8	5000000	6000000
              chr8	6000000	7000000
              chr8	7000000	8000000
              chr8	8000000	9000000
              chr8	9000000	10000000
              chr8	10000000	11000000
              chr8	11000000	12000000
              chr8	12000000	13000000
              chr8	13000000	14000000
              chr8	14000000	15000000
              chr8	15000000	16000000
              chr8	16000000	17000000
              chr8	17000000	18000000
              chr8	18000000	19000000
              chr8	19000000	20000000
              chr8	20000000	21000000
              chr8	21000000	22000000
              chr8	22000000	23000000
              chr8	23000000	24000000
              chr8	24000000	25000000
              chr8	25000000	26000000
              chr8	26000000	27000000
              chr8	27000000	28000000
              chr8	28000000	29000000
              chr8	29000000	30000000
              chr8	30000000	31000000
              chr8	31000000	32000000
              chr8	32000000	33000000
              chr8	33000000	34000000
              chr8	34000000	35000000
              chr8	35000000	36000000
              chr8	36000000	37000000
              chr8	37000000	38000000
              chr8	38000000	39000000
              chr8	39000000	40000000
              chr8	40000000	41000000
              chr8	41000000	42000000
              chr8	42000000	43000000
              chr8	43000000	44000000
              chr8	44000000	45000000
              chr8	45000000	46000000
              chr8	46000000	47000000
              chr8	47000000	48000000
              chr8	48000000	49000000
              chr8	49000000	50000000
              chr8	50000000	51000000
              chr8	51000000	52000000
              chr8	52000000	53000000
              chr8	53000000	54000000
              chr8	54000000	55000000
              chr8	55000000	56000000
              chr8	56000000	57000000
              chr8	57000000	58000000
              chr8	58000000	59000000
              chr8	59000000	60000000
              chr8	60000000	61000000
              chr8	61000000	62000000
              chr8	62000000	63000000
              chr8	63000000	64000000
              chr8	64000000	65000000
              chr8	65000000	66000000
              chr8	66000000	67000000
              chr8	67000000	68000000
              chr8	68000000	69000000
              chr8	69000000	70000000
              chr8	70000000	71000000
              chr8	71000000	72000000
              chr8	72000000	73000000
              chr8	73000000	74000000
              chr8	74000000	75000000
              chr8	75000000	76000000
              chr8	76000000	77000000
              chr8	77000000	78000000
              chr8	78000000	79000000
              chr8	79000000	80000000
              chr8	80000000	81000000
              chr8	81000000	82000000
              chr8	82000000	83000000
              chr8	83000000	84000000
              chr8	84000000	85000000
              chr8	85000000	86000000
              chr8	86000000	87000000
              chr8	87000000	88000000
              chr8	88000000	89000000
              chr8	89000000	90000000
              chr8	90000000	91000000
              chr8	91000000	92000000
              chr8	92000000	93000000
              chr8	93000000	94000000
              chr8	94000000	95000000
              chr8	95000000	96000000
              chr8	96000000	97000000
              chr8	97000000	98000000
              chr8	98000000	99000000
              chr8	99000000	100000000
              chr8	100000000	101000000
              chr8	101000000	102000000
              chr8	102000000	103000000
              chr8	103000000	104000000
              chr8	104000000	105000000
              chr8	105000000	106000000
              chr8	106000000	107000000
              chr8	107000000	108000000
              chr8	108000000	109000000
              chr8	109000000	110000000
              chr8	110000000	111000000
              chr8	111000000	112000000
              chr8	112000000	113000000
              chr8	113000000	114000000
              chr8	114000000	115000000
              chr8	115000000	116000000
              chr8	116000000	117000000
              chr8	117000000	118000000
              chr8	118000000	119000000
              chr8	119000000	120000000
              chr8	120000000	121000000
              chr8	121000000	122000000
              chr8	122000000	123000000
              chr8	123000000	124000000
              chr8	124000000	125000000
              chr8	125000000	126000000
              chr8	126000000	127000000
              chr8	127000000	128000000
              chr8	128000000	129000000
              chr8	129000000	130000000
              chr8	130000000	131000000
              chr8	131000000	132000000
              chr8	132000000	133000000
              chr8	133000000	134000000
              chr8	134000000	135000000
              chr8	135000000	136000000
              chr8	136000000	137000000
              chr8	137000000	138000000
              chr8	138000000	139000000
              chr8	139000000	140000000
              chr8	140000000	141000000
              chr8	141000000	142000000
              chr8	142000000	143000000
              chr8	143000000	144000000
              chr8	144000000	145000000
              chr8	145000000	145138636
              chr8_KI270810v1_alt	0	374415
              chr8_KI270811v1_alt	0	292436
              chr8_KI270812v1_alt	0	282736
              chr8_KI270813v1_alt	0	300230
              chr8_KI270814v1_alt	0	141812
              chr8_KI270815v1_alt	0	132244
              chr8_KI270816v1_alt	0	305841
              chr8_KI270817v1_alt	0	158983
              chr8_KI270818v1_alt	0	145606
              chr8_KI270819v1_alt	0	133535
              chr8_KI270820v1_alt	0	36640
              chr8_KI270821v1_alt	0	985506
              chr8_KI270822v1_alt	0	624492
              chr8_KI270900v1_alt	0	318687
              chr8_KI270901v1_alt	0	136959
              chr8_KI270926v1_alt	0	229282
              chr9	0	1000000
              chr9	1000000	2000000
              chr9	2000000	3000000
              chr9	3000000	4000000
              chr9	4000000	5000000
              chr9	5000000	6000000
              chr9	6000000	7000000
              chr9	7000000	8000000
              chr9	8000000	9000000
              chr9	9000000	10000000
              chr9	10000000	11000000
              chr9	11000000	12000000
              chr9	12000000	13000000
              chr9	13000000	14000000
              chr9	14000000	15000000
              chr9	15000000	16000000
              chr9	16000000	17000000
              chr9	17000000	18000000
              chr9	18000000	19000000
              chr9	19000000	20000000
              chr9	20000000	21000000
              chr9	21000000	22000000
              chr9	22000000	23000000
              chr9	23000000	24000000
              chr9	24000000	25000000
              chr9	25000000	26000000
              chr9	26000000	27000000
              chr9	27000000	28000000
              chr9	28000000	29000000
              chr9	29000000	30000000
              chr9	30000000	31000000
              chr9	31000000	32000000
              chr9	32000000	33000000
              chr9	33000000	34000000
              chr9	34000000	35000000
              chr9	35000000	36000000
              chr9	36000000	37000000
              chr9	37000000	38000000
              chr9	38000000	39000000
              chr9	39000000	40000000
              chr9	40000000	41000000
              chr9	41000000	42000000
              chr9	42000000	43000000
              chr9	43000000	44000000
              chr9	44000000	45000000
              chr9	45000000	46000000
              chr9	46000000	47000000
              chr9	47000000	48000000
              chr9	48000000	49000000
              chr9	49000000	50000000
              chr9	50000000	51000000
              chr9	51000000	52000000
              chr9	52000000	53000000
              chr9	53000000	54000000
              chr9	54000000	55000000
              chr9	55000000	56000000
              chr9	56000000	57000000
              chr9	57000000	58000000
              chr9	58000000	59000000
              chr9	59000000	60000000
              chr9	60000000	61000000
              chr9	61000000	62000000
              chr9	62000000	63000000
              chr9	63000000	64000000
              chr9	64000000	65000000
              chr9	65000000	66000000
              chr9	66000000	67000000
              chr9	67000000	68000000
              chr9	68000000	69000000
              chr9	69000000	70000000
              chr9	70000000	71000000
              chr9	71000000	72000000
              chr9	72000000	73000000
              chr9	73000000	74000000
              chr9	74000000	75000000
              chr9	75000000	76000000
              chr9	76000000	77000000
              chr9	77000000	78000000
              chr9	78000000	79000000
              chr9	79000000	80000000
              chr9	80000000	81000000
              chr9	81000000	82000000
              chr9	82000000	83000000
              chr9	83000000	84000000
              chr9	84000000	85000000
              chr9	85000000	86000000
              chr9	86000000	87000000
              chr9	87000000	88000000
              chr9	88000000	89000000
              chr9	89000000	90000000
              chr9	90000000	91000000
              chr9	91000000	92000000
              chr9	92000000	93000000
              chr9	93000000	94000000
              chr9	94000000	95000000
              chr9	95000000	96000000
              chr9	96000000	97000000
              chr9	97000000	98000000
              chr9	98000000	99000000
              chr9	99000000	100000000
              chr9	100000000	101000000
              chr9	101000000	102000000
              chr9	102000000	103000000
              chr9	103000000	104000000
              chr9	104000000	105000000
              chr9	105000000	106000000
              chr9	106000000	107000000
              chr9	107000000	108000000
              chr9	108000000	109000000
              chr9	109000000	110000000
              chr9	110000000	111000000
              chr9	111000000	112000000
              chr9	112000000	113000000
              chr9	113000000	114000000
              chr9	114000000	115000000
              chr9	115000000	116000000
              chr9	116000000	117000000
              chr9	117000000	118000000
              chr9	118000000	119000000
              chr9	119000000	120000000
              chr9	120000000	121000000
              chr9	121000000	122000000
              chr9	122000000	123000000
              chr9	123000000	124000000
              chr9	124000000	125000000
              chr9	125000000	126000000
              chr9	126000000	127000000
              chr9	127000000	128000000
              chr9	128000000	129000000
              chr9	129000000	130000000
              chr9	130000000	131000000
              chr9	131000000	132000000
              chr9	132000000	133000000
              chr9	133000000	134000000
              chr9	134000000	135000000
              chr9	135000000	136000000
              chr9	136000000	137000000
              chr9	137000000	138000000
              chr9	138000000	138394717
              chr9_GL383539v1_alt	0	162988
              chr9_GL383540v1_alt	0	71551
              chr9_GL383541v1_alt	0	171286
              chr9_GL383542v1_alt	0	60032
              chr9_KI270717v1_random	0	40062
              chr9_KI270718v1_random	0	38054
              chr9_KI270719v1_random	0	176845
              chr9_KI270720v1_random	0	39050
              chr9_KI270823v1_alt	0	439082
              chrM	0	16569
              chrUn_GL000195v1	0	182896
              chrUn_GL000213v1	0	164239
              chrUn_GL000214v1	0	137718
              chrUn_GL000216v2	0	176608
              chrUn_GL000218v1	0	161147
              chrUn_GL000219v1	0	179198
              chrUn_GL000220v1	0	161802
              chrUn_GL000224v1	0	179693
              chrUn_GL000226v1	0	15008
              chrUn_KI270302v1	0	2274
              chrUn_KI270303v1	0	1942
              chrUn_KI270304v1	0	2165
              chrUn_KI270305v1	0	1472
              chrUn_KI270310v1	0	1201
              chrUn_KI270311v1	0	12399
              chrUn_KI270312v1	0	998
              chrUn_KI270315v1	0	2276
              chrUn_KI270316v1	0	1444
              chrUn_KI270317v1	0	37690
              chrUn_KI270320v1	0	4416
              chrUn_KI270322v1	0	21476
              chrUn_KI270329v1	0	1040
              chrUn_KI270330v1	0	1652
              chrUn_KI270333v1	0	2699
              chrUn_KI270334v1	0	1368
              chrUn_KI270335v1	0	1048
              chrUn_KI270336v1	0	1026
              chrUn_KI270337v1	0	1121
              chrUn_KI270338v1	0	1428
              chrUn_KI270340v1	0	1428
              chrUn_KI270362v1	0	3530
              chrUn_KI270363v1	0	1803
              chrUn_KI270364v1	0	2855
              chrUn_KI270366v1	0	8320
              chrUn_KI270371v1	0	2805
              chrUn_KI270372v1	0	1650
              chrUn_KI270373v1	0	1451
              chrUn_KI270374v1	0	2656
              chrUn_KI270375v1	0	2378
              chrUn_KI270376v1	0	1136
              chrUn_KI270378v1	0	1048
              chrUn_KI270379v1	0	1045
              chrUn_KI270381v1	0	1930
              chrUn_KI270382v1	0	4215
              chrUn_KI270383v1	0	1750
              chrUn_KI270384v1	0	1658
              chrUn_KI270385v1	0	990
              chrUn_KI270386v1	0	1788
              chrUn_KI270387v1	0	1537
              chrUn_KI270388v1	0	1216
              chrUn_KI270389v1	0	1298
              chrUn_KI270390v1	0	2387
              chrUn_KI270391v1	0	1484
              chrUn_KI270392v1	0	971
              chrUn_KI270393v1	0	1308
              chrUn_KI270394v1	0	970
              chrUn_KI270395v1	0	1143
              chrUn_KI270396v1	0	1880
              chrUn_KI270411v1	0	2646
              chrUn_KI270412v1	0	1179
              chrUn_KI270414v1	0	2489
              chrUn_KI270417v1	0	2043
              chrUn_KI270418v1	0	2145
              chrUn_KI270419v1	0	1029
              chrUn_KI270420v1	0	2321
              chrUn_KI270422v1	0	1445
              chrUn_KI270423v1	0	981
              chrUn_KI270424v1	0	2140
              chrUn_KI270425v1	0	1884
              chrUn_KI270429v1	0	1361
              chrUn_KI270435v1	0	92983
              chrUn_KI270438v1	0	112505
              chrUn_KI270442v1	0	392061
              chrUn_KI270448v1	0	7992
              chrUn_KI270465v1	0	1774
              chrUn_KI270466v1	0	1233
              chrUn_KI270467v1	0	3920
              chrUn_KI270468v1	0	4055
              chrUn_KI270507v1	0	5353
              chrUn_KI270508v1	0	1951
              chrUn_KI270509v1	0	2318
              chrUn_KI270510v1	0	2415
              chrUn_KI270511v1	0	8127
              chrUn_KI270512v1	0	22689
              chrUn_KI270515v1	0	6361
              chrUn_KI270516v1	0	1300
              chrUn_KI270517v1	0	3253
              chrUn_KI270518v1	0	2186
              chrUn_KI270519v1	0	138126
              chrUn_KI270521v1	0	7642
              chrUn_KI270522v1	0	5674
              chrUn_KI270528v1	0	2983
              chrUn_KI270529v1	0	1899
              chrUn_KI270530v1	0	2168
              chrUn_KI270538v1	0	91309
              chrUn_KI270539v1	0	993
              chrUn_KI270544v1	0	1202
              chrUn_KI270548v1	0	1599
              chrUn_KI270579v1	0	31033
              chrUn_KI270580v1	0	1553
              chrUn_KI270581v1	0	7046
              chrUn_KI270582v1	0	6504
              chrUn_KI270583v1	0	1400
              chrUn_KI270584v1	0	4513
              chrUn_KI270587v1	0	2969
              chrUn_KI270588v1	0	6158
              chrUn_KI270589v1	0	44474
              chrUn_KI270590v1	0	4685
              chrUn_KI270591v1	0	5796
              chrUn_KI270593v1	0	3041
              chrUn_KI270741v1	0	157432
              chrUn_KI270742v1	0	186739
              chrUn_KI270743v1	0	210658
              chrUn_KI270744v1	0	168472
              chrUn_KI270745v1	0	41891
              chrUn_KI270746v1	0	66486
              chrUn_KI270747v1	0	198735
              chrUn_KI270748v1	0	93321
              chrUn_KI270749v1	0	158759
              chrUn_KI270750v1	0	148850
              chrUn_KI270751v1	0	150742
              chrUn_KI270752v1	0	27745
              chrUn_KI270753v1	0	62944
              chrUn_KI270754v1	0	40191
              chrUn_KI270755v1	0	36723
              chrUn_KI270756v1	0	79590
              chrUn_KI270757v1	0	71251
              chrX	0	1000000
              chrX	1000000	2000000
              chrX	2000000	3000000
              chrX	3000000	4000000
              chrX	4000000	5000000
              chrX	5000000	6000000
              chrX	6000000	7000000
              chrX	7000000	8000000
              chrX	8000000	9000000
              chrX	9000000	10000000
              chrX	10000000	11000000
              chrX	11000000	12000000
              chrX	12000000	13000000
              chrX	13000000	14000000
              chrX	14000000	15000000
              chrX	15000000	16000000
              chrX	16000000	17000000
              chrX	17000000	18000000
              chrX	18000000	19000000
              chrX	19000000	20000000
              chrX	20000000	21000000
              chrX	21000000	22000000
              chrX	22000000	23000000
              chrX	23000000	24000000
              chrX	24000000	25000000
              chrX	25000000	26000000
              chrX	26000000	27000000
              chrX	27000000	28000000
              chrX	28000000	29000000
              chrX	29000000	30000000
              chrX	30000000	31000000
              chrX	31000000	32000000
              chrX	32000000	33000000
              chrX	33000000	34000000
              chrX	34000000	35000000
              chrX	35000000	36000000
              chrX	36000000	37000000
              chrX	37000000	38000000
              chrX	38000000	39000000
              chrX	39000000	40000000
              chrX	40000000	41000000
              chrX	41000000	42000000
              chrX	42000000	43000000
              chrX	43000000	44000000
              chrX	44000000	45000000
              chrX	45000000	46000000
              chrX	46000000	47000000
              chrX	47000000	48000000
              chrX	48000000	49000000
              chrX	49000000	50000000
              chrX	50000000	51000000
              chrX	51000000	52000000
              chrX	52000000	53000000
              chrX	53000000	54000000
              chrX	54000000	55000000
              chrX	55000000	56000000
              chrX	56000000	57000000
              chrX	57000000	58000000
              chrX	58000000	59000000
              chrX	59000000	60000000
              chrX	60000000	61000000
              chrX	61000000	62000000
              chrX	62000000	63000000
              chrX	63000000	64000000
              chrX	64000000	65000000
              chrX	65000000	66000000
              chrX	66000000	67000000
              chrX	67000000	68000000
              chrX	68000000	69000000
              chrX	69000000	70000000
              chrX	70000000	71000000
              chrX	71000000	72000000
              chrX	72000000	73000000
              chrX	73000000	74000000
              chrX	74000000	75000000
              chrX	75000000	76000000
              chrX	76000000	77000000
              chrX	77000000	78000000
              chrX	78000000	79000000
              chrX	79000000	80000000
              chrX	80000000	81000000
              chrX	81000000	82000000
              chrX	82000000	83000000
              chrX	83000000	84000000
              chrX	84000000	85000000
              chrX	85000000	86000000
              chrX	86000000	87000000
              chrX	87000000	88000000
              chrX	88000000	89000000
              chrX	89000000	90000000
              chrX	90000000	91000000
              chrX	91000000	92000000
              chrX	92000000	93000000
              chrX	93000000	94000000
              chrX	94000000	95000000
              chrX	95000000	96000000
              chrX	96000000	97000000
              chrX	97000000	98000000
              chrX	98000000	99000000
              chrX	99000000	100000000
              chrX	100000000	101000000
              chrX	101000000	102000000
              chrX	102000000	103000000
              chrX	103000000	104000000
              chrX	104000000	105000000
              chrX	105000000	106000000
              chrX	106000000	107000000
              chrX	107000000	108000000
              chrX	108000000	109000000
              chrX	109000000	110000000
              chrX	110000000	111000000
              chrX	111000000	112000000
              chrX	112000000	113000000
              chrX	113000000	114000000
              chrX	114000000	115000000
              chrX	115000000	116000000
              chrX	116000000	117000000
              chrX	117000000	118000000
              chrX	118000000	119000000
              chrX	119000000	120000000
              chrX	120000000	121000000
              chrX	121000000	122000000
              chrX	122000000	123000000
              chrX	123000000	124000000
              chrX	124000000	125000000
              chrX	125000000	126000000
              chrX	126000000	127000000
              chrX	127000000	128000000
              chrX	128000000	129000000
              chrX	129000000	130000000
              chrX	130000000	131000000
              chrX	131000000	132000000
              chrX	132000000	133000000
              chrX	133000000	134000000
              chrX	134000000	135000000
              chrX	135000000	136000000
              chrX	136000000	137000000
              chrX	137000000	138000000
              chrX	138000000	139000000
              chrX	139000000	140000000
              chrX	140000000	141000000
              chrX	141000000	142000000
              chrX	142000000	143000000
              chrX	143000000	144000000
              chrX	144000000	145000000
              chrX	145000000	146000000
              chrX	146000000	147000000
              chrX	147000000	148000000
              chrX	148000000	149000000
              chrX	149000000	150000000
              chrX	150000000	151000000
              chrX	151000000	152000000
              chrX	152000000	153000000
              chrX	153000000	154000000
              chrX	154000000	155000000
              chrX	155000000	156000000
              chrX	156000000	156040895
              chrX_KI270880v1_alt	0	284869
              chrX_KI270881v1_alt	0	144206
              chrX_KI270913v1_alt	0	274009
              chrY	0	1000000
              chrY	1000000	2000000
              chrY	2000000	3000000
              chrY	3000000	4000000
              chrY	4000000	5000000
              chrY	5000000	6000000
              chrY	6000000	7000000
              chrY	7000000	8000000
              chrY	8000000	9000000
              chrY	9000000	10000000
              chrY	10000000	11000000
              chrY	11000000	12000000
              chrY	12000000	13000000
              chrY	13000000	14000000
              chrY	14000000	15000000
              chrY	15000000	16000000
              chrY	16000000	17000000
              chrY	17000000	18000000
              chrY	18000000	19000000
              chrY	19000000	20000000
              chrY	20000000	21000000
              chrY	21000000	22000000
              chrY	22000000	23000000
              chrY	23000000	24000000
              chrY	24000000	25000000
              chrY	25000000	26000000
              chrY	26000000	27000000
              chrY	27000000	28000000
              chrY	28000000	29000000
              chrY	29000000	30000000
              chrY	30000000	31000000
              chrY	31000000	32000000
              chrY	32000000	33000000
              chrY	33000000	34000000
              chrY	34000000	35000000
              chrY	35000000	36000000
              chrY	36000000	37000000
              chrY	37000000	38000000
              chrY	38000000	39000000
              chrY	39000000	40000000
              chrY	40000000	41000000
              chrY	41000000	42000000
              chrY	42000000	43000000
              chrY	43000000	44000000
              chrY	44000000	45000000
              chrY	45000000	46000000
              chrY	46000000	47000000
              chrY	47000000	48000000
              chrY	48000000	49000000
              chrY	49000000	50000000
              chrY	50000000	51000000
              chrY	51000000	52000000
              chrY	52000000	53000000
              chrY	53000000	54000000
              chrY	54000000	55000000
              chrY	55000000	56000000
              chrY	56000000	57000000
              chrY	57000000	57227415
              chrY_KI270740v1_random	0	37240
            filename: hg38_chrom_sizes.window.1000000.bed

        inputs:
        - id: '#Input_BedFile'
          type:
          - File
          inputBinding:
            position: 0
          description: Bed file must be sorted
          sbg:fileTypes: BED

        outputs:
        - id: '#OverlapBed'
          type:
          - 'null'
          - File
          outputBinding:
            glob:
              class: Expression
              engine: '#cwl-js-engine'
              script: |
                {
                  //sufix = "test";
                    //output_file=$job.inputs.Input_BedFile.path + "_hex.coverage_" + $job.inputs.Input_BedFile.nameroot + ".bed"

                    filepath_cap = $job.inputs.Input_BedFile.path
                    filename_cap = filepath_cap.split("/").pop()
                    basename_cap = filename_cap.substr(0,filename_cap.lastIndexOf("."))
                  
                    output = basename_cap + ".sorted.bed"
                    
                    return output
                }
        stdout:
          class: Expression
          engine: '#cwl-js-engine'
          script: |
            {
              //sufix = "test";
                //output_file=$job.inputs.Input_BedFile.path + "_hex.coverage_" + $job.inputs.Input_BedFile.nameroot + ".bed"

                filepath_cap = $job.inputs.Input_BedFile.path
                filename_cap = filepath_cap.split("/").pop()
                basename_cap = filename_cap.substr(0,filename_cap.lastIndexOf("."))
              
                output = basename_cap + ".sorted.bed"
                
                return output
            }

        baseCommand:
        - sort-bed

        hints:
        - class: DockerRequirement
          dockerPull: pgc-images.sbgenomics.com/d3b-bixu/bedops:v2
        id: michael_lloyd/mantis/bedops/5
        sbg:appVersion:
        - sbg:draft-2
        sbg:content_hash: aa14eb666275c9c3a65e0380ee06aaf642d78fbc8cfda521fe9f4266a1106bb12
        sbg:contributors:
        - michael_lloyd
        sbg:createdBy: michael_lloyd
        sbg:createdOn: 1565973380
        sbg:id: michael_lloyd/mantis/bedops/5
        sbg:image_url:
        sbg:latestRevision: 5
        sbg:modifiedBy: michael_lloyd
        sbg:modifiedOn: 1566228757
        sbg:project: michael_lloyd/mantis
        sbg:projectName: MSI, HRD, TMB
        sbg:publisher: sbg
        sbg:revision: 5
        sbg:revisionNotes: ''
        sbg:revisionsInfo:
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1565973380
          sbg:revision: 0
          sbg:revisionNotes: Copy of michael_lloyd/jax-pdx-wes-tumor-only/bedops/4
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1565973742
          sbg:revision: 1
          sbg:revisionNotes: hard coded hg38 window size file
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1565974852
          sbg:revision: 2
          sbg:revisionNotes: updated output grob
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1566228513
          sbg:revision: 3
          sbg:revisionNotes: changed to sorted region bed file
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1566228733
          sbg:revision: 4
          sbg:revisionNotes: ''
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1566228757
          sbg:revision: 5
          sbg:revisionNotes: ''
        sbg:sbgMaintained: false
        sbg:validationErrors: []
      inputs:
      - id: '#bedops.Input_BedFile'
        source: '#Coverage_BedFile'
      outputs:
      - id: '#bedops.OverlapBed'
      sbg:x: 1352.3988037109375
      sbg:y: -823.6986083984375
    - id: '#filter_trim_py_tumoronly'
      label: PY-allele_depth_min_and_AF_from_ADs-tumoronly
      in: []
      run:
        cwlVersion: sbg:draft-2
        class: CommandLineTool
        label: PY-allele_depth_min_and_AF_from_ADs-tumoronly
        $namespaces:
          sbg: https://sevenbridges.com

        requirements:
        - id: '#cwl-js-engine'
          class: ExpressionEngineRequirement
          requirements:
          - class: DockerRequirement
            dockerPull: rabix/js-engine
        - class: InlineJavascriptRequirement

        inputs:
        - id: '#raw_vcf'
          type:
          - 'null'
          - File
        - id: '#Allele_Depth_Filter'
          type:
          - 'null'
          - string
          inputBinding:
            position: 0
          description: |-
            for WES allele depth filter = 15, for higher depth sequencing, increase this parameter
          sbg:toolDefaultValue: '15'

        outputs:
        - id: '#filtered_vcf'
          type:
          - 'null'
          - File
          outputBinding:
            glob: '*.vcf'
            sbg:inheritMetadataFrom: '#raw_vcf'

        baseCommand:
        - python
        - /allele_depth_min_and_AF_from_ADs.py
        - class: Expression
          engine: '#cwl-js-engine'
          script: $job.inputs.raw_vcf.path
        - class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              filename = $job.inputs.raw_vcf.path.split('/').slice(-1)[0];
              file = filename.split('.vcf').slice(0)[0].concat('.pyfiltered.vcf');
              return file
            }

        hints:
        - class: sbg:CPURequirement
          value: 1
        - class: sbg:MemRequirement
          value: 1000
        - class: DockerRequirement
          dockerPull: pgc-images.sbgenomics.com/d3b-bixu/allele_depth_filter:v2
        id: michael_lloyd/mantis/filter-trim-py-tumoronly/2
        sbg:appVersion:
        - sbg:draft-2
        sbg:cmdPreview: |-
          python /allele_depth_min_and_AF_from_ADs.py /path/to/DN2-1-T_out_bwa.bam.aln.sorted.deduped.sorted.vcf DN2-1-T_out_bwa.pyfiltered.vcf 15
        sbg:content_hash: ac393f34d7e89ae1b9f5e870a38c7def7ed73e6152f3696a8719d55ce64aba825
        sbg:contributors:
        - michael_lloyd
        sbg:createdBy: michael_lloyd
        sbg:createdOn: 1567603421
        sbg:id: michael_lloyd/mantis/filter-trim-py-tumoronly/2
        sbg:image_url:
        sbg:job:
          allocatedResources:
            cpu: 1
            mem: 1000
          inputs:
            raw_vcf:
              class: File
              secondaryFiles: []
              path: /path/to/DN2-1-T_out_bwa.bam.aln.sorted.deduped.sorted.vcf
              size: 0
        sbg:latestRevision: 2
        sbg:modifiedBy: michael_lloyd
        sbg:modifiedOn: 1567603487
        sbg:project: michael_lloyd/mantis
        sbg:projectName: MSI, HRD, TMB
        sbg:publisher: sbg
        sbg:revision: 2
        sbg:revisionNotes: ''
        sbg:revisionsInfo:
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1567603421
          sbg:revision: 0
          sbg:revisionNotes: Copy of michael_lloyd/mantis/filter-trim-py/3
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1567603473
          sbg:revision: 1
          sbg:revisionNotes: changed python script
        - sbg:modifiedBy: michael_lloyd
          sbg:modifiedOn: 1567603487
          sbg:revision: 2
          sbg:revisionNotes: ''
        sbg:sbgMaintained: false
        sbg:validationErrors: []
        successCodes:
        - 0
        temporaryFailCodes:
        - 10
      inputs:
      - id: '#filter_trim_py_tumoronly.raw_vcf'
        source: '#raw_vcf'
      - id: '#filter_trim_py_tumoronly.Allele_Depth_Filter'
        default: '15'
      outputs:
      - id: '#filter_trim_py_tumoronly.filtered_vcf'
      sbg:x: -499.84161376953125
      sbg:y: -747.7218627929688
    id: michael_lloyd/mantis/tmb-tumoronly/1
    sbg:appVersion:
    - sbg:draft-2
    sbg:content_hash: afac419175d0b28b588cb8310658e7030559e2ff7af336666f6dfbb8b4501b45f
    sbg:contributors:
    - michael_lloyd
    sbg:createdBy: michael_lloyd
    sbg:createdOn: 1567603517
    sbg:id: michael_lloyd/mantis/tmb-tumoronly/1
    sbg:image_url:
    sbg:latestRevision: 1
    sbg:modifiedBy: michael_lloyd
    sbg:modifiedOn: 1567603585
    sbg:original_source: michael_lloyd/mantis/tmb-tumoronly/1
    sbg:project: michael_lloyd/mantis
    sbg:projectName: MSI, HRD, TMB
    sbg:publisher: sbg
    sbg:revision: 1
    sbg:revisionNotes: changed to tumor only version of filtering script
    sbg:revisionsInfo:
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1567603517
      sbg:revision: 0
      sbg:revisionNotes: Copy of michael_lloyd/mantis/tmb/23
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1567603585
      sbg:revision: 1
      sbg:revisionNotes: changed to tumor only version of filtering script
    sbg:sbgMaintained: false
    sbg:validationErrors: []
  inputs:
  - id: '#tmb_tumoronly.snpEFF_Database'
    source: '#database'
  - id: '#tmb_tumoronly.COSMIC_snpSift_Database'
    source: '#database_for_annotation'
  - id: '#tmb_tumoronly.raw_vcf'
    source: '#snpsift_split_4_3k.joined_vcf'
  - id: '#tmb_tumoronly.dbNSFP_Database'
    source: '#database_1'
  - id: '#tmb_tumoronly.Coverage_BedFile'
    source: '#bed'
  outputs:
  - id: '#tmb_tumoronly.TMB_Score'
  sbg:x: 1752.3887939453125
  sbg:y: 165.85398864746094
- id: '#vcftools_sort_0_1_14'
  label: VCFtools Sort
  in: []
  run:
    cwlVersion: sbg:draft-2
    class: CommandLineTool
    label: VCFtools Sort
    $namespaces:
      sbg: https://sevenbridges.com

    requirements:
    - id: '#cwl-js-engine'
      class: ExpressionEngineRequirement
      requirements:
      - class: DockerRequirement
        dockerPull: rabix/js-engine
      engineCommand: cwl-engine.js
    - class: InlineJavascriptRequirement

    inputs:
    - id: '#input_file'
      label: Input file
      type:
      - File
      inputBinding:
        position: 1
        separate: false
        sbg:cmdInclude: true
      description: Input file (vcf or vcf.gz)
      sbg:fileTypes: VCF, VCF.GZ
    - id: '#chromosomal_order'
      label: Chromosomal order
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -c
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Use natural ordering (1,2,10,MT,X) rather then the default (1,10,2,MT,X). This requires                                      new version of the unix "sort" command which supports the --version-sort option.
    - id: '#parallel'
      label: Parallel threads
      type:
      - 'null'
      - int
      inputBinding:
        prefix: -p
        position: 2
        separate: true
        sbg:cmdInclude: true
      description: Change the number of sorts run concurrently to <int>.
      sbg:category: Execution
      sbg:toolDefaultValue: '1'
    - id: '#mem_mb'
      label: Memory in MB
      type:
      - 'null'
      - int
      description: Memory in MB for execution.
      sbg:category: Execution
      sbg:stageInput:
      sbg:toolDefaultValue: N/A
    - id: '#compressed'
      label: Compressed output
      type:
      - 'null'
      - boolean
      description: Check to make the output compressed (usually for further processing).
      sbg:category: Execution
      sbg:stageInput:
      sbg:toolDefaultValue: 'FALSE'

    outputs:
    - id: '#output_file'
      label: Output file
      type:
      - 'null'
      - File
      outputBinding:
        glob:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              filepath = $job.inputs.input_file.path

              filename = filepath.split("/").pop();

              file_dot_sep = filename.split(".");
              file_ext = file_dot_sep[file_dot_sep.length-1];

              new_filename = filename.substr(0,filename.lastIndexOf(".vcf")) + ".sorted.vcf";

              if ($job.inputs.compressed) {
                  new_filename += ".gz";
              }
              
              return new_filename;
              
            }
        sbg:inheritMetadataFrom: '#input_file'
        sbg:metadata:
          __inherit__: input_file
      sbg:fileTypes: VCF, VCF.GZ

    baseCommand:
    - vcf-sort
    arguments:
    - position: 100
      valueFrom:
        class: Expression
        engine: '#cwl-js-engine'
        script: |-
          {


            if ($job.inputs.compressed) {
              
            	filepath = $job.inputs.input_file.path

            	filename = filepath.split("/").pop();

            	file_dot_sep = filename.split(".");
            	file_ext = file_dot_sep[file_dot_sep.length-1];

            	new_filename = filename.substr(0,filename.lastIndexOf(".vcf")) + ".sorted.vcf";
            	return '&& bgzip -c -f ' + new_filename + ' > ' + new_filename + '.gz'
                
            }
            
            
            
          }
      separate: true
    - position: 50
      valueFrom:
        class: Expression
        engine: '#cwl-js-engine'
        script: |-
          {
            filepath = $job.inputs.input_file.path

            filename = filepath.split("/").pop();

            file_dot_sep = filename.split(".");
            file_ext = file_dot_sep[file_dot_sep.length-1];

            new_filename = filename.substr(0,filename.lastIndexOf(".vcf")) + ".sorted.vcf";
            
            return '> '+new_filename;
            
          }
      separate: true

    hints:
    - class: sbg:CPURequirement
      value:
        class: Expression
        engine: '#cwl-js-engine'
        script: |-
          {
            if ($job.inputs.parallel) {
            	return $job.inputs.parallel
            } else {
              return 1
            }  
          }
    - class: sbg:MemRequirement
      value:
        class: Expression
        engine: '#cwl-js-engine'
        script: |-
          {
            if ($job.inputs.mem_mb) {
              
              return $job.inputs.mem_mb
              
             }  else {
                
                return 1000
                
                }
          }
    - class: DockerRequirement
      dockerPull: images.sbgenomics.com/ognjenm/vcftools:0.1.14
      dockerImageId: 54cb823ba25a
    id: michael_lloyd/mantis/vcftools-sort-0-1-14/0
    description: VCFtools sort sorts a VCF file.
    sbg:appVersion:
    - sbg:draft-2
    sbg:categories:
    - VCF-Processing
    sbg:cmdPreview: |-
      vcf-sort sample1.vcf  > sample1.sorted.vcf  && bgzip -c -f sample1.sorted.vcf > sample1.sorted.vcf.gz
    sbg:content_hash:
    sbg:contributors:
    - michael_lloyd
    sbg:copyOf: admin/sbg-public-data/vcftools-sort-0-1-14/6
    sbg:createdBy: michael_lloyd
    sbg:createdOn: 1568123250
    sbg:id: michael_lloyd/mantis/vcftools-sort-0-1-14/0
    sbg:image_url:
    sbg:job:
      allocatedResources:
        cpu: 2
        mem: 100
      inputs:
        compressed: true
        input_file:
          path: sample1.vcf
        mem_mb: 100
        parallel: 2
    sbg:latestRevision: 0
    sbg:license: GNU General Public License version 3.0 (GPLv3)
    sbg:links:
    - id: https://vcftools.github.io
      label: Homepage
    - id: https://github.com/vcftools/vcftools
      label: Source code
    - id: http://bioinformatics.oxfordjournals.org/content/27/15/2156
      label: Publications
    sbg:modifiedBy: michael_lloyd
    sbg:modifiedOn: 1568123250
    sbg:project: michael_lloyd/mantis
    sbg:projectName: MSI, HRD, TMB
    sbg:publisher: sbg
    sbg:revision: 0
    sbg:revisionNotes: Copy of admin/sbg-public-data/vcftools-sort-0-1-14/6
    sbg:revisionsInfo:
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1568123250
      sbg:revision: 0
      sbg:revisionNotes: Copy of admin/sbg-public-data/vcftools-sort-0-1-14/6
    sbg:sbgMaintained: false
    sbg:toolAuthor: Adam Auton, Petr Danecek, Anthony Marcketta
    sbg:toolkit: VCFtools
    sbg:toolkitVersion: 0.1.14
    sbg:validationErrors: []
  inputs:
  - id: '#vcftools_sort_0_1_14.input_file'
    source: '#SnpSift_Annotate.annotated_vcf'
  - id: '#vcftools_sort_0_1_14.chromosomal_order'
    default: true
  outputs:
  - id: '#vcftools_sort_0_1_14.output_file'
  sbg:x: 1882.15283203125
  sbg:y: 277.21221923828125
- id: '#msisensor_msi_0_2'
  label: MSIsensor2 msi
  in: []
  run:
    cwlVersion: sbg:draft-2
    class: CommandLineTool
    label: MSIsensor2 msi
    $namespaces:
      sbg: https://sevenbridges.com

    requirements:
    - id: '#cwl-js-engine'
      class: ExpressionEngineRequirement
      requirements:
      - class: DockerRequirement
        dockerPull: rabix/js-engine
    - class: InlineJavascriptRequirement

    inputs:
    - id: '#tumor_bam'
      label: Tumor BAM file
      type:
      - File
      inputBinding:
        secondaryFiles:
        - .bai
        prefix: -t
        position: 8
        separate: true
        sbg:cmdInclude: true
      description: |-
        Reads from the tumor sample, aligned to the reference genome. The BAM file needs to be indexed, and the corresponding BAM.BAI file present in project files.
      sbg:category: Input and output options
      sbg:fileTypes: BAM
    - id: '#out_prefix'
      label: Prefix for output files
      type:
      - 'null'
      - string
      inputBinding:
        prefix: -o
        position: 4
        valueFrom:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              if ($job.inputs.out_prefix) return $job.inputs.out_prefix;
              
              var t_sample = $job.inputs.tumor_bam.path.split('/').pop();
              var base_name = t_sample.slice(0, t_sample.length - 4);
              return base_name + '_msi';
            }
        separate: true
        sbg:cmdInclude: true
      description: Prefix for output file names.
      sbg:category: Input and output options
      sbg:toolDefaultValue: '[same as tumor BAM]_msi'
    - id: '#min_coverage'
      label: Coverage threshold
      type:
      - 'null'
      - int
      inputBinding:
        prefix: -c
        position: 7
        separate: true
        sbg:cmdInclude: true
      description: |-
        Minimum read coverage for MSI analysis at a given locus. Recommended: 20 for WXS, 15 for WGS data.
      sbg:category: Config Inputs
      sbg:stageInput:
      sbg:toolDefaultValue: '20'
    - id: '#only_homopolymers'
      label: Output homopolymers only
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -x
        position: 16
        valueFrom:
          class: Expression
          engine: '#cwl-js-engine'
          script: "{\n  if ($job.inputs.only_homopolymers) return 1;\n}"
        separate: true
        sbg:cmdInclude: true
      description: Only output homopolymers, do not analyze microsatellites of period
        > 1.
      sbg:category: Input and output options
      sbg:stageInput:
      sbg:toolDefaultValue: 'False'
    - id: '#only_microsats'
      label: Output microsatellites only
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: -y
        position: 17
        valueFrom:
          class: Expression
          engine: '#cwl-js-engine'
          script: "{\n  if ($job.inputs.only_microsats) return 1;\n}"
        separate: true
        sbg:cmdInclude: true
      description: Only output microsatellites, do not analyze homopolymers.
      sbg:category: Input and output options
      sbg:stageInput:
      sbg:toolDefaultValue: 'False'

    outputs:
    - id: '#distribution'
      label: Distribution of microsatellite and homopolymer lengths
      type:
      - 'null'
      - File
      outputBinding:
        glob:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              if ($job.inputs.out_prefix) return $job.inputs.out_prefix + '_dis';
              
              var t_sample = $job.inputs.tumor_bam.path.split('/').pop();
              var base_name = t_sample.slice(0, t_sample.length - 4);
              return base_name + '_msi_dis';
            }
      description: |-
        Read count distribution across microsatellite and homopolymer lengths, for each locus specified in the input microsatellites list.
      sbg:fileTypes: TXT
    - id: '#somatic_loci'
      label: Somatic loci
      type:
      - 'null'
      - File
      outputBinding:
        glob:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              if ($job.inputs.out_prefix) return $job.inputs.out_prefix + '_somatic';
              
              var t_sample = $job.inputs.tumor_bam.path.split('/').pop();
              var base_name = t_sample.slice(0, t_sample.length - 4);
              return base_name + '_msi_somatic';
            }
      description: Somatic sites detected.
      sbg:fileTypes: TXT
    - id: '#msi_score'
      label: MSI score
      type:
      - 'null'
      - File
      outputBinding:
        glob:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            {
              if ($job.inputs.out_prefix) return $job.inputs.out_prefix;
              
              var t_sample = $job.inputs.tumor_bam.path.split('/').pop();
              var base_name = t_sample.slice(0, t_sample.length - 4);
              return base_name + '_msi';
            }
        sbg:inheritMetadataFrom: '#tumor_bam'
      description: MSI score.
      sbg:fileTypes: TXT

    baseCommand:
    - /msisensor2
    - msi
    arguments:
    - prefix: -M
      position: 1
      valueFrom: /models_hg38
      separate: true

    hints:
    - class: sbg:MemRequirement
      value: 4000
    - class: DockerRequirement
      dockerPull: pgc-images.sbgenomics.com/d3b-bixu/msisensor2:0.1
    id: michael_lloyd/mantis/msisensor-msi-0-2/3
    description: |-
      **MSIsensor2** is a C++ program for automatically detecting somati variants at microsatellite regions for tumor only samples.
    sbg:appVersion:
    - sbg:draft-2
    sbg:categories:
    - Analysis
    - Microsatellites
    - DNA
    - WES-(WXS)
    - WGS
    sbg:cmdPreview: |-
      /opt/msisensor-0.2/msisensor msi -b 1 -t /path/to/tumor_bam.bam -d /path/to/microsats_list.txt
    sbg:content_hash: ad8bf551d647aaf459a70c3f1949e0d0e15772c826bfc0e645f5d6a85be976858
    sbg:contributors:
    - michael_lloyd
    sbg:createdBy: michael_lloyd
    sbg:createdOn: 1572283831
    sbg:id: michael_lloyd/mantis/msisensor-msi-0-2/3
    sbg:image_url:
    sbg:job:
      allocatedResources:
        cpu: 1
        mem: 4000
      inputs:
        homopolymer_dist_lbound:
        homopolymer_dist_ubound:
        microsats_list:
          class: File
          secondaryFiles: []
          path: /path/to/microsats_list.txt
          size: 0
        min_coverage:
        min_homopolymer_size:
        min_repeats:
        ms_repeats_dist_lbound:
        ms_repeats_dist_ubound:
        normal_bam:
          class: File
          secondaryFiles: []
          path: /path/to/normal_bam.bam
          size: 0
        num_threads: 0
        only_homopolymers: false
        only_microsats: false
        out_prefix: ''
        p_val:
        region_sel: ''
        span_size:
        tumor_bam:
          class: File
          secondaryFiles:
          - path: .bai
          path: /path/to/tumor_bam.bam
          size: 0
    sbg:latestRevision: 3
    sbg:license: MIT License
    sbg:links:
    - id: http://tvap.genome.wustl.edu/tools/msisensor/
      label: Homepage
    - id: https://github.com/ding-lab/msisensor
      label: Source code
    - id: https://github.com/ding-lab/msisensor/archive/master.zip
      label: Download
    - id: http://bioinformatics.oxfordjournals.org/content/30/7/1015
      label: Publication
    sbg:modifiedBy: michael_lloyd
    sbg:modifiedOn: 1572284787
    sbg:project: michael_lloyd/mantis
    sbg:projectName: MSI, HRD, TMB
    sbg:publisher: sbg
    sbg:revision: 3
    sbg:revisionNotes: ''
    sbg:revisionsInfo:
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1572283831
      sbg:revision: 0
      sbg:revisionNotes: Copy of admin/sbg-public-data/msisensor-msi-0-2/5
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1572284560
      sbg:revision: 1
      sbg:revisionNotes: updated to msisensor2
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1572284594
      sbg:revision: 2
      sbg:revisionNotes: ''
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1572284787
      sbg:revision: 3
      sbg:revisionNotes: ''
    sbg:sbgMaintained: false
    sbg:toolAuthor: Beifang Niu and Kai Ye, Washington University in St. Louis
    sbg:toolkit: MSIsensor
    sbg:toolkitVersion: '0.2'
    sbg:validationErrors: []
  inputs:
  - id: '#msisensor_msi_0_2.tumor_bam'
    source: '#SAMtools_Index_BAM_1.output_bam_file'
  outputs:
  - id: '#msisensor_msi_0_2.distribution'
  - id: '#msisensor_msi_0_2.somatic_loci'
  - id: '#msisensor_msi_0_2.msi_score'
  sbg:x: 907.3973999023438
  sbg:y: -241.63319396972656
- id: '#gatk_filtermutectcalls_4051'
  label: GATK FilterMutectCalls - 4.0.5.1
  in: []
  scatter: '#gatk_filtermutectcalls_4051.variant'
  run:
    cwlVersion: sbg:draft-2
    class: CommandLineTool
    label: GATK FilterMutectCalls - 4.0.5.1
    $namespaces:
      sbg: https://sevenbridges.com

    requirements:
    - id: '#cwl-js-engine'
      class: ExpressionEngineRequirement
      requirements:
      - class: DockerRequirement
        dockerPull: rabix/js-engine
    - class: InlineJavascriptRequirement

    inputs:
    - id: '#verbosity'
      label: Verbosity
      type:
      - 'null'
      - name: 'null'
        type: enum
        symbols:
        - ERROR
        - WARNING
        - INFO
        - DEBUG
      inputBinding:
        prefix: --verbosity
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Control verbosity of logging. Default value: INFO. Possible values: {ERROR, WARNING, INFO, DEBUG}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: INFO
    - id: '#variant'
      label: Variant
      type:
      - File
      inputBinding:
        prefix: --variant
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: A VCF file containing variants Required.
      required: true
      sbg:altPrefix: ''
      sbg:category: Required Arguments
    - id: '#use_new_af_calculator'
      label: Use New Af Calculator
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --use-new-af-calculator
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        If provided, we will use the new AF model instead of the so-called exact model Default value: false. Possible values: {true, false}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#use_jdk_inflater'
      label: Use Jdk Inflater
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --use-jdk-inflater
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Whether to use the JdkInflater (as opposed to IntelInflater) Default value: false. Possible values: {true, false}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#use_jdk_deflater'
      label: Use Jdk Deflater
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --use-jdk-deflater
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Whether to use the JdkDeflater (as opposed to IntelDeflater) Default value: false. Possible values: {true, false}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#use_filtered_reads_for_annotations'
      label: Use Filtered Reads For Annotations
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --use-filtered-reads-for-annotations
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Use the contamination-filtered read maps for the purposes of annotating variants Default value: false. Possible values: {true, false}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#use_double_precision'
      label: Use Double Precision
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --use-double-precision
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Use double precision in the native pairHmm. This is slower but matches the java implementation better Default value: false. Possible values: {true, false}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#uniqueAltReadCount'
      label: UniqueAltReadCount
      type:
      - 'null'
      - int
      inputBinding:
        prefix: -unique
        position: 0
        valueFrom:
          class: Expression
          engine: '#cwl-js-engine'
          script: |-
            { if ($job.inputs.uniqueAltReadCount == undefined)
            {
              return 5
            }
             
             else
             {
               return $job.inputs.uniqueAltReadCount
             }

            }
        separate: true
        sbg:cmdInclude: true
      description: |-
        Filter a variant if a site contains fewer than this many unique (i.e. deduplicated) reads supporting the alternate allele
      required: false
      sbg:altPrefix: --uniqueAltReadCount
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '0'
    - id: '#tumor_lod'
      label: Tumor Lod
      type:
      - 'null'
      - float
      inputBinding:
        prefix: --tumor-lod
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: 'LOD threshold for calling tumor variant Default value: 5.3.'
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '5'
    - id: '#strand_artifact_posterior_probability'
      label: Strand Artifact Posterior Probability
      type:
      - 'null'
      - float
      inputBinding:
        prefix: --strand-artifact-posterior-probability
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Filter a variant if the probability of strand artifact exceeds this number Default value: 0.99.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '0'
    - id: '#strand_artifact_allele_fraction'
      label: Strand Artifact Allele Fraction
      type:
      - 'null'
      - float
      inputBinding:
        prefix: --strand-artifact-allele-fraction
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Only filter a variant if the MAP estimate of allele fraction given artifact is below this number Default value: 0.01.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '0'
    - id: '#standard_min_confidence_threshold_for_calling'
      label: Standard Min Confidence Threshold For Calling
      type:
      - 'null'
      - float
      inputBinding:
        prefix: --standard-min-confidence-threshold-for-calling
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        The minimum phred-scaled confidence threshold at which variants should be called Default value: 10.0.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '10'
    - id: '#seconds_between_progress_updates'
      label: Seconds Between Progress Updates
      type:
      - 'null'
      - float
      inputBinding:
        prefix: --seconds-between-progress-updates
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Output traversal statistics every time this many seconds elapse Default value: 10.0.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '10'
    - id: '#sample_ploidy'
      label: Sample Ploidy
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --sample-ploidy
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Ploidy (number of chromosomes) per sample. For pooled data, set to (Number of samples in each pool * Sample Ploidy). Default value: 2.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '2'
    - id: '#sample'
      label: Sample
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --sample
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        The name of the sample(s) to keep, filtering out all others This argument must be specified at least once. Required.
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
      sbg:toolDefaultValue:
    - id: '#reference'
      label: Reference
      type:
      - 'null'
      - File
      inputBinding:
        secondaryFiles:
        - .fai
        - ^.dict
        prefix: --reference
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: 'Reference sequence Default value: null.'
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
    - id: '#recover_dangling_heads'
      label: Recover Dangling Heads
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --recover-dangling-heads
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        This argument is deprecated since version 3.3 Default value: false. Possible values: {true, false}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#read_validation_stringency'
      label: Read Validation Stringency
      type:
      - 'null'
      - name: 'null'
        type: enum
        symbols:
        - STRICT
        - LENIENT
        - SILENT
      inputBinding:
        prefix: --read-validation-stringency
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Validation stringency for all SAM/BAM/CRAM/SRA files read by this program. The default stringency value SILENT can improve performance when processing a BAM file in which variable-length data (read, qualities, tags) do not otherwise need to be decoded. Default value: SILENT. Possible values: {STRICT, LENIENT, SILENT}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: SILENT
    - id: '#read_name'
      label: Read Name
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --read-name
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Keep only reads with this read name Required.
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
      sbg:toolDefaultValue:
    - id: '#read_index'
      label: Read Index
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --read-index
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Indices to use for the read inputs. If specified, an index must be provided for every read input and in the same order as the read inputs. If this argument is not specified, the path to the index for each input will be inferred automatically. This argument may be specified 0 or more times. Default value: null.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'null'
    - id: '#read_filter'
      label: Read Filter
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --read-filter
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Read filters to be applied before analysis This argument may be specified 0 or more times. Default value: null.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'null'
    - id: '#quiet'
      label: Quiet
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --quiet
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Whether to suppress job-summary info on System.err. Default value: false. Possible values: {true, false}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#pl_filter_name'
      label: Pl Filter Name
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --pl-filter-name
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Keep reads with RG:PL attribute containing this string This argument must be specified at least once. Required.
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
      sbg:toolDefaultValue:
    - id: '#phred_scaled_global_read_mismapping_rate'
      label: Phred Scaled Global Read Mismapping Rate
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --phred-scaled-global-read-mismapping-rate
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: 'The global assumed mismapping rate for reads Default value: 45.'
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: '45'
    - id: '#pcr_indel_model'
      label: Pcr Indel Model
      type:
      - 'null'
      - name: 'null'
        type: enum
        symbols:
        - NONE
        - HOSTILE
        - AGGRESSIVE
        - CONSERVATIVE
      inputBinding:
        prefix: --pcr-indel-model
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        The PCR indel model to use Default value: CONSERVATIVE. Possible values: {NONE, HOSTILE, AGGRESSIVE, CONSERVATIVE}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: CONSERVATIVE
    - id: '#output_mode'
      label: Output Mode
      type:
      - 'null'
      - name: 'null'
        type: enum
        symbols:
        - EMIT_VARIANTS_ONLY
        - EMIT_ALL_CONFIDENT_SITES
        - EMIT_ALL_SITES
      inputBinding:
        prefix: --output-mode
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Specifies which type of calls we should output Default value: EMIT_VARIANTS_ONLY. Possible values: {EMIT_VARIANTS_ONLY, EMIT_ALL_CONFIDENT_SITES, EMIT_ALL_SITES}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: EMIT_VARIANTS_ONLY
    - id: '#num_pruning_samples'
      label: Num Pruning Samples
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --num-pruning-samples
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: 'Number of samples that must pass the minPruning threshold Default
        value: 1.'
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: '1'
    - id: '#normal_artifact_lod'
      label: Normal Artifact Lod
      type:
      - 'null'
      - float
      inputBinding:
        prefix: --normal-artifact-lod
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: 'LOD threshold for calling normal artifacts Default value: 0.0.'
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '0'
    - id: '#native_pair_hmm_threads'
      label: Native Pair Hmm Threads
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --native-pair-hmm-threads
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: 'How many threads should a native pairHMM implementation use Default
        value: 4.'
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '4'
    - id: '#minimum_mapping_quality'
      label: Minimum Mapping Quality
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --minimum-mapping-quality
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: 'Minimum mapping quality to keep (inclusive) Default value: 10.'
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
      sbg:toolDefaultValue: '10'
    - id: '#min_read_length'
      label: Min Read Length
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --min-read-length
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Keep only reads with length at least equal to the specified value Default value: 1.
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
      sbg:toolDefaultValue: '1'
    - id: '#min_pruning'
      label: Min Pruning
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --min-pruning
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: 'Minimum support to not prune paths in the graph Default value:
        2.'
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: '2'
    - id: '#min_median_read_position'
      label: Min Median Read Position
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --min-median-read-position
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Filter variants for which the median position of alt alleles within reads is too near the end of reads. Default value: 5.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '5'
    - id: '#min_median_mapping_quality'
      label: Min Median Mapping Quality
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --min-median-mapping-quality
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Integer filter variants for which alt reads' median mapping quality
        is too low.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '30'
    - id: '#min_median_base_quality'
      label: Min Median Base Quality
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --min-median-base-quality
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Integerfilter variants for which alt reads' median base quality
        is too low.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: '20'
    - id: '#min_dangling_branch_length'
      label: Min Dangling Branch Length
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --min-dangling-branch-length
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: 'Minimum length of a dangling branch to attempt recovery Default
        value: 4.'
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: '4'
    - id: '#min_base_quality_score'
      label: Min Base Quality Score
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --min-base-quality-score
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: 'Minimum base quality required to consider a base for calling Default
        value: 10.'
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '10'
    - id: '#memory_per_job'
      label: Memory Per Job
      type:
      - 'null'
      - int
      description: Memory Per Job
      required: false
      sbg:category: Execution
    - id: '#memory_overhead_per_job'
      label: Memory Overhead Per Job
      type:
      - 'null'
      - int
      description: Memory Overhead Per Job
      required: false
      sbg:category: Execution
    - id: '#maximum_mapping_quality'
      label: Maximum Mapping Quality
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --maximum-mapping-quality
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: 'Maximum mapping quality to keep (inclusive) Default value: null.'
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
      sbg:toolDefaultValue: 'null'
    - id: '#max_read_length'
      label: Max Read Length
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --max-read-length
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Keep only reads with length at most equal to the specified value
        Required.
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
      sbg:toolDefaultValue:
    - id: '#max_num_haplotypes_in_population'
      label: Max Num Haplotypes In Population
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --max-num-haplotypes-in-population
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: 'Maximum number of haplotypes to consider for your population Default
        value: 128.'
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: '128'
    - id: '#max_median_fragment_length_difference'
      label: Max Median Fragment Length Difference
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --max-median-fragment-length-difference
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Filter variants for which alt reads' median fragment length is very different from the median for ref reads. Default value: 10000.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '10000'
    - id: '#max_germline_posterior'
      label: Max Germline Posterior
      type:
      - 'null'
      - float
      inputBinding:
        prefix: --max-germline-posterior
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Maximum posterior probability that an allele is a germline variant Default value: 0.025.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '0'
    - id: '#max_genotype_count'
      label: Max Genotype Count
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --max-genotype-count
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: 'Maximum number of genotypes to consider at any site Default value:
        1024.'
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: '1024'
    - id: '#max_fragment_length'
      label: Max Fragment Length
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --max-fragment-length
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Keep only read pairs with fragment length at most equal to the given value Default value: 1000000.
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
      sbg:toolDefaultValue: '1000000'
    - id: '#max_events_in_haplotype'
      label: Max Events In Haplotype
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --max-events-in-haplotype
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Coming from a haplotype with more than this many events are filtered Default value: 2.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '2'
    - id: '#max_alternate_alleles'
      label: Max Alternate Alleles
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --max-alternate-alleles
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: 'Maximum number of alternate alleles to genotype Default value:
        6.'
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: '6'
    - id: '#max_alt_alleles_threshold'
      label: Max Alt Alleles Threshold
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --max-alt-alleles-threshold
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: 'Filter variants with too many alt alleles Default value: 1.'
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '1'
    - id: '#library'
      label: Library
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --library
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: The name of the library to keep Required.
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
      sbg:toolDefaultValue:
    - id: '#lenient'
      label: Lenient
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --lenient
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Lenient processing of VCF files Default value: false. Possible values: {true, false}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#kmer_size'
      label: Kmer Size
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --kmer-size
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Kmer size to use in the read threading assembler This argument may be specified 0 or more times. Default value: [10, 25].
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: '[10, 25]'
    - id: '#keep_reverse'
      label: Keep Reverse
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --keep-reverse
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: 'Keep only reads on the reverse strand Required. Possible values:
        {true, false}.'
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
      sbg:stageInput:
    - id: '#keep_read_group'
      label: Keep Read Group
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --keep-read-group
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: The name of the read group to keep Required.
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
      sbg:toolDefaultValue:
    - id: '#intervals'
      label: Intervals
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --intervals
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        One or more genomic intervals over which to operate This argument may be specified 0 or more times. Default value: null.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'null'
    - id: '#interval_set_rule'
      label: Interval Set Rule
      type:
      - 'null'
      - name: 'null'
        type: enum
        symbols:
        - UNION
        - INTERSECTION
      inputBinding:
        prefix: --interval-set-rule
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Set merging approach to use for combining interval inputs Default value: UNION. Possible values: {UNION, INTERSECTION}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: UNION
    - id: '#interval_padding'
      label: Interval Padding
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --interval-padding
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: 'Of padding (in bp) to add to each interval you are including.
        Default value: 0.'
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '0'
    - id: '#interval_merging_rule'
      label: Interval Merging Rule
      type:
      - 'null'
      - name: interval_merging_rule
        type: enum
        symbols:
        - ALL
        - OVERLAPPING_ONLY
      inputBinding:
        prefix: --interval-merging-rule
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Interval merging rule for abutting intervals.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: ALL
    - id: '#interval_exclusion_padding'
      label: Interval Exclusion Padding
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --interval-exclusion-padding
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Amount of padding (in bp) to add to each interval you are excluding. Default value: 0.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '0'
    - id: '#input_prior'
      label: Input Prior
      type:
      - 'null'
      - float
      inputBinding:
        prefix: --input-prior
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Input prior for calls This argument may be specified 0 or more times. Default value: null.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: 'null'
    - id: '#input'
      label: Input
      type:
      - 'null'
      - File
      inputBinding:
        prefix: --input
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        BAM/SAM/CRAM file containing reads This argument may be specified 0 or more times. Default value: null.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue:
    - id: '#indel_heterozygosity'
      label: Indel Heterozygosity
      type:
      - 'null'
      - float
      inputBinding:
        prefix: --indel-heterozygosity
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Heterozygosity for indel calling. See the GATKDocs for heterozygosity for full details on the meaning of this population genetics concept Default value: 1.25E-4.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '1'
    - id: '#heterozygosity_stdev'
      label: Heterozygosity Stdev
      type:
      - 'null'
      - float
      inputBinding:
        prefix: --heterozygosity-stdev
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Standard deviation of eterozygosity for SNP and indel calling. Default value: 0.01.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '0'
    - id: '#heterozygosity'
      label: Heterozygosity
      type:
      - 'null'
      - float
      inputBinding:
        prefix: --heterozygosity
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Heterozygosity value used to compute prior likelihoods for any locus. See the GATKDocs for full details on the meaning of this population genetics concept Default value: 0.001.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '0'
    - id: '#graph_output'
      label: Graph Output
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --graph-output
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: 'Write debug assembly graph information to this file Default value:
        null.'
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'null'
    - id: '#genotyping_mode'
      label: Genotyping Mode
      type:
      - 'null'
      - name: 'null'
        type: enum
        symbols:
        - DISCOVERY
        - GENOTYPE_GIVEN_ALLELES
      inputBinding:
        prefix: --genotyping-mode
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Specifies how to determine the alternate alleles to use for genotyping Default value: DISCOVERY. Possible values: {DISCOVERY, GENOTYPE_GIVEN_ALLELES}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: DISCOVERY
    - id: '#gcs_max_retries'
      label: Gcs Max Retries
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --gcs-max-retries
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        If the GCS bucket channel errors out, how many times it will attempt to re-initiate the connection.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: '20'
    - id: '#gcp_hmm'
      label: Gcp Hmm
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --gcp-hmm
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: 'Flat gap continuation penalty for use in the Pair HMM Default
        value: 10.'
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: '10'
    - id: '#filter_too_short'
      label: Filter Too Short
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --filter-too-short
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Value for which reads with less than this number of aligned bases is considered too short Default value: 30.
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
      sbg:toolDefaultValue: '30'
    - id: '#exclude_intervals'
      label: Exclude Intervals
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --exclude-intervals
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        One or more genomic intervals to exclude from processing This argument may be specified 0 or more times. Default value: null.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'null'
    - id: '#exclude_annotation'
      label: Exclude Annotation
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --exclude-annotation
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Or more specific annotations to exclude This argument may be specified 0 or more times. Default value: null.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: 'null'
    - id: '#emit_ref_confidence'
      label: Emit Ref Confidence
      type:
      - 'null'
      - name: 'null'
        type: enum
        symbols:
        - NONE
        - BP_RESOLUTION
        - GVCF
      inputBinding:
        prefix: --emit-ref-confidence
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Mode for emitting reference confidence scores Default value: NONE. Possible values: {NONE, BP_RESOLUTION, GVCF}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: NONE
    - id: '#dont_use_soft_clipped_bases'
      label: Dont Use Soft Clipped Bases
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --dont-use-soft-clipped-bases
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Do not analyze soft clipped bases in the reads Default value: false. Possible values: {true, false}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#dont_trim_active_regions'
      label: Dont Trim Active Regions
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --dont-trim-active-regions
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        If specified, we will not trim down the active region from the full region (active + extension) to just the active interval for genotyping Default value: false. Possible values: {true, false}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#dont_require_soft_clips_both_ends'
      label: Dont Require Soft Clips Both Ends
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --dont-require-soft-clips-both-ends
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Allow a read to be filtered out based on having only 1 soft-clipped block. By default, both ends must have a soft-clipped block, setting this flag requires only 1 soft-clipped block. Default value: false. Possible values: {true, false}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#dont_increase_kmer_sizes_for_cycles'
      label: Dont Increase Kmer Sizes For Cycles
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --dont-increase-kmer-sizes-for-cycles
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Disable iterating over kmer sizes when graph cycles are detected Default value: false. Possible values: {true, false}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#do_not_run_physical_phasing'
      label: Do Not Run Physical Phasing
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --do-not-run-physical-phasing
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: 'Disable physical phasing Default value: false. Possible values:
        {true, false}.'
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#disable_tool_default_read_filters'
      label: Disable Tool Default Read Filters
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --disable-tool-default-read-filters
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Disable all tool default read filters Default value: false. Possible values: {true, false}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#disable_sequence_dictionary_validation'
      label: Disable Sequence Dictionary Validation
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --disable-sequence-dictionary-validation
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        If specified, do not check the sequence dictionaries from our inputs for compatibility. Use at your own risk! Default value: false. Possible values: {true, false}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#disable_read_filter'
      label: Disable Read Filter
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --disable-read-filter
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Filters to be disabled before analysis This argument may be specified 0 or more times. Default value: null.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'null'
    - id: '#disable_optimizations'
      label: Disable Optimizations
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --disable-optimizations
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Don't skip calculations in ActiveRegions with no variants Default value: false. Possible values: {true, false}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#disable_bam_index_caching'
      label: Disable Bam Index Caching
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --disable-bam-index-caching
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        If true, don't cache bam indexes, this will reduce memory requirements but may harm performance if many intervals are specified. Caching is automatically disabled if there are no intervals specified. Default value: false. Possible values: {true, false}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#debug'
      label: Debug
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --debug
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Print out very verbose debug information about each triggering active region Default value: false. Possible values: {true, false}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#create_output_variant_md5'
      label: Create Output Variant Md5
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --create-output-variant-md5
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        If true, create a a MD5 digest any VCF file created. Default value: false. Possible values: {true, false}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#create_output_variant_index'
      label: Create Output Variant Index
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --create-output-variant-index
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        If true, create a VCF index when writing a coordinate-sorted VCF file. Default value: true. Possible values: {true, false}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'true'
    - id: '#create_output_bam_md5'
      label: Create Output Bam Md5
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --create-output-bam-md5
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        If true, create a MD5 digest for any BAM/SAM/CRAM file created Default value: false. Possible values: {true, false}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#create_output_bam_index'
      label: Create Output Bam Index
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --create-output-bam-index
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        If true, create a BAM/CRAM index when writing a coordinate-sorted BAM/CRAM file. Default value: true. Possible values: {true, false}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'true'
    - id: '#contamination_table'
      label: Contamination Table
      type:
      - 'null'
      - File
      inputBinding:
        prefix: --contamination-table
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: 'Table containing contamination information. Default value: null.'
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'null'
    - id: '#contamination_fraction_to_filter'
      label: Contamination Fraction To Filter
      type:
      - 'null'
      - float
      inputBinding:
        prefix: --contamination-fraction-to-filter
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Fraction of contamination in sequencing data (for all samples) to aggressively remove Default value: 0.0.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '0'
    - id: '#contamination_fraction_per_sample_file'
      label: Contamination Fraction Per Sample File
      type:
      - 'null'
      - File
      inputBinding:
        prefix: --contamination-fraction-per-sample-file
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Tab-separated File containing fraction of contamination in sequencing data (per sample) to aggressively remove. Format should be "<SampleID><TAB><Contamination>" (Contamination is double) per line; No header. Default value: null.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: 'null'
    - id: '#consensus'
      label: Consensus
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --consensus
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: 'Consensus mode Default value: false. Possible values: {true, false}.'
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#comp'
      label: Comp
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --comp
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Comparison VCF file(s) This argument may be specified 0 or more times. Default value: null.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: 'null'
    - id: '#cloud_prefetch_buffer'
      label: Cloud Prefetch Buffer
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --cloud-prefetch-buffer
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: 'Size of the cloud-only prefetch buffer (in MB; 0 to disable).
        Default value: 40.'
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '40'
    - id: '#cloud_index_prefetch_buffer'
      label: Cloud Index Prefetch Buffer
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --cloud-index-prefetch-buffer
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Size of the cloud-only prefetch buffer (in MB; 0 to disable). Defaults to cloudPrefetchBuffer if unset. Default value: -1.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '-1'
    - id: '#black_listed_lanes'
      label: Black Listed Lanes
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --black-listed-lanes
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Keep reads with platform units not on the list This argument must be specified at least once. Required.
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
      sbg:toolDefaultValue:
    - id: '#black_list'
      label: Black List
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --black-list
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: This argument must be specified at least once. Required.
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
      sbg:toolDefaultValue:
    - id: '#base_quality_score_threshold'
      label: Base Quality Score Threshold
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --base-quality-score-threshold
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Base qualities below this threshold will be reduced to the minimum (6) Default value: 18.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '18'
    - id: '#bam_writer_type'
      label: Bam Writer Type
      type:
      - 'null'
      - name: 'null'
        type: enum
        symbols:
        - ALL_POSSIBLE_HAPLOTYPES
        - CALLED_HAPLOTYPES
      inputBinding:
        prefix: --bam-writer-type
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Which haplotypes should be written to the BAM Default value: CALLED_HAPLOTYPES. Possible values: {ALL_POSSIBLE_HAPLOTYPES, CALLED_HAPLOTYPES}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: CALLED_HAPLOTYPES
    - id: '#bam_output'
      label: Bam Output
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --bam-output
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: 'File to which assembled haplotypes should be written Default value:
        null.'
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: 'null'
    - id: '#annotate_nda'
      label: Annotate Nda
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --annotate-nda
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        If provided, we will annotate records with the number of alternate alleles that were discovered (but not necessarily genotyped) at a given site Default value: false. Possible values: {true, false}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#ambig_filter_frac'
      label: Ambig Filter Frac
      type:
      - 'null'
      - float
      inputBinding:
        prefix: --ambig-filter-frac
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: 'Threshold fraction of ambiguous bases Default value: 0.05.'
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
      sbg:toolDefaultValue: '0'
    - id: '#allow_non_unique_kmers_in_ref'
      label: Allow Non Unique Kmers In Ref
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --allow-non-unique-kmers-in-ref
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Allow graphs that have non-unique kmers in the reference Default value: false. Possible values: {true, false}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#alleles'
      label: Alleles
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --alleles
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        The set of alleles at which to genotype when --genotyping_mode is GENOTYPE_GIVEN_ALLELES Default value: null.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'null'
    - id: '#all_site_p_ls'
      label: All Site P Ls
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --all-site-p-ls
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Annotate all sites with PLs Default value: false. Possible values: {true, false}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#add_output_vcf_command_line'
      label: Add Output Vcf Command Line
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --add-output-vcf-command-line
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        If true, adds a command line header line to created VCF files. Default value: true. Possible values: {true, false}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'true'
    - id: '#add_output_sam_program_record'
      label: Add Output Sam Program Record
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --add-output-sam-program-record
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        If true, adds a PG tag to created SAM/BAM/CRAM files. Default value: true. Possible values: {true, false}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'true'

    outputs:
    - id: '#output'
      label: Filtered VCF
      type:
      - 'null'
      - File
      outputBinding:
        glob: '*.vcf'
        sbg:inheritMetadataFrom: '#variant'
      description: The output filtered VCF file.
      sbg:fileTypes: VCF

    baseCommand:
    - /gatk/gatk
    - --java-options
    - class: Expression
      engine: '#cwl-js-engine'
      script: |-
        {
          if($job.inputs.memory_per_job){
          	return '\"-Xmx'.concat($job.inputs.memory_per_job, 'M') + '\"'
          }
          	return '\"-Xmx2048M\"'
        }
    - FilterMutectCalls
    arguments:
    - prefix: --output
      position: 0
      valueFrom:
        class: Expression
        engine: '#cwl-js-engine'
        script: |-
          {
            variant_name = [].concat($job.inputs.variant)[0].path.replace(/^.*[\\\/]/, '').split('.')
            variant_namebase = variant_name.slice(0, variant_name.length-1).join('.')
            
            return variant_namebase + '.filtered.vcf'
          }
      separate: true

    hints:
    - class: sbg:CPURequirement
      value: 1
    - class: sbg:MemRequirement
      value:
        class: Expression
        engine: '#cwl-js-engine'
        script: |-
          {
            if($job.inputs.memory_per_job){
              if($job.inputs.memory_overhead_per_job){
              	return $job.inputs.memory_per_job + $job.inputs.memory_overhead_per_job
              }
              else
            		return $job.inputs.memory_per_job
            }
            else if(!$job.inputs.memory_per_job && $job.inputs.memory_overhead_per_job){
          		return 2048 + $job.inputs.memory_overhead_per_job  
            }
            else
            	return 2048
          }
    - class: DockerRequirement
      dockerPull: pgc-images.sbgenomics.com/d3b-bixu/gatk:4.0.5.1
    id: michael_lloyd/mantis/gatk-filtermutectcalls-4051/0
    appUrl: |-
      /u/anuj_srivastava/pdx-wxs-tumor-normal/apps/#anuj_srivastava/pdx-wxs-tumor-normal/filter-mutect2-latest-gatk4/14
    description: |-
      ###**Overview**  

      Filters somatic SNVs and indels called by Mutect2. 

      ###**Inputs**  

      A VCF file containing variants called by Mutect2.

      ###**Output**  

      A VCF file containing filtered variants.

      ###**Usage example**  

          java -jar gatk.jar \  
               FilterMutectCalls \  
               --reference reference.fasta \  
               --variant variant.vcf \   
               --output output.vcf

      ###**IMPORTANT NOTICE**  

      Tools in GATK that require a fasta reference file also look for the reference file's corresponding *.fai* (fasta index) and *.dict* (fasta dictionary) files. The fasta index file allows random access to reference bases and the dictionary file is a dictionary of the contig names and sizes contained within the fasta reference. These two secondary files are essential for GATK to work properly. To append these two files to your fasta reference please use the '***SBG FASTA Indices***' tool within your GATK based workflow before using any of the GATK tools.
    sbg:appVersion:
    - sbg:draft-2
    sbg:categories:
    - GATK-4.0.5.1
    sbg:cmdPreview: |-
      /gatk/gatk --java-options "-Xmx9M" FilterMutectCalls --variant /path/to/variant.ext --output variant.filtered.vcf
    sbg:content_hash: a55e4243765cb489fa42e0c88c0524d197434b1a4b8958e33d48a29d475a4a5d8
    sbg:contributors:
    - michael_lloyd
    sbg:copyOf: michael_lloyd/jax-pdx-wes-tumor-only/gatk-filtermutectcalls-4051/2
    sbg:createdBy: michael_lloyd
    sbg:createdOn: 1565889735
    sbg:id: michael_lloyd/mantis/gatk-filtermutectcalls-4051/0
    sbg:image_url:
    sbg:job:
      allocatedResources:
        cpu: 1
        mem: 10
      inputs:
        QUIET: true
        addOutputSAMProgramRecord: true
        addOutputVCFCommandLine: true
        allSitePLs: true
        allowNonUniqueKmersInRef: true
        annotateNDA: true
        consensus: true
        createOutputBamIndex: true
        createOutputBamMD5: true
        createOutputVariantIndex: true
        createOutputVariantMD5: true
        debug: true
        disableBamIndexCaching: true
        disableOptimizations: true
        disableSequenceDictionaryValidation: true
        disableToolDefaultReadFilters: true
        doNotRunPhysicalPhasing: true
        dontIncreaseKmerSizesForCycles: true
        dontRequireSoftClipsBothEnds: true
        dontTrimActiveRegions: true
        dontUseSoftClippedBases: true
        gcs_max_retries: 9
        input:
          class: File
          secondaryFiles: []
          path: /path/to/input.ext
          size: 0
        interval_merging_rule: ALL
        keepReverse: true
        lenient: true
        memory_overhead_per_job: 1
        memory_per_job: 9
        minMedianBaseQuality: 7
        minMedianMappingQuality: 6
        recoverDanglingHeads: true
        reference:
          class: File
          secondaryFiles:
          - path: .fai
          - path: ^.dict
          path: /path/to/reference.ext
          size: 0
        uniqueAltReadCount: 5
        useDoublePrecision: true
        useFilteredReadsForAnnotations: true
        useNewAFCalculator: true
        use_jdk_deflater: true
        use_jdk_inflater: true
        variant:
          class: File
          secondaryFiles: []
          path: /path/to/variant.ext
          size: 0
    sbg:latestRevision: 0
    sbg:license: Open source BSD (3-clause) license
    sbg:links:
    - id: https://software.broadinstitute.org/gatk/
      label: Homepage
    - id: https://software.broadinstitute.org/gatk/documentation/tooldocs/current/
      label: Documentation
    - id: https://software.broadinstitute.org/gatk/download/
      label: Download
    sbg:modifiedBy: michael_lloyd
    sbg:modifiedOn: 1565889735
    sbg:project: michael_lloyd/mantis
    sbg:projectName: MSI, HRD, TMB
    sbg:publisher: sbg
    sbg:revision: 0
    sbg:revisionNotes: Copy of michael_lloyd/jax-pdx-wes-tumor-only/gatk-filtermutectcalls-4051/2
    sbg:revisionsInfo:
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1565889735
      sbg:revision: 0
      sbg:revisionNotes: Copy of michael_lloyd/jax-pdx-wes-tumor-only/gatk-filtermutectcalls-4051/2
    sbg:sbgMaintained: false
    sbg:toolAuthor: Broad Institute
    sbg:toolkit: GATK
    sbg:toolkitVersion: 4.0.5.1
    sbg:validationErrors: []
  inputs:
  - id: '#gatk_filtermutectcalls_4051.variant'
    source: '#gatk_mutect2_4051.vcf_output'
  - id: '#gatk_filtermutectcalls_4051.uniqueAltReadCount'
    default: 5
  - id: '#gatk_filtermutectcalls_4051.reference'
    source: '#reference'
  - id: '#gatk_filtermutectcalls_4051.input'
    source: '#SAMtools_Index_BAM_1.output_bam_file'
  outputs:
  - id: '#gatk_filtermutectcalls_4051.output'
  sbg:x: 1149.28857421875
  sbg:y: 99.29027557373047
- id: '#gatk_mutect2_4051'
  label: GATK Mutect2 latest - tumor only - 4.0.5.1
  in: []
  scatter: '#gatk_mutect2_4051.intervals'
  run:
    cwlVersion: sbg:draft-2
    class: CommandLineTool
    label: GATK Mutect2 latest - tumor only - 4.0.5.1
    $namespaces:
      sbg: https://sevenbridges.com

    requirements:
    - id: '#cwl-js-engine'
      class: ExpressionEngineRequirement
      requirements:
      - class: DockerRequirement
        dockerPull: rabix/js-engine
    - class: InlineJavascriptRequirement

    inputs:
    - id: '#verbosity'
      label: Verbosity
      type:
      - 'null'
      - name: verbosity
        type: enum
        symbols:
        - ERROR
        - WARNING
        - INFO
        - DEBUG
      inputBinding:
        prefix: --verbosity
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Control verbosity of logging.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: INFO
    - id: '#use_new_af_calculator'
      label: Use New Af Calculator
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --use-new-af-calculator
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: If provided, we will use the new AF model instead of the so-called
        exact model.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#use_jdk_inflater'
      label: Use Jdk Inflater
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --use-jdk-inflater
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Whether to use the JdkInflater (as opposed to IntelInflater).
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#use_jdk_deflater'
      label: Use Jdk Deflater
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --use-jdk-deflater
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Whether to use the JdkDeflater (as opposed to IntelDeflater).
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#use_filtered_reads_for_annotations'
      label: Use Filtered Reads For Annotations
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --use-filtered-reads-for-annotations
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Use the contamination-filtered read maps for the purposes of annotating variants.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: 'false'
    - id: '#use_double_precision'
      label: Use Double Precision
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --use-double-precision
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Use double precision in the native pairHmm. This is slower but matches the java implementation better.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#tumor_sample'
      label: Tumor Sample Name
      type:
      - string
      inputBinding:
        prefix: --tumor-sample
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: BAM sample name of tumor.
      required: true
      sbg:category: Required Arguments
    - id: '#tumor_lod_to_emit'
      label: Tumor Lod To Emit
      type:
      - 'null'
      - float
      inputBinding:
        prefix: --tumor-lod-to-emit
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: LOD threshold to emit tumor variant to VCF.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '3.0'
    - id: '#standard_min_confidence_threshold_for_calling'
      label: Standard Min Confidence Threshold For Calling
      type:
      - 'null'
      - float
      inputBinding:
        prefix: --standard-min-confidence-threshold-for-calling
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        The minimum phred-scaled confidence threshold at which variants should be called.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '10.0'
    - id: '#seconds_between_progress_updates'
      label: Seconds Between Progress Updates
      type:
      - 'null'
      - float
      inputBinding:
        prefix: --seconds-between-progress-updates
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Output traversal statistics every time this many seconds elapse.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '10.0'
    - id: '#sample_ploidy'
      label: Sample Ploidy
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --sample-ploidy
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Ploidy (number of chromosomes) per sample. For pooled data, set to (Number of samples in each pool * Sample Ploidy).
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '2'
    - id: '#sample'
      label: Sample
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --sample
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        The name of the sample(s) to keep, filtering out all others This argument must be specified at least once.
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
    - id: '#reference'
      label: Reference
      type:
      - File
      inputBinding:
        secondaryFiles:
        - .fai
        - ^.dict
        prefix: --reference
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Reference sequence file.
      required: true
      sbg:altPrefix: ''
      sbg:category: Required Arguments
      sbg:fileTypes: FASTA, FA
    - id: '#recover_dangling_heads'
      label: Recover Dangling Heads
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --recover-dangling-heads
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: This argument is deprecated since version 3.3
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#reads'
      label: Reads
      type:
      - File
      inputBinding:
        secondaryFiles:
        - .bai
        prefix: --input
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        BAM/SAM/CRAM file containing reads. This argument must be specified at least once.
      required: true
      sbg:category: Required Arguments
      sbg:fileTypes: CRAM, BAM, SAM
    - id: '#read_validation_stringency'
      label: Read Validation Stringency
      type:
      - 'null'
      - name: readValidationStringency
        type: enum
        symbols:
        - STRICT
        - LENIENT
        - SILENT
      inputBinding:
        prefix: --read-validation-stringency
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Validation stringency for all SAM/BAM/CRAM/SRA files read by this program. The default stringency value SILENT can improve performance when processing a BAM file in which variable-length data (read, qualities, tags) do not otherwise need to be decoded.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: SILENT
    - id: '#read_shard_size'
      label: Read Shard Size
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --read-shard-size
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Maximum size of each read shard, in bases. For good performance, this should be much larger than the maximum assembly region size.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '5000'
    - id: '#read_shard_padding'
      label: Read Shard Padding
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --read-shard-padding
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Each read shard has this many bases of extra context on each side. Read shards must have as much or more padding than assembly regions.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '100'
    - id: '#read_name'
      label: Read Name
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --read-name
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Keep only reads with this read name.
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
    - id: '#read_index'
      label: Read Index
      type:
      - 'null'
      - type: array
        items: File
      inputBinding:
        prefix: --read-index
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Indices to use for the read inputs. If specified, an index must be provided for every read input and in the same order as the read inputs. If this argument is not specified, the path to the index for each input will be inferred automatically. This argument may be specified 0 or more times.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue:
    - id: '#read_filter'
      label: Read Filter
      type:
      - 'null'
      - type: array
        items: string
      inputBinding:
        prefix: --read-filter
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Read filters to be applied before analysis. This argument may be specified 0 or more times.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
    - id: '#quiet'
      label: Quiet
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --quiet
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Whether to suppress job-summary info on System.err.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#pl_filter_name'
      label: Pl Filter Name
      type:
      - 'null'
      - type: array
        items: string
      inputBinding:
        prefix: --pl-filter-name
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Keep reads with RG:PL attribute containing this string This argument must be specified at least once.
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
      sbg:stageInput:
    - id: '#phred_scaled_global_read_mismapping_rate'
      label: Phred Scaled Global Read Mismapping Rate
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --phred-scaled-global-read-mismapping-rate
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: The global assumed mismapping rate for reads.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: '45'
    - id: '#pcr_indel_model'
      label: Pcr Indel Model
      type:
      - 'null'
      - name: pcr_indel_model
        type: enum
        symbols:
        - NONE
        - HOSTILE
        - AGGRESSIVE
        - CONSERVATIVE
      inputBinding:
        prefix: --pcr-indel-model
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: The PCR indel model to use.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: CONSERVATIVE
    - id: '#output_mode'
      label: Output Mode
      type:
      - 'null'
      - name: output_mode
        type: enum
        symbols:
        - EMIT_VARIANTS_ONLY
        - EMIT_ALL_CONFIDENT_SITES
        - EMIT_ALL_SITES
      inputBinding:
        prefix: --output-mode
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Specifies which type of calls we should output.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: EMIT_VARIANTS_ONLY
    - id: '#num_pruning_samples'
      label: Num Pruning Samples
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --num-pruning-samples
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Number of samples that must pass the minPruning threshold.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: '1'
    - id: '#normal_sample'
      label: Normal Sample Name
      type:
      - 'null'
      - string
      description: BAM sample name of normal.
      required: true
      sbg:category: Optional Arguments
    - id: '#normal_panel'
      label: Normal Panel
      type:
      - 'null'
      - File
      inputBinding:
        prefix: --normal-panel
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: VCF file of sites observed in normal.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
    - id: '#normal_lod'
      label: Normal Lod
      type:
      - 'null'
      - float
      inputBinding:
        prefix: --normal-lod
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: LOD threshold for calling normal variant non-germline.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '2.2'
    - id: '#native_pair_hmm_threads'
      label: Native Pair Hmm Threads
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --native-pair-hmm-threads
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: How many threads should a native pairHMM implementation use.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '4'
    - id: '#minimum_mapping_quality'
      label: Minimum Mapping Quality
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --minimum-mapping-quality
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Minimum mapping quality to keep (inclusive).
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
      sbg:toolDefaultValue: '20'
    - id: '#min_read_length'
      label: Min Read Length
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --min-read-length
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Keep only reads with length at least equal to the specified value.
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
      sbg:toolDefaultValue: '1'
    - id: '#min_pruning'
      label: Min Pruning
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --min-pruning
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Minimum support to not prune paths in the graph.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: '2'
    - id: '#min_dangling_branch_length'
      label: Min Dangling Branch Length
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --min-dangling-branch-length
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Minimum length of a dangling branch to attempt recovery.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: '4'
    - id: '#min_base_quality_score'
      label: Min Base Quality Score
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --min-base-quality-score
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Minimum base quality required to consider a base for calling.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '10'
    - id: '#min_assembly_region_size'
      label: Min Assembly Region Size
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --min-assembly-region-size
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Minimum size of an assembly region.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '50'
    - id: '#memory_overhead_per_job'
      label: Memory Overhead Per Job
      type:
      - 'null'
      - int
      description: |-
        Memory overhead per job. By default this parameter value is set to '0' (zero megabytes). This parameter value is added to the Memory per job parameter value. This results in the allocation of the sum total (Memory per job and Memory overhead per job) amount of memory per job. By default the memory per job parameter value is set to 2048 megabytes, unless specified otherwise.
      required: false
      sbg:category: Execution
      sbg:stageInput:
      sbg:toolDefaultValue: '0'
    - id: '#maximum_mapping_quality'
      label: Maximum Mapping Quality
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --maximum-mapping-quality
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Maximum mapping quality to keep (inclusive).
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
    - id: '#max_reads_per_alignment_start'
      label: Max Reads Per Alignment Start
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --max-reads-per-alignment-start
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Maximum number of reads to retain per alignment start position. Reads above this threshold will be downsampled. Set to 0 to disable.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '50'
    - id: '#max_read_length'
      label: Max Read Length
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --max-read-length
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Keep only reads with length at most equal to the specified value.
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
    - id: '#max_prob_propagation_distance'
      label: Max Prob Propagation Distance
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --max-prob-propagation-distance
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Upper limit on how many bases away probability mass can be moved around when calculating the boundaries between active and inactive assembly regions.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: '50'
    - id: '#max_population_af'
      label: Max Population Af
      type:
      - 'null'
      - float
      inputBinding:
        prefix: --max-population-af
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Maximum population allele frequency in tumor-only mode.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: '0.01'
    - id: '#max_num_haplotypes_in_population'
      label: Max Num Haplotypes In Population
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --max-num-haplotypes-in-population
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Maximum number of haplotypes to consider for your population.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: '128'
    - id: '#max_genotype_count'
      label: Max Genotype Count
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --max-genotype-count
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Maximum number of genotypes to consider at any site.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: '1024'
    - id: '#max_fragment_length'
      label: Max Fragment Length
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --max-fragment-length
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Keep only read pairs with fragment length at most equal to the
        given value.
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
      sbg:toolDefaultValue: '1000000'
    - id: '#max_assembly_region_size'
      label: Max Assembly Region Size
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --max-assembly-region-size
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Maximum size of an assembly region.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '300'
    - id: '#max_alternate_alleles'
      label: Max Alternate Alleles
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --max-alternate-alleles
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Maximum number of alternate alleles to genotype.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: '6'
    - id: '#log_somatic_prior'
      label: Log Somatic Prior
      type:
      - 'null'
      - float
      inputBinding:
        prefix: --log-somatic-prior
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Prior probability that a given site has a somatic allele.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '-6.0'
    - id: '#library'
      label: Library
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --library
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: The name of the library to keep.
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
    - id: '#lenient'
      label: Lenient
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --lenient
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Lenient processing of VCF files.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#kmer_size'
      label: Kmer Size
      type:
      - 'null'
      - type: array
        items: int
      inputBinding:
        prefix: --kmer-size
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Kmer size to use in the read threading assembler This argument may be specified 0 or more times. Default value: [10, 25].
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:stageInput:
    - id: '#keep_reverse'
      label: Keep Reverse
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --keep-reverse
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Keep only reads on the reverse strand.
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
      sbg:stageInput:
    - id: '#keep_read_group'
      label: Keep Read Group
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --keep-read-group
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: The name of the read group to keep Required.
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
      sbg:toolDefaultValue:
    - id: '#intervals'
      label: Intervals
      type:
      - 'null'
      - type: array
        items: File
      inputBinding:
        prefix: --intervals
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        One or more genomic intervals over which to operate. This argument may be specified 0 or more times.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:fileTypes: TXT, BED, VCF
      sbg:stageInput:
    - id: '#interval_set_rule'
      label: Interval Set Rule
      type:
      - 'null'
      - name: interval_set_rule
        type: enum
        symbols:
        - UNION
        - INTERSECTION
      inputBinding:
        prefix: --interval-set-rule
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Set merging approach to use for combining interval inputs.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: UNION
    - id: '#interval_padding'
      label: Interval Padding
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --interval-padding
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Of padding (in bp) to add to each interval you are including.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '0'
    - id: '#interval_merging_rule'
      label: Interval Merging Rule
      type:
      - 'null'
      - name: interval_merging_rule
        type: enum
        symbols:
        - ALL
        - OVERLAPPING_ONLY
      inputBinding:
        prefix: --interval-merging-rule
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Interval merging rule for abutting intervals.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: ALL
    - id: '#interval_exclusion_padding'
      label: Interval Exclusion Padding
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --interval-exclusion-padding
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Amount of padding (in bp) to add to each interval you are excluding.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '0'
    - id: '#input_prior'
      label: Input Prior
      type:
      - 'null'
      - type: array
        items: float
      inputBinding:
        prefix: --input-prior
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Input prior for calls. This argument may be specified 0 or more
        times.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:stageInput:
    - id: '#initial_tumor_lod'
      label: Initial Tumor Lod
      type:
      - 'null'
      - float
      inputBinding:
        prefix: --initial-tumor-lod
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: LOD threshold to consider pileup active.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: '2.0'
    - id: '#indel_heterozygosity'
      label: Indel Heterozygosity
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --indel-heterozygosity
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Heterozygosity for indel calling. See the GATKDocs for heterozygosity for full details on the meaning of this population genetics concept.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '1.25E-4'
    - id: '#heterozygosity_stdev'
      label: Heterozygosity Stdev
      type:
      - 'null'
      - float
      inputBinding:
        prefix: --heterozygosity-stdev
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Standard deviation of heterozygosity for SNP and indel calling.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '0.01'
    - id: '#heterozygosity'
      label: Heterozygosity
      type:
      - 'null'
      - float
      inputBinding:
        prefix: --heterozygosity
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Heterozygosity value used to compute prior likelihoods for any locus. See the GATKDocs for full details on the meaning of this population genetics concept.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '0.001'
    - id: '#group'
      label: Group
      type:
      - 'null'
      - type: array
        items: string
      inputBinding:
        prefix: --group
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        One or more classes/groups of annotations to apply to variant calls. This argument may be specified 0 or more times.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
    - id: '#graph_output'
      label: Graph Output
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --graph-output
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Write debug assembly graph information to this file.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
    - id: '#germline_resource'
      label: Germline Resource
      type:
      - 'null'
      - File
      inputBinding:
        secondaryFiles:
        - .tbi
        prefix: --germline-resource
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Population vcf of germline sequencing containing allele fractions.
      required: false
      sbg:category: Optional Arguments
    - id: '#genotyping_mode'
      label: Genotyping Mode
      type:
      - 'null'
      - name: genotyping_mode
        type: enum
        symbols:
        - DISCOVERY
        - GENOTYPE_GIVEN_ALLELES
      inputBinding:
        prefix: --genotyping-mode
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Specifies how to determine the alternate alleles to use for genotyping.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: DISCOVERY
    - id: '#genotype_pon_sites'
      label: Genotype Pon Sites
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --genotype-pon-sites
        position: 0
        separate: false
        sbg:cmdInclude: true
      description: Whether to call sites in the PoN even though they will ultimately
        be filtered.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'false'
    - id: '#gcs_max_retries'
      label: Gcs Max Retries
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --gcs-max-retries
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        If the GCS bucket channel errors out, how many times it will attempt to re-initiate the connection.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: '20'
    - id: '#gcp_hmm'
      label: Gcp Hmm
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --gcp-hmm
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Flat gap continuation penalty for use in the Pair HMM.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: '10'
    - id: '#filter_too_short'
      label: Filter Too Short
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --filter-too-short
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Value for which reads with less than this number of aligned bases is considered too short.
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
      sbg:toolDefaultValue: '30'
    - id: '#exclude_intervals'
      label: Exclude Intervals
      type:
      - 'null'
      - type: array
        items: File
      inputBinding:
        prefix: --exclude-intervals
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        One or more genomic intervals to exclude from processing. This argument may be specified 0 or more times.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:fileTypes: TXT, BED, VCF
      sbg:stageInput:
    - id: '#exclude_annotation'
      label: Exclude Annotation
      type:
      - 'null'
      - type: array
        items: string
      inputBinding:
        prefix: --exclude-annotation
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Or more specific annotations to exclude This argument may be specified 0 or more times.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:stageInput:
      sbg:toolDefaultValue:
    - id: '#emit_ref_confidence'
      label: Emit Ref Confidence
      type:
      - 'null'
      - name: emitRefConfidence
        type: enum
        symbols:
        - NONE
        - BP_RESOLUTION
        - GVCF
      inputBinding:
        prefix: --emit-ref-confidence
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Mode for emitting reference confidence scores.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: NONE
    - id: '#dont_use_soft_clipped_bases'
      label: Dont Use Soft Clipped Bases
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --dont-use-soft-clipped-bases
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Do not analyze soft clipped bases in the reads.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: 'false'
    - id: '#dont_trim_active_regions'
      label: Dont Trim Active Regions
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --dont-trim-active-regions
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        If specified, we will not trim down the active region from the full region (active + extension) to just the active interval for genotyping.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: 'false'
    - id: '#dont_require_soft_clips_both_ends'
      label: Dont Require Soft Clips Both Ends
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --dont-require-soft-clips-both-ends
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Allow a read to be filtered out based on having only 1 soft-clipped block. By default, both ends must have a soft-clipped block, setting this flag requires only 1 soft-clipped block.
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#dont_increase_kmer_sizes_for_cycles'
      label: Dont Increase Kmer Sizes For Cycles
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --dont-increase-kmer-sizes-for-cycles
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Disable iterating over kmer sizes when graph cycles are detected.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: 'false'
    - id: '#do_not_run_physical_phasing'
      label: Do Not Run Physical Phasing
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --do-not-run-physical-phasing
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Disable physical phasing.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: 'false'
    - id: '#disable_tool_default_read_filters'
      label: Disable Tool Default Read Filters
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --disable-tool-default-read-filters
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Disable all tool default read filters.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#disable_sequence_dictionary_validation'
      label: Disable Sequence Dictionary Validation
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --disable-sequence-dictionary-validation
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        If specified, do not check the sequence dictionaries from our inputs for compatibility. Use at your own risk!
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#disable_read_filter'
      label: Disable Read Filter
      type:
      - 'null'
      - name: 'null'
        type: enum
        symbols:
        - GoodCigarReadFilter
        - MappedReadFilter
        - MappingQualityAvailableReadFilter
        - MappingQualityReadFilter
        - NonZeroReferenceLengthAlignmentReadFilter
        - NotDuplicateReadFilter
        - NotSecondaryAlignmentReadFilter
        - PassesVendorQualityCheckReadFilter
        - WellformedReadFilter
      inputBinding:
        prefix: --disable-read-filter
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Filters to be disabled before analysis This argument may be specified 0 or more times.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
    - id: '#disable_optimizations'
      label: Disable Optimizations
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --disable-optimizations
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Don't skip calculations in ActiveRegions with no variants.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: 'false'
    - id: '#disable_bam_index_caching'
      label: Disable Bam Index Caching
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --disable-bam-index-caching
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        If true, don't cache bam indexes, this will reduce memory requirements but may harm performance if many intervals are specified. Caching is automatically disabled if there are no intervals specified.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#debug'
      label: Debug
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --debug
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Print out very verbose debug information about each triggering
        active region.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#create_output_variant_md5'
      label: Create Output Variant Md5
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --create-output-variant-md5
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        If true, create a a MD5 digest any VCF file created. Default value: false. Possible values: {true, false}.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#create_output_variant_index'
      label: Create Output Variant Index
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --create-output-variant-index
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: If true, create a VCF index when writing a coordinate-sorted VCF
        file.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'true'
    - id: '#create_output_bam_md5'
      label: Create Output Bam Md5
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --create-output-bam-md5
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: If true, create a MD5 digest for any BAM/SAM/CRAM file created.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'false'
    - id: '#create_output_bam_index'
      label: Create Output Bam Index
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --create-output-bam-index
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: If true, create a BAM/CRAM index when writing a coordinate-sorted
        BAM/CRAM file.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:stageInput:
      sbg:toolDefaultValue: 'true'
    - id: '#cpu_per_job'
      label: Cpu Per Job
      type:
      - 'null'
      - int
      description: Number of CPUs per job.
      required: false
      sbg:category: Execution
      sbg:stageInput:
      sbg:toolDefaultValue: '1'
    - id: '#contamination_fraction_to_filter'
      label: Contamination Fraction To Filter
      type:
      - 'null'
      - float
      inputBinding:
        prefix: --contamination-fraction-to-filter
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Fraction of contamination in sequencing data (for all samples) to aggressively remove.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '0.0'
    - id: '#contamination_fraction_per_sample_file'
      label: Contamination Fraction Per Sample File
      type:
      - 'null'
      - File
      inputBinding:
        prefix: --contamination-fraction-per-sample-file
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Tab-separated File containing fraction of contamination in sequencing data (per sample) to aggressively remove. Format should be "<SampleID><TAB><Contamination>" (Contamination is double) per line; No header.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue:
    - id: '#consensus'
      label: Consensus
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --consensus
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Consensus mode
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: 'false'
    - id: '#comp'
      label: Comp
      type:
      - 'null'
      - type: array
        items: File
      inputBinding:
        prefix: --comp
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Comparison VCF file(s) This argument may be specified 0 or more
        times.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:stageInput:
    - id: '#cloud_prefetch_buffer'
      label: Cloud Prefetch Buffer
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --cloud-prefetch-buffer
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Size of the cloud-only prefetch buffer (in MB; 0 to disable).
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '40'
    - id: '#cloud_index_prefetch_buffer'
      label: Cloud Index Prefetch Buffer
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --cloud-index-prefetch-buffer
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Size of the cloud-only prefetch buffer (in MB; 0 to disable). Defaults to cloudPrefetchBuffer if unset.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '-1'
    - id: '#black_listed_lanes'
      label: Black Listed Lanes
      type:
      - 'null'
      - type: array
        items: string
      inputBinding:
        prefix: --black-listed-lanes
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Keep reads with platform units not on the list This argument must be specified at least once.
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
      sbg:stageInput:
    - id: '#black_list'
      label: Black List
      type:
      - 'null'
      - type: array
        items: string
      inputBinding:
        prefix: --black-list
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: This argument must be specified at least once.
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
      sbg:stageInput:
    - id: '#base_quality_score_threshold'
      label: Base Quality Score Threshold
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --base-quality-score-threshold
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Base qualities below this threshold will be reduced to the minimum
        (6).
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '18'
    - id: '#bam_writer_type'
      label: Bam Writer Type
      type:
      - 'null'
      - name: bamWriterType
        type: enum
        symbols:
        - ALL_POSSIBLE_HAPLOTYPES
        - CALLED_HAPLOTYPES
      inputBinding:
        prefix: --bam-writer-type
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Which haplotypes should be written to the BAM.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: CALLED_HAPLOTYPES
    - id: '#bam_output'
      label: Bam Output
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --bam-output
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: File to which assembled haplotypes should be written.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
    - id: '#assembly_region_padding'
      label: Assembly Region Padding
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --assembly-region-padding
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Number of additional bases of context to include around each assembly
        region.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '100'
    - id: '#annotation'
      label: Annotation
      type:
      - 'null'
      - type: array
        items: string
      inputBinding:
        prefix: --annotation
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        One or more specific annotations to apply to variant calls. This argument may be specified 0 or more times. Default value: [Coverage, DepthPerAlleleBySample, TandemRepeat, OxoGReadCounts, ReadPosition, BaseQuality, MappingQuality, FragmentLength, StrandArtifact].
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
    - id: '#annotate_nda'
      label: Annotate Nda
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --annotate-nda
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        If provided, we will annotate records with the number of alternate alleles that were discovered (but not necessarily genotyped) at a given site.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'false'
    - id: '#ambig_filter_frac'
      label: Ambig Filter Frac
      type:
      - 'null'
      - float
      inputBinding:
        prefix: --ambig-filter-frac
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Threshold fraction of ambiguous bases. Cannot be used in conjunction with argument(s) maxAmbiguousBases.
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
      sbg:toolDefaultValue: '0.05'
    - id: '#ambig_filter_bases'
      label: Ambig Filter Bases
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --ambig-filter-bases
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Threshold number of ambiguous bases. If null, uses threshold fraction; otherwise, overrides threshold fraction. Cannot be used in conjunction with argument(s) maxAmbiguousBaseFraction.
      required: false
      sbg:altPrefix: ''
      sbg:category: Conditional Arguments for readFilter
      sbg:stageInput:
    - id: '#allow_non_unique_kmers_in_ref'
      label: Allow Non Unique Kmers In Ref
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --allow-non-unique-kmers-in-ref
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Allow graphs that have non-unique kmers in the reference.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: 'false'
    - id: '#alleles'
      label: Alleles
      type:
      - 'null'
      - File
      inputBinding:
        prefix: --alleles
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        The set of alleles at which to genotype when --genotyping_mode is GENOTYPE_GIVEN_ALLELES.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
    - id: '#all_site_p_ls'
      label: All Site P Ls
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --all-site-p-ls
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Annotate all sites with PLs
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: 'false'
    - id: '#af_of_alleles_not_in_resource'
      label: Af Of Alleles Not In Resource
      type:
      - 'null'
      - float
      inputBinding:
        prefix: --af-of-alleles-not-in-resource
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: |-
        Population allele fraction assigned to alleles not found in germline resource. A reasonable value is1/(2* number of samples in resource) if a germline resource is available; otherwise an average heterozygosity rate such as 0.001 is reasonable.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '0.001'
    - id: '#add_output_vcf_command_line'
      label: Add Output Vcf Command Line
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --add-output-vcf-command-line
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: If true, adds a command line header line to created VCF files.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'true'
    - id: '#add_output_sam_program_record'
      label: Add Output Sam Program Record
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --add-output-sam-program-record
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: If true, adds a PG tag to created SAM/BAM/CRAM files.
      required: false
      sbg:altPrefix: ''
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'true'
    - id: '#active_probability_threshold'
      label: Active Probability Threshold
      type:
      - 'null'
      - float
      inputBinding:
        prefix: --active-probability-threshold
        position: 0
        separate: true
        sbg:cmdInclude: true
      description: Minimum probability for a locus to be considered active.
      required: false
      sbg:altPrefix: ''
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: '0.002'

    outputs:
    - id: '#vcf_output'
      label: Variant VCF
      type:
      - 'null'
      - File
      outputBinding:
        secondaryFiles:
        - .idx
        glob: '*.vcf'
        sbg:inheritMetadataFrom: '#reads'
        sbg:metadata:
          sample_id:
            class: Expression
            engine: '#cwl-js-engine'
            script: |-
              {
                test = [].concat($job.inputs.reads)
                if (((test[0].metadata) && (test[0].metadata['sample_id'])))
                {
                  return test[0].metadata['sample_id'].concat('_only')
                }
                else
                {
                  return 'tumor'
                }
              }
      description: File to which variants should be written.
      sbg:fileTypes: VCF
    - id: '#vcf_md5'
      label: Variant md5
      type:
      - 'null'
      - File
      outputBinding:
        glob: '*.vcf.md5'
      description: MD5 digest of created VCF file.
      sbg:fileTypes: MD5
    - id: '#vcf_idx'
      label: Variant index
      type:
      - 'null'
      - File
      outputBinding:
        glob: '*.vcf.idx'
      description: VCF index file.
      sbg:fileTypes: IDX
    - id: '#output_graph'
      label: Assembly graph
      type:
      - 'null'
      - File
      outputBinding:
        glob:
          class: Expression
          engine: '#cwl-js-engine'
          script: "{\n  if ($job.inputs.graphOutput) {\n    return $job.inputs.graphOutput\n\
            \  }\n}"
      description: Debug assembly graph information.

    baseCommand:
    - /gatk/gatk
    - --java-options
    - class: Expression
      engine: '#cwl-js-engine'
      script: |-
        {
          if($job.inputs.memory_per_job){
          	return '\"-Xmx'.concat($job.inputs.memory_per_job, 'M') + '\"'
          }
          	return '\"-Xmx2048M\"'
        }
    - Mutect2
    arguments:
    - prefix: --output
      position: 0
      valueFrom:
        class: Expression
        engine: '#cwl-js-engine'
        script: |
          {
            test = [].concat($job.inputs.reads)
            tempout = ''
            if ((test[0].metadata) && (test[0].metadata['sample_id']))
            {
              tempout = tempout.concat(test[0].metadata['sample_id'], '_')
            } else {
              tempout = 'missing_sample_id_metadata'
            }
            tempout = tempout.concat("somatic.vcf")
            return tempout
          }
      separate: true

    hints:
    - class: sbg:CPURequirement
      value: 1
    - class: sbg:MemRequirement
      value:
        class: Expression
        engine: '#cwl-js-engine'
        script: |-
          {
            if($job.inputs.memory_per_job){
              if($job.inputs.memory_overhead_per_job){
              	return $job.inputs.memory_per_job + $job.inputs.memory_overhead_per_job
              }
              else
            		return $job.inputs.memory_per_job
            }
            else if(!$job.inputs.memory_per_job && $job.inputs.memory_overhead_per_job){
          		return 2048 + $job.inputs.memory_overhead_per_job  
            }
            else
            	return 2048
          }
    - class: DockerRequirement
      dockerPull: pgc-images.sbgenomics.com/d3b-bixu/gatk:4.0.5.1
    id: michael_lloyd/mantis/gatk-mutect2-4051/0
    appUrl: |-
      /u/anuj_srivastava/pdx-wxs-tumor-normal/apps/#anuj_srivastava/pdx-wxs-tumor-normal/mutect2-latest-gatk4/4
    description: |-
      ###**Overview**  

      Call somatic SNVs and indels via local assembly of haplotypes.

      ###**Inputs**  

      A BAM file containing normal data.

      A BAM file containing tumor data.

      The name of the normal sample.

      The name of the tumor sample.

      ###**Output**  

      A VCF file with called variants.

      ###**Usage example**  

          java -jar gatk.jar \  
               Mutect2 \  
               --reference reference.fasta \  
               --input normal.bam \   
               --input tumor.bam \ 
               --normalSampleName normal \
               --tumorSampleName tumor \
               --output variant.vcf

      ###**IMPORTANT NOTICE**  

      Tools in GATK that require a fasta reference file also look for the reference file's corresponding *.fai* (fasta index) and *.dict* (fasta dictionary) files. The fasta index file allows random access to reference bases and the dictionary file is a dictionary of the contig names and sizes contained within the fasta reference. These two secondary files are essential for GATK to work properly. To append these two files to your fasta reference please use the '***SBG FASTA Indices***' tool within your GATK based workflow before using any of the GATK tools.
    sbg:appVersion:
    - sbg:draft-2
    sbg:categories:
    - GATK-4.0.5.1
    sbg:cmdPreview: |-
      /gatk/gatk --java-options "-Xmx2048M" Mutect2 --tumor-sample tumor --reference /path/to/reference.ext --input /path/to/reads.ext --output missing_sample_id_metadatasomatic.vcf
    sbg:content_hash: a947a89717e7b263f938303e20edab9c7f4fd442fc1414f183d3d3f4a00cb3c53
    sbg:contributors:
    - michael_lloyd
    sbg:copyOf: michael_lloyd/jax-pdx-wes-tumor-only/gatk-mutect2-4051/2
    sbg:createdBy: michael_lloyd
    sbg:createdOn: 1565889732
    sbg:id: michael_lloyd/mantis/gatk-mutect2-4051/0
    sbg:image_url:
    sbg:job:
      allocatedResources:
        cpu: 1
        mem: 2056
      inputs:
        PLFilterName:
        QUIET: false
        addOutputSAMProgramRecord: false
        addOutputVCFCommandLine: false
        allSitePLs: false
        allowNonUniqueKmersInRef: false
        ambigFilterBases:
        annotateNDA: false
        annotation:
        blackList:
        blackListedLanes:
        comp:
        consensus: false
        cpu_per_job: 4
        createOutputBamIndex: false
        createOutputBamMD5: false
        createOutputVariantIndex: false
        createOutputVariantMD5: true
        debug: false
        disableBamIndexCaching: false
        disableOptimizations: false
        disableReadFilter:
        disableSequenceDictionaryValidation: false
        disableToolDefaultReadFilters: false
        doNotRunPhysicalPhasing: false
        dontIncreaseKmerSizesForCycles: false
        dontRequireSoftClipsBothEnds: false
        dontTrimActiveRegions: false
        dontUseSoftClippedBases: false
        excludeAnnotation:
        excludeIntervals:
        gcs_max_retries: 3
        genotypePonSites: false
        group:
        indel_heterozygosity: ''
        initial_tumor_lod:
        input_prior:
        interval_merging_rule: ALL
        intervals:
        keepReverse: false
        kmerSize:
        lenient: false
        max_population_af:
        memory_overhead_per_job: 8
        normalSampleName: normalSampleName-string-value
        normal_sample: normal
        readFilter:
        readIndex:
        reads:
          class: File
          secondaryFiles:
          - path: .bai
          path: /path/to/reads.ext
          size: 0
        recoverDanglingHeads: false
        reference:
          class: File
          secondaryFiles:
          - path: .fai
          - path: ^.dict
          path: /path/to/reference.ext
          size: 0
        tumorSampleName: tumorSampleName-string-value
        tumor_sample: tumor
        useDoublePrecision: false
        useFilteredReadsForAnnotations: false
        useNewAFCalculator: false
        use_jdk_deflater: false
        use_jdk_inflater: false
    sbg:latestRevision: 0
    sbg:license: Open source BSD (3-clause) license
    sbg:links:
    - id: https://software.broadinstitute.org/gatk/
      label: Homepage
    - id: https://software.broadinstitute.org/gatk/documentation/tooldocs/current/
      label: Documentation
    - id: https://software.broadinstitute.org/gatk/download/
      label: Download
    sbg:modifiedBy: michael_lloyd
    sbg:modifiedOn: 1565889732
    sbg:project: michael_lloyd/mantis
    sbg:projectName: MSI, HRD, TMB
    sbg:publisher: sbg
    sbg:revision: 0
    sbg:revisionNotes: Copy of michael_lloyd/jax-pdx-wes-tumor-only/gatk-mutect2-4051/2
    sbg:revisionsInfo:
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1565889732
      sbg:revision: 0
      sbg:revisionNotes: Copy of michael_lloyd/jax-pdx-wes-tumor-only/gatk-mutect2-4051/2
    sbg:sbgMaintained: false
    sbg:toolAuthor: Broad Institute
    sbg:toolkit: GATK
    sbg:toolkitVersion: 4.0.5.1
    sbg:validationErrors: []
    sbg:wrapperAuthor: filip_tubic
  inputs:
  - id: '#gatk_mutect2_4051.tumor_sample'
    default: tumor
  - id: '#gatk_mutect2_4051.sample_ploidy'
    source: '#sample_ploidy'
  - id: '#gatk_mutect2_4051.reference'
    source: '#reference'
  - id: '#gatk_mutect2_4051.reads'
    source: '#SAMtools_Index_BAM_1.output_bam_file'
  - id: '#gatk_mutect2_4051.intervals'
    source:
    - '#SBG_Prepare_Intervals.intervals'
  - id: '#gatk_mutect2_4051.germline_resource'
    source: '#germline_resource'
  - id: '#gatk_mutect2_4051.dont_use_soft_clipped_bases'
    default: true
  - id: '#gatk_mutect2_4051.annotation'
    default:
    - QualByDepth
    - RMSMappingQuality
    - FisherStrand
    - MappingQualityRankSumTest
    - ReadPosRankSumTest
  - id: '#gatk_mutect2_4051.af_of_alleles_not_in_resource'
    default: 8.2364e-06
  outputs:
  - id: '#gatk_mutect2_4051.vcf_output'
  - id: '#gatk_mutect2_4051.vcf_md5'
  - id: '#gatk_mutect2_4051.vcf_idx'
  - id: '#gatk_mutect2_4051.output_graph'
  sbg:x: 934.2427978515625
  sbg:y: 245.24783325195312
- id: '#gatk_variantannotator'
  label: GATK VariantAnnotator
  in: []
  run:
    cwlVersion: sbg:draft-2
    class: CommandLineTool
    label: GATK VariantAnnotator
    $namespaces:
      sbg: https://sevenbridges.com

    requirements:
    - id: '#cwl-js-engine'
      class: ExpressionEngineRequirement
      requirements:
      - class: DockerRequirement
        dockerPull: rabix/js-engine
    - class: InlineJavascriptRequirement

    inputs:
    - id: '#variant'
      label: Variant
      type:
      - File
      inputBinding:
        prefix: --variant
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: A VCF file containing variants.
      sbg:altPrefix: -V
      sbg:category: Required Arguments
      sbg:fileTypes: VCF
    - id: '#add_output_sam_program_record'
      label: Add Output Sam Program Record
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --add-output-sam-program-record
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: If true, adds a PG tag to created SAM/BAM/CRAM files.
      sbg:altPrefix: -add-output-sam-program-record
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'true'
    - id: '#add_output_vcf_command_line'
      label: Add Output Vcf Command Line
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --add-output-vcf-command-line
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: If true, adds a command line header line to created VCF files.
      sbg:altPrefix: -add-output-vcf-command-line
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'true'
    - id: '#annotation'
      label: Annotation
      type:
      - 'null'
      - type: array
        items:
          name: annotation
          type: enum
          symbols:
          - AS_BaseQualityRankSumTest
          - AS_FisherStrand
          - AS_InbreedingCoeff
          - AS_MappingQualityRankSumTest
          - AS_QualByDepth
          - AS_ReadPosRankSumTest
          - AS_RMSMappingQuality
          - AS_StrandOddsRatio
          - BaseQuality
          - BaseQualityRankSumTest
          - ChromosomeCounts
          - ClippingRankSumTest
          - Coverage
          - DepthPerAlleleBySample
          - DepthPerSampleHC
          - ExcessHet
          - FisherStrand
          - FragmentLength
          - GenotypeSummaries
          - InbreedingCoeff
          - LikelihoodRankSumTest
          - MappingQuality
          - MappingQualityRankSumTest
          - MappingQualityZero
          - OxoGReadCounts
          - PossibleDeNovo
          - QualByDepth
          - ReadPosition
          - ReadPosRankSumTest
          - ReferenceBases
          - RMSMappingQuality
          - SampleList
          - StrandArtifact
          - StrandBiasBySample
          - StrandOddsRatio
          - TandemRepeat
          - UniqueAltReadCount
      inputBinding:
        prefix: --annotation
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: One or more specific annotations to add to variant calls.
      sbg:altPrefix: -A
      sbg:category: Optional Arguments
    - id: '#annotation_group'
      label: Annotation Group
      type:
      - 'null'
      - type: array
        items:
          name: annotation_group
          type: enum
          symbols:
          - AS_StandardAnnotation
          - ReducibleAnnotation
          - StandardAnnotation
          - StandardHCAnnotation
          - StandardMutectAnnotation
      inputBinding:
        prefix: --annotation-group
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: One or more groups of annotations to apply to variant calls.
      sbg:altPrefix: -G
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'null'
    - id: '#annotations_to_exclude'
      label: Annotations To Exclude
      type:
      - 'null'
      - type: array
        items: string
      inputBinding:
        prefix: --annotations-to-exclude
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: |-
        One or more specific annotations to exclude from variant calls Any value allowed.
      sbg:altPrefix: -AX
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'null'
    - id: '#create_output_bam_index'
      label: Create Output Bam Index
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --create-output-bam-index
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: If true, create a BAM/CRAM index when writing a coordinate-sorted
        BAM/CRAM file.
      sbg:altPrefix: -OBI
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'true'
    - id: '#create_output_bam_md5'
      label: Create Output Bam Md5
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --create-output-bam-md5
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: If true, create a MD5 digest for any BAM/SAM/CRAM file created.
      sbg:altPrefix: -OBM
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'false'
    - id: '#create_output_variant_index'
      label: Create Output Variant Index
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --create-output-variant-index
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: If true, create a VCF index when writing a coordinate-sorted VCF
        file.
      sbg:altPrefix: -OVI
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'true'
    - id: '#create_output_variant_md5'
      label: Create Output Variant Md5
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --create-output-variant-md5
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: If true, create a a MD5 digest any VCF file created.
      sbg:altPrefix: -OVM
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'false'
    - id: '#dbsnp'
      label: Dbsnp
      type:
      - 'null'
      - File
      inputBinding:
        secondaryFiles:
        - .tbi
        prefix: --dbsnp
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: dbSNP file.
      sbg:altPrefix: -D
      sbg:category: Optional Arguments
      sbg:fileTypes: VCF.GZ
      sbg:toolDefaultValue:
    - id: '#disable_bam_index_caching'
      label: Disable Bam Index Caching
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --disable-bam-index-caching
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: |-
        If true, don't cache bam indexes, this will reduce memory requirements but may harm performance if many intervals are specified. Caching is automatically disabled if there are no intervals specified.
      sbg:altPrefix: -DBIC
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'false'
    - id: '#disable_read_filter'
      label: Disable Read Filter
      type:
      - 'null'
      - type: array
        items:
          name: disable_read_filter
          type: enum
          symbols:
          - MappedReadFilter
          - NotDuplicateReadFilter
          - PassesVendorQualityCheckReadFilter
          - PrimaryLineReadFilter
          - WellformedReadFilter
      inputBinding:
        prefix: --disable-read-filter
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: Read filters to be disabled before analysis.
      sbg:altPrefix: -DF
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'null'
    - id: '#disable_sequence_dictionary_validation'
      label: Disable Sequence Dictionary Validation
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --disable-sequence-dictionary-validation
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: |-
        If specified, do not check the sequence dictionaries from our inputs for compatibility. Use at your own risk!
      sbg:altPrefix: -disable-sequence-dictionary-validation
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'false'
    - id: '#exclude_intervals'
      label: Exclude Intervals
      type:
      - 'null'
      - type: array
        items: string
      inputBinding:
        prefix: --exclude-intervals
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: or more genomic intervals to exclude from processing.
      sbg:altPrefix: -XL
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'null'
    - id: '#expression'
      label: Expression
      type:
      - 'null'
      - type: array
        items: string
      inputBinding:
        prefix: --expression
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: One or more specific expressions to apply to variant calls.
      sbg:altPrefix: -E
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'null'
    - id: '#founder_id'
      label: Founder Id
      type:
      - 'null'
      - type: array
        items: string
      inputBinding:
        prefix: --founder-id
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: Samples representing the population "founders".
      sbg:altPrefix: -founder-id
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'null'
    - id: '#gcs_max_retries'
      label: Gcs Max Retries
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --gcs-max-retries
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: |-
        If the GCS bucket channel errors out, how many times it will attempt to re-initiate the connection.
      sbg:altPrefix: -gcs-retries
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '20'
    - id: '#input'
      label: Input
      type:
      - 'null'
      - type: array
        items: File
      inputBinding:
        secondaryFiles:
        - .bai
        prefix: --input
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: BAM/SAM/CRAM file containing reads.
      sbg:altPrefix: -I
      sbg:category: Optional Arguments
      sbg:fileTypes: BAM
      sbg:toolDefaultValue:
    - id: '#interval_exclusion_padding'
      label: Interval Exclusion Padding
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --interval-exclusion-padding
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: Amount of padding (in bp) to add to each interval you are excluding.
      sbg:altPrefix: -ixp
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '0'
    - id: '#interval_merging_rule'
      label: Interval Merging Rule
      type:
      - 'null'
      - name: interval_merging_rule
        type: enum
        symbols:
        - ALL
        - OVERLAPPING_ONLY
      inputBinding:
        prefix: --interval-merging-rule
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: Interval merging rule for abutting intervals.
      sbg:altPrefix: -imr
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: ALL
    - id: '#interval_padding'
      label: Interval Padding
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --interval-padding
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: of padding (in bp) to add to each interval you are including.
      sbg:altPrefix: -ip
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: '0'
    - id: '#interval_set_rule'
      label: Interval Set Rule
      type:
      - 'null'
      - name: interval_set_rule
        type: enum
        symbols:
        - UNION
        - INTERSECTION
      inputBinding:
        prefix: --interval-set-rule
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: Set merging approach to use for combining interval inputs.
      sbg:altPrefix: -isr
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: UNION
    - id: '#intervals'
      label: Intervals
      type:
      - 'null'
      - type: array
        items: string
      inputBinding:
        prefix: --intervals
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: One or more genomic intervals over which to operate.
      sbg:altPrefix: -L
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'null'
    - id: '#lenient'
      label: Lenient
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --lenient
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: Lenient processing of VCF files.
      sbg:altPrefix: -LE
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'false'
    - id: '#pedigree'
      label: Pedigree
      type:
      - 'null'
      - File
      inputBinding:
        prefix: --pedigree
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: Pedigree file for determining the population "founders".
      sbg:altPrefix: -ped
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'null'
    - id: '#read_filter'
      label: Read Filter
      type:
      - 'null'
      - type: array
        items:
          name: read_filter
          type: enum
          symbols:
          - AlignmentAgreesWithHeaderReadFilter
          - AllowAllReadsReadFilter
          - AmbiguousBaseReadFilter
          - CigarContainsNoNOperator
          - FirstOfPairReadFilter
          - FragmentLengthReadFilter
          - GoodCigarReadFilter
          - HasReadGroupReadFilter
          - LibraryReadFilter
          - MappedReadFilter
          - MappingQualityAvailableReadFilter
          - MappingQualityNotZeroReadFilter
          - MappingQualityReadFilter
          - MatchingBasesAndQualsReadFilter
          - MateDifferentStrandReadFilter
          - MateOnSameContigOrNoMappedMateReadFilter
          - MetricsReadFilter
          - NonZeroFragmentLengthReadFilter
          - NonZeroReferenceLengthAlignmentReadFilter
          - NotDuplicateReadFilter
          - NotOpticalDuplicateReadFilter
          - NotSecondaryAlignmentReadFilter
          - NotSupplementaryAlignmentReadFilter
          - OverclippedReadFilter
          - PairedReadFilter
          - PassesVendorQualityCheckReadFilter
          - PlatformReadFilter
          - PlatformUnitReadFilter
          - PrimaryLineReadFilter
          - ProperlyPairedReadFilter
          - ReadGroupBlackListReadFilter
          - ReadGroupReadFilter
          - ReadLengthEqualsCigarLengthReadFilter
          - ReadLengthReadFilter
          - ReadNameReadFilter
          - ReadStrandFilter
          - SampleReadFilter
          - SecondOfPairReadFilter
          - SeqIsStoredReadFilter
          - ValidAlignmentEndReadFilter
          - ValidAlignmentStartReadFilter
          - WellformedReadFilter
      inputBinding:
        prefix: --read-filter
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: Read filters to be applied before analysis.
      sbg:altPrefix: -RF
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'null'
    - id: '#read_index'
      label: Read Index
      type:
      - 'null'
      - type: array
        items: string
      inputBinding:
        prefix: --read-index
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: |-
        Indices to use for the read inputs. If specified, an index must be provided for every read input and in the same order as the read inputs. If this argument is not specified, the path to the index for each input will be inferred automatically.
      sbg:altPrefix: -read-index
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'null'
    - id: '#read_validation_stringency'
      label: Read Validation Stringency
      type:
      - 'null'
      - name: read_validation_stringency
        type: enum
        symbols:
        - STRICT
        - LENIENT
        - SILENT
      inputBinding:
        prefix: --read-validation-stringency
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: |-
        Validation stringency for all SAM/BAM/CRAM/SRA files read by this program. The default stringency value SILENT can improve performance when processing a BAM file in which variable-length data (read, qualities, tags) do not otherwise need to be decoded.
      sbg:altPrefix: -VS
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: SILENT
    - id: '#reference'
      label: Reference
      type:
      - 'null'
      - File
      inputBinding:
        secondaryFiles:
        - .fai
        - ^.dict
        prefix: --reference
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: Reference sequence.
      sbg:altPrefix: -R
      sbg:category: Optional Arguments
      sbg:fileTypes: FASTA, FA
      sbg:toolDefaultValue:
    - id: '#resource'
      label: Resource
      type:
      - 'null'
      - type: array
        items: string
      inputBinding:
        prefix: --resource
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: External resource VCF file.
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'null'
    - id: '#resource_allele_concordance'
      label: Resource Allele Concordance
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --resource-allele-concordance
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: Check for allele concordances when using an external resource VCF
        file.
      sbg:altPrefix: -rac
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'false'
    - id: '#sites_only_vcf_output'
      label: Sites Only Vcf Output
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --sites-only-vcf-output
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: If true, don't emit genotype fields when writing vcf file output.
      sbg:category: Optional Arguments
      sbg:toolDefaultValue: 'false'
    - id: '#comp'
      label: Comp
      type:
      - 'null'
      - type: array
        items: string
      inputBinding:
        prefix: --comp
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: Comparison VCF file(s).
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: 'null'
    - id: '#disable_tool_default_annotations'
      label: Disable Tool Default Annotations
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --disable-tool-default-annotations
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: Disable all tool default annotations.
      sbg:altPrefix: -disable-tool-default-annotations
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: 'false'
    - id: '#disable_tool_default_read_filters'
      label: Disable Tool Default Read Filters
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --disable-tool-default-read-filters
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: |-
        Disable all tool default read filters (WARNING: many tools will not function correctly without their default read filters on).
      sbg:altPrefix: -disable-tool-default-read-filters
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: 'false'
    - id: '#enable_all_annotations'
      label: Enable All Annotations
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --enable-all-annotations
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: Use all possible annotations (not for the faint of heart).
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: 'false'
    - id: '#showhidden'
      label: Showhidden
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --showHidden
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: display hidden arguments.
      sbg:altPrefix: -showHidden
      sbg:category: Advanced Arguments
      sbg:toolDefaultValue: 'false'
    - id: '#ambig_filter_bases'
      label: Ambig Filter Bases
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --ambig-filter-bases
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: |-
        Threshold number of ambiguous bases. If null, uses threshold fraction; otherwise, overrides threshold fraction. Cannot be used in conjuction with argument(s) maxAmbiguousBaseFraction. Valid only if "AmbiguousBaseReadFilter" is specified.
      sbg:category: Conditional Arguments
      sbg:toolDefaultValue: 'null'
    - id: '#ambig_filter_frac'
      label: Ambig Filter Frac
      type:
      - 'null'
      - float
      inputBinding:
        prefix: --ambig-filter-frac
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: |-
        Threshold fraction of ambiguous bases 05. Cannot be used in conjuction with argument(s) maxAmbiguousBases. Valid only if "AmbiguousBaseReadFilter" is specified.
      sbg:category: Conditional Arguments
      sbg:toolDefaultValue: '0'
    - id: '#max_fragment_length'
      label: Max Fragment Length
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --max-fragment-length
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: |-
        Maximum length of fragment (insert size). Valid only if "FragmentLengthReadFilter" is specified.
      sbg:category: Conditional Arguments
      sbg:toolDefaultValue: '1000000'
    - id: '#library'
      label: Library
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --library
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: |-
        Name of the library to keep This argument must be specified at least once. Valid only if "LibraryReadFilter" is specified.
      sbg:altPrefix: -library
      sbg:category: Conditional Arguments
    - id: '#maximum_mapping_quality'
      label: Maximum Mapping Quality
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --maximum-mapping-quality
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: |-
        Maximum mapping quality to keep (inclusive). Valid only if "MappingQualityReadFilter" is specified.
      sbg:category: Conditional Arguments
      sbg:toolDefaultValue: 'null'
    - id: '#minimum_mapping_quality'
      label: Minimum Mapping Quality
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --minimum-mapping-quality
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: |-
        Minimum mapping quality to keep (inclusive). Valid only if "MappingQualityReadFilter" is specified.
      sbg:category: Conditional Arguments
      sbg:toolDefaultValue: '10'
    - id: '#dont_require_soft_clips_both_ends'
      label: Dont Require Soft Clips Both Ends
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --dont-require-soft-clips-both-ends
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: |-
        Allow a read to be filtered out based on having only 1 soft-clipped block. By default, both ends must have a soft-clipped block, setting this flag requires only 1 soft-clipped block. Valid only if "OverclippedReadFilter" is specified.
      sbg:category: Conditional Arguments
      sbg:toolDefaultValue: 'false'
    - id: '#filter_too_short'
      label: Filter Too Short
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --filter-too-short
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: |-
        Minimum number of aligned bases. Valid only if "OverclippedReadFilter" is specified.
      sbg:category: Conditional Arguments
      sbg:toolDefaultValue: '30'
    - id: '#platform_filter_name'
      label: Platform Filter Name
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --platform-filter-name
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: |-
        Platform attribute (PL) to match This argument must be specified at least once. Valid only if "PlatformReadFilter" is specified.
      sbg:category: Conditional Arguments
    - id: '#black_listed_lanes'
      label: Black Listed Lanes
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --black-listed-lanes
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: |-
        Platform unit (PU) to filter out This argument must be specified at least once. Valid only if "PlatformUnitReadFilter" is specified.
      sbg:category: Conditional Arguments
    - id: '#read_group_black_list'
      label: Read Group Black List
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --read-group-black-list
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: |-
        name of the read group to filter out This argument must be specified at least once. Valid only if "ReadGroupBlackListReadFilter" is specified.
      sbg:category: Conditional Arguments
    - id: '#keep_read_group'
      label: Keep Read Group
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --keep-read-group
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: |-
        The name of the read group to keep. Valid only if "ReadGroupReadFilter" is specified.
      sbg:category: Conditional Arguments
    - id: '#max_read_length'
      label: Max Read Length
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --max-read-length
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: |-
        Keep only reads with length at most equal to the specified value. Valid only if "ReadLengthReadFilter" is specified.
      sbg:category: Conditional Arguments
    - id: '#min_read_length'
      label: Min Read Length
      type:
      - 'null'
      - int
      inputBinding:
        prefix: --min-read-length
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: |-
        Keep only reads with length at least equal to the specified value. Valid only if "ReadLengthReadFilter" is specified.
      sbg:category: Conditional Arguments
      sbg:toolDefaultValue: '1'
    - id: '#read_name'
      label: Read Name
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --read-name
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: |-
        Keep only reads with this read name. Valid only if "ReadNameReadFilter" is specified.
      sbg:category: Conditional Arguments
    - id: '#keep_reverse_strand_only'
      label: Keep Reverse Strand Only
      type:
      - 'null'
      - boolean
      inputBinding:
        prefix: --keep-reverse-strand-only
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: |-
        Keep only reads on the reverse strand. Valid only if "ReadStrandFilter" is specified.
      sbg:category: Conditional Arguments
    - id: '#sample'
      label: Sample
      type:
      - 'null'
      - string
      inputBinding:
        prefix: --sample
        position: 0
        separate: true
        shellQuote: false
        sbg:cmdInclude: true
      description: |-
        The name of the sample(s) to keep, filtering out all others This argument must be specified at least once. Valid only if "SampleReadFilter" is specified.
      sbg:altPrefix: -sample
      sbg:category: Conditional Arguments
    - id: '#memory_per_job'
      label: Memory Per Job
      type:
      - 'null'
      - int
      description: Memory Per Job.
      sbg:category: Execution
    - id: '#memory_overhead_per_job'
      label: Memory Overhead Per Job
      type:
      - 'null'
      - int
      description: Memory Overhead Per Job.
      sbg:category: Execution

    outputs:
    - id: '#annotated_variants'
      label: Annotated Variants
      type:
      - 'null'
      - File
      outputBinding:
        secondaryFiles:
        - .idx
        glob: '*.vcf'
        sbg:inheritMetadataFrom: '#variant'
      description: Annotated variant VCF file
      sbg:fileTypes: VCF

    baseCommand:
    - /gatk/gatk
    - --java-options
    - class: Expression
      engine: '#cwl-js-engine'
      script: |-
        {
          if($job.inputs.memory_per_job){
          	return '\"-Xmx'.concat($job.inputs.memory_per_job, 'M') + '\"'
          }
          	return '\"-Xmx2048M\"'
        }
    - VariantAnnotator
    arguments:
    - prefix: --output
      position: 0
      valueFrom:
        class: Expression
        engine: '#cwl-js-engine'
        script: |-
          { 
            output_name = [].concat($job.inputs.variant)[0].path.split('/').slice(-1)[0]
            output_name = output_name.split('.')
            output_name = output_name.slice(0, -1).join('.') + '.dbsnp_anno.vcf'
            
            return output_name
          }
      separate: true
    - position: 0
      valueFrom: --create-output-variant-index
      separate: true

    hints:
    - class: sbg:CPURequirement
      value: 1
    - class: sbg:MemRequirement
      value:
        class: Expression
        engine: '#cwl-js-engine'
        script: |-
          {
            if($job.inputs.memory_per_job){
              if($job.inputs.memory_overhead_per_job){
              	return $job.inputs.memory_per_job + $job.inputs.memory_overhead_per_job
              }
              else
            		return $job.inputs.memory_per_job
            }
            else if(!$job.inputs.memory_per_job && $job.inputs.memory_overhead_per_job){
          		return 2048 + $job.inputs.memory_overhead_per_job  
            }
            else
            	return 2048
          }
    - class: DockerRequirement
      dockerPull: pgc-images.sbgenomics.com/d3b-bixu/gatk:4.1.4.0
    id: michael_lloyd/mantis/gatk-variantannotator/6
    description: |-
      ###**Overview**  

      Tool for adding annotations to VCF files.

      ###**Inputs**  

      A variant set to annotate and optionally one or more BAM files.

      ###**Output**  

      An annotated VCF.

      ###**Usage example**  

          java -jar gatk.jar \  
               VariantAnnotator \ 
               --variant variant.vcf \
               --input input.bam \
               --reference reference.fasta \
               --dbsnp dbsnp.vcf
    sbg:appVersion:
    - sbg:draft-2
    sbg:categories:
    - GATK-4
    sbg:cmdPreview: |-
      /opt/gatk --java-options "-Xmx2M" VariantAnnotator --variant /path/to/variant.ext --output variant.annotated.vcf  --create-output-variant-index
    sbg:content_hash: a0eefd96bfe8274f874c15b0772a9d3670c81d37cc0590a57aeae515faeb3a4f1
    sbg:contributors:
    - michael_lloyd
    sbg:createdBy: michael_lloyd
    sbg:createdOn: 1572878282
    sbg:id: michael_lloyd/mantis/gatk-variantannotator/6
    sbg:image_url:
    sbg:job:
      allocatedResources:
        cpu: 1
        mem: 3
      inputs:
        annotation:
        - TODO
        dbsnp:
          class: File
          secondaryFiles:
          - path: .idx
          path: /path/to/dbsnp.ext
          size: 0
        input:
        - class: File
          secondaryFiles: []
          path: /path/to/input-1.ext
          size: 0
        - class: File
          secondaryFiles: []
          path: /path/to/input-2.ext
          size: 0
        memory_overhead_per_job: 1
        memory_per_job: 2
        reference:
          class: File
          secondaryFiles:
          - path: .fai
          - path: ^.dict
          path: /path/to/reference.ext
          size: 0
        variant:
          class: File
          secondaryFiles: []
          path: /path/to/variant.ext
          size: 0
    sbg:latestRevision: 6
    sbg:license: Open source BSD (3-clause) license
    sbg:links:
    - id: https://software.broadinstitute.org/gatk/
      label: Homepage
    - id: https://software.broadinstitute.org/gatk/documentation/tooldocs/current/
      label: Documentation
    - id: https://software.broadinstitute.org/gatk/download/
      label: Download
    sbg:modifiedBy: michael_lloyd
    sbg:modifiedOn: 1572884073
    sbg:project: michael_lloyd/mantis
    sbg:projectName: MSI, HRD, TMB
    sbg:publisher: sbg
    sbg:revision: 6
    sbg:revisionNotes: changed naming
    sbg:revisionsInfo:
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1572878282
      sbg:revision: 0
      sbg:revisionNotes: Copy of admin/sbg-public-data/gatk-variantannotator/2
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1572882320
      sbg:revision: 1
      sbg:revisionNotes: changed secondary dbsnp file requirement
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1572882363
      sbg:revision: 2
      sbg:revisionNotes: updated to GATK 4.1.4.0
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1572882437
      sbg:revision: 3
      sbg:revisionNotes: changed expected input
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1572882485
      sbg:revision: 4
      sbg:revisionNotes: updated naming convention
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1572883175
      sbg:revision: 5
      sbg:revisionNotes: updated tool call
    - sbg:modifiedBy: michael_lloyd
      sbg:modifiedOn: 1572884073
      sbg:revision: 6
      sbg:revisionNotes: changed naming
    sbg:sbgMaintained: false
    sbg:toolAuthor: Broad Institute
    sbg:toolkit: GATK
    sbg:toolkitVersion: 4.1.4.0
    sbg:validationErrors: []
  inputs:
  - id: '#gatk_variantannotator.variant'
    source: '#snpsift_split_4_3k.joined_vcf'
  - id: '#gatk_variantannotator.dbsnp'
    source: '#dbsnp'
  outputs:
  - id: '#gatk_variantannotator.annotated_variants'
  sbg:x: 1393.0169677734375
  sbg:y: 232.15281677246094
id: |-
  https://cavatica-api.sbgenomics.com/v2/apps/d3b-bixu/open-targets-pdx-workflow-dev/pdx-wes-tumor-only-xenome-with-variant-calling-msi-and-tmb-score/3/raw/
description: |-
  This Whole Exome Sequencing (WES) tumor-normal workflow first uses the [Broad Institute's](https://software.broadinstitute.org/gatk/best-practices/) best-practices workflow for read alignment, and then analyzes those data in several ways. 

  1.  Identifies variants from a human exome experiment with GATK-4 [Mutect2](https://software.broadinstitute.org/gatk/documentation/tooldocs/4.0.5.1/org_broadinstitute_hellbender_tools_walkers_mutect_Mutect2.php) for variant calling. 
  2. Calculates microsatellite instability (MSI) status using [MSIsensor2](https://github.com/niu-lab/msisensor2)
  3. Calculates tumor mutation burden (TMB) score using filtered variants.

  Note: This workflow utilizes the tool `Xenome` to removed mouse-reads from the raw-read data. `Xenome` uses host and graft reference sequences to characterize the set of all possible k-mers according to whether they belong to: only the graft (and not the host), only the host (and not the graft), both references, neither reference, and marginal asignments. This workflow uses those reads classified as 'human-only'.

  ---

  ### Essential Requirements

    The following metadata fields are required and should be assigned to input read files:

  1. **Sample ID**: Any string. As this is the biospecimen identifier, it should be different for PDX (tumor) and normal sample. The workflow will run even if the strings are the same, however, please consider using different identifiers.
  2. **Paired-end**: 1 or 2
  3. **Sample type**: Any string. Please make sure to also provide the chosen tumor and normal sample types to the *SBG Split Pair by Metadata* tool (see below for details), if this is the field you wish to use to separate tumor and normal files. 

  #### Optional (but recommended) Metadata
  1. **Case ID**. Any string. This allows for the tracking of case's. It is not required for running; however, the workflow will append this metadata field to the front of most outputs if present. 

  The workflow will process both uncompressed and compressed FASTQ files.

  ---

  #### The following output files will be generated:

         ---VARIANT OUTPUTS---
         Tumor FASTQC reports
         Tumor BAM file
         Tumor QC reports - integrated and exome
         Final annotated VCF and TAB files

         ---MSI---
         MSI Score File

         ---TMB---
        TMB Score File

  ---

  ### Reference Files and Workflow Details

  Required reference input files:

  1. **Xenome** is used to classify reads as human or mouse. Xenome indices are built on hg38 and pseudoNOD genome (based on SNP incorporation into mm10 genome from Sanger [ftp://ftp-mouse.sanger.ac.uk/REL-1505-SNPs_Indels/]). The default value of k=25 is used during the indices preparation. 
  Default file input: Xenome_WGS_indices.tar.gz

  2. Reference FASTA file and secondary files (.FAI, .DICT). 
  Default file input: Homo_sapiens_assembly38.fasta (Homo_sapiens_assembly38.fasta.fai, Homo_sapiens_assembly38.dict)
  Chromosome naming in the default input file: chr1, chr2... chrX, chrY, chrM.

  3. BWA indices were prepared using bwakit/0.7.15. GRCh38 files from the Broad GATK resource bundle (hg38_201601) were used. 
  Default file input: BWA_ref_files.tar.gz

  4. Additional reference input files (hg38-specific) from the Broad Institute GATK resource bundle (known indels):               
       Homo_sapiens_assembly38.known_indels.vcf.gz (Homo_sapiens_assembly38.known_indels.vcf.gz.tbi)     
       Mills_and_1000G_gold_standard.indels.hg38.vcf.gz (Mills_and_1000G_gold_standard.indels.hg38.vcf.gz.tbi)
       hsa\_dbSNP\_v151\_20170710chr\_renamed.vcf.gz (hsa\_dbSNP\_v151\_20170710chr\_renamed.vcf.gz.tbi)     

  5. dbSNP v151 (20170710) file:
  Default file input: hsa_dbSNP_v151_20170710chr_renamed.vcf.gz (hsa_dbSNP_v151_20170710chr_renamed.vcf.gz.tbi)

  6. ExAC sites (lifted over to GRCh38 and formatted to match the reference FASTA). Original file was downloaded from Ensembl (http://ftp.ensembl.org/pub/data_files/homo_sapiens/GRCh38/variation_genotype/ExAC.0.3.GRCh38.vcf.gz).
  Default file input: ExAC.0.3.GRCh38_chr_added_bad_lift_over_removed_reorder_primary_only.vcf.gz (ExAC.0.3.GRCh38_chr_added_bad_lift_over_removed_reorder_primary_only.vcf.gz.tbi)

  7. Annotation database file for SnpEff (v4.3):
       snpEff_v4_3_hg38.zip

  8. Annotation database files for SnpSift:
       a) dbNSFP (by default, the workflow uses dbNSFP v3.2 (academic release): dbNSFP3.2a.txt.gz)
       b) COSMIC (default file: Sorted_Cosmicv80_Coding_Noncoding.processed.vcf.gz, with .tbi index)

  9. Target region files (BED and INTERVALS_LIST format)
  The BED file should correspond to the data being processed. Chromosome naming in the BED file should match the reference FASTA used to process the data (if using the default input FASTA file, please ensure that chromosome names begin with 'chr'). If a corresponding INTERVALS_LIST file (as used by Picard toolkit) is not available, it can easily be generated using the **GATK BedToIntervalList** tool.

  ---

  ### Workflow Steps

  #### Step 1: **SBG Split Pair by Metadata**

  The first step separates tumor and normal FASTQ files for downstream processing, based on specified metadata criteria which should be supplied as inputs (normal_metadata and tumor_metadata) in the format metadata_key:value. For example, for files to be classified based on Sample type metadata (sample_type field), with the values "tumor" and "normal", the inputs should read: sample_type:normal and sample_type:tumor (for normal_metadata and tumor_metadata parameters, respectively).

  #### Step 2:  **Tumor Alignment and Target Coverage** 

  In the next step, tumor (PDX model) and normal FASTQ files are QC-checked (**FASTQC**), trimmed (**Trimmomatic**), aligned (**BWA**, alt-aware), sorted (**Picard SortSam**) and prepared for variant calling (**Picard MarkDuplicates**, **GATK BaseRecalibrator**, **GATK ApplyBQSR**). QC is also performed on the processed BAM files using **Exome Coverage QC 1.0** tool and **Picard CalculateHsMetrics**. QC reports are aggregated with the **QC Integrate** script. The two workflows are similar, except for **Xenome** preprocessing of PDX-derived data (tumor workflow) to remove mouse reads. 

  Important notes: 

  1. **BWA** is set to hardcode "tumor" and "normal" as sample names (SM) in the output BAMs. This can be changed via the read group input parameter. However, please make sure to also adjust **Mutect** parameters tumor and normal sample name, if this is changed.

  2. By default, QC checks require that at least 75 % of every target region be covered at 20X. If this requirement is not met, the task will fail (See Target Coverage parameter of the **QC Integrate** tool to adjust this).

  #### Step 3: Indexing BAM files (**Samtools Index BAM**)  


  #### Step 4: Variant calling (**GATK 4 Mutect2** and **FilterMutectCalls**)

  The variant calling step has been paralellized by invoking **Mutect2** and **GATK FilterMutectCalls** on smaller intervals in paralel (scatter). The intervals are prepared with the **SBG Prepare Intervals** tool and the output VCFs are collected and merged with the **SnpSift split** tool before annotation. 

  Important default parameter: As ExAC 0.3 is used, AF Of Alleles Not In Resource was set to (ExAC 0.3 ~ 60,706Exomes) = 1/(2* 60706)= 0.0000082364  

  #### Step 5: Annotation 

  The somatic VCF is annotated with SnpEff/SnpSift tools (v4.3). Basic annotation is done with **SnpEff** (using hg38 database). If you wish to use a different database, please make sure to also alter the Assembly input parameter of the SnpEff tool. 
  **SnpSift dbNSFP** and **SnpSift annotate** tools are used to add additional annotations from the [dbNSFP](https://sites.google.com/site/jpopgen/dbNSFP) and [COSMIC](https://cancer.sanger.ac.uk/cosmic). Please note that these two annotation sources should be provided as tabix-indexed VCF.GZ files (with their .TBI index present)

  For convenience, the output VCF is reformatted to hold one effect per line (script vcfEffOnePerLine.pl from SnpEff toolkit) and also converted to a tab-separated file (**SnpSift extractFields tool**).

  **Please note:** the GT genotype field present in the final VCF does not represent genotype in the traditional sense. Per [GATK documentation](https://software.broadinstitute.org/gatk/documentation/article?id=11127): A somatic caller should detect low fraction alleles, can make no explicit ploidy assumption and omits genotyping in the traditional sense. Mutect2 adheres to all of these criteria. A number of cancer sample characteristics necessitate such caller features. For one, biopsied tumor samples are commonly contaminated with normal cells, and the normal fraction can be much higher than the tumor fraction of a sample. Second, a tumor can be heterogeneous in its mutations. Third, these mutations not uncommonly include aneuploid events that change the copy number of a cell's genome in patchwork fashion.  

  #### Step 6: Microsattelite instability (MSI) Status Calculation

  The tumor BAM file are passed to MSIsensor2 which calls MSI status. Note that the threshold as determined by the authors of the tool is- MSI high: msi score >= 20%.

  #### Step 7: Tumor Mutation Burden (TMB) Calculation
  TMB is calucated as the number of coding mutations per Mb of the genome. This is assessed using variants that meet all quality criteria (coverage, AF, mapping quality, strand bias etc.), are somatic and non-polymorphic, and are defined in SnpEff as 'high' or 'moderate' functional impact. As only a porition of the genome was sequenced, genome coverage (Mb) is calculated from the input target coverage BED file.
sbg:appVersion:
- sbg:draft-2
sbg:canvas_x: 249
sbg:canvas_y: 138
sbg:canvas_zoom: 0.5999999999999996
sbg:content_hash: ac1013d143b40e479672fee5abfdc72670093f8246bbe316838cca468ef43195f
sbg:contributors:
- brownm28
sbg:createdBy: brownm28
sbg:createdOn: 1638471467
sbg:id: |-
  d3b-bixu/open-targets-pdx-workflow-dev/pdx-wes-tumor-only-xenome-with-variant-calling-msi-and-tmb-score/3
sbg:image_url:
sbg:latestRevision: 3
sbg:modifiedBy: brownm28
sbg:modifiedOn: 1638804993
sbg:original_source: |-
  https://cgc-api.sbgenomics.com/v2/apps/pdxnet/pdx-wf-commit2/pdx-wes-tumor-only-xenome-with-variant-calling-msi-and-tmb-score/6/raw/
sbg:project: d3b-bixu/open-targets-pdx-workflow-dev
sbg:projectName: Open Targets PDX Workflow Dev
sbg:publisher: sbg
sbg:revision: 3
sbg:revisionNotes: reverted xenome arg - changed default instance from c4 to c5
sbg:revisionsInfo:
- sbg:modifiedBy: brownm28
  sbg:modifiedOn: 1638471467
  sbg:revision: 0
  sbg:revisionNotes: |-
    Uploaded using sbpack v2020.10.05. 
    Source: pdx-wes-tumor-only-xenome-with-variant-calling-msi-and-tmb-score.cwl
- sbg:modifiedBy: brownm28
  sbg:modifiedOn: 1638557767
  sbg:revision: 1
  sbg:revisionNotes: changed docker pull locations
- sbg:modifiedBy: brownm28
  sbg:modifiedOn: 1638741978
  sbg:revision: 2
  sbg:revisionNotes: swapped arg "--pairs" for "--preserve-read-order"
- sbg:modifiedBy: brownm28
  sbg:modifiedOn: 1638804993
  sbg:revision: 3
  sbg:revisionNotes: reverted xenome arg - changed default instance from c4 to c5
sbg:sbgMaintained: false
sbg:toolAuthor: JAX
sbg:validationErrors: []
