# Open Targets PDX Processing Pipelines

This repo contains code and information related to the processing of DNA and RNA data from patient-derived xenografts (PDX).

## DNA Workflows
Simply ported over from the Seven Bridges Cancer Genomics Cloud (SB CGC), [PDX WES Tumor-Only (Xenome) with Variant Calling, MSI, and TMB scores](https://cgc.sbgenomics.com/public/apps/pdxnet/pdx-wf-commit2/pdx-wes-tumor-only-xenome-with-variant-calling-msi-and-tmb-score).
### Overview
This workflow takes reads generated from a PDX model and performs variant calling as well as calculates microsatellite instability (MSI) and tumor mutation burden (TMB).
It was developed by Jackson Labs and implemented using common workflow language (CWL).
Open Targets group has ported over this workflow code into the Seven Bridges Cavatica environment, the source if which can be found in `pdx-wes-tumor-only.cwl`.
A brief outline of what the workflow does, using fastq reads generated from the PDX:
1. Run QC check on reads using FastQC
1. Filter out mouse contaminants using xenome tool
1. Align to reference (default hg38)
1. Run alignment QC using picard metrics
1. Call variants with Mutect2
1. Calculate microsatellite instability (MSI)
1. Calculate tumor mutation burden (TMB)
1. Summarize results

Dtailed documentation, including specific reference inputs, generated by the authors can be found [here](docs/jax_pdx_wes_README.md)

## RNA Workflows
Currently two workflows are present as we are comparing their relative efficacy.
### JAX PDX RNAseq Worfklow Overview
As above, the first was developed by Jackson labs and ported over from SB CGC [PDX RNA Expression Estimation Workflow](https://cgc.sbgenomics.com/public/apps/pdxnet/pdx-wf-commit2/pdx-rna-expression-estimation-workflow).
Open Targets group has ported over this workflow code into the Seven Bridges Cavatica environment, the source if which can be found in `pdx-rna-tumor-only.cwl`.
A brief outline of what the workflow does, using fastq reads generated from the PDX:
1. Run QC check on reads using FastQC
1. Filter out mouse contaminants using xenome tool
1. Use RSEM to align reads (via STAR one-pass) to reference (default GRCh38, gene model ENSEMBL91) and quantify expression
1. Run Somalier to generate inputs for relatedness calculations

Detailed documentation, including specific reference inputs, generated by the authors can be found [here](docs/jax_pdx_rna_README.md)

Also ported over from https://cgc.sbgenomics.com/public/apps/ and additionally, [Kids First DRC RNAseq](https://github.com/kids-first/kf-rnaseq-workflow) used as a submodule

### Modified Kids First Workflow Overview
The second workflow being compared is from the Kids First Data Resource Center (KFDRC)
This workflow has been modified to add xenome filtering up front, but the rest remains the same.
It has the added benefit of also performing fusion calling.
Open Targets group has ported over this workflow code into the Seven Bridges Cavatica environment, the source if which can be found in `workflows/kfdrc-pdx-RNAseq-workflow.cwl`.

A brief outline of what the workflow does, using fastq reads generated from the PDX:
1. Quantify expression with kallisto 
1. Align using STAR two pass against reference (default GENCODE 27, a GRCh38 reference)
1. Call fusions using STAR-Fusion and arriba
1. Collate and filter fusions using annofuse
1. Additionally quantify expression using RSEM

Detailed documentation, including specific reference inputs, generated by the authors can be found [here](docs/kf_pdx_rnaseq_README.md)

## [Cavatica Development Project](https://cavatica.sbgenomics.com/u/d3b-bixu/open-targets-pdx-workflow-dev) 
PDX data from the paper "Genomic Profiling of Childhood Tumor Patient-Derived Xenograft Models to Enable Rational Clinical Trial Design", DOI: https://doi.org/10.1016/j.celrep.2019.09.071 are here as well as workflows used
